{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <span style='font-family:\"Times New Roman\"'> <span styel=''> **MASTER FILE CREATION**\n",
    "\n",
    "*Emile Cohen* \n",
    "    \n",
    "*March 2020*\n",
    "\n",
    "**Goal:** In this Notebook, we create a master file that summarizes all useful information.\n",
    "\n",
    "The Notebook is divided in 4 parts, representing the four parts of our Master file:\n",
    "   \n",
    "* **1. Patient/Sample Information**\n",
    "* **2. TP53 Mutations**\n",
    "* **3. TP53 Copy Numbers**\n",
    "* **4. TP53 Computed Metrics**\n",
    "* **5. Subgroup columns creation**\n",
    "* **6. Merge tables**\n",
    "\n",
    "**NB1:** In each part, you must run the cells from the begining in order to initialize the variables\n",
    "\n",
    "**NB2:** In order to launch the last script (Merge Tables), you have to define the functions in each part.\n",
    "\n",
    "**NB3:** All functions used for the plots are located in utils/custom_tools.py\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Setup environment... done!\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "<span style=\"color:green\">✅ Working on **mskimpact_env** conda environment.</span>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%run -i '../../../utils/setup_environment.ipy'\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "from scipy.stats import fisher_exact, ranksums, chi2, norm\n",
    "from statsmodels.sandbox.stats.multicomp import multipletests\n",
    "import matplotlib.gridspec as gridspec\n",
    "import pickle\n",
    "\n",
    "data_path = '../../../data/'\n",
    "data_wgd = data_path + 'impact-facets-tp53/processed/wgd/'\n",
    "data_no_wgd = data_path + 'impact-facets-tp53/processed/no_wgd/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [],
   "source": [
    "# first we load all files for WGD\n",
    "maf_cohort_wgd = pd.read_csv(data_wgd + 'maf_cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_wgd = pd.read_csv(data_wgd + 'cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_wgd = pd.read_csv(data_wgd + 'arm_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_wgd = pd.read_csv(data_wgd + 'gene_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We load all files for non WGD\n",
    "\n",
    "maf_cohort_nowgd = pd.read_csv(data_no_wgd + 'maf_cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_nowgd = pd.read_csv(data_no_wgd + 'cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_nowgd = pd.read_csv(data_no_wgd + 'arm_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_nowgd = pd.read_csv(data_no_wgd + 'gene_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 203,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Creating keys for mutations in maf files\n",
    "\n",
    "# First we need to create a sample_mut_key to identify duplicated mutations\n",
    "maf_cohort_wgd['mut_key'] = maf_cohort_wgd.apply(lambda h: str(h.Chromosome)+'_'+str(h.Start_Position)+'_'+str(h.Reference_Allele)+'_'+str(h.Tumor_Seq_Allele2), axis=1) \n",
    "maf_cohort_nowgd['mut_key'] = maf_cohort_nowgd.apply(lambda h: str(h.Chromosome)+'_'+str(h.Start_Position)+'_'+str(h.Reference_Allele)+'_'+str(h.Tumor_Seq_Allele2), axis=1) \n",
    "\n",
    "# Create a sample key to differentiate duplicates\n",
    "maf_cohort_wgd['sample_mut_key'] = maf_cohort_wgd.apply(lambda h: h.Tumor_Sample_Barcode + h.mut_key, axis = 1)\n",
    "maf_cohort_nowgd['sample_mut_key'] = maf_cohort_nowgd.apply(lambda h: h.Tumor_Sample_Barcode + h.mut_key, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load clinical data\n",
    "clinical_data = pd.read_csv(data_path + 'cbioportal/raw/mskimpact_clinical_data-2.tsv', sep= '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filtering the clinical data\n",
    "samples_wgd = list(set(cohort_wgd.tumor_sample))\n",
    "samples_nowgd = list(set(cohort_nowgd.tumor_sample))\n",
    "\n",
    "clinical_wgd = clinical_data[clinical_data['Sample ID'].isin(samples_wgd)]\n",
    "clinical_nowgd = clinical_data[clinical_data['Sample ID'].isin(samples_nowgd)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IMPORTANT: Parameter Definition\n",
    "In this script we can create two master files: one for WGD samples and one for non WGD samples.\n",
    "So this parameter allows to select the type of master you want.\n",
    "On top of that you can specify where you want to store the file you created"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 264,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_type = 'wgd'\n",
    "\n",
    "\n",
    "\n",
    "output_path = data_path + 'impact-facets-tp53/processed/'\n",
    "\n",
    "total_output = output_path + cohort_type + '/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "# Patient/Sample Information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we focus on clinical information exported from CbioPortal.\n",
    "\n",
    "The following columns are selected:\n",
    "* Sample_Id\n",
    "* Tumor_Id\n",
    "* Patient_Id\n",
    "* Patient Current Age\n",
    "* Cancer_Type\n",
    "* Cancer_Type_Detailed\n",
    "* Sample_Type\n",
    "* purity\n",
    "* ploidy\n",
    "* Overall Survival Status\n",
    "* Overall Survival (Months)\n",
    "* MSI Score\n",
    "* MSI Type\n",
    "* Tumor Mutational Burden"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_sample_info(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    cohort_filt = cohort[['sample_id', 'tumor_sample', 'patient', 'ploidy']]\n",
    "    clinical_filt = clinical[['Sample ID', 'Patient Current Age', 'Cancer Type', 'Cancer Type Detailed', 'Sample Type',\n",
    "                  'Overall Survival (Months)', 'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "    purity = maf_cohort.drop_duplicates('Tumor_Sample_Barcode')[['Tumor_Sample_Barcode', 'purity']]\n",
    "\n",
    "    # Merging these files\n",
    "    sample_info = pd.merge(cohort_filt, purity, left_on='tumor_sample', right_on='Tumor_Sample_Barcode')\n",
    "    sample_info =  pd.merge(sample_info, clinical_filt, left_on='tumor_sample', right_on='Sample ID').drop(['Tumor_Sample_Barcode', 'Sample ID'], axis=1)\n",
    "\n",
    "\n",
    "    sample_info = sample_info[['sample_id', 'tumor_sample', 'patient', 'Cancer Type', 'Cancer Type Detailed',\n",
    "                               'Patient Current Age','Sample Type', 'purity','ploidy','Overall Survival (Months)', \n",
    "                               'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "\n",
    "    sample_info.columns = ['Sample_Id', 'Tumor_Id', 'Patient_Id','Cancer_Type', 'Cancer_Type_Detailed', 'Patient_Current_Age',\n",
    "                          'Sample_Type', 'purity', 'ploidy', 'Overall_Survival_Months', 'Overall_Survival_Status',\n",
    "                          'MSI_Score', 'MSI_Type', 'TMB_Score']\n",
    "    \n",
    "    return sample_info\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [],
   "source": [
    "sample_info = create_sample_info(cohort=cohort_type)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Mutations\n",
    "In this part, we focus on tp53 mutation information.\n",
    "\n",
    "We gather all mutations per sample, and split it into different columns. We have the following columns:\n",
    "* Tumor_Id\t\n",
    "* key_1 (2,3,4,5) --> Mutation key allowing to filter duplicates\n",
    "* vc_1 (2,3,4,5) --> Variant Classification\n",
    "* ccf_1 (2,3,4,5) --> Cancer Cell Fraction of the mutation\n",
    "* vaf_1 (2,3,4,5) --> Variant Allele Frequency of the mutation\n",
    "* HGVSp_1 (2,3,4,5) --> protein change\n",
    "* spot_1 (2,3,4,5) --> Integer that defines the spot of the tp53 mutation\n",
    "* tp53_count --> Number of tp53 mutations of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def f_(x):\n",
    "    # This function helps us to group mutations together in a single cell per patient\n",
    "    return pd.DataFrame(dict(Tumor_Sample_Barcode = x['Tumor_Sample_Barcode'],  \n",
    "                        muts = \"%s\" % ','.join(x['sample_mut_key_vc_ccf_vaf_hgv_spot'])))\n",
    "\n",
    "def count_tp53_muts(x):\n",
    "    count = 0\n",
    "    for i in range(1,6):\n",
    "        if x['tp53_key_' + str(i)]:\n",
    "            count+= 1\n",
    "    return count\n",
    "\n",
    "# WARNING: THis function needs sample_info to work\n",
    "def create_tp53_muts(cohort:str, sample_info):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        print('yes')\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    '''\n",
    "    This function aims to gather all tp53 mutation characteristics.\n",
    "    For each sample we gather the tp53 mutations and their characteristics for all patients.\n",
    "    '''\n",
    "    # We load the  table created in maf_tp53_creation.ipynb\n",
    "    maf_tp53 = maf_cohort[maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    maf_tp53['mut_spot'] = maf_tp53.HGVSp.str.extract('(\\d+)')\n",
    "\n",
    "    # We select only intresting columns\n",
    "    maf_tp53_filtered = maf_tp53[['Tumor_Sample_Barcode','sample_mut_key', 'Variant_Classification',\\\n",
    "                                        'ccf_expected_copies', 't_var_freq', 'HGVSp','mut_spot' ]]\n",
    "\n",
    "    # Let's Merge mut_key,Variant_classification, CF, CCF, and VAF to gather them\n",
    "    maf_tp53_filtered['sample_mut_key_vc_ccf_vaf_hgv_spot'] = maf_tp53_filtered.apply(lambda x: str(x.sample_mut_key)+'%'+str(x.Variant_Classification)+'%'+str(x.ccf_expected_copies)+'%'+str(x.t_var_freq)+'%'+str(x.HGVSp)+'%'+str(x.mut_spot), axis=1)\n",
    "\n",
    "    # We Select important columns\n",
    "    final = maf_tp53_filtered[['Tumor_Sample_Barcode', 'sample_mut_key_vc_ccf_vaf_hgv_spot']]\n",
    "    # We groupby Patient_Id and apply the function above to group mutations\n",
    "    final = final.groupby(['Tumor_Sample_Barcode'], sort=False).apply(f_)\n",
    "\n",
    "    # We separate the different mutations into 5 different columns (5 is the max number of tp53 mutations in our cohort)\n",
    "    final[['mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5']] = final.muts.str.split(',', expand=True)\n",
    "    #final = final.drop(['mut_key_6'],axis=1)\n",
    "    # Split the columns into mut_key_ and vc_\n",
    "    final[['tp53_key_1','tp53_vc_1','tp53_ccf_1','tp53_vaf_1','tp53_HGVSp_1', 'tp53_spot_1']] = final.mut_key_1.str.split('%', expand=True)\n",
    "    final[['tp53_key_2','tp53_vc_2','tp53_ccf_2','tp53_vaf_2','tp53_HGVSp_2', 'tp53_spot_2']] = final.mut_key_2.str.split('%', expand=True)\n",
    "    final[['tp53_key_3','tp53_vc_3','tp53_ccf_3','tp53_vaf_3','tp53_HGVSp_3', 'tp53_spot_3']] = final.mut_key_3.str.split('%', expand=True)\n",
    "    final[['tp53_key_4','tp53_vc_4','tp53_ccf_4','tp53_vaf_4','tp53_HGVSp_4', 'tp53_spot_4']] = final.mut_key_4.str.split('%', expand=True)\n",
    "    final[['tp53_key_5','tp53_vc_5','tp53_ccf_5','tp53_vaf_5','tp53_HGVSp_5', 'tp53_spot_5']] = final.mut_key_5.str.split('%', expand=True)\n",
    "\n",
    "    # We remove the muts column\n",
    "    final = final.drop(['muts','mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5'], axis=1)\n",
    "\n",
    "    # We remove duplicates\n",
    "    final = final.drop_duplicates('Tumor_Sample_Barcode')\n",
    "\n",
    "    # We add the cohort patients that are not tp53 positive\n",
    "    #First we create a dataframe with all missing samples\n",
    "\n",
    "    cohort_samples = set(sample_info.Tumor_Id)\n",
    "    final_samples = set(final.Tumor_Sample_Barcode)\n",
    "    missing_samp = pd.DataFrame(cohort_samples - final_samples, columns = ['Tumor_Sample_Barcode'])\n",
    "    #Then we append the two datframe\n",
    "    final = final.append(missing_samp)\n",
    "\n",
    "    # We rename the Tumor_Sample_Barcode column to have the same key as in other datframes\n",
    "    final = final.rename(columns={'Tumor_Sample_Barcode': 'Tumor_Id'})\n",
    "\n",
    "    # We add a last column tp53_count that represents the number of tp53 mutations per sample\n",
    "    final = final.where(final.notnull(), None)\n",
    "    final['tp53_count'] = final.apply(count_tp53_muts, axis = 1)\n",
    "\n",
    "    # We change the type of vafs column to float64 instead of strings\n",
    "    final = final.astype({'tp53_vaf_1': 'float64', 'tp53_vaf_2': 'float64', 'tp53_vaf_3': 'float64', 'tp53_vaf_4': 'float64', 'tp53_vaf_5': 'float64',\n",
    "                       'tp53_ccf_1': 'float64', 'tp53_ccf_2': 'float64', 'tp53_ccf_3': 'float64', 'tp53_ccf_4': 'float64', 'tp53_ccf_5': 'float64'})\n",
    "\n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yes\n",
      "18706\n",
      "6885\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P-0027609-T01-IM6</td>\n",
       "      <td>P-0027609-T01-IM617_7577548_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.118574</td>\n",
       "      <td>p.Gly245Ser</td>\n",
       "      <td>245</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>P-0027408-T01-IM6</td>\n",
       "      <td>P-0027408-T01-IM617_7578409_CT_TC</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.168901</td>\n",
       "      <td>p.Arg174Glu</td>\n",
       "      <td>174</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>P-0025997-T01-IM6</td>\n",
       "      <td>P-0025997-T01-IM617_7578471_G_-</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.912621</td>\n",
       "      <td>p.Gly154AlafsTer16</td>\n",
       "      <td>154</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>P-0036570-T01-IM6</td>\n",
       "      <td>P-0036570-T01-IM617_7578236_A_G</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.718605</td>\n",
       "      <td>p.Tyr205His</td>\n",
       "      <td>205</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>P-0036570-T02-IM6</td>\n",
       "      <td>P-0036570-T02-IM617_7577529_A_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.156652</td>\n",
       "      <td>p.Ile251Asn</td>\n",
       "      <td>251</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11816</th>\n",
       "      <td>P-0018720-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11817</th>\n",
       "      <td>P-0019192-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11818</th>\n",
       "      <td>P-0017107-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11819</th>\n",
       "      <td>P-0018918-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11820</th>\n",
       "      <td>P-0012675-T02-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18706 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Tumor_Id                         tp53_key_1  \\\n",
       "8      P-0027609-T01-IM6    P-0027609-T01-IM617_7577548_C_T   \n",
       "18     P-0027408-T01-IM6  P-0027408-T01-IM617_7578409_CT_TC   \n",
       "101    P-0025997-T01-IM6    P-0025997-T01-IM617_7578471_G_-   \n",
       "104    P-0036570-T01-IM6    P-0036570-T01-IM617_7578236_A_G   \n",
       "109    P-0036570-T02-IM6    P-0036570-T02-IM617_7577529_A_T   \n",
       "...                  ...                                ...   \n",
       "11816  P-0018720-T01-IM6                               None   \n",
       "11817  P-0019192-T01-IM6                               None   \n",
       "11818  P-0017107-T01-IM6                               None   \n",
       "11819  P-0018918-T01-IM6                               None   \n",
       "11820  P-0012675-T02-IM6                               None   \n",
       "\n",
       "               tp53_vc_1  tp53_ccf_1  tp53_vaf_1        tp53_HGVSp_1  \\\n",
       "8      Missense_Mutation        0.79    0.118574         p.Gly245Ser   \n",
       "18     Missense_Mutation        1.00    0.168901         p.Arg174Glu   \n",
       "101      Frame_Shift_Del        1.00    0.912621  p.Gly154AlafsTer16   \n",
       "104    Missense_Mutation        1.00    0.718605         p.Tyr205His   \n",
       "109    Missense_Mutation        1.00    0.156652         p.Ile251Asn   \n",
       "...                  ...         ...         ...                 ...   \n",
       "11816               None         NaN         NaN                None   \n",
       "11817               None         NaN         NaN                None   \n",
       "11818               None         NaN         NaN                None   \n",
       "11819               None         NaN         NaN                None   \n",
       "11820               None         NaN         NaN                None   \n",
       "\n",
       "      tp53_spot_1 tp53_key_2 tp53_vc_2  tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2  \\\n",
       "8             245       None      None         NaN         NaN         None   \n",
       "18            174       None      None         NaN         NaN         None   \n",
       "101           154       None      None         NaN         NaN         None   \n",
       "104           205       None      None         NaN         NaN         None   \n",
       "109           251       None      None         NaN         NaN         None   \n",
       "...           ...        ...       ...         ...         ...          ...   \n",
       "11816        None       None      None         NaN         NaN         None   \n",
       "11817        None       None      None         NaN         NaN         None   \n",
       "11818        None       None      None         NaN         NaN         None   \n",
       "11819        None       None      None         NaN         NaN         None   \n",
       "11820        None       None      None         NaN         NaN         None   \n",
       "\n",
       "      tp53_spot_2 tp53_key_3 tp53_vc_3  tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3  \\\n",
       "8            None       None      None         NaN         NaN         None   \n",
       "18           None       None      None         NaN         NaN         None   \n",
       "101          None       None      None         NaN         NaN         None   \n",
       "104          None       None      None         NaN         NaN         None   \n",
       "109          None       None      None         NaN         NaN         None   \n",
       "...           ...        ...       ...         ...         ...          ...   \n",
       "11816        None       None      None         NaN         NaN         None   \n",
       "11817        None       None      None         NaN         NaN         None   \n",
       "11818        None       None      None         NaN         NaN         None   \n",
       "11819        None       None      None         NaN         NaN         None   \n",
       "11820        None       None      None         NaN         NaN         None   \n",
       "\n",
       "      tp53_spot_3 tp53_key_4 tp53_vc_4  tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4  \\\n",
       "8            None       None      None         NaN         NaN         None   \n",
       "18           None       None      None         NaN         NaN         None   \n",
       "101          None       None      None         NaN         NaN         None   \n",
       "104          None       None      None         NaN         NaN         None   \n",
       "109          None       None      None         NaN         NaN         None   \n",
       "...           ...        ...       ...         ...         ...          ...   \n",
       "11816        None       None      None         NaN         NaN         None   \n",
       "11817        None       None      None         NaN         NaN         None   \n",
       "11818        None       None      None         NaN         NaN         None   \n",
       "11819        None       None      None         NaN         NaN         None   \n",
       "11820        None       None      None         NaN         NaN         None   \n",
       "\n",
       "      tp53_spot_4 tp53_key_5 tp53_vc_5  tp53_ccf_5  tp53_vaf_5 tp53_HGVSp_5  \\\n",
       "8            None       None      None         NaN         NaN         None   \n",
       "18           None       None      None         NaN         NaN         None   \n",
       "101          None       None      None         NaN         NaN         None   \n",
       "104          None       None      None         NaN         NaN         None   \n",
       "109          None       None      None         NaN         NaN         None   \n",
       "...           ...        ...       ...         ...         ...          ...   \n",
       "11816        None       None      None         NaN         NaN         None   \n",
       "11817        None       None      None         NaN         NaN         None   \n",
       "11818        None       None      None         NaN         NaN         None   \n",
       "11819        None       None      None         NaN         NaN         None   \n",
       "11820        None       None      None         NaN         NaN         None   \n",
       "\n",
       "      tp53_spot_5  tp53_count  \n",
       "8            None           1  \n",
       "18           None           1  \n",
       "101          None           1  \n",
       "104          None           1  \n",
       "109          None           1  \n",
       "...           ...         ...  \n",
       "11816        None           0  \n",
       "11817        None           0  \n",
       "11818        None           0  \n",
       "11819        None           0  \n",
       "11820        None           0  \n",
       "\n",
       "[18706 rows x 32 columns]"
      ]
     },
     "execution_count": 224,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_tp53_muts(cohort=cohort_type, sample_info=sample_info)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Copy Numbers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we gather the information from gene_level table.\n",
    "We creaste the following columns:\n",
    "* Sample_Id \n",
    "* tcn --> total copy number\n",
    "* mcn --> major copy number\n",
    "* lcn --> lower copy number\n",
    "* seg_length --> length of the segment\n",
    "* cn_state --> copy number state\n",
    "* cf --> Cell fraction of the cn_state\n",
    "* wgd --> Wholde Genome Doubling (True or False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [],
   "source": [
    "# WARNING: THis function needs sample_info to work\n",
    "def create_copy_number_state(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    # We want TP53 locus so we have to filter the gene\n",
    "    gene_level = gene_level[gene_level['gene'] == 'TP53']\n",
    "\n",
    "    gene_level['Tumor_Id'] = gene_level['sample'].str[:17]\n",
    "    gene_level_subset = gene_level[['sample','tcn','mcn','lcn','seg_length','cn_state', 'cf.em']]\n",
    "    \n",
    "    # We rename the cf.em column \n",
    "    gene_level_subset = gene_level_subset.rename(columns={'cf.em': 'tp53_cf', \n",
    "                                                          'sample':'Sample_Id',\n",
    "                                                          'tcn': 'tp53_tcn',\n",
    "                                                          'mcn': 'tp53_mcn',\n",
    "                                                          'lcn': 'tp53_lcn',\n",
    "                                                          'seg_length': 'tp53_seg_length',\n",
    "                                                          'cn_state':'tp53_cn_state'})\n",
    "    \n",
    "    # We add WGD information\n",
    "    wgd = cohort[['sample_id', 'wgd']]\n",
    "    \n",
    "    final = pd.merge(gene_level_subset, wgd, left_on='Sample_Id', right_on='sample_id').drop(['sample_id'], axis=1)\n",
    "    \n",
    "    \n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 38.5 ms, sys: 5.41 ms, total: 43.9 ms\n",
      "Wall time: 42.4 ms\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>tp53_tcn</th>\n",
       "      <th>tp53_mcn</th>\n",
       "      <th>tp53_lcn</th>\n",
       "      <th>tp53_seg_length</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>tp53_cf</th>\n",
       "      <th>wgd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668592</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0009819-T01-IM5_P-0009819-N01-IM5</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668300</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0027609-T01-IM6_P-0027609-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18461192</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0027408-T01-IM6_P-0027408-N01-IM6</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26256025</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.275073</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0006554-T01-IM5_P-0006554-N01-IM5</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>40254480</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18702</th>\n",
       "      <td>P-0050747-T01-IM6_P-0050747-N01-IM6</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18461117</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.254038</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18703</th>\n",
       "      <td>P-0050745-T01-IM6_P-0050745-N01-IM6</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29307621</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.602685</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18704</th>\n",
       "      <td>P-0047736-T01-IM6_P-0047736-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>80668548</td>\n",
       "      <td>CNLOH</td>\n",
       "      <td>0.184186</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705</th>\n",
       "      <td>P-0047148-T01-IM6_P-0047148-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668400</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18706</th>\n",
       "      <td>P-0046531-T01-IM6_P-0046531-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668500</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18707 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Sample_Id  tp53_tcn  tp53_mcn  tp53_lcn  \\\n",
       "0      P-0034223-T01-IM6_P-0034223-N01-IM6         2       1.0       1.0   \n",
       "1      P-0009819-T01-IM5_P-0009819-N01-IM5         2       1.0       1.0   \n",
       "2      P-0027609-T01-IM6_P-0027609-N01-IM6         2       1.0       1.0   \n",
       "3      P-0027408-T01-IM6_P-0027408-N01-IM6         1       1.0       0.0   \n",
       "4      P-0006554-T01-IM5_P-0006554-N01-IM5         2       1.0       1.0   \n",
       "...                                    ...       ...       ...       ...   \n",
       "18702  P-0050747-T01-IM6_P-0050747-N01-IM6         1       1.0       0.0   \n",
       "18703  P-0050745-T01-IM6_P-0050745-N01-IM6         1       1.0       0.0   \n",
       "18704  P-0047736-T01-IM6_P-0047736-N01-IM6         2       2.0       0.0   \n",
       "18705  P-0047148-T01-IM6_P-0047148-N01-IM6         2       1.0       1.0   \n",
       "18706  P-0046531-T01-IM6_P-0046531-N01-IM6         2       1.0       1.0   \n",
       "\n",
       "       tp53_seg_length tp53_cn_state   tp53_cf    wgd  \n",
       "0             80668592       DIPLOID  1.000000  False  \n",
       "1             80668300       DIPLOID  1.000000  False  \n",
       "2             18461192       DIPLOID  1.000000  False  \n",
       "3             26256025       HETLOSS  0.275073  False  \n",
       "4             40254480       DIPLOID  1.000000  False  \n",
       "...                ...           ...       ...    ...  \n",
       "18702         18461117       HETLOSS  0.254038  False  \n",
       "18703         29307621       HETLOSS  0.602685  False  \n",
       "18704         80668548         CNLOH  0.184186  False  \n",
       "18705         80668400       DIPLOID  1.000000  False  \n",
       "18706         80668500       DIPLOID  1.000000  False  \n",
       "\n",
       "[18707 rows x 8 columns]"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time \n",
    "copy_number_info = create_copy_number_state(cohort = cohort_type)\n",
    "copy_number_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computed metrics\n",
    "In this part we define functions to be applied on the master file to compute specific metrics.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes, mutations and max_vaf\n",
    "* *create_gene_count*: count of mutated genes for the given sample\n",
    "* *create_mut_count*: count of mutations for the given sample\n",
    "* *get_max_vaf*: the maximum of Variant Allele Frequency within all mutation of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [],
   "source": [
    "# These 2 first functions allow  to count the total number of genes/mutations per sample.\n",
    "def create_gene_count(x):\n",
    "    '''\n",
    "    This function create the count of genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    \n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "\n",
    "    nb_genes = selected_cohort.groupby('Entrez_Gene_Id').size().shape[0]\n",
    "\n",
    "    return nb_genes\n",
    "\n",
    "def create_mut_count(x):\n",
    "    '''\n",
    "    This function computes the dataframe of all mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    \n",
    "    return selected_cohort.shape[0]\n",
    "\n",
    "# The following calculates the max_vaf for a given sample\n",
    "def get_max_vaf(x):\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    \n",
    "    return selected_cohort['t_var_freq'].max()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genome Instability (frac_genome_aletered, chr_affected by loss, cnkloh, gain )\n",
    "This script is very long to run, so I decide to run it apart and save the results in a pickle fil that we will merge with our master.\n",
    "\n",
    "* For no_wgd: 40 minutes (MacBook Pro 2019)\n",
    "* For wgd:  minutes (MacBook Pro 2019)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 253,
   "metadata": {},
   "outputs": [],
   "source": [
    "def condition_CNLOH(x):\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    if x.cn_state in CNLOH:\n",
    "        return 'CNLOH' + x.chr\n",
    "    else:\n",
    "        return 'NO_CNLOH'  + x.chr \n",
    "    \n",
    "def condition_GAIN(x):\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    if x.cn_state in GAIN:\n",
    "        return 'GAIN'+ x.chr\n",
    "    else:\n",
    "        return 'NO_GAIN'+ x.chr\n",
    "\n",
    "def condition_LOSS(x):\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    if x.cn_state in LOSS:\n",
    "        return 'LOSS'+ x.chr\n",
    "    else:\n",
    "        return 'NO_LOSS'+ x.chr\n",
    "\n",
    "def compute_frac_genome(x, arm_level):\n",
    "    Sample_Id = x.Sample_Id\n",
    "    wgd = x.wgd\n",
    "    lookup_table = arm_level[arm_level['sample'] == Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "        \n",
    "    altered_length = lookup_table_altered.cn_length.sum()\n",
    "    total_length = lookup_table.arm_length.sum()\n",
    "    \n",
    "    frac_gen_altered = round(altered_length/total_length,3)\n",
    "    \n",
    "    return frac_gen_altered\n",
    "\n",
    "# Here is the function that allws to compute genome instability columns\n",
    "def chr_computations(x, arm_level):\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    wgd=x.wgd\n",
    "    arm_level_samples = list(set(arm_level['sample']))\n",
    "    \n",
    "    if x.Sample_Id not in arm_level_samples:\n",
    "        return ['NaN','NaN','NaN','NaN', 'NaN']\n",
    "    \n",
    "    lookup_table = arm_level[arm_level['sample'] == x.Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "    lookup_table['state_chr'] = lookup_table['cn_state']+lookup_table['chr']\n",
    "    \n",
    "    # If only DIPLOID or TETRAPLOID\n",
    "    if lookup_table.empty:\n",
    "        return [float(0)]*5\n",
    "    \n",
    "    lookup_table['cnloh_chr'] = lookup_table.apply(condition_CNLOH, axis=1)\n",
    "    lookup_table['loss_chr'] = lookup_table.apply(condition_LOSS, axis=1)\n",
    "    lookup_table['gain_chr'] = lookup_table.apply(condition_GAIN, axis=1)\n",
    "\n",
    "    #chr_affected colum\n",
    "    lookup_table_chr = lookup_table.drop_duplicates(subset=['chr'])\n",
    "    chr_affected = len(lookup_table_chr)\n",
    "    \n",
    "    #chr_loss, chr_gain, chr_cnloh columns\n",
    "    lookup_table_cnloh = lookup_table.drop_duplicates(subset=['cnloh_chr'])['cnloh_chr']\n",
    "    lookup_table_loss = lookup_table.drop_duplicates(subset=['loss_chr'])['loss_chr']\n",
    "    lookup_table_gain = lookup_table.drop_duplicates(subset=['gain_chr'])['gain_chr']\n",
    "\n",
    "    chr_loss = len(lookup_table_loss[lookup_table_loss.str.startswith('LOSS')])\n",
    "    chr_gain = len(lookup_table_gain[lookup_table_gain.str.startswith('GAIN')])\n",
    "    chr_cnloh = len(lookup_table_cnloh[lookup_table_cnloh.str.startswith('CNLOH')])\n",
    "    \n",
    "    #frac_gen_altered column\n",
    "    frac_gen_altered = compute_frac_genome(x, arm_level=arm_level)\n",
    "    \n",
    "    return [chr_affected, chr_loss, chr_gain, chr_cnloh, frac_gen_altered]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So we compute it and save it in a file:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 256,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "8d5faba857e44a3da1e4f16288b37f23",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=1.0, bar_style='info', max=1.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "2f058d83ece645348ed4523af9d14f4a",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=0.0, max=18707.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "checkpoint 1\n",
      "checkpoint 2\n"
     ]
    }
   ],
   "source": [
    "from tqdm import tqdm,tqdm_notebook\n",
    "\n",
    "def compute_genome_instability(cohort:str):\n",
    "    tqdm_notebook().pandas()\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    copy_number_info = create_copy_number_state(cohort = cohort_type)\n",
    "    copy_number_info_ = copy_number_info\n",
    "    copy_number_info_['chr_comput'] = copy_number_info_.progress_apply(chr_computations,arm_level=arm_level, axis=1)\n",
    "    print('checkpoint 1')\n",
    "    copy_number_info[['chr_affected', 'chr_loss', 'chr_gain', 'chr_cnloh', 'frac_genome_altered']] = pd.DataFrame(copy_number_info_.chr_comput.values.tolist(), index= copy_number_info_.index)\n",
    "    print('checkpoint 2')\n",
    "    \n",
    "    return copy_number_info_\n",
    "\n",
    "copy_number_info = compute_genome_instability(cohort=cohort_type)\n",
    "copy_number_info[['Sample_Id','chr_affected', 'chr_loss', 'chr_gain', 'chr_cnloh', 'frac_genome_altered']].to_pickle(total_output + 'gi_metrics_{}.pkl'.format(cohort_type))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 260,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>chr_affected</th>\n",
       "      <th>chr_loss</th>\n",
       "      <th>chr_gain</th>\n",
       "      <th>chr_cnloh</th>\n",
       "      <th>frac_genome_altered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0009819-T01-IM5_P-0009819-N01-IM5</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0027609-T01-IM6_P-0027609-N01-IM6</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0027408-T01-IM6_P-0027408-N01-IM6</td>\n",
       "      <td>12.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0006554-T01-IM5_P-0006554-N01-IM5</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18702</th>\n",
       "      <td>P-0050747-T01-IM6_P-0050747-N01-IM6</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18703</th>\n",
       "      <td>P-0050745-T01-IM6_P-0050745-N01-IM6</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18704</th>\n",
       "      <td>P-0047736-T01-IM6_P-0047736-N01-IM6</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705</th>\n",
       "      <td>P-0047148-T01-IM6_P-0047148-N01-IM6</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18706</th>\n",
       "      <td>P-0046531-T01-IM6_P-0046531-N01-IM6</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.202</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18707 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Sample_Id  chr_affected  chr_loss  chr_gain  \\\n",
       "0      P-0034223-T01-IM6_P-0034223-N01-IM6           3.0       2.0       2.0   \n",
       "1      P-0009819-T01-IM5_P-0009819-N01-IM5           4.0       4.0       0.0   \n",
       "2      P-0027609-T01-IM6_P-0027609-N01-IM6           4.0       0.0       4.0   \n",
       "3      P-0027408-T01-IM6_P-0027408-N01-IM6          12.0      11.0       3.0   \n",
       "4      P-0006554-T01-IM5_P-0006554-N01-IM5           5.0       4.0       1.0   \n",
       "...                                    ...           ...       ...       ...   \n",
       "18702  P-0050747-T01-IM6_P-0050747-N01-IM6           5.0       5.0       1.0   \n",
       "18703  P-0050745-T01-IM6_P-0050745-N01-IM6           7.0       7.0       0.0   \n",
       "18704  P-0047736-T01-IM6_P-0047736-N01-IM6           7.0       2.0       1.0   \n",
       "18705  P-0047148-T01-IM6_P-0047148-N01-IM6           5.0       2.0       3.0   \n",
       "18706  P-0046531-T01-IM6_P-0046531-N01-IM6           5.0       4.0       0.0   \n",
       "\n",
       "       chr_cnloh  frac_genome_altered  \n",
       "0            0.0                0.080  \n",
       "1            0.0                0.137  \n",
       "2            0.0                0.138  \n",
       "3            0.0                0.444  \n",
       "4            0.0                0.173  \n",
       "...          ...                  ...  \n",
       "18702        0.0                0.192  \n",
       "18703        0.0                0.668  \n",
       "18704        3.0                0.293  \n",
       "18705        0.0                0.249  \n",
       "18706        1.0                0.202  \n",
       "\n",
       "[18707 rows x 6 columns]"
      ]
     },
     "execution_count": 260,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_pickle(total_output + 'gi_metrics_{}.pkl'.format(cohort_type))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>arm</th>\n",
       "      <th>tcn</th>\n",
       "      <th>lcn</th>\n",
       "      <th>cn_length</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>frac_of_arm</th>\n",
       "      <th>cn_state</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>1p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>120534257</td>\n",
       "      <td>120534257</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>1q</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>101461100</td>\n",
       "      <td>125135032</td>\n",
       "      <td>0.81</td>\n",
       "      <td>GAIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>2p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>87764506</td>\n",
       "      <td>87764506</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>2q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>150656179</td>\n",
       "      <td>150656179</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>3p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>89055004</td>\n",
       "      <td>89055004</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>3q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>106857923</td>\n",
       "      <td>106857923</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>4p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>48257537</td>\n",
       "      <td>48257537</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>4q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>140602481</td>\n",
       "      <td>140602481</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>5p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>46187241</td>\n",
       "      <td>46187241</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>5q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>134121897</td>\n",
       "      <td>134121897</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>6p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>58437159</td>\n",
       "      <td>58437159</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>6q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>111968888</td>\n",
       "      <td>111968888</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>7p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>56381732</td>\n",
       "      <td>56381732</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>7q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>100573669</td>\n",
       "      <td>100573669</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>8q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>102347589</td>\n",
       "      <td>102347589</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>9p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>46945819</td>\n",
       "      <td>46945819</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>9q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>92452921</td>\n",
       "      <td>92452921</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>10p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>37744888</td>\n",
       "      <td>37744888</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>10q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>95926530</td>\n",
       "      <td>95926530</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>11p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>51319187</td>\n",
       "      <td>51319187</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>12p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>22819044</td>\n",
       "      <td>22848994</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>12q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>98407356</td>\n",
       "      <td>98407356</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>13q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>93122091</td>\n",
       "      <td>93122091</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>14q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>86418558</td>\n",
       "      <td>86418558</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>15q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>80605814</td>\n",
       "      <td>80605814</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>16p</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>33258581</td>\n",
       "      <td>35220670</td>\n",
       "      <td>0.94</td>\n",
       "      <td>GAIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>16q</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>54904790</td>\n",
       "      <td>54904790</td>\n",
       "      <td>1.00</td>\n",
       "      <td>HETLOSS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>17p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>22148337</td>\n",
       "      <td>22148337</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>17q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>58520255</td>\n",
       "      <td>58520255</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>18p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>14736648</td>\n",
       "      <td>14736648</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>18q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>59002560</td>\n",
       "      <td>59002560</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>19p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>23475032</td>\n",
       "      <td>23475032</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>19q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>34414558</td>\n",
       "      <td>34414558</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>20p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>26023705</td>\n",
       "      <td>26023705</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>20q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>36311101</td>\n",
       "      <td>36311101</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>21q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>34964871</td>\n",
       "      <td>34964871</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>22q</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>31302129</td>\n",
       "      <td>31302129</td>\n",
       "      <td>1.00</td>\n",
       "      <td>HETLOSS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>23p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>58180962</td>\n",
       "      <td>58180962</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>23q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>94232388</td>\n",
       "      <td>94232388</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 sample  arm  tcn  lcn  cn_length  arm_length  \\\n",
       "0   P-0034223-T01-IM6_P-0034223-N01-IM6   1p    2    1  120534257   120534257   \n",
       "1   P-0034223-T01-IM6_P-0034223-N01-IM6   1q    4    1  101461100   125135032   \n",
       "2   P-0034223-T01-IM6_P-0034223-N01-IM6   2p    2    1   87764506    87764506   \n",
       "3   P-0034223-T01-IM6_P-0034223-N01-IM6   2q    2    1  150656179   150656179   \n",
       "4   P-0034223-T01-IM6_P-0034223-N01-IM6   3p    2    1   89055004    89055004   \n",
       "5   P-0034223-T01-IM6_P-0034223-N01-IM6   3q    2    1  106857923   106857923   \n",
       "6   P-0034223-T01-IM6_P-0034223-N01-IM6   4p    2    1   48257537    48257537   \n",
       "7   P-0034223-T01-IM6_P-0034223-N01-IM6   4q    2    1  140602481   140602481   \n",
       "8   P-0034223-T01-IM6_P-0034223-N01-IM6   5p    2    1   46187241    46187241   \n",
       "9   P-0034223-T01-IM6_P-0034223-N01-IM6   5q    2    1  134121897   134121897   \n",
       "10  P-0034223-T01-IM6_P-0034223-N01-IM6   6p    2    1   58437159    58437159   \n",
       "11  P-0034223-T01-IM6_P-0034223-N01-IM6   6q    2    1  111968888   111968888   \n",
       "12  P-0034223-T01-IM6_P-0034223-N01-IM6   7p    2    1   56381732    56381732   \n",
       "13  P-0034223-T01-IM6_P-0034223-N01-IM6   7q    2    1  100573669   100573669   \n",
       "14  P-0034223-T01-IM6_P-0034223-N01-IM6   8q    2    1  102347589   102347589   \n",
       "15  P-0034223-T01-IM6_P-0034223-N01-IM6   9p    2    1   46945819    46945819   \n",
       "16  P-0034223-T01-IM6_P-0034223-N01-IM6   9q    2    1   92452921    92452921   \n",
       "17  P-0034223-T01-IM6_P-0034223-N01-IM6  10p    2    1   37744888    37744888   \n",
       "18  P-0034223-T01-IM6_P-0034223-N01-IM6  10q    2    1   95926530    95926530   \n",
       "19  P-0034223-T01-IM6_P-0034223-N01-IM6  11p    2    1   51319187    51319187   \n",
       "20  P-0034223-T01-IM6_P-0034223-N01-IM6  12p    2    1   22819044    22848994   \n",
       "21  P-0034223-T01-IM6_P-0034223-N01-IM6  12q    2    1   98407356    98407356   \n",
       "22  P-0034223-T01-IM6_P-0034223-N01-IM6  13q    2    1   93122091    93122091   \n",
       "23  P-0034223-T01-IM6_P-0034223-N01-IM6  14q    2    1   86418558    86418558   \n",
       "24  P-0034223-T01-IM6_P-0034223-N01-IM6  15q    2    1   80605814    80605814   \n",
       "25  P-0034223-T01-IM6_P-0034223-N01-IM6  16p    3    1   33258581    35220670   \n",
       "26  P-0034223-T01-IM6_P-0034223-N01-IM6  16q    1    0   54904790    54904790   \n",
       "27  P-0034223-T01-IM6_P-0034223-N01-IM6  17p    2    1   22148337    22148337   \n",
       "28  P-0034223-T01-IM6_P-0034223-N01-IM6  17q    2    1   58520255    58520255   \n",
       "29  P-0034223-T01-IM6_P-0034223-N01-IM6  18p    2    1   14736648    14736648   \n",
       "30  P-0034223-T01-IM6_P-0034223-N01-IM6  18q    2    1   59002560    59002560   \n",
       "31  P-0034223-T01-IM6_P-0034223-N01-IM6  19p    2    1   23475032    23475032   \n",
       "32  P-0034223-T01-IM6_P-0034223-N01-IM6  19q    2    1   34414558    34414558   \n",
       "33  P-0034223-T01-IM6_P-0034223-N01-IM6  20p    2    1   26023705    26023705   \n",
       "34  P-0034223-T01-IM6_P-0034223-N01-IM6  20q    2    1   36311101    36311101   \n",
       "35  P-0034223-T01-IM6_P-0034223-N01-IM6  21q    2    1   34964871    34964871   \n",
       "36  P-0034223-T01-IM6_P-0034223-N01-IM6  22q    1    0   31302129    31302129   \n",
       "37  P-0034223-T01-IM6_P-0034223-N01-IM6  23p    2    1   58180962    58180962   \n",
       "38  P-0034223-T01-IM6_P-0034223-N01-IM6  23q    2    1   94232388    94232388   \n",
       "\n",
       "    frac_of_arm cn_state  \n",
       "0          1.00  DIPLOID  \n",
       "1          0.81     GAIN  \n",
       "2          1.00  DIPLOID  \n",
       "3          1.00  DIPLOID  \n",
       "4          1.00  DIPLOID  \n",
       "5          1.00  DIPLOID  \n",
       "6          1.00  DIPLOID  \n",
       "7          1.00  DIPLOID  \n",
       "8          1.00  DIPLOID  \n",
       "9          1.00  DIPLOID  \n",
       "10         1.00  DIPLOID  \n",
       "11         1.00  DIPLOID  \n",
       "12         1.00  DIPLOID  \n",
       "13         1.00  DIPLOID  \n",
       "14         1.00  DIPLOID  \n",
       "15         1.00  DIPLOID  \n",
       "16         1.00  DIPLOID  \n",
       "17         1.00  DIPLOID  \n",
       "18         1.00  DIPLOID  \n",
       "19         1.00  DIPLOID  \n",
       "20         1.00  DIPLOID  \n",
       "21         1.00  DIPLOID  \n",
       "22         1.00  DIPLOID  \n",
       "23         1.00  DIPLOID  \n",
       "24         1.00  DIPLOID  \n",
       "25         0.94     GAIN  \n",
       "26         1.00  HETLOSS  \n",
       "27         1.00  DIPLOID  \n",
       "28         1.00  DIPLOID  \n",
       "29         1.00  DIPLOID  \n",
       "30         1.00  DIPLOID  \n",
       "31         1.00  DIPLOID  \n",
       "32         1.00  DIPLOID  \n",
       "33         1.00  DIPLOID  \n",
       "34         1.00  DIPLOID  \n",
       "35         1.00  DIPLOID  \n",
       "36         1.00  HETLOSS  \n",
       "37         1.00  DIPLOID  \n",
       "38         1.00  DIPLOID  "
      ]
     },
     "execution_count": 259,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arm_level_nowgd[arm_level_nowgd['sample'] == 'P-0034223-T01-IM6_P-0034223-N01-IM6']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TP53 Residual"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following function needs to be called on the complete master file because it needs info from different parts\n",
    "# It computes the expected number of tp53 mutant copies in a cell\n",
    "def create_copies_tp53_muts(master):\n",
    "    master['tp53_exp_nb_1'] = master.apply(lambda x:(x.tp53_vaf_1 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_2'] = master.apply(lambda x:(x.tp53_vaf_2 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_3'] = master.apply(lambda x:(x.tp53_vaf_3 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_4'] = master.apply(lambda x:(x.tp53_vaf_4 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_5'] = master.apply(lambda x:(x.tp53_vaf_5 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    \n",
    "    return master\n",
    "\n",
    "\n",
    "# The following computes the expected number of copies of tp53 residuals \n",
    "def create_tp53_res(master):\n",
    "    master['tp53_res_1'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_1, axis = 1)\n",
    "    master['tp53_res_2'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_2, axis = 1)\n",
    "    master['tp53_res_3'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_3, axis = 1)\n",
    "    master['tp53_res_4'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_4, axis = 1)\n",
    "    master['tp53_res_5'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_5, axis = 1)\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mutation Types Grouping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following functions allow to group the Mutation Types\n",
    "def vc_group_cond_1(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_1 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_1 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_1 in missense: \n",
    "        if x.tp53_spot_1 in ['273','248','175']: return x.tp53_spot_1\n",
    "        elif x.tp53_spot_1 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_2(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_2 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_2 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_2 in missense: \n",
    "        if x.tp53_spot_2 in ['273','248','175']: return x.tp53_spot_2\n",
    "        elif x.tp53_spot_2 in['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'   \n",
    "def vc_group_cond_3(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_3 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_3 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_3 in missense: \n",
    "        if x.tp53_spot_3 in ['273','248','175']: return x.tp53_spot_3\n",
    "        elif x.tp53_spot_3 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense' \n",
    "def vc_group_cond_4(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_4 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_4 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_4 in missense: \n",
    "        if x.tp53_spot_4 in ['273','248','175']: return x.tp53_spot_4\n",
    "        elif x.tp53_spot_4 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_5(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_5 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_5 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_5 in missense: \n",
    "        if x.tp53_spot_5 in ['273','248','175']: return x.tp53_spot_5\n",
    "        elif x.tp53_spot_5 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Non-WGD specific functions\n",
    "\n",
    "* *cn_group_cond*\n",
    "\n",
    "First, we group the different Copy Number States *cn_state* in subgroups, under the column *cn_group*:\n",
    "    * Group 1: cnLOH gathering ['CNLOH']\n",
    "    * Group 2: LOSS gathering ['HETLOSS']\n",
    "    * Group 3: HOMDEL gathering ['HOMDEL']\n",
    "    * Group 4: WILD_TYPE gathering ['DIPLOID', 'TETRAPLOID']\n",
    "    * Group 5: GAIN gathering ['GAIN']\n",
    "    * Group 6: OTHER gathering ['AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']\n",
    "\n",
    "* *mut_cn_group_cond*\n",
    "\n",
    "Based on this first column we define 7 final groups of patients adding the mutational information. These groups will be under the column *tp53_group*.\n",
    "    * Group 1: Samples with 0 tp53 mutations and HETLOSS\n",
    "    * Group 2: Samples with HOMDEL\n",
    "    * Group 3: Samples with 1 tp53 mutation and WILD_TYPE (DIPLOID, LOSS AFTER, TETRAPLOID)\n",
    "    * Group 4: Samples with 1 tp53 mutation or more and LOSS\n",
    "    * Group 5: Samples with 1 tp53 mutation or more and cnLOH\n",
    "    * Group 6: Samples with 2/3/4/5 tp53 mutations and WILD_TYPE or GAIN\n",
    "\n",
    "* *tp53_residual_group*: defines if we have residual tp53 or not based on the expected WT tp53 residual\n",
    "\n",
    "\n",
    "* *get_loh_nowgd*\n",
    "\n",
    "The last function is meant to compute the LOH state"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [],
   "source": [
    "def cn_group_cond(x):\n",
    "    if x.tp53_cn_state in ['CNLOH']:\n",
    "        return 'cnLOH'\n",
    "    if x.tp53_cn_state in ['HETLOSS']:\n",
    "        return 'LOSS'\n",
    "    if x.tp53_cn_state == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_cn_state in ['DIPLOID', 'TETRAPLOID']:\n",
    "        return 'WILD_TYPE'\n",
    "    if x.tp53_cn_state == 'GAIN':\n",
    "        return 'GAIN'\n",
    "    if x.tp53_cn_state in ['AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']:\n",
    "        return 'OTHER'\n",
    "\n",
    "def mut_cn_group_cond(x):\n",
    "    if x.tp53_cn_state == 'HETLOSS' and x.tp53_count == 0:\n",
    "        return '0_HETLOSS'\n",
    "    if x.tp53_first_group == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_first_group == 'WILD_TYPE' and x.tp53_count == 1 :\n",
    "        return '1_WILD_TYPE'\n",
    "    if x.tp53_first_group == 'LOSS' and x.tp53_count >=1:\n",
    "        return '>=1_LOSS'\n",
    "    if x.tp53_first_group == 'cnLOH' and x.tp53_count >=1:\n",
    "        return '>=1_cnLOH'\n",
    "    if (x.tp53_first_group == 'WILD_TYPE' or x.tp53_first_group == 'GAIN') and x.tp53_count > 1:\n",
    "        return '>1muts'\n",
    "    \n",
    "    \n",
    "def tp53_residual_group(x):\n",
    "    if x.tp53_group == '1_WILD_TYPE' or x.tp53_group == '0_HETLOSS':\n",
    "        return 'tp53_res'\n",
    "    if x.tp53_group == 'HOMDEL':\n",
    "        return 'no_tp53_res'\n",
    "    if x.tp53_group == '>=1_LOSS' or x.tp53_group == '>=1_cnLOH':\n",
    "        if (x.tp53_res_1 < 0.5) or (x.tp53_res_2 < 0.5):\n",
    "            return 'no_tp53_res'\n",
    "        else:\n",
    "            if (x.tp53_cf + max(x.tp53_ccf_1, x.tp53_ccf_2, x.tp53_ccf_3, x.tp53_ccf_4, x.tp53_ccf_5)) > 1:\n",
    "                return 'no_tp53_res'\n",
    "            else:\n",
    "                return 'uncertain'\n",
    "    if x.tp53_group == '>1muts':\n",
    "        if (x.tp53_res_1 + x.tp53_res_2 < 2.5):\n",
    "            if (x.tp53_ccf_1 + x.tp53_ccf_2 > 1):\n",
    "                return 'no_tp53_res'\n",
    "            else: \n",
    "                return 'uncertain'\n",
    "           \n",
    "        elif (x.tp53_res_1 + x.tp53_res_2 > 2.5):\n",
    "            return 'tp53_res'\n",
    "        \n",
    "        \n",
    "def get_loh_nowgd(x):\n",
    "    if x.tp53_cn_state in ['CNLOH', 'CNLOH & GAIN', 'AMP (LOH)']:\n",
    "        return True\n",
    "    else: return False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WGD specific functions\n",
    "Finally we have functions to compute TP53 allelic state BEFORE WGD (for the WGD cohort only):\n",
    "* *get_bi_nobi* allows to say if the pre WGD allelic state was Bi-Allelic or not\n",
    "* *get_mono* allows to say if, within Not Bi-allelic samples, the state was mono-allelic or with 2 WT alleles\n",
    "* *get_loh_wgd* that computes if the samples is in LOH state or not"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "            NUMBER OF TP53 residual ASSOCIATED WITH ALLELIC STATE BEFORE WGD\n",
    "            \n",
    "            \n",
    "                                     BI-ALLELIC | MONO - ALLELIC | 2WT\n",
    "\n",
    "                LOSS BEFORE :            0      |        1       | -\n",
    "                CNLOH BEFORE & LOSS:     0      |       1,2      | 2\n",
    "                CNLOH BEFORE:            0      |        2       | 3\n",
    "                LOSS AFTER:              -      |       1,2      | 2\n",
    "                DOUBLE LOSS AFTER:       -      |       0,1      | 1\n",
    "                TETRAPLOID:              -      |        2       | 3\n",
    "                CNLOH AFTER:             -      |       1,3      | 3\n",
    "                CNLOH BEFORE & GAIN:     0      |        2,3     | 3,4\n",
    "                \n",
    "\n",
    "                        THRESHOLDS BETWEEN ALLELIC STATES BEFORE WGD\n",
    "                    \n",
    "                                     BI-ALLELIC | MONO - ALLELIC | 2WT\n",
    "\n",
    "                LOSS BEFORE :            <0.4   |        >0.6    | -\n",
    "                CNLOH BEFORE & LOSS:     <0.4   |      0.6< <1.5 | uncertain\n",
    "                CNLOH BEFORE:            <1.5   |   1.5< <2.5    | >2.5\n",
    "                LOSS AFTER:              -      |      <1.5      | uncertain \n",
    "                DOUBLE LOSS AFTER:       -      |       <0.5     | uncertain\n",
    "                TETRAPLOID:              -      |       <2.5     | >2.5\n",
    "                CNLOH AFTER:             -      |       <1.5     | uncertain\n",
    "                CNLOH BEFORE & GAIN:     <1.4   |   1.6< <2.5    | uncertain\n",
    "                \n",
    "    \n",
    "\n",
    " '''\n",
    "\n",
    "\n",
    "# The following functions are for WGD cohort, computing the tp53 allelic state before WGD\n",
    "def get_bi_nobi(x):\n",
    "    tumor = x.Tumor_Id\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    \n",
    "    #samples without TP53 mutation\n",
    "    if nb_tp53muts == 0:\n",
    "        return 'no_bi'\n",
    "    \n",
    "    # Samples with only one tp53 mutation\n",
    "    if nb_tp53muts == 1:\n",
    "        tp53_res = x.tp53_res_1\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE' or cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr = 0.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr:\n",
    "                return 'bi'\n",
    "            elif tp53_res >= thr:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state in ['LOSS AFTER','DOUBLE LOSS AFTER','TETRAPLOID','CNLOH AFTER']: \n",
    "            return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "            \n",
    "            \n",
    "            \n",
    "    # Samples with 2 tp53 mutations\n",
    "    elif nb_tp53muts == 2:\n",
    "        tp53_res_1 = x.tp53_res_1\n",
    "        tp53_res_2 = x.tp53_res_2\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr - 0.1) or (tp53_res_2 < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr + 0.1) and (tp53_res_2 > thr + 0.1):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (tp53_res_1 < thr_1 - 0.1) or (tp53_res_2 < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (tp53_res_1 < thr_1) or (tp53_res_2 < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if tp53_res_1 < thr_1 - 0.1 or tp53_res_2 < thr_1 - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr_2) and (tp53_res_2 > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "            \n",
    "    # Samples with 3 tp53 mutations\n",
    "    elif nb_tp53muts == 3:\n",
    "        tp53_res_1 = x.tp53_res_1\n",
    "        tp53_res_2 = x.tp53_res_2\n",
    "        tp53_res_3 = x.tp53_res_3\n",
    "        tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "        tp53_res.sort()\n",
    "    \n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (min(tp53_res) < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (max(tp53_res) > thr + 0.1) :\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (min(tp53_res) < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2): # we want that exactly 2 mutation arose before WGD\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (min(tp53_res) < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if min(tp53_res) < thr_1 - 0.1 :\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "    \n",
    "    else: return 'uncertain'\n",
    "\n",
    "\n",
    "# This functions can only be called after get_bi_nobi() has been called on master\n",
    "def get_mono(x):\n",
    "    tumor = x.Tumor_Id\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    bi_state = x.bi_allelic_state\n",
    "    \n",
    "    if bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    \n",
    "    elif bi_state == 'uncertain':\n",
    "        return 'uncertain_bi'\n",
    "    \n",
    "    \n",
    "    elif bi_state == 'no_bi': # We are already in the  mono/2WT distinction \n",
    "        \n",
    "        if nb_tp53muts == 0:\n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            else: return '2WT'\n",
    "            \n",
    "            \n",
    "        #1 mut samples\n",
    "        if nb_tp53muts == 1:\n",
    "            tp53_res = x.tp53_res_1\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "\n",
    "        \n",
    "        # 2 mut samples   \n",
    "        elif nb_tp53muts == 2:\n",
    "            tp53_res_1 = x.tp53_res_1\n",
    "            tp53_res_2 = x.tp53_res_2\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res_1 < 2.5 or tp53_res_2 < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res_1 > 2.5 and tp53_res_2 > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "            \n",
    "        # 3 mut samples\n",
    "        elif nb_tp53muts == 3:\n",
    "            tp53_res_1 = x.tp53_res_1\n",
    "            tp53_res_2 = x.tp53_res_2\n",
    "            tp53_res_3 = x.tp53_res_3\n",
    "            tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "            tp53_res.sort()\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if min(tp53_res) < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res[0] > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "        \n",
    "        else: return 'uncertain_mono' \n",
    "        \n",
    "def get_loh_wgd(x):\n",
    "    if x.tp53_cn_state in ['LOSS BEFORE', 'CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS BEFORE & AFTER', 'CNLOH BEFORE & GAIN']:\n",
    "        return True\n",
    "    else: return False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge Tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 261,
   "metadata": {},
   "outputs": [],
   "source": [
    "def merge_tables(cohort_type: str):\n",
    "    \n",
    "    if cohort_type == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort_type == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    sample_info = create_sample_info(cohort=cohort_type)\n",
    "    tp53_muts = create_tp53_muts(cohort=cohort_type, sample_info=sample_info)\n",
    "    copy_number_info = create_copy_number_state(cohort=cohort_type)\n",
    "    \n",
    "    master = pd.merge(sample_info, tp53_muts, on='Tumor_Id')\n",
    "    master = pd.merge(master, copy_number_info, on='Sample_Id')\n",
    "    \n",
    "    # Now we compute the columns\n",
    "    master['gene_count'] = master.apply(create_gene_count, axis=1)\n",
    "    master['mutation_count'] = master.apply(create_mut_count, axis=1)\n",
    "    master['max_vaf'] = master.apply(get_max_vaf, axis=1)\n",
    "    \n",
    "    # Tp53 residual\n",
    "    master = create_copies_tp53_muts(master)\n",
    "    master = create_tp53_res(master)\n",
    "    \n",
    "    # Mutation Type Grouping\n",
    "    master['tp53_vc_group_1'] = master.apply(vc_group_cond_1, axis = 1)\n",
    "    master['tp53_vc_group_2'] = master.apply(vc_group_cond_2, axis = 1)\n",
    "    master['tp53_vc_group_3'] = master.apply(vc_group_cond_3, axis = 1)\n",
    "    master['tp53_vc_group_4'] = master.apply(vc_group_cond_4, axis = 1)\n",
    "    master['tp53_vc_group_5'] = master.apply(vc_group_cond_5, axis = 1)\n",
    "    \n",
    "    \n",
    "    if cohort_type == 'wgd':\n",
    "        # These groups are BEFORE WGD !!!\n",
    "        master['bi_allelic_state'] = master.apply(get_bi_nobi, axis=1)\n",
    "        master['allelic_state_wgd'] = master.apply(get_mono,axis=1)\n",
    "        # This is after WGD\n",
    "        master['loh_status'] = master.apply(get_loh_wgd, axis=1)\n",
    "        \n",
    "    if cohort_type == 'no_wgd':\n",
    "        master['tp53_first_group'] = master.apply(cn_group_cond, axis = 1)\n",
    "        master['tp53_group'] = master.apply(mut_cn_group_cond, axis = 1)\n",
    "        master['tp53_res_group'] = master.apply(tp53_residual_group, axis = 1)\n",
    "        master['loh_status'] = master.apply(get_loh_nowgd, axis=1)\n",
    "        \n",
    "    # Finally, we merge Genome Instability metrics\n",
    "    gi_path = total_output + 'gi_metrics_{}.pkl'.format(cohort_type)\n",
    "    chr_metrics =  pd.read_pickle(gi_path)\n",
    "    master = pd.merge(master, chr_metrics, on=['Sample_Id'])\n",
    "    \n",
    "    master.to_pickle(total_output + 'master_{}.pkl'.format(cohort_type))\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 262,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yes\n",
      "18706\n",
      "6885\n",
      "CPU times: user 4min 24s, sys: 1.07 s, total: 4min 25s\n",
      "Wall time: 4min 25s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "master = merge_tables(cohort_type=cohort_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 263,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>Overall_Survival_Months</th>\n",
       "      <th>Overall_Survival_Status</th>\n",
       "      <th>MSI_Score</th>\n",
       "      <th>MSI_Type</th>\n",
       "      <th>TMB_Score</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_count</th>\n",
       "      <th>tp53_tcn</th>\n",
       "      <th>tp53_mcn</th>\n",
       "      <th>tp53_lcn</th>\n",
       "      <th>tp53_seg_length</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>tp53_cf</th>\n",
       "      <th>wgd</th>\n",
       "      <th>gene_count</th>\n",
       "      <th>mutation_count</th>\n",
       "      <th>max_vaf</th>\n",
       "      <th>tp53_exp_nb_1</th>\n",
       "      <th>tp53_exp_nb_2</th>\n",
       "      <th>tp53_exp_nb_3</th>\n",
       "      <th>tp53_exp_nb_4</th>\n",
       "      <th>tp53_exp_nb_5</th>\n",
       "      <th>tp53_res_1</th>\n",
       "      <th>tp53_res_2</th>\n",
       "      <th>tp53_res_3</th>\n",
       "      <th>tp53_res_4</th>\n",
       "      <th>tp53_res_5</th>\n",
       "      <th>tp53_vc_group_1</th>\n",
       "      <th>tp53_vc_group_2</th>\n",
       "      <th>tp53_vc_group_3</th>\n",
       "      <th>tp53_vc_group_4</th>\n",
       "      <th>tp53_vc_group_5</th>\n",
       "      <th>tp53_first_group</th>\n",
       "      <th>tp53_group</th>\n",
       "      <th>tp53_res_group</th>\n",
       "      <th>loh_status</th>\n",
       "      <th>chr_affected</th>\n",
       "      <th>chr_loss</th>\n",
       "      <th>chr_gain</th>\n",
       "      <th>chr_cnloh</th>\n",
       "      <th>frac_genome_altered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>P-0034223-T01-IM6</td>\n",
       "      <td>P-0034223</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>63.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.946448</td>\n",
       "      <td>2.241830</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.55</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668592</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0009819-T01-IM5_P-0009819-N01-IM5</td>\n",
       "      <td>P-0009819-T01-IM5</td>\n",
       "      <td>P-0009819</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>72.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.278140</td>\n",
       "      <td>2.681075</td>\n",
       "      <td>23.441</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>1.0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668300</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0027609-T01-IM6_P-0027609-N01-IM6</td>\n",
       "      <td>P-0027609-T01-IM6</td>\n",
       "      <td>P-0027609</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Colon Adenocarcinoma</td>\n",
       "      <td>40.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>2.110732</td>\n",
       "      <td>23.014</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "      <td>P-0027609-T01-IM617_7577548_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.790</td>\n",
       "      <td>0.118574</td>\n",
       "      <td>p.Gly245Ser</td>\n",
       "      <td>245</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18461192</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.790492</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.209508</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hotspot</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>1_WILD_TYPE</td>\n",
       "      <td>tp53_res</td>\n",
       "      <td>False</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0027408-T01-IM6_P-0027408-N01-IM6</td>\n",
       "      <td>P-0027408-T01-IM6</td>\n",
       "      <td>P-0027408</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>67.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.275073</td>\n",
       "      <td>1.811066</td>\n",
       "      <td>22.586</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.27</td>\n",
       "      <td>Stable</td>\n",
       "      <td>17.6</td>\n",
       "      <td>P-0027408-T01-IM617_7578409_CT_TC</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.168901</td>\n",
       "      <td>p.Arg174Glu</td>\n",
       "      <td>174</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26256025</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.275073</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.059141</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.059141</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LOSS</td>\n",
       "      <td>&gt;=1_LOSS</td>\n",
       "      <td>no_tp53_res</td>\n",
       "      <td>False</td>\n",
       "      <td>12.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0006554-T01-IM5_P-0006554-N01-IM5</td>\n",
       "      <td>P-0006554-T01-IM5</td>\n",
       "      <td>P-0006554</td>\n",
       "      <td>Glioma</td>\n",
       "      <td>Anaplastic Oligodendroglioma</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.775152</td>\n",
       "      <td>1.910719</td>\n",
       "      <td>26.170</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.30</td>\n",
       "      <td>Stable</td>\n",
       "      <td>46.2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>40254480</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18701</th>\n",
       "      <td>P-0050747-T01-IM6_P-0050747-N01-IM6</td>\n",
       "      <td>P-0050747-T01-IM6</td>\n",
       "      <td>P-0050747</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>72.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.305687</td>\n",
       "      <td>2.187990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Stable</td>\n",
       "      <td>6.1</td>\n",
       "      <td>P-0050747-T01-IM617_7577570_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.937</td>\n",
       "      <td>0.168975</td>\n",
       "      <td>p.Met237Ile</td>\n",
       "      <td>237</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18461117</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.254038</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.936567</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.063433</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LOSS</td>\n",
       "      <td>&gt;=1_LOSS</td>\n",
       "      <td>no_tp53_res</td>\n",
       "      <td>False</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18702</th>\n",
       "      <td>P-0050745-T01-IM6_P-0050745-N01-IM6</td>\n",
       "      <td>P-0050745-T01-IM6</td>\n",
       "      <td>P-0050745</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.602685</td>\n",
       "      <td>1.808634</td>\n",
       "      <td>1.841</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.85</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>P-0050745-T01-IM617_7577079_C_A</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>0.994</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>p.Glu287Ter</td>\n",
       "      <td>287</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29307621</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.602685</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.993635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006365</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>truncated</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LOSS</td>\n",
       "      <td>&gt;=1_LOSS</td>\n",
       "      <td>no_tp53_res</td>\n",
       "      <td>False</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18703</th>\n",
       "      <td>P-0047736-T01-IM6_P-0047736-N01-IM6</td>\n",
       "      <td>P-0047736-T01-IM6</td>\n",
       "      <td>P-0047736</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Colon Adenocarcinoma</td>\n",
       "      <td>46.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.381915</td>\n",
       "      <td>3.314959</td>\n",
       "      <td>4.405</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.38</td>\n",
       "      <td>Stable</td>\n",
       "      <td>7.9</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>80668548</td>\n",
       "      <td>CNLOH</td>\n",
       "      <td>0.184186</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>cnLOH</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>True</td>\n",
       "      <td>7.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18704</th>\n",
       "      <td>P-0047148-T01-IM6_P-0047148-N01-IM6</td>\n",
       "      <td>P-0047148-T01-IM6</td>\n",
       "      <td>P-0047148</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.480940</td>\n",
       "      <td>2.574443</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.29</td>\n",
       "      <td>Stable</td>\n",
       "      <td>1.8</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668400</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705</th>\n",
       "      <td>P-0046531-T01-IM6_P-0046531-N01-IM6</td>\n",
       "      <td>P-0046531-T01-IM6</td>\n",
       "      <td>P-0046531</td>\n",
       "      <td>Glioma</td>\n",
       "      <td>Anaplastic Oligodendroglioma</td>\n",
       "      <td>49.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.700208</td>\n",
       "      <td>1.981912</td>\n",
       "      <td>5.622</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.26</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668500</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.202</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18706 rows × 79 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0      P-0034223-T01-IM6_P-0034223-N01-IM6  P-0034223-T01-IM6  P-0034223   \n",
       "1      P-0009819-T01-IM5_P-0009819-N01-IM5  P-0009819-T01-IM5  P-0009819   \n",
       "2      P-0027609-T01-IM6_P-0027609-N01-IM6  P-0027609-T01-IM6  P-0027609   \n",
       "3      P-0027408-T01-IM6_P-0027408-N01-IM6  P-0027408-T01-IM6  P-0027408   \n",
       "4      P-0006554-T01-IM5_P-0006554-N01-IM5  P-0006554-T01-IM5  P-0006554   \n",
       "...                                    ...                ...        ...   \n",
       "18701  P-0050747-T01-IM6_P-0050747-N01-IM6  P-0050747-T01-IM6  P-0050747   \n",
       "18702  P-0050745-T01-IM6_P-0050745-N01-IM6  P-0050745-T01-IM6  P-0050745   \n",
       "18703  P-0047736-T01-IM6_P-0047736-N01-IM6  P-0047736-T01-IM6  P-0047736   \n",
       "18704  P-0047148-T01-IM6_P-0047148-N01-IM6  P-0047148-T01-IM6  P-0047148   \n",
       "18705  P-0046531-T01-IM6_P-0046531-N01-IM6  P-0046531-T01-IM6  P-0046531   \n",
       "\n",
       "                      Cancer_Type              Cancer_Type_Detailed  \\\n",
       "0                   Breast Cancer         Invasive Breast Carcinoma   \n",
       "1                 Prostate Cancer           Prostate Adenocarcinoma   \n",
       "2               Colorectal Cancer              Colon Adenocarcinoma   \n",
       "3      Non-Small Cell Lung Cancer        Non-Small Cell Lung Cancer   \n",
       "4                          Glioma      Anaplastic Oligodendroglioma   \n",
       "...                           ...                               ...   \n",
       "18701           Pancreatic Cancer         Pancreatic Adenocarcinoma   \n",
       "18702               Breast Cancer  Breast Invasive Ductal Carcinoma   \n",
       "18703           Colorectal Cancer              Colon Adenocarcinoma   \n",
       "18704             Prostate Cancer           Prostate Adenocarcinoma   \n",
       "18705                      Glioma      Anaplastic Oligodendroglioma   \n",
       "\n",
       "       Patient_Current_Age Sample_Type    purity    ploidy  \\\n",
       "0                     63.0  Metastasis  0.946448  2.241830   \n",
       "1                     72.0     Primary  0.278140  2.681075   \n",
       "2                     40.0  Metastasis  0.300000  2.110732   \n",
       "3                     67.0  Metastasis  0.275073  1.811066   \n",
       "4                     55.0     Primary  0.775152  1.910719   \n",
       "...                    ...         ...       ...       ...   \n",
       "18701                 72.0     Primary  0.305687  2.187990   \n",
       "18702                 68.0     Primary  0.602685  1.808634   \n",
       "18703                 46.0     Primary  0.381915  3.314959   \n",
       "18704                 53.0     Primary  0.480940  2.574443   \n",
       "18705                 49.0  Metastasis  0.700208  1.981912   \n",
       "\n",
       "       Overall_Survival_Months Overall_Survival_Status  MSI_Score MSI_Type  \\\n",
       "0                          NaN                  LIVING       0.55   Stable   \n",
       "1                       23.441                  LIVING       0.00   Stable   \n",
       "2                       23.014                  LIVING       0.00   Stable   \n",
       "3                       22.586                  LIVING       0.27   Stable   \n",
       "4                       26.170                  LIVING       1.30   Stable   \n",
       "...                        ...                     ...        ...      ...   \n",
       "18701                      NaN                  LIVING       0.05   Stable   \n",
       "18702                    1.841                  LIVING       1.85   Stable   \n",
       "18703                    4.405                  LIVING       0.38   Stable   \n",
       "18704                      NaN                  LIVING       0.29   Stable   \n",
       "18705                    5.622                  LIVING       0.26   Stable   \n",
       "\n",
       "       TMB_Score                         tp53_key_1          tp53_vc_1  \\\n",
       "0            5.3                               None               None   \n",
       "1            1.0                               None               None   \n",
       "2            3.5    P-0027609-T01-IM617_7577548_C_T  Missense_Mutation   \n",
       "3           17.6  P-0027408-T01-IM617_7578409_CT_TC  Missense_Mutation   \n",
       "4           46.2                               None               None   \n",
       "...          ...                                ...                ...   \n",
       "18701        6.1    P-0050747-T01-IM617_7577570_C_T  Missense_Mutation   \n",
       "18702        5.3    P-0050745-T01-IM617_7577079_C_A  Nonsense_Mutation   \n",
       "18703        7.9                               None               None   \n",
       "18704        1.8                               None               None   \n",
       "18705        5.3                               None               None   \n",
       "\n",
       "       tp53_ccf_1  tp53_vaf_1 tp53_HGVSp_1 tp53_spot_1 tp53_key_2 tp53_vc_2  \\\n",
       "0             NaN         NaN         None        None       None      None   \n",
       "1             NaN         NaN         None        None       None      None   \n",
       "2           0.790    0.118574  p.Gly245Ser         245       None      None   \n",
       "3           1.000    0.168901  p.Arg174Glu         174       None      None   \n",
       "4             NaN         NaN         None        None       None      None   \n",
       "...           ...         ...          ...         ...        ...       ...   \n",
       "18701       0.937    0.168975  p.Met237Ile         237       None      None   \n",
       "18702       0.994    0.428571  p.Glu287Ter         287       None      None   \n",
       "18703         NaN         NaN         None        None       None      None   \n",
       "18704         NaN         NaN         None        None       None      None   \n",
       "18705         NaN         NaN         None        None       None      None   \n",
       "\n",
       "       tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2 tp53_spot_2 tp53_key_3 tp53_vc_3  \\\n",
       "0             NaN         NaN         None        None       None      None   \n",
       "1             NaN         NaN         None        None       None      None   \n",
       "2             NaN         NaN         None        None       None      None   \n",
       "3             NaN         NaN         None        None       None      None   \n",
       "4             NaN         NaN         None        None       None      None   \n",
       "...           ...         ...          ...         ...        ...       ...   \n",
       "18701         NaN         NaN         None        None       None      None   \n",
       "18702         NaN         NaN         None        None       None      None   \n",
       "18703         NaN         NaN         None        None       None      None   \n",
       "18704         NaN         NaN         None        None       None      None   \n",
       "18705         NaN         NaN         None        None       None      None   \n",
       "\n",
       "       tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3 tp53_spot_3 tp53_key_4 tp53_vc_4  \\\n",
       "0             NaN         NaN         None        None       None      None   \n",
       "1             NaN         NaN         None        None       None      None   \n",
       "2             NaN         NaN         None        None       None      None   \n",
       "3             NaN         NaN         None        None       None      None   \n",
       "4             NaN         NaN         None        None       None      None   \n",
       "...           ...         ...          ...         ...        ...       ...   \n",
       "18701         NaN         NaN         None        None       None      None   \n",
       "18702         NaN         NaN         None        None       None      None   \n",
       "18703         NaN         NaN         None        None       None      None   \n",
       "18704         NaN         NaN         None        None       None      None   \n",
       "18705         NaN         NaN         None        None       None      None   \n",
       "\n",
       "       tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4 tp53_spot_4 tp53_key_5 tp53_vc_5  \\\n",
       "0             NaN         NaN         None        None       None      None   \n",
       "1             NaN         NaN         None        None       None      None   \n",
       "2             NaN         NaN         None        None       None      None   \n",
       "3             NaN         NaN         None        None       None      None   \n",
       "4             NaN         NaN         None        None       None      None   \n",
       "...           ...         ...          ...         ...        ...       ...   \n",
       "18701         NaN         NaN         None        None       None      None   \n",
       "18702         NaN         NaN         None        None       None      None   \n",
       "18703         NaN         NaN         None        None       None      None   \n",
       "18704         NaN         NaN         None        None       None      None   \n",
       "18705         NaN         NaN         None        None       None      None   \n",
       "\n",
       "       tp53_ccf_5  tp53_vaf_5 tp53_HGVSp_5 tp53_spot_5  tp53_count  tp53_tcn  \\\n",
       "0             NaN         NaN         None        None           0         2   \n",
       "1             NaN         NaN         None        None           0         2   \n",
       "2             NaN         NaN         None        None           1         2   \n",
       "3             NaN         NaN         None        None           1         1   \n",
       "4             NaN         NaN         None        None           0         2   \n",
       "...           ...         ...          ...         ...         ...       ...   \n",
       "18701         NaN         NaN         None        None           1         1   \n",
       "18702         NaN         NaN         None        None           1         1   \n",
       "18703         NaN         NaN         None        None           0         2   \n",
       "18704         NaN         NaN         None        None           0         2   \n",
       "18705         NaN         NaN         None        None           0         2   \n",
       "\n",
       "       tp53_mcn  tp53_lcn  tp53_seg_length tp53_cn_state   tp53_cf    wgd  \\\n",
       "0           1.0       1.0         80668592       DIPLOID  1.000000  False   \n",
       "1           1.0       1.0         80668300       DIPLOID  1.000000  False   \n",
       "2           1.0       1.0         18461192       DIPLOID  1.000000  False   \n",
       "3           1.0       0.0         26256025       HETLOSS  0.275073  False   \n",
       "4           1.0       1.0         40254480       DIPLOID  1.000000  False   \n",
       "...         ...       ...              ...           ...       ...    ...   \n",
       "18701       1.0       0.0         18461117       HETLOSS  0.254038  False   \n",
       "18702       1.0       0.0         29307621       HETLOSS  0.602685  False   \n",
       "18703       2.0       0.0         80668548         CNLOH  0.184186  False   \n",
       "18704       1.0       1.0         80668400       DIPLOID  1.000000  False   \n",
       "18705       1.0       1.0         80668500       DIPLOID  1.000000  False   \n",
       "\n",
       "       gene_count  mutation_count  max_vaf  tp53_exp_nb_1  tp53_exp_nb_2  \\\n",
       "0               0               0      NaN            NaN            NaN   \n",
       "1               0               0      NaN            NaN            NaN   \n",
       "2               0               0      NaN       0.790492            NaN   \n",
       "3               0               0      NaN       1.059141            NaN   \n",
       "4               0               0      NaN            NaN            NaN   \n",
       "...           ...             ...      ...            ...            ...   \n",
       "18701           0               0      NaN       0.936567            NaN   \n",
       "18702           0               0      NaN       0.993635            NaN   \n",
       "18703           0               0      NaN            NaN            NaN   \n",
       "18704           0               0      NaN            NaN            NaN   \n",
       "18705           0               0      NaN            NaN            NaN   \n",
       "\n",
       "       tp53_exp_nb_3  tp53_exp_nb_4  tp53_exp_nb_5  tp53_res_1  tp53_res_2  \\\n",
       "0                NaN            NaN            NaN         NaN         NaN   \n",
       "1                NaN            NaN            NaN         NaN         NaN   \n",
       "2                NaN            NaN            NaN    1.209508         NaN   \n",
       "3                NaN            NaN            NaN   -0.059141         NaN   \n",
       "4                NaN            NaN            NaN         NaN         NaN   \n",
       "...              ...            ...            ...         ...         ...   \n",
       "18701            NaN            NaN            NaN    0.063433         NaN   \n",
       "18702            NaN            NaN            NaN    0.006365         NaN   \n",
       "18703            NaN            NaN            NaN         NaN         NaN   \n",
       "18704            NaN            NaN            NaN         NaN         NaN   \n",
       "18705            NaN            NaN            NaN         NaN         NaN   \n",
       "\n",
       "       tp53_res_3  tp53_res_4  tp53_res_5 tp53_vc_group_1 tp53_vc_group_2  \\\n",
       "0             NaN         NaN         NaN            None            None   \n",
       "1             NaN         NaN         NaN            None            None   \n",
       "2             NaN         NaN         NaN         hotspot            None   \n",
       "3             NaN         NaN         NaN        missense            None   \n",
       "4             NaN         NaN         NaN            None            None   \n",
       "...           ...         ...         ...             ...             ...   \n",
       "18701         NaN         NaN         NaN        missense            None   \n",
       "18702         NaN         NaN         NaN       truncated            None   \n",
       "18703         NaN         NaN         NaN            None            None   \n",
       "18704         NaN         NaN         NaN            None            None   \n",
       "18705         NaN         NaN         NaN            None            None   \n",
       "\n",
       "      tp53_vc_group_3 tp53_vc_group_4 tp53_vc_group_5 tp53_first_group  \\\n",
       "0                None            None            None        WILD_TYPE   \n",
       "1                None            None            None        WILD_TYPE   \n",
       "2                None            None            None        WILD_TYPE   \n",
       "3                None            None            None             LOSS   \n",
       "4                None            None            None        WILD_TYPE   \n",
       "...               ...             ...             ...              ...   \n",
       "18701            None            None            None             LOSS   \n",
       "18702            None            None            None             LOSS   \n",
       "18703            None            None            None            cnLOH   \n",
       "18704            None            None            None        WILD_TYPE   \n",
       "18705            None            None            None        WILD_TYPE   \n",
       "\n",
       "        tp53_group tp53_res_group  loh_status  chr_affected  chr_loss  \\\n",
       "0             None           None       False           3.0       2.0   \n",
       "1             None           None       False           4.0       4.0   \n",
       "2      1_WILD_TYPE       tp53_res       False           4.0       0.0   \n",
       "3         >=1_LOSS    no_tp53_res       False          12.0      11.0   \n",
       "4             None           None       False           5.0       4.0   \n",
       "...            ...            ...         ...           ...       ...   \n",
       "18701     >=1_LOSS    no_tp53_res       False           5.0       5.0   \n",
       "18702     >=1_LOSS    no_tp53_res       False           7.0       7.0   \n",
       "18703         None           None        True           7.0       2.0   \n",
       "18704         None           None       False           5.0       2.0   \n",
       "18705         None           None       False           5.0       4.0   \n",
       "\n",
       "       chr_gain  chr_cnloh  frac_genome_altered  \n",
       "0           2.0        0.0                0.080  \n",
       "1           0.0        0.0                0.137  \n",
       "2           4.0        0.0                0.138  \n",
       "3           3.0        0.0                0.444  \n",
       "4           1.0        0.0                0.173  \n",
       "...         ...        ...                  ...  \n",
       "18701       1.0        0.0                0.192  \n",
       "18702       0.0        0.0                0.668  \n",
       "18703       1.0        3.0                0.293  \n",
       "18704       3.0        0.0                0.249  \n",
       "18705       0.0        1.0                0.202  \n",
       "\n",
       "[18706 rows x 79 columns]"
      ]
     },
     "execution_count": 263,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master = pd.read_pickle(total_output + 'master_{}.pkl'.format(cohort_type))\n",
    "master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 234,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tp53_group</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0_HETLOSS</th>\n",
       "      <td>2641</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_WILD_TYPE</th>\n",
       "      <td>1548</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;1muts</th>\n",
       "      <td>517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;=1_LOSS</th>\n",
       "      <td>3841</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;=1_cnLOH</th>\n",
       "      <td>780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOMDEL</th>\n",
       "      <td>289</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             count\n",
       "tp53_group        \n",
       "0_HETLOSS     2641\n",
       "1_WILD_TYPE   1548\n",
       ">1muts         517\n",
       ">=1_LOSS      3841\n",
       ">=1_cnLOH      780\n",
       "HOMDEL         289"
      ]
     },
     "execution_count": 234,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(master, 'tp53_group', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 235,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>AMP</th>\n",
       "      <td>25</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP (BALANCED)</th>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP (LOH)</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH</th>\n",
       "      <td>932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH &amp; GAIN</th>\n",
       "      <td>57</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DIPLOID</th>\n",
       "      <td>10486</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GAIN</th>\n",
       "      <td>374</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HETLOSS</th>\n",
       "      <td>6482</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOMDEL</th>\n",
       "      <td>289</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TETRAPLOID</th>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                count\n",
       "tp53_cn_state        \n",
       "AMP                25\n",
       "AMP (BALANCED)      5\n",
       "AMP (LOH)           1\n",
       "CNLOH             932\n",
       "CNLOH & GAIN       57\n",
       "DIPLOID         10486\n",
       "GAIN              374\n",
       "HETLOSS          6482\n",
       "HOMDEL            289\n",
       "TETRAPLOID         55"
      ]
     },
     "execution_count": 235,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(master, 'tp53_cn_state', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 236,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tp53_count</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>81</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            count\n",
       "tp53_count       \n",
       "0           11821\n",
       "1            6192\n",
       "2             594\n",
       "3              81\n",
       "4              16\n",
       "5               2"
      ]
     },
     "execution_count": 236,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(master, 'tp53_count', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.31216931216931204"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_cohort_wgd[maf_cohort_wgd['Tumor_Sample_Barcode'] == 'P-0036909-T01-IM6']['t_var_freq'].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == 'P-0025956-T01-IM6']\n",
    "selected_cohort.groupby('Entrez_Gene_Id').size().shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "336px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
