{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <span style='font-family:\"Times New Roman\"'> <span styel=''> **MASTER FILE CREATION**\n",
    "\n",
    "*Emile Cohen* \n",
    "    \n",
    "*May 2020*\n",
    "\n",
    "**Goal:** In this Notebook, we create a master file that summarizes all useful information.\n",
    "\n",
    "The Notebook is divided in 4 parts, representing the four parts of our Master file:\n",
    "   \n",
    "* **1. Patient/Sample Information**\n",
    "* **2. TP53 Mutations**\n",
    "* **3. TP53 Copy Numbers**\n",
    "* **4. TP53 Computed Metrics**\n",
    "* **5. Subgroup columns creation**\n",
    "* **6. Merge tables**\n",
    "\n",
    "**NB1:** In each part, you must run the cells from the begining in order to initialize the variables\n",
    "\n",
    "**NB2:** In order to launch the last script (Merge Tables), you have to define the functions in each part.\n",
    "\n",
    "**NB3:** All functions used for the plots are located in utils/custom_tools.py\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 136,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Setup environment... done!\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "<span style=\"color:green\">✅ Working on **mskimpact_env** conda environment.</span>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%run -i '../../../utils/setup_environment.ipy'\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "from scipy.stats import fisher_exact, ranksums, chi2, norm\n",
    "from statsmodels.sandbox.stats.multicomp import multipletests\n",
    "import matplotlib.gridspec as gridspec\n",
    "import pickle\n",
    "\n",
    "data_path = '../../../data/'\n",
    "data_wgd = data_path + 'impact-facets-tp53/processed/wgd/'\n",
    "data_no_wgd = data_path + 'impact-facets-tp53/processed/no_wgd/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 205,
   "metadata": {},
   "outputs": [],
   "source": [
    "# first we load all files for WGD\n",
    "maf_cohort_wgd = pd.read_csv(data_wgd + 'maf_cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_wgd = pd.read_csv(data_wgd + 'cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_wgd = pd.read_csv(data_wgd + 'arm_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_wgd = pd.read_csv(data_wgd + 'gene_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We load all files for non WGD\n",
    "\n",
    "maf_cohort_nowgd = pd.read_csv(data_no_wgd + 'maf_cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_nowgd = pd.read_csv(data_no_wgd + 'cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_nowgd = pd.read_csv(data_no_wgd + 'arm_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_nowgd = pd.read_csv(data_no_wgd + 'gene_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "def removing_germ_na_mut(maf_cohort: pd.DataFrame):\n",
    "    '''\n",
    "    This function is meant to remove non TP53 GERMLINE mutations from the mafs, to have only SOMATIC mut \n",
    "    except for TP53 muts where we allow GERMLINE status.\n",
    "    '''\n",
    "    maf_cohort_final = maf_cohort[maf_cohort['Mutation_Status'] != 'UNKNOWN']\n",
    "\n",
    "    key_list = list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] != 'TP53'][maf_cohort_final['Mutation_Status'] != 'GERMLINE'].sample_mut_key)\n",
    "    key_list = key_list + list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] == 'TP53'].sample_mut_key)\n",
    "    maf_cohort_final = maf_cohort_final[maf_cohort_final.sample_mut_key.isin(key_list)]\n",
    "    \n",
    "    return maf_cohort_final\n",
    "\n",
    "maf_cohort_wgd = removing_germ_na_mut(maf_cohort_wgd)\n",
    "maf_cohort_nowgd = removing_germ_na_mut(maf_cohort_nowgd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load clinical data\n",
    "clinical_data = pd.read_csv(data_path + 'cbioportal/raw/mskimpact_clinical_data-2.tsv', sep= '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filtering the clinical data\n",
    "samples_wgd = list(set(cohort_wgd.tumor_sample))\n",
    "samples_nowgd = list(set(cohort_nowgd.tumor_sample))\n",
    "\n",
    "clinical_wgd = clinical_data[clinical_data['Sample ID'].isin(samples_wgd)]\n",
    "clinical_nowgd = clinical_data[clinical_data['Sample ID'].isin(samples_nowgd)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IMPORTANT: Parameter Definition\n",
    "In this script we can create two master files: one for WGD samples and one for non WGD samples.\n",
    "So this parameter allows to select the type of master you want.\n",
    "On top of that you can specify where you want to store the file you created"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_type = 'no_wgd'\n",
    "\n",
    "\n",
    "\n",
    "output_path = data_path + 'impact-facets-tp53/processed/'\n",
    "\n",
    "total_output = output_path + cohort_type + '/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "# Patient/Sample Information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we focus on clinical information exported from CbioPortal.\n",
    "\n",
    "The following columns are selected:\n",
    "* Sample_Id\n",
    "* Tumor_Id\n",
    "* Patient_Id\n",
    "* Patient Current Age\n",
    "* Cancer_Type\n",
    "* Cancer_Type_Detailed\n",
    "* Sample_Type\n",
    "* purity\n",
    "* ploidy\n",
    "* Overall Survival Status\n",
    "* Overall Survival (Months)\n",
    "* MSI Score\n",
    "* MSI Type\n",
    "* Tumor Mutational Burden"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 206,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_sample_info(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    cohort_filt = cohort[['sample_id', 'tumor_sample', 'patient', 'ploidy']]\n",
    "    clinical_filt = clinical[['Sample ID', 'Patient Current Age', 'Cancer Type', 'Cancer Type Detailed', 'Sample Type',\n",
    "                  'Overall Survival (Months)', 'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "    purity = maf_cohort.drop_duplicates('Tumor_Sample_Barcode')[['Tumor_Sample_Barcode', 'purity']]\n",
    "\n",
    "    # Merging these files\n",
    "    sample_info = pd.merge(cohort_filt, purity, how='left', left_on='tumor_sample', right_on='Tumor_Sample_Barcode')\n",
    "    sample_info =  pd.merge(sample_info, clinical_filt, left_on='tumor_sample', right_on='Sample ID').drop(['Tumor_Sample_Barcode', 'Sample ID'], axis=1)\n",
    "\n",
    "\n",
    "    sample_info = sample_info[['sample_id', 'tumor_sample', 'patient', 'Cancer Type', 'Cancer Type Detailed',\n",
    "                               'Patient Current Age','Sample Type', 'purity','ploidy','Overall Survival (Months)', \n",
    "                               'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "\n",
    "    sample_info.columns = ['Sample_Id', 'Tumor_Id', 'Patient_Id','Cancer_Type', 'Cancer_Type_Detailed', 'Patient_Current_Age',\n",
    "                          'Sample_Type', 'purity', 'ploidy', 'Overall_Survival_Months', 'Overall_Survival_Status',\n",
    "                          'MSI_Score', 'MSI_Type', 'TMB_Score']\n",
    "    \n",
    "    return sample_info\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>Overall_Survival_Months</th>\n",
       "      <th>Overall_Survival_Status</th>\n",
       "      <th>MSI_Score</th>\n",
       "      <th>MSI_Type</th>\n",
       "      <th>TMB_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>P-0025956</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>71.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>3.496971</td>\n",
       "      <td>3.584</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.391316</td>\n",
       "      <td>2.871793</td>\n",
       "      <td>14.137</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Neuroendocrine Carcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.865628</td>\n",
       "      <td>3.115253</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>2.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>3.136841</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.82</td>\n",
       "      <td>Stable</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Colon Adenocarcinoma</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.351778</td>\n",
       "      <td>3.727190</td>\n",
       "      <td>34.060</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.45</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6362</th>\n",
       "      <td>P-0050671-T01-IM6_P-0050671-N01-IM6</td>\n",
       "      <td>P-0050671-T01-IM6</td>\n",
       "      <td>P-0050671</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>34.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.300360</td>\n",
       "      <td>3.776265</td>\n",
       "      <td>2.005</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Stable</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6363</th>\n",
       "      <td>P-0009110-T02-IM6_P-0009110-N01-IM6</td>\n",
       "      <td>P-0009110-T02-IM6</td>\n",
       "      <td>P-0009110</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.239690</td>\n",
       "      <td>3.480349</td>\n",
       "      <td>49.414</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>30.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6364</th>\n",
       "      <td>P-0050668-T01-IM6_P-0050668-N01-IM6</td>\n",
       "      <td>P-0050668-T01-IM6</td>\n",
       "      <td>P-0050668</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Squamous Cell Carcinoma</td>\n",
       "      <td>73.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.211949</td>\n",
       "      <td>3.144416</td>\n",
       "      <td>1.710</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>6.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6365</th>\n",
       "      <td>P-0050675-T01-IM6_P-0050675-N01-IM6</td>\n",
       "      <td>P-0050675-T01-IM6</td>\n",
       "      <td>P-0050675</td>\n",
       "      <td>Cancer of Unknown Primary</td>\n",
       "      <td>Cancer of Unknown Primary</td>\n",
       "      <td>64.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.386871</td>\n",
       "      <td>2.760248</td>\n",
       "      <td>1.973</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.32</td>\n",
       "      <td>Stable</td>\n",
       "      <td>23.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6366</th>\n",
       "      <td>P-0050660-T01-IM6_P-0050660-N01-IM6</td>\n",
       "      <td>P-0050660-T01-IM6</td>\n",
       "      <td>P-0050660</td>\n",
       "      <td>Bladder Cancer</td>\n",
       "      <td>Bladder Urothelial Carcinoma</td>\n",
       "      <td>70.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>3.459433</td>\n",
       "      <td>2.038</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.41</td>\n",
       "      <td>Stable</td>\n",
       "      <td>6.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6367 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6  P-0025956-T01-IM6  P-0025956   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6  P-0036909-T01-IM6  P-0036909   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6  P-0023546-T01-IM6  P-0023546   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6  P-0023546-T02-IM6  P-0023546   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6  P-0018837-T01-IM6  P-0018837   \n",
       "...                                   ...                ...        ...   \n",
       "6362  P-0050671-T01-IM6_P-0050671-N01-IM6  P-0050671-T01-IM6  P-0050671   \n",
       "6363  P-0009110-T02-IM6_P-0009110-N01-IM6  P-0009110-T02-IM6  P-0009110   \n",
       "6364  P-0050668-T01-IM6_P-0050668-N01-IM6  P-0050668-T01-IM6  P-0050668   \n",
       "6365  P-0050675-T01-IM6_P-0050675-N01-IM6  P-0050675-T01-IM6  P-0050675   \n",
       "6366  P-0050660-T01-IM6_P-0050660-N01-IM6  P-0050660-T01-IM6  P-0050660   \n",
       "\n",
       "                     Cancer_Type               Cancer_Type_Detailed  \\\n",
       "0     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "1     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "2                Prostate Cancer  Prostate Neuroendocrine Carcinoma   \n",
       "3                Prostate Cancer            Prostate Adenocarcinoma   \n",
       "4              Colorectal Cancer               Colon Adenocarcinoma   \n",
       "...                          ...                                ...   \n",
       "6362               Breast Cancer          Invasive Breast Carcinoma   \n",
       "6363  Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "6364  Non-Small Cell Lung Cancer       Lung Squamous Cell Carcinoma   \n",
       "6365   Cancer of Unknown Primary          Cancer of Unknown Primary   \n",
       "6366              Bladder Cancer       Bladder Urothelial Carcinoma   \n",
       "\n",
       "      Patient_Current_Age Sample_Type    purity    ploidy  \\\n",
       "0                    71.0     Primary  0.273767  3.496971   \n",
       "1                    47.0  Metastasis  0.391316  2.871793   \n",
       "2                    50.0     Primary  0.865628  3.115253   \n",
       "3                    50.0     Primary  0.312907  3.136841   \n",
       "4                    60.0     Primary  0.351778  3.727190   \n",
       "...                   ...         ...       ...       ...   \n",
       "6362                 34.0  Metastasis  0.300360  3.776265   \n",
       "6363                 57.0  Metastasis  0.239690  3.480349   \n",
       "6364                 73.0     Primary  0.211949  3.144416   \n",
       "6365                 64.0     Primary  0.386871  2.760248   \n",
       "6366                 70.0     Primary  0.683828  3.459433   \n",
       "\n",
       "      Overall_Survival_Months Overall_Survival_Status  MSI_Score MSI_Type  \\\n",
       "0                       3.584                DECEASED       0.00   Stable   \n",
       "1                      14.137                  LIVING       0.37   Stable   \n",
       "2                       4.800                DECEASED       2.37   Stable   \n",
       "3                       4.800                DECEASED       0.82   Stable   \n",
       "4                      34.060                  LIVING       0.45   Stable   \n",
       "...                       ...                     ...        ...      ...   \n",
       "6362                    2.005                  LIVING       0.05   Stable   \n",
       "6363                   49.414                  LIVING       0.00   Stable   \n",
       "6364                    1.710                  LIVING       0.00   Stable   \n",
       "6365                    1.973                  LIVING       1.32   Stable   \n",
       "6366                    2.038                  LIVING       0.41   Stable   \n",
       "\n",
       "      TMB_Score  \n",
       "0           5.3  \n",
       "1           3.5  \n",
       "2           3.5  \n",
       "3           2.6  \n",
       "4           5.3  \n",
       "...         ...  \n",
       "6362        7.0  \n",
       "6363       30.7  \n",
       "6364        6.1  \n",
       "6365       23.7  \n",
       "6366        6.1  \n",
       "\n",
       "[6367 rows x 14 columns]"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info = create_sample_info(cohort=cohort_type)\n",
    "sample_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 190,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>Overall_Survival_Months</th>\n",
       "      <th>Overall_Survival_Status</th>\n",
       "      <th>MSI_Score</th>\n",
       "      <th>MSI_Type</th>\n",
       "      <th>TMB_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>226</th>\n",
       "      <td>P-0013968-T01-IM5_P-0013968-N01-IM5</td>\n",
       "      <td>P-0013968-T01-IM5</td>\n",
       "      <td>P-0013968</td>\n",
       "      <td>Hepatobiliary Cancer</td>\n",
       "      <td>Extrahepatic Cholangiocarcinoma</td>\n",
       "      <td>90.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.831027</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>25.03</td>\n",
       "      <td>Instable</td>\n",
       "      <td>28.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>261</th>\n",
       "      <td>P-0002304-T01-IM3_P-0002304-N01-IM3</td>\n",
       "      <td>P-0002304-T01-IM3</td>\n",
       "      <td>P-0002304</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>76.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.703556</td>\n",
       "      <td>55.529</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.43</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>276</th>\n",
       "      <td>P-0030484-T01-IM6_P-0030484-N01-IM6</td>\n",
       "      <td>P-0030484-T01-IM6</td>\n",
       "      <td>P-0030484</td>\n",
       "      <td>CNS Cancer</td>\n",
       "      <td>Atypical Meningioma</td>\n",
       "      <td>30.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.023616</td>\n",
       "      <td>20.449</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>321</th>\n",
       "      <td>P-0000735-T01-IM3_P-0000735-N01-IM3</td>\n",
       "      <td>P-0000735-T01-IM3</td>\n",
       "      <td>P-0000735</td>\n",
       "      <td>Thymic Tumor</td>\n",
       "      <td>Thymic Carcinoma</td>\n",
       "      <td>80.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.180875</td>\n",
       "      <td>63.090</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.23</td>\n",
       "      <td>Stable</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>336</th>\n",
       "      <td>P-0002625-T01-IM3_P-0002625-N01-IM3</td>\n",
       "      <td>P-0002625-T01-IM3</td>\n",
       "      <td>P-0002625</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.199868</td>\n",
       "      <td>9.929</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.83</td>\n",
       "      <td>Stable</td>\n",
       "      <td>6.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14367</th>\n",
       "      <td>P-0002078-T01-IM3_P-0002078-N01-IM3</td>\n",
       "      <td>P-0002078-T01-IM3</td>\n",
       "      <td>P-0002078</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Mucinous Adenocarcinoma of the Colon and Rectum</td>\n",
       "      <td>21.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.878995</td>\n",
       "      <td>1.216</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15288</th>\n",
       "      <td>P-0047901-T01-IM6_P-0047901-N01-IM6</td>\n",
       "      <td>P-0047901-T01-IM6</td>\n",
       "      <td>P-0047901</td>\n",
       "      <td>Endometrial Cancer</td>\n",
       "      <td>Endometrial Carcinoma</td>\n",
       "      <td>62.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.604396</td>\n",
       "      <td>0.789</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>22.54</td>\n",
       "      <td>Instable</td>\n",
       "      <td>332.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15368</th>\n",
       "      <td>P-0048298-T02-IM6_P-0048298-N01-IM6</td>\n",
       "      <td>P-0048298-T02-IM6</td>\n",
       "      <td>P-0048298</td>\n",
       "      <td>Bladder Cancer</td>\n",
       "      <td>Bladder Urothelial Carcinoma</td>\n",
       "      <td>67.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.831071</td>\n",
       "      <td>3.551</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.80</td>\n",
       "      <td>Stable</td>\n",
       "      <td>411.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15386</th>\n",
       "      <td>P-0047836-T02-IM6_P-0047836-N02-IM6</td>\n",
       "      <td>P-0047836-T02-IM6</td>\n",
       "      <td>P-0047836</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>76.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.602599</td>\n",
       "      <td>3.222</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15924</th>\n",
       "      <td>P-0050213-T01-IM6_P-0050213-N01-IM6</td>\n",
       "      <td>P-0050213-T01-IM6</td>\n",
       "      <td>P-0050213</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Rectal Adenocarcinoma</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2.145242</td>\n",
       "      <td>1.775</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.31</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>218 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Sample_Id           Tumor_Id Patient_Id  \\\n",
       "226    P-0013968-T01-IM5_P-0013968-N01-IM5  P-0013968-T01-IM5  P-0013968   \n",
       "261    P-0002304-T01-IM3_P-0002304-N01-IM3  P-0002304-T01-IM3  P-0002304   \n",
       "276    P-0030484-T01-IM6_P-0030484-N01-IM6  P-0030484-T01-IM6  P-0030484   \n",
       "321    P-0000735-T01-IM3_P-0000735-N01-IM3  P-0000735-T01-IM3  P-0000735   \n",
       "336    P-0002625-T01-IM3_P-0002625-N01-IM3  P-0002625-T01-IM3  P-0002625   \n",
       "...                                    ...                ...        ...   \n",
       "14367  P-0002078-T01-IM3_P-0002078-N01-IM3  P-0002078-T01-IM3  P-0002078   \n",
       "15288  P-0047901-T01-IM6_P-0047901-N01-IM6  P-0047901-T01-IM6  P-0047901   \n",
       "15368  P-0048298-T02-IM6_P-0048298-N01-IM6  P-0048298-T02-IM6  P-0048298   \n",
       "15386  P-0047836-T02-IM6_P-0047836-N02-IM6  P-0047836-T02-IM6  P-0047836   \n",
       "15924  P-0050213-T01-IM6_P-0050213-N01-IM6  P-0050213-T01-IM6  P-0050213   \n",
       "\n",
       "                      Cancer_Type  \\\n",
       "226          Hepatobiliary Cancer   \n",
       "261               Prostate Cancer   \n",
       "276                    CNS Cancer   \n",
       "321                  Thymic Tumor   \n",
       "336                 Breast Cancer   \n",
       "...                           ...   \n",
       "14367           Colorectal Cancer   \n",
       "15288          Endometrial Cancer   \n",
       "15368              Bladder Cancer   \n",
       "15386  Non-Small Cell Lung Cancer   \n",
       "15924           Colorectal Cancer   \n",
       "\n",
       "                                  Cancer_Type_Detailed  Patient_Current_Age  \\\n",
       "226                    Extrahepatic Cholangiocarcinoma                 90.0   \n",
       "261                            Prostate Adenocarcinoma                 76.0   \n",
       "276                                Atypical Meningioma                 30.0   \n",
       "321                                   Thymic Carcinoma                 80.0   \n",
       "336                   Breast Invasive Ductal Carcinoma                 53.0   \n",
       "...                                                ...                  ...   \n",
       "14367  Mucinous Adenocarcinoma of the Colon and Rectum                 21.0   \n",
       "15288                            Endometrial Carcinoma                 62.0   \n",
       "15368                     Bladder Urothelial Carcinoma                 67.0   \n",
       "15386                              Lung Adenocarcinoma                 76.0   \n",
       "15924                            Rectal Adenocarcinoma                 53.0   \n",
       "\n",
       "      Sample_Type  purity    ploidy  Overall_Survival_Months  \\\n",
       "226    Metastasis     NaN  1.831027                      NaN   \n",
       "261    Metastasis     NaN  1.703556                   55.529   \n",
       "276       Primary     NaN  2.023616                   20.449   \n",
       "321    Metastasis     NaN  2.180875                   63.090   \n",
       "336    Metastasis     NaN  2.199868                    9.929   \n",
       "...           ...     ...       ...                      ...   \n",
       "14367  Metastasis     NaN  1.878995                    1.216   \n",
       "15288     Primary     NaN  2.604396                    0.789   \n",
       "15368     Primary     NaN  2.831071                    3.551   \n",
       "15386     Primary     NaN  2.602599                    3.222   \n",
       "15924     Primary     NaN  2.145242                    1.775   \n",
       "\n",
       "      Overall_Survival_Status  MSI_Score  MSI_Type  TMB_Score  \n",
       "226                    LIVING      25.03  Instable       28.5  \n",
       "261                    LIVING       0.43    Stable        5.6  \n",
       "276                    LIVING       0.00    Stable        NaN  \n",
       "321                    LIVING       0.23    Stable        NaN  \n",
       "336                  DECEASED       0.83    Stable        6.7  \n",
       "...                       ...        ...       ...        ...  \n",
       "14367                DECEASED       0.00    Stable        NaN  \n",
       "15288                  LIVING      22.54  Instable      332.7  \n",
       "15368                  LIVING       1.80    Stable      411.6  \n",
       "15386                  LIVING       0.00    Stable        2.6  \n",
       "15924                  LIVING       0.31    Stable        5.3  \n",
       "\n",
       "[218 rows x 14 columns]"
      ]
     },
     "execution_count": 190,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h = sample_info.purity.isna()\n",
    "index = h[h == True].index.tolist()\n",
    "sample_info[sample_info.index.isin(index)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Mutations\n",
    "In this part, we focus on tp53 mutation information.\n",
    "\n",
    "We gather all mutations per sample, and split it into different columns. We have the following columns:\n",
    "* Tumor_Id\t\n",
    "* key_1 (2,3,4,5) --> Mutation key allowing to filter duplicates\n",
    "* vc_1 (2,3,4,5) --> Variant Classification\n",
    "* ccf_1 (2,3,4,5) --> Cancer Cell Fraction of the mutation\n",
    "* vaf_1 (2,3,4,5) --> Variant Allele Frequency of the mutation\n",
    "* HGVSp_1 (2,3,4,5) --> protein change\n",
    "* spot_1 (2,3,4,5) --> Integer that defines the spot of the tp53 mutation\n",
    "* tp53_count --> Number of tp53 mutations of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 191,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def f_(x):\n",
    "    # This function helps us to group mutations together in a single cell per patient\n",
    "    return pd.DataFrame(dict(Tumor_Sample_Barcode = x['Tumor_Sample_Barcode'],  \n",
    "                        muts = \"%s\" % ','.join(x['sample_mut_key_vc_ccf_vaf_hgv_spot'])))\n",
    "\n",
    "def count_tp53_muts(x):\n",
    "    count = 0\n",
    "    for i in range(1,6):\n",
    "        if x['tp53_key_' + str(i)]:\n",
    "            count+= 1\n",
    "    return count\n",
    "\n",
    "# WARNING: THis function needs sample_info to work\n",
    "def create_tp53_muts(cohort:str, sample_info):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        print('yes')\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    '''\n",
    "    This function aims to gather all tp53 mutation characteristics.\n",
    "    For each sample we gather the tp53 mutations and their characteristics for all patients.\n",
    "    '''\n",
    "    # We load the  table created in maf_tp53_creation.ipynb\n",
    "    maf_tp53 = maf_cohort[maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    maf_tp53['mut_spot'] = maf_tp53.HGVSp.str.extract('(\\d+)')\n",
    "\n",
    "    # We select only intresting columns\n",
    "    maf_tp53_filtered = maf_tp53[['Tumor_Sample_Barcode','sample_mut_key', 'Variant_Classification',\\\n",
    "                                        'ccf_expected_copies', 't_var_freq', 'HGVSp','mut_spot' ]]\n",
    "\n",
    "    # Let's Merge mut_key,Variant_classification, CF, CCF, and VAF to gather them\n",
    "    maf_tp53_filtered['sample_mut_key_vc_ccf_vaf_hgv_spot'] = maf_tp53_filtered.apply(lambda x: str(x.sample_mut_key)+'%'+str(x.Variant_Classification)+'%'+str(x.ccf_expected_copies)+'%'+str(x.t_var_freq)+'%'+str(x.HGVSp)+'%'+str(x.mut_spot), axis=1)\n",
    "\n",
    "    # We Select important columns\n",
    "    final = maf_tp53_filtered[['Tumor_Sample_Barcode', 'sample_mut_key_vc_ccf_vaf_hgv_spot']]\n",
    "    # We groupby Patient_Id and apply the function above to group mutations\n",
    "    final = final.groupby(['Tumor_Sample_Barcode'], sort=False).apply(f_)\n",
    "\n",
    "    # We separate the different mutations into 5 different columns (5 is the max number of tp53 mutations in our cohort)\n",
    "    final[['mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5']] = final.muts.str.split(',', expand=True)\n",
    "    #final = final.drop(['mut_key_6'],axis=1)\n",
    "    # Split the columns into mut_key_ and vc_\n",
    "    final[['tp53_key_1','tp53_vc_1','tp53_ccf_1','tp53_vaf_1','tp53_HGVSp_1', 'tp53_spot_1']] = final.mut_key_1.str.split('%', expand=True)\n",
    "    final[['tp53_key_2','tp53_vc_2','tp53_ccf_2','tp53_vaf_2','tp53_HGVSp_2', 'tp53_spot_2']] = final.mut_key_2.str.split('%', expand=True)\n",
    "    final[['tp53_key_3','tp53_vc_3','tp53_ccf_3','tp53_vaf_3','tp53_HGVSp_3', 'tp53_spot_3']] = final.mut_key_3.str.split('%', expand=True)\n",
    "    final[['tp53_key_4','tp53_vc_4','tp53_ccf_4','tp53_vaf_4','tp53_HGVSp_4', 'tp53_spot_4']] = final.mut_key_4.str.split('%', expand=True)\n",
    "    final[['tp53_key_5','tp53_vc_5','tp53_ccf_5','tp53_vaf_5','tp53_HGVSp_5', 'tp53_spot_5']] = final.mut_key_5.str.split('%', expand=True)\n",
    "\n",
    "    # We remove the muts column\n",
    "    final = final.drop(['muts','mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5'], axis=1)\n",
    "\n",
    "    # We remove duplicates\n",
    "    final = final.drop_duplicates('Tumor_Sample_Barcode')\n",
    "\n",
    "    # We add the cohort patients that are not tp53 positive\n",
    "    #First we create a dataframe with all missing samples\n",
    "\n",
    "    cohort_samples = set(sample_info.Tumor_Id)\n",
    "    final_samples = set(final.Tumor_Sample_Barcode)\n",
    "    missing_samp = pd.DataFrame(cohort_samples - final_samples, columns = ['Tumor_Sample_Barcode'])\n",
    "    #Then we append the two datframe\n",
    "    final = final.append(missing_samp)\n",
    "\n",
    "    # We rename the Tumor_Sample_Barcode column to have the same key as in other datframes\n",
    "    final = final.rename(columns={'Tumor_Sample_Barcode': 'Tumor_Id'})\n",
    "\n",
    "    # We add a last column tp53_count that represents the number of tp53 mutations per sample\n",
    "    final = final.where(final.notnull(), None)\n",
    "    final['tp53_count'] = final.apply(count_tp53_muts, axis = 1)\n",
    "\n",
    "    # We change the type of vafs column to float64 instead of strings\n",
    "    final = final.astype({'tp53_vaf_1': 'float64', 'tp53_vaf_2': 'float64', 'tp53_vaf_3': 'float64', 'tp53_vaf_4': 'float64', 'tp53_vaf_5': 'float64',\n",
    "                       'tp53_ccf_1': 'float64', 'tp53_ccf_2': 'float64', 'tp53_ccf_3': 'float64', 'tp53_ccf_4': 'float64', 'tp53_ccf_5': 'float64'})\n",
    "\n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 224,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yes\n",
      "18706\n",
      "6885\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P-0027609-T01-IM6</td>\n",
       "      <td>P-0027609-T01-IM617_7577548_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.79</td>\n",
       "      <td>0.118574</td>\n",
       "      <td>p.Gly245Ser</td>\n",
       "      <td>245</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>P-0027408-T01-IM6</td>\n",
       "      <td>P-0027408-T01-IM617_7578409_CT_TC</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.168901</td>\n",
       "      <td>p.Arg174Glu</td>\n",
       "      <td>174</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>101</th>\n",
       "      <td>P-0025997-T01-IM6</td>\n",
       "      <td>P-0025997-T01-IM617_7578471_G_-</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.912621</td>\n",
       "      <td>p.Gly154AlafsTer16</td>\n",
       "      <td>154</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>P-0036570-T01-IM6</td>\n",
       "      <td>P-0036570-T01-IM617_7578236_A_G</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.718605</td>\n",
       "      <td>p.Tyr205His</td>\n",
       "      <td>205</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>109</th>\n",
       "      <td>P-0036570-T02-IM6</td>\n",
       "      <td>P-0036570-T02-IM617_7577529_A_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.00</td>\n",
       "      <td>0.156652</td>\n",
       "      <td>p.Ile251Asn</td>\n",
       "      <td>251</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11816</th>\n",
       "      <td>P-0018720-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11817</th>\n",
       "      <td>P-0019192-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11818</th>\n",
       "      <td>P-0017107-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11819</th>\n",
       "      <td>P-0018918-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11820</th>\n",
       "      <td>P-0012675-T02-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18706 rows × 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                Tumor_Id                         tp53_key_1  \\\n",
       "8      P-0027609-T01-IM6    P-0027609-T01-IM617_7577548_C_T   \n",
       "18     P-0027408-T01-IM6  P-0027408-T01-IM617_7578409_CT_TC   \n",
       "101    P-0025997-T01-IM6    P-0025997-T01-IM617_7578471_G_-   \n",
       "104    P-0036570-T01-IM6    P-0036570-T01-IM617_7578236_A_G   \n",
       "109    P-0036570-T02-IM6    P-0036570-T02-IM617_7577529_A_T   \n",
       "...                  ...                                ...   \n",
       "11816  P-0018720-T01-IM6                               None   \n",
       "11817  P-0019192-T01-IM6                               None   \n",
       "11818  P-0017107-T01-IM6                               None   \n",
       "11819  P-0018918-T01-IM6                               None   \n",
       "11820  P-0012675-T02-IM6                               None   \n",
       "\n",
       "               tp53_vc_1  tp53_ccf_1  tp53_vaf_1        tp53_HGVSp_1  \\\n",
       "8      Missense_Mutation        0.79    0.118574         p.Gly245Ser   \n",
       "18     Missense_Mutation        1.00    0.168901         p.Arg174Glu   \n",
       "101      Frame_Shift_Del        1.00    0.912621  p.Gly154AlafsTer16   \n",
       "104    Missense_Mutation        1.00    0.718605         p.Tyr205His   \n",
       "109    Missense_Mutation        1.00    0.156652         p.Ile251Asn   \n",
       "...                  ...         ...         ...                 ...   \n",
       "11816               None         NaN         NaN                None   \n",
       "11817               None         NaN         NaN                None   \n",
       "11818               None         NaN         NaN                None   \n",
       "11819               None         NaN         NaN                None   \n",
       "11820               None         NaN         NaN                None   \n",
       "\n",
       "      tp53_spot_1 tp53_key_2 tp53_vc_2  tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2  \\\n",
       "8             245       None      None         NaN         NaN         None   \n",
       "18            174       None      None         NaN         NaN         None   \n",
       "101           154       None      None         NaN         NaN         None   \n",
       "104           205       None      None         NaN         NaN         None   \n",
       "109           251       None      None         NaN         NaN         None   \n",
       "...           ...        ...       ...         ...         ...          ...   \n",
       "11816        None       None      None         NaN         NaN         None   \n",
       "11817        None       None      None         NaN         NaN         None   \n",
       "11818        None       None      None         NaN         NaN         None   \n",
       "11819        None       None      None         NaN         NaN         None   \n",
       "11820        None       None      None         NaN         NaN         None   \n",
       "\n",
       "      tp53_spot_2 tp53_key_3 tp53_vc_3  tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3  \\\n",
       "8            None       None      None         NaN         NaN         None   \n",
       "18           None       None      None         NaN         NaN         None   \n",
       "101          None       None      None         NaN         NaN         None   \n",
       "104          None       None      None         NaN         NaN         None   \n",
       "109          None       None      None         NaN         NaN         None   \n",
       "...           ...        ...       ...         ...         ...          ...   \n",
       "11816        None       None      None         NaN         NaN         None   \n",
       "11817        None       None      None         NaN         NaN         None   \n",
       "11818        None       None      None         NaN         NaN         None   \n",
       "11819        None       None      None         NaN         NaN         None   \n",
       "11820        None       None      None         NaN         NaN         None   \n",
       "\n",
       "      tp53_spot_3 tp53_key_4 tp53_vc_4  tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4  \\\n",
       "8            None       None      None         NaN         NaN         None   \n",
       "18           None       None      None         NaN         NaN         None   \n",
       "101          None       None      None         NaN         NaN         None   \n",
       "104          None       None      None         NaN         NaN         None   \n",
       "109          None       None      None         NaN         NaN         None   \n",
       "...           ...        ...       ...         ...         ...          ...   \n",
       "11816        None       None      None         NaN         NaN         None   \n",
       "11817        None       None      None         NaN         NaN         None   \n",
       "11818        None       None      None         NaN         NaN         None   \n",
       "11819        None       None      None         NaN         NaN         None   \n",
       "11820        None       None      None         NaN         NaN         None   \n",
       "\n",
       "      tp53_spot_4 tp53_key_5 tp53_vc_5  tp53_ccf_5  tp53_vaf_5 tp53_HGVSp_5  \\\n",
       "8            None       None      None         NaN         NaN         None   \n",
       "18           None       None      None         NaN         NaN         None   \n",
       "101          None       None      None         NaN         NaN         None   \n",
       "104          None       None      None         NaN         NaN         None   \n",
       "109          None       None      None         NaN         NaN         None   \n",
       "...           ...        ...       ...         ...         ...          ...   \n",
       "11816        None       None      None         NaN         NaN         None   \n",
       "11817        None       None      None         NaN         NaN         None   \n",
       "11818        None       None      None         NaN         NaN         None   \n",
       "11819        None       None      None         NaN         NaN         None   \n",
       "11820        None       None      None         NaN         NaN         None   \n",
       "\n",
       "      tp53_spot_5  tp53_count  \n",
       "8            None           1  \n",
       "18           None           1  \n",
       "101          None           1  \n",
       "104          None           1  \n",
       "109          None           1  \n",
       "...           ...         ...  \n",
       "11816        None           0  \n",
       "11817        None           0  \n",
       "11818        None           0  \n",
       "11819        None           0  \n",
       "11820        None           0  \n",
       "\n",
       "[18706 rows x 32 columns]"
      ]
     },
     "execution_count": 224,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_tp53_muts(cohort=cohort_type, sample_info=sample_info)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Copy Numbers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we gather the information from gene_level table.\n",
    "We creaste the following columns:\n",
    "* Sample_Id \n",
    "* tcn --> total copy number\n",
    "* mcn --> major copy number\n",
    "* lcn --> lower copy number\n",
    "* seg_length --> length of the segment\n",
    "* cn_state --> copy number state\n",
    "* cf --> Cell fraction of the cn_state\n",
    "* wgd --> Wholde Genome Doubling (True or False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {},
   "outputs": [],
   "source": [
    "# WARNING: THis function needs sample_info to work\n",
    "def create_copy_number_state(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    # We want TP53 locus so we have to filter the gene\n",
    "    gene_level = gene_level[gene_level['gene'] == 'TP53']\n",
    "\n",
    "    gene_level['Tumor_Id'] = gene_level['sample'].str[:17]\n",
    "    gene_level_subset = gene_level[['sample','tcn','mcn','lcn','seg_length','cn_state', 'cf.em']]\n",
    "    \n",
    "    # We rename the cf.em column \n",
    "    gene_level_subset = gene_level_subset.rename(columns={'cf.em': 'tp53_cf', \n",
    "                                                          'sample':'Sample_Id',\n",
    "                                                          'tcn': 'tp53_tcn',\n",
    "                                                          'mcn': 'tp53_mcn',\n",
    "                                                          'lcn': 'tp53_lcn',\n",
    "                                                          'seg_length': 'tp53_seg_length',\n",
    "                                                          'cn_state':'tp53_cn_state'})\n",
    "    \n",
    "    # We add WGD information\n",
    "    wgd = cohort[['sample_id', 'wgd']]\n",
    "    \n",
    "    final = pd.merge(gene_level_subset, wgd, left_on='Sample_Id', right_on='sample_id').drop(['sample_id'], axis=1)\n",
    "    \n",
    "    \n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 226,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 38.5 ms, sys: 5.41 ms, total: 43.9 ms\n",
      "Wall time: 42.4 ms\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>tp53_tcn</th>\n",
       "      <th>tp53_mcn</th>\n",
       "      <th>tp53_lcn</th>\n",
       "      <th>tp53_seg_length</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>tp53_cf</th>\n",
       "      <th>wgd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668592</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0009819-T01-IM5_P-0009819-N01-IM5</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668300</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0027609-T01-IM6_P-0027609-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>18461192</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0027408-T01-IM6_P-0027408-N01-IM6</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26256025</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.275073</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0006554-T01-IM5_P-0006554-N01-IM5</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>40254480</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18702</th>\n",
       "      <td>P-0050747-T01-IM6_P-0050747-N01-IM6</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18461117</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.254038</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18703</th>\n",
       "      <td>P-0050745-T01-IM6_P-0050745-N01-IM6</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29307621</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.602685</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18704</th>\n",
       "      <td>P-0047736-T01-IM6_P-0047736-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>80668548</td>\n",
       "      <td>CNLOH</td>\n",
       "      <td>0.184186</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18705</th>\n",
       "      <td>P-0047148-T01-IM6_P-0047148-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668400</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18706</th>\n",
       "      <td>P-0046531-T01-IM6_P-0046531-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668500</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18707 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Sample_Id  tp53_tcn  tp53_mcn  tp53_lcn  \\\n",
       "0      P-0034223-T01-IM6_P-0034223-N01-IM6         2       1.0       1.0   \n",
       "1      P-0009819-T01-IM5_P-0009819-N01-IM5         2       1.0       1.0   \n",
       "2      P-0027609-T01-IM6_P-0027609-N01-IM6         2       1.0       1.0   \n",
       "3      P-0027408-T01-IM6_P-0027408-N01-IM6         1       1.0       0.0   \n",
       "4      P-0006554-T01-IM5_P-0006554-N01-IM5         2       1.0       1.0   \n",
       "...                                    ...       ...       ...       ...   \n",
       "18702  P-0050747-T01-IM6_P-0050747-N01-IM6         1       1.0       0.0   \n",
       "18703  P-0050745-T01-IM6_P-0050745-N01-IM6         1       1.0       0.0   \n",
       "18704  P-0047736-T01-IM6_P-0047736-N01-IM6         2       2.0       0.0   \n",
       "18705  P-0047148-T01-IM6_P-0047148-N01-IM6         2       1.0       1.0   \n",
       "18706  P-0046531-T01-IM6_P-0046531-N01-IM6         2       1.0       1.0   \n",
       "\n",
       "       tp53_seg_length tp53_cn_state   tp53_cf    wgd  \n",
       "0             80668592       DIPLOID  1.000000  False  \n",
       "1             80668300       DIPLOID  1.000000  False  \n",
       "2             18461192       DIPLOID  1.000000  False  \n",
       "3             26256025       HETLOSS  0.275073  False  \n",
       "4             40254480       DIPLOID  1.000000  False  \n",
       "...                ...           ...       ...    ...  \n",
       "18702         18461117       HETLOSS  0.254038  False  \n",
       "18703         29307621       HETLOSS  0.602685  False  \n",
       "18704         80668548         CNLOH  0.184186  False  \n",
       "18705         80668400       DIPLOID  1.000000  False  \n",
       "18706         80668500       DIPLOID  1.000000  False  \n",
       "\n",
       "[18707 rows x 8 columns]"
      ]
     },
     "execution_count": 226,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time \n",
    "copy_number_info = create_copy_number_state(cohort = cohort_type)\n",
    "copy_number_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computed metrics\n",
    "In this part we define functions to be applied on the master file to compute specific metrics.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes, mutations and max_vaf\n",
    "* *create_gene_count*: count of mutated genes for the given sample\n",
    "* *create_mut_count*: count of mutations for the given sample\n",
    "* *get_max_vaf*: the maximum of Variant Allele Frequency within all mutation of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "metadata": {},
   "outputs": [],
   "source": [
    "# These 2 first functions allow  to count the total number of genes/mutations per sample.\n",
    "def create_gene_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function create the count of genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    \n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "\n",
    "    nb_genes = selected_cohort.groupby('Entrez_Gene_Id').size().shape[0]\n",
    "\n",
    "    return nb_genes\n",
    "\n",
    "def create_mut_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "    \n",
    "    return selected_cohort.shape[0]\n",
    "\n",
    "# The following calculates the max_vaf for a given sample\n",
    "def get_max_vaf(x, maf_cohort):\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    \n",
    "    return selected_cohort['t_var_freq'].max()\n",
    "\n",
    "\n",
    "def create_driver_gene_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function create the count of driver mut genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    \n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['driver'] == True][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "\n",
    "    nb_genes = selected_cohort.groupby('Entrez_Gene_Id').size().shape[0]\n",
    "\n",
    "    return nb_genes\n",
    "\n",
    "def create_driver_mut_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all driver mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['driver'] == True][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "    \n",
    "    return selected_cohort.shape[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genome Instability (frac_genome_aletered, chr_affected by loss, cnkloh, gain )\n",
    "This script is very long to run, so I decide to run it apart and save the results in a pickle fil that we will merge with our master.\n",
    "\n",
    "* For no_wgd: 40 minutes (MacBook Pro 2019)\n",
    "* For wgd: 11 minutes (MacBook Pro 2019)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 194,
   "metadata": {},
   "outputs": [],
   "source": [
    "def condition_CNLOH(x):\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    if x.cn_state in CNLOH:\n",
    "        return 'CNLOH' + x.chr\n",
    "    else:\n",
    "        return 'NO_CNLOH'  + x.chr \n",
    "    \n",
    "def condition_GAIN(x):\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    if x.cn_state in GAIN:\n",
    "        return 'GAIN'+ x.chr\n",
    "    else:\n",
    "        return 'NO_GAIN'+ x.chr\n",
    "\n",
    "def condition_LOSS(x):\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    if x.cn_state in LOSS:\n",
    "        return 'LOSS'+ x.chr\n",
    "    else:\n",
    "        return 'NO_LOSS'+ x.chr\n",
    "\n",
    "def compute_frac_genome(x, arm_level):\n",
    "    Sample_Id = x.Sample_Id\n",
    "    wgd = x.wgd\n",
    "    lookup_table = arm_level[arm_level['sample'] == Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "        \n",
    "    altered_length = lookup_table_altered.cn_length.sum()\n",
    "    total_length = lookup_table.arm_length.sum()\n",
    "    \n",
    "    frac_gen_altered = round(altered_length/total_length,3)\n",
    "    \n",
    "    return frac_gen_altered\n",
    "\n",
    "# Here is the function that allws to compute genome instability columns\n",
    "def chr_computations(x, arm_level):\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    wgd=x.wgd\n",
    "    arm_level_samples = list(set(arm_level['sample']))\n",
    "    \n",
    "    if x.Sample_Id not in arm_level_samples:\n",
    "        return ['NaN','NaN','NaN','NaN', 'NaN']\n",
    "    \n",
    "    lookup_table = arm_level[arm_level['sample'] == x.Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "    lookup_table['state_chr'] = lookup_table['cn_state']+lookup_table['chr']\n",
    "    \n",
    "    # If only DIPLOID or TETRAPLOID\n",
    "    if lookup_table.empty:\n",
    "        return [float(0)]*5\n",
    "    \n",
    "    lookup_table['cnloh_chr'] = lookup_table.apply(condition_CNLOH, axis=1)\n",
    "    lookup_table['loss_chr'] = lookup_table.apply(condition_LOSS, axis=1)\n",
    "    lookup_table['gain_chr'] = lookup_table.apply(condition_GAIN, axis=1)\n",
    "\n",
    "    #chr_affected colum\n",
    "    lookup_table_chr = lookup_table.drop_duplicates(subset=['chr'])\n",
    "    chr_affected = len(lookup_table_chr)\n",
    "    \n",
    "    #chr_loss, chr_gain, chr_cnloh columns\n",
    "    lookup_table_cnloh = lookup_table.drop_duplicates(subset=['cnloh_chr'])['cnloh_chr']\n",
    "    lookup_table_loss = lookup_table.drop_duplicates(subset=['loss_chr'])['loss_chr']\n",
    "    lookup_table_gain = lookup_table.drop_duplicates(subset=['gain_chr'])['gain_chr']\n",
    "\n",
    "    chr_loss = len(lookup_table_loss[lookup_table_loss.str.startswith('LOSS')])\n",
    "    chr_gain = len(lookup_table_gain[lookup_table_gain.str.startswith('GAIN')])\n",
    "    chr_cnloh = len(lookup_table_cnloh[lookup_table_cnloh.str.startswith('CNLOH')])\n",
    "    \n",
    "    #frac_gen_altered column\n",
    "    frac_gen_altered = compute_frac_genome(x, arm_level=arm_level)\n",
    "    \n",
    "    return [chr_affected, chr_loss, chr_gain, chr_cnloh, frac_gen_altered]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You do not need to run the following, excpet if you want to recompute the genome instability metrics.\n",
    "\n",
    "The following cell allows to run the above functions and to save it in a file that we will call in the final merging cell."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a296c923389d418d9ae57f5a9c875a4f",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=1.0, bar_style='info', max=1.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "ef0cf49b8fa345ea8d99699ed6bdb721",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=0.0, max=6367.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "checkpoint 1\n",
      "checkpoint 2\n"
     ]
    }
   ],
   "source": [
    "'''from tqdm import tqdm,tqdm_notebook\n",
    "\n",
    "def compute_genome_instability(cohort:str):\n",
    "    tqdm_notebook().pandas()\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    copy_number_info = create_copy_number_state(cohort = cohort_type)\n",
    "    copy_number_info_ = copy_number_info\n",
    "    copy_number_info_['chr_comput'] = copy_number_info_.progress_apply(chr_computations,arm_level=arm_level, axis=1)\n",
    "    print('checkpoint 1')\n",
    "    copy_number_info[['chr_affected', 'chr_loss', 'chr_gain', 'chr_cnloh', 'frac_genome_altered']] = pd.DataFrame(copy_number_info_.chr_comput.values.tolist(), index= copy_number_info_.index)\n",
    "    print('checkpoint 2')\n",
    "    \n",
    "    return copy_number_info_\n",
    "\n",
    "copy_number_info = compute_genome_instability(cohort=cohort_type)\n",
    "copy_number_info[['Sample_Id','chr_affected', 'chr_loss', 'chr_gain', 'chr_cnloh', 'frac_genome_altered']].to_pickle(total_output + 'gi_metrics_{}.pkl'.format(cohort_type))\n",
    "'''"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TP53 Residual"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following function needs to be called on the complete master file because it needs info from different parts\n",
    "# It computes the expected number of tp53 mutant copies in a cell\n",
    "def create_copies_tp53_muts(master):\n",
    "    master['tp53_exp_nb_1'] = master.apply(lambda x:(x.tp53_vaf_1 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_2'] = master.apply(lambda x:(x.tp53_vaf_2 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_3'] = master.apply(lambda x:(x.tp53_vaf_3 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_4'] = master.apply(lambda x:(x.tp53_vaf_4 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_5'] = master.apply(lambda x:(x.tp53_vaf_5 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    \n",
    "    return master\n",
    "\n",
    "\n",
    "# The following computes the expected number of copies of tp53 residuals \n",
    "def create_tp53_res(master):\n",
    "    master['tp53_res_1'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_1, axis = 1)\n",
    "    master['tp53_res_2'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_2, axis = 1)\n",
    "    master['tp53_res_3'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_3, axis = 1)\n",
    "    master['tp53_res_4'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_4, axis = 1)\n",
    "    master['tp53_res_5'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_5, axis = 1)\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mutation Types Grouping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following functions allow to group the Mutation Types\n",
    "def vc_group_cond_1(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_1 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_1 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_1 in missense: \n",
    "        if x.tp53_spot_1 in ['273','248','175']: return x.tp53_spot_1\n",
    "        elif x.tp53_spot_1 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_2(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_2 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_2 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_2 in missense: \n",
    "        if x.tp53_spot_2 in ['273','248','175']: return x.tp53_spot_2\n",
    "        elif x.tp53_spot_2 in['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'   \n",
    "def vc_group_cond_3(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_3 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_3 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_3 in missense: \n",
    "        if x.tp53_spot_3 in ['273','248','175']: return x.tp53_spot_3\n",
    "        elif x.tp53_spot_3 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense' \n",
    "def vc_group_cond_4(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_4 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_4 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_4 in missense: \n",
    "        if x.tp53_spot_4 in ['273','248','175']: return x.tp53_spot_4\n",
    "        elif x.tp53_spot_4 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_5(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_5 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_5 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_5 in missense: \n",
    "        if x.tp53_spot_5 in ['273','248','175']: return x.tp53_spot_5\n",
    "        elif x.tp53_spot_5 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Non-WGD specific functions\n",
    "\n",
    "* *cn_group_cond*\n",
    "\n",
    "First, we group the different Copy Number States *cn_state* in subgroups, under the column *cn_group*:\n",
    "    * Group 1: cnLOH gathering ['CNLOH']\n",
    "    * Group 2: LOSS gathering ['HETLOSS']\n",
    "    * Group 3: HOMDEL gathering ['HOMDEL']\n",
    "    * Group 4: WILD_TYPE gathering ['DIPLOID', 'TETRAPLOID']\n",
    "    * Group 5: GAIN gathering ['GAIN']\n",
    "    * Group 6: OTHER gathering ['AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']\n",
    "\n",
    "* *mut_cn_group_cond*\n",
    "\n",
    "Based on this first column we define 7 final groups of patients adding the mutational information. These groups will be under the column *tp53_group*.\n",
    "    * Group 1: Samples with 0 tp53 mutations and HETLOSS\n",
    "    * Group 2: Samples with HOMDEL\n",
    "    * Group 3: Samples with 1 tp53 mutation and WILD_TYPE (DIPLOID, LOSS AFTER, TETRAPLOID)\n",
    "    * Group 4: Samples with 1 tp53 mutation or more and LOSS\n",
    "    * Group 5: Samples with 1 tp53 mutation or more and cnLOH\n",
    "    * Group 6: Samples with 2/3/4/5 tp53 mutations and WILD_TYPE or GAIN\n",
    "\n",
    "* *tp53_residual_group*: defines if we have residual tp53 or not based on the expected WT tp53 residual\n",
    "\n",
    "\n",
    "* *get_loh_nowgd*\n",
    "\n",
    "The last function is meant to compute the LOH state"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [],
   "source": [
    "def cn_group_cond(x):\n",
    "    if x.tp53_cn_state in ['CNLOH']:\n",
    "        return 'cnLOH'\n",
    "    if x.tp53_cn_state in ['HETLOSS']:\n",
    "        return 'LOSS'\n",
    "    if x.tp53_cn_state == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_cn_state in ['DIPLOID', 'TETRAPLOID']:\n",
    "        return 'WILD_TYPE'\n",
    "    if x.tp53_cn_state == 'GAIN':\n",
    "        return 'GAIN'\n",
    "    if x.tp53_cn_state in ['AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']:\n",
    "        return 'OTHER'\n",
    "\n",
    "def mut_cn_group_cond(x):\n",
    "    if x.tp53_cn_state == 'HETLOSS' and x.tp53_count == 0:\n",
    "        return '0_HETLOSS'\n",
    "    if x.tp53_first_group == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_first_group == 'WILD_TYPE' and x.tp53_count == 1 :\n",
    "        return '1_WILD_TYPE'\n",
    "    if x.tp53_first_group == 'LOSS' and x.tp53_count >=1:\n",
    "        return '>=1_LOSS'\n",
    "    if x.tp53_first_group == 'cnLOH' and x.tp53_count >=1:\n",
    "        return '>=1_cnLOH'\n",
    "    if (x.tp53_first_group == 'WILD_TYPE' or x.tp53_first_group == 'GAIN') and x.tp53_count > 1:\n",
    "        return '>1muts'\n",
    "    \n",
    "    \n",
    "def tp53_residual_group(x):\n",
    "    if x.tp53_group == '1_WILD_TYPE' or x.tp53_group == '0_HETLOSS':\n",
    "        return 'tp53_res'\n",
    "    if x.tp53_group == 'HOMDEL':\n",
    "        return 'no_tp53_res'\n",
    "    if x.tp53_group == '>=1_LOSS' or x.tp53_group == '>=1_cnLOH':\n",
    "        if (x.tp53_res_1 < 0.5) or (x.tp53_res_2 < 0.5):\n",
    "            return 'no_tp53_res'\n",
    "        else:\n",
    "            if (x.tp53_cf + max(x.tp53_ccf_1, x.tp53_ccf_2, x.tp53_ccf_3, x.tp53_ccf_4, x.tp53_ccf_5)) > 1:\n",
    "                return 'no_tp53_res'\n",
    "            else:\n",
    "                return 'uncertain'\n",
    "    if x.tp53_group == '>1muts':\n",
    "        if (x.tp53_res_1 + x.tp53_res_2 < 2.5):\n",
    "            if (x.tp53_ccf_1 + x.tp53_ccf_2 > 1):\n",
    "                return 'no_tp53_res'\n",
    "            else: \n",
    "                return 'uncertain'\n",
    "           \n",
    "        elif (x.tp53_res_1 + x.tp53_res_2 > 2.5):\n",
    "            return 'tp53_res'\n",
    "        \n",
    "        \n",
    "def get_loh_nowgd(x):\n",
    "    if x.tp53_lcn == 0:\n",
    "        return True\n",
    "    else: return False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WGD specific functions\n",
    "Finally we have functions to compute TP53 allelic state BEFORE WGD (for the WGD cohort only):\n",
    "* *get_bi_nobi* allows to say if the pre WGD allelic state was Bi-Allelic or not\n",
    "* *get_mono* allows to say if, within Not Bi-allelic samples, the state was mono-allelic or with 2 WT alleles\n",
    "* *get_loh_wgd* that computes if the samples is in LOH state or not"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "            NUMBER OF TP53 residual ASSOCIATED WITH ALLELIC STATE BEFORE WGD\n",
    "            \n",
    "            \n",
    "                                     BI-ALLELIC | MONO - ALLELIC | 2WT\n",
    "\n",
    "                LOSS BEFORE :            0      |        1       | -\n",
    "                CNLOH BEFORE & LOSS:     0      |       1,2      | 2\n",
    "                CNLOH BEFORE:            0      |        2       | 3\n",
    "                LOSS AFTER:              -      |       1,2      | 2\n",
    "                DOUBLE LOSS AFTER:       -      |       0,1      | 1\n",
    "                TETRAPLOID:              -      |        2       | 3\n",
    "                CNLOH AFTER:             -      |       1,3      | 3\n",
    "                CNLOH BEFORE & GAIN:     0      |        2,3     | 3,4\n",
    "                \n",
    "\n",
    "                        THRESHOLDS BETWEEN ALLELIC STATES BEFORE WGD\n",
    "                    \n",
    "                                     BI-ALLELIC | MONO - ALLELIC | 2WT\n",
    "\n",
    "                LOSS BEFORE :            <0.4   |        >0.6    | -\n",
    "                CNLOH BEFORE & LOSS:     <0.4   |      0.6< <1.5 | uncertain\n",
    "                CNLOH BEFORE:            <1.5   |   1.5< <2.5    | >2.5\n",
    "                LOSS AFTER:              -      |      <1.5      | uncertain \n",
    "                DOUBLE LOSS AFTER:       -      |       <0.5     | uncertain\n",
    "                TETRAPLOID:              -      |       <2.5     | >2.5\n",
    "                CNLOH AFTER:             -      |       <1.5     | uncertain\n",
    "                CNLOH BEFORE & GAIN:     <1.4   |   1.6< <2.5    | uncertain\n",
    "                \n",
    "    \n",
    "\n",
    " '''\n",
    "\n",
    "\n",
    "# The following functions are for WGD cohort, computing the tp53 allelic state before WGD\n",
    "def get_bi_nobi(x, maf_cohort):\n",
    "    tumor = x.Tumor_Id\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    \n",
    "    #samples without TP53 mutation\n",
    "    if nb_tp53muts == 0:\n",
    "        return 'no_bi'\n",
    "    \n",
    "    # Samples with only one tp53 mutation\n",
    "    if nb_tp53muts == 1:\n",
    "        tp53_res = x.tp53_res_1\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE' or cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr = 0.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr:\n",
    "                return 'bi'\n",
    "            elif tp53_res >= thr:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state in ['LOSS AFTER','DOUBLE LOSS AFTER','TETRAPLOID','CNLOH AFTER']: \n",
    "            return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "            \n",
    "            \n",
    "            \n",
    "    # Samples with 2 tp53 mutations\n",
    "    elif nb_tp53muts == 2:\n",
    "        tp53_res_1 = x.tp53_res_1\n",
    "        tp53_res_2 = x.tp53_res_2\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr - 0.1) or (tp53_res_2 < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr + 0.1) and (tp53_res_2 > thr + 0.1):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (tp53_res_1 < thr_1 - 0.1) or (tp53_res_2 < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (tp53_res_1 < thr_1) or (tp53_res_2 < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if tp53_res_1 < thr_1 - 0.1 or tp53_res_2 < thr_1 - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr_2) and (tp53_res_2 > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "            \n",
    "    # Samples with 3 tp53 mutations\n",
    "    elif nb_tp53muts == 3:\n",
    "        tp53_res_1 = x.tp53_res_1\n",
    "        tp53_res_2 = x.tp53_res_2\n",
    "        tp53_res_3 = x.tp53_res_3\n",
    "        tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "        tp53_res.sort()\n",
    "    \n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (min(tp53_res) < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (max(tp53_res) > thr + 0.1) :\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (min(tp53_res) < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2): # we want that exactly 2 mutation arose before WGD\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (min(tp53_res) < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if min(tp53_res) < thr_1 - 0.1 :\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "    \n",
    "    else: return 'uncertain'\n",
    "\n",
    "\n",
    "# This functions can only be called after get_bi_nobi() has been called on master\n",
    "def get_mono(x, maf_cohort):\n",
    "    tumor = x.Tumor_Id\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    bi_state = x.tp53_bi_state\n",
    "    \n",
    "    if bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    \n",
    "    elif bi_state == 'uncertain':\n",
    "        return 'uncertain_bi'\n",
    "    \n",
    "    \n",
    "    elif bi_state == 'no_bi': # We are already in the  mono/2WT distinction \n",
    "        \n",
    "        if nb_tp53muts == 0:\n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            else: return '2WT'\n",
    "            \n",
    "            \n",
    "        #1 mut samples\n",
    "        if nb_tp53muts == 1:\n",
    "            tp53_res = x.tp53_res_1\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "\n",
    "        \n",
    "        # 2 mut samples   \n",
    "        elif nb_tp53muts == 2:\n",
    "            tp53_res_1 = x.tp53_res_1\n",
    "            tp53_res_2 = x.tp53_res_2\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res_1 < 2.5 or tp53_res_2 < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res_1 > 2.5 and tp53_res_2 > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "            \n",
    "        # 3 mut samples\n",
    "        elif nb_tp53muts == 3:\n",
    "            tp53_res_1 = x.tp53_res_1\n",
    "            tp53_res_2 = x.tp53_res_2\n",
    "            tp53_res_3 = x.tp53_res_3\n",
    "            tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "            tp53_res.sort()\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if min(tp53_res) < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res[0] > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "        \n",
    "        else: return 'uncertain_mono' \n",
    "        \n",
    "def get_loh_wgd(x):\n",
    "    if x.tp53_cn_state in ['LOSS BEFORE', 'CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS BEFORE & AFTER', 'CNLOH BEFORE & GAIN']:\n",
    "        return True\n",
    "    else: return False\n",
    "    \n",
    "    \n",
    "def get_prewgd_tp53_group_1(x):\n",
    "    if x.tp53_bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    elif x.tp53_bi_state == 'no_bi':\n",
    "        return 'tp53_res'\n",
    "    \n",
    "    elif x.tp53_count == 0:\n",
    "        return 'tp53_res'\n",
    "    \n",
    "    \n",
    "def get_prewgd_tp53_group_2(x):\n",
    "    if x.tp53_bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    \n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == True and x.tp53_cn_state == 'LOSS BEFORE':\n",
    "        return 'mono'\n",
    "    elif x.tp53_state_wgd == 'mono':\n",
    "        return 'mono'\n",
    "    \n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == True and x.tp53_cn_state in ['CNLOH BEFORE', 'CNLOH BEFORE & GAIN', 'CNLOH BEFORE & LOSS']:\n",
    "        return 'cnloh_2WT'\n",
    "    \n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == False:\n",
    "        return '2WT'\n",
    "    elif x.tp53_state_wgd == '2WT':\n",
    "        return '2WT'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge and save tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [],
   "source": [
    "def merge_tables(cohort_type: str):\n",
    "    \n",
    "    if cohort_type == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort_type == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    sample_info = create_sample_info(cohort=cohort_type)\n",
    "    tp53_muts = create_tp53_muts(cohort=cohort_type, sample_info=sample_info)\n",
    "    copy_number_info = create_copy_number_state(cohort=cohort_type)\n",
    "    \n",
    "    master = pd.merge(sample_info, tp53_muts, on='Tumor_Id')\n",
    "    master = pd.merge(master, copy_number_info, on='Sample_Id')\n",
    "    \n",
    "    # Now we compute the columns\n",
    "    master['gene_count'] = master.apply(create_gene_count, maf_cohort=maf_cohort, axis=1)\n",
    "    master['mutation_count'] = master.apply(create_mut_count,maf_cohort=maf_cohort, axis=1)\n",
    "    master['driver_gene_count'] = master.apply(create_driver_gene_count, maf_cohort=maf_cohort, axis=1)\n",
    "    master['driver_mutation_count'] = master.apply(create_driver_mut_count,maf_cohort=maf_cohort, axis=1)\n",
    "    master['max_vaf'] = master.apply(get_max_vaf,maf_cohort=maf_cohort, axis=1)\n",
    "    \n",
    "    # Tp53 residual\n",
    "    master = create_copies_tp53_muts(master)\n",
    "    master = create_tp53_res(master)\n",
    "    \n",
    "    # Mutation Type Grouping\n",
    "    master['tp53_vc_group_1'] = master.apply(vc_group_cond_1, axis = 1)\n",
    "    master['tp53_vc_group_2'] = master.apply(vc_group_cond_2, axis = 1)\n",
    "    master['tp53_vc_group_3'] = master.apply(vc_group_cond_3, axis = 1)\n",
    "    master['tp53_vc_group_4'] = master.apply(vc_group_cond_4, axis = 1)\n",
    "    master['tp53_vc_group_5'] = master.apply(vc_group_cond_5, axis = 1)\n",
    "    \n",
    "    \n",
    "    if cohort_type == 'wgd':\n",
    "        # These groups are BEFORE WGD !!!\n",
    "        master['tp53_bi_state'] = master.apply(get_bi_nobi, maf_cohort=maf_cohort, axis=1)\n",
    "        master['tp53_state_wgd'] = master.apply(get_mono, maf_cohort=maf_cohort,axis=1)\n",
    "        # This is after WGD\n",
    "        master['tp53_loh_status'] = master.apply(get_loh_wgd, axis=1)\n",
    "        \n",
    "        # Pre WGD Allelic groups\n",
    "        master['prewgd_tp53_group_1'] = master.apply(get_prewgd_tp53_group_1, axis=1)\n",
    "        master['prewgd_tp53_group_2'] = master.apply(get_prewgd_tp53_group_2, axis=1)\n",
    "        \n",
    "    if cohort_type == 'no_wgd':\n",
    "        master['tp53_first_group'] = master.apply(cn_group_cond, axis = 1)\n",
    "        master['tp53_group'] = master.apply(mut_cn_group_cond, axis = 1)\n",
    "        master['tp53_res_group'] = master.apply(tp53_residual_group, axis = 1)\n",
    "        master['tp53_loh_status'] = master.apply(get_loh_nowgd, axis=1)\n",
    "        \n",
    "    # Finally, we merge Genome Instability metrics\n",
    "    gi_path = total_output + 'gi_metrics_{}.pkl'.format(cohort_type)\n",
    "    chr_metrics =  pd.read_pickle(gi_path)\n",
    "    master = pd.merge(master, chr_metrics, on=['Sample_Id'])\n",
    "    \n",
    "    master.to_pickle(total_output + 'master_{}.pkl'.format(cohort_type))\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "yes\n",
      "CPU times: user 19min 8s, sys: 2.38 s, total: 19min 11s\n",
      "Wall time: 19min 12s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "master = merge_tables(cohort_type=cohort_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>Overall_Survival_Months</th>\n",
       "      <th>Overall_Survival_Status</th>\n",
       "      <th>MSI_Score</th>\n",
       "      <th>MSI_Type</th>\n",
       "      <th>TMB_Score</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_count</th>\n",
       "      <th>tp53_tcn</th>\n",
       "      <th>tp53_mcn</th>\n",
       "      <th>tp53_lcn</th>\n",
       "      <th>tp53_seg_length</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>tp53_cf</th>\n",
       "      <th>wgd</th>\n",
       "      <th>gene_count</th>\n",
       "      <th>mutation_count</th>\n",
       "      <th>driver_gene_count</th>\n",
       "      <th>driver_mutation_count</th>\n",
       "      <th>max_vaf</th>\n",
       "      <th>tp53_exp_nb_1</th>\n",
       "      <th>tp53_exp_nb_2</th>\n",
       "      <th>tp53_exp_nb_3</th>\n",
       "      <th>tp53_exp_nb_4</th>\n",
       "      <th>tp53_exp_nb_5</th>\n",
       "      <th>tp53_res_1</th>\n",
       "      <th>tp53_res_2</th>\n",
       "      <th>tp53_res_3</th>\n",
       "      <th>tp53_res_4</th>\n",
       "      <th>tp53_res_5</th>\n",
       "      <th>tp53_vc_group_1</th>\n",
       "      <th>tp53_vc_group_2</th>\n",
       "      <th>tp53_vc_group_3</th>\n",
       "      <th>tp53_vc_group_4</th>\n",
       "      <th>tp53_vc_group_5</th>\n",
       "      <th>tp53_first_group</th>\n",
       "      <th>tp53_group</th>\n",
       "      <th>tp53_res_group</th>\n",
       "      <th>tp53_loh_status</th>\n",
       "      <th>chr_affected</th>\n",
       "      <th>chr_loss</th>\n",
       "      <th>chr_gain</th>\n",
       "      <th>chr_cnloh</th>\n",
       "      <th>frac_genome_altered</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>P-0034223-T01-IM6</td>\n",
       "      <td>P-0034223</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>63.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.946448</td>\n",
       "      <td>2.241830</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.55</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668592</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>0.901899</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.080</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0009819-T01-IM5_P-0009819-N01-IM5</td>\n",
       "      <td>P-0009819-T01-IM5</td>\n",
       "      <td>P-0009819</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>72.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.278140</td>\n",
       "      <td>2.681075</td>\n",
       "      <td>23.441</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>1.0</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668300</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.148014</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0027408-T01-IM6_P-0027408-N01-IM6</td>\n",
       "      <td>P-0027408-T01-IM6</td>\n",
       "      <td>P-0027408</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>67.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.275073</td>\n",
       "      <td>1.811066</td>\n",
       "      <td>22.586</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.27</td>\n",
       "      <td>Stable</td>\n",
       "      <td>17.6</td>\n",
       "      <td>P-0027408-T01-IM6-17_7578409_CT_TC</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.168901</td>\n",
       "      <td>p.Arg174Glu</td>\n",
       "      <td>174</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>26256025</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.275073</td>\n",
       "      <td>False</td>\n",
       "      <td>17</td>\n",
       "      <td>19</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "      <td>0.192475</td>\n",
       "      <td>1.059141</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.059141</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LOSS</td>\n",
       "      <td>&gt;=1_LOSS</td>\n",
       "      <td>no_tp53_res</td>\n",
       "      <td>True</td>\n",
       "      <td>12.0</td>\n",
       "      <td>11.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.444</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0006554-T01-IM5_P-0006554-N01-IM5</td>\n",
       "      <td>P-0006554-T01-IM5</td>\n",
       "      <td>P-0006554</td>\n",
       "      <td>Glioma</td>\n",
       "      <td>Anaplastic Oligodendroglioma</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.775152</td>\n",
       "      <td>1.910719</td>\n",
       "      <td>26.170</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.30</td>\n",
       "      <td>Stable</td>\n",
       "      <td>46.2</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>40254480</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>39</td>\n",
       "      <td>47</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>0.706897</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0036980-T01-IM6_P-0036980-N01-IM6</td>\n",
       "      <td>P-0036980-T01-IM6</td>\n",
       "      <td>P-0036980</td>\n",
       "      <td>Hepatobiliary Cancer</td>\n",
       "      <td>Intrahepatic Cholangiocarcinoma</td>\n",
       "      <td>54.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.845207</td>\n",
       "      <td>2.267573</td>\n",
       "      <td>9.501</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.45</td>\n",
       "      <td>Stable</td>\n",
       "      <td>4.4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668381</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0.814815</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>4.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16029</th>\n",
       "      <td>P-0050741-T01-IM6_P-0050741-N01-IM6</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>P-0050741</td>\n",
       "      <td>Small Cell Lung Cancer</td>\n",
       "      <td>Small Cell Lung Cancer</td>\n",
       "      <td>46.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.833058</td>\n",
       "      <td>2.006039</td>\n",
       "      <td>1.940</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.22</td>\n",
       "      <td>Stable</td>\n",
       "      <td>7.0</td>\n",
       "      <td>P-0050741-T01-IM6-17_7578394_T_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.757801</td>\n",
       "      <td>p.His179Leu</td>\n",
       "      <td>179</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21694879</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.824066</td>\n",
       "      <td>False</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0.757801</td>\n",
       "      <td>1.061523</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.061523</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LOSS</td>\n",
       "      <td>&gt;=1_LOSS</td>\n",
       "      <td>no_tp53_res</td>\n",
       "      <td>True</td>\n",
       "      <td>6.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.222</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16030</th>\n",
       "      <td>P-0050747-T01-IM6_P-0050747-N01-IM6</td>\n",
       "      <td>P-0050747-T01-IM6</td>\n",
       "      <td>P-0050747</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>72.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.305687</td>\n",
       "      <td>2.187990</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Stable</td>\n",
       "      <td>6.1</td>\n",
       "      <td>P-0050747-T01-IM6-17_7577570_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.937</td>\n",
       "      <td>0.168975</td>\n",
       "      <td>p.Met237Ile</td>\n",
       "      <td>237</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18461117</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.254038</td>\n",
       "      <td>False</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "      <td>0.270471</td>\n",
       "      <td>0.936567</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.063433</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>missense</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LOSS</td>\n",
       "      <td>&gt;=1_LOSS</td>\n",
       "      <td>no_tp53_res</td>\n",
       "      <td>True</td>\n",
       "      <td>5.0</td>\n",
       "      <td>5.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16031</th>\n",
       "      <td>P-0050745-T01-IM6_P-0050745-N01-IM6</td>\n",
       "      <td>P-0050745-T01-IM6</td>\n",
       "      <td>P-0050745</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.602685</td>\n",
       "      <td>1.808634</td>\n",
       "      <td>1.841</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.85</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>P-0050745-T01-IM6-17_7577079_C_A</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>0.994</td>\n",
       "      <td>0.428571</td>\n",
       "      <td>p.Glu287Ter</td>\n",
       "      <td>287</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>29307621</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>0.602685</td>\n",
       "      <td>False</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.451400</td>\n",
       "      <td>0.993635</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.006365</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>truncated</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>LOSS</td>\n",
       "      <td>&gt;=1_LOSS</td>\n",
       "      <td>no_tp53_res</td>\n",
       "      <td>True</td>\n",
       "      <td>7.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.668</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16032</th>\n",
       "      <td>P-0047148-T01-IM6_P-0047148-N01-IM6</td>\n",
       "      <td>P-0047148-T01-IM6</td>\n",
       "      <td>P-0047148</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.480940</td>\n",
       "      <td>2.574443</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.29</td>\n",
       "      <td>Stable</td>\n",
       "      <td>1.8</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668400</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0.228625</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.249</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16033</th>\n",
       "      <td>P-0046531-T01-IM6_P-0046531-N01-IM6</td>\n",
       "      <td>P-0046531-T01-IM6</td>\n",
       "      <td>P-0046531</td>\n",
       "      <td>Glioma</td>\n",
       "      <td>Anaplastic Oligodendroglioma</td>\n",
       "      <td>49.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.700208</td>\n",
       "      <td>1.981912</td>\n",
       "      <td>5.622</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.26</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668500</td>\n",
       "      <td>DIPLOID</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>False</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>0.582822</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>WILD_TYPE</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>False</td>\n",
       "      <td>5.0</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.202</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16034 rows × 81 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0      P-0034223-T01-IM6_P-0034223-N01-IM6  P-0034223-T01-IM6  P-0034223   \n",
       "1      P-0009819-T01-IM5_P-0009819-N01-IM5  P-0009819-T01-IM5  P-0009819   \n",
       "2      P-0027408-T01-IM6_P-0027408-N01-IM6  P-0027408-T01-IM6  P-0027408   \n",
       "3      P-0006554-T01-IM5_P-0006554-N01-IM5  P-0006554-T01-IM5  P-0006554   \n",
       "4      P-0036980-T01-IM6_P-0036980-N01-IM6  P-0036980-T01-IM6  P-0036980   \n",
       "...                                    ...                ...        ...   \n",
       "16029  P-0050741-T01-IM6_P-0050741-N01-IM6  P-0050741-T01-IM6  P-0050741   \n",
       "16030  P-0050747-T01-IM6_P-0050747-N01-IM6  P-0050747-T01-IM6  P-0050747   \n",
       "16031  P-0050745-T01-IM6_P-0050745-N01-IM6  P-0050745-T01-IM6  P-0050745   \n",
       "16032  P-0047148-T01-IM6_P-0047148-N01-IM6  P-0047148-T01-IM6  P-0047148   \n",
       "16033  P-0046531-T01-IM6_P-0046531-N01-IM6  P-0046531-T01-IM6  P-0046531   \n",
       "\n",
       "                      Cancer_Type              Cancer_Type_Detailed  \\\n",
       "0                   Breast Cancer         Invasive Breast Carcinoma   \n",
       "1                 Prostate Cancer           Prostate Adenocarcinoma   \n",
       "2      Non-Small Cell Lung Cancer        Non-Small Cell Lung Cancer   \n",
       "3                          Glioma      Anaplastic Oligodendroglioma   \n",
       "4            Hepatobiliary Cancer   Intrahepatic Cholangiocarcinoma   \n",
       "...                           ...                               ...   \n",
       "16029      Small Cell Lung Cancer            Small Cell Lung Cancer   \n",
       "16030           Pancreatic Cancer         Pancreatic Adenocarcinoma   \n",
       "16031               Breast Cancer  Breast Invasive Ductal Carcinoma   \n",
       "16032             Prostate Cancer           Prostate Adenocarcinoma   \n",
       "16033                      Glioma      Anaplastic Oligodendroglioma   \n",
       "\n",
       "       Patient_Current_Age Sample_Type    purity    ploidy  \\\n",
       "0                     63.0  Metastasis  0.946448  2.241830   \n",
       "1                     72.0     Primary  0.278140  2.681075   \n",
       "2                     67.0  Metastasis  0.275073  1.811066   \n",
       "3                     55.0     Primary  0.775152  1.910719   \n",
       "4                     54.0     Primary  0.845207  2.267573   \n",
       "...                    ...         ...       ...       ...   \n",
       "16029                 46.0  Metastasis  0.833058  2.006039   \n",
       "16030                 72.0     Primary  0.305687  2.187990   \n",
       "16031                 68.0     Primary  0.602685  1.808634   \n",
       "16032                 53.0     Primary  0.480940  2.574443   \n",
       "16033                 49.0  Metastasis  0.700208  1.981912   \n",
       "\n",
       "       Overall_Survival_Months Overall_Survival_Status  MSI_Score MSI_Type  \\\n",
       "0                          NaN                  LIVING       0.55   Stable   \n",
       "1                       23.441                  LIVING       0.00   Stable   \n",
       "2                       22.586                  LIVING       0.27   Stable   \n",
       "3                       26.170                  LIVING       1.30   Stable   \n",
       "4                        9.501                  LIVING       0.45   Stable   \n",
       "...                        ...                     ...        ...      ...   \n",
       "16029                    1.940                  LIVING       0.22   Stable   \n",
       "16030                      NaN                  LIVING       0.05   Stable   \n",
       "16031                    1.841                  LIVING       1.85   Stable   \n",
       "16032                      NaN                  LIVING       0.29   Stable   \n",
       "16033                    5.622                  LIVING       0.26   Stable   \n",
       "\n",
       "       TMB_Score                          tp53_key_1          tp53_vc_1  \\\n",
       "0            5.3                                None               None   \n",
       "1            1.0                                None               None   \n",
       "2           17.6  P-0027408-T01-IM6-17_7578409_CT_TC  Missense_Mutation   \n",
       "3           46.2                                None               None   \n",
       "4            4.4                                None               None   \n",
       "...          ...                                 ...                ...   \n",
       "16029        7.0    P-0050741-T01-IM6-17_7578394_T_A  Missense_Mutation   \n",
       "16030        6.1    P-0050747-T01-IM6-17_7577570_C_T  Missense_Mutation   \n",
       "16031        5.3    P-0050745-T01-IM6-17_7577079_C_A  Nonsense_Mutation   \n",
       "16032        1.8                                None               None   \n",
       "16033        5.3                                None               None   \n",
       "\n",
       "       tp53_ccf_1  tp53_vaf_1 tp53_HGVSp_1 tp53_spot_1 tp53_key_2 tp53_vc_2  \\\n",
       "0             NaN         NaN         None        None       None      None   \n",
       "1             NaN         NaN         None        None       None      None   \n",
       "2           1.000    0.168901  p.Arg174Glu         174       None      None   \n",
       "3             NaN         NaN         None        None       None      None   \n",
       "4             NaN         NaN         None        None       None      None   \n",
       "...           ...         ...          ...         ...        ...       ...   \n",
       "16029       1.000    0.757801  p.His179Leu         179       None      None   \n",
       "16030       0.937    0.168975  p.Met237Ile         237       None      None   \n",
       "16031       0.994    0.428571  p.Glu287Ter         287       None      None   \n",
       "16032         NaN         NaN         None        None       None      None   \n",
       "16033         NaN         NaN         None        None       None      None   \n",
       "\n",
       "       tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2 tp53_spot_2 tp53_key_3 tp53_vc_3  \\\n",
       "0             NaN         NaN         None        None       None      None   \n",
       "1             NaN         NaN         None        None       None      None   \n",
       "2             NaN         NaN         None        None       None      None   \n",
       "3             NaN         NaN         None        None       None      None   \n",
       "4             NaN         NaN         None        None       None      None   \n",
       "...           ...         ...          ...         ...        ...       ...   \n",
       "16029         NaN         NaN         None        None       None      None   \n",
       "16030         NaN         NaN         None        None       None      None   \n",
       "16031         NaN         NaN         None        None       None      None   \n",
       "16032         NaN         NaN         None        None       None      None   \n",
       "16033         NaN         NaN         None        None       None      None   \n",
       "\n",
       "       tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3 tp53_spot_3 tp53_key_4 tp53_vc_4  \\\n",
       "0             NaN         NaN         None        None       None      None   \n",
       "1             NaN         NaN         None        None       None      None   \n",
       "2             NaN         NaN         None        None       None      None   \n",
       "3             NaN         NaN         None        None       None      None   \n",
       "4             NaN         NaN         None        None       None      None   \n",
       "...           ...         ...          ...         ...        ...       ...   \n",
       "16029         NaN         NaN         None        None       None      None   \n",
       "16030         NaN         NaN         None        None       None      None   \n",
       "16031         NaN         NaN         None        None       None      None   \n",
       "16032         NaN         NaN         None        None       None      None   \n",
       "16033         NaN         NaN         None        None       None      None   \n",
       "\n",
       "       tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4 tp53_spot_4 tp53_key_5 tp53_vc_5  \\\n",
       "0             NaN         NaN         None        None       None      None   \n",
       "1             NaN         NaN         None        None       None      None   \n",
       "2             NaN         NaN         None        None       None      None   \n",
       "3             NaN         NaN         None        None       None      None   \n",
       "4             NaN         NaN         None        None       None      None   \n",
       "...           ...         ...          ...         ...        ...       ...   \n",
       "16029         NaN         NaN         None        None       None      None   \n",
       "16030         NaN         NaN         None        None       None      None   \n",
       "16031         NaN         NaN         None        None       None      None   \n",
       "16032         NaN         NaN         None        None       None      None   \n",
       "16033         NaN         NaN         None        None       None      None   \n",
       "\n",
       "       tp53_ccf_5  tp53_vaf_5 tp53_HGVSp_5 tp53_spot_5  tp53_count  tp53_tcn  \\\n",
       "0             NaN         NaN         None        None           0         2   \n",
       "1             NaN         NaN         None        None           0         2   \n",
       "2             NaN         NaN         None        None           1         1   \n",
       "3             NaN         NaN         None        None           0         2   \n",
       "4             NaN         NaN         None        None           0         2   \n",
       "...           ...         ...          ...         ...         ...       ...   \n",
       "16029         NaN         NaN         None        None           1         1   \n",
       "16030         NaN         NaN         None        None           1         1   \n",
       "16031         NaN         NaN         None        None           1         1   \n",
       "16032         NaN         NaN         None        None           0         2   \n",
       "16033         NaN         NaN         None        None           0         2   \n",
       "\n",
       "       tp53_mcn  tp53_lcn  tp53_seg_length tp53_cn_state   tp53_cf    wgd  \\\n",
       "0           1.0       1.0         80668592       DIPLOID  1.000000  False   \n",
       "1           1.0       1.0         80668300       DIPLOID  1.000000  False   \n",
       "2           1.0       0.0         26256025       HETLOSS  0.275073  False   \n",
       "3           1.0       1.0         40254480       DIPLOID  1.000000  False   \n",
       "4           1.0       1.0         80668381       DIPLOID  1.000000  False   \n",
       "...         ...       ...              ...           ...       ...    ...   \n",
       "16029       1.0       0.0         21694879       HETLOSS  0.824066  False   \n",
       "16030       1.0       0.0         18461117       HETLOSS  0.254038  False   \n",
       "16031       1.0       0.0         29307621       HETLOSS  0.602685  False   \n",
       "16032       1.0       1.0         80668400       DIPLOID  1.000000  False   \n",
       "16033       1.0       1.0         80668500       DIPLOID  1.000000  False   \n",
       "\n",
       "       gene_count  mutation_count  driver_gene_count  driver_mutation_count  \\\n",
       "0               6               6                  5                      5   \n",
       "1               1               1                  0                      0   \n",
       "2              17              19                  4                      4   \n",
       "3              39              47                  6                      6   \n",
       "4               5               5                  3                      3   \n",
       "...           ...             ...                ...                    ...   \n",
       "16029           7               7                  2                      2   \n",
       "16030           5               5                  3                      3   \n",
       "16031           5               5                  0                      0   \n",
       "16032           2               2                  1                      1   \n",
       "16033           6               6                  5                      5   \n",
       "\n",
       "        max_vaf  tp53_exp_nb_1  tp53_exp_nb_2  tp53_exp_nb_3  tp53_exp_nb_4  \\\n",
       "0      0.901899            NaN            NaN            NaN            NaN   \n",
       "1      0.148014            NaN            NaN            NaN            NaN   \n",
       "2      0.192475       1.059141            NaN            NaN            NaN   \n",
       "3      0.706897            NaN            NaN            NaN            NaN   \n",
       "4      0.814815            NaN            NaN            NaN            NaN   \n",
       "...         ...            ...            ...            ...            ...   \n",
       "16029  0.757801       1.061523            NaN            NaN            NaN   \n",
       "16030  0.270471       0.936567            NaN            NaN            NaN   \n",
       "16031  0.451400       0.993635            NaN            NaN            NaN   \n",
       "16032  0.228625            NaN            NaN            NaN            NaN   \n",
       "16033  0.582822            NaN            NaN            NaN            NaN   \n",
       "\n",
       "       tp53_exp_nb_5  tp53_res_1  tp53_res_2  tp53_res_3  tp53_res_4  \\\n",
       "0                NaN         NaN         NaN         NaN         NaN   \n",
       "1                NaN         NaN         NaN         NaN         NaN   \n",
       "2                NaN   -0.059141         NaN         NaN         NaN   \n",
       "3                NaN         NaN         NaN         NaN         NaN   \n",
       "4                NaN         NaN         NaN         NaN         NaN   \n",
       "...              ...         ...         ...         ...         ...   \n",
       "16029            NaN   -0.061523         NaN         NaN         NaN   \n",
       "16030            NaN    0.063433         NaN         NaN         NaN   \n",
       "16031            NaN    0.006365         NaN         NaN         NaN   \n",
       "16032            NaN         NaN         NaN         NaN         NaN   \n",
       "16033            NaN         NaN         NaN         NaN         NaN   \n",
       "\n",
       "       tp53_res_5 tp53_vc_group_1 tp53_vc_group_2 tp53_vc_group_3  \\\n",
       "0             NaN            None            None            None   \n",
       "1             NaN            None            None            None   \n",
       "2             NaN        missense            None            None   \n",
       "3             NaN            None            None            None   \n",
       "4             NaN            None            None            None   \n",
       "...           ...             ...             ...             ...   \n",
       "16029         NaN        missense            None            None   \n",
       "16030         NaN        missense            None            None   \n",
       "16031         NaN       truncated            None            None   \n",
       "16032         NaN            None            None            None   \n",
       "16033         NaN            None            None            None   \n",
       "\n",
       "      tp53_vc_group_4 tp53_vc_group_5 tp53_first_group tp53_group  \\\n",
       "0                None            None        WILD_TYPE       None   \n",
       "1                None            None        WILD_TYPE       None   \n",
       "2                None            None             LOSS   >=1_LOSS   \n",
       "3                None            None        WILD_TYPE       None   \n",
       "4                None            None        WILD_TYPE       None   \n",
       "...               ...             ...              ...        ...   \n",
       "16029            None            None             LOSS   >=1_LOSS   \n",
       "16030            None            None             LOSS   >=1_LOSS   \n",
       "16031            None            None             LOSS   >=1_LOSS   \n",
       "16032            None            None        WILD_TYPE       None   \n",
       "16033            None            None        WILD_TYPE       None   \n",
       "\n",
       "      tp53_res_group  tp53_loh_status  chr_affected  chr_loss  chr_gain  \\\n",
       "0               None            False           3.0       2.0       2.0   \n",
       "1               None            False           4.0       4.0       0.0   \n",
       "2        no_tp53_res             True          12.0      11.0       3.0   \n",
       "3               None            False           5.0       4.0       1.0   \n",
       "4               None            False           4.0       4.0       1.0   \n",
       "...              ...              ...           ...       ...       ...   \n",
       "16029    no_tp53_res             True           6.0       5.0       2.0   \n",
       "16030    no_tp53_res             True           5.0       5.0       1.0   \n",
       "16031    no_tp53_res             True           7.0       7.0       0.0   \n",
       "16032           None            False           5.0       2.0       3.0   \n",
       "16033           None            False           5.0       4.0       0.0   \n",
       "\n",
       "       chr_cnloh  frac_genome_altered  \n",
       "0            0.0                0.080  \n",
       "1            0.0                0.137  \n",
       "2            0.0                0.444  \n",
       "3            0.0                0.173  \n",
       "4            0.0                0.187  \n",
       "...          ...                  ...  \n",
       "16029        0.0                0.222  \n",
       "16030        0.0                0.192  \n",
       "16031        0.0                0.668  \n",
       "16032        0.0                0.249  \n",
       "16033        1.0                0.202  \n",
       "\n",
       "[16034 rows x 81 columns]"
      ]
     },
     "execution_count": 221,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master = pd.read_pickle(total_output + 'master_{}.pkl'.format(cohort_type))\n",
    "master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>dbSNP_RS</th>\n",
       "      <th>dbSNP_Val_Status</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Match_Norm_Seq_Allele1</th>\n",
       "      <th>Match_Norm_Seq_Allele2</th>\n",
       "      <th>Tumor_Validation_Allele1</th>\n",
       "      <th>Tumor_Validation_Allele2</th>\n",
       "      <th>Match_Norm_Validation_Allele1</th>\n",
       "      <th>Match_Norm_Validation_Allele2</th>\n",
       "      <th>Verification_Status</th>\n",
       "      <th>Validation_Status</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>Sequencing_Phase</th>\n",
       "      <th>Sequence_Source</th>\n",
       "      <th>Validation_Method</th>\n",
       "      <th>Score</th>\n",
       "      <th>BAM_File</th>\n",
       "      <th>Sequencer</th>\n",
       "      <th>Tumor_Sample_UUID</th>\n",
       "      <th>Matched_Norm_Sample_UUID</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Exon_Number</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>n_depth</th>\n",
       "      <th>n_ref_count</th>\n",
       "      <th>n_alt_count</th>\n",
       "      <th>all_effects</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>ALLELE_NUM</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND_VEP</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>RefSeq</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>GMAF</th>\n",
       "      <th>AFR_MAF</th>\n",
       "      <th>AMR_MAF</th>\n",
       "      <th>ASN_MAF</th>\n",
       "      <th>EAS_MAF</th>\n",
       "      <th>EUR_MAF</th>\n",
       "      <th>SAS_MAF</th>\n",
       "      <th>AA_MAF</th>\n",
       "      <th>EA_MAF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>PICK</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>TSL</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>MINIMISED</th>\n",
       "      <th>ExAC_AF</th>\n",
       "      <th>ExAC_AF_AFR</th>\n",
       "      <th>ExAC_AF_AMR</th>\n",
       "      <th>ExAC_AF_EAS</th>\n",
       "      <th>ExAC_AF_FIN</th>\n",
       "      <th>ExAC_AF_NFE</th>\n",
       "      <th>ExAC_AF_OTH</th>\n",
       "      <th>ExAC_AF_SAS</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>flanking_bps</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_qual</th>\n",
       "      <th>ExAC_AF_Adj</th>\n",
       "      <th>ExAC_AC_AN_Adj</th>\n",
       "      <th>ExAC_AC_AN</th>\n",
       "      <th>ExAC_AC_AN_AFR</th>\n",
       "      <th>ExAC_AC_AN_AMR</th>\n",
       "      <th>ExAC_AC_AN_EAS</th>\n",
       "      <th>ExAC_AC_AN_FIN</th>\n",
       "      <th>ExAC_AC_AN_NFE</th>\n",
       "      <th>ExAC_AC_AN_OTH</th>\n",
       "      <th>ExAC_AC_AN_SAS</th>\n",
       "      <th>ExAC_FILTER</th>\n",
       "      <th>Caller</th>\n",
       "      <th>is-a-hotspot</th>\n",
       "      <th>is-a-3d-hotspot</th>\n",
       "      <th>mutation_effect</th>\n",
       "      <th>oncogenic</th>\n",
       "      <th>LEVEL_1</th>\n",
       "      <th>LEVEL_2A</th>\n",
       "      <th>LEVEL_2B</th>\n",
       "      <th>LEVEL_3A</th>\n",
       "      <th>LEVEL_3B</th>\n",
       "      <th>LEVEL_4</th>\n",
       "      <th>LEVEL_R1</th>\n",
       "      <th>LEVEL_R2</th>\n",
       "      <th>LEVEL_R3</th>\n",
       "      <th>Highest_level</th>\n",
       "      <th>citations</th>\n",
       "      <th>driver</th>\n",
       "      <th>tcn</th>\n",
       "      <th>lcn</th>\n",
       "      <th>cf</th>\n",
       "      <th>purity</th>\n",
       "      <th>t_var_freq</th>\n",
       "      <th>expected_alt_copies</th>\n",
       "      <th>ccf_Mcopies</th>\n",
       "      <th>ccf_Mcopies_lower</th>\n",
       "      <th>ccf_Mcopies_upper</th>\n",
       "      <th>ccf_Mcopies_prob95</th>\n",
       "      <th>ccf_Mcopies_prob90</th>\n",
       "      <th>ccf_1copy</th>\n",
       "      <th>ccf_1copy_lower</th>\n",
       "      <th>ccf_1copy_upper</th>\n",
       "      <th>ccf_1copy_prob95</th>\n",
       "      <th>ccf_1copy_prob90</th>\n",
       "      <th>ccf_expected_copies</th>\n",
       "      <th>ccf_expected_copies_lower</th>\n",
       "      <th>ccf_expected_copies_upper</th>\n",
       "      <th>ccf_expected_copies_prob95</th>\n",
       "      <th>ccf_expected_copies_prob90</th>\n",
       "      <th>facets_fit</th>\n",
       "      <th>reviewer_set_purity</th>\n",
       "      <th>use_only_purity_run</th>\n",
       "      <th>use_edited_cncf</th>\n",
       "      <th>cncf_file_used</th>\n",
       "      <th>mut_key</th>\n",
       "      <th>sample_mut_key</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>147876</th>\n",
       "      <td>EPHB1</td>\n",
       "      <td>2047</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>3</td>\n",
       "      <td>134670483</td>\n",
       "      <td>134670483</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.394G&gt;T</td>\n",
       "      <td>p.Ala132Ser</td>\n",
       "      <td>p.A132S</td>\n",
       "      <td>ENST00000398015</td>\n",
       "      <td>3/16</td>\n",
       "      <td>811.0</td>\n",
       "      <td>511.0</td>\n",
       "      <td>300.0</td>\n",
       "      <td>384</td>\n",
       "      <td>384</td>\n",
       "      <td>0</td>\n",
       "      <td>EPHB1,missense_variant,p.Ala132Ser,ENST00000398015,NM_004441.4;EPHB1,missense_variant,p.Ala110Ser,ENST00000460895,;EPHB1,missense_variant,p.Ala110Ser,ENST00000474732,;EPHB1,downstream_gene_variant,,ENST00000473867,;EPHB1,downstream_gene_variant,,ENST00000497173,;EPHB1,non_coding_transcript_exon_variant,,ENST00000488154,;EPHB1,missense_variant,p.Ala132Ser,ENST00000482618,;</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000154928</td>\n",
       "      <td>ENST00000398015</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>764/4672</td>\n",
       "      <td>394/2955</td>\n",
       "      <td>132/984</td>\n",
       "      <td>A/S</td>\n",
       "      <td>Gcc/Tcc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>EPHB1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>3392.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS46921.1</td>\n",
       "      <td>ENSP00000381097</td>\n",
       "      <td>P54762</td>\n",
       "      <td>C9K090,C9JB88,C9J6S4,C9J466,B3KTB2</td>\n",
       "      <td>UPI000012A07E</td>\n",
       "      <td>NM_004441.4</td>\n",
       "      <td>tolerated(0.74)</td>\n",
       "      <td>benign(0.012)</td>\n",
       "      <td>3/16</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROSITE_profiles:PS51550,hmmpanther:PTHR24416,hmmpanther:PTHR24416:SF173,Pfam_domain:PF01404,Gene3D:2.60.120.260,PIRSF_domain:PIRSF000666,SMART_domains:SM00615,Superfamily_domains:SSF49785</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.833058</td>\n",
       "      <td>0.369914</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.888</td>\n",
       "      <td>0.840</td>\n",
       "      <td>0.937</td>\n",
       "      <td>0.065682</td>\n",
       "      <td>0.392463</td>\n",
       "      <td>0.888</td>\n",
       "      <td>0.840</td>\n",
       "      <td>0.937</td>\n",
       "      <td>0.065682</td>\n",
       "      <td>0.392463</td>\n",
       "      <td>0.888</td>\n",
       "      <td>0.840</td>\n",
       "      <td>0.937</td>\n",
       "      <td>0.065682</td>\n",
       "      <td>0.392463</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>3_134670483_G_T</td>\n",
       "      <td>P-0050741-T01-IM6-3_134670483_G_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147877</th>\n",
       "      <td>ROS1</td>\n",
       "      <td>6098</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>6</td>\n",
       "      <td>117715500</td>\n",
       "      <td>117715500</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.989T&gt;C</td>\n",
       "      <td>p.Val330Ala</td>\n",
       "      <td>p.V330A</td>\n",
       "      <td>ENST00000368508</td>\n",
       "      <td>10/43</td>\n",
       "      <td>219.0</td>\n",
       "      <td>116.0</td>\n",
       "      <td>103.0</td>\n",
       "      <td>255</td>\n",
       "      <td>255</td>\n",
       "      <td>0</td>\n",
       "      <td>ROS1,missense_variant,p.Val330Ala,ENST00000368508,NM_002944.2;ROS1,missense_variant,p.Val339Ala,ENST00000368507,;GOPC,intron_variant,,ENST00000467125,;</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000047936</td>\n",
       "      <td>ENST00000368508</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1188/7435</td>\n",
       "      <td>989/7044</td>\n",
       "      <td>330/2347</td>\n",
       "      <td>V/A</td>\n",
       "      <td>gTt/gCt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>ROS1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>10261.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS5116.1</td>\n",
       "      <td>ENSP00000357494</td>\n",
       "      <td>P08922</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UPI000013D467</td>\n",
       "      <td>NM_002944.2</td>\n",
       "      <td>tolerated(0.14)</td>\n",
       "      <td>benign(0.024)</td>\n",
       "      <td>10/43</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.120.10.30,hmmpanther:PTHR24416,hmmpanther:PTHR24416:SF304,Superfamily_domains:SSF63825</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>AAC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.833058</td>\n",
       "      <td>0.470320</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.969</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.776482</td>\n",
       "      <td>0.965241</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.969</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.776482</td>\n",
       "      <td>0.965241</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.969</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.776482</td>\n",
       "      <td>0.965241</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>6_117715500_A_G</td>\n",
       "      <td>P-0050741-T01-IM6-6_117715500_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147878</th>\n",
       "      <td>PIK3CG</td>\n",
       "      <td>5294</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>7</td>\n",
       "      <td>106509775</td>\n",
       "      <td>106509775</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1769C&gt;T</td>\n",
       "      <td>p.Pro590Leu</td>\n",
       "      <td>p.P590L</td>\n",
       "      <td>ENST00000359195</td>\n",
       "      <td>2/11</td>\n",
       "      <td>663.0</td>\n",
       "      <td>388.0</td>\n",
       "      <td>275.0</td>\n",
       "      <td>430</td>\n",
       "      <td>429</td>\n",
       "      <td>1</td>\n",
       "      <td>PIK3CG,missense_variant,p.Pro590Leu,ENST00000359195,NM_002649.2,NM_001282427.1;PIK3CG,missense_variant,p.Pro590Leu,ENST00000496166,NM_001282426.1;PIK3CG,missense_variant,p.Pro590Leu,ENST00000440650,;PIK3CG,intron_variant,,ENST00000473541,;</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000105851</td>\n",
       "      <td>ENST00000359195</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2079/5377</td>\n",
       "      <td>1769/3309</td>\n",
       "      <td>590/1102</td>\n",
       "      <td>P/L</td>\n",
       "      <td>cCt/cTt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PIK3CG</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>8978.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS5739.1</td>\n",
       "      <td>ENSP00000352121</td>\n",
       "      <td>P48736</td>\n",
       "      <td>Q24M88,E9PDN7</td>\n",
       "      <td>UPI00000746B8</td>\n",
       "      <td>NM_002649.2,NM_001282427.1</td>\n",
       "      <td>deleterious(0.05)</td>\n",
       "      <td>probably_damaging(0.998)</td>\n",
       "      <td>2/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:1.25.40.70,Pfam_domain:PF00613,PROSITE_profiles:PS51545,hmmpanther:PTHR10048,hmmpanther:PTHR10048:SF34,SMART_domains:SM00145,Superfamily_domains:SSF48371</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.00011</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>1/106208</td>\n",
       "      <td>1/106210</td>\n",
       "      <td>1/9064</td>\n",
       "      <td>0/11216</td>\n",
       "      <td>0/7866</td>\n",
       "      <td>0/6614</td>\n",
       "      <td>0/54346</td>\n",
       "      <td>0/694</td>\n",
       "      <td>0/16408</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.833058</td>\n",
       "      <td>0.414781</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.996</td>\n",
       "      <td>0.942</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.714943</td>\n",
       "      <td>0.969240</td>\n",
       "      <td>0.996</td>\n",
       "      <td>0.942</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.714943</td>\n",
       "      <td>0.969240</td>\n",
       "      <td>0.996</td>\n",
       "      <td>0.942</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.714943</td>\n",
       "      <td>0.969240</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>7_106509775_C_T</td>\n",
       "      <td>P-0050741-T01-IM6-7_106509775_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147879</th>\n",
       "      <td>SMO</td>\n",
       "      <td>6608</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>7</td>\n",
       "      <td>128846073</td>\n",
       "      <td>128846073</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1003C&gt;T</td>\n",
       "      <td>p.Leu335Phe</td>\n",
       "      <td>p.L335F</td>\n",
       "      <td>ENST00000249373</td>\n",
       "      <td>5/12</td>\n",
       "      <td>1111.0</td>\n",
       "      <td>660.0</td>\n",
       "      <td>451.0</td>\n",
       "      <td>542</td>\n",
       "      <td>542</td>\n",
       "      <td>0</td>\n",
       "      <td>SMO,missense_variant,p.Leu335Phe,ENST00000249373,NM_005631.4;RP11-286H14.8,downstream_gene_variant,,ENST00000466717,;SMO,missense_variant,p.Leu29Phe,ENST00000462420,;SMO,upstream_gene_variant,,ENST00000495998,;SMO,upstream_gene_variant,,ENST00000475779,;</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000128602</td>\n",
       "      <td>ENST00000249373</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1283/3738</td>\n",
       "      <td>1003/2364</td>\n",
       "      <td>335/787</td>\n",
       "      <td>L/F</td>\n",
       "      <td>Ctc/Ttc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SMO</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11119.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS5811.1</td>\n",
       "      <td>ENSP00000249373</td>\n",
       "      <td>Q99835</td>\n",
       "      <td>A4D1K5</td>\n",
       "      <td>UPI0000050447</td>\n",
       "      <td>NM_005631.4</td>\n",
       "      <td>tolerated(0.16)</td>\n",
       "      <td>possibly_damaging(0.646)</td>\n",
       "      <td>5/12</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Transmembrane_helices:TMhelix,PROSITE_profiles:PS50261,hmmpanther:PTHR11309,hmmpanther:PTHR11309:SF35,Pfam_domain:PF01534,Prints_domain:PR00489</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.833058</td>\n",
       "      <td>0.405941</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.975</td>\n",
       "      <td>0.933</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.691664</td>\n",
       "      <td>0.979469</td>\n",
       "      <td>0.975</td>\n",
       "      <td>0.933</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.691664</td>\n",
       "      <td>0.979469</td>\n",
       "      <td>0.975</td>\n",
       "      <td>0.933</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.691664</td>\n",
       "      <td>0.979469</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>7_128846073_C_T</td>\n",
       "      <td>P-0050741-T01-IM6-7_128846073_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147880</th>\n",
       "      <td>PIK3C2G</td>\n",
       "      <td>5288</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>12</td>\n",
       "      <td>18715643</td>\n",
       "      <td>18715643</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.3474A&gt;G</td>\n",
       "      <td>p.Ile1158Met</td>\n",
       "      <td>p.I1158M</td>\n",
       "      <td>ENST00000266497</td>\n",
       "      <td>25/31</td>\n",
       "      <td>276.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>106.0</td>\n",
       "      <td>219</td>\n",
       "      <td>219</td>\n",
       "      <td>0</td>\n",
       "      <td>PIK3C2G,missense_variant,p.Ile1158Met,ENST00000433979,NM_004570.4;PIK3C2G,missense_variant,p.Ile1158Met,ENST00000266497,;PIK3C2G,missense_variant,p.Ile1199Met,ENST00000538779,NM_001288772.1;PIK3C2G,3_prime_UTR_variant,,ENST00000546003,NM_001288774.1;</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000139144</td>\n",
       "      <td>ENST00000266497</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>3512/4754</td>\n",
       "      <td>3474/4338</td>\n",
       "      <td>1158/1445</td>\n",
       "      <td>I/M</td>\n",
       "      <td>atA/atG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PIK3C2G</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>8973.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS44839.1</td>\n",
       "      <td>ENSP00000266497</td>\n",
       "      <td>O75747</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UPI000013D6EF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.999)</td>\n",
       "      <td>25/31</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:1.10.1070.11,PROSITE_profiles:PS50290,hmmpanther:PTHR10048,hmmpanther:PTHR10048:SF29,SMART_domains:SM00146,Superfamily_domains:SSF56112</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TAA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.105024</td>\n",
       "      <td>0.833058</td>\n",
       "      <td>0.384058</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>12_18715643_A_G</td>\n",
       "      <td>P-0050741-T01-IM6-12_18715643_A_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147881</th>\n",
       "      <td>TBX3</td>\n",
       "      <td>6926</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>12</td>\n",
       "      <td>115112079</td>\n",
       "      <td>115112079</td>\n",
       "      <td>+</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1661C&gt;A</td>\n",
       "      <td>p.Ser554Ter</td>\n",
       "      <td>p.S554*</td>\n",
       "      <td>ENST00000257566</td>\n",
       "      <td>7/8</td>\n",
       "      <td>1354.0</td>\n",
       "      <td>811.0</td>\n",
       "      <td>543.0</td>\n",
       "      <td>599</td>\n",
       "      <td>598</td>\n",
       "      <td>1</td>\n",
       "      <td>TBX3,stop_gained,p.Ser534Ter,ENST00000349155,NM_005996.3;TBX3,stop_gained,p.Ser554Ter,ENST00000257566,NM_016569.3;TBX3,downstream_gene_variant,,ENST00000548503,;,regulatory_region_variant,,ENSR00000077728,;</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000135111</td>\n",
       "      <td>ENST00000257566</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>2051/4208</td>\n",
       "      <td>1661/2232</td>\n",
       "      <td>554/743</td>\n",
       "      <td>S/*</td>\n",
       "      <td>tCg/tAg</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TBX3</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11602.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS9176.1</td>\n",
       "      <td>ENSP00000257566</td>\n",
       "      <td>O15119</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UPI0000136AA0</td>\n",
       "      <td>NM_016569.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7/8</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hmmpanther:PTHR11267,hmmpanther:PTHR11267:SF91</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VQSRTrancheSNP99.60to99.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>11689487;26579496</td>\n",
       "      <td>True</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.833058</td>\n",
       "      <td>0.401034</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.963</td>\n",
       "      <td>0.925</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.617843</td>\n",
       "      <td>0.974354</td>\n",
       "      <td>0.963</td>\n",
       "      <td>0.925</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.617843</td>\n",
       "      <td>0.974354</td>\n",
       "      <td>0.963</td>\n",
       "      <td>0.925</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.617843</td>\n",
       "      <td>0.974354</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>12_115112079_G_T</td>\n",
       "      <td>P-0050741-T01-IM6-12_115112079_G_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147882</th>\n",
       "      <td>RB1</td>\n",
       "      <td>5925</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>13</td>\n",
       "      <td>48951162</td>\n",
       "      <td>48951162</td>\n",
       "      <td>+</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1324G&gt;T</td>\n",
       "      <td>p.Gly442Ter</td>\n",
       "      <td>p.G442*</td>\n",
       "      <td>ENST00000267163</td>\n",
       "      <td>13/27</td>\n",
       "      <td>211.0</td>\n",
       "      <td>53.0</td>\n",
       "      <td>158.0</td>\n",
       "      <td>254</td>\n",
       "      <td>254</td>\n",
       "      <td>0</td>\n",
       "      <td>RB1,stop_gained,p.Gly442Ter,ENST00000267163,NM_000321.2;</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000139687</td>\n",
       "      <td>ENST00000267163</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1462/4840</td>\n",
       "      <td>1324/2787</td>\n",
       "      <td>442/928</td>\n",
       "      <td>G/*</td>\n",
       "      <td>Gga/Tga</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>RB1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>9884.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS31973.1</td>\n",
       "      <td>ENSP00000267163</td>\n",
       "      <td>P06400</td>\n",
       "      <td>Q7KZ61,Q7KZ60,Q6LE82,Q6LE81,Q6LE80,Q6LE79,Q6LE77,Q6LE65,Q6DUF0,P78495,B4DRB3</td>\n",
       "      <td>UPI0000001C79</td>\n",
       "      <td>NM_000321.2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>13/27</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hmmpanther:PTHR13742,Pfam_domain:PF01858,Gene3D:1.10.472.10,Superfamily_domains:SSF47954</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TGG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25621664;30206110;22293180;22205104</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.833058</td>\n",
       "      <td>0.833058</td>\n",
       "      <td>0.748815</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.977</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.904269</td>\n",
       "      <td>0.996372</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.977</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.904269</td>\n",
       "      <td>0.996372</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.977</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.904269</td>\n",
       "      <td>0.996372</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>13_48951162_G_T</td>\n",
       "      <td>P-0050741-T01-IM6-13_48951162_G_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>147883</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578394</td>\n",
       "      <td>7578394</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.536A&gt;T</td>\n",
       "      <td>p.His179Leu</td>\n",
       "      <td>p.H179L</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>5/11</td>\n",
       "      <td>673.0</td>\n",
       "      <td>163.0</td>\n",
       "      <td>510.0</td>\n",
       "      <td>479</td>\n",
       "      <td>479</td>\n",
       "      <td>0</td>\n",
       "      <td>TP53,missense_variant,p.His179Leu,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.His179Leu,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.His179Leu,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.His179Leu,ENST00000445888,;TP53,missense_variant,p.His179Leu,ENST00000359597,;TP53,missense_variant,p.His179Leu,ENST00000413465,;TP53,missense_variant,p.His47Leu,ENST00000509690,;TP53,missense_variant,p.His86Leu,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,EN...</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>726/2579</td>\n",
       "      <td>536/1182</td>\n",
       "      <td>179/393</td>\n",
       "      <td>H/L</td>\n",
       "      <td>cAt/cTt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>tolerated(0.12)</td>\n",
       "      <td>probably_damaging(1)</td>\n",
       "      <td>5/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ATG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23150443</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.824066</td>\n",
       "      <td>0.833058</td>\n",
       "      <td>0.757801</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.993</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.999502</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.993</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.999502</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.993</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.999502</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>17_7578394_T_A</td>\n",
       "      <td>P-0050741-T01-IM6-17_7578394_T_A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  \\\n",
       "147876       EPHB1            2047  MSKCC     GRCh37          3   \n",
       "147877        ROS1            6098  MSKCC     GRCh37          6   \n",
       "147878      PIK3CG            5294  MSKCC     GRCh37          7   \n",
       "147879         SMO            6608  MSKCC     GRCh37          7   \n",
       "147880     PIK3C2G            5288  MSKCC     GRCh37         12   \n",
       "147881        TBX3            6926  MSKCC     GRCh37         12   \n",
       "147882         RB1            5925  MSKCC     GRCh37         13   \n",
       "147883        TP53            7157  MSKCC     GRCh37         17   \n",
       "\n",
       "        Start_Position  End_Position Strand Variant_Classification  \\\n",
       "147876       134670483     134670483      +      Missense_Mutation   \n",
       "147877       117715500     117715500      +      Missense_Mutation   \n",
       "147878       106509775     106509775      +      Missense_Mutation   \n",
       "147879       128846073     128846073      +      Missense_Mutation   \n",
       "147880        18715643      18715643      +      Missense_Mutation   \n",
       "147881       115112079     115112079      +      Nonsense_Mutation   \n",
       "147882        48951162      48951162      +      Nonsense_Mutation   \n",
       "147883         7578394       7578394      +      Missense_Mutation   \n",
       "\n",
       "       Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2  \\\n",
       "147876          SNP                G                 G                 T   \n",
       "147877          SNP                A                 A                 G   \n",
       "147878          SNP                C                 C                 T   \n",
       "147879          SNP                C                 C                 T   \n",
       "147880          SNP                A                 A                 G   \n",
       "147881          SNP                G                 G                 T   \n",
       "147882          SNP                G                 G                 T   \n",
       "147883          SNP                T                 T                 A   \n",
       "\n",
       "       dbSNP_RS  dbSNP_Val_Status Tumor_Sample_Barcode  \\\n",
       "147876    novel               NaN    P-0050741-T01-IM6   \n",
       "147877    novel               NaN    P-0050741-T01-IM6   \n",
       "147878    novel               NaN    P-0050741-T01-IM6   \n",
       "147879    novel               NaN    P-0050741-T01-IM6   \n",
       "147880    novel               NaN    P-0050741-T01-IM6   \n",
       "147881    novel               NaN    P-0050741-T01-IM6   \n",
       "147882    novel               NaN    P-0050741-T01-IM6   \n",
       "147883    novel               NaN    P-0050741-T01-IM6   \n",
       "\n",
       "        Matched_Norm_Sample_Barcode Match_Norm_Seq_Allele1  \\\n",
       "147876                          NaN                    NaN   \n",
       "147877                          NaN                    NaN   \n",
       "147878                          NaN                    NaN   \n",
       "147879                          NaN                    NaN   \n",
       "147880                          NaN                    NaN   \n",
       "147881                          NaN                    NaN   \n",
       "147882                          NaN                    NaN   \n",
       "147883                          NaN                    NaN   \n",
       "\n",
       "       Match_Norm_Seq_Allele2  Tumor_Validation_Allele1  \\\n",
       "147876                    NaN                       NaN   \n",
       "147877                    NaN                       NaN   \n",
       "147878                    NaN                       NaN   \n",
       "147879                    NaN                       NaN   \n",
       "147880                    NaN                       NaN   \n",
       "147881                    NaN                       NaN   \n",
       "147882                    NaN                       NaN   \n",
       "147883                    NaN                       NaN   \n",
       "\n",
       "        Tumor_Validation_Allele2  Match_Norm_Validation_Allele1  \\\n",
       "147876                       NaN                            NaN   \n",
       "147877                       NaN                            NaN   \n",
       "147878                       NaN                            NaN   \n",
       "147879                       NaN                            NaN   \n",
       "147880                       NaN                            NaN   \n",
       "147881                       NaN                            NaN   \n",
       "147882                       NaN                            NaN   \n",
       "147883                       NaN                            NaN   \n",
       "\n",
       "        Match_Norm_Validation_Allele2  Verification_Status Validation_Status  \\\n",
       "147876                            NaN                  NaN           Unknown   \n",
       "147877                            NaN                  NaN           Unknown   \n",
       "147878                            NaN                  NaN           Unknown   \n",
       "147879                            NaN                  NaN           Unknown   \n",
       "147880                            NaN                  NaN           Unknown   \n",
       "147881                            NaN                  NaN           Unknown   \n",
       "147882                            NaN                  NaN           Unknown   \n",
       "147883                            NaN                  NaN           Unknown   \n",
       "\n",
       "       Mutation_Status  Sequencing_Phase  Sequence_Source  Validation_Method  \\\n",
       "147876         SOMATIC               NaN              NaN                NaN   \n",
       "147877         SOMATIC               NaN              NaN                NaN   \n",
       "147878         SOMATIC               NaN              NaN                NaN   \n",
       "147879         SOMATIC               NaN              NaN                NaN   \n",
       "147880         SOMATIC               NaN              NaN                NaN   \n",
       "147881         SOMATIC               NaN              NaN                NaN   \n",
       "147882         SOMATIC               NaN              NaN                NaN   \n",
       "147883         SOMATIC               NaN              NaN                NaN   \n",
       "\n",
       "             Score  BAM_File  Sequencer  Tumor_Sample_UUID  \\\n",
       "147876  MSK-IMPACT       NaN        NaN                NaN   \n",
       "147877  MSK-IMPACT       NaN        NaN                NaN   \n",
       "147878  MSK-IMPACT       NaN        NaN                NaN   \n",
       "147879  MSK-IMPACT       NaN        NaN                NaN   \n",
       "147880  MSK-IMPACT       NaN        NaN                NaN   \n",
       "147881  MSK-IMPACT       NaN        NaN                NaN   \n",
       "147882  MSK-IMPACT       NaN        NaN                NaN   \n",
       "147883  MSK-IMPACT       NaN        NaN                NaN   \n",
       "\n",
       "        Matched_Norm_Sample_UUID      HGVSc         HGVSp HGVSp_Short  \\\n",
       "147876                       NaN   c.394G>T   p.Ala132Ser     p.A132S   \n",
       "147877                       NaN   c.989T>C   p.Val330Ala     p.V330A   \n",
       "147878                       NaN  c.1769C>T   p.Pro590Leu     p.P590L   \n",
       "147879                       NaN  c.1003C>T   p.Leu335Phe     p.L335F   \n",
       "147880                       NaN  c.3474A>G  p.Ile1158Met    p.I1158M   \n",
       "147881                       NaN  c.1661C>A   p.Ser554Ter     p.S554*   \n",
       "147882                       NaN  c.1324G>T   p.Gly442Ter     p.G442*   \n",
       "147883                       NaN   c.536A>T   p.His179Leu     p.H179L   \n",
       "\n",
       "          Transcript_ID Exon_Number  t_depth  t_ref_count  t_alt_count  \\\n",
       "147876  ENST00000398015        3/16    811.0        511.0        300.0   \n",
       "147877  ENST00000368508       10/43    219.0        116.0        103.0   \n",
       "147878  ENST00000359195        2/11    663.0        388.0        275.0   \n",
       "147879  ENST00000249373        5/12   1111.0        660.0        451.0   \n",
       "147880  ENST00000266497       25/31    276.0        170.0        106.0   \n",
       "147881  ENST00000257566         7/8   1354.0        811.0        543.0   \n",
       "147882  ENST00000267163       13/27    211.0         53.0        158.0   \n",
       "147883  ENST00000269305        5/11    673.0        163.0        510.0   \n",
       "\n",
       "        n_depth n_ref_count n_alt_count  \\\n",
       "147876      384         384           0   \n",
       "147877      255         255           0   \n",
       "147878      430         429           1   \n",
       "147879      542         542           0   \n",
       "147880      219         219           0   \n",
       "147881      599         598           1   \n",
       "147882      254         254           0   \n",
       "147883      479         479           0   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    all_effects  \\\n",
       "147876                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   EPHB1,missense_variant,p.Ala132Ser,ENST00000398015,NM_004441.4;EPHB1,missense_variant,p.Ala110Ser,ENST00000460895,;EPHB1,missense_variant,p.Ala110Ser,ENST00000474732,;EPHB1,downstream_gene_variant,,ENST00000473867,;EPHB1,downstream_gene_variant,,ENST00000497173,;EPHB1,non_coding_transcript_exon_variant,,ENST00000488154,;EPHB1,missense_variant,p.Ala132Ser,ENST00000482618,;   \n",
       "147877                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  ROS1,missense_variant,p.Val330Ala,ENST00000368508,NM_002944.2;ROS1,missense_variant,p.Val339Ala,ENST00000368507,;GOPC,intron_variant,,ENST00000467125,;   \n",
       "147878                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          PIK3CG,missense_variant,p.Pro590Leu,ENST00000359195,NM_002649.2,NM_001282427.1;PIK3CG,missense_variant,p.Pro590Leu,ENST00000496166,NM_001282426.1;PIK3CG,missense_variant,p.Pro590Leu,ENST00000440650,;PIK3CG,intron_variant,,ENST00000473541,;   \n",
       "147879                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           SMO,missense_variant,p.Leu335Phe,ENST00000249373,NM_005631.4;RP11-286H14.8,downstream_gene_variant,,ENST00000466717,;SMO,missense_variant,p.Leu29Phe,ENST00000462420,;SMO,upstream_gene_variant,,ENST00000495998,;SMO,upstream_gene_variant,,ENST00000475779,;   \n",
       "147880                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               PIK3C2G,missense_variant,p.Ile1158Met,ENST00000433979,NM_004570.4;PIK3C2G,missense_variant,p.Ile1158Met,ENST00000266497,;PIK3C2G,missense_variant,p.Ile1199Met,ENST00000538779,NM_001288772.1;PIK3C2G,3_prime_UTR_variant,,ENST00000546003,NM_001288774.1;   \n",
       "147881                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           TBX3,stop_gained,p.Ser534Ter,ENST00000349155,NM_005996.3;TBX3,stop_gained,p.Ser554Ter,ENST00000257566,NM_016569.3;TBX3,downstream_gene_variant,,ENST00000548503,;,regulatory_region_variant,,ENSR00000077728,;   \n",
       "147882                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 RB1,stop_gained,p.Gly442Ter,ENST00000267163,NM_000321.2;   \n",
       "147883  TP53,missense_variant,p.His179Leu,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.His179Leu,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.His179Leu,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.His179Leu,ENST00000445888,;TP53,missense_variant,p.His179Leu,ENST00000359597,;TP53,missense_variant,p.His179Leu,ENST00000413465,;TP53,missense_variant,p.His47Leu,ENST00000509690,;TP53,missense_variant,p.His86Leu,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,EN...   \n",
       "\n",
       "       Allele             Gene          Feature Feature_type  \\\n",
       "147876      T  ENSG00000154928  ENST00000398015   Transcript   \n",
       "147877      G  ENSG00000047936  ENST00000368508   Transcript   \n",
       "147878      T  ENSG00000105851  ENST00000359195   Transcript   \n",
       "147879      T  ENSG00000128602  ENST00000249373   Transcript   \n",
       "147880      G  ENSG00000139144  ENST00000266497   Transcript   \n",
       "147881      T  ENSG00000135111  ENST00000257566   Transcript   \n",
       "147882      T  ENSG00000139687  ENST00000267163   Transcript   \n",
       "147883      A  ENSG00000141510  ENST00000269305   Transcript   \n",
       "\n",
       "             Consequence cDNA_position CDS_position Protein_position  \\\n",
       "147876  missense_variant      764/4672     394/2955          132/984   \n",
       "147877  missense_variant     1188/7435     989/7044         330/2347   \n",
       "147878  missense_variant     2079/5377    1769/3309         590/1102   \n",
       "147879  missense_variant     1283/3738    1003/2364          335/787   \n",
       "147880  missense_variant     3512/4754    3474/4338        1158/1445   \n",
       "147881       stop_gained     2051/4208    1661/2232          554/743   \n",
       "147882       stop_gained     1462/4840    1324/2787          442/928   \n",
       "147883  missense_variant      726/2579     536/1182          179/393   \n",
       "\n",
       "       Amino_acids   Codons  Existing_variation  ALLELE_NUM  DISTANCE  \\\n",
       "147876         A/S  Gcc/Tcc                 NaN           1       NaN   \n",
       "147877         V/A  gTt/gCt                 NaN           1       NaN   \n",
       "147878         P/L  cCt/cTt                 NaN           1       NaN   \n",
       "147879         L/F  Ctc/Ttc                 NaN           1       NaN   \n",
       "147880         I/M  atA/atG                 NaN           1       NaN   \n",
       "147881         S/*  tCg/tAg                 NaN           1       NaN   \n",
       "147882         G/*  Gga/Tga                 NaN           1       NaN   \n",
       "147883         H/L  cAt/cTt                 NaN           1       NaN   \n",
       "\n",
       "        STRAND_VEP   SYMBOL SYMBOL_SOURCE  HGNC_ID         BIOTYPE CANONICAL  \\\n",
       "147876           1    EPHB1          HGNC   3392.0  protein_coding       YES   \n",
       "147877          -1     ROS1          HGNC  10261.0  protein_coding       YES   \n",
       "147878           1   PIK3CG          HGNC   8978.0  protein_coding       YES   \n",
       "147879           1      SMO          HGNC  11119.0  protein_coding       YES   \n",
       "147880           1  PIK3C2G          HGNC   8973.0  protein_coding       NaN   \n",
       "147881          -1     TBX3          HGNC  11602.0  protein_coding       YES   \n",
       "147882           1      RB1          HGNC   9884.0  protein_coding       YES   \n",
       "147883          -1     TP53          HGNC  11998.0  protein_coding       YES   \n",
       "\n",
       "               CCDS             ENSP SWISSPROT  \\\n",
       "147876  CCDS46921.1  ENSP00000381097    P54762   \n",
       "147877   CCDS5116.1  ENSP00000357494    P08922   \n",
       "147878   CCDS5739.1  ENSP00000352121    P48736   \n",
       "147879   CCDS5811.1  ENSP00000249373    Q99835   \n",
       "147880  CCDS44839.1  ENSP00000266497    O75747   \n",
       "147881   CCDS9176.1  ENSP00000257566    O15119   \n",
       "147882  CCDS31973.1  ENSP00000267163    P06400   \n",
       "147883  CCDS11118.1  ENSP00000269305    P04637   \n",
       "\n",
       "                                                                                                                                                                                TREMBL  \\\n",
       "147876                                                                                                                                              C9K090,C9JB88,C9J6S4,C9J466,B3KTB2   \n",
       "147877                                                                                                                                                                             NaN   \n",
       "147878                                                                                                                                                                   Q24M88,E9PDN7   \n",
       "147879                                                                                                                                                                          A4D1K5   \n",
       "147880                                                                                                                                                                             NaN   \n",
       "147881                                                                                                                                                                             NaN   \n",
       "147882                                                                                                    Q7KZ61,Q7KZ60,Q6LE82,Q6LE81,Q6LE80,Q6LE79,Q6LE77,Q6LE65,Q6DUF0,P78495,B4DRB3   \n",
       "147883  S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "\n",
       "              UNIPARC  \\\n",
       "147876  UPI000012A07E   \n",
       "147877  UPI000013D467   \n",
       "147878  UPI00000746B8   \n",
       "147879  UPI0000050447   \n",
       "147880  UPI000013D6EF   \n",
       "147881  UPI0000136AA0   \n",
       "147882  UPI0000001C79   \n",
       "147883  UPI000002ED67   \n",
       "\n",
       "                                                                         RefSeq  \\\n",
       "147876                                                              NM_004441.4   \n",
       "147877                                                              NM_002944.2   \n",
       "147878                                               NM_002649.2,NM_001282427.1   \n",
       "147879                                                              NM_005631.4   \n",
       "147880                                                                      NaN   \n",
       "147881                                                              NM_016569.3   \n",
       "147882                                                              NM_000321.2   \n",
       "147883  NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "\n",
       "                     SIFT                  PolyPhen   EXON INTRON  \\\n",
       "147876    tolerated(0.74)             benign(0.012)   3/16    NaN   \n",
       "147877    tolerated(0.14)             benign(0.024)  10/43    NaN   \n",
       "147878  deleterious(0.05)  probably_damaging(0.998)   2/11    NaN   \n",
       "147879    tolerated(0.16)  possibly_damaging(0.646)   5/12    NaN   \n",
       "147880     deleterious(0)  probably_damaging(0.999)  25/31    NaN   \n",
       "147881                NaN                       NaN    7/8    NaN   \n",
       "147882                NaN                       NaN  13/27    NaN   \n",
       "147883    tolerated(0.12)      probably_damaging(1)   5/11    NaN   \n",
       "\n",
       "                                                                                                                                                                                             DOMAINS  \\\n",
       "147876  PROSITE_profiles:PS51550,hmmpanther:PTHR24416,hmmpanther:PTHR24416:SF173,Pfam_domain:PF01404,Gene3D:2.60.120.260,PIRSF_domain:PIRSF000666,SMART_domains:SM00615,Superfamily_domains:SSF49785   \n",
       "147877                                                                                               Gene3D:2.120.10.30,hmmpanther:PTHR24416,hmmpanther:PTHR24416:SF304,Superfamily_domains:SSF63825   \n",
       "147878                              Gene3D:1.25.40.70,Pfam_domain:PF00613,PROSITE_profiles:PS51545,hmmpanther:PTHR10048,hmmpanther:PTHR10048:SF34,SMART_domains:SM00145,Superfamily_domains:SSF48371   \n",
       "147879                                               Transmembrane_helices:TMhelix,PROSITE_profiles:PS50261,hmmpanther:PTHR11309,hmmpanther:PTHR11309:SF35,Pfam_domain:PF01534,Prints_domain:PR00489   \n",
       "147880                                                Gene3D:1.10.1070.11,PROSITE_profiles:PS50290,hmmpanther:PTHR10048,hmmpanther:PTHR10048:SF29,SMART_domains:SM00146,Superfamily_domains:SSF56112   \n",
       "147881                                                                                                                                                hmmpanther:PTHR11267,hmmpanther:PTHR11267:SF91   \n",
       "147882                                                                                                      hmmpanther:PTHR13742,Pfam_domain:PF01858,Gene3D:1.10.472.10,Superfamily_domains:SSF47954   \n",
       "147883                                                       Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "\n",
       "        GMAF  AFR_MAF  AMR_MAF  ASN_MAF  EAS_MAF  EUR_MAF  SAS_MAF  AA_MAF  \\\n",
       "147876   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "147877   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "147878   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "147879   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "147880   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "147881   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "147882   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "147883   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "\n",
       "        EA_MAF  CLIN_SIG  SOMATIC  PUBMED  MOTIF_NAME  MOTIF_POS  \\\n",
       "147876     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "147877     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "147878     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "147879     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "147880     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "147881     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "147882     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "147883     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "\n",
       "        HIGH_INF_POS  MOTIF_SCORE_CHANGE    IMPACT  PICK VARIANT_CLASS  TSL  \\\n",
       "147876           NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "147877           NaN                 NaN  MODERATE   1.0           SNV  NaN   \n",
       "147878           NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "147879           NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "147880           NaN                 NaN  MODERATE   NaN           SNV  NaN   \n",
       "147881           NaN                 NaN      HIGH   1.0         indel  NaN   \n",
       "147882           NaN                 NaN      HIGH   1.0         indel  NaN   \n",
       "147883           NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "\n",
       "        HGVS_OFFSET  PHENO  MINIMISED   ExAC_AF  ExAC_AF_AFR  ExAC_AF_AMR  \\\n",
       "147876          NaN    NaN        NaN       NaN          NaN          NaN   \n",
       "147877          NaN    NaN        1.0       NaN          NaN          NaN   \n",
       "147878          NaN    NaN        NaN  0.000009      0.00011          0.0   \n",
       "147879          NaN    NaN        NaN       NaN          NaN          NaN   \n",
       "147880          NaN    NaN        1.0       NaN          NaN          NaN   \n",
       "147881          NaN    NaN        NaN       NaN          NaN          NaN   \n",
       "147882          NaN    NaN        NaN       NaN          NaN          NaN   \n",
       "147883          NaN    NaN        NaN       NaN          NaN          NaN   \n",
       "\n",
       "        ExAC_AF_EAS  ExAC_AF_FIN  ExAC_AF_NFE  ExAC_AF_OTH  ExAC_AF_SAS  \\\n",
       "147876          NaN          NaN          NaN          NaN          NaN   \n",
       "147877          NaN          NaN          NaN          NaN          NaN   \n",
       "147878          0.0          0.0          0.0          0.0          0.0   \n",
       "147879          NaN          NaN          NaN          NaN          NaN   \n",
       "147880          NaN          NaN          NaN          NaN          NaN   \n",
       "147881          NaN          NaN          NaN          NaN          NaN   \n",
       "147882          NaN          NaN          NaN          NaN          NaN   \n",
       "147883          NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "        GENE_PHENO FILTER flanking_bps variant_id variant_qual  ExAC_AF_Adj  \\\n",
       "147876         1.0    NaN          GGC          .            .          NaN   \n",
       "147877         1.0    NaN          AAC          .            .          NaN   \n",
       "147878         NaN    NaN          CCT          .            .     0.000009   \n",
       "147879         1.0    NaN          CCT          .            .          NaN   \n",
       "147880         NaN    NaN          TAA          .            .          NaN   \n",
       "147881         1.0    NaN          CGA          .            .          NaN   \n",
       "147882         1.0    NaN          TGG          .            .          NaN   \n",
       "147883         1.0    NaN          ATG          .            .          NaN   \n",
       "\n",
       "       ExAC_AC_AN_Adj ExAC_AC_AN ExAC_AC_AN_AFR ExAC_AC_AN_AMR ExAC_AC_AN_EAS  \\\n",
       "147876            NaN        NaN            NaN            NaN            NaN   \n",
       "147877            NaN        NaN            NaN            NaN            NaN   \n",
       "147878       1/106208   1/106210         1/9064        0/11216         0/7866   \n",
       "147879            NaN        NaN            NaN            NaN            NaN   \n",
       "147880            NaN        NaN            NaN            NaN            NaN   \n",
       "147881            NaN        NaN            NaN            NaN            NaN   \n",
       "147882            NaN        NaN            NaN            NaN            NaN   \n",
       "147883            NaN        NaN            NaN            NaN            NaN   \n",
       "\n",
       "       ExAC_AC_AN_FIN ExAC_AC_AN_NFE ExAC_AC_AN_OTH ExAC_AC_AN_SAS  \\\n",
       "147876            NaN            NaN            NaN            NaN   \n",
       "147877            NaN            NaN            NaN            NaN   \n",
       "147878         0/6614        0/54346          0/694        0/16408   \n",
       "147879            NaN            NaN            NaN            NaN   \n",
       "147880            NaN            NaN            NaN            NaN   \n",
       "147881            NaN            NaN            NaN            NaN   \n",
       "147882            NaN            NaN            NaN            NaN   \n",
       "147883            NaN            NaN            NaN            NaN   \n",
       "\n",
       "                       ExAC_FILTER  Caller is-a-hotspot is-a-3d-hotspot  \\\n",
       "147876                         NaN     NaN          NaN             NaN   \n",
       "147877                         NaN     NaN          NaN             NaN   \n",
       "147878                        PASS     NaN          NaN             NaN   \n",
       "147879                         NaN     NaN          NaN             NaN   \n",
       "147880                         NaN     NaN          NaN             NaN   \n",
       "147881  VQSRTrancheSNP99.60to99.80     NaN          NaN             NaN   \n",
       "147882                         NaN     NaN          NaN             NaN   \n",
       "147883                         NaN     NaN            Y               Y   \n",
       "\n",
       "                mutation_effect         oncogenic LEVEL_1 LEVEL_2A LEVEL_2B  \\\n",
       "147876                      NaN               NaN     NaN      NaN      NaN   \n",
       "147877                      NaN               NaN     NaN      NaN      NaN   \n",
       "147878                      NaN               NaN     NaN      NaN      NaN   \n",
       "147879                      NaN               NaN     NaN      NaN      NaN   \n",
       "147880                      NaN               NaN     NaN      NaN      NaN   \n",
       "147881  Likely Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "147882  Likely Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "147883         Loss-of-function         Oncogenic     NaN      NaN      NaN   \n",
       "\n",
       "       LEVEL_3A LEVEL_3B LEVEL_4 LEVEL_R1 LEVEL_R2  LEVEL_R3 Highest_level  \\\n",
       "147876      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "147877      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "147878      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "147879      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "147880      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "147881      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "147882      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "147883      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "\n",
       "                                  citations  driver  tcn  lcn        cf  \\\n",
       "147876                                  NaN   False  2.0  1.0  1.000000   \n",
       "147877                                  NaN   False  2.0  1.0  1.000000   \n",
       "147878                                  NaN   False  2.0  1.0  1.000000   \n",
       "147879                                  NaN   False  2.0  1.0  1.000000   \n",
       "147880                                  NaN   False  0.0  0.0  0.105024   \n",
       "147881                    11689487;26579496    True  2.0  1.0  1.000000   \n",
       "147882  25621664;30206110;22293180;22205104    True  1.0  0.0  0.833058   \n",
       "147883                             23150443    True  1.0  0.0  0.824066   \n",
       "\n",
       "          purity  t_var_freq  expected_alt_copies  ccf_Mcopies  \\\n",
       "147876  0.833058    0.369914                  1.0        0.888   \n",
       "147877  0.833058    0.470320                  1.0        1.000   \n",
       "147878  0.833058    0.414781                  1.0        0.996   \n",
       "147879  0.833058    0.405941                  1.0        0.975   \n",
       "147880  0.833058    0.384058                  1.0          NaN   \n",
       "147881  0.833058    0.401034                  1.0        0.963   \n",
       "147882  0.833058    0.748815                  1.0        1.000   \n",
       "147883  0.833058    0.757801                  1.0        1.000   \n",
       "\n",
       "        ccf_Mcopies_lower  ccf_Mcopies_upper  ccf_Mcopies_prob95  \\\n",
       "147876              0.840              0.937            0.065682   \n",
       "147877              0.969              1.000            0.776482   \n",
       "147878              0.942              1.000            0.714943   \n",
       "147879              0.933              1.000            0.691664   \n",
       "147880                NaN              1.000                 NaN   \n",
       "147881              0.925              1.000            0.617843   \n",
       "147882              0.977              1.000            0.904269   \n",
       "147883              0.993              1.000            0.999502   \n",
       "\n",
       "        ccf_Mcopies_prob90  ccf_1copy  ccf_1copy_lower  ccf_1copy_upper  \\\n",
       "147876            0.392463      0.888            0.840            0.937   \n",
       "147877            0.965241      1.000            0.969            1.000   \n",
       "147878            0.969240      0.996            0.942            1.000   \n",
       "147879            0.979469      0.975            0.933            1.000   \n",
       "147880                 NaN        NaN              NaN            1.000   \n",
       "147881            0.974354      0.963            0.925            1.000   \n",
       "147882            0.996372      1.000            0.977            1.000   \n",
       "147883            1.000000      1.000            0.993            1.000   \n",
       "\n",
       "        ccf_1copy_prob95  ccf_1copy_prob90  ccf_expected_copies  \\\n",
       "147876          0.065682          0.392463                0.888   \n",
       "147877          0.776482          0.965241                1.000   \n",
       "147878          0.714943          0.969240                0.996   \n",
       "147879          0.691664          0.979469                0.975   \n",
       "147880               NaN               NaN                  NaN   \n",
       "147881          0.617843          0.974354                0.963   \n",
       "147882          0.904269          0.996372                1.000   \n",
       "147883          0.999502          1.000000                1.000   \n",
       "\n",
       "        ccf_expected_copies_lower  ccf_expected_copies_upper  \\\n",
       "147876                      0.840                      0.937   \n",
       "147877                      0.969                      1.000   \n",
       "147878                      0.942                      1.000   \n",
       "147879                      0.933                      1.000   \n",
       "147880                        NaN                      1.000   \n",
       "147881                      0.925                      1.000   \n",
       "147882                      0.977                      1.000   \n",
       "147883                      0.993                      1.000   \n",
       "\n",
       "        ccf_expected_copies_prob95  ccf_expected_copies_prob90  \\\n",
       "147876                    0.065682                    0.392463   \n",
       "147877                    0.776482                    0.965241   \n",
       "147878                    0.714943                    0.969240   \n",
       "147879                    0.691664                    0.979469   \n",
       "147880                         NaN                         NaN   \n",
       "147881                    0.617843                    0.974354   \n",
       "147882                    0.904269                    0.996372   \n",
       "147883                    0.999502                    1.000000   \n",
       "\n",
       "                                                                                                    facets_fit  \\\n",
       "147876  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default   \n",
       "147877  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default   \n",
       "147878  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default   \n",
       "147879  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default   \n",
       "147880  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default   \n",
       "147881  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default   \n",
       "147882  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default   \n",
       "147883  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default   \n",
       "\n",
       "        reviewer_set_purity  use_only_purity_run  use_edited_cncf  \\\n",
       "147876                  NaN                False            False   \n",
       "147877                  NaN                False            False   \n",
       "147878                  NaN                False            False   \n",
       "147879                  NaN                False            False   \n",
       "147880                  NaN                False            False   \n",
       "147881                  NaN                False            False   \n",
       "147882                  NaN                False            False   \n",
       "147883                  NaN                False            False   \n",
       "\n",
       "                                                                                                                                                    cncf_file_used  \\\n",
       "147876  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt   \n",
       "147877  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt   \n",
       "147878  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt   \n",
       "147879  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt   \n",
       "147880  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt   \n",
       "147881  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt   \n",
       "147882  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt   \n",
       "147883  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050741-T01-IM6_P-0050741-N01-IM6//default/P-0050741-T01-IM6_P-0050741-N01-IM6_hisens.cncf.txt   \n",
       "\n",
       "                 mut_key                      sample_mut_key  \n",
       "147876   3_134670483_G_T   P-0050741-T01-IM6-3_134670483_G_T  \n",
       "147877   6_117715500_A_G   P-0050741-T01-IM6-6_117715500_A_G  \n",
       "147878   7_106509775_C_T   P-0050741-T01-IM6-7_106509775_C_T  \n",
       "147879   7_128846073_C_T   P-0050741-T01-IM6-7_128846073_C_T  \n",
       "147880   12_18715643_A_G   P-0050741-T01-IM6-12_18715643_A_G  \n",
       "147881  12_115112079_G_T  P-0050741-T01-IM6-12_115112079_G_T  \n",
       "147882   13_48951162_G_T   P-0050741-T01-IM6-13_48951162_G_T  \n",
       "147883    17_7578394_T_A    P-0050741-T01-IM6-17_7578394_T_A  "
      ]
     },
     "execution_count": 222,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_cohort_nowgd[maf_cohort_nowgd['Tumor_Sample_Barcode'] == 'P-0050741-T01-IM6']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_nowgd = pd.read_csv(data_no_wgd + 'cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GERMLINE</th>\n",
       "      <td>799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOMATIC</th>\n",
       "      <td>147625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UNKNOWN</th>\n",
       "      <td>1380</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  count\n",
       "Mutation_Status        \n",
       "GERMLINE            799\n",
       "SOMATIC          147625\n",
       "UNKNOWN            1380"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(maf_cohort_nowgd, 'Mutation_Status', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "#maf_cohort_wgd\n",
    "\n",
    "def removing_germ_na_mut(maf_cohort: pd.DataFrame):\n",
    "    '''\n",
    "    This function is meant to remove non TP53 GERMLINE mutations from the mafs, to have only SOMATIC mut \n",
    "    except for TP53 muts where we allow GERMLINE status.\n",
    "    '''\n",
    "    maf_cohort_final = maf_cohort[maf_cohort['Mutation_Status'] != 'UNKNOWN']\n",
    "\n",
    "    key_list = list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] != 'TP53'][maf_cohort_final['Mutation_Status'] != 'GERMLINE'].sample_mut_key)\n",
    "    key_list = key_list + list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] == 'TP53'].sample_mut_key)\n",
    "    maf_cohort_final = maf_cohort_final[maf_cohort_final.sample_mut_key.isin(key_list)]\n",
    "    \n",
    "    return maf_cohort_final\n",
    "\n",
    "new = removing_germ_na_mut(maf_cohort_nowgd)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GERMLINE</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOMATIC</th>\n",
       "      <td>147625</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  count\n",
       "Mutation_Status        \n",
       "GERMLINE              6\n",
       "SOMATIC          147625"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(new, 'Mutation_Status', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "336px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
