{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <span style='font-family:\"Times New Roman\"'> <span styel=''> **MASTER FILE CREATION**\n",
    "\n",
    "*Emile Cohen* \n",
    "    \n",
    "*March 2020*\n",
    "\n",
    "**Goal:** In this Notebook, we create a master file that summarizes all useful information.\n",
    "\n",
    "The Notebook is divided in 4 parts, representing the four parts of our Master file:\n",
    "   \n",
    "* **1. Patient/Sample Information**\n",
    "* **2. TP53 Mutations**\n",
    "* **3. TP53 Copy Numbers**\n",
    "* **4. TP53 Computed Metrics**\n",
    "* **5. Subgroup columns creation**\n",
    "* **6. Merge tables**\n",
    "\n",
    "**NB1:** In each part, you must run the cells from the begining in order to initialize the variables\n",
    "\n",
    "**NB2:** In order to launch the last script (Merge Tables), you have to define the functions in each part.\n",
    "\n",
    "**NB3:** All functions used for the plots are located in utils/custom_tools.py\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Setup environment... done!\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "<span style=\"color:green\">âœ… Working on **mskimpact_env** conda environment.</span>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%run -i '../../../utils/setup_environment.ipy'\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "from scipy.stats import fisher_exact, ranksums, chi2, norm\n",
    "from statsmodels.sandbox.stats.multicomp import multipletests\n",
    "import matplotlib.gridspec as gridspec\n",
    "import pickle\n",
    "\n",
    "data_path = '../../../data/'\n",
    "data_wgd = data_path + 'impact-facets-tp53/processed/wgd/'\n",
    "data_no_wgd = data_path + 'impact-facets-tp53/processed/no_wgd/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "# first we load all files for WGD\n",
    "maf_cohort_wgd = pd.read_csv(data_wgd + 'maf_cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_wgd = pd.read_csv(data_wgd + 'cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_wgd = pd.read_csv(data_wgd + 'arm_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_wgd = pd.read_csv(data_wgd + 'gene_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We load all files for non WGD\n",
    "\n",
    "maf_cohort_nowgd = pd.read_csv(data_no_wgd + 'maf_cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_nowgd = pd.read_csv(data_no_wgd + 'cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_nowgd = pd.read_csv(data_no_wgd + 'arm_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_nowgd = pd.read_csv(data_no_wgd + 'gene_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Creating keys for mutations in maf files\n",
    "\n",
    "# First we need to create a sample_mut_key to identify duplicated mutations\n",
    "maf_cohort_wgd['mut_key'] = maf_cohort_wgd.apply(lambda h: str(h.Chromosome)+'_'+str(h.Start_Position)+'_'+str(h.Reference_Allele)+'_'+str(h.Tumor_Seq_Allele2), axis=1) \n",
    "maf_cohort_nowgd['mut_key'] = maf_cohort_nowgd.apply(lambda h: str(h.Chromosome)+'_'+str(h.Start_Position)+'_'+str(h.Reference_Allele)+'_'+str(h.Tumor_Seq_Allele2), axis=1) \n",
    "\n",
    "# Create a sample key to differentiate duplicates\n",
    "maf_cohort_wgd['sample_mut_key'] = maf_cohort_wgd.apply(lambda h: h.Tumor_Sample_Barcode + h.mut_key, axis = 1)\n",
    "maf_cohort_nowgd['sample_mut_key'] = maf_cohort_nowgd.apply(lambda h: h.Tumor_Sample_Barcode + h.mut_key, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load clinical data\n",
    "clinical_data = pd.read_csv(data_path + 'cbioportal/raw/mskimpact_clinical_data-2.tsv', sep= '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filtering the clinical data\n",
    "samples_wgd = list(set(cohort_wgd.tumor_sample))\n",
    "samples_nowgd = list(set(cohort_nowgd.tumor_sample))\n",
    "\n",
    "clinical_wgd = clinical_data[clinical_data['Sample ID'].isin(samples_wgd)]\n",
    "clinical_nowgd = clinical_data[clinical_data['Sample ID'].isin(samples_nowgd)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IMPORTANT: Parameter Definition\n",
    "In this script we can create two master files: one for WGD samples and one for non WGD samples.\n",
    "So this parameter allows to select the type of master you want.\n",
    "On top of that you can specify where you want to store the file you created"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 146,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_type = 'wgd'\n",
    "output_path = data_path + 'impact-facets-tp53/processed/'\n",
    "\n",
    "total_output = output_path + cohort_type + '/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "# Patient/Sample Information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we focus on clinical information exported from CbioPortal.\n",
    "\n",
    "The following columns are selected:\n",
    "* Sample_Id\n",
    "* Tumor_Id\n",
    "* Patient_Id\n",
    "* Patient Current Age\n",
    "* Cancer_Type\n",
    "* Cancer_Type_Detailed\n",
    "* Sample_Type\n",
    "* purity\n",
    "* ploidy\n",
    "* Overall Survival Status\n",
    "* Overall Survival (Months)\n",
    "* MSI Score\n",
    "* MSI Type\n",
    "* Tumor Mutational Burden"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_sample_info(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    cohort_filt = cohort[['sample_id', 'tumor_sample', 'patient', 'ploidy']]\n",
    "    clinical_filt = clinical[['Sample ID', 'Patient Current Age', 'Cancer Type', 'Cancer Type Detailed', 'Sample Type',\n",
    "                  'Overall Survival (Months)', 'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "    purity = maf_cohort.drop_duplicates('Tumor_Sample_Barcode')[['Tumor_Sample_Barcode', 'purity']]\n",
    "\n",
    "    # Merging these files\n",
    "    sample_info = pd.merge(cohort_filt, purity, left_on='tumor_sample', right_on='Tumor_Sample_Barcode')\n",
    "    sample_info =  pd.merge(sample_info, clinical_filt, left_on='tumor_sample', right_on='Sample ID').drop(['Tumor_Sample_Barcode', 'Sample ID'], axis=1)\n",
    "\n",
    "\n",
    "    sample_info = sample_info[['sample_id', 'tumor_sample', 'patient', 'Cancer Type', 'Cancer Type Detailed',\n",
    "                               'Patient Current Age','Sample Type', 'purity','ploidy','Overall Survival (Months)', \n",
    "                               'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "\n",
    "    sample_info.columns = ['Sample_Id', 'Tumor_Id', 'Patient_Id','Cancer_Type', 'Cancer_Type_Detailed', 'Patient_Current_Age',\n",
    "                          'Sample_Type', 'purity', 'ploidy', 'Overall_Survival_Months', 'Overall_Survival_Status',\n",
    "                          'MSI_Score', 'MSI_Type', 'TMB_Score']\n",
    "    \n",
    "    return sample_info\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>Overall_Survival_Months</th>\n",
       "      <th>Overall_Survival_Status</th>\n",
       "      <th>MSI_Score</th>\n",
       "      <th>MSI_Type</th>\n",
       "      <th>TMB_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>P-0025956</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>71.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>3.496971</td>\n",
       "      <td>3.584</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.391316</td>\n",
       "      <td>2.871793</td>\n",
       "      <td>14.137</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Neuroendocrine Carcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.865628</td>\n",
       "      <td>3.115253</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>2.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>3.136841</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.82</td>\n",
       "      <td>Stable</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Colon Adenocarcinoma</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.351778</td>\n",
       "      <td>3.727190</td>\n",
       "      <td>34.060</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.45</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9893</th>\n",
       "      <td>P-0050750-T01-IM6_P-0050750-N01-IM6</td>\n",
       "      <td>P-0050750-T01-IM6</td>\n",
       "      <td>P-0050750</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.301523</td>\n",
       "      <td>3.845451</td>\n",
       "      <td>0.099</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9894</th>\n",
       "      <td>P-0050643-T01-IM6_P-0050643-N01-IM6</td>\n",
       "      <td>P-0050643-T01-IM6</td>\n",
       "      <td>P-0050643</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.667936</td>\n",
       "      <td>4.419218</td>\n",
       "      <td>1.315</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.30</td>\n",
       "      <td>Stable</td>\n",
       "      <td>4.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9895</th>\n",
       "      <td>P-0050223-T02-IM6_P-0050223-N02-IM6</td>\n",
       "      <td>P-0050223-T02-IM6</td>\n",
       "      <td>P-0050223</td>\n",
       "      <td>Hepatobiliary Cancer</td>\n",
       "      <td>Cholangiocarcinoma</td>\n",
       "      <td>67.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.215128</td>\n",
       "      <td>4.725790</td>\n",
       "      <td>1.940</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9896</th>\n",
       "      <td>P-0048760-T01-IM6_P-0048760-N01-IM6</td>\n",
       "      <td>P-0048760-T01-IM6</td>\n",
       "      <td>P-0048760</td>\n",
       "      <td>Bone Cancer</td>\n",
       "      <td>Osteosarcoma</td>\n",
       "      <td>18.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.614603</td>\n",
       "      <td>3.947721</td>\n",
       "      <td>3.682</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>5.79</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>6.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9897</th>\n",
       "      <td>P-0050744-T01-IM6_P-0050744-N01-IM6</td>\n",
       "      <td>P-0050744-T01-IM6</td>\n",
       "      <td>P-0050744</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>69.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>2.426452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.06</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9898 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6  P-0025956-T01-IM6  P-0025956   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6  P-0036909-T01-IM6  P-0036909   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6  P-0023546-T01-IM6  P-0023546   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6  P-0023546-T02-IM6  P-0023546   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6  P-0018837-T01-IM6  P-0018837   \n",
       "...                                   ...                ...        ...   \n",
       "9893  P-0050750-T01-IM6_P-0050750-N01-IM6  P-0050750-T01-IM6  P-0050750   \n",
       "9894  P-0050643-T01-IM6_P-0050643-N01-IM6  P-0050643-T01-IM6  P-0050643   \n",
       "9895  P-0050223-T02-IM6_P-0050223-N02-IM6  P-0050223-T02-IM6  P-0050223   \n",
       "9896  P-0048760-T01-IM6_P-0048760-N01-IM6  P-0048760-T01-IM6  P-0048760   \n",
       "9897  P-0050744-T01-IM6_P-0050744-N01-IM6  P-0050744-T01-IM6  P-0050744   \n",
       "\n",
       "                     Cancer_Type               Cancer_Type_Detailed  \\\n",
       "0     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "1     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "2                Prostate Cancer  Prostate Neuroendocrine Carcinoma   \n",
       "3                Prostate Cancer            Prostate Adenocarcinoma   \n",
       "4              Colorectal Cancer               Colon Adenocarcinoma   \n",
       "...                          ...                                ...   \n",
       "9893           Pancreatic Cancer          Pancreatic Adenocarcinoma   \n",
       "9894               Breast Cancer   Breast Invasive Ductal Carcinoma   \n",
       "9895        Hepatobiliary Cancer                 Cholangiocarcinoma   \n",
       "9896                 Bone Cancer                       Osteosarcoma   \n",
       "9897           Pancreatic Cancer          Pancreatic Adenocarcinoma   \n",
       "\n",
       "      Patient_Current_Age Sample_Type    purity    ploidy  \\\n",
       "0                    71.0     Primary  0.273767  3.496971   \n",
       "1                    47.0  Metastasis  0.391316  2.871793   \n",
       "2                    50.0     Primary  0.865628  3.115253   \n",
       "3                    50.0     Primary  0.312907  3.136841   \n",
       "4                    60.0     Primary  0.351778  3.727190   \n",
       "...                   ...         ...       ...       ...   \n",
       "9893                 55.0  Metastasis  0.301523  3.845451   \n",
       "9894                 53.0     Primary  0.667936  4.419218   \n",
       "9895                 67.0     Primary  0.215128  4.725790   \n",
       "9896                 18.0     Primary  0.614603  3.947721   \n",
       "9897                 69.0     Primary  0.300000  2.426452   \n",
       "\n",
       "      Overall_Survival_Months Overall_Survival_Status  MSI_Score  \\\n",
       "0                       3.584                DECEASED       0.00   \n",
       "1                      14.137                  LIVING       0.37   \n",
       "2                       4.800                DECEASED       2.37   \n",
       "3                       4.800                DECEASED       0.82   \n",
       "4                      34.060                  LIVING       0.45   \n",
       "...                       ...                     ...        ...   \n",
       "9893                    0.099                  LIVING       0.05   \n",
       "9894                    1.315                  LIVING       0.30   \n",
       "9895                    1.940                  LIVING       0.00   \n",
       "9896                    3.682                  LIVING       5.79   \n",
       "9897                      NaN                  LIVING       0.06   \n",
       "\n",
       "           MSI_Type  TMB_Score  \n",
       "0            Stable        5.3  \n",
       "1            Stable        3.5  \n",
       "2            Stable        3.5  \n",
       "3            Stable        2.6  \n",
       "4            Stable        5.3  \n",
       "...             ...        ...  \n",
       "9893         Stable        5.3  \n",
       "9894         Stable        4.4  \n",
       "9895         Stable        3.5  \n",
       "9896  Indeterminate        6.1  \n",
       "9897         Stable        5.3  \n",
       "\n",
       "[9898 rows x 14 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_sample_info(cohort=cohort_type)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Mutations\n",
    "In this part, we focus on tp53 mutation information.\n",
    "\n",
    "We gather all mutations per sample, and split it into different columns. We have the following columns:\n",
    "* Tumor_Id\t\n",
    "* key_1 (2,3,4,5) --> Mutation key allowing to filter duplicates\n",
    "* vc_1 (2,3,4,5) --> Variant Classification\n",
    "* ccf_1 (2,3,4,5) --> Cancer Cell Fraction of the mutation\n",
    "* vaf_1 (2,3,4,5) --> Variant Allele Frequency of the mutation\n",
    "* HGVSp_1 (2,3,4,5) --> protein change\n",
    "* spot_1 (2,3,4,5) --> Integer that defines the spot of the tp53 mutation\n",
    "* tp53_count --> Number of tp53 mutations of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def f_(x):\n",
    "    # This function helps us to group mutations together in a single cell per patient\n",
    "    return pd.DataFrame(dict(Tumor_Sample_Barcode = x['Tumor_Sample_Barcode'],  \n",
    "                        muts = \"%s\" % ','.join(x['sample_mut_key_vc_ccf_vaf_hgv_spot'])))\n",
    "\n",
    "def count_tp53_muts(x):\n",
    "    count = 0\n",
    "    for i in range(1,6):\n",
    "        if x['tp53_key_' + str(i)]:\n",
    "            count+= 1\n",
    "    return count\n",
    "\n",
    "cohort = cohort_type\n",
    "\n",
    "\n",
    "# WARNING: THis function needs sample_info to work\n",
    "def create_tp53_muts(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    '''\n",
    "    This function aims to gather all tp53 mutation characteristics.\n",
    "    For each sample we gather the tp53 mutations and their characteristics for all patients.\n",
    "    '''\n",
    "    # We load the  table created in maf_tp53_creation.ipynb\n",
    "    maf_tp53 = maf_cohort[maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    maf_tp53['mut_spot'] = maf_tp53.HGVSp.str.extract('(\\d+)')\n",
    "\n",
    "    # We select only intresting columns\n",
    "    maf_tp53_filtered = maf_tp53[['Tumor_Sample_Barcode','sample_mut_key', 'Variant_Classification',\\\n",
    "                                        'ccf_expected_copies', 't_var_freq', 'HGVSp','mut_spot' ]]\n",
    "\n",
    "    # Let's Merge mut_key,Variant_classification, CF, CCF, and VAF to gather them\n",
    "    maf_tp53_filtered['sample_mut_key_vc_ccf_vaf_hgv_spot'] = maf_tp53_filtered.apply(lambda x: str(x.sample_mut_key)+'%'+str(x.Variant_Classification)+'%'+str(x.ccf_expected_copies)+'%'+str(x.t_var_freq)+'%'+str(x.HGVSp)+'%'+str(x.mut_spot), axis=1)\n",
    "\n",
    "    # We Select important columns\n",
    "    final = maf_tp53_filtered[['Tumor_Sample_Barcode', 'sample_mut_key_vc_ccf_vaf_hgv_spot']]\n",
    "    # We groupby Patient_Id and apply the function above to group mutations\n",
    "    final = final.groupby(['Tumor_Sample_Barcode'], sort=False).apply(f_)\n",
    "\n",
    "    # We separate the different mutations into 5 different columns (5 is the max number of tp53 mutations in our cohort)\n",
    "    final[['mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5']] = final.muts.str.split(',', expand=True)\n",
    "    #final = final.drop(['mut_key_6'],axis=1)\n",
    "    # Split the columns into mut_key_ and vc_\n",
    "    final[['tp53_key_1','tp53_vc_1','tp53_ccf_1','tp53_vaf_1','tp53_HGVSp_1', 'tp53_spot_1']] = final.mut_key_1.str.split('%', expand=True)\n",
    "    final[['tp53_key_2','tp53_vc_2','tp53_ccf_2','tp53_vaf_2','tp53_HGVSp_2', 'tp53_spot_2']] = final.mut_key_2.str.split('%', expand=True)\n",
    "    final[['tp53_key_3','tp53_vc_3','tp53_ccf_3','tp53_vaf_3','tp53_HGVSp_3', 'tp53_spot_3']] = final.mut_key_3.str.split('%', expand=True)\n",
    "    final[['tp53_key_4','tp53_vc_4','tp53_ccf_4','tp53_vaf_4','tp53_HGVSp_4', 'tp53_spot_4']] = final.mut_key_4.str.split('%', expand=True)\n",
    "    final[['tp53_key_5','tp53_vc_5','tp53_ccf_5','tp53_vaf_5','tp53_HGVSp_5', 'tp53_spot_5']] = final.mut_key_5.str.split('%', expand=True)\n",
    "\n",
    "    # We remove the muts column\n",
    "    final = final.drop(['muts','mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5'], axis=1)\n",
    "\n",
    "    # We remove duplicates\n",
    "    final = final.drop_duplicates('Tumor_Sample_Barcode')\n",
    "\n",
    "    # We add the cohort patients that are not tp53 positive\n",
    "    #First we create a dataframe with all missing samples\n",
    "\n",
    "    cohort_samples = set(sample_info.Tumor_Id)\n",
    "    final_samples = set(final.Tumor_Sample_Barcode)\n",
    "    missing_samp = pd.DataFrame(cohort_samples - final_samples, columns = ['Tumor_Sample_Barcode'])\n",
    "    #Then we append the two datframe\n",
    "    final = final.append(missing_samp)\n",
    "\n",
    "    # We rename the Tumor_Sample_Barcode column to have the same key as in other datframes\n",
    "    final = final.rename(columns={'Tumor_Sample_Barcode': 'Tumor_Id'})\n",
    "\n",
    "    # We add a last column tp53_count that represents the number of tp53 mutations per sample\n",
    "    final = final.where(final.notnull(), None)\n",
    "    final['tp53_count'] = final.apply(count_tp53_muts, axis = 1)\n",
    "\n",
    "    # We change the type of vafs column to float64 instead of strings\n",
    "    final = final.astype({'tp53_vaf_1': 'float64', 'tp53_vaf_2': 'float64', 'tp53_vaf_3': 'float64', 'tp53_vaf_4': 'float64', 'tp53_vaf_5': 'float64',\n",
    "                       'tp53_ccf_1': 'float64', 'tp53_ccf_2': 'float64', 'tp53_ccf_3': 'float64', 'tp53_ccf_4': 'float64', 'tp53_ccf_5': 'float64'})\n",
    "\n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909-T01-IM617_7577121_G_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.798</td>\n",
       "      <td>0.312169</td>\n",
       "      <td>p.Arg273Cys</td>\n",
       "      <td>273</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546-T01-IM617_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.845070</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546-T02-IM617_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.636735</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837-T01-IM617_7578406_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.325843</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>175</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>P-0019444-T01-IM6</td>\n",
       "      <td>P-0019444-T01-IM617_7578461_C_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.169002</td>\n",
       "      <td>p.Val157Phe</td>\n",
       "      <td>157</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3190</th>\n",
       "      <td>P-0029543-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3191</th>\n",
       "      <td>P-0023447-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3192</th>\n",
       "      <td>P-0002901-T01-IM3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3193</th>\n",
       "      <td>P-0027887-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3194</th>\n",
       "      <td>P-0012282-T02-IM5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9898 rows Ã— 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Tumor_Id                       tp53_key_1          tp53_vc_1  \\\n",
       "8     P-0036909-T01-IM6  P-0036909-T01-IM617_7577121_G_A  Missense_Mutation   \n",
       "13    P-0023546-T01-IM6  P-0023546-T01-IM617_7578442_T_C  Missense_Mutation   \n",
       "16    P-0023546-T02-IM6  P-0023546-T02-IM617_7578442_T_C  Missense_Mutation   \n",
       "21    P-0018837-T01-IM6  P-0018837-T01-IM617_7578406_C_T  Missense_Mutation   \n",
       "48    P-0019444-T01-IM6  P-0019444-T01-IM617_7578461_C_A  Missense_Mutation   \n",
       "...                 ...                              ...                ...   \n",
       "3190  P-0029543-T01-IM6                             None               None   \n",
       "3191  P-0023447-T01-IM6                             None               None   \n",
       "3192  P-0002901-T01-IM3                             None               None   \n",
       "3193  P-0027887-T01-IM6                             None               None   \n",
       "3194  P-0012282-T02-IM5                             None               None   \n",
       "\n",
       "      tp53_ccf_1  tp53_vaf_1 tp53_HGVSp_1 tp53_spot_1 tp53_key_2 tp53_vc_2  \\\n",
       "8          0.798    0.312169  p.Arg273Cys         273       None      None   \n",
       "13         0.933    0.845070  p.Tyr163Cys         163       None      None   \n",
       "16         1.000    0.636735  p.Tyr163Cys         163       None      None   \n",
       "21         1.000    0.325843  p.Arg175His         175       None      None   \n",
       "48         1.000    0.169002  p.Val157Phe         157       None      None   \n",
       "...          ...         ...          ...         ...        ...       ...   \n",
       "3190         NaN         NaN         None        None       None      None   \n",
       "3191         NaN         NaN         None        None       None      None   \n",
       "3192         NaN         NaN         None        None       None      None   \n",
       "3193         NaN         NaN         None        None       None      None   \n",
       "3194         NaN         NaN         None        None       None      None   \n",
       "\n",
       "      tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2 tp53_spot_2 tp53_key_3 tp53_vc_3  \\\n",
       "8            NaN         NaN         None        None       None      None   \n",
       "13           NaN         NaN         None        None       None      None   \n",
       "16           NaN         NaN         None        None       None      None   \n",
       "21           NaN         NaN         None        None       None      None   \n",
       "48           NaN         NaN         None        None       None      None   \n",
       "...          ...         ...          ...         ...        ...       ...   \n",
       "3190         NaN         NaN         None        None       None      None   \n",
       "3191         NaN         NaN         None        None       None      None   \n",
       "3192         NaN         NaN         None        None       None      None   \n",
       "3193         NaN         NaN         None        None       None      None   \n",
       "3194         NaN         NaN         None        None       None      None   \n",
       "\n",
       "      tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3 tp53_spot_3 tp53_key_4 tp53_vc_4  \\\n",
       "8            NaN         NaN         None        None       None      None   \n",
       "13           NaN         NaN         None        None       None      None   \n",
       "16           NaN         NaN         None        None       None      None   \n",
       "21           NaN         NaN         None        None       None      None   \n",
       "48           NaN         NaN         None        None       None      None   \n",
       "...          ...         ...          ...         ...        ...       ...   \n",
       "3190         NaN         NaN         None        None       None      None   \n",
       "3191         NaN         NaN         None        None       None      None   \n",
       "3192         NaN         NaN         None        None       None      None   \n",
       "3193         NaN         NaN         None        None       None      None   \n",
       "3194         NaN         NaN         None        None       None      None   \n",
       "\n",
       "      tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4 tp53_spot_4 tp53_key_5 tp53_vc_5  \\\n",
       "8            NaN         NaN         None        None       None      None   \n",
       "13           NaN         NaN         None        None       None      None   \n",
       "16           NaN         NaN         None        None       None      None   \n",
       "21           NaN         NaN         None        None       None      None   \n",
       "48           NaN         NaN         None        None       None      None   \n",
       "...          ...         ...          ...         ...        ...       ...   \n",
       "3190         NaN         NaN         None        None       None      None   \n",
       "3191         NaN         NaN         None        None       None      None   \n",
       "3192         NaN         NaN         None        None       None      None   \n",
       "3193         NaN         NaN         None        None       None      None   \n",
       "3194         NaN         NaN         None        None       None      None   \n",
       "\n",
       "      tp53_ccf_5  tp53_vaf_5 tp53_HGVSp_5 tp53_spot_5  tp53_count  \n",
       "8            NaN         NaN         None        None           1  \n",
       "13           NaN         NaN         None        None           1  \n",
       "16           NaN         NaN         None        None           1  \n",
       "21           NaN         NaN         None        None           1  \n",
       "48           NaN         NaN         None        None           1  \n",
       "...          ...         ...          ...         ...         ...  \n",
       "3190         NaN         NaN         None        None           0  \n",
       "3191         NaN         NaN         None        None           0  \n",
       "3192         NaN         NaN         None        None           0  \n",
       "3193         NaN         NaN         None        None           0  \n",
       "3194         NaN         NaN         None        None           0  \n",
       "\n",
       "[9898 rows x 32 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_tp53_muts(cohort=cohort_type)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Copy Numbers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we gather the information from gene_level table.\n",
    "We creaste the following columns:\n",
    "* Sample_Id \n",
    "* tcn --> total copy number\n",
    "* mcn --> major copy number\n",
    "* lcn --> lower copy number\n",
    "* seg_length --> length of the segment\n",
    "* cn_state --> copy number state\n",
    "* cf --> Cell fraction of the cn_state\n",
    "* wgd --> Wholde Genome Doubling (True or False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "# WARNING: THis function needs sample_info to work\n",
    "def create_copy_number_state(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    # We want TP53 locus so we have to filter the gene\n",
    "    gene_level = gene_level[gene_level['gene'] == 'TP53']\n",
    "\n",
    "    gene_level['Tumor_Id'] = gene_level['sample'].str[:17]\n",
    "    gene_level_subset = gene_level[['sample','tcn','mcn','lcn','seg_length','cn_state', 'cf.em']]\n",
    "    \n",
    "    # We rename the cf.em column \n",
    "    gene_level_subset = gene_level_subset.rename(columns={'cf.em': 'tp53_cf', \n",
    "                                                          'sample':'Sample_Id',\n",
    "                                                          'tcn': 'tp53_tcn',\n",
    "                                                          'mcn': 'tp53_mcn',\n",
    "                                                          'lcn': 'tp53_lcn',\n",
    "                                                          'seg_length': 'tp53_seg_length',\n",
    "                                                          'cn_state':'tp53_cn_state'})\n",
    "    \n",
    "    # We add WGD information\n",
    "    wgd = cohort[['sample_id', 'wgd']]\n",
    "    \n",
    "    final = pd.merge(gene_level_subset, wgd, left_on='Sample_Id', right_on='sample_id').drop(['sample_id'], axis=1)\n",
    "    \n",
    "    \n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 19.6 ms, sys: 2.26 ms, total: 21.8 ms\n",
      "Wall time: 20.4 ms\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>tp53_tcn</th>\n",
       "      <th>tp53_mcn</th>\n",
       "      <th>tp53_lcn</th>\n",
       "      <th>tp53_seg_length</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>tp53_cf</th>\n",
       "      <th>wgd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668431</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>0.169170</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18101929</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.306121</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25250470</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.835504</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25237770</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25231975</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.257400</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9893</th>\n",
       "      <td>P-0050750-T01-IM6_P-0050750-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13849975</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.286332</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9894</th>\n",
       "      <td>P-0050643-T01-IM6_P-0050643-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11917531</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.545007</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9895</th>\n",
       "      <td>P-0050223-T02-IM6_P-0050223-N02-IM6</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>80668650</td>\n",
       "      <td>TETRAPLOID</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9896</th>\n",
       "      <td>P-0048760-T01-IM6_P-0048760-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7438283</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.598599</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9897</th>\n",
       "      <td>P-0050744-T01-IM6_P-0050744-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30207093</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.205450</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9898 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id  tp53_tcn  tp53_mcn  tp53_lcn  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6         3       2.0       1.0   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6         2       2.0       0.0   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6         3       3.0       0.0   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6         2       2.0       0.0   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6         3       3.0       0.0   \n",
       "...                                   ...       ...       ...       ...   \n",
       "9893  P-0050750-T01-IM6_P-0050750-N01-IM6         2       2.0       0.0   \n",
       "9894  P-0050643-T01-IM6_P-0050643-N01-IM6         3       3.0       0.0   \n",
       "9895  P-0050223-T02-IM6_P-0050223-N02-IM6         4       2.0       2.0   \n",
       "9896  P-0048760-T01-IM6_P-0048760-N01-IM6         3       3.0       0.0   \n",
       "9897  P-0050744-T01-IM6_P-0050744-N01-IM6         2       2.0       0.0   \n",
       "\n",
       "      tp53_seg_length        tp53_cn_state   tp53_cf   wgd  \n",
       "0            80668431           LOSS AFTER  0.169170  True  \n",
       "1            18101929          LOSS BEFORE  0.306121  True  \n",
       "2            25250470  CNLOH BEFORE & LOSS  0.835504  True  \n",
       "3            25237770          LOSS BEFORE  0.312907  True  \n",
       "4            25231975  CNLOH BEFORE & LOSS  0.257400  True  \n",
       "...               ...                  ...       ...   ...  \n",
       "9893         13849975          LOSS BEFORE  0.286332  True  \n",
       "9894         11917531  CNLOH BEFORE & LOSS  0.545007  True  \n",
       "9895         80668650           TETRAPLOID       NaN  True  \n",
       "9896          7438283  CNLOH BEFORE & LOSS  0.598599  True  \n",
       "9897         30207093          LOSS BEFORE  0.205450  True  \n",
       "\n",
       "[9898 rows x 8 columns]"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time \n",
    "copy_number_info = create_copy_number_state(cohort = cohort_type)\n",
    "copy_number_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computed metrics\n",
    "In this part we define functions to be applied on the master file to compute specific metrics.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes, mutations and max_vaf\n",
    "* *create_gene_count*: count of mutated genes for the given sample\n",
    "* *create_mut_count*: count of mutations for the given sample\n",
    "* *get_max_vaf*: the maximum of Variant Allele Frequency within all mutation of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# These 2 first functions allow  to count the total number of genes/mutations per sample.\n",
    "def create_gene_count(x):\n",
    "    '''\n",
    "    This function create the count of genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    \n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "\n",
    "    nb_genes = selected_cohort.groupby('Entrez_Gene_Id').size().shape[0]\n",
    "\n",
    "    return nb_genes\n",
    "\n",
    "def create_mut_count(x):\n",
    "    '''\n",
    "    This function computes the dataframe of all mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    \n",
    "    return selected_cohort.shape[0]\n",
    "\n",
    "# The following calculates the max_vaf for a given sample\n",
    "def get_max_vaf(x):\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    \n",
    "    return selected_cohort['t_var_freq'].max()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genome Instability (frac_genome_aletered, chr_affected by loss, cnkloh, gain )\n",
    "This script is very long to run, so I decide to run it apart and save the results in a pickle fil that we will merge with our master."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def compute_frac_genome(x):\n",
    "    Sample_Id = x.Sample_Id\n",
    "    wgd_state = x.wgd\n",
    "    lookup_table = arm_level[arm_level['sample'] == Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "        \n",
    "    altered_length = lookup_table_altered.cn_length.sum()\n",
    "    total_length = lookup_table.arm_length.sum()\n",
    "    \n",
    "    frac_gen_altered = round(altered_length/total_length,3)\n",
    "    \n",
    "    return frac_gen_altered\n",
    "\n",
    "# Here is the function that allws to compute genome instability columns\n",
    "def chr_computations(x):\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    \n",
    "    arm_level_samples = list(set(arm_level['sample']))\n",
    "    \n",
    "    if x.Sample_Id not in arm_level_samples:\n",
    "        return ['NaN','NaN','NaN','NaN', 'NaN']\n",
    "    \n",
    "    lookup_table = arm_level[arm_level['sample'] == x.Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "    lookup_table['state_chr'] = lookup_table['cn_state']+lookup_table['chr']\n",
    "    \n",
    "    # If only DIPLOID or TETRAPLOID\n",
    "    if lookup_table.empty:\n",
    "        return [float(0)]*5\n",
    "    \n",
    "    lookup_table['cnloh_chr'] = lookup_table.apply(condition_CNLOH, axis=1)\n",
    "    lookup_table['loss_chr'] = lookup_table.apply(condition_LOSS, axis=1)\n",
    "    lookup_table['gain_chr'] = lookup_table.apply(condition_GAIN, axis=1)\n",
    "\n",
    "    #chr_affected colum\n",
    "    lookup_table_chr = lookup_table.drop_duplicates(subset=['chr'])\n",
    "    chr_affected = len(lookup_table_chr)\n",
    "    \n",
    "    #chr_loss, chr_gain, chr_cnloh columns\n",
    "    lookup_table_cnloh = lookup_table.drop_duplicates(subset=['cnloh_chr'])['cnloh_chr']\n",
    "    lookup_table_loss = lookup_table.drop_duplicates(subset=['loss_chr'])['loss_chr']\n",
    "    lookup_table_gain = lookup_table.drop_duplicates(subset=['gain_chr'])['gain_chr']\n",
    "\n",
    "    chr_loss = len(lookup_table_loss[lookup_table_loss.str.startswith('LOSS')])\n",
    "    chr_gain = len(lookup_table_gain[lookup_table_gain.str.startswith('GAIN')])\n",
    "    chr_cnloh = len(lookup_table_cnloh[lookup_table_cnloh.str.startswith('CNLOH')])\n",
    "    \n",
    "    #frac_gen_altered column\n",
    "    frac_gen_altered = compute_frac_genome(x, arm_level)\n",
    "    \n",
    "    return [chr_affected, chr_loss, chr_gain, chr_cnloh, frac_gen_altered]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TP53 Residual"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following function needs to be called on the complete master file because it needs info from different parts\n",
    "# It computes the expected number of tp53 mutant copies in a cell\n",
    "def create_copies_tp53_muts(master):\n",
    "    master['tp53_exp_nb_1'] = master.apply(lambda x:(x.tp53_vaf_1 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_2'] = master.apply(lambda x:(x.tp53_vaf_2 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_3'] = master.apply(lambda x:(x.tp53_vaf_3 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_4'] = master.apply(lambda x:(x.tp53_vaf_4 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_5'] = master.apply(lambda x:(x.tp53_vaf_5 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    \n",
    "    return master\n",
    "\n",
    "\n",
    "# The following computes the expected number of copies of tp53 residuals \n",
    "def create_tp53_res(master):\n",
    "    master['tp53_res_1'] = master.apply(lambda x:x.tcn - x.tp53_exp_nb_1, axis = 1)\n",
    "    master['tp53_res_2'] = master.apply(lambda x:x.tcn - x.tp53_exp_nb_2, axis = 1)\n",
    "    master['tp53_res_3'] = master.apply(lambda x:x.tcn - x.tp53_exp_nb_3, axis = 1)\n",
    "    master['tp53_res_4'] = master.apply(lambda x:x.tcn - x.tp53_exp_nb_4, axis = 1)\n",
    "    master['tp53_res_5'] = master.apply(lambda x:x.tcn - x.tp53_exp_nb_5, axis = 1)\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mutation Types Grouping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following functions allow to group the Mutation Types\n",
    "def vc_group_cond_1(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_1 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_1 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_1 in missense: \n",
    "        if x.tp53_spot_1 in ['273','248','175']: return x.tp53_spot_1\n",
    "        elif x.tp53_spot_1 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_2(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_2 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_2 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_2 in missense: \n",
    "        if x.tp53_spot_2 in ['273','248','175']: return x.tp53_spot_2\n",
    "        elif x.tp53_spot_2 in['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'   \n",
    "def vc_group_cond_3(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_3 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_3 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_3 in missense: \n",
    "        if x.tp53_spot_3 in ['273','248','175']: return x.tp53_spot_3\n",
    "        elif x.tp53_spot_3 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense' \n",
    "def vc_group_cond_4(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_4 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_4 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_4 in missense: \n",
    "        if x.tp53_spot_4 in ['273','248','175']: return x.tp53_spot_4\n",
    "        elif x.tp53_spot_4 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_5(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_5 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_5 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_5 in missense: \n",
    "        if x.tp53_spot_5 in ['273','248','175']: return x.tp53_spot_5\n",
    "        elif x.tp53_spot_5 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Non-WGD specific functions\n",
    "\n",
    "* *cn_group_cond*\n",
    "\n",
    "First, we group the different Copy Number States *cn_state* in subgroups, under the column *cn_group*:\n",
    "    * Group 1: cnLOH gathering ['CNLOH']\n",
    "    * Group 2: LOSS gathering ['HETLOSS']\n",
    "    * Group 3: HOMDEL gathering ['HOMDEL']\n",
    "    * Group 4: WILD_TYPE gathering ['DIPLOID', 'TETRAPLOID']\n",
    "    * Group 5: GAIN gathering ['GAIN']\n",
    "    * Group 6: OTHER gathering ['AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']\n",
    "\n",
    "* *mut_cn_group_cond*\n",
    "\n",
    "Based on this first column we define 7 final groups of patients adding the mutational information. These groups will be under the column *tp53_group*.\n",
    "    * Group 1: Samples with 0 tp53 mutations and HETLOSS\n",
    "    * Group 2: Samples with HOMDEL\n",
    "    * Group 3: Samples with 1 tp53 mutation and WILD_TYPE (DIPLOID, LOSS AFTER, TETRAPLOID)\n",
    "    * Group 4: Samples with 1 tp53 mutation or more and LOSS\n",
    "    * Group 5: Samples with 1 tp53 mutation or more and cnLOH\n",
    "    * Group 6: Samples with 2/3/4/5 tp53 mutations and WILD_TYPE or GAIN\n",
    "\n",
    "* *tp53_residual_group*: defines if we have residual tp53 or not based on the expected WT tp53 residual\n",
    "\n",
    "\n",
    "* *get_loh_nowgd*\n",
    "\n",
    "The last function is meant to compute the LOH state"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [],
   "source": [
    "def cn_group_cond(x):\n",
    "    if x.tp53_cn_state in ['CNLOH']:\n",
    "        return 'cnLOH'\n",
    "    if x.tp53_cn_state in ['HETLOSS']:\n",
    "        return 'LOSS'\n",
    "    if x.tp53_cn_state == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_cn_state in ['DIPLOID', 'TETRAPLOID']:\n",
    "        return 'WILD_TYPE'\n",
    "    if x.tp53_cn_state == 'GAIN':\n",
    "        return 'GAIN'\n",
    "    if x.tp53_cn_state in ['AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']:\n",
    "        return 'OTHER'\n",
    "\n",
    "def mut_cn_group_cond(x):\n",
    "    if x.tp53_cn_state == 'HETLOSS' and x.tp53_count == 0:\n",
    "        return '0_HETLOSS'\n",
    "    if x.tp53_first_group == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_first_group == 'WILD_TYPE' and x.tp53_count == 1 :\n",
    "        return '1_WILD_TYPE'\n",
    "    if x.tp53_first_group == 'LOSS' and x.tp53_count >=1:\n",
    "        return '>=1_LOSS'\n",
    "    if x.tp53_first_group == 'cnLOH' and x.tp53_count >=1:\n",
    "        return '>=1_cnLOH'\n",
    "    if (x.tp53_first_group == 'WILD_TYPE' or x.tp53_first_group == 'GAIN') and x.tp53_count > 1:\n",
    "        return '>1muts'\n",
    "    \n",
    "    \n",
    "def tp53_residual_group(x):\n",
    "    if x.tp53_group == '1_WILD_TYPE' or x.tp53_group == '0_HETLOSS':\n",
    "        return 'tp53_res'\n",
    "    if x.tp53_group == 'HOMDEL':\n",
    "        return 'no_tp53_res'\n",
    "    if x.tp53_group == '>=1_LOSS' or x.tp53_group == '>=1_cnLOH':\n",
    "        if (x.tp53_residual_1 < 0.5) or (x.tp53_residual_2 < 0.5):\n",
    "            return 'no_tp53_res'\n",
    "        else:\n",
    "            if (x.tp53_cf + max(x.tp53_ccf_1, x.tp53_ccf_2, x.tp53_ccf_3, x.tp53_ccf_4, x.tp53_ccf_5)) > 1:\n",
    "                return 'no_tp53_res'\n",
    "            else:\n",
    "                return 'uncertain'\n",
    "    if x.tp53_group == '>1muts':\n",
    "        if (x.tp53_residual_1 + x.tp53_residual_2 < 2.5):\n",
    "            if (x.tp53_ccf_1 + x.tp53_ccf_2 > 1):\n",
    "                return 'no_tp53_res'\n",
    "            else: \n",
    "                return 'uncertain'\n",
    "           \n",
    "        elif (x.tp53_residual_1 + x.tp53_residual_2 > 2.5):\n",
    "            return 'tp53_res'\n",
    "        \n",
    "        \n",
    "def get_loh_nowgd(x):\n",
    "    if tp53_cn_state in ['CNLOH', 'CNLOH & GAIN', 'AMP (LOH)']:\n",
    "        return True\n",
    "    else: return False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WGD specific functions\n",
    "Finally we have functions to compute TP53 allelic state BEFORE WGD (for the WGD cohort only):\n",
    "* *get_bi_nobi* allows to say if the pre WGD allelic state was Bi-Allelic or not\n",
    "* *get_mono* allows to say if, within Not Bi-allelic samples, the state was mono-allelic or with 2 WT alleles\n",
    "* *get_loh_wgd* that computes if the samples is in LOH state or not"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "            NUMBER OF TP53 residual ASSOCIATED WITH ALLELIC STATE BEFORE WGD\n",
    "            \n",
    "            \n",
    "                                     BI-ALLELIC | MONO - ALLELIC | 2WT\n",
    "\n",
    "                LOSS BEFORE :            0      |        1       | -\n",
    "                CNLOH BEFORE & LOSS:     0      |       1,2      | 2\n",
    "                CNLOH BEFORE:            0      |        2       | 3\n",
    "                LOSS AFTER:              -      |       1,2      | 2\n",
    "                DOUBLE LOSS AFTER:       -      |       0,1      | 1\n",
    "                TETRAPLOID:              -      |        2       | 3\n",
    "                CNLOH AFTER:             -      |       1,3      | 3\n",
    "                CNLOH BEFORE & GAIN:     0      |        2,3     | 3,4\n",
    "                \n",
    "\n",
    "                        THRESHOLDS BETWEEN ALLELIC STATES BEFORE WGD\n",
    "                    \n",
    "                                     BI-ALLELIC | MONO - ALLELIC | 2WT\n",
    "\n",
    "                LOSS BEFORE :            <0.4   |        >0.6    | -\n",
    "                CNLOH BEFORE & LOSS:     <0.4   |      0.6< <1.5 | uncertain\n",
    "                CNLOH BEFORE:            <1.5   |   1.5< <2.5    | >2.5\n",
    "                LOSS AFTER:              -      |      <1.5      | uncertain \n",
    "                DOUBLE LOSS AFTER:       -      |       <0.5     | uncertain\n",
    "                TETRAPLOID:              -      |       <2.5     | >2.5\n",
    "                CNLOH AFTER:             -      |       <1.5     | uncertain\n",
    "                CNLOH BEFORE & GAIN:     <1.4   |   1.6< <2.5    | uncertain\n",
    "                \n",
    "    \n",
    "\n",
    " '''\n",
    "\n",
    "\n",
    "# The following functions are for WGD cohort, computing the tp53 allelic state before WGD\n",
    "def get_bi_nobi(x):\n",
    "    tumor = x.tumor_sample\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    \n",
    "    # Samples with only one tp53 mutation\n",
    "    if nb_tp53muts == 1:\n",
    "        tp53_res = maf_muts.tp53_res.values[0]\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE' or cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr = 0.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr:\n",
    "                return 'bi'\n",
    "            elif tp53_res >= thr:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state in ['LOSS AFTER','DOUBLE LOSS AFTER','TETRAPLOID','CNLOH AFTER']: \n",
    "            return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "            \n",
    "            \n",
    "            \n",
    "    # Samples with 2 tp53 mutations\n",
    "    elif nb_tp53muts == 2:\n",
    "        tp53_res_1 = maf_muts.tp53_res.values[0]\n",
    "        tp53_res_2 = maf_muts.tp53_res.values[1]\n",
    "        timing_1 = maf_muts.tp53_timing_wgd.values[0]\n",
    "        timing_2 = maf_muts.tp53_timing_wgd.values[1]\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr - 0.1) or (tp53_res_2 < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr + 0.1) and (tp53_res_2 > thr + 0.1):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (tp53_res_1 < thr_1 - 0.1) or (tp53_res_2 < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (tp53_res_1 < thr_1) or (tp53_res_2 < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if tp53_res_1 < thr_1 - 0.1 or tp53_res_2 < thr_1 - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr_2) and (tp53_res_2 > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "            \n",
    "    # Samples with 3 tp53 mutations\n",
    "    elif nb_tp53muts == 3:\n",
    "        tp53_res_1 = maf_muts.tp53_res.values[0]\n",
    "        tp53_res_2 = maf_muts.tp53_res.values[1]\n",
    "        tp53_res_3 = maf_muts.tp53_res.values[2]\n",
    "        tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "        tp53_res.sort()\n",
    "        \n",
    "        timing_1 = maf_muts.tp53_timing_wgd.values[0]\n",
    "        timing_2 = maf_muts.tp53_timing_wgd.values[1]\n",
    "        timing_3 = maf_muts.tp53_timing_wgd.values[2]\n",
    "        timings = [timing_1, timing_2, timing_3]\n",
    "        BEF = [1 for x in timings if x=='BEFORE']\n",
    "        \n",
    "        # Number of mutations happening before wgd\n",
    "        nb_bef = len(BEF)\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (min(tp53_res) < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (max(tp53_res) > thr + 0.1) :\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (min(tp53_res) < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2): # we want that exactly 2 mutation arose before WGD\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (min(tp53_res) < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if min(tp53_res) < thr_1 - 0.1 :\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "    \n",
    "    else: return 'uncertain'\n",
    "\n",
    "\n",
    "# This functions can only be called after get_bi_nobi() has been called on master\n",
    "def get_mono(x):\n",
    "    tumor = x.tumor_sample\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    bi_state = x.bi_state_wgd\n",
    "    \n",
    "    if bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    \n",
    "    elif bi_state == 'uncertain':\n",
    "        return 'uncertain_bi'\n",
    "    \n",
    "    \n",
    "    elif bi_state == 'no_bi': # We are already in the  mono/2WT distinction \n",
    "        \n",
    "        #1 mut samples\n",
    "        if nb_tp53muts == 1:\n",
    "            tp53_res = maf_muts.tp53_res.values[0]\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "\n",
    "        \n",
    "        # 2 mut samples   \n",
    "        elif nb_tp53muts == 2:\n",
    "            tp53_res_1 = maf_muts.tp53_res.values[0]\n",
    "            tp53_res_2 = maf_muts.tp53_res.values[1]\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res_1 < 2.5 or tp53_res_2 < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res_1 > 2.5 and tp53_res_2 > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "            \n",
    "        # 3 mut samples\n",
    "        elif nb_tp53muts == 3:\n",
    "            tp53_res_1 = maf_muts.tp53_res.values[0]\n",
    "            tp53_res_2 = maf_muts.tp53_res.values[1]\n",
    "            tp53_res_3 = maf_muts.tp53_res.values[2]\n",
    "            tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "            tp53_res.sort()\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if min(tp53_res) < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res[0] > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "        \n",
    "        else: return 'uncertain_mono' \n",
    "        \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge Tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "def merge_tables(cohort_type: str):\n",
    "    \n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    sample_info = create_sample_info(cohort=cohort_type)\n",
    "    tp53_muts = create_tp53_muts(cohort=cohort_type)\n",
    "    copy_number_info = create_copy_number_state(cohort=cohort_type)\n",
    "    \n",
    "    master = pd.merge(sample_info, tp53_muts, on='Tumor_Id')\n",
    "    master = pd.merge(master, copy_number_info, on='Sample_Id')\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 5.65 s, sys: 71.6 ms, total: 5.73 s\n",
      "Wall time: 5.7 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "master = merge_tables(cohort_type=cohort_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_max_vaf(x):\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    \n",
    "    return selected_cohort['t_var_freq'].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "metadata": {},
   "outputs": [],
   "source": [
    "master['gene_count'] = master.apply(create_gene_count, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>Overall_Survival_Months</th>\n",
       "      <th>Overall_Survival_Status</th>\n",
       "      <th>MSI_Score</th>\n",
       "      <th>MSI_Type</th>\n",
       "      <th>TMB_Score</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_count</th>\n",
       "      <th>tp53_tcn</th>\n",
       "      <th>tp53_mcn</th>\n",
       "      <th>tp53_lcn</th>\n",
       "      <th>tp53_seg_length</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>tp53_cf</th>\n",
       "      <th>wgd</th>\n",
       "      <th>mut_count</th>\n",
       "      <th>gene_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>P-0025956</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>71.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>3.496971</td>\n",
       "      <td>3.584</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668431</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>0.169170</td>\n",
       "      <td>True</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.391316</td>\n",
       "      <td>2.871793</td>\n",
       "      <td>14.137</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "      <td>P-0036909-T01-IM617_7577121_G_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.798</td>\n",
       "      <td>0.312169</td>\n",
       "      <td>p.Arg273Cys</td>\n",
       "      <td>273</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18101929</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.306121</td>\n",
       "      <td>True</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Neuroendocrine Carcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.865628</td>\n",
       "      <td>3.115253</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>2.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "      <td>P-0023546-T01-IM617_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.845070</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25250470</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.835504</td>\n",
       "      <td>True</td>\n",
       "      <td>4</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>3.136841</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.82</td>\n",
       "      <td>Stable</td>\n",
       "      <td>2.6</td>\n",
       "      <td>P-0023546-T02-IM617_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.636735</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25237770</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>True</td>\n",
       "      <td>3</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Colon Adenocarcinoma</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.351778</td>\n",
       "      <td>3.727190</td>\n",
       "      <td>34.060</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.45</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>P-0018837-T01-IM617_7578406_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.325843</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>175</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25231975</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.257400</td>\n",
       "      <td>True</td>\n",
       "      <td>6</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9893</th>\n",
       "      <td>P-0050750-T01-IM6_P-0050750-N01-IM6</td>\n",
       "      <td>P-0050750-T01-IM6</td>\n",
       "      <td>P-0050750</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.301523</td>\n",
       "      <td>3.845451</td>\n",
       "      <td>0.099</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>P-0050750-T01-IM617_7578406_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0.285229</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>175</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13849975</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.286332</td>\n",
       "      <td>True</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9894</th>\n",
       "      <td>P-0050643-T01-IM6_P-0050643-N01-IM6</td>\n",
       "      <td>P-0050643-T01-IM6</td>\n",
       "      <td>P-0050643</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.667936</td>\n",
       "      <td>4.419218</td>\n",
       "      <td>1.315</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.30</td>\n",
       "      <td>Stable</td>\n",
       "      <td>4.4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11917531</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.545007</td>\n",
       "      <td>True</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9895</th>\n",
       "      <td>P-0050223-T02-IM6_P-0050223-N02-IM6</td>\n",
       "      <td>P-0050223-T02-IM6</td>\n",
       "      <td>P-0050223</td>\n",
       "      <td>Hepatobiliary Cancer</td>\n",
       "      <td>Cholangiocarcinoma</td>\n",
       "      <td>67.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.215128</td>\n",
       "      <td>4.725790</td>\n",
       "      <td>1.940</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>80668650</td>\n",
       "      <td>TETRAPLOID</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>4</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9896</th>\n",
       "      <td>P-0048760-T01-IM6_P-0048760-N01-IM6</td>\n",
       "      <td>P-0048760-T01-IM6</td>\n",
       "      <td>P-0048760</td>\n",
       "      <td>Bone Cancer</td>\n",
       "      <td>Osteosarcoma</td>\n",
       "      <td>18.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.614603</td>\n",
       "      <td>3.947721</td>\n",
       "      <td>3.682</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>5.79</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>6.1</td>\n",
       "      <td>P-0048760-T01-IM617_7578413_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.958</td>\n",
       "      <td>0.675637</td>\n",
       "      <td>p.Val173Met</td>\n",
       "      <td>173</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7438283</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.598599</td>\n",
       "      <td>True</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9897</th>\n",
       "      <td>P-0050744-T01-IM6_P-0050744-N01-IM6</td>\n",
       "      <td>P-0050744-T01-IM6</td>\n",
       "      <td>P-0050744</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>69.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>2.426452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.06</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>P-0050744-T01-IM617_7577580_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.797</td>\n",
       "      <td>0.239130</td>\n",
       "      <td>p.Tyr234Cys</td>\n",
       "      <td>234</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30207093</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.205450</td>\n",
       "      <td>True</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9898 rows Ã— 54 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6  P-0025956-T01-IM6  P-0025956   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6  P-0036909-T01-IM6  P-0036909   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6  P-0023546-T01-IM6  P-0023546   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6  P-0023546-T02-IM6  P-0023546   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6  P-0018837-T01-IM6  P-0018837   \n",
       "...                                   ...                ...        ...   \n",
       "9893  P-0050750-T01-IM6_P-0050750-N01-IM6  P-0050750-T01-IM6  P-0050750   \n",
       "9894  P-0050643-T01-IM6_P-0050643-N01-IM6  P-0050643-T01-IM6  P-0050643   \n",
       "9895  P-0050223-T02-IM6_P-0050223-N02-IM6  P-0050223-T02-IM6  P-0050223   \n",
       "9896  P-0048760-T01-IM6_P-0048760-N01-IM6  P-0048760-T01-IM6  P-0048760   \n",
       "9897  P-0050744-T01-IM6_P-0050744-N01-IM6  P-0050744-T01-IM6  P-0050744   \n",
       "\n",
       "                     Cancer_Type               Cancer_Type_Detailed  \\\n",
       "0     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "1     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "2                Prostate Cancer  Prostate Neuroendocrine Carcinoma   \n",
       "3                Prostate Cancer            Prostate Adenocarcinoma   \n",
       "4              Colorectal Cancer               Colon Adenocarcinoma   \n",
       "...                          ...                                ...   \n",
       "9893           Pancreatic Cancer          Pancreatic Adenocarcinoma   \n",
       "9894               Breast Cancer   Breast Invasive Ductal Carcinoma   \n",
       "9895        Hepatobiliary Cancer                 Cholangiocarcinoma   \n",
       "9896                 Bone Cancer                       Osteosarcoma   \n",
       "9897           Pancreatic Cancer          Pancreatic Adenocarcinoma   \n",
       "\n",
       "      Patient_Current_Age Sample_Type    purity    ploidy  \\\n",
       "0                    71.0     Primary  0.273767  3.496971   \n",
       "1                    47.0  Metastasis  0.391316  2.871793   \n",
       "2                    50.0     Primary  0.865628  3.115253   \n",
       "3                    50.0     Primary  0.312907  3.136841   \n",
       "4                    60.0     Primary  0.351778  3.727190   \n",
       "...                   ...         ...       ...       ...   \n",
       "9893                 55.0  Metastasis  0.301523  3.845451   \n",
       "9894                 53.0     Primary  0.667936  4.419218   \n",
       "9895                 67.0     Primary  0.215128  4.725790   \n",
       "9896                 18.0     Primary  0.614603  3.947721   \n",
       "9897                 69.0     Primary  0.300000  2.426452   \n",
       "\n",
       "      Overall_Survival_Months Overall_Survival_Status  MSI_Score  \\\n",
       "0                       3.584                DECEASED       0.00   \n",
       "1                      14.137                  LIVING       0.37   \n",
       "2                       4.800                DECEASED       2.37   \n",
       "3                       4.800                DECEASED       0.82   \n",
       "4                      34.060                  LIVING       0.45   \n",
       "...                       ...                     ...        ...   \n",
       "9893                    0.099                  LIVING       0.05   \n",
       "9894                    1.315                  LIVING       0.30   \n",
       "9895                    1.940                  LIVING       0.00   \n",
       "9896                    3.682                  LIVING       5.79   \n",
       "9897                      NaN                  LIVING       0.06   \n",
       "\n",
       "           MSI_Type  TMB_Score                       tp53_key_1  \\\n",
       "0            Stable        5.3                             None   \n",
       "1            Stable        3.5  P-0036909-T01-IM617_7577121_G_A   \n",
       "2            Stable        3.5  P-0023546-T01-IM617_7578442_T_C   \n",
       "3            Stable        2.6  P-0023546-T02-IM617_7578442_T_C   \n",
       "4            Stable        5.3  P-0018837-T01-IM617_7578406_C_T   \n",
       "...             ...        ...                              ...   \n",
       "9893         Stable        5.3  P-0050750-T01-IM617_7578406_C_T   \n",
       "9894         Stable        4.4                             None   \n",
       "9895         Stable        3.5                             None   \n",
       "9896  Indeterminate        6.1  P-0048760-T01-IM617_7578413_C_T   \n",
       "9897         Stable        5.3  P-0050744-T01-IM617_7577580_T_C   \n",
       "\n",
       "              tp53_vc_1  tp53_ccf_1  tp53_vaf_1 tp53_HGVSp_1 tp53_spot_1  \\\n",
       "0                  None         NaN         NaN         None        None   \n",
       "1     Missense_Mutation       0.798    0.312169  p.Arg273Cys         273   \n",
       "2     Missense_Mutation       0.933    0.845070  p.Tyr163Cys         163   \n",
       "3     Missense_Mutation       1.000    0.636735  p.Tyr163Cys         163   \n",
       "4     Missense_Mutation       1.000    0.325843  p.Arg175His         175   \n",
       "...                 ...         ...         ...          ...         ...   \n",
       "9893  Missense_Mutation       0.946    0.285229  p.Arg175His         175   \n",
       "9894               None         NaN         NaN         None        None   \n",
       "9895               None         NaN         NaN         None        None   \n",
       "9896  Missense_Mutation       0.958    0.675637  p.Val173Met         173   \n",
       "9897  Missense_Mutation       0.797    0.239130  p.Tyr234Cys         234   \n",
       "\n",
       "     tp53_key_2 tp53_vc_2  tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2 tp53_spot_2  \\\n",
       "0          None      None         NaN         NaN         None        None   \n",
       "1          None      None         NaN         NaN         None        None   \n",
       "2          None      None         NaN         NaN         None        None   \n",
       "3          None      None         NaN         NaN         None        None   \n",
       "4          None      None         NaN         NaN         None        None   \n",
       "...         ...       ...         ...         ...          ...         ...   \n",
       "9893       None      None         NaN         NaN         None        None   \n",
       "9894       None      None         NaN         NaN         None        None   \n",
       "9895       None      None         NaN         NaN         None        None   \n",
       "9896       None      None         NaN         NaN         None        None   \n",
       "9897       None      None         NaN         NaN         None        None   \n",
       "\n",
       "     tp53_key_3 tp53_vc_3  tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3 tp53_spot_3  \\\n",
       "0          None      None         NaN         NaN         None        None   \n",
       "1          None      None         NaN         NaN         None        None   \n",
       "2          None      None         NaN         NaN         None        None   \n",
       "3          None      None         NaN         NaN         None        None   \n",
       "4          None      None         NaN         NaN         None        None   \n",
       "...         ...       ...         ...         ...          ...         ...   \n",
       "9893       None      None         NaN         NaN         None        None   \n",
       "9894       None      None         NaN         NaN         None        None   \n",
       "9895       None      None         NaN         NaN         None        None   \n",
       "9896       None      None         NaN         NaN         None        None   \n",
       "9897       None      None         NaN         NaN         None        None   \n",
       "\n",
       "     tp53_key_4 tp53_vc_4  tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4 tp53_spot_4  \\\n",
       "0          None      None         NaN         NaN         None        None   \n",
       "1          None      None         NaN         NaN         None        None   \n",
       "2          None      None         NaN         NaN         None        None   \n",
       "3          None      None         NaN         NaN         None        None   \n",
       "4          None      None         NaN         NaN         None        None   \n",
       "...         ...       ...         ...         ...          ...         ...   \n",
       "9893       None      None         NaN         NaN         None        None   \n",
       "9894       None      None         NaN         NaN         None        None   \n",
       "9895       None      None         NaN         NaN         None        None   \n",
       "9896       None      None         NaN         NaN         None        None   \n",
       "9897       None      None         NaN         NaN         None        None   \n",
       "\n",
       "     tp53_key_5 tp53_vc_5  tp53_ccf_5  tp53_vaf_5 tp53_HGVSp_5 tp53_spot_5  \\\n",
       "0          None      None         NaN         NaN         None        None   \n",
       "1          None      None         NaN         NaN         None        None   \n",
       "2          None      None         NaN         NaN         None        None   \n",
       "3          None      None         NaN         NaN         None        None   \n",
       "4          None      None         NaN         NaN         None        None   \n",
       "...         ...       ...         ...         ...          ...         ...   \n",
       "9893       None      None         NaN         NaN         None        None   \n",
       "9894       None      None         NaN         NaN         None        None   \n",
       "9895       None      None         NaN         NaN         None        None   \n",
       "9896       None      None         NaN         NaN         None        None   \n",
       "9897       None      None         NaN         NaN         None        None   \n",
       "\n",
       "      tp53_count  tp53_tcn  tp53_mcn  tp53_lcn  tp53_seg_length  \\\n",
       "0              0         3       2.0       1.0         80668431   \n",
       "1              1         2       2.0       0.0         18101929   \n",
       "2              1         3       3.0       0.0         25250470   \n",
       "3              1         2       2.0       0.0         25237770   \n",
       "4              1         3       3.0       0.0         25231975   \n",
       "...          ...       ...       ...       ...              ...   \n",
       "9893           1         2       2.0       0.0         13849975   \n",
       "9894           0         3       3.0       0.0         11917531   \n",
       "9895           0         4       2.0       2.0         80668650   \n",
       "9896           1         3       3.0       0.0          7438283   \n",
       "9897           1         2       2.0       0.0         30207093   \n",
       "\n",
       "            tp53_cn_state   tp53_cf   wgd  mut_count  gene_count  \n",
       "0              LOSS AFTER  0.169170  True          6           6  \n",
       "1             LOSS BEFORE  0.306121  True          4           3  \n",
       "2     CNLOH BEFORE & LOSS  0.835504  True          4           4  \n",
       "3             LOSS BEFORE  0.312907  True          3           3  \n",
       "4     CNLOH BEFORE & LOSS  0.257400  True          6           6  \n",
       "...                   ...       ...   ...        ...         ...  \n",
       "9893          LOSS BEFORE  0.286332  True          5           5  \n",
       "9894  CNLOH BEFORE & LOSS  0.545007  True          5           5  \n",
       "9895           TETRAPLOID       NaN  True          4           3  \n",
       "9896  CNLOH BEFORE & LOSS  0.598599  True          7           7  \n",
       "9897          LOSS BEFORE  0.205450  True          5           5  \n",
       "\n",
       "[9898 rows x 54 columns]"
      ]
     },
     "execution_count": 141,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.31216931216931204"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_cohort_wgd[maf_cohort_wgd['Tumor_Sample_Barcode'] == 'P-0036909-T01-IM6']['t_var_freq'].max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 137,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6"
      ]
     },
     "execution_count": 137,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == 'P-0025956-T01-IM6']\n",
    "selected_cohort.groupby('Entrez_Gene_Id').size().shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "336px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
