{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <span style='font-family:\"Times New Roman\"'> <span styel=''> **MASTER FILE CREATION**\n",
    "\n",
    "*Emile Cohen* \n",
    "    \n",
    "*May 2020*\n",
    "\n",
    "**Goal:** In this Notebook, we create a master file that summarizes all useful information.\n",
    "\n",
    "The Notebook is divided in 4 parts, representing the four parts of our Master file:\n",
    "   \n",
    "* **1. Patient/Sample Information**\n",
    "* **2. TP53 Mutations**\n",
    "* **3. TP53 Copy Numbers**\n",
    "* **4. TP53 Computed Metrics**\n",
    "* **5. Subgroup columns creation**\n",
    "* **6. Merge tables**\n",
    "\n",
    "**NB1:** In each part, you must run the cells from the begining in order to initialize the variables\n",
    "\n",
    "**NB2:** In order to launch the last script (Merge Tables), you have to define the functions in each part.\n",
    "\n",
    "**NB3:** All functions used for the plots are located in utils/custom_tools.py\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Setup environment... done!\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "<span style=\"color:green\">✅ Working on **mskimpact_env** conda environment.</span>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%run -i '../../../utils/setup_environment.ipy'\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "from scipy.stats import fisher_exact, ranksums, chi2, norm\n",
    "from statsmodels.sandbox.stats.multicomp import multipletests\n",
    "import matplotlib.gridspec as gridspec\n",
    "import pickle\n",
    "\n",
    "data_path = '../../../data/'\n",
    "data_wgd = data_path + 'impact-facets-tp53/processed/wgd/'\n",
    "data_no_wgd = data_path + 'impact-facets-tp53/processed/no_wgd/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "# first we load all files for WGD\n",
    "maf_cohort_wgd = pd.read_csv(data_wgd + 'maf_cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_wgd = pd.read_csv(data_wgd + 'cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_wgd = pd.read_csv(data_wgd + 'arm_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_wgd = pd.read_csv(data_wgd + 'gene_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We load all files for non WGD\n",
    "\n",
    "maf_cohort_nowgd = pd.read_csv(data_no_wgd + 'maf_cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_nowgd = pd.read_csv(data_no_wgd + 'cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_nowgd = pd.read_csv(data_no_wgd + 'arm_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_nowgd = pd.read_csv(data_no_wgd + 'gene_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def removing_germ_na_mut(maf_cohort: pd.DataFrame):\n",
    "    '''\n",
    "    This function is meant to remove non TP53 GERMLINE mutations from the mafs, to have only SOMATIC mut \n",
    "    except for TP53 muts where we allow GERMLINE status.\n",
    "    '''\n",
    "    maf_cohort_final = maf_cohort[maf_cohort['Mutation_Status'] != 'UNKNOWN']\n",
    "\n",
    "    key_list = list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] != 'TP53'][maf_cohort_final['Mutation_Status'] != 'GERMLINE'].sample_mut_key)\n",
    "    key_list = key_list + list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] == 'TP53'].sample_mut_key)\n",
    "    maf_cohort_final = maf_cohort_final[maf_cohort_final.sample_mut_key.isin(key_list)]\n",
    "    \n",
    "    return maf_cohort_final\n",
    "\n",
    "maf_cohort_wgd = removing_germ_na_mut(maf_cohort_wgd)\n",
    "maf_cohort_nowgd = removing_germ_na_mut(maf_cohort_nowgd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load clinical data\n",
    "clinical_data = pd.read_csv(data_path + 'cbioportal/raw/mskimpact_clinical_data-2.tsv', sep= '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filtering the clinical data\n",
    "samples_wgd = list(set(cohort_wgd.tumor_sample))\n",
    "samples_nowgd = list(set(cohort_nowgd.tumor_sample))\n",
    "\n",
    "clinical_wgd = clinical_data[clinical_data['Sample ID'].isin(samples_wgd)]\n",
    "clinical_nowgd = clinical_data[clinical_data['Sample ID'].isin(samples_nowgd)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IMPORTANT: Parameter Definition\n",
    "In this script we can create two master files: one for WGD samples and one for non WGD samples.\n",
    "So this parameter allows to select the type of master you want.\n",
    "On top of that you can specify where you want to store the file you created"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_type = 'wgd'\n",
    "\n",
    "\n",
    "\n",
    "output_path = data_path + 'impact-facets-tp53/processed/'\n",
    "\n",
    "total_output = output_path + cohort_type + '/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "# Patient/Sample Information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we focus on clinical information exported from CbioPortal.\n",
    "\n",
    "The following columns are selected:\n",
    "* Sample_Id\n",
    "* Tumor_Id\n",
    "* Patient_Id\n",
    "* Patient Current Age\n",
    "* Cancer_Type\n",
    "* Cancer_Type_Detailed\n",
    "* Sample_Type\n",
    "* purity\n",
    "* ploidy\n",
    "* Overall Survival Status\n",
    "* Overall Survival (Months)\n",
    "* MSI Score\n",
    "* MSI Type\n",
    "* Tumor Mutational Burden"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_sample_info(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    cohort_filt = cohort[['sample_id', 'tumor_sample', 'patient', 'ploidy']]\n",
    "    clinical_filt = clinical[['Sample ID', 'Patient Current Age', 'Sex', 'Cancer Type', 'Cancer Type Detailed', 'Sample Type',\n",
    "                  'Overall Survival (Months)', 'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "    purity = maf_cohort.drop_duplicates('Tumor_Sample_Barcode')[['Tumor_Sample_Barcode', 'purity']]\n",
    "\n",
    "    # Merging these files\n",
    "    sample_info = pd.merge(cohort_filt, purity, how='left', left_on='tumor_sample', right_on='Tumor_Sample_Barcode')\n",
    "    sample_info =  pd.merge(sample_info, clinical_filt, left_on='tumor_sample', right_on='Sample ID').drop(['Tumor_Sample_Barcode', 'Sample ID'], axis=1)\n",
    "\n",
    "    sample_info = sample_info[['sample_id', 'tumor_sample', 'patient', 'Cancer Type', 'Cancer Type Detailed',\n",
    "                               'Patient Current Age', 'Sex','Sample Type', 'purity','ploidy','Overall Survival (Months)', \n",
    "                               'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "\n",
    "    sample_info.columns = ['Sample_Id', 'Tumor_Id', 'Patient_Id','Cancer_Type', 'Cancer_Type_Detailed', 'Patient_Current_Age',\n",
    "                          'Sex', 'Sample_Type', 'purity', 'ploidy', 'Overall_Survival_Months', 'Overall_Survival_Status',\n",
    "                          'MSI_Score', 'MSI_Type', 'TMB_Score']\n",
    "    \n",
    "    return sample_info\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>Overall_Survival_Months</th>\n",
       "      <th>Overall_Survival_Status</th>\n",
       "      <th>MSI_Score</th>\n",
       "      <th>MSI_Type</th>\n",
       "      <th>TMB_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>P-0025956</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>71.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>3.496971</td>\n",
       "      <td>3.584</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.391316</td>\n",
       "      <td>2.871793</td>\n",
       "      <td>14.137</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Neuroendocrine Carcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.865628</td>\n",
       "      <td>3.115253</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>2.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>3.136841</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.82</td>\n",
       "      <td>Stable</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Colon Adenocarcinoma</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.351778</td>\n",
       "      <td>3.727190</td>\n",
       "      <td>34.060</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.45</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6362</th>\n",
       "      <td>P-0050671-T01-IM6_P-0050671-N01-IM6</td>\n",
       "      <td>P-0050671-T01-IM6</td>\n",
       "      <td>P-0050671</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>34.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.300360</td>\n",
       "      <td>3.776265</td>\n",
       "      <td>2.005</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Stable</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6363</th>\n",
       "      <td>P-0009110-T02-IM6_P-0009110-N01-IM6</td>\n",
       "      <td>P-0009110-T02-IM6</td>\n",
       "      <td>P-0009110</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.239690</td>\n",
       "      <td>3.480349</td>\n",
       "      <td>49.414</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>30.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6364</th>\n",
       "      <td>P-0050668-T01-IM6_P-0050668-N01-IM6</td>\n",
       "      <td>P-0050668-T01-IM6</td>\n",
       "      <td>P-0050668</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Squamous Cell Carcinoma</td>\n",
       "      <td>73.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.211949</td>\n",
       "      <td>3.144416</td>\n",
       "      <td>1.710</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>6.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6365</th>\n",
       "      <td>P-0050675-T01-IM6_P-0050675-N01-IM6</td>\n",
       "      <td>P-0050675-T01-IM6</td>\n",
       "      <td>P-0050675</td>\n",
       "      <td>Cancer of Unknown Primary</td>\n",
       "      <td>Cancer of Unknown Primary</td>\n",
       "      <td>64.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.386871</td>\n",
       "      <td>2.760248</td>\n",
       "      <td>1.973</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.32</td>\n",
       "      <td>Stable</td>\n",
       "      <td>23.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6366</th>\n",
       "      <td>P-0050660-T01-IM6_P-0050660-N01-IM6</td>\n",
       "      <td>P-0050660-T01-IM6</td>\n",
       "      <td>P-0050660</td>\n",
       "      <td>Bladder Cancer</td>\n",
       "      <td>Bladder Urothelial Carcinoma</td>\n",
       "      <td>70.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>3.459433</td>\n",
       "      <td>2.038</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.41</td>\n",
       "      <td>Stable</td>\n",
       "      <td>6.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6367 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6  P-0025956-T01-IM6  P-0025956   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6  P-0036909-T01-IM6  P-0036909   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6  P-0023546-T01-IM6  P-0023546   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6  P-0023546-T02-IM6  P-0023546   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6  P-0018837-T01-IM6  P-0018837   \n",
       "...                                   ...                ...        ...   \n",
       "6362  P-0050671-T01-IM6_P-0050671-N01-IM6  P-0050671-T01-IM6  P-0050671   \n",
       "6363  P-0009110-T02-IM6_P-0009110-N01-IM6  P-0009110-T02-IM6  P-0009110   \n",
       "6364  P-0050668-T01-IM6_P-0050668-N01-IM6  P-0050668-T01-IM6  P-0050668   \n",
       "6365  P-0050675-T01-IM6_P-0050675-N01-IM6  P-0050675-T01-IM6  P-0050675   \n",
       "6366  P-0050660-T01-IM6_P-0050660-N01-IM6  P-0050660-T01-IM6  P-0050660   \n",
       "\n",
       "                     Cancer_Type               Cancer_Type_Detailed  \\\n",
       "0     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "1     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "2                Prostate Cancer  Prostate Neuroendocrine Carcinoma   \n",
       "3                Prostate Cancer            Prostate Adenocarcinoma   \n",
       "4              Colorectal Cancer               Colon Adenocarcinoma   \n",
       "...                          ...                                ...   \n",
       "6362               Breast Cancer          Invasive Breast Carcinoma   \n",
       "6363  Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "6364  Non-Small Cell Lung Cancer       Lung Squamous Cell Carcinoma   \n",
       "6365   Cancer of Unknown Primary          Cancer of Unknown Primary   \n",
       "6366              Bladder Cancer       Bladder Urothelial Carcinoma   \n",
       "\n",
       "      Patient_Current_Age     Sex Sample_Type    purity    ploidy  \\\n",
       "0                    71.0  Female     Primary  0.273767  3.496971   \n",
       "1                    47.0  Female  Metastasis  0.391316  2.871793   \n",
       "2                    50.0    Male     Primary  0.865628  3.115253   \n",
       "3                    50.0    Male     Primary  0.312907  3.136841   \n",
       "4                    60.0    Male     Primary  0.351778  3.727190   \n",
       "...                   ...     ...         ...       ...       ...   \n",
       "6362                 34.0  Female  Metastasis  0.300360  3.776265   \n",
       "6363                 57.0  Female  Metastasis  0.239690  3.480349   \n",
       "6364                 73.0    Male     Primary  0.211949  3.144416   \n",
       "6365                 64.0  Female     Primary  0.386871  2.760248   \n",
       "6366                 70.0    Male     Primary  0.683828  3.459433   \n",
       "\n",
       "      Overall_Survival_Months Overall_Survival_Status  MSI_Score MSI_Type  \\\n",
       "0                       3.584                DECEASED       0.00   Stable   \n",
       "1                      14.137                  LIVING       0.37   Stable   \n",
       "2                       4.800                DECEASED       2.37   Stable   \n",
       "3                       4.800                DECEASED       0.82   Stable   \n",
       "4                      34.060                  LIVING       0.45   Stable   \n",
       "...                       ...                     ...        ...      ...   \n",
       "6362                    2.005                  LIVING       0.05   Stable   \n",
       "6363                   49.414                  LIVING       0.00   Stable   \n",
       "6364                    1.710                  LIVING       0.00   Stable   \n",
       "6365                    1.973                  LIVING       1.32   Stable   \n",
       "6366                    2.038                  LIVING       0.41   Stable   \n",
       "\n",
       "      TMB_Score  \n",
       "0           5.3  \n",
       "1           3.5  \n",
       "2           3.5  \n",
       "3           2.6  \n",
       "4           5.3  \n",
       "...         ...  \n",
       "6362        7.0  \n",
       "6363       30.7  \n",
       "6364        6.1  \n",
       "6365       23.7  \n",
       "6366        6.1  \n",
       "\n",
       "[6367 rows x 15 columns]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info = create_sample_info(cohort=cohort_type)\n",
    "sample_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>dbSNP_RS</th>\n",
       "      <th>dbSNP_Val_Status</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Match_Norm_Seq_Allele1</th>\n",
       "      <th>Match_Norm_Seq_Allele2</th>\n",
       "      <th>Tumor_Validation_Allele1</th>\n",
       "      <th>Tumor_Validation_Allele2</th>\n",
       "      <th>Match_Norm_Validation_Allele1</th>\n",
       "      <th>Match_Norm_Validation_Allele2</th>\n",
       "      <th>Verification_Status</th>\n",
       "      <th>Validation_Status</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>Sequencing_Phase</th>\n",
       "      <th>Sequence_Source</th>\n",
       "      <th>Validation_Method</th>\n",
       "      <th>Score</th>\n",
       "      <th>BAM_File</th>\n",
       "      <th>Sequencer</th>\n",
       "      <th>Tumor_Sample_UUID</th>\n",
       "      <th>Matched_Norm_Sample_UUID</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Exon_Number</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>n_depth</th>\n",
       "      <th>n_ref_count</th>\n",
       "      <th>n_alt_count</th>\n",
       "      <th>all_effects</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>ALLELE_NUM</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND_VEP</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>RefSeq</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>GMAF</th>\n",
       "      <th>AFR_MAF</th>\n",
       "      <th>AMR_MAF</th>\n",
       "      <th>ASN_MAF</th>\n",
       "      <th>EAS_MAF</th>\n",
       "      <th>EUR_MAF</th>\n",
       "      <th>SAS_MAF</th>\n",
       "      <th>AA_MAF</th>\n",
       "      <th>EA_MAF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>PICK</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>TSL</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>MINIMISED</th>\n",
       "      <th>ExAC_AF</th>\n",
       "      <th>ExAC_AF_AFR</th>\n",
       "      <th>ExAC_AF_AMR</th>\n",
       "      <th>ExAC_AF_EAS</th>\n",
       "      <th>ExAC_AF_FIN</th>\n",
       "      <th>ExAC_AF_NFE</th>\n",
       "      <th>ExAC_AF_OTH</th>\n",
       "      <th>ExAC_AF_SAS</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>flanking_bps</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_qual</th>\n",
       "      <th>ExAC_AF_Adj</th>\n",
       "      <th>ExAC_AC_AN_Adj</th>\n",
       "      <th>ExAC_AC_AN</th>\n",
       "      <th>ExAC_AC_AN_AFR</th>\n",
       "      <th>ExAC_AC_AN_AMR</th>\n",
       "      <th>ExAC_AC_AN_EAS</th>\n",
       "      <th>ExAC_AC_AN_FIN</th>\n",
       "      <th>ExAC_AC_AN_NFE</th>\n",
       "      <th>ExAC_AC_AN_OTH</th>\n",
       "      <th>ExAC_AC_AN_SAS</th>\n",
       "      <th>ExAC_FILTER</th>\n",
       "      <th>Caller</th>\n",
       "      <th>is-a-hotspot</th>\n",
       "      <th>is-a-3d-hotspot</th>\n",
       "      <th>mutation_effect</th>\n",
       "      <th>oncogenic</th>\n",
       "      <th>LEVEL_1</th>\n",
       "      <th>LEVEL_2A</th>\n",
       "      <th>LEVEL_2B</th>\n",
       "      <th>LEVEL_3A</th>\n",
       "      <th>LEVEL_3B</th>\n",
       "      <th>LEVEL_4</th>\n",
       "      <th>LEVEL_R1</th>\n",
       "      <th>LEVEL_R2</th>\n",
       "      <th>LEVEL_R3</th>\n",
       "      <th>Highest_level</th>\n",
       "      <th>citations</th>\n",
       "      <th>driver</th>\n",
       "      <th>tcn</th>\n",
       "      <th>lcn</th>\n",
       "      <th>cf</th>\n",
       "      <th>purity</th>\n",
       "      <th>t_var_freq</th>\n",
       "      <th>expected_alt_copies</th>\n",
       "      <th>ccf_Mcopies</th>\n",
       "      <th>ccf_Mcopies_lower</th>\n",
       "      <th>ccf_Mcopies_upper</th>\n",
       "      <th>ccf_Mcopies_prob95</th>\n",
       "      <th>ccf_Mcopies_prob90</th>\n",
       "      <th>ccf_1copy</th>\n",
       "      <th>ccf_1copy_lower</th>\n",
       "      <th>ccf_1copy_upper</th>\n",
       "      <th>ccf_1copy_prob95</th>\n",
       "      <th>ccf_1copy_prob90</th>\n",
       "      <th>ccf_expected_copies</th>\n",
       "      <th>ccf_expected_copies_lower</th>\n",
       "      <th>ccf_expected_copies_upper</th>\n",
       "      <th>ccf_expected_copies_prob95</th>\n",
       "      <th>ccf_expected_copies_prob90</th>\n",
       "      <th>facets_fit</th>\n",
       "      <th>reviewer_set_purity</th>\n",
       "      <th>use_only_purity_run</th>\n",
       "      <th>use_edited_cncf</th>\n",
       "      <th>cncf_file_used</th>\n",
       "      <th>mut_key</th>\n",
       "      <th>sample_mut_key</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>3845</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>12</td>\n",
       "      <td>25398285</td>\n",
       "      <td>25398285</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.34G&gt;T</td>\n",
       "      <td>p.Gly12Cys</td>\n",
       "      <td>p.G12C</td>\n",
       "      <td>ENST00000256078</td>\n",
       "      <td>2/6</td>\n",
       "      <td>588.0</td>\n",
       "      <td>495.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>908</td>\n",
       "      <td>908</td>\n",
       "      <td>0</td>\n",
       "      <td>KRAS,missense_variant,p.Gly12Cys,ENST00000311936,NM_004985.3;KRAS,missense_variant,p.Gly12Cys,ENST00000556131,;KRAS,missense_variant,p.Gly12Cys,ENST00000256078,NM_033360.2;KRAS,missense_variant,p.Gly12Cys,ENST00000557334,;</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000133703</td>\n",
       "      <td>ENST00000256078</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>98/1119</td>\n",
       "      <td>34/570</td>\n",
       "      <td>12/189</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Ggt/Tgt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>6407.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS8703.1</td>\n",
       "      <td>ENSP00000256078</td>\n",
       "      <td>P01116</td>\n",
       "      <td>Q9UM97,Q71SP6,P78460,L7RSL8,I1SRC5</td>\n",
       "      <td>UPI0000133132</td>\n",
       "      <td>NM_033360.2</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>probably_damaging(0.993)</td>\n",
       "      <td>2/6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:3.40.50.300,Pfam_domain:PF00071,Prints_domain:PR00449,PROSITE_profiles:PS51421,hmmpanther:PTHR24070,hmmpanther:PTHR24070:SF186,Low_complexity_(Seg):seg,SMART_domains:SM00173,SMART_domains:SM00174,SMART_domains:SM00175,SMART_domains:SM00176,Superfamily_domains:SSF52540,TIGRFAM_domain:TIGR00231</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Gain-of-function</td>\n",
       "      <td>Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Binimetinib,Cobimetinib,Trametinib</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LEVEL_4</td>\n",
       "      <td>16051643;25705018;26841430;24256730;28783719;29247021;Bhagwat et al. Abstract# 4973, AACR 2017(http://cancerres.aacrjournals.org/content/77/13_Supplement/4973);Robarge et al. Abstract# DDT02-03, AACR 2014(http://cancerres.aacrjournals.org/content/74/19_Supplement/DDT02-03);Burrows et al. Abstract# 5168, AACR 2017(http://cancerres.aacrjournals.org/content/77/13_Supplement/5168)</td>\n",
       "      <td>True</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.138437</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.158163</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.736</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.821</td>\n",
       "      <td>2.235549e-03</td>\n",
       "      <td>1.483415e-02</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.860769</td>\n",
       "      <td>0.984692</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.860769</td>\n",
       "      <td>0.984692</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>12_25398285_C_A</td>\n",
       "      <td>P-0025956-T01-IM6-12_25398285_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>STK11</td>\n",
       "      <td>6794</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>19</td>\n",
       "      <td>1221314</td>\n",
       "      <td>1221314</td>\n",
       "      <td>+</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.837delC</td>\n",
       "      <td>p.Pro281ArgfsTer6</td>\n",
       "      <td>p.P281Rfs*6</td>\n",
       "      <td>ENST00000326873</td>\n",
       "      <td>6/10</td>\n",
       "      <td>615.0</td>\n",
       "      <td>492.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>885</td>\n",
       "      <td>884</td>\n",
       "      <td>1</td>\n",
       "      <td>STK11,frameshift_variant,p.Pro281ArgfsTer6,ENST00000326873,NM_000455.4;STK11,frameshift_variant,p.Pro37ArgfsTer6,ENST00000586243,;STK11,downstream_gene_variant,,ENST00000585851,;STK11,downstream_gene_variant,,ENST00000585748,;STK11,3_prime_UTR_variant,,ENST00000593219,;STK11,non_coding_transcript_exon_variant,,ENST00000589152,;STK11,non_coding_transcript_exon_variant,,ENST00000591133,;STK11,non_coding_transcript_exon_variant,,ENST00000586358,;STK11,upstream_gene_variant,,ENST00000585465,;,regulatory_region_variant,,ENSR00000377818,;</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000118046</td>\n",
       "      <td>ENST00000326873</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>2010/3328</td>\n",
       "      <td>837/1302</td>\n",
       "      <td>279/433</td>\n",
       "      <td>G/X</td>\n",
       "      <td>ggC/gg</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>STK11</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11389.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS45896.1</td>\n",
       "      <td>ENSP00000324856</td>\n",
       "      <td>Q15831</td>\n",
       "      <td>Q9NS52</td>\n",
       "      <td>UPI0000136105</td>\n",
       "      <td>NM_000455.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:1.10.510.10,Pfam_domain:PF00069,PROSITE_profiles:PS50011,hmmpanther:PTHR24347,hmmpanther:PTHR24347:SF1,SMART_domains:SM00220,Superfamily_domains:SSF56112</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>1.0</td>\n",
       "      <td>sequence_alteration</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGCC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24652667;19892943;25079552;21516316;19340305</td>\n",
       "      <td>True</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.183720</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.731</td>\n",
       "      <td>0.662</td>\n",
       "      <td>0.802</td>\n",
       "      <td>2.663028e-04</td>\n",
       "      <td>3.570729e-03</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.984</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.926527</td>\n",
       "      <td>0.996079</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.984</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.926527</td>\n",
       "      <td>0.996079</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>19_1221314_C_-</td>\n",
       "      <td>P-0025956-T01-IM6-19_1221314_C_-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KEAP1</td>\n",
       "      <td>9817</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>19</td>\n",
       "      <td>10602584</td>\n",
       "      <td>10602584</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.994G&gt;A</td>\n",
       "      <td>p.Gly332Ser</td>\n",
       "      <td>p.G332S</td>\n",
       "      <td>ENST00000171111</td>\n",
       "      <td>3/6</td>\n",
       "      <td>641.0</td>\n",
       "      <td>523.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>471</td>\n",
       "      <td>471</td>\n",
       "      <td>0</td>\n",
       "      <td>KEAP1,missense_variant,p.Gly332Ser,ENST00000171111,NM_203500.1;KEAP1,missense_variant,p.Gly332Ser,ENST00000393623,NM_012289.3;KEAP1,upstream_gene_variant,,ENST00000592478,;KEAP1,downstream_gene_variant,,ENST00000591419,;KEAP1,downstream_gene_variant,,ENST00000592055,;CTC-429L19.3,upstream_gene_variant,,ENST00000592671,;KEAP1,downstream_gene_variant,,ENST00000588024,;KEAP1,downstream_gene_variant,,ENST00000585845,;KEAP1,upstream_gene_variant,,ENST00000590593,;KEAP1,upstream_gene_variant,,ENST00000590237,;</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000079999</td>\n",
       "      <td>ENST00000171111</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1542/2955</td>\n",
       "      <td>994/1875</td>\n",
       "      <td>332/624</td>\n",
       "      <td>G/S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>KEAP1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>23177.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS12239.1</td>\n",
       "      <td>ENSP00000171111</td>\n",
       "      <td>Q14145</td>\n",
       "      <td>K7ESE0,K7EJD8,K7EJ49</td>\n",
       "      <td>UPI000007139C</td>\n",
       "      <td>NM_203500.1</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.998)</td>\n",
       "      <td>3/6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hmmpanther:PTHR24412:SF162,hmmpanther:PTHR24412,Pfam_domain:PF01344,Gene3D:1k3iA02,PIRSF_domain:PIRSF037037,SMART_domains:SM00612,Superfamily_domains:0052715</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.183720</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.184087</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.672</td>\n",
       "      <td>0.608</td>\n",
       "      <td>0.740</td>\n",
       "      <td>3.215666e-06</td>\n",
       "      <td>9.339940e-05</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.880435</td>\n",
       "      <td>0.989551</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.880435</td>\n",
       "      <td>0.989551</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>19_10602584_C_T</td>\n",
       "      <td>P-0025956-T01-IM6-19_10602584_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMARCA4</td>\n",
       "      <td>6597</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>19</td>\n",
       "      <td>11113836</td>\n",
       "      <td>11113836</td>\n",
       "      <td>+</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1943+1G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.X648_splice</td>\n",
       "      <td>ENST00000344626</td>\n",
       "      <td>NaN</td>\n",
       "      <td>445.0</td>\n",
       "      <td>371.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>362</td>\n",
       "      <td>361</td>\n",
       "      <td>1</td>\n",
       "      <td>SMARCA4,splice_donor_variant,,ENST00000358026,NM_001128849.1;SMARCA4,splice_donor_variant,,ENST00000429416,NM_001128844.1;SMARCA4,splice_donor_variant,,ENST00000344626,NM_003072.3;SMARCA4,splice_donor_variant,,ENST00000413806,NM_001128845.1,NM_001128847.1;SMARCA4,splice_donor_variant,,ENST00000450717,NM_001128846.1,NM_001128848.1;SMARCA4,splice_donor_variant,,ENST00000590574,;SMARCA4,splice_donor_variant,,ENST00000589677,;SMARCA4,splice_donor_variant,,ENST00000541122,;SMARCA4,splice_donor_variant,,ENST00000444061,;SMARCA4,splice_donor_variant,,ENST00000591545,;</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000127616</td>\n",
       "      <td>ENST00000344626</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>-/5392</td>\n",
       "      <td>1943/4944</td>\n",
       "      <td>648/1647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SMARCA4</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11100.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS12253.1</td>\n",
       "      <td>ENSP00000343896</td>\n",
       "      <td>P51532</td>\n",
       "      <td>B4DSI8,A7E2E1</td>\n",
       "      <td>UPI000006F973</td>\n",
       "      <td>NM_003072.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/34</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24658002;24658001;24658004;25060813;18301784</td>\n",
       "      <td>True</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.183720</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.166292</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.607</td>\n",
       "      <td>0.534</td>\n",
       "      <td>0.686</td>\n",
       "      <td>1.397430e-06</td>\n",
       "      <td>2.653238e-05</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.959</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.659996</td>\n",
       "      <td>0.904988</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.959</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.659996</td>\n",
       "      <td>0.904988</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>19_11113836_G_A</td>\n",
       "      <td>P-0025956-T01-IM6-19_11113836_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PTPRT</td>\n",
       "      <td>11122</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>20</td>\n",
       "      <td>40735560</td>\n",
       "      <td>40735560</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.3313G&gt;T</td>\n",
       "      <td>p.Ala1105Ser</td>\n",
       "      <td>p.A1105S</td>\n",
       "      <td>ENST00000373198</td>\n",
       "      <td>25/32</td>\n",
       "      <td>471.0</td>\n",
       "      <td>378.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>349</td>\n",
       "      <td>349</td>\n",
       "      <td>0</td>\n",
       "      <td>PTPRT,missense_variant,p.Ala1105Ser,ENST00000373198,NM_133170.3;PTPRT,missense_variant,p.Ala1076Ser,ENST00000373201,;PTPRT,missense_variant,p.Ala1089Ser,ENST00000373193,NM_007050.5;PTPRT,missense_variant,p.Ala1085Ser,ENST00000373190,;PTPRT,missense_variant,p.Ala1096Ser,ENST00000373184,;PTPRT,missense_variant,p.Ala1095Ser,ENST00000356100,;PTPRT,missense_variant,p.Ala1086Ser,ENST00000373187,;</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000196090</td>\n",
       "      <td>ENST00000373198</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>3549/12746</td>\n",
       "      <td>3313/4383</td>\n",
       "      <td>1105/1460</td>\n",
       "      <td>A/S</td>\n",
       "      <td>Gct/Tct</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>PTPRT</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>9682.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS68127.1</td>\n",
       "      <td>ENSP00000362294</td>\n",
       "      <td>O14522</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UPI00001AF6FA</td>\n",
       "      <td>NM_133170.3</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.973)</td>\n",
       "      <td>25/32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:3.90.190.10,Pfam_domain:PF00102,Prints_domain:PR00700,PROSITE_patterns:PS00383,PROSITE_profiles:PS50055,PROSITE_profiles:PS50056,hmmpanther:PTHR19134,hmmpanther:PTHR19134:SF208,SMART_domains:SM00194,SMART_domains:SM00404,Superfamily_domains:SSF52799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.183720</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.197452</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.721</td>\n",
       "      <td>0.644</td>\n",
       "      <td>0.803</td>\n",
       "      <td>7.615616e-04</td>\n",
       "      <td>6.531934e-03</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.859171</td>\n",
       "      <td>0.984420</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.859171</td>\n",
       "      <td>0.984420</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>20_40735560_C_A</td>\n",
       "      <td>P-0025956-T01-IM6-20_40735560_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52110</th>\n",
       "      <td>LATS1</td>\n",
       "      <td>9113</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>6</td>\n",
       "      <td>150001566</td>\n",
       "      <td>150001566</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050660-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.2038C&gt;G</td>\n",
       "      <td>p.Gln680Glu</td>\n",
       "      <td>p.Q680E</td>\n",
       "      <td>ENST00000253339</td>\n",
       "      <td>4/7</td>\n",
       "      <td>355.0</td>\n",
       "      <td>211.0</td>\n",
       "      <td>144.0</td>\n",
       "      <td>296</td>\n",
       "      <td>296</td>\n",
       "      <td>0</td>\n",
       "      <td>LATS1,missense_variant,p.Gln680Glu,ENST00000543571,NM_004690.3;LATS1,missense_variant,p.Gln680Glu,ENST00000253339,;LATS1,downstream_gene_variant,,ENST00000392273,NM_001270519.1;LATS1,downstream_gene_variant,,ENST00000458696,;LATS1,non_coding_transcript_exon_variant,,ENST00000542747,;LATS1,3_prime_UTR_variant,,ENST00000441107,;LATS1,downstream_gene_variant,,ENST00000542720,;</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000131023</td>\n",
       "      <td>ENST00000253339</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>2266/4256</td>\n",
       "      <td>2038/3393</td>\n",
       "      <td>680/1130</td>\n",
       "      <td>Q/E</td>\n",
       "      <td>Caa/Gaa</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>LATS1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>6514.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS34551.1</td>\n",
       "      <td>ENSP00000253339</td>\n",
       "      <td>O95835</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UPI0000073DC2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tolerated(0.07)</td>\n",
       "      <td>benign(0.443)</td>\n",
       "      <td>4/7</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hmmpanther:PTHR24356:SF138,hmmpanther:PTHR24356</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TGA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.652662</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>0.405634</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.999</td>\n",
       "      <td>0.923</td>\n",
       "      <td>1.000</td>\n",
       "      <td>5.737254e-01</td>\n",
       "      <td>8.879509e-01</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.992</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.992534</td>\n",
       "      <td>0.999960</td>\n",
       "      <td>0.999</td>\n",
       "      <td>0.923</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.573725</td>\n",
       "      <td>0.887951</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default/P-0050660-T01-IM6_P-0050660-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>6_150001566_G_C</td>\n",
       "      <td>P-0050660-T01-IM6-6_150001566_G_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52111</th>\n",
       "      <td>PREX2</td>\n",
       "      <td>80243</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>8</td>\n",
       "      <td>69104718</td>\n",
       "      <td>69104719</td>\n",
       "      <td>+</td>\n",
       "      <td>Frame_Shift_Ins</td>\n",
       "      <td>INS</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>G</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050660-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.4565dupG</td>\n",
       "      <td>p.Ala1523ArgfsTer109</td>\n",
       "      <td>p.A1523Rfs*109</td>\n",
       "      <td>ENST00000288368</td>\n",
       "      <td>37/40</td>\n",
       "      <td>841.0</td>\n",
       "      <td>336.0</td>\n",
       "      <td>505.0</td>\n",
       "      <td>472</td>\n",
       "      <td>472</td>\n",
       "      <td>0</td>\n",
       "      <td>PREX2,frameshift_variant,p.Ala1523ArgfsTer109,ENST00000288368,NM_024870.2,NM_025170.4;PREX2,non_coding_transcript_exon_variant,,ENST00000520235,;PREX2,non_coding_transcript_exon_variant,,ENST00000522247,;</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000046889</td>\n",
       "      <td>ENST00000288368</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>4839-4840/10750</td>\n",
       "      <td>4562-4563/4821</td>\n",
       "      <td>1521/1606</td>\n",
       "      <td>L/LX</td>\n",
       "      <td>ctg/ctGg</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PREX2</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>22950.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS6201.1</td>\n",
       "      <td>ENSP00000288368</td>\n",
       "      <td>Q70Z35</td>\n",
       "      <td>Q56UR8</td>\n",
       "      <td>UPI0000375435</td>\n",
       "      <td>NM_024870.2,NM_025170.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>37/40</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hmmpanther:PTHR22829,hmmpanther:PTHR22829:SF1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>1.0</td>\n",
       "      <td>insertion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CTG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>VQSRTrancheSNP99.60to99.80</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>0.600476</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.986</td>\n",
       "      <td>0.952</td>\n",
       "      <td>1.000</td>\n",
       "      <td>8.608698e-01</td>\n",
       "      <td>9.984730e-01</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.998</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.986</td>\n",
       "      <td>0.952</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.860870</td>\n",
       "      <td>0.998473</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default/P-0050660-T01-IM6_P-0050660-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>8_69104718_-_G</td>\n",
       "      <td>P-0050660-T01-IM6-8_69104718_-_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52112</th>\n",
       "      <td>ATM</td>\n",
       "      <td>472</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>11</td>\n",
       "      <td>108186598</td>\n",
       "      <td>108186598</td>\n",
       "      <td>+</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050660-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.6055delT</td>\n",
       "      <td>p.Tyr2019MetfsTer28</td>\n",
       "      <td>p.Y2019Mfs*28</td>\n",
       "      <td>ENST00000278616</td>\n",
       "      <td>41/63</td>\n",
       "      <td>264.0</td>\n",
       "      <td>205.0</td>\n",
       "      <td>59.0</td>\n",
       "      <td>233</td>\n",
       "      <td>233</td>\n",
       "      <td>0</td>\n",
       "      <td>ATM,frameshift_variant,p.Tyr2019MetfsTer28,ENST00000278616,NM_000051.3;ATM,frameshift_variant,p.Tyr2019MetfsTer28,ENST00000452508,;C11orf65,intron_variant,,ENST00000525729,;ATM,non_coding_transcript_exon_variant,,ENST00000532765,;ATM,3_prime_UTR_variant,,ENST00000529588,;ATM,non_coding_transcript_exon_variant,,ENST00000524792,;ATM,non_coding_transcript_exon_variant,,ENST00000533690,;</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000149311</td>\n",
       "      <td>ENST00000278616</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>6440/13147</td>\n",
       "      <td>6055/9171</td>\n",
       "      <td>2019/3056</td>\n",
       "      <td>Y/X</td>\n",
       "      <td>Tat/at</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>ATM</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>795.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS31669.1</td>\n",
       "      <td>ENSP00000278616</td>\n",
       "      <td>Q13315</td>\n",
       "      <td>M0QXY8,E9PRG7,E9PIN0</td>\n",
       "      <td>UPI0000DBEF44</td>\n",
       "      <td>NM_000051.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>41/63</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hmmpanther:PTHR11139,hmmpanther:PTHR11139:SF66,PROSITE_profiles:PS51189</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>1.0</td>\n",
       "      <td>sequence_alteration</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TGTA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Olaparib</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LEVEL_4</td>\n",
       "      <td>16832357;16998505;15928302;26510020</td>\n",
       "      <td>True</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>0.223485</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.439</td>\n",
       "      <td>0.381</td>\n",
       "      <td>0.500</td>\n",
       "      <td>1.502461e-18</td>\n",
       "      <td>1.324905e-15</td>\n",
       "      <td>0.877</td>\n",
       "      <td>0.762</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.147953</td>\n",
       "      <td>0.350775</td>\n",
       "      <td>0.877</td>\n",
       "      <td>0.762</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.147953</td>\n",
       "      <td>0.350775</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default/P-0050660-T01-IM6_P-0050660-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>11_108186598_T_-</td>\n",
       "      <td>P-0050660-T01-IM6-11_108186598_T_-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52113</th>\n",
       "      <td>KMT2D</td>\n",
       "      <td>8085</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>12</td>\n",
       "      <td>49422690</td>\n",
       "      <td>49422690</td>\n",
       "      <td>+</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>-</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050660-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.14303delA</td>\n",
       "      <td>p.Lys4768SerfsTer29</td>\n",
       "      <td>p.K4768Sfs*29</td>\n",
       "      <td>ENST00000301067</td>\n",
       "      <td>45/54</td>\n",
       "      <td>762.0</td>\n",
       "      <td>356.0</td>\n",
       "      <td>406.0</td>\n",
       "      <td>584</td>\n",
       "      <td>584</td>\n",
       "      <td>0</td>\n",
       "      <td>KMT2D,frameshift_variant,p.Lys4768SerfsTer29,ENST00000301067,NM_003482.3;KMT2D,upstream_gene_variant,,ENST00000526209,;KMT2D,downstream_gene_variant,,ENST00000552391,;</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000167548</td>\n",
       "      <td>ENST00000301067</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>14303/19419</td>\n",
       "      <td>14303/16614</td>\n",
       "      <td>4768/5537</td>\n",
       "      <td>K/X</td>\n",
       "      <td>aAg/ag</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>KMT2D</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>7133.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS44873.1</td>\n",
       "      <td>ENSP00000301067</td>\n",
       "      <td>O14686</td>\n",
       "      <td>Q6PIA1,Q59FG6,F8VWW4</td>\n",
       "      <td>UPI0000EE84D6</td>\n",
       "      <td>NM_003482.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>45/54</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hmmpanther:PTHR22884,hmmpanther:PTHR22884:SF324</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>1.0</td>\n",
       "      <td>sequence_alteration</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCTT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24633898;24240169;26366712</td>\n",
       "      <td>True</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>0.532808</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.983</td>\n",
       "      <td>1.000</td>\n",
       "      <td>9.665077e-01</td>\n",
       "      <td>9.998115e-01</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.997</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.983</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.966508</td>\n",
       "      <td>0.999811</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default/P-0050660-T01-IM6_P-0050660-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>12_49422690_T_-</td>\n",
       "      <td>P-0050660-T01-IM6-12_49422690_T_-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52114</th>\n",
       "      <td>NTHL1</td>\n",
       "      <td>4913</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>16</td>\n",
       "      <td>2090190</td>\n",
       "      <td>2090190</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050660-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.759G&gt;C</td>\n",
       "      <td>p.Lys253Asn</td>\n",
       "      <td>p.K253N</td>\n",
       "      <td>ENST00000219066</td>\n",
       "      <td>5/6</td>\n",
       "      <td>895.0</td>\n",
       "      <td>413.0</td>\n",
       "      <td>482.0</td>\n",
       "      <td>691</td>\n",
       "      <td>691</td>\n",
       "      <td>0</td>\n",
       "      <td>NTHL1,missense_variant,p.Lys253Asn,ENST00000219066,NM_002528.5;NTHL1,missense_variant,p.Lys177Asn,ENST00000566380,;SLC9A3R2,downstream_gene_variant,,ENST00000424542,NM_001130012.2,NM_004785.5;SLC9A3R2,downstream_gene_variant,,ENST00000432365,;SLC9A3R2,downstream_gene_variant,,ENST00000566198,NM_001252073.1,NM_001252076.1,NM_001252075.1;SLC9A3R2,downstream_gene_variant,,ENST00000563587,;SLC9A3R2,downstream_gene_variant,,ENST00000565855,;SLC9A3R2,downstream_gene_variant,,ENST00000561844,;SLC9A3R2,downstream_gene_variant,,ENST00000567504,;NTHL1,non_coding_transcript_exon_variant,,ENST00000562951,;SLC9A3R2,downstream_gene_variant,,ENST00000565086,;NTHL1,3_prime_UTR_variant,,ENST00000561841,;NTHL1,3_prime_UTR_variant,,ENST00000568513,;NTHL1,non_coding_transcript_exon_variant,,ENST00000565406,;NTHL1,non_coding_transcript_exon_variant,,ENST00000567727,;NTHL1,non_coding_transcript_exon_variant,,ENST00000561862,;SLC9A3R2,downstream_gene_variant,,ENST00000564033,;NTHL1,downstream_gene_varian...</td>\n",
       "      <td>G</td>\n",
       "      <td>ENSG00000065057</td>\n",
       "      <td>ENST00000219066</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>778/1067</td>\n",
       "      <td>759/939</td>\n",
       "      <td>253/312</td>\n",
       "      <td>K/N</td>\n",
       "      <td>aaG/aaC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>NTHL1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>8028.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS10457.1</td>\n",
       "      <td>ENSP00000219066</td>\n",
       "      <td>P78549</td>\n",
       "      <td>E5KTI5</td>\n",
       "      <td>UPI0000130586</td>\n",
       "      <td>NM_002528.5</td>\n",
       "      <td>deleterious(0.01)</td>\n",
       "      <td>benign(0.068)</td>\n",
       "      <td>5/6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:1.10.1670.10,Pfam_domain:PF00730,hmmpanther:PTHR10359,SMART_domains:SM00478,Superfamily_domains:SSF48150</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TCT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>0.538547</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.988</td>\n",
       "      <td>1.000</td>\n",
       "      <td>9.876220e-01</td>\n",
       "      <td>9.999818e-01</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.997</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.988</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.987622</td>\n",
       "      <td>0.999982</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default/P-0050660-T01-IM6_P-0050660-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>16_2090190_C_G</td>\n",
       "      <td>P-0050660-T01-IM6-16_2090190_C_G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>52115 rows × 168 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  \\\n",
       "0            KRAS            3845  MSKCC     GRCh37         12   \n",
       "1           STK11            6794  MSKCC     GRCh37         19   \n",
       "2           KEAP1            9817  MSKCC     GRCh37         19   \n",
       "3         SMARCA4            6597  MSKCC     GRCh37         19   \n",
       "4           PTPRT           11122  MSKCC     GRCh37         20   \n",
       "...           ...             ...    ...        ...        ...   \n",
       "52110       LATS1            9113  MSKCC     GRCh37          6   \n",
       "52111       PREX2           80243  MSKCC     GRCh37          8   \n",
       "52112         ATM             472  MSKCC     GRCh37         11   \n",
       "52113       KMT2D            8085  MSKCC     GRCh37         12   \n",
       "52114       NTHL1            4913  MSKCC     GRCh37         16   \n",
       "\n",
       "       Start_Position  End_Position Strand Variant_Classification  \\\n",
       "0            25398285      25398285      +      Missense_Mutation   \n",
       "1             1221314       1221314      +        Frame_Shift_Del   \n",
       "2            10602584      10602584      +      Missense_Mutation   \n",
       "3            11113836      11113836      +            Splice_Site   \n",
       "4            40735560      40735560      +      Missense_Mutation   \n",
       "...               ...           ...    ...                    ...   \n",
       "52110       150001566     150001566      +      Missense_Mutation   \n",
       "52111        69104718      69104719      +        Frame_Shift_Ins   \n",
       "52112       108186598     108186598      +        Frame_Shift_Del   \n",
       "52113        49422690      49422690      +        Frame_Shift_Del   \n",
       "52114         2090190       2090190      +      Missense_Mutation   \n",
       "\n",
       "      Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2  \\\n",
       "0              SNP                C                 C                 A   \n",
       "1              DEL                C                 C                 -   \n",
       "2              SNP                C                 C                 T   \n",
       "3              SNP                G                 G                 A   \n",
       "4              SNP                C                 C                 A   \n",
       "...            ...              ...               ...               ...   \n",
       "52110          SNP                G                 G                 C   \n",
       "52111          INS                -                 -                 G   \n",
       "52112          DEL                T                 T                 -   \n",
       "52113          DEL                T                 T                 -   \n",
       "52114          SNP                C                 C                 G   \n",
       "\n",
       "      dbSNP_RS  dbSNP_Val_Status Tumor_Sample_Barcode  \\\n",
       "0        novel               NaN    P-0025956-T01-IM6   \n",
       "1        novel               NaN    P-0025956-T01-IM6   \n",
       "2        novel               NaN    P-0025956-T01-IM6   \n",
       "3        novel               NaN    P-0025956-T01-IM6   \n",
       "4        novel               NaN    P-0025956-T01-IM6   \n",
       "...        ...               ...                  ...   \n",
       "52110    novel               NaN    P-0050660-T01-IM6   \n",
       "52111    novel               NaN    P-0050660-T01-IM6   \n",
       "52112    novel               NaN    P-0050660-T01-IM6   \n",
       "52113    novel               NaN    P-0050660-T01-IM6   \n",
       "52114    novel               NaN    P-0050660-T01-IM6   \n",
       "\n",
       "       Matched_Norm_Sample_Barcode Match_Norm_Seq_Allele1  \\\n",
       "0                              NaN                    NaN   \n",
       "1                              NaN                    NaN   \n",
       "2                              NaN                    NaN   \n",
       "3                              NaN                    NaN   \n",
       "4                              NaN                    NaN   \n",
       "...                            ...                    ...   \n",
       "52110                          NaN                    NaN   \n",
       "52111                          NaN                    NaN   \n",
       "52112                          NaN                    NaN   \n",
       "52113                          NaN                    NaN   \n",
       "52114                          NaN                    NaN   \n",
       "\n",
       "      Match_Norm_Seq_Allele2  Tumor_Validation_Allele1  \\\n",
       "0                        NaN                       NaN   \n",
       "1                        NaN                       NaN   \n",
       "2                        NaN                       NaN   \n",
       "3                        NaN                       NaN   \n",
       "4                        NaN                       NaN   \n",
       "...                      ...                       ...   \n",
       "52110                    NaN                       NaN   \n",
       "52111                    NaN                       NaN   \n",
       "52112                    NaN                       NaN   \n",
       "52113                    NaN                       NaN   \n",
       "52114                    NaN                       NaN   \n",
       "\n",
       "       Tumor_Validation_Allele2  Match_Norm_Validation_Allele1  \\\n",
       "0                           NaN                            NaN   \n",
       "1                           NaN                            NaN   \n",
       "2                           NaN                            NaN   \n",
       "3                           NaN                            NaN   \n",
       "4                           NaN                            NaN   \n",
       "...                         ...                            ...   \n",
       "52110                       NaN                            NaN   \n",
       "52111                       NaN                            NaN   \n",
       "52112                       NaN                            NaN   \n",
       "52113                       NaN                            NaN   \n",
       "52114                       NaN                            NaN   \n",
       "\n",
       "       Match_Norm_Validation_Allele2  Verification_Status Validation_Status  \\\n",
       "0                                NaN                  NaN           Unknown   \n",
       "1                                NaN                  NaN           Unknown   \n",
       "2                                NaN                  NaN           Unknown   \n",
       "3                                NaN                  NaN           Unknown   \n",
       "4                                NaN                  NaN           Unknown   \n",
       "...                              ...                  ...               ...   \n",
       "52110                            NaN                  NaN           Unknown   \n",
       "52111                            NaN                  NaN           Unknown   \n",
       "52112                            NaN                  NaN           Unknown   \n",
       "52113                            NaN                  NaN           Unknown   \n",
       "52114                            NaN                  NaN           Unknown   \n",
       "\n",
       "      Mutation_Status  Sequencing_Phase  Sequence_Source  Validation_Method  \\\n",
       "0             SOMATIC               NaN              NaN                NaN   \n",
       "1             SOMATIC               NaN              NaN                NaN   \n",
       "2             SOMATIC               NaN              NaN                NaN   \n",
       "3             SOMATIC               NaN              NaN                NaN   \n",
       "4             SOMATIC               NaN              NaN                NaN   \n",
       "...               ...               ...              ...                ...   \n",
       "52110         SOMATIC               NaN              NaN                NaN   \n",
       "52111         SOMATIC               NaN              NaN                NaN   \n",
       "52112         SOMATIC               NaN              NaN                NaN   \n",
       "52113         SOMATIC               NaN              NaN                NaN   \n",
       "52114         SOMATIC               NaN              NaN                NaN   \n",
       "\n",
       "            Score  BAM_File  Sequencer  Tumor_Sample_UUID  \\\n",
       "0      MSK-IMPACT       NaN        NaN                NaN   \n",
       "1      MSK-IMPACT       NaN        NaN                NaN   \n",
       "2      MSK-IMPACT       NaN        NaN                NaN   \n",
       "3      MSK-IMPACT       NaN        NaN                NaN   \n",
       "4      MSK-IMPACT       NaN        NaN                NaN   \n",
       "...           ...       ...        ...                ...   \n",
       "52110  MSK-IMPACT       NaN        NaN                NaN   \n",
       "52111  MSK-IMPACT       NaN        NaN                NaN   \n",
       "52112  MSK-IMPACT       NaN        NaN                NaN   \n",
       "52113  MSK-IMPACT       NaN        NaN                NaN   \n",
       "52114  MSK-IMPACT       NaN        NaN                NaN   \n",
       "\n",
       "       Matched_Norm_Sample_UUID        HGVSc                 HGVSp  \\\n",
       "0                           NaN      c.34G>T            p.Gly12Cys   \n",
       "1                           NaN    c.837delC     p.Pro281ArgfsTer6   \n",
       "2                           NaN     c.994G>A           p.Gly332Ser   \n",
       "3                           NaN  c.1943+1G>A                   NaN   \n",
       "4                           NaN    c.3313G>T          p.Ala1105Ser   \n",
       "...                         ...          ...                   ...   \n",
       "52110                       NaN    c.2038C>G           p.Gln680Glu   \n",
       "52111                       NaN   c.4565dupG  p.Ala1523ArgfsTer109   \n",
       "52112                       NaN   c.6055delT   p.Tyr2019MetfsTer28   \n",
       "52113                       NaN  c.14303delA   p.Lys4768SerfsTer29   \n",
       "52114                       NaN     c.759G>C           p.Lys253Asn   \n",
       "\n",
       "          HGVSp_Short    Transcript_ID Exon_Number  t_depth  t_ref_count  \\\n",
       "0              p.G12C  ENST00000256078         2/6    588.0        495.0   \n",
       "1         p.P281Rfs*6  ENST00000326873        6/10    615.0        492.0   \n",
       "2             p.G332S  ENST00000171111         3/6    641.0        523.0   \n",
       "3       p.X648_splice  ENST00000344626         NaN    445.0        371.0   \n",
       "4            p.A1105S  ENST00000373198       25/32    471.0        378.0   \n",
       "...               ...              ...         ...      ...          ...   \n",
       "52110         p.Q680E  ENST00000253339         4/7    355.0        211.0   \n",
       "52111  p.A1523Rfs*109  ENST00000288368       37/40    841.0        336.0   \n",
       "52112   p.Y2019Mfs*28  ENST00000278616       41/63    264.0        205.0   \n",
       "52113   p.K4768Sfs*29  ENST00000301067       45/54    762.0        356.0   \n",
       "52114         p.K253N  ENST00000219066         5/6    895.0        413.0   \n",
       "\n",
       "       t_alt_count  n_depth n_ref_count n_alt_count  \\\n",
       "0             93.0      908         908           0   \n",
       "1            123.0      885         884           1   \n",
       "2            118.0      471         471           0   \n",
       "3             74.0      362         361           1   \n",
       "4             93.0      349         349           0   \n",
       "...            ...      ...         ...         ...   \n",
       "52110        144.0      296         296           0   \n",
       "52111        505.0      472         472           0   \n",
       "52112         59.0      233         233           0   \n",
       "52113        406.0      584         584           0   \n",
       "52114        482.0      691         691           0   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   all_effects  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               KRAS,missense_variant,p.Gly12Cys,ENST00000311936,NM_004985.3;KRAS,missense_variant,p.Gly12Cys,ENST00000556131,;KRAS,missense_variant,p.Gly12Cys,ENST00000256078,NM_033360.2;KRAS,missense_variant,p.Gly12Cys,ENST00000557334,;   \n",
       "1                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   STK11,frameshift_variant,p.Pro281ArgfsTer6,ENST00000326873,NM_000455.4;STK11,frameshift_variant,p.Pro37ArgfsTer6,ENST00000586243,;STK11,downstream_gene_variant,,ENST00000585851,;STK11,downstream_gene_variant,,ENST00000585748,;STK11,3_prime_UTR_variant,,ENST00000593219,;STK11,non_coding_transcript_exon_variant,,ENST00000589152,;STK11,non_coding_transcript_exon_variant,,ENST00000591133,;STK11,non_coding_transcript_exon_variant,,ENST00000586358,;STK11,upstream_gene_variant,,ENST00000585465,;,regulatory_region_variant,,ENSR00000377818,;   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                KEAP1,missense_variant,p.Gly332Ser,ENST00000171111,NM_203500.1;KEAP1,missense_variant,p.Gly332Ser,ENST00000393623,NM_012289.3;KEAP1,upstream_gene_variant,,ENST00000592478,;KEAP1,downstream_gene_variant,,ENST00000591419,;KEAP1,downstream_gene_variant,,ENST00000592055,;CTC-429L19.3,upstream_gene_variant,,ENST00000592671,;KEAP1,downstream_gene_variant,,ENST00000588024,;KEAP1,downstream_gene_variant,,ENST00000585845,;KEAP1,upstream_gene_variant,,ENST00000590593,;KEAP1,upstream_gene_variant,,ENST00000590237,;   \n",
       "3                                                                                                                                                                                                                                                                                                                                                                                                                                                      SMARCA4,splice_donor_variant,,ENST00000358026,NM_001128849.1;SMARCA4,splice_donor_variant,,ENST00000429416,NM_001128844.1;SMARCA4,splice_donor_variant,,ENST00000344626,NM_003072.3;SMARCA4,splice_donor_variant,,ENST00000413806,NM_001128845.1,NM_001128847.1;SMARCA4,splice_donor_variant,,ENST00000450717,NM_001128846.1,NM_001128848.1;SMARCA4,splice_donor_variant,,ENST00000590574,;SMARCA4,splice_donor_variant,,ENST00000589677,;SMARCA4,splice_donor_variant,,ENST00000541122,;SMARCA4,splice_donor_variant,,ENST00000444061,;SMARCA4,splice_donor_variant,,ENST00000591545,;   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    PTPRT,missense_variant,p.Ala1105Ser,ENST00000373198,NM_133170.3;PTPRT,missense_variant,p.Ala1076Ser,ENST00000373201,;PTPRT,missense_variant,p.Ala1089Ser,ENST00000373193,NM_007050.5;PTPRT,missense_variant,p.Ala1085Ser,ENST00000373190,;PTPRT,missense_variant,p.Ala1096Ser,ENST00000373184,;PTPRT,missense_variant,p.Ala1095Ser,ENST00000356100,;PTPRT,missense_variant,p.Ala1086Ser,ENST00000373187,;   \n",
       "...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        ...   \n",
       "52110                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 LATS1,missense_variant,p.Gln680Glu,ENST00000543571,NM_004690.3;LATS1,missense_variant,p.Gln680Glu,ENST00000253339,;LATS1,downstream_gene_variant,,ENST00000392273,NM_001270519.1;LATS1,downstream_gene_variant,,ENST00000458696,;LATS1,non_coding_transcript_exon_variant,,ENST00000542747,;LATS1,3_prime_UTR_variant,,ENST00000441107,;LATS1,downstream_gene_variant,,ENST00000542720,;   \n",
       "52111                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             PREX2,frameshift_variant,p.Ala1523ArgfsTer109,ENST00000288368,NM_024870.2,NM_025170.4;PREX2,non_coding_transcript_exon_variant,,ENST00000520235,;PREX2,non_coding_transcript_exon_variant,,ENST00000522247,;   \n",
       "52112                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       ATM,frameshift_variant,p.Tyr2019MetfsTer28,ENST00000278616,NM_000051.3;ATM,frameshift_variant,p.Tyr2019MetfsTer28,ENST00000452508,;C11orf65,intron_variant,,ENST00000525729,;ATM,non_coding_transcript_exon_variant,,ENST00000532765,;ATM,3_prime_UTR_variant,,ENST00000529588,;ATM,non_coding_transcript_exon_variant,,ENST00000524792,;ATM,non_coding_transcript_exon_variant,,ENST00000533690,;   \n",
       "52113                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                  KMT2D,frameshift_variant,p.Lys4768SerfsTer29,ENST00000301067,NM_003482.3;KMT2D,upstream_gene_variant,,ENST00000526209,;KMT2D,downstream_gene_variant,,ENST00000552391,;   \n",
       "52114  NTHL1,missense_variant,p.Lys253Asn,ENST00000219066,NM_002528.5;NTHL1,missense_variant,p.Lys177Asn,ENST00000566380,;SLC9A3R2,downstream_gene_variant,,ENST00000424542,NM_001130012.2,NM_004785.5;SLC9A3R2,downstream_gene_variant,,ENST00000432365,;SLC9A3R2,downstream_gene_variant,,ENST00000566198,NM_001252073.1,NM_001252076.1,NM_001252075.1;SLC9A3R2,downstream_gene_variant,,ENST00000563587,;SLC9A3R2,downstream_gene_variant,,ENST00000565855,;SLC9A3R2,downstream_gene_variant,,ENST00000561844,;SLC9A3R2,downstream_gene_variant,,ENST00000567504,;NTHL1,non_coding_transcript_exon_variant,,ENST00000562951,;SLC9A3R2,downstream_gene_variant,,ENST00000565086,;NTHL1,3_prime_UTR_variant,,ENST00000561841,;NTHL1,3_prime_UTR_variant,,ENST00000568513,;NTHL1,non_coding_transcript_exon_variant,,ENST00000565406,;NTHL1,non_coding_transcript_exon_variant,,ENST00000567727,;NTHL1,non_coding_transcript_exon_variant,,ENST00000561862,;SLC9A3R2,downstream_gene_variant,,ENST00000564033,;NTHL1,downstream_gene_varian...   \n",
       "\n",
       "      Allele             Gene          Feature Feature_type  \\\n",
       "0          A  ENSG00000133703  ENST00000256078   Transcript   \n",
       "1          -  ENSG00000118046  ENST00000326873   Transcript   \n",
       "2          T  ENSG00000079999  ENST00000171111   Transcript   \n",
       "3          A  ENSG00000127616  ENST00000344626   Transcript   \n",
       "4          A  ENSG00000196090  ENST00000373198   Transcript   \n",
       "...      ...              ...              ...          ...   \n",
       "52110      C  ENSG00000131023  ENST00000253339   Transcript   \n",
       "52111      G  ENSG00000046889  ENST00000288368   Transcript   \n",
       "52112      -  ENSG00000149311  ENST00000278616   Transcript   \n",
       "52113      -  ENSG00000167548  ENST00000301067   Transcript   \n",
       "52114      G  ENSG00000065057  ENST00000219066   Transcript   \n",
       "\n",
       "                                  Consequence    cDNA_position  \\\n",
       "0                            missense_variant          98/1119   \n",
       "1                          frameshift_variant        2010/3328   \n",
       "2                            missense_variant        1542/2955   \n",
       "3                        splice_donor_variant           -/5392   \n",
       "4      missense_variant,splice_region_variant       3549/12746   \n",
       "...                                       ...              ...   \n",
       "52110                        missense_variant        2266/4256   \n",
       "52111                      frameshift_variant  4839-4840/10750   \n",
       "52112                      frameshift_variant       6440/13147   \n",
       "52113                      frameshift_variant      14303/19419   \n",
       "52114                        missense_variant         778/1067   \n",
       "\n",
       "         CDS_position Protein_position Amino_acids    Codons  \\\n",
       "0              34/570           12/189         G/C   Ggt/Tgt   \n",
       "1            837/1302          279/433         G/X    ggC/gg   \n",
       "2            994/1875          332/624         G/S   Ggc/Agc   \n",
       "3           1943/4944         648/1647         NaN       NaN   \n",
       "4           3313/4383        1105/1460         A/S   Gct/Tct   \n",
       "...               ...              ...         ...       ...   \n",
       "52110       2038/3393         680/1130         Q/E   Caa/Gaa   \n",
       "52111  4562-4563/4821        1521/1606        L/LX  ctg/ctGg   \n",
       "52112       6055/9171        2019/3056         Y/X    Tat/at   \n",
       "52113     14303/16614        4768/5537         K/X    aAg/ag   \n",
       "52114         759/939          253/312         K/N   aaG/aaC   \n",
       "\n",
       "       Existing_variation  ALLELE_NUM  DISTANCE  STRAND_VEP   SYMBOL  \\\n",
       "0                     NaN           1       NaN          -1     KRAS   \n",
       "1                     NaN           1       NaN           1    STK11   \n",
       "2                     NaN           1       NaN          -1    KEAP1   \n",
       "3                     NaN           1       NaN           1  SMARCA4   \n",
       "4                     NaN           1       NaN          -1    PTPRT   \n",
       "...                   ...         ...       ...         ...      ...   \n",
       "52110                 NaN           1       NaN          -1    LATS1   \n",
       "52111                 NaN           1       NaN           1    PREX2   \n",
       "52112                 NaN           1       NaN           1      ATM   \n",
       "52113                 NaN           1       NaN          -1    KMT2D   \n",
       "52114                 NaN           1       NaN          -1    NTHL1   \n",
       "\n",
       "      SYMBOL_SOURCE  HGNC_ID         BIOTYPE CANONICAL         CCDS  \\\n",
       "0              HGNC   6407.0  protein_coding       YES   CCDS8703.1   \n",
       "1              HGNC  11389.0  protein_coding       YES  CCDS45896.1   \n",
       "2              HGNC  23177.0  protein_coding       YES  CCDS12239.1   \n",
       "3              HGNC  11100.0  protein_coding       NaN  CCDS12253.1   \n",
       "4              HGNC   9682.0  protein_coding       NaN  CCDS68127.1   \n",
       "...             ...      ...             ...       ...          ...   \n",
       "52110          HGNC   6514.0  protein_coding       NaN  CCDS34551.1   \n",
       "52111          HGNC  22950.0  protein_coding       YES   CCDS6201.1   \n",
       "52112          HGNC    795.0  protein_coding       YES  CCDS31669.1   \n",
       "52113          HGNC   7133.0  protein_coding       YES  CCDS44873.1   \n",
       "52114          HGNC   8028.0  protein_coding       YES  CCDS10457.1   \n",
       "\n",
       "                  ENSP SWISSPROT                              TREMBL  \\\n",
       "0      ENSP00000256078    P01116  Q9UM97,Q71SP6,P78460,L7RSL8,I1SRC5   \n",
       "1      ENSP00000324856    Q15831                              Q9NS52   \n",
       "2      ENSP00000171111    Q14145                K7ESE0,K7EJD8,K7EJ49   \n",
       "3      ENSP00000343896    P51532                       B4DSI8,A7E2E1   \n",
       "4      ENSP00000362294    O14522                                 NaN   \n",
       "...                ...       ...                                 ...   \n",
       "52110  ENSP00000253339    O95835                                 NaN   \n",
       "52111  ENSP00000288368    Q70Z35                              Q56UR8   \n",
       "52112  ENSP00000278616    Q13315                M0QXY8,E9PRG7,E9PIN0   \n",
       "52113  ENSP00000301067    O14686                Q6PIA1,Q59FG6,F8VWW4   \n",
       "52114  ENSP00000219066    P78549                              E5KTI5   \n",
       "\n",
       "             UNIPARC                   RefSeq               SIFT  \\\n",
       "0      UPI0000133132              NM_033360.2  deleterious(0.04)   \n",
       "1      UPI0000136105              NM_000455.4                NaN   \n",
       "2      UPI000007139C              NM_203500.1     deleterious(0)   \n",
       "3      UPI000006F973              NM_003072.3                NaN   \n",
       "4      UPI00001AF6FA              NM_133170.3     deleterious(0)   \n",
       "...              ...                      ...                ...   \n",
       "52110  UPI0000073DC2                      NaN    tolerated(0.07)   \n",
       "52111  UPI0000375435  NM_024870.2,NM_025170.4                NaN   \n",
       "52112  UPI0000DBEF44              NM_000051.3                NaN   \n",
       "52113  UPI0000EE84D6              NM_003482.3                NaN   \n",
       "52114  UPI0000130586              NM_002528.5  deleterious(0.01)   \n",
       "\n",
       "                       PolyPhen   EXON INTRON  \\\n",
       "0      probably_damaging(0.993)    2/6    NaN   \n",
       "1                           NaN   6/10    NaN   \n",
       "2      probably_damaging(0.998)    3/6    NaN   \n",
       "3                           NaN    NaN  12/34   \n",
       "4      probably_damaging(0.973)  25/32    NaN   \n",
       "...                         ...    ...    ...   \n",
       "52110             benign(0.443)    4/7    NaN   \n",
       "52111                       NaN  37/40    NaN   \n",
       "52112                       NaN  41/63    NaN   \n",
       "52113                       NaN  45/54    NaN   \n",
       "52114             benign(0.068)    5/6    NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                            DOMAINS  \\\n",
       "0      Gene3D:3.40.50.300,Pfam_domain:PF00071,Prints_domain:PR00449,PROSITE_profiles:PS51421,hmmpanther:PTHR24070,hmmpanther:PTHR24070:SF186,Low_complexity_(Seg):seg,SMART_domains:SM00173,SMART_domains:SM00174,SMART_domains:SM00175,SMART_domains:SM00176,Superfamily_domains:SSF52540,TIGRFAM_domain:TIGR00231   \n",
       "1                                                                                                                                                  Gene3D:1.10.510.10,Pfam_domain:PF00069,PROSITE_profiles:PS50011,hmmpanther:PTHR24347,hmmpanther:PTHR24347:SF1,SMART_domains:SM00220,Superfamily_domains:SSF56112   \n",
       "2                                                                                                                                                     hmmpanther:PTHR24412:SF162,hmmpanther:PTHR24412,Pfam_domain:PF01344,Gene3D:1k3iA02,PIRSF_domain:PIRSF037037,SMART_domains:SM00612,Superfamily_domains:0052715   \n",
       "3                                                                                                                                                                                                                                                                                                               NaN   \n",
       "4                                                  Gene3D:3.90.190.10,Pfam_domain:PF00102,Prints_domain:PR00700,PROSITE_patterns:PS00383,PROSITE_profiles:PS50055,PROSITE_profiles:PS50056,hmmpanther:PTHR19134,hmmpanther:PTHR19134:SF208,SMART_domains:SM00194,SMART_domains:SM00404,Superfamily_domains:SSF52799   \n",
       "...                                                                                                                                                                                                                                                                                                             ...   \n",
       "52110                                                                                                                                                                                                                                                               hmmpanther:PTHR24356:SF138,hmmpanther:PTHR24356   \n",
       "52111                                                                                                                                                                                                                                                                 hmmpanther:PTHR22829,hmmpanther:PTHR22829:SF1   \n",
       "52112                                                                                                                                                                                                                                       hmmpanther:PTHR11139,hmmpanther:PTHR11139:SF66,PROSITE_profiles:PS51189   \n",
       "52113                                                                                                                                                                                                                                                               hmmpanther:PTHR22884,hmmpanther:PTHR22884:SF324   \n",
       "52114                                                                                                                                                                                               Gene3D:1.10.1670.10,Pfam_domain:PF00730,hmmpanther:PTHR10359,SMART_domains:SM00478,Superfamily_domains:SSF48150   \n",
       "\n",
       "       GMAF  AFR_MAF  AMR_MAF  ASN_MAF  EAS_MAF  EUR_MAF  SAS_MAF  AA_MAF  \\\n",
       "0       NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "1       NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "2       NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "3       NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "4       NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "...     ...      ...      ...      ...      ...      ...      ...     ...   \n",
       "52110   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "52111   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "52112   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "52113   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "52114   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "\n",
       "       EA_MAF  CLIN_SIG  SOMATIC  PUBMED  MOTIF_NAME  MOTIF_POS  HIGH_INF_POS  \\\n",
       "0         NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "1         NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "2         NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "3         NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "4         NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "...       ...       ...      ...     ...         ...        ...           ...   \n",
       "52110     NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "52111     NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "52112     NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "52113     NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "52114     NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "\n",
       "       MOTIF_SCORE_CHANGE    IMPACT  PICK        VARIANT_CLASS  TSL  \\\n",
       "0                     NaN  MODERATE   1.0                indel  NaN   \n",
       "1                     NaN      HIGH   1.0  sequence_alteration  NaN   \n",
       "2                     NaN  MODERATE   1.0                indel  NaN   \n",
       "3                     NaN      HIGH   NaN                indel  NaN   \n",
       "4                     NaN  MODERATE   NaN                indel  NaN   \n",
       "...                   ...       ...   ...                  ...  ...   \n",
       "52110                 NaN  MODERATE   NaN                indel  NaN   \n",
       "52111                 NaN      HIGH   1.0            insertion  NaN   \n",
       "52112                 NaN      HIGH   1.0  sequence_alteration  NaN   \n",
       "52113                 NaN      HIGH   1.0  sequence_alteration  NaN   \n",
       "52114                 NaN  MODERATE   1.0                indel  NaN   \n",
       "\n",
       "       HGVS_OFFSET  PHENO  MINIMISED  ExAC_AF  ExAC_AF_AFR  ExAC_AF_AMR  \\\n",
       "0              NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "1              NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "2              NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "3              NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "4              NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "...            ...    ...        ...      ...          ...          ...   \n",
       "52110          NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "52111          3.0    NaN        NaN      NaN          NaN          NaN   \n",
       "52112          NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "52113          NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "52114          NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "\n",
       "       ExAC_AF_EAS  ExAC_AF_FIN  ExAC_AF_NFE  ExAC_AF_OTH  ExAC_AF_SAS  \\\n",
       "0              NaN          NaN          NaN          NaN          NaN   \n",
       "1              NaN          NaN          NaN          NaN          NaN   \n",
       "2              NaN          NaN          NaN          NaN          NaN   \n",
       "3              NaN          NaN          NaN          NaN          NaN   \n",
       "4              NaN          NaN          NaN          NaN          NaN   \n",
       "...            ...          ...          ...          ...          ...   \n",
       "52110          NaN          NaN          NaN          NaN          NaN   \n",
       "52111          NaN          NaN          NaN          NaN          NaN   \n",
       "52112          NaN          NaN          NaN          NaN          NaN   \n",
       "52113          NaN          NaN          NaN          NaN          NaN   \n",
       "52114          NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "       GENE_PHENO FILTER flanking_bps variant_id variant_qual  ExAC_AF_Adj  \\\n",
       "0             1.0    NaN          CCA          .            .          NaN   \n",
       "1             1.0    NaN         GGCC          .            .          NaN   \n",
       "2             NaN    NaN          CCC          .            .          NaN   \n",
       "3             1.0    NaN          GGT          .            .          NaN   \n",
       "4             1.0    NaN          GCA          .            .          NaN   \n",
       "...           ...    ...          ...        ...          ...          ...   \n",
       "52110         1.0    NaN          TGA          .            .          NaN   \n",
       "52111         NaN    NaN          CTG          .            .          NaN   \n",
       "52112         1.0    NaN         TGTA          .            .          NaN   \n",
       "52113         1.0    NaN         GCTT          .            .          NaN   \n",
       "52114         1.0    NaN          TCT          .            .          NaN   \n",
       "\n",
       "      ExAC_AC_AN_Adj ExAC_AC_AN ExAC_AC_AN_AFR ExAC_AC_AN_AMR ExAC_AC_AN_EAS  \\\n",
       "0                NaN        NaN            NaN            NaN            NaN   \n",
       "1                NaN        NaN            NaN            NaN            NaN   \n",
       "2                NaN        NaN            NaN            NaN            NaN   \n",
       "3                NaN        NaN            NaN            NaN            NaN   \n",
       "4                NaN        NaN            NaN            NaN            NaN   \n",
       "...              ...        ...            ...            ...            ...   \n",
       "52110            NaN        NaN            NaN            NaN            NaN   \n",
       "52111            NaN        NaN            NaN            NaN            NaN   \n",
       "52112            NaN        NaN            NaN            NaN            NaN   \n",
       "52113            NaN        NaN            NaN            NaN            NaN   \n",
       "52114            NaN        NaN            NaN            NaN            NaN   \n",
       "\n",
       "      ExAC_AC_AN_FIN ExAC_AC_AN_NFE ExAC_AC_AN_OTH ExAC_AC_AN_SAS  \\\n",
       "0                NaN            NaN            NaN            NaN   \n",
       "1                NaN            NaN            NaN            NaN   \n",
       "2                NaN            NaN            NaN            NaN   \n",
       "3                NaN            NaN            NaN            NaN   \n",
       "4                NaN            NaN            NaN            NaN   \n",
       "...              ...            ...            ...            ...   \n",
       "52110            NaN            NaN            NaN            NaN   \n",
       "52111            NaN            NaN            NaN            NaN   \n",
       "52112            NaN            NaN            NaN            NaN   \n",
       "52113            NaN            NaN            NaN            NaN   \n",
       "52114            NaN            NaN            NaN            NaN   \n",
       "\n",
       "                      ExAC_FILTER  Caller is-a-hotspot is-a-3d-hotspot  \\\n",
       "0                             NaN     NaN            Y               Y   \n",
       "1                             NaN     NaN          NaN             NaN   \n",
       "2                             NaN     NaN          NaN             NaN   \n",
       "3                             NaN     NaN          NaN             NaN   \n",
       "4                             NaN     NaN          NaN             NaN   \n",
       "...                           ...     ...          ...             ...   \n",
       "52110                         NaN     NaN          NaN             NaN   \n",
       "52111  VQSRTrancheSNP99.60to99.80     NaN          NaN             NaN   \n",
       "52112                         NaN     NaN          NaN             NaN   \n",
       "52113                         NaN     NaN          NaN             NaN   \n",
       "52114                         NaN     NaN          NaN             NaN   \n",
       "\n",
       "               mutation_effect         oncogenic LEVEL_1 LEVEL_2A LEVEL_2B  \\\n",
       "0             Gain-of-function         Oncogenic     NaN      NaN      NaN   \n",
       "1      Likely Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "2                          NaN               NaN     NaN      NaN      NaN   \n",
       "3      Likely Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "4                          NaN               NaN     NaN      NaN      NaN   \n",
       "...                        ...               ...     ...      ...      ...   \n",
       "52110                      NaN               NaN     NaN      NaN      NaN   \n",
       "52111                      NaN               NaN     NaN      NaN      NaN   \n",
       "52112  Likely Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "52113  Likely Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "52114                      NaN               NaN     NaN      NaN      NaN   \n",
       "\n",
       "      LEVEL_3A LEVEL_3B                             LEVEL_4 LEVEL_R1 LEVEL_R2  \\\n",
       "0          NaN      NaN  Binimetinib,Cobimetinib,Trametinib      NaN      NaN   \n",
       "1          NaN      NaN                                 NaN      NaN      NaN   \n",
       "2          NaN      NaN                                 NaN      NaN      NaN   \n",
       "3          NaN      NaN                                 NaN      NaN      NaN   \n",
       "4          NaN      NaN                                 NaN      NaN      NaN   \n",
       "...        ...      ...                                 ...      ...      ...   \n",
       "52110      NaN      NaN                                 NaN      NaN      NaN   \n",
       "52111      NaN      NaN                                 NaN      NaN      NaN   \n",
       "52112      NaN      NaN                            Olaparib      NaN      NaN   \n",
       "52113      NaN      NaN                                 NaN      NaN      NaN   \n",
       "52114      NaN      NaN                                 NaN      NaN      NaN   \n",
       "\n",
       "       LEVEL_R3 Highest_level  \\\n",
       "0           NaN       LEVEL_4   \n",
       "1           NaN           NaN   \n",
       "2           NaN           NaN   \n",
       "3           NaN           NaN   \n",
       "4           NaN           NaN   \n",
       "...         ...           ...   \n",
       "52110       NaN           NaN   \n",
       "52111       NaN           NaN   \n",
       "52112       NaN       LEVEL_4   \n",
       "52113       NaN           NaN   \n",
       "52114       NaN           NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                         citations  \\\n",
       "0      16051643;25705018;26841430;24256730;28783719;29247021;Bhagwat et al. Abstract# 4973, AACR 2017(http://cancerres.aacrjournals.org/content/77/13_Supplement/4973);Robarge et al. Abstract# DDT02-03, AACR 2014(http://cancerres.aacrjournals.org/content/74/19_Supplement/DDT02-03);Burrows et al. Abstract# 5168, AACR 2017(http://cancerres.aacrjournals.org/content/77/13_Supplement/5168)   \n",
       "1                                                                                                                                                                                                                                                                                                                                                     24652667;19892943;25079552;21516316;19340305   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                              NaN   \n",
       "3                                                                                                                                                                                                                                                                                                                                                     24658002;24658001;24658004;25060813;18301784   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                              NaN   \n",
       "...                                                                                                                                                                                                                                                                                                                                                                                            ...   \n",
       "52110                                                                                                                                                                                                                                                                                                                                                                                          NaN   \n",
       "52111                                                                                                                                                                                                                                                                                                                                                                                          NaN   \n",
       "52112                                                                                                                                                                                                                                                                                                                                                          16832357;16998505;15928302;26510020   \n",
       "52113                                                                                                                                                                                                                                                                                                                                                                   24633898;24240169;26366712   \n",
       "52114                                                                                                                                                                                                                                                                                                                                                                                          NaN   \n",
       "\n",
       "       driver  tcn  lcn        cf    purity  t_var_freq  expected_alt_copies  \\\n",
       "0        True  4.0  2.0  0.138437  0.273767    0.158163                  1.0   \n",
       "1        True  2.0  0.0  0.183720  0.273767    0.200000                  1.0   \n",
       "2       False  2.0  0.0  0.183720  0.273767    0.184087                  1.0   \n",
       "3        True  2.0  0.0  0.183720  0.273767    0.166292                  1.0   \n",
       "4       False  2.0  0.0  0.183720  0.273767    0.197452                  1.0   \n",
       "...       ...  ...  ...       ...       ...         ...                  ...   \n",
       "52110   False  4.0  2.0  0.652662  0.683828    0.405634                  2.0   \n",
       "52111   False  4.0  1.0  0.683828  0.683828    0.600476                  3.0   \n",
       "52112    True  3.0  1.0  0.683828  0.683828    0.223485                  1.0   \n",
       "52113    True  3.0  1.0  0.683828  0.683828    0.532808                  2.0   \n",
       "52114   False  3.0  1.0  0.683828  0.683828    0.538547                  2.0   \n",
       "\n",
       "       ccf_Mcopies  ccf_Mcopies_lower  ccf_Mcopies_upper  ccf_Mcopies_prob95  \\\n",
       "0            0.736              0.655              0.821        2.235549e-03   \n",
       "1            0.731              0.662              0.802        2.663028e-04   \n",
       "2            0.672              0.608              0.740        3.215666e-06   \n",
       "3            0.607              0.534              0.686        1.397430e-06   \n",
       "4            0.721              0.644              0.803        7.615616e-04   \n",
       "...            ...                ...                ...                 ...   \n",
       "52110        0.999              0.923              1.000        5.737254e-01   \n",
       "52111        0.986              0.952              1.000        8.608698e-01   \n",
       "52112        0.439              0.381              0.500        1.502461e-18   \n",
       "52113        1.000              0.983              1.000        9.665077e-01   \n",
       "52114        1.000              0.988              1.000        9.876220e-01   \n",
       "\n",
       "       ccf_Mcopies_prob90  ccf_1copy  ccf_1copy_lower  ccf_1copy_upper  \\\n",
       "0            1.483415e-02      1.000            0.979              1.0   \n",
       "1            3.570729e-03      1.000            0.984              1.0   \n",
       "2            9.339940e-05      1.000            0.980              1.0   \n",
       "3            2.653238e-05      1.000            0.959              1.0   \n",
       "4            6.531934e-03      1.000            0.979              1.0   \n",
       "...                   ...        ...              ...              ...   \n",
       "52110        8.879509e-01      1.000            0.992              1.0   \n",
       "52111        9.984730e-01      1.000            0.998              1.0   \n",
       "52112        1.324905e-15      0.877            0.762              1.0   \n",
       "52113        9.998115e-01      1.000            0.997              1.0   \n",
       "52114        9.999818e-01      1.000            0.997              1.0   \n",
       "\n",
       "       ccf_1copy_prob95  ccf_1copy_prob90  ccf_expected_copies  \\\n",
       "0              0.860769          0.984692                1.000   \n",
       "1              0.926527          0.996079                1.000   \n",
       "2              0.880435          0.989551                1.000   \n",
       "3              0.659996          0.904988                1.000   \n",
       "4              0.859171          0.984420                1.000   \n",
       "...                 ...               ...                  ...   \n",
       "52110          0.992534          0.999960                0.999   \n",
       "52111          1.000000          1.000000                0.986   \n",
       "52112          0.147953          0.350775                0.877   \n",
       "52113          1.000000          1.000000                1.000   \n",
       "52114          1.000000          1.000000                1.000   \n",
       "\n",
       "       ccf_expected_copies_lower  ccf_expected_copies_upper  \\\n",
       "0                          0.979                        1.0   \n",
       "1                          0.984                        1.0   \n",
       "2                          0.980                        1.0   \n",
       "3                          0.959                        1.0   \n",
       "4                          0.979                        1.0   \n",
       "...                          ...                        ...   \n",
       "52110                      0.923                        1.0   \n",
       "52111                      0.952                        1.0   \n",
       "52112                      0.762                        1.0   \n",
       "52113                      0.983                        1.0   \n",
       "52114                      0.988                        1.0   \n",
       "\n",
       "       ccf_expected_copies_prob95  ccf_expected_copies_prob90  \\\n",
       "0                        0.860769                    0.984692   \n",
       "1                        0.926527                    0.996079   \n",
       "2                        0.880435                    0.989551   \n",
       "3                        0.659996                    0.904988   \n",
       "4                        0.859171                    0.984420   \n",
       "...                           ...                         ...   \n",
       "52110                    0.573725                    0.887951   \n",
       "52111                    0.860870                    0.998473   \n",
       "52112                    0.147953                    0.350775   \n",
       "52113                    0.966508                    0.999811   \n",
       "52114                    0.987622                    0.999982   \n",
       "\n",
       "                                                                                                   facets_fit  \\\n",
       "0      /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "1      /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "2      /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "3      /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "4      /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "...                                                                                                       ...   \n",
       "52110  /juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default   \n",
       "52111  /juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default   \n",
       "52112  /juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default   \n",
       "52113  /juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default   \n",
       "52114  /juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default   \n",
       "\n",
       "       reviewer_set_purity  use_only_purity_run  use_edited_cncf  \\\n",
       "0                      NaN                False            False   \n",
       "1                      NaN                False            False   \n",
       "2                      NaN                False            False   \n",
       "3                      NaN                False            False   \n",
       "4                      NaN                False            False   \n",
       "...                    ...                  ...              ...   \n",
       "52110                  NaN                False            False   \n",
       "52111                  NaN                False            False   \n",
       "52112                  NaN                False            False   \n",
       "52113                  NaN                False            False   \n",
       "52114                  NaN                False            False   \n",
       "\n",
       "                                                                                                                                                   cncf_file_used  \\\n",
       "0      /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "1      /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "2      /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "3      /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "4      /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "...                                                                                                                                                           ...   \n",
       "52110  /juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default/P-0050660-T01-IM6_P-0050660-N01-IM6_hisens.cncf.txt   \n",
       "52111  /juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default/P-0050660-T01-IM6_P-0050660-N01-IM6_hisens.cncf.txt   \n",
       "52112  /juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default/P-0050660-T01-IM6_P-0050660-N01-IM6_hisens.cncf.txt   \n",
       "52113  /juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default/P-0050660-T01-IM6_P-0050660-N01-IM6_hisens.cncf.txt   \n",
       "52114  /juno/work/ccs/shared/resources/impact/facets/all/P-00506/P-0050660-T01-IM6_P-0050660-N01-IM6//default/P-0050660-T01-IM6_P-0050660-N01-IM6_hisens.cncf.txt   \n",
       "\n",
       "                mut_key                      sample_mut_key  \n",
       "0       12_25398285_C_A   P-0025956-T01-IM6-12_25398285_C_A  \n",
       "1        19_1221314_C_-    P-0025956-T01-IM6-19_1221314_C_-  \n",
       "2       19_10602584_C_T   P-0025956-T01-IM6-19_10602584_C_T  \n",
       "3       19_11113836_G_A   P-0025956-T01-IM6-19_11113836_G_A  \n",
       "4       20_40735560_C_A   P-0025956-T01-IM6-20_40735560_C_A  \n",
       "...                 ...                                 ...  \n",
       "52110   6_150001566_G_C   P-0050660-T01-IM6-6_150001566_G_C  \n",
       "52111    8_69104718_-_G    P-0050660-T01-IM6-8_69104718_-_G  \n",
       "52112  11_108186598_T_-  P-0050660-T01-IM6-11_108186598_T_-  \n",
       "52113   12_49422690_T_-   P-0050660-T01-IM6-12_49422690_T_-  \n",
       "52114    16_2090190_C_G    P-0050660-T01-IM6-16_2090190_C_G  \n",
       "\n",
       "[52115 rows x 168 columns]"
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_cohort_wgd"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Mutations\n",
    "In this part, we focus on tp53 mutation information.\n",
    "\n",
    "We gather all mutations per sample, and split it into different columns. We have the following columns:\n",
    "* Tumor_Id\t\n",
    "* key_1 (2,3,4,5) --> Mutation key allowing to filter duplicates\n",
    "* vc_1 (2,3,4,5) --> Variant Classification\n",
    "* ccf_1 (2,3,4,5) --> Cancer Cell Fraction of the mutation\n",
    "* vaf_1 (2,3,4,5) --> Variant Allele Frequency of the mutation\n",
    "* HGVSp_1 (2,3,4,5) --> protein change\n",
    "* spot_1 (2,3,4,5) --> Integer that defines the spot of the tp53 mutation\n",
    "* tp53_count --> Number of tp53 mutations of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def f_(x):\n",
    "    # This function helps us to group mutations together in a single cell per patient\n",
    "    return pd.DataFrame(dict(Tumor_Sample_Barcode = x['Tumor_Sample_Barcode'],  \n",
    "                        muts = \"%s\" % ','.join(x['sample_mut_key_vc_ccf_vaf_hgv_spot_vt'])))\n",
    "\n",
    "def count_tp53_muts(x):\n",
    "    count = 0\n",
    "    for i in range(1,6):\n",
    "        if x['tp53_key_' + str(i)]:\n",
    "            count+= 1\n",
    "    return count\n",
    "\n",
    "# WARNING: THis function needs sample_info to work\n",
    "def create_tp53_muts(cohort:str, sample_info):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        print('yes')\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    '''\n",
    "    This function aims to gather all tp53 mutation characteristics.\n",
    "    For each sample we gather the tp53 mutations and their characteristics for all patients.\n",
    "    '''\n",
    "    # We load the  table created in maf_tp53_creation.ipynb\n",
    "    maf_tp53 = maf_cohort[maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    maf_tp53['mut_spot'] = maf_tp53.HGVSp.str.extract('(\\d+)')\n",
    "\n",
    "    # We select only intresting columns\n",
    "    maf_tp53_filtered = maf_tp53[['Tumor_Sample_Barcode','sample_mut_key', 'Variant_Classification',\\\n",
    "                                        'ccf_expected_copies', 't_var_freq', 'HGVSp','mut_spot', 'Variant_Type' ]]\n",
    "\n",
    "    # Let's Merge mut_key,Variant_classification, CF, CCF, and VAF to gather them\n",
    "    maf_tp53_filtered['sample_mut_key_vc_ccf_vaf_hgv_spot_vt'] = maf_tp53_filtered.apply(lambda x: str(x.sample_mut_key)+'%'+str(x.Variant_Classification)+'%'+str(x.ccf_expected_copies)+'%'+str(x.t_var_freq)+'%'+str(x.HGVSp)+'%'+str(x.mut_spot)+'%'+str(x.Variant_Type), axis=1)\n",
    "\n",
    "    # We Select important columns\n",
    "    final = maf_tp53_filtered[['Tumor_Sample_Barcode', 'sample_mut_key_vc_ccf_vaf_hgv_spot_vt']]\n",
    "    # We groupby Patient_Id and apply the function above to group mutations\n",
    "    final = final.groupby(['Tumor_Sample_Barcode'], sort=False).apply(f_)\n",
    "\n",
    "    # We separate the different mutations into 5 different columns (5 is the max number of tp53 mutations in our cohort)\n",
    "    final[['mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5']] = final.muts.str.split(',', expand=True)\n",
    "    #final = final.drop(['mut_key_6'],axis=1)\n",
    "    # Split the columns into mut_key_ and vc_\n",
    "    final[['tp53_key_1','tp53_vc_1','tp53_ccf_1','tp53_vaf_1','tp53_HGVSp_1', 'tp53_spot_1','tp53_vt_1']] = final.mut_key_1.str.split('%', expand=True)\n",
    "    final[['tp53_key_2','tp53_vc_2','tp53_ccf_2','tp53_vaf_2','tp53_HGVSp_2', 'tp53_spot_2','tp53_vt_2']] = final.mut_key_2.str.split('%', expand=True)\n",
    "    final[['tp53_key_3','tp53_vc_3','tp53_ccf_3','tp53_vaf_3','tp53_HGVSp_3', 'tp53_spot_3','tp53_vt_3']] = final.mut_key_3.str.split('%', expand=True)\n",
    "    final[['tp53_key_4','tp53_vc_4','tp53_ccf_4','tp53_vaf_4','tp53_HGVSp_4', 'tp53_spot_4','tp53_vt_4']] = final.mut_key_4.str.split('%', expand=True)\n",
    "    final[['tp53_key_5','tp53_vc_5','tp53_ccf_5','tp53_vaf_5','tp53_HGVSp_5', 'tp53_spot_5','tp53_vt_5']] = final.mut_key_5.str.split('%', expand=True)\n",
    "\n",
    "    # We remove the muts column\n",
    "    final = final.drop(['muts','mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5'], axis=1)\n",
    "\n",
    "    # We remove duplicates\n",
    "    final = final.drop_duplicates('Tumor_Sample_Barcode')\n",
    "\n",
    "    # We add the cohort patients that are not tp53 positive\n",
    "    #First we create a dataframe with all missing samples\n",
    "\n",
    "    cohort_samples = set(sample_info.Tumor_Id)\n",
    "    final_samples = set(final.Tumor_Sample_Barcode)\n",
    "    missing_samp = pd.DataFrame(cohort_samples - final_samples, columns = ['Tumor_Sample_Barcode'])\n",
    "    #Then we append the two datframe\n",
    "    final = final.append(missing_samp)\n",
    "\n",
    "    # We rename the Tumor_Sample_Barcode column to have the same key as in other datframes\n",
    "    final = final.rename(columns={'Tumor_Sample_Barcode': 'Tumor_Id'})\n",
    "\n",
    "    # We add a last column tp53_count that represents the number of tp53 mutations per sample\n",
    "    final = final.where(final.notnull(), None)\n",
    "    final['tp53_count'] = final.apply(count_tp53_muts, axis = 1)\n",
    "\n",
    "    # We change the type of vafs column to float64 instead of strings\n",
    "    final = final.astype({'tp53_vaf_1': 'float64', 'tp53_vaf_2': 'float64', 'tp53_vaf_3': 'float64', 'tp53_vaf_4': 'float64', 'tp53_vaf_5': 'float64',\n",
    "                       'tp53_ccf_1': 'float64', 'tp53_ccf_2': 'float64', 'tp53_ccf_3': 'float64', 'tp53_ccf_4': 'float64', 'tp53_ccf_5': 'float64'})\n",
    "\n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_vt_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_vt_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_vt_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_vt_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_vt_5</th>\n",
       "      <th>tp53_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909-T01-IM6-17_7577121_G_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.798</td>\n",
       "      <td>0.312169</td>\n",
       "      <td>p.Arg273Cys</td>\n",
       "      <td>273</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546-T01-IM6-17_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.845070</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546-T02-IM6-17_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.636735</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837-T01-IM6-17_7578406_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.325843</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>175</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>P-0019444-T01-IM6</td>\n",
       "      <td>P-0019444-T01-IM6-17_7578461_C_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.169002</td>\n",
       "      <td>p.Val157Phe</td>\n",
       "      <td>157</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2134</th>\n",
       "      <td>P-0031854-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2135</th>\n",
       "      <td>P-0021436-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2136</th>\n",
       "      <td>P-0025567-T02-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2137</th>\n",
       "      <td>P-0016892-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2138</th>\n",
       "      <td>P-0044556-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6367 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Tumor_Id                        tp53_key_1          tp53_vc_1  \\\n",
       "8     P-0036909-T01-IM6  P-0036909-T01-IM6-17_7577121_G_A  Missense_Mutation   \n",
       "13    P-0023546-T01-IM6  P-0023546-T01-IM6-17_7578442_T_C  Missense_Mutation   \n",
       "16    P-0023546-T02-IM6  P-0023546-T02-IM6-17_7578442_T_C  Missense_Mutation   \n",
       "21    P-0018837-T01-IM6  P-0018837-T01-IM6-17_7578406_C_T  Missense_Mutation   \n",
       "48    P-0019444-T01-IM6  P-0019444-T01-IM6-17_7578461_C_A  Missense_Mutation   \n",
       "...                 ...                               ...                ...   \n",
       "2134  P-0031854-T01-IM6                              None               None   \n",
       "2135  P-0021436-T01-IM6                              None               None   \n",
       "2136  P-0025567-T02-IM6                              None               None   \n",
       "2137  P-0016892-T01-IM6                              None               None   \n",
       "2138  P-0044556-T01-IM6                              None               None   \n",
       "\n",
       "      tp53_ccf_1  tp53_vaf_1 tp53_HGVSp_1 tp53_spot_1 tp53_vt_1 tp53_key_2  \\\n",
       "8          0.798    0.312169  p.Arg273Cys         273       SNP       None   \n",
       "13         0.933    0.845070  p.Tyr163Cys         163       SNP       None   \n",
       "16         1.000    0.636735  p.Tyr163Cys         163       SNP       None   \n",
       "21         1.000    0.325843  p.Arg175His         175       SNP       None   \n",
       "48         1.000    0.169002  p.Val157Phe         157       SNP       None   \n",
       "...          ...         ...          ...         ...       ...        ...   \n",
       "2134         NaN         NaN         None        None      None       None   \n",
       "2135         NaN         NaN         None        None      None       None   \n",
       "2136         NaN         NaN         None        None      None       None   \n",
       "2137         NaN         NaN         None        None      None       None   \n",
       "2138         NaN         NaN         None        None      None       None   \n",
       "\n",
       "     tp53_vc_2  tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2 tp53_spot_2 tp53_vt_2  \\\n",
       "8         None         NaN         NaN         None        None      None   \n",
       "13        None         NaN         NaN         None        None      None   \n",
       "16        None         NaN         NaN         None        None      None   \n",
       "21        None         NaN         NaN         None        None      None   \n",
       "48        None         NaN         NaN         None        None      None   \n",
       "...        ...         ...         ...          ...         ...       ...   \n",
       "2134      None         NaN         NaN         None        None      None   \n",
       "2135      None         NaN         NaN         None        None      None   \n",
       "2136      None         NaN         NaN         None        None      None   \n",
       "2137      None         NaN         NaN         None        None      None   \n",
       "2138      None         NaN         NaN         None        None      None   \n",
       "\n",
       "     tp53_key_3 tp53_vc_3  tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3 tp53_spot_3  \\\n",
       "8          None      None         NaN         NaN         None        None   \n",
       "13         None      None         NaN         NaN         None        None   \n",
       "16         None      None         NaN         NaN         None        None   \n",
       "21         None      None         NaN         NaN         None        None   \n",
       "48         None      None         NaN         NaN         None        None   \n",
       "...         ...       ...         ...         ...          ...         ...   \n",
       "2134       None      None         NaN         NaN         None        None   \n",
       "2135       None      None         NaN         NaN         None        None   \n",
       "2136       None      None         NaN         NaN         None        None   \n",
       "2137       None      None         NaN         NaN         None        None   \n",
       "2138       None      None         NaN         NaN         None        None   \n",
       "\n",
       "     tp53_vt_3 tp53_key_4 tp53_vc_4  tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4  \\\n",
       "8         None       None      None         NaN         NaN         None   \n",
       "13        None       None      None         NaN         NaN         None   \n",
       "16        None       None      None         NaN         NaN         None   \n",
       "21        None       None      None         NaN         NaN         None   \n",
       "48        None       None      None         NaN         NaN         None   \n",
       "...        ...        ...       ...         ...         ...          ...   \n",
       "2134      None       None      None         NaN         NaN         None   \n",
       "2135      None       None      None         NaN         NaN         None   \n",
       "2136      None       None      None         NaN         NaN         None   \n",
       "2137      None       None      None         NaN         NaN         None   \n",
       "2138      None       None      None         NaN         NaN         None   \n",
       "\n",
       "     tp53_spot_4 tp53_vt_4 tp53_key_5 tp53_vc_5  tp53_ccf_5  tp53_vaf_5  \\\n",
       "8           None      None       None      None         NaN         NaN   \n",
       "13          None      None       None      None         NaN         NaN   \n",
       "16          None      None       None      None         NaN         NaN   \n",
       "21          None      None       None      None         NaN         NaN   \n",
       "48          None      None       None      None         NaN         NaN   \n",
       "...          ...       ...        ...       ...         ...         ...   \n",
       "2134        None      None       None      None         NaN         NaN   \n",
       "2135        None      None       None      None         NaN         NaN   \n",
       "2136        None      None       None      None         NaN         NaN   \n",
       "2137        None      None       None      None         NaN         NaN   \n",
       "2138        None      None       None      None         NaN         NaN   \n",
       "\n",
       "     tp53_HGVSp_5 tp53_spot_5 tp53_vt_5  tp53_count  \n",
       "8            None        None      None           1  \n",
       "13           None        None      None           1  \n",
       "16           None        None      None           1  \n",
       "21           None        None      None           1  \n",
       "48           None        None      None           1  \n",
       "...           ...         ...       ...         ...  \n",
       "2134         None        None      None           0  \n",
       "2135         None        None      None           0  \n",
       "2136         None        None      None           0  \n",
       "2137         None        None      None           0  \n",
       "2138         None        None      None           0  \n",
       "\n",
       "[6367 rows x 37 columns]"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_tp53_muts(cohort=cohort_type, sample_info=sample_info)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Copy Numbers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we gather the information from gene_level table.\n",
    "We creaste the following columns:\n",
    "* Sample_Id \n",
    "* tcn --> total copy number\n",
    "* mcn --> major copy number\n",
    "* lcn --> lower copy number\n",
    "* seg_length --> length of the segment\n",
    "* cn_state --> copy number state\n",
    "* cf --> Cell fraction of the cn_state\n",
    "* wgd --> Wholde Genome Doubling (True or False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [],
   "source": [
    "# WARNING: THis function needs sample_info to work\n",
    "def create_copy_number_state(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    # We want TP53 locus so we have to filter the gene\n",
    "    gene_level = gene_level[gene_level['gene'] == 'TP53']\n",
    "\n",
    "    gene_level['Tumor_Id'] = gene_level['sample'].str[:17]\n",
    "    gene_level_subset = gene_level[['sample','tcn','mcn','lcn','seg_length','cn_state', 'cf.em']]\n",
    "    \n",
    "    # We rename the cf.em column \n",
    "    gene_level_subset = gene_level_subset.rename(columns={'cf.em': 'tp53_cf', \n",
    "                                                          'sample':'Sample_Id',\n",
    "                                                          'tcn': 'tp53_tcn',\n",
    "                                                          'mcn': 'tp53_mcn',\n",
    "                                                          'lcn': 'tp53_lcn',\n",
    "                                                          'seg_length': 'tp53_seg_length',\n",
    "                                                          'cn_state':'tp53_cn_state'})\n",
    "    \n",
    "    # We add WGD information\n",
    "    wgd = cohort[['sample_id', 'wgd']]\n",
    "    \n",
    "    final = pd.merge(gene_level_subset, wgd, left_on='Sample_Id', right_on='sample_id').drop(['sample_id'], axis=1)\n",
    "    \n",
    "    \n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 19.1 ms, sys: 5.4 ms, total: 24.5 ms\n",
      "Wall time: 23.3 ms\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>tp53_tcn</th>\n",
       "      <th>tp53_mcn</th>\n",
       "      <th>tp53_lcn</th>\n",
       "      <th>tp53_seg_length</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>tp53_cf</th>\n",
       "      <th>wgd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668431</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>0.169170</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18101929</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.306121</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25250470</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.835504</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25237770</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25231975</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.257400</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6362</th>\n",
       "      <td>P-0050671-T01-IM6_P-0050671-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>46689400</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.300360</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6363</th>\n",
       "      <td>P-0009110-T02-IM6_P-0009110-N01-IM6</td>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7438495</td>\n",
       "      <td>CNLOH BEFORE</td>\n",
       "      <td>0.239690</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6364</th>\n",
       "      <td>P-0050668-T01-IM6_P-0050668-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>80668561</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.197170</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6365</th>\n",
       "      <td>P-0050675-T01-IM6_P-0050675-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21375632</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.376084</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6366</th>\n",
       "      <td>P-0050660-T01-IM6_P-0050660-N01-IM6</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>29371203</td>\n",
       "      <td>TETRAPLOID</td>\n",
       "      <td>0.657819</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6367 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id  tp53_tcn  tp53_mcn  tp53_lcn  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6         3       2.0       1.0   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6         2       2.0       0.0   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6         3       3.0       0.0   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6         2       2.0       0.0   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6         3       3.0       0.0   \n",
       "...                                   ...       ...       ...       ...   \n",
       "6362  P-0050671-T01-IM6_P-0050671-N01-IM6         2       2.0       0.0   \n",
       "6363  P-0009110-T02-IM6_P-0009110-N01-IM6         4       4.0       0.0   \n",
       "6364  P-0050668-T01-IM6_P-0050668-N01-IM6         3       3.0       0.0   \n",
       "6365  P-0050675-T01-IM6_P-0050675-N01-IM6         2       2.0       0.0   \n",
       "6366  P-0050660-T01-IM6_P-0050660-N01-IM6         4       2.0       2.0   \n",
       "\n",
       "      tp53_seg_length        tp53_cn_state   tp53_cf   wgd  \n",
       "0            80668431           LOSS AFTER  0.169170  True  \n",
       "1            18101929          LOSS BEFORE  0.306121  True  \n",
       "2            25250470  CNLOH BEFORE & LOSS  0.835504  True  \n",
       "3            25237770          LOSS BEFORE  0.312907  True  \n",
       "4            25231975  CNLOH BEFORE & LOSS  0.257400  True  \n",
       "...               ...                  ...       ...   ...  \n",
       "6362         46689400          LOSS BEFORE  0.300360  True  \n",
       "6363          7438495         CNLOH BEFORE  0.239690  True  \n",
       "6364         80668561  CNLOH BEFORE & LOSS  0.197170  True  \n",
       "6365         21375632          LOSS BEFORE  0.376084  True  \n",
       "6366         29371203           TETRAPLOID  0.657819  True  \n",
       "\n",
       "[6367 rows x 8 columns]"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time \n",
    "copy_number_info = create_copy_number_state(cohort = cohort_type)\n",
    "copy_number_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computed metrics\n",
    "In this part we define functions to be applied on the master file to compute specific metrics.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes, mutations and max_vaf\n",
    "* *create_gene_count*: count of mutated genes for the given sample\n",
    "* *create_mut_count*: count of mutations for the given sample\n",
    "* *get_max_vaf*: the maximum of Variant Allele Frequency within all mutation of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [],
   "source": [
    "# These 2 first functions allow  to count the total number of genes/mutations per sample.\n",
    "def create_gene_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function create the count of genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    \n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "\n",
    "    nb_genes = selected_cohort.groupby('Entrez_Gene_Id').size().shape[0]\n",
    "\n",
    "    return nb_genes\n",
    "\n",
    "def create_mut_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "    \n",
    "    return selected_cohort.shape[0]\n",
    "\n",
    "# The following calculates the max_vaf for a given sample\n",
    "def get_max_vaf(x, maf_cohort):\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    \n",
    "    return selected_cohort['t_var_freq'].max()\n",
    "\n",
    "\n",
    "def create_driver_gene_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function create the count of driver mut genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    \n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['driver'] == True][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "\n",
    "    nb_genes = selected_cohort.groupby('Entrez_Gene_Id').size().shape[0]\n",
    "\n",
    "    return nb_genes\n",
    "\n",
    "def create_driver_mut_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all driver mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['driver'] == True][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "    \n",
    "    return selected_cohort.shape[0]\n",
    "\n",
    "def create_snv_driver_mut_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all driver mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['driver'] == True][maf_cohort['Variant_Type'].isin(['SNP', 'DNP', 'TNP', 'ONP'])][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "    \n",
    "    return selected_cohort.shape[0]\n",
    "\n",
    "def create_indel_driver_mut_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all driver mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['driver'] == True][maf_cohort['Variant_Type'].isin(['DEL', 'INS'])][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "    \n",
    "    return selected_cohort.shape[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genome Instability (frac_genome_aletered, chr_affected by loss, cnkloh, gain )\n",
    "This script is very long to run, so I decide to run it apart and save the results in a pickle fil that we will merge with our master.\n",
    "\n",
    "* For no_wgd: 40 minutes (MacBook Pro 2019)\n",
    "* For wgd: 11 minutes (MacBook Pro 2019)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "def condition_CNLOH(x):\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    if x.cn_state in CNLOH:\n",
    "        return 'CNLOH' + x.chr\n",
    "    else:\n",
    "        return 'NO_CNLOH'  + x.chr \n",
    "    \n",
    "def condition_GAIN(x):\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    if x.cn_state in GAIN:\n",
    "        return 'GAIN'+ x.chr\n",
    "    else:\n",
    "        return 'NO_GAIN'+ x.chr\n",
    "\n",
    "def condition_LOSS(x):\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    if x.cn_state in LOSS:\n",
    "        return 'LOSS'+ x.chr\n",
    "    else:\n",
    "        return 'NO_LOSS'+ x.chr\n",
    "\n",
    "def compute_frac_genome(x, arm_level):\n",
    "    Sample_Id = x.Sample_Id\n",
    "    wgd = x.wgd\n",
    "    lookup_table = arm_level[arm_level['sample'] == Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "        \n",
    "    altered_length = lookup_table_altered.cn_length.sum()\n",
    "    total_length = lookup_table.arm_length.sum()\n",
    "    \n",
    "    frac_gen_altered = round(altered_length/total_length,3)\n",
    "    \n",
    "    return frac_gen_altered\n",
    "\n",
    "def compute_frac_genome_without_cnloh(x, arm_level):\n",
    "    Sample_Id = x.Sample_Id\n",
    "    wgd = x.wgd\n",
    "    lookup_table = arm_level[arm_level['sample'] == Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table_altered = lookup_table[~lookup_table['cn_state'].isin(['TETRAPLOID', 'CNLOH BEFORE', 'CNLOH AFTER'])][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table_altered = lookup_table[~lookup_table['cn_state'].isin(['DIPLOID', 'CNLOH', 'CNLOH & GAIN'])][lookup_table['chr'] != '17']\n",
    "        \n",
    "    altered_length = lookup_table_altered.cn_length.sum()\n",
    "    total_length = lookup_table.arm_length.sum()\n",
    "    \n",
    "    frac_gen_altered = round(altered_length/total_length,3)\n",
    "    \n",
    "    return frac_gen_altered\n",
    "\n",
    "\n",
    "# Here is the function that allws to compute genome instability columns\n",
    "def chr_computations(x, arm_level):\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    wgd=x.wgd\n",
    "    arm_level_samples = list(set(arm_level['sample']))\n",
    "    \n",
    "    if x.Sample_Id not in arm_level_samples:\n",
    "        return ['NaN','NaN','NaN','NaN', 'NaN']\n",
    "    \n",
    "    lookup_table = arm_level[arm_level['sample'] == x.Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "    lookup_table['state_chr'] = lookup_table['cn_state']+lookup_table['chr']\n",
    "    \n",
    "    # If only DIPLOID or TETRAPLOID\n",
    "    if lookup_table.empty:\n",
    "        return [float(0)]*5\n",
    "    \n",
    "    lookup_table['cnloh_chr'] = lookup_table.apply(condition_CNLOH, axis=1)\n",
    "    lookup_table['loss_chr'] = lookup_table.apply(condition_LOSS, axis=1)\n",
    "    lookup_table['gain_chr'] = lookup_table.apply(condition_GAIN, axis=1)\n",
    "\n",
    "    #chr_affected colum\n",
    "    lookup_table_chr = lookup_table.drop_duplicates(subset=['chr'])\n",
    "    chr_affected = len(lookup_table_chr)\n",
    "    \n",
    "    #chr_loss, chr_gain, chr_cnloh columns\n",
    "    lookup_table_cnloh = lookup_table.drop_duplicates(subset=['cnloh_chr'])['cnloh_chr']\n",
    "    lookup_table_loss = lookup_table.drop_duplicates(subset=['loss_chr'])['loss_chr']\n",
    "    lookup_table_gain = lookup_table.drop_duplicates(subset=['gain_chr'])['gain_chr']\n",
    "\n",
    "    chr_loss = len(lookup_table_loss[lookup_table_loss.str.startswith('LOSS')])\n",
    "    chr_gain = len(lookup_table_gain[lookup_table_gain.str.startswith('GAIN')])\n",
    "    chr_cnloh = len(lookup_table_cnloh[lookup_table_cnloh.str.startswith('CNLOH')])\n",
    "    \n",
    "    #frac_gen_altered column\n",
    "    frac_gen_altered = compute_frac_genome(x, arm_level=arm_level)\n",
    "    \n",
    "    # frac_gen_altered_without_cnloh\n",
    "    frac_gen_altered_without_cnloh = compute_frac_genome(x, arm_level=arm_level)\n",
    "    \n",
    "    return [chr_affected, chr_loss, chr_gain, chr_cnloh, frac_gen_altered, frac_gen_altered_without_cnloh]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You do not need to run the following, excpet if you want to recompute the genome instability metrics.\n",
    "\n",
    "The following cell allows to run the above functions and to save it in a file that we will call in the final merging cell."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "97e63fca7e104277b23775a8f152a1ff",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=1.0, bar_style='info', max=1.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f279732644b4476fa6047dc69b203b29",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=0.0, max=6367.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "checkpoint 1\n",
      "checkpoint 2\n"
     ]
    }
   ],
   "source": [
    "from tqdm import tqdm,tqdm_notebook\n",
    "\n",
    "def compute_genome_instability(cohort:str):\n",
    "    tqdm_notebook().pandas()\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    copy_number_info = create_copy_number_state(cohort = cohort_type)\n",
    "    copy_number_info_ = copy_number_info\n",
    "    copy_number_info_['chr_comput'] = copy_number_info_.progress_apply(chr_computations,arm_level=arm_level, axis=1)\n",
    "    print('checkpoint 1')\n",
    "    copy_number_info[['chr_affected', 'chr_loss', 'chr_gain', 'chr_cnloh', 'frac_genome_altered', 'frac_gen_altered_without_cnloh']] = pd.DataFrame(copy_number_info_.chr_comput.values.tolist(), index= copy_number_info_.index)\n",
    "    print('checkpoint 2')\n",
    "    \n",
    "    return copy_number_info_\n",
    "\n",
    "copy_number_info = compute_genome_instability(cohort=cohort_type)\n",
    "copy_number_info[['Sample_Id','chr_affected', 'chr_loss', 'chr_gain', 'chr_cnloh', 'frac_genome_altered', 'frac_gen_altered_without_cnloh']].to_pickle(total_output + 'gi_metrics_{}.pkl'.format(cohort_type))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TP53 Residual"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following function needs to be called on the complete master file because it needs info from different parts\n",
    "# It computes the expected number of tp53 mutant copies in a cell\n",
    "def create_copies_tp53_muts(master):\n",
    "    master['tp53_exp_nb_1'] = master.apply(lambda x:(x.tp53_vaf_1 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_2'] = master.apply(lambda x:(x.tp53_vaf_2 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_3'] = master.apply(lambda x:(x.tp53_vaf_3 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_4'] = master.apply(lambda x:(x.tp53_vaf_4 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_5'] = master.apply(lambda x:(x.tp53_vaf_5 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    \n",
    "    return master\n",
    "\n",
    "\n",
    "# The following computes the expected number of copies of tp53 residuals \n",
    "def create_tp53_res(master):\n",
    "    master['tp53_res_1'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_1, axis = 1)\n",
    "    master['tp53_res_2'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_2, axis = 1)\n",
    "    master['tp53_res_3'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_3, axis = 1)\n",
    "    master['tp53_res_4'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_4, axis = 1)\n",
    "    master['tp53_res_5'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_5, axis = 1)\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mutation Types + Variant Type Grouping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following functions allow to group the Mutation Types\n",
    "def vc_group_cond_1(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_1 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_1 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_1 in missense: \n",
    "        if x.tp53_spot_1 in ['273','248','175']: return x.tp53_spot_1\n",
    "        elif x.tp53_spot_1 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_2(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_2 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_2 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_2 in missense: \n",
    "        if x.tp53_spot_2 in ['273','248','175']: return x.tp53_spot_2\n",
    "        elif x.tp53_spot_2 in['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'   \n",
    "def vc_group_cond_3(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_3 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_3 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_3 in missense: \n",
    "        if x.tp53_spot_3 in ['273','248','175']: return x.tp53_spot_3\n",
    "        elif x.tp53_spot_3 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense' \n",
    "def vc_group_cond_4(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_4 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_4 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_4 in missense: \n",
    "        if x.tp53_spot_4 in ['273','248','175']: return x.tp53_spot_4\n",
    "        elif x.tp53_spot_4 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_5(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_5 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_5 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_5 in missense: \n",
    "        if x.tp53_spot_5 in ['273','248','175']: return x.tp53_spot_5\n",
    "        elif x.tp53_spot_5 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "        \n",
    "\n",
    "#############################################################################################################\n",
    "\n",
    "def vt_group_cond_1(x):\n",
    "    indel = ['DEL', 'INS']\n",
    "    snv = ['SNP', 'DNP', 'TNP', 'ONP']\n",
    "    if x.tp53_vt_1 in indel:\n",
    "        return 'INDEL'\n",
    "    elif x.tp53_vt_1 in snv:\n",
    "        return 'SNV'\n",
    "    \n",
    "def vt_group_cond_2(x):\n",
    "    indel = ['DEL', 'INS']\n",
    "    snv = ['SNP', 'DNP', 'TNP', 'ONP']\n",
    "    if x.tp53_vt_2 in indel:\n",
    "        return 'INDEL'\n",
    "    elif x.tp53_vt_2 in snv:\n",
    "        return 'SNV'\n",
    "    \n",
    "def vt_group_cond_3(x):\n",
    "    indel = ['DEL', 'INS']\n",
    "    snv = ['SNP', 'DNP', 'TNP', 'ONP']\n",
    "    if x.tp53_vt_3 in indel:\n",
    "        return 'INDEL'\n",
    "    elif x.tp53_vt_3 in snv:\n",
    "        return 'SNV'\n",
    "    \n",
    "def vt_group_cond_4(x):\n",
    "    indel = ['DEL', 'INS']\n",
    "    snv = ['SNP', 'DNP', 'TNP', 'ONP']\n",
    "    if x.tp53_vt_4 in indel:\n",
    "        return 'INDEL'\n",
    "    elif x.tp53_vt_4 in snv:\n",
    "        return 'SNV'\n",
    "    \n",
    "def vt_group_cond_5(x):\n",
    "    indel = ['DEL', 'INS']\n",
    "    snv = ['SNP', 'DNP', 'TNP', 'ONP']\n",
    "    if x.tp53_vt_5 in indel:\n",
    "        return 'INDEL'\n",
    "    elif x.tp53_vt_5 in snv:\n",
    "        return 'SNV'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Non-WGD specific functions\n",
    "\n",
    "* *cn_group_cond*\n",
    "\n",
    "First, we group the different Copy Number States *cn_state* in subgroups, under the column *cn_group*:\n",
    "    * Group 1: cnLOH gathering ['CNLOH']\n",
    "    * Group 2: LOSS gathering ['HETLOSS']\n",
    "    * Group 3: HOMDEL gathering ['HOMDEL']\n",
    "    * Group 4: WILD_TYPE gathering ['DIPLOID', 'TETRAPLOID']\n",
    "    * Group 5: GAIN gathering ['GAIN']\n",
    "    * Group 6: OTHER gathering ['AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']\n",
    "\n",
    "* *mut_cn_group_cond*\n",
    "\n",
    "Based on this first column we define 7 final groups of patients adding the mutational information. These groups will be under the column *tp53_group*.\n",
    "    * Group 1: Samples with 0 tp53 mutations and HETLOSS\n",
    "    * Group 2: Samples with HOMDEL\n",
    "    * Group 3: Samples with 1 tp53 mutation and WILD_TYPE (DIPLOID, LOSS AFTER, TETRAPLOID)\n",
    "    * Group 4: Samples with 1 tp53 mutation or more and LOSS\n",
    "    * Group 5: Samples with 1 tp53 mutation or more and cnLOH\n",
    "    * Group 6: Samples with 2/3/4/5 tp53 mutations and WILD_TYPE or GAIN\n",
    "\n",
    "* *tp53_residual_group*: defines if we have residual tp53 or not based on the expected WT tp53 residual\n",
    "\n",
    "\n",
    "* *get_loh_nowgd*\n",
    "\n",
    "The last function is meant to compute the LOH state"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [],
   "source": [
    "def cn_group_cond(x):\n",
    "    if x.tp53_cn_state in ['CNLOH']:\n",
    "        return 'cnLOH'\n",
    "    if x.tp53_cn_state in ['HETLOSS']:\n",
    "        return 'LOSS'\n",
    "    if x.tp53_cn_state == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_cn_state in ['DIPLOID', 'TETRAPLOID']:\n",
    "        return 'WILD_TYPE'\n",
    "    if x.tp53_cn_state == 'GAIN':\n",
    "        return 'GAIN'\n",
    "    if x.tp53_cn_state in ['AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']:\n",
    "        return 'OTHER'\n",
    "\n",
    "def mut_cn_group_cond(x):\n",
    "    if x.tp53_cn_state == 'HETLOSS' and x.tp53_count == 0:\n",
    "        return '0_HETLOSS'\n",
    "    if x.tp53_first_group == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_first_group == 'WILD_TYPE' and x.tp53_count == 1 :\n",
    "        return '1_WILD_TYPE'\n",
    "    if x.tp53_first_group == 'LOSS' and x.tp53_count >=1:\n",
    "        return '>=1_LOSS'\n",
    "    if x.tp53_first_group == 'cnLOH' and x.tp53_count >=1:\n",
    "        return '>=1_cnLOH'\n",
    "    if (x.tp53_first_group == 'WILD_TYPE' or x.tp53_first_group == 'GAIN') and x.tp53_count > 1:\n",
    "        return '>1muts'\n",
    "    \n",
    "    \n",
    "def tp53_residual_group(x):\n",
    "    if x.tp53_group == '1_WILD_TYPE' or x.tp53_group == '0_HETLOSS':\n",
    "        return 'tp53_res'\n",
    "    if x.tp53_group == 'HOMDEL':\n",
    "        return 'no_tp53_res'\n",
    "    if x.tp53_group == '>=1_LOSS' or x.tp53_group == '>=1_cnLOH':\n",
    "        if (x.tp53_res_1 < 0.5) or (x.tp53_res_2 < 0.5):\n",
    "            return 'no_tp53_res'\n",
    "        else:\n",
    "            if (x.tp53_cf + max(x.tp53_ccf_1, x.tp53_ccf_2, x.tp53_ccf_3, x.tp53_ccf_4, x.tp53_ccf_5)) > 1:\n",
    "                return 'no_tp53_res'\n",
    "            else:\n",
    "                return 'uncertain'\n",
    "    if x.tp53_group == '>1muts':\n",
    "        if (x.tp53_res_1 + x.tp53_res_2 < 2.5):\n",
    "            if (x.tp53_ccf_1 + x.tp53_ccf_2 > 1):\n",
    "                return 'no_tp53_res'\n",
    "            else: \n",
    "                return 'uncertain'\n",
    "           \n",
    "        elif (x.tp53_res_1 + x.tp53_res_2 > 2.5):\n",
    "            return 'tp53_res'\n",
    "        \n",
    "        \n",
    "def get_loh_nowgd(x):\n",
    "    if x.tp53_lcn == 0:\n",
    "        return True\n",
    "    else: return False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WGD specific functions\n",
    "Finally we have functions to compute TP53 allelic state BEFORE WGD (for the WGD cohort only):\n",
    "* *get_bi_nobi* allows to say if the pre WGD allelic state was Bi-Allelic or not\n",
    "* *get_mono* allows to say if, within Not Bi-allelic samples, the state was mono-allelic or with 2 WT alleles\n",
    "* *get_loh_wgd* that computes if the samples is in LOH state or not"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "            NUMBER OF TP53 residual ASSOCIATED WITH ALLELIC STATE BEFORE WGD\n",
    "            \n",
    "            \n",
    "                                     BI-ALLELIC | MONO - ALLELIC | 2WT\n",
    "\n",
    "                LOSS BEFORE :            0      |        1       | -\n",
    "                CNLOH BEFORE & LOSS:     0      |       1,2      | 2\n",
    "                CNLOH BEFORE:            0      |        2       | 3\n",
    "                LOSS AFTER:              -      |       1,2      | 2\n",
    "                DOUBLE LOSS AFTER:       -      |       0,1      | 1\n",
    "                TETRAPLOID:              -      |        2       | 3\n",
    "                CNLOH AFTER:             -      |       1,3      | 3\n",
    "                CNLOH BEFORE & GAIN:     0      |        2,3     | 3,4\n",
    "                \n",
    "\n",
    "                        THRESHOLDS BETWEEN ALLELIC STATES BEFORE WGD\n",
    "                    \n",
    "                                     BI-ALLELIC | MONO - ALLELIC | 2WT\n",
    "\n",
    "                LOSS BEFORE :            <0.4   |        >0.6    | -\n",
    "                CNLOH BEFORE & LOSS:     <0.4   |      0.6< <1.5 | uncertain\n",
    "                CNLOH BEFORE:            <1.5   |   1.5< <2.5    | >2.5\n",
    "                LOSS AFTER:              -      |      <1.5      | uncertain \n",
    "                DOUBLE LOSS AFTER:       -      |       <0.5     | uncertain\n",
    "                TETRAPLOID:              -      |       <2.5     | >2.5\n",
    "                CNLOH AFTER:             -      |       <1.5     | uncertain\n",
    "                CNLOH BEFORE & GAIN:     <1.4   |   1.6< <2.5    | uncertain\n",
    "                \n",
    "    \n",
    "\n",
    " '''\n",
    "\n",
    "\n",
    "# The following functions are for WGD cohort, computing the tp53 allelic state before WGD\n",
    "def get_bi_nobi(x, maf_cohort):\n",
    "    tumor = x.Tumor_Id\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    \n",
    "    #samples without TP53 mutation\n",
    "    if nb_tp53muts == 0:\n",
    "        return 'no_bi'\n",
    "    \n",
    "    # Samples with only one tp53 mutation\n",
    "    if nb_tp53muts == 1:\n",
    "        tp53_res = x.tp53_res_1\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE' or cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr = 0.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr:\n",
    "                return 'bi'\n",
    "            elif tp53_res >= thr:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state in ['LOSS AFTER','DOUBLE LOSS AFTER','TETRAPLOID','CNLOH AFTER']: \n",
    "            return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "            \n",
    "            \n",
    "            \n",
    "    # Samples with 2 tp53 mutations\n",
    "    elif nb_tp53muts == 2:\n",
    "        tp53_res_1 = x.tp53_res_1\n",
    "        tp53_res_2 = x.tp53_res_2\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr - 0.1) or (tp53_res_2 < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr + 0.1) and (tp53_res_2 > thr + 0.1):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (tp53_res_1 < thr_1 - 0.1) or (tp53_res_2 < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (tp53_res_1 < thr_1) or (tp53_res_2 < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if tp53_res_1 < thr_1 - 0.1 or tp53_res_2 < thr_1 - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr_2) and (tp53_res_2 > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "            \n",
    "    # Samples with 3 tp53 mutations\n",
    "    elif nb_tp53muts == 3:\n",
    "        tp53_res_1 = x.tp53_res_1\n",
    "        tp53_res_2 = x.tp53_res_2\n",
    "        tp53_res_3 = x.tp53_res_3\n",
    "        tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "        tp53_res.sort()\n",
    "    \n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (min(tp53_res) < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (max(tp53_res) > thr + 0.1) :\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (min(tp53_res) < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2): # we want that exactly 2 mutation arose before WGD\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (min(tp53_res) < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if min(tp53_res) < thr_1 - 0.1 :\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "    \n",
    "    else: return 'uncertain'\n",
    "\n",
    "\n",
    "# This functions can only be called after get_bi_nobi() has been called on master\n",
    "def get_mono(x, maf_cohort):\n",
    "    tumor = x.Tumor_Id\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    bi_state = x.tp53_bi_state\n",
    "    \n",
    "    if bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    \n",
    "    elif bi_state == 'uncertain':\n",
    "        return 'uncertain_bi'\n",
    "    \n",
    "    \n",
    "    elif bi_state == 'no_bi': # We are already in the  mono/2WT distinction \n",
    "        \n",
    "        if nb_tp53muts == 0:\n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            else: return '2WT'\n",
    "            \n",
    "            \n",
    "        #1 mut samples\n",
    "        if nb_tp53muts == 1:\n",
    "            tp53_res = x.tp53_res_1\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "\n",
    "        \n",
    "        # 2 mut samples   \n",
    "        elif nb_tp53muts == 2:\n",
    "            tp53_res_1 = x.tp53_res_1\n",
    "            tp53_res_2 = x.tp53_res_2\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res_1 < 2.5 or tp53_res_2 < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res_1 > 2.5 and tp53_res_2 > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "            \n",
    "        # 3 mut samples\n",
    "        elif nb_tp53muts == 3:\n",
    "            tp53_res_1 = x.tp53_res_1\n",
    "            tp53_res_2 = x.tp53_res_2\n",
    "            tp53_res_3 = x.tp53_res_3\n",
    "            tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "            tp53_res.sort()\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if min(tp53_res) < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res[0] > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "        \n",
    "        else: return 'uncertain_mono' \n",
    "        \n",
    "def get_loh_wgd(x):\n",
    "    if x.tp53_cn_state in ['LOSS BEFORE', 'CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS BEFORE & AFTER', 'CNLOH BEFORE & GAIN']:\n",
    "        return True\n",
    "    else: return False\n",
    "    \n",
    "    \n",
    "def get_prewgd_tp53_group_1(x):\n",
    "    if x.tp53_bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    elif x.tp53_bi_state == 'no_bi':\n",
    "        return 'tp53_res'\n",
    "    \n",
    "    elif x.tp53_count == 0:\n",
    "        return 'tp53_res'\n",
    "    \n",
    "    \n",
    "def get_prewgd_tp53_group_2(x):\n",
    "    if x.tp53_bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    \n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == True and x.tp53_cn_state == 'LOSS BEFORE':\n",
    "        return 'mono'\n",
    "    elif x.tp53_state_wgd == 'mono':\n",
    "        return 'mono'\n",
    "    \n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == True and x.tp53_cn_state in ['CNLOH BEFORE', 'CNLOH BEFORE & GAIN', 'CNLOH BEFORE & LOSS']:\n",
    "        return 'cnloh_2WT'\n",
    "    \n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == False:\n",
    "        return '2WT'\n",
    "    elif x.tp53_state_wgd == '2WT':\n",
    "        return '2WT'\n",
    "    \n",
    "    \n",
    "def get_loh_mut_status(x):\n",
    "    if x.tp53_count == 0 and x.tp53_loh_status == True:\n",
    "        return '0_loh'\n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == False:\n",
    "        return '0_noloh'\n",
    "    elif x.tp53_count >= 1 and x.tp53_loh_status == True:\n",
    "        return '1_loh'\n",
    "    elif x.tp53_count >= 1 and x.tp53_loh_status == False:\n",
    "        return '1_noloh'\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge and save tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "def merge_tables(cohort_type: str):\n",
    "    \n",
    "    if cohort_type == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort_type == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    sample_info = create_sample_info(cohort=cohort_type)\n",
    "    tp53_muts = create_tp53_muts(cohort=cohort_type, sample_info=sample_info)\n",
    "    copy_number_info = create_copy_number_state(cohort=cohort_type)\n",
    "    \n",
    "    master = pd.merge(sample_info, tp53_muts, on='Tumor_Id')\n",
    "    master = pd.merge(master, copy_number_info, on='Sample_Id')\n",
    "    print('checkpoint_1')\n",
    "    # Now we compute the columns\n",
    "    master['gene_count'] = master.apply(create_gene_count, maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_1')\n",
    "    master['mutation_count'] = master.apply(create_mut_count,maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_1')\n",
    "    master['driver_gene_count'] = master.apply(create_driver_gene_count, maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_1')\n",
    "    master['driver_mutation_count'] = master.apply(create_driver_mut_count,maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_1')\n",
    "    master['snv_driver_mutation_count'] = master.apply(create_snv_driver_mut_count, maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_1')\n",
    "    master['indel_driver_mutation_count'] = master.apply(create_indel_driver_mut_count, maf_cohort=maf_cohort, axis=1)\n",
    "    master['max_vaf'] = master.apply(get_max_vaf,maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_2')\n",
    "    # Tp53 residual\n",
    "    master = create_copies_tp53_muts(master)\n",
    "    master = create_tp53_res(master)\n",
    "    print('checkpoint_3')\n",
    "    # Mutation Type Grouping\n",
    "    master['tp53_vc_group_1'] = master.apply(vc_group_cond_1, axis = 1)\n",
    "    master['tp53_vc_group_2'] = master.apply(vc_group_cond_2, axis = 1)\n",
    "    master['tp53_vc_group_3'] = master.apply(vc_group_cond_3, axis = 1)\n",
    "    master['tp53_vc_group_4'] = master.apply(vc_group_cond_4, axis = 1)\n",
    "    master['tp53_vc_group_5'] = master.apply(vc_group_cond_5, axis = 1)\n",
    "    print('checkpoint_4')\n",
    "    # Variant Type Grouping\n",
    "    master['tp53_vt_group_1'] = master.apply(vt_group_cond_1, axis = 1)\n",
    "    master['tp53_vt_group_2'] = master.apply(vt_group_cond_2, axis = 1)\n",
    "    master['tp53_vt_group_3'] = master.apply(vt_group_cond_3, axis = 1)\n",
    "    master['tp53_vt_group_4'] = master.apply(vt_group_cond_4, axis = 1)\n",
    "    master['tp53_vt_group_5'] = master.apply(vt_group_cond_5, axis = 1)\n",
    "    print('checkpoint_5')\n",
    "    \n",
    "    if cohort_type == 'wgd':\n",
    "        # These groups are BEFORE WGD !!!\n",
    "        master['tp53_bi_state'] = master.apply(get_bi_nobi, maf_cohort=maf_cohort, axis=1)\n",
    "        master['tp53_state_wgd'] = master.apply(get_mono, maf_cohort=maf_cohort,axis=1)\n",
    "        # This is after WGD\n",
    "        master['tp53_loh_status'] = master.apply(get_loh_wgd, axis=1)\n",
    "        master['tp53_mut_loh_group'] = master.apply(get_loh_mut_status, axis=1)\n",
    "        \n",
    "        # Pre WGD Allelic groups\n",
    "        master['prewgd_tp53_group_1'] = master.apply(get_prewgd_tp53_group_1, axis=1)\n",
    "        master['prewgd_tp53_group_2'] = master.apply(get_prewgd_tp53_group_2, axis=1)\n",
    "        \n",
    "    if cohort_type == 'no_wgd':\n",
    "        master['tp53_first_group'] = master.apply(cn_group_cond, axis = 1)\n",
    "        master['tp53_group'] = master.apply(mut_cn_group_cond, axis = 1)\n",
    "        master['tp53_res_group'] = master.apply(tp53_residual_group, axis = 1)\n",
    "        master['tp53_loh_status'] = master.apply(get_loh_nowgd, axis=1)\n",
    "        print('checkpoint_6')\n",
    "        \n",
    "    # Finally, we merge Genome Instability metrics\n",
    "    gi_path = total_output + 'gi_metrics_{}.pkl'.format(cohort_type)\n",
    "    chr_metrics =  pd.read_pickle(gi_path)\n",
    "    print('checkpoint_7')\n",
    "    master = pd.merge(master, chr_metrics, on=['Sample_Id'])\n",
    "    print('checkpoint_8')\n",
    "    master.to_pickle(total_output + 'master_{}.pkl'.format(cohort_type))\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "checkpoint_1\n",
      "checkpoint_1\n",
      "checkpoint_1\n",
      "checkpoint_1\n",
      "checkpoint_1\n",
      "checkpoint_1\n",
      "checkpoint_2\n",
      "checkpoint_3\n",
      "checkpoint_4\n",
      "checkpoint_5\n",
      "checkpoint_7\n",
      "checkpoint_8\n",
      "CPU times: user 6min 20s, sys: 902 ms, total: 6min 21s\n",
      "Wall time: 6min 21s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "master = merge_tables(cohort_type=cohort_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tp53_mut_loh_group</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0_loh</th>\n",
       "      <td>2139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_loh</th>\n",
       "      <td>3814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_noloh</th>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    count\n",
       "tp53_mut_loh_group       \n",
       "0_loh                2139\n",
       "1_loh                3814\n",
       "1_noloh               414"
      ]
     },
     "execution_count": 311,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master = pd.read_pickle(total_output + 'master_{}.pkl'.format(cohort_type))\n",
    "#master_tp53_loh = master[master['tp53_count']==0][master['tp53_loh_status']==True][~master['tp53_cn_state'].isin(['LOSS BEFORE'])]\n",
    "#get_groupby(master_tp53_loh, 'prewgd_tp53_group_1', 'count')\n",
    "#get_groupby(master,'tp53_vt_group_1','count')\n",
    "get_groupby(master, 'tp53_mut_loh_group', 'count')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_nowgd = pd.read_csv(data_no_wgd + 'cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GERMLINE</th>\n",
       "      <td>799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOMATIC</th>\n",
       "      <td>147625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UNKNOWN</th>\n",
       "      <td>1380</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  count\n",
       "Mutation_Status        \n",
       "GERMLINE            799\n",
       "SOMATIC          147625\n",
       "UNKNOWN            1380"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(maf_cohort_nowgd, 'Mutation_Status', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "#maf_cohort_wgd\n",
    "\n",
    "def removing_germ_na_mut(maf_cohort: pd.DataFrame):\n",
    "    '''\n",
    "    This function is meant to remove non TP53 GERMLINE mutations from the mafs, to have only SOMATIC mut \n",
    "    except for TP53 muts where we allow GERMLINE status.\n",
    "    '''\n",
    "    maf_cohort_final = maf_cohort[maf_cohort['Mutation_Status'] != 'UNKNOWN']\n",
    "\n",
    "    key_list = list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] != 'TP53'][maf_cohort_final['Mutation_Status'] != 'GERMLINE'].sample_mut_key)\n",
    "    key_list = key_list + list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] == 'TP53'].sample_mut_key)\n",
    "    maf_cohort_final = maf_cohort_final[maf_cohort_final.sample_mut_key.isin(key_list)]\n",
    "    \n",
    "    return maf_cohort_final\n",
    "\n",
    "new = removing_germ_na_mut(maf_cohort_nowgd)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GERMLINE</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOMATIC</th>\n",
       "      <td>147625</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  count\n",
       "Mutation_Status        \n",
       "GERMLINE              6\n",
       "SOMATIC          147625"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(new, 'Mutation_Status', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "336px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
