{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <span style='font-family:\"Times New Roman\"'> <span styel=''>**WHOLE GENOME DOUBLING ANALYSIS**\n",
    "\n",
    "*Emile Cohen*\n",
    "    \n",
    " *May 2020*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Goal:** In this notebook, we analyse all MSK-Imoact Cohort to know the proportion of WGD Samples, the tp53 mutated samples. Then we output a diagram with important information.\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Setup environment... done!\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "<span style=\"color:green\">âœ… Working on **mskimpact_env** conda environment.</span>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%run -i '../../../utils/setup_environment.ipy'\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "from scipy.stats import fisher_exact, ranksums, chi2, norm\n",
    "from statsmodels.sandbox.stats.multicomp import multipletests\n",
    "import matplotlib.gridspec as gridspec\n",
    "import pickle\n",
    "\n",
    "data_path = '../../../data/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# MSK-Impact Cohort"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Loading the data\n",
    "maf_cohort = pd.read_csv(data_path + 'impact-facets-tp53/new_data_facets/msk_impact_facets_annotated.ccf.maf', sep='\\t')\n",
    "annotated_cohort = pd.read_csv(data_path + 'impact-facets-tp53/new_data_facets/msk_impact_facets_annotated.cohort.txt', sep='\\t')\n",
    "arm_level = pd.read_csv(data_path + 'impact-facets-tp53/new_data_facets/msk_impact_facets_annotated.arm_level.txt', sep='\\t')\n",
    "gene_level = pd.read_csv(data_path + 'impact-facets-tp53/new_data_facets/msk_impact_facets_annotated.gene_level_grepTP53.txt', sep='\\t')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "The following cell shows the size of the total cohort through the different input files. We see that the cohort is more than 40 000 samples but only 38 465 have mutational information."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAiAAAAEFCAQAAABwCZSPAAAAAmJLR0QA/4ePzL8AAAAJcEhZcwAAAJYAAACWAHFG/vAAAAAHdElNRQfkBQgNBiQu1ISvAAAKeHpUWHRSYXcgcHJvZmlsZSB0eXBlIGljYwAAWIWdl22SZKkNRf+zCi+BTwktBwSK8P434ENWd0+33TNjOyuIzHoPhJCu7hXpn+7pH3zqUEn5fVbJPz7167ccudq1jtq115rHHDZWzX/2SVKkadPcy8gjd//TmX/xCXZ9Hv1w57R6/h9DH4/+x/lLugxt0r758u0E6omDZa3aP/8XnV8v6lQlQvn78/XNTulSZf/xfPfvzxPh/ITx63+fPxboz8+P/2Ho5+eRfzbUycyXqzV/7TCzY+j3z/9kfvr1zN8/tfbyDiwmwvtJ+puECMdV4Y2MmrV9h0a33lJvTCJKbxrLXMbvo/x3ptN/2v5vTf/6+dv06zv6/JYlPh0/yJqxkYkKb9j+efTXcWi15VYakP1diUQbD8zlu0eliPvf1dL3z+/mSaz6OVqb8RWHZr+fWM3e99b5mVfmWf8+72Oo9m/IjfmJxRYPED/Ikvxi8Uek8jP4FsUDI8MwVC6m2isLBkVL0jJ1k9v+WtlZ9HbqLBo8GHg3WPOwJ/MRDil5R1N9RQc8CdrEg4mBdxLDgGHAMLAwsHi4MLDrOySDNc4aZ41vDD3mOCw6GGBevvy+++M1TMPY5OX9KeOQmsYwRuRSB4P3DY9Km4zLUXkIsRWyXnC/YKMIi4V3yju8LhMjeFyMOXhboNaCp2UXDG1+4GJxvg/fh+/L9+U7WBCL4mwMh4Y741AvwghCO8lUYXA0qpnBS3avykNlIdmr8+ZqTCTHdWFks5gNq29yMnJ9OSIEFei0l/6WN+AVklXyo9rGLtQbI3KDd5rwTvFJL4Djf+N/jDcC3zb/u+Z2Goaw3K7nFka2hcJpmfphHApr594nCEAXSHfH447BPp36XqCCd3javafcDxOIyYNJjwvUTh7F8yAboy2gA9zHzIOjD6AygMjAq7EYG+lxxhkJbPGDNH/+OKJUzY/IBU+E7ImsLLrBnmexk2VFFn84LFluo9DgnKwpK5hQdtd24IzIVD4Y7VnZWakxJdC6eX4gLjbVmFDrBr+RJ1Uwu+Q5VgLMN084ZOLuXAtg8z+L5tU8AaMBXgN4xjGNjUx6NrVsk98g3gi4eaRs7GIsWKXkxbEWni0gsTjSomwWEFhkaBGLhZqseHnmD0Ld0MWGk7ZQtJu620ze+5UP3wR+k0EvQLCu7EDBh2cH3Q62fGn2V2YA1zF63l9Fsk9/pbbyIS6HiQfIH2fC4TfxuMDhgr5L9i7Huhr52qYcJV9CcO+lLPEoOH8A84AaAlQHsYrdUOPIcV95E6VKBjqMK5xfcdk2bvP86FtYKOTE4LsHfHtKmV7KIlpupdzJ4bRQV6X2Uar0QumUulqpzriQ+SP0ykDXCuIIATAWmPYBEQxKU0qn8Ho3RHqVPnfp60AOlz0hh1LLaHRCQwqyAVnsVMY+hVO9ait0CEVYLOJFZhTZFUd5Fqso1KC9FJVBr2FF1y1gq2homQVDFHqZvJxzlbkCYuc3Cz+Uw5FMdjFOahvonkNj0suqqyxCs1Sho1uARiqLgOJ42W2XzTE3Bjee7LPKYyAgUHzwrbs48XH34gT4QFqHKj76KMwSHUsrB2O3SLl4d4nJtV4ugLrXSpCNaLeE8JvnsaPEXfVDpcSewqvAPIE6SAOyI1UQ4OTQbL+Ipt/Kqlqr1jpGrZOfK2o9B81ZFd6qcFVt1mvzmmqLx5ZRez90Eo7G7drPetVVB5OHMJD64YxAyetTc8bU17xVuZP84pF2q6pUGQb0OOp26mxB8wdsFo6cXu2JLUYJPKJ7KmxC8eAgbcxio0X6oeOARGrdTaBlq5uJIKI+avNm1eVWx6AfhTO9HuJyVOph43PBJaC53VPFMzhcKzVTOSBcvmpYqcFRImCuNmAvim9RvWdTB0C5kz5CVDbfURu+pValtWob3u+Nma1Bzk2jtT1bI2UdX+mRWrfb+pl0Mq0N+HlM+jOvbcShODQ1UYK/bpNriEVv+kTDvOnRNktvNCBtTm/T52tWPkkyNrLNwQO6w8zSnhpHRVmiceK2BViu1fadZFQbbV9zjuS3tVNro1oaOG0wTLso0mXTiyLBJIn8lBZMoFlqcSvK2KjZ/ijykQ+hBYVCRS8HpRd/UCpcr3sQUCUe7KSHrhaJ6shhpx3tc3Uq/JEGUkZDDSmPc+nSa389oazdJZA2oqS6gR0Sh2BNJLtTyH1Cj0blmBDTZZ1OhrxoX3o6jvQN/Dfx3hjeeE39dZLafa8OpDqzUj9GMo73SxNw5Xag8KWVtMrEssd5Qg9hKxex/ageqkAKoYNBYQ5AMCqXGlCnA1ob5BFhXYOAjd6xSmPZz6bK5hjKQZ1qgVcFaZVlgy55EIyhVBIqnsYEglPPmL6HwTImBuEheVnHYtlajBhjE7VtjIvNxoDE/Mg4eHt0pnHcBtQ0rvi4+wwoHwUvAwGg1cIJLqwIG844/MubBY3iWCWi1bjkoOCPswV0SUNb+ku6denXQA9bGUV+VYTflKBQ5YKsixoYZg6FLaizzOvyLjVitsTiIWVy9KBHUNnsvBffEfip4otrK+J+6DHONqFW5cqW66CBiAdHk4DTaccQevqWS24AfLGh9AgkmGpeOEIH2YgE9QdC+9fd0skSZEPnrsQmvXOpwOwSXD9pgnQ3BAah4Lo+mWx1qU3ahgtrcbEksTQ5XeF33dQRvKo+MeRPVbjfUEP6+tcLBV4mwA50MF3j0mV1LrtrvpZiolGz+IFEMkwHAUeHEjRNqhT9PBOsz34pdhaNtemOXnQrgeGW9c5kMbE4pxhkcKdB2mb4GndSlmkuXxOpn8Rw7vDpAmPw7EBdhzUnYt5Pcu6MhmwafTO9G+0a3QbSQvNZ1kyGfEDay9DyVywGl0A59FSToqNOxggbbp8yJL1GB2UE04iDze42N47VnvAum4UDgmnrAGq4fq8wZNCcOR5qB4ShQobu2V0XtBwOui2CFk9ob89MdAiKtAr0zjBZEDSFz0ApO1VFmVOAc43FXrQqBGCBGVB2F16tiZBM2uMFwTLFaGZ8LUQfRVmbMtvXkHRfTid4Or0IWn7RjovsP/zi0X53O0qSrmulTRuyy0GwOorvMH0j9utyQurUqOTS9piL/gy/1TbEBujmxhtKm/I+3Gbgo20shqX32gNLlx8PZ2W77dfw7ENrywmgcTgtUH6UNIKmklYyXzoKURqHlmCZQPWQBIikHS4DtP3QrY++ORlo6Fz9nRtHfw0J+GjH53ZHP9jLaFCmE4vksIVvbrFYcg7iKJbDZwiH+H2326YeHIDbzMmbtq05h6ENbXG4LR3Y/iA3iTgafkBE/Z5xiNYYRw4sjj3icKYgixdsCg0xeSddZ8Um9jS/3EJ8LtqvnA4zkHA/tDwnaA9icbNBLvPmcee64/Q3Axk7GyfbhbsuMnJ7OFUIzedzxSRd+OICACSRNmA7PRbYPyQUUl0X0oRcNvGGWi997z3mdAnzktcbKF84ffSYie57RKFfKBH0MoSkWEBJ0REQdAe2hnvPDZET8pJGozmZMwEdrQ4loAGzpFi08ls1yCeFMomgxaFGbt9xj8ORlG1E+hftkQTIS62KtQAAGGZJREFUeNrt3e11ozrbhuEzs3YDmhJIBy8pAZdASsAlQAmmBFNCKMGUYN4OQgmjEnh+IBPwJ8Z28Md1ZK29MzaWyWS4LIS49VYjIjLOf1w1QSqqqX8iETnIw7tmc8V/l73eUlBSYSmm/FsRkTMYfDwMAcGFLb3VI3sgOSUFZftnHwOY6+abiFxVhaV/phDgE+KPa64YESAlORm2ffPg2t0iEbm5kqrTCfCICM8/js8NkIKUAjCEBIRT/x2IyIUsOQU5ABHxeSFyToBswsMnIsRM/XOLyNVUZORUnBkiQwOkIiEHAuKLh11E5D5lpFRATDysizAsQFJSLB5LhYfIkzvraC/+nNrC8kkCxHwrPkSeXsw3ERUzkgFbn+iBlHxS4bMce5lHRB5QzhxLwNfxU5njPZCMGRUxK8WHyEsJWRNQ8NGZ67XPkQDJmGNZstAVF5GX47FypzLHIuRggGTMMSyJpv45RGQiS2Ls0Qg5ECBNfKwUHyIvbcHyaITsDZCCOfClkQ+Rlxe5Xojd++yeAKn4BM35EBEAFkRYZnuf2xMgn1gWOnkREWdJSLl3XshOgCSUBMRT77GI3JElhtTdcte1FSAFKYavqfdW7lDBG29XKBz1xtugOY5yT5pUmO+MhGwFSDN4qnkfz+LUQX+tULiPn0ZuKSDG7kR/L0ASKkINnorIHjEe2VaAd2qiWjIMi6n38oVlrtDcplJlAkSU5IDf1ova/2jz+hwwBK5ey6ZMTE7lBsX777D9fE6OxcPrjIFZEoqB1TMzCiw+XmcIfnufNpr3Cog6j+5u+/OzBng7P438LsOCT9L+v4S6FdfUcS3T+Ff7Ne2XX9d1XVNTR+1jntty/6P913v1d938RpuvYO879J+POs8G9T/Xqtc+Zmpq6tWg/Y8O7lPtvu//pMe2jd1/+3sr0wj6/wZWbYD8q01t3D8b+X1xTW3qRb1yB/JXXbvDJa6/3MH1VdcHH20Or7D+ct9FdV1/14uamnpRr/e+Q/f5Zdvq5oCt602o/LR6OEA277qsg3a7ffv0s/8rt/+rg/u/2dbUQb3c+mlkGqt+gP8EiPof01rWsTuUVu0B/HMgrToH9f5HTefTvDnsv9stVgff4ef5sPP6sKY27XsFvVb3B8i/zpb/6qAO6uXBffrZ59P73/RRNh9rq/pYhMnv6PVBVu0YSA46s5xQREFBCluDVM1YwPZNBduPFlhoz02bUZFqq7LloXfYtAAlb+2fbbvd5l2O3dhQdrYwrAbu08+59LFtA10VvCshBXn7u3IBUlCNKeouV5OQ0iz4E5PuPLv/EDIHnm8O5O0r9sffwbK7alm/1XP/dQzZpzHbyrQiEvK2yIcLkBy0SMOEKlIgYtl+fx5/qzXYPihPvYPBEm5dg9vX6n7990pg56rNvn06Z//lfkSk5O58xc0DyTEKkAk1h0xzGGXA+Z/3AbjlviwZzcXQc96heX1zKvLJjHm7fbfVQ3x32Ria2cxNbe9T+3TO/sv9CPk5Df4PoMRq/GNSm3kfOdYd6ucuUh5TYHnHd8sWdu9mSghc32L/OyQERORYZu3rv1yrn71WD4tIKXnHowQ3E+TYPp2z/30/P41Mw8drA+QP4JaLkimtMFgKDKtRs4EDlq6FzboeAD4+UFIeeIef5wPWeO71hoXrj4YsaZZQtycO2gUxULmpZKsj+3TO/vd1fxqZUoB1v4W3uoZPcr41hDoxS3VhjFu37KjfGz8o2sHR/e9QdAZPm8+V7ZOfauex/SoKPEzvPfbv0zn7f3hvZSoZ8+Yjo1lY6i+G76n36aXNLm/i1/zb89jfX96HhXrMk6p4J+QLiv/AYvXrmJjuMD2PLvFOy2MzhvZfMwlIATKtwSuci9yFwH3o/WmSRNfcRWQ4Q5McLkDUAxGR4TYnMScX1xYROUQBIiKj/WkGUTVtWESG82muHaoHIiJn21x2UYCIyGgKEBEZTQEiIqMpQERkNAWIiIymABGR0RQgIjLaf5c3IY8mxfQWn2yqiB1avLIiO1lCsCCnwsPrLVUpz089kJeTkLhVZsHywZwCSJntrLzeSLdquL/z1n41hZByZmRYcpKHKo0kl1MP5MWUvTjIKIlZABUzUqJeucCKimKnGnt3BaHNct/GVVRNSclUoPuFKEBeiuWzt6xUgXGnJ55bMKhbzDjbs35MBUS9k52SihgPMERuQQd5FTqFeSlzTC8ivE5vYbcmzIKaes/6Mj4pSVsb3RD3ttEYyCtRD+SFZOSsewf4svfskLuySwzvWCDF5wsPz/VhCkoyfJ3AvBT1QF5GRXKwmnnFjJx4QGU6iyVgxZqIks/OMzMSKhWGeDHqgbyMT/wDSzslZMDXwMVNQ7dq3RJLTt6+qqYiI8X2+jXy3NQDeREpJR4JCQlNb6Spql3yQUrE98D4WLj4ANw6cT88FvjtJWJ5BeqBvAgLnQuyFSkQUDHDYz2yqLYPeCSkrNpTF08LT74U9UBeRHNFpfmCgJoFzZzU1RnxUfHWmW5WsBl2rTpb6CrMK1EP5IVZMvzeXI/gxCCoh0futirJCPAISUkJMUBCqaswL0UB8sJKdte6P3UV5YtPZnh4FHgsAZ+YlL/4bvHuU/fNyDNRgLygzUKawaAlNVe9P/msybBA6HoduKFTiyE6cJ1HnpUCRM5k9oREOPAajjwbDaKKyGgKEBEZTQEiIqMpQERkNAWIiIymABGR0RQgIjKaAkRERlOAiMhoChARGU0BIiKjKUBEZDQFiIiMpgARkdEUICIymgJEREZTgIjIaAoQERlNASIioylARGS0uyiqXFJSgdY0E7kyg4fBv9mi5xMHSE5B0VnXTERuwRDitwtxXM9kAWJJ3foiEOIRcHpRIxE5T4mlpCInA+ZExHhXbH+SANmEhyEi0IoiIjfj03wwLynJycnIrhoiEwyi5ryTAjHfLBUfIr/CZ8E3Szwy3jtLpF/m1wNkzieWiG8WWsdd5JdFrIkxpMzcAMJlfjVALDMyDGuWCg+RSRgWfONT8HGF656/GCAlHxT4fOP/3puKyA7DmoiK2cUR8msBYplTEbFW30PkDixZYC+OkF8LkE9KIpa/9XYickLMEsvnRWMhvxQgcwp8Fr/zZiIySOROZMb7lQDJyTCsdPIicmeW+JQXXNT9hQCxJKD4ELlLKwzp6NtJfiFAUioiXXkRuUuGBTAf+eqbB4glwxD/7t+J/Io33q4wo/E6rch4ER4FxajX3jxAUizRVW/fEZHrWgDpqFfePEDU/xC5dyE+xagZITe+G7fAEmn49C5kbqAscGUTEiCiJCcg6vwJPHe3RIFPOHD0KifH4uG5j4ucEojd7z4BfMI928l9CCkpRoxU3jxAVOXjHnRnHKb4rNl0WVPAc/+3ZG6bAp8MKMhYDzgBnbevhIIvDNa1HAEFKbDcu53ch5CEfESo3/gUJkcBcg9SSgwLVkTg+hnN44agDYic2PUZSnIW+HRD5bCMDIj5ImYTF02Zhia0CpqKWPu2k/vg4VOOmJN60wCpqG5QRE3O57nDNugd2ODz7UIFIGLBgi/3fcx6YOsFTbWJkAUhkAHG9T02z4eYvdvJvQig/WAZ7sYBgq6/3IWIgIIZs61py8GeeN8+Dz49tFYAJW+88UYO7nMsACpKKkr3p/3byX1ofl/nuukYSPOJI9NLSAGDT3zyxOH8HqMFvJ2PihCDpcDQnMAc2k7ug8fdBQiM+eco11aRgrsXurrByIPBEu65VTIkcwESHd1O7oHHmD7hTU9hStQDuQfN50rzm2jGHa7bCwiAzJ3qfDJrp0U3oyDNCMix7eQ+eCNmgqgH8gI28z5yrAuT667EE5FjmeFTUYEbhgUfjwqL58Lr0HZyH7wR09m1tOVLWLXjESvCq19YD1jjYSmoMO4KS6P5Ljq5nTyqu1jaUm4t4JvK9QM2n/t1b4t67/f10VZ/nvX5dp9e/XBabI147Nvu+HvIfVOAPL2qPWEZd78lwP/veez/9m459D22tzMaLXtICpCnlz3EjM+A1dS7ICMoQJ7eY9xKoNkhj0kB8vSCB4kQeUS6CiMioylARGQ0BYiIjKYxEGml7ib8RkaBxT84hlKRDbiv5VQr8tjUAxEnIWnrQVg+mFMAKbMDNdPTrcvD7+5G/TfeXNGAIa3IY1MPRAAoe3GQURKzACpmpFt19Ssqip1yQBVhu5U3oBV5BuqBCGD57NXDLNxyQ5uqpv1KVRmznclpFZuKZgsW7kToeCvyDBQgAsy3Ft/wOmMhu1PMF9TUW2MaFeCTknRuCT/eijwDncIIGTnrXuGFZe/ZIbNZSwzvWJqq7194o1qRR6MeyMurSFwF9n3PzciJB/QfLJaAFWsiSj5HtiKPRj2Ql/eJf2A9kIQM+BpYtyN0hQKWWHLy9lXntSKPRT2QF5dS4pGQkND0Rpob7Us+SIn4HnjgLzr1xXw2tdzPbUUejXogL87SXZ+lKbkcUDHDYz3ypMNnU+P7klbkEagH8uKaKyrNFwTUbqV2w+qMA7/irTNRbLOg6bmtyONRD0R2WDL83kyPUxPRPTxyt1VJRoA3ohV5PAoQ2VEC5VaJ/1OH/hefzPDwKPBYjmxFHo0CRFqb8sbBoELH/RKEPmsyLBC69ZCHtSKPTQEiV2IOXAyWZ6ZBVBEZTQEiIqMpQERkNAWIiIymABGR0RQgIjKaAkRERlOAiMhoChARGU0BIiKjKUBEZDQFiIiMpgARkdEUICIymgJEREZTgIjIaAoQERlNASIioylARGQ0BYiIjKYAEZHRFCAiMpoCRERGU4CIyGgKEBEZ7Y5WpivJt1ZSFZHrMfiEeFdt804CJCehmnonRJ5cTkLA8oohcgcBYvmkAAJCgivno4hslBRkFLwTs7hSm5OPgZR8UOCzYkWk+BC5GZ+Yb5YYUuZXanPiALHMqYhYE0y7IyIvImKNT3alCJk4QOaURCyn3QmRl+KxwiMjvUJbkwZIQY5/tbMxERnG8AWk2ItbmjRAUmCBmXIXRF6ST4Qlu7idCQOkosDX2IfIJGIgv7iVCQOkAMLp3l7kpXn4lBfPvpq0BwL+dG8v8uICeOQAKVGAiEzHg4tvHpn4Mq4GUEWm4sHF12Emn4kqIo9LASIioylARGQ0BYiIjKYAEZHRFCAiMpoCRERGU4CIyGgKEBEZTQEiW1L+8sbs4nbeeCOZ+oeRG1OASE9JcoUyM5cpeOONYuq/ChngDqqyyz1pwuNLdVpkEAXIE0uAiJIc8IgxpBT4hJ27oDN3Q3dAAGTu7swCS3Sw3YwCi4/X2SYjBwwBYe8WyZwcS0DUeXR32589DfBcmZuc6sg+yJ2o66Cmvo3jLd/ufaVBTR3VuC+//d7U33Vd1/W/2m+fpfbr5nfSfAUH2uy/JtrzmOdab77vtn5829j9Nz65D3Idq5o67vz5nCPSvXalMZAnlxMTY2iWDl3gQ1sLM6XEsGBF1D7flLheHCx1nVICEUvXYynax0K+iIGqU+27ImaFT7OoEUe3TTEEeEQn90HuiHogz4v2E2bV+f7nu2Ud16ut55vvVgda/NfpGfyrgzqol3Vdm04Po+nlfB9898Pb+vU/9/jxfZBruUYPRGMgL2G78lsz0hFRUJDC4CseZac1wwpoxktoB12btQWr3hqDPwOyx7YNVF7qAekU5iXsPzQTZu6UIr5a20242CtsK49APZCX1Yw/NOsCVgNXKesHQMLm6k231d3tfvhnbCuPQD2Ql9Ucvs0h3Qyqnl7a3MewOeEpSEmpwA2nWjbDs+bILJJztpX7pwB5Wc2BmzDjwwXIkBL/zfWad2Z8gpsJEgOWd2Z8nDwhGr5toonwD0AB8sJWGCwFhhXhwJ7Awl1+baaSNcOoAUvXUgXERwNkyLa+u/B76ZIDcnsaA3li9d7vf74L+KZyJzFf7WP1iVYXRBR4mM6IRkRIARh3knP43U9vC4Y1Bd6AUyqZmgLkpZkjS3vtu7Tb9FK8PVPMzRnLlA7ZViMjj0EBIgccDhCRDQWIHKCJ5HKaBlFFZDQFiIiMpgARkdEUICIymgZRxSnIqfDwtqqHNVPG/COXXlNM58JuRQZ4BJrH8QLUAxEAcmZkWHKStiK75YM5JZDyeXBieULiihACZLyTUTLnvX30nbf26/Jq73JP1AMRABIMazwsKSkZEU31sM3U9Tlpr5bqRtm7j7dk3t7fOyNxvZaKsO2NqFfyXBQgApRUxHiAIXL32NK52c0jpKDYCRDLJ3EnQjI8N3/EY0GJxVABkaagPSmdwghgiHuHuGn/u7lD18Ke/scc07sdLicEMhIKQhZtCz4piW6Oe0LqgQi0/YaCkgzfDYnGFHwQ0dTu2O1FZOSsOwOuJRaYuVETjy93V63hHQuk+HzpJOapqAciHTMSqjYofKJ2TISdqzAViavyvtGEhOGLNTEVc/eoJWDFmoiSz6l/RLkq9UCko6YiI8WyBBJSIhYYChJmrHp9kE/8PdU8PFcjxMeSURDQLOIAsMSSu9MceQ7qgUiPxwKfnKZOqs8SAwSs2BQ+bKSUeCSublhF4u7eDTstNcOwi7bWCO6URp6HeiBC09dYdZZbKKGtdtow+L366ZZuoGwKNHf57F603feYPDL1QMSpOt81vY7+kgtl79Bf/KxPSVPHbIGH6fQvCsCn4q0zBa1ANUWeiwJEaE48UhcXCaU7EQnI2l7GHI5UL9uI3EJVUJIR4OHhkbvTm81j8jwUIAL4xFT85YN3Uvx2dVqPOX+Z8U5GtKeQ4bYFEQl/mfGBcWMfX8DM1WE3LKf+UeWqNAYiAG7o1GKI2msrPmu3fLZPfCQ+uiWRl65kckjoZog0rdjeY/IsFCDihHsur5oTizTsE+yMcpiLls6Ue6ZTGBEZTQEiIqMpQERktD/Nxbni4oZE5HU0Q+vqgYjISAYFiIhc4E+TI7rBSUSGq2h7ID79ex5+01TvKyKbEBhjU6Huz3S3WOvWbpEpNcUmx9nUx/3Tr3x5fYf7GB66+iMynYJ+cYVqcH/kp+/yByCgukmEHO9jBChARKZSUeJvBcjQ/sjmIq4LkFvOBDnWAwkoO0sSicjvSenXuT1nRKRkU9fF9UBuMxpxqt0YSDSQKvLrSrLegqRNgAyt1fJTGKoNkFv0QE6NrgSEbe1uEfktlhm4ercb2yMih1Wdyv1uIllEdYOTCR+P/GgPY4lPzly9EJFfUzLDEm2VbxhebrK7yIcLkFv1QcIT7RpW+GTMNJwq8isyZpRuBeMfu0Oqh+XsBEiIOdFXGOf08KxhRUjJjFm7JquIXF9BwjtzLIud0pJNFblhrVTdynK1E9fUcX19pjb1v5NbreqgRl/60tfNv6L6e88RGNTU60FHdFBTr9oD9612BS0r3jH8u3rqJaTErkjvcRW5W11VRK7Pwz9QlbZg5hYPO6XkA4/v9oVtgMCcjOWAytvnsbwD3yqmK3K33qm2Fi49ZCslis7t/LeZlWGIsJ2lhUTkvmRUe0ph71OQ4fU6GZ0A8Yixblmga4oxZLptTuQuNcf8kEEGSHa27BUUijGkVz/Um8WEPjW6IXKHPqmIB90Fk1ISbF2r6QVIc6hff1pXSETFbNq/JxHZMafAH7RuT0m6Z2XBrZKGISHlDUYslviUmrQuclcyMgyrAZc4LHMs8c5Us52aqEu8zpLK17PCI1OEiNyNjPnA+Ngsub7bU9kJkGZJ5OQGIyFfeGR8aCxE5A7MmWNYDBr9yMjw9i6Mvqcqu88Sy+zqEeKzxqfkQ1dkRCZlmZHhsRo076vpqXzt7ansXdYhYnGTCDGsiaj4UBUQkcmkvFO4D/TTmoGH1YFtD6wLExPdJEJgyQJDyrtCROTXZe7Ii1kPGvto4mN5MGo6U9m3zckwfA2sEXAOS0qGxRANnAEnIpcpycmpgGjP1ZT9UhI4doNLcSRANumzGHSV+FwVqbvWYwjx8fAGF1QTkWEq95W7UhnhwEljYEncRd4j2x8PEMhIsIRbxc+uxVJQ3KQOiYj0hQQEgz+kS+aUeHwdj5tTAbJpyLAcWG5kjIISS6UgEbk6H84eKEhczfaTHYfTAdKMWKRAwFInGSJPrmBOhSEeMnRR/Dm9jWHBCp+Cd+YqOijytApmzKgIWQ8c+RzQA9lISbFARKgrJyJPJiN3Szsshx/fQ05h+m+SugVoQiKd0Ig8gZLMXcoIiM/rHJwbIAAZhVtDxiPAxx+9wreITKegpKTA0kymGHFmMSZAACwZRWfBBh+jGBF5CNbNDWkYAoKxtZDHBkj7egoqrJaFEnkoxk3eDC774L80QHptTf13IiIDXPFsoXirhywGISKyq3y7WgdERF7O/wAqc9OtKAfoiQAAACV0RVh0ZGF0ZTpjcmVhdGUAMjAyMC0wNS0wOFQxMzowNjozNSswMDowMLBpobwAAAAldEVYdGRhdGU6bW9kaWZ5ADIwMjAtMDUtMDhUMTM6MDY6MzUrMDA6MDDBNBkAAAAALXRFWHRpY2M6Y29weXJpZ2h0AENvcHlyaWdodCBBcnRpZmV4IFNvZnR3YXJlIDIwMTEIusW0AAAAMXRFWHRpY2M6ZGVzY3JpcHRpb24AQXJ0aWZleCBTb2Z0d2FyZSBzUkdCIElDQyBQcm9maWxlEwwBhgAAACZ0RVh0cGRmOkhpUmVzQm91bmRpbmdCb3gAMjYxLjEyeDEyNS4wNCswKzCuO5DaAAAAE3RFWHRwZGY6VmVyc2lvbgBQREYtMS40JDFqVwAAAEp0RVh0c2lnbmF0dXJlADU1MzljYzNjNDNmNzMwNmM2MjA1NTk2MDc2NmYyZThiNjJhMWY5OWVmYWU0ZTZmMDk3ZjMyMTBlNWFhYTU4NWLf+RhMAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<wand.image.Image: 5539cc3 'PDF' (544x261)>"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "img = WImage(filename='total_cohort.pdf', resolution=150)\n",
    "img"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WGD column\n",
    "The annotated_cohort have a column called *wgd*. On the other side, for the TP53 mutated cohort, we computed the WGD states by hand. Let's see if the WGD column is accurate."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [],
   "source": [
    "#Loading the master_file\n",
    "master = pd.read_pickle(data_path + 'merged_data/master_file.pkl')\n",
    "#master = master[['Tumor_Id', 'wgd']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotated_cohort_wgd = annotated_cohort[['tumor_sample', 'wgd']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>wgd_x</th>\n",
       "      <th>tumor_sample</th>\n",
       "      <th>wgd_y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0034223-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0034223-T01-IM6</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0009819-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0009819-T01-IM5</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>1</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0027408-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0027408-T01-IM6</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0006554-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0006554-T01-IM5</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Tumor_Id  wgd_x       tumor_sample  wgd_y\n",
       "0  P-0034223-T01-IM6     -1  P-0034223-T01-IM6  False\n",
       "1  P-0009819-T01-IM5     -1  P-0009819-T01-IM5  False\n",
       "2  P-0025956-T01-IM6      1  P-0025956-T01-IM6   True\n",
       "3  P-0027408-T01-IM6     -1  P-0027408-T01-IM6  False\n",
       "4  P-0006554-T01-IM5     -1  P-0006554-T01-IM5  False"
      ]
     },
     "execution_count": 58,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# left merge \n",
    "merged = pd.merge(left=master, right=annotated_cohort_wgd, how='left', left_on='Tumor_Id', right_on='tumor_sample')\n",
    "merged.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34491"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(set(annotated_cohort[annotated_cohort['purity'] >= 0.2].tumor_sample))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now the idea is to compare *wgd_x* and *wgd_y* to see if there are inconsistencies."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We define a function to create a new column\n",
    "def check_inconsistencies(x):\n",
    "    if (x.wgd_x == 1 and x.wgd_y == True) or (x.wgd_x == -1 and x.wgd_y == False):\n",
    "        return 0\n",
    "    elif: return 1\n",
    "    \n",
    "merged['inconsistencies'] = merged.apply(check_inconsistencies, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 61,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "121"
      ]
     },
     "execution_count": 61,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# now we sum the inconsistencies column to see if we have any inconsistency. We should have 0 for no inconsitencies at all.\n",
    "merged.inconsistencies.sum()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So we have 121 samples with problems: let's take a look."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>wgd_x</th>\n",
       "      <th>tumor_sample</th>\n",
       "      <th>wgd_y</th>\n",
       "      <th>inconsistencies</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2596</th>\n",
       "      <td>P-0011205-T03-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0011205-T03-IM6</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3906</th>\n",
       "      <td>P-0035720-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0035720-T01-IM6</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4201</th>\n",
       "      <td>P-0020803-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0020803-T02-IM6</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6153</th>\n",
       "      <td>P-0022193-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0022193-T02-IM6</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11524</th>\n",
       "      <td>P-0016952-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0016952-T02-IM6</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20395</th>\n",
       "      <td>P-0005192-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0005192-T01-IM5</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21326</th>\n",
       "      <td>P-0018318-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0018318-T01-IM6</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23992</th>\n",
       "      <td>P-0031102-T06-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0031102-T06-IM6</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26782</th>\n",
       "      <td>P-0046593-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0046593-T02-IM6</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27950</th>\n",
       "      <td>P-0031010-T03-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0031010-T03-IM6</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Tumor_Id  wgd_x       tumor_sample wgd_y  inconsistencies\n",
       "2596   P-0011205-T03-IM6     -1  P-0011205-T03-IM6  True                1\n",
       "3906   P-0035720-T01-IM6     -1  P-0035720-T01-IM6  True                1\n",
       "4201   P-0020803-T02-IM6     -1  P-0020803-T02-IM6  True                1\n",
       "6153   P-0022193-T02-IM6     -1  P-0022193-T02-IM6  True                1\n",
       "11524  P-0016952-T02-IM6     -1  P-0016952-T02-IM6  True                1\n",
       "20395  P-0005192-T01-IM5     -1  P-0005192-T01-IM5  True                1\n",
       "21326  P-0018318-T01-IM6     -1  P-0018318-T01-IM6  True                1\n",
       "23992  P-0031102-T06-IM6     -1  P-0031102-T06-IM6  True                1\n",
       "26782  P-0046593-T02-IM6     -1  P-0046593-T02-IM6  True                1\n",
       "27950  P-0031010-T03-IM6     -1  P-0031010-T03-IM6  True                1"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "merged[merged['inconsistencies'] == 1][~merged.wgd_y.isna()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We see that those inconsistencies come from the fact that some tumor samples have different Normal matched with one of them being with a WGD. They annotate the sample as a WGD sample.\n",
    "\n",
    "So let's filter annotated_cohort - keep the highest purity matched normal - and see if the inconsistencies are still present."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def normal_samp_duplicates_filter(df, sample_name, purity_name):\n",
    "    '''\n",
    "    This function aims to filter out the samples that have the same tumor id but different normal samples.\n",
    "    We choose to keep only the one with the highest purity in case of duplicates.\n",
    "    \n",
    "    Arguments:\n",
    "        - df: the dataframe we want to clean\n",
    "        - sample_name: the name of the column containing the sample. Ex: P-0006554-T01-IM5_P-0006554-N01-IM5\n",
    "        - purity_name: the name of the column containing the purity\n",
    "    '''\n",
    "    \n",
    "    # We first restrain to the interesting columns\n",
    "    sub_df = df[[sample_name, purity_name]]\n",
    "    #Create the column with the tumor_id and set it as index to identify the duplicates\n",
    "    sub_df['Tumor_Id'] = sub_df[sample_name].str[:17]\n",
    "    sub_df = sub_df.set_index('Tumor_Id')\n",
    "    # We use the duplicated() method to create a False/True Series if the sample is duplicated\n",
    "    duplicates_series = sub_df[sample_name].str[:17].duplicated()\n",
    "    # We then select only samples that are duplicated to obtain the list of duplicated samples\n",
    "    duplicated_samples = list(duplicates_series[duplicates_series == True].index)\n",
    "\n",
    "    # We put the sample_name column as Index\n",
    "    sub_df = sub_df.set_index(sample_name)\n",
    "\n",
    "    # We can now filter out the samples that are listed in duplicated_samples\n",
    "    to_be_filtered = []\n",
    "    for sample in duplicated_samples:\n",
    "        to_be_filtered.append(sub_df[sub_df.index.str[:17] == sample][['purity']].idxmin()['purity'])\n",
    "    \n",
    "    #print(to_be_filtered)\n",
    "    # Now we filter out the lines that are in the list\n",
    "    df = df[~df[sample_name].isin(to_be_filtered)]\n",
    "    \n",
    "    return df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotated_cohort_filt = normal_samp_duplicates_filter(annotated_cohort, 'sample_id', 'purity')\n",
    "annotated_cohort_filt = normal_samp_duplicates_filter(annotated_cohort_filt, 'sample_id', 'purity')\n",
    "# the last duplicated samples are samples with purity = NaN so we simply use deplicated method\n",
    "annotated_cohort_filt = annotated_cohort_filt.drop_duplicates(subset='tumor_sample')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>wgd_x</th>\n",
       "      <th>tumor_sample</th>\n",
       "      <th>wgd_y</th>\n",
       "      <th>inconsistencies</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>P-0031435-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1282</th>\n",
       "      <td>P-0010672-T02-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1855</th>\n",
       "      <td>P-0027242-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2906</th>\n",
       "      <td>P-0013827-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2957</th>\n",
       "      <td>P-0010083-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3263</th>\n",
       "      <td>P-0020798-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3271</th>\n",
       "      <td>P-0017938-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3284</th>\n",
       "      <td>P-0031170-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3898</th>\n",
       "      <td>P-0035720-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0035720-T01-IM6</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3980</th>\n",
       "      <td>P-0036528-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5936</th>\n",
       "      <td>P-0017305-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6401</th>\n",
       "      <td>P-0017240-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6473</th>\n",
       "      <td>P-0016745-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6480</th>\n",
       "      <td>P-0006119-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6533</th>\n",
       "      <td>P-0025776-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6540</th>\n",
       "      <td>P-0019560-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7324</th>\n",
       "      <td>P-0011588-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9995</th>\n",
       "      <td>P-0000328-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10549</th>\n",
       "      <td>P-0019219-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11679</th>\n",
       "      <td>P-0000416-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12234</th>\n",
       "      <td>P-0027992-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12546</th>\n",
       "      <td>P-0023638-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12765</th>\n",
       "      <td>P-0014014-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13272</th>\n",
       "      <td>P-0028566-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13757</th>\n",
       "      <td>P-0015914-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15355</th>\n",
       "      <td>P-0029600-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15765</th>\n",
       "      <td>P-0022650-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16231</th>\n",
       "      <td>P-0004094-T03-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16448</th>\n",
       "      <td>P-0026525-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17460</th>\n",
       "      <td>P-0000244-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17839</th>\n",
       "      <td>P-0014773-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17950</th>\n",
       "      <td>P-0024980-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17969</th>\n",
       "      <td>P-0000657-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19907</th>\n",
       "      <td>P-0000801-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19961</th>\n",
       "      <td>P-0001378-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19965</th>\n",
       "      <td>P-0001401-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20016</th>\n",
       "      <td>P-0001785-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20108</th>\n",
       "      <td>P-0002550-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20112</th>\n",
       "      <td>P-0002576-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20132</th>\n",
       "      <td>P-0002742-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20137</th>\n",
       "      <td>P-0002748-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20138</th>\n",
       "      <td>P-0002750-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20144</th>\n",
       "      <td>P-0002760-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20169</th>\n",
       "      <td>P-0002936-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20170</th>\n",
       "      <td>P-0002938-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20171</th>\n",
       "      <td>P-0002940-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20172</th>\n",
       "      <td>P-0002942-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20173</th>\n",
       "      <td>P-0002944-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20174</th>\n",
       "      <td>P-0002945-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20175</th>\n",
       "      <td>P-0002946-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20176</th>\n",
       "      <td>P-0002947-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20178</th>\n",
       "      <td>P-0002953-T02-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20234</th>\n",
       "      <td>P-0003697-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20238</th>\n",
       "      <td>P-0003702-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20331</th>\n",
       "      <td>P-0004842-T02-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20362</th>\n",
       "      <td>P-0005192-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>P-0005192-T01-IM5</td>\n",
       "      <td>True</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20404</th>\n",
       "      <td>P-0005666-T02-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20658</th>\n",
       "      <td>P-0007961-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21085</th>\n",
       "      <td>P-0014831-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21118</th>\n",
       "      <td>P-0015364-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21147</th>\n",
       "      <td>P-0015881-T05-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21247</th>\n",
       "      <td>P-0017878-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21352</th>\n",
       "      <td>P-0019732-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21356</th>\n",
       "      <td>P-0019817-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21402</th>\n",
       "      <td>P-0024195-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21406</th>\n",
       "      <td>P-0031102-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21706</th>\n",
       "      <td>P-0038201-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22078</th>\n",
       "      <td>P-0039073-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22185</th>\n",
       "      <td>P-0039361-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23053</th>\n",
       "      <td>P-0040935-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23118</th>\n",
       "      <td>P-0028971-T03-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23205</th>\n",
       "      <td>P-0041308-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23218</th>\n",
       "      <td>P-0041349-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23392</th>\n",
       "      <td>P-0040989-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23393</th>\n",
       "      <td>P-0040989-T03-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23648</th>\n",
       "      <td>P-0042229-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23779</th>\n",
       "      <td>P-0042670-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24406</th>\n",
       "      <td>P-0043869-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25605</th>\n",
       "      <td>P-0040699-T03-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25731</th>\n",
       "      <td>P-0020747-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26045</th>\n",
       "      <td>P-0045498-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26062</th>\n",
       "      <td>P-0045536-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26234</th>\n",
       "      <td>P-0045745-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26442</th>\n",
       "      <td>P-0046202-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26789</th>\n",
       "      <td>P-0039879-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26792</th>\n",
       "      <td>P-0046981-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26879</th>\n",
       "      <td>P-0035894-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27032</th>\n",
       "      <td>P-0047440-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27419</th>\n",
       "      <td>P-0047819-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27497</th>\n",
       "      <td>P-0047363-T02-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27620</th>\n",
       "      <td>P-0000088-T01-IM3</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27623</th>\n",
       "      <td>P-0040058-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27650</th>\n",
       "      <td>P-0025278-T01-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27651</th>\n",
       "      <td>P-0026935-T03-IM6</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27665</th>\n",
       "      <td>P-0008769-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27666</th>\n",
       "      <td>P-0008759-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27667</th>\n",
       "      <td>P-0008493-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27668</th>\n",
       "      <td>P-0008557-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27669</th>\n",
       "      <td>P-0008531-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27670</th>\n",
       "      <td>P-0008536-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27671</th>\n",
       "      <td>P-0008670-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27672</th>\n",
       "      <td>P-0008761-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27674</th>\n",
       "      <td>P-0008488-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27675</th>\n",
       "      <td>P-0008492-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27676</th>\n",
       "      <td>P-0008679-T02-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27677</th>\n",
       "      <td>P-0008484-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27678</th>\n",
       "      <td>P-0009175-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27679</th>\n",
       "      <td>P-0008459-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27680</th>\n",
       "      <td>P-0008758-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27682</th>\n",
       "      <td>P-0008458-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27683</th>\n",
       "      <td>P-0009324-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27684</th>\n",
       "      <td>P-0008457-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27685</th>\n",
       "      <td>P-0008558-T01-IM5</td>\n",
       "      <td>-1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Tumor_Id  wgd_x       tumor_sample wgd_y  inconsistencies\n",
       "495    P-0031435-T01-IM6     -1                NaN   NaN                1\n",
       "1282   P-0010672-T02-IM5     -1                NaN   NaN                1\n",
       "1855   P-0027242-T01-IM6     -1                NaN   NaN                1\n",
       "2906   P-0013827-T01-IM5     -1                NaN   NaN                1\n",
       "2957   P-0010083-T01-IM5     -1                NaN   NaN                1\n",
       "3263   P-0020798-T01-IM6     -1                NaN   NaN                1\n",
       "3271   P-0017938-T01-IM6     -1                NaN   NaN                1\n",
       "3284   P-0031170-T01-IM6     -1                NaN   NaN                1\n",
       "3898   P-0035720-T01-IM6     -1  P-0035720-T01-IM6  True                1\n",
       "3980   P-0036528-T01-IM6     -1                NaN   NaN                1\n",
       "5936   P-0017305-T01-IM5     -1                NaN   NaN                1\n",
       "6401   P-0017240-T02-IM6     -1                NaN   NaN                1\n",
       "6473   P-0016745-T01-IM6     -1                NaN   NaN                1\n",
       "6480   P-0006119-T01-IM5     -1                NaN   NaN                1\n",
       "6533   P-0025776-T01-IM6     -1                NaN   NaN                1\n",
       "6540   P-0019560-T01-IM6     -1                NaN   NaN                1\n",
       "7324   P-0011588-T01-IM5     -1                NaN   NaN                1\n",
       "9995   P-0000328-T02-IM6     -1                NaN   NaN                1\n",
       "10549  P-0019219-T01-IM6     -1                NaN   NaN                1\n",
       "11679  P-0000416-T02-IM3     -1                NaN   NaN                1\n",
       "12234  P-0027992-T01-IM6     -1                NaN   NaN                1\n",
       "12546  P-0023638-T02-IM6     -1                NaN   NaN                1\n",
       "12765  P-0014014-T01-IM6     -1                NaN   NaN                1\n",
       "13272  P-0028566-T01-IM6     -1                NaN   NaN                1\n",
       "13757  P-0015914-T01-IM6     -1                NaN   NaN                1\n",
       "15355  P-0029600-T02-IM6     -1                NaN   NaN                1\n",
       "15765  P-0022650-T01-IM6     -1                NaN   NaN                1\n",
       "16231  P-0004094-T03-IM6     -1                NaN   NaN                1\n",
       "16448  P-0026525-T02-IM6     -1                NaN   NaN                1\n",
       "17460  P-0000244-T01-IM3     -1                NaN   NaN                1\n",
       "17839  P-0014773-T01-IM6     -1                NaN   NaN                1\n",
       "17950  P-0024980-T01-IM6     -1                NaN   NaN                1\n",
       "17969  P-0000657-T01-IM3     -1                NaN   NaN                1\n",
       "19907  P-0000801-T02-IM3     -1                NaN   NaN                1\n",
       "19961  P-0001378-T01-IM3     -1                NaN   NaN                1\n",
       "19965  P-0001401-T01-IM3     -1                NaN   NaN                1\n",
       "20016  P-0001785-T01-IM3     -1                NaN   NaN                1\n",
       "20108  P-0002550-T01-IM3     -1                NaN   NaN                1\n",
       "20112  P-0002576-T01-IM3     -1                NaN   NaN                1\n",
       "20132  P-0002742-T01-IM3     -1                NaN   NaN                1\n",
       "20137  P-0002748-T01-IM3     -1                NaN   NaN                1\n",
       "20138  P-0002750-T01-IM3     -1                NaN   NaN                1\n",
       "20144  P-0002760-T01-IM3     -1                NaN   NaN                1\n",
       "20169  P-0002936-T02-IM3     -1                NaN   NaN                1\n",
       "20170  P-0002938-T02-IM3     -1                NaN   NaN                1\n",
       "20171  P-0002940-T02-IM3     -1                NaN   NaN                1\n",
       "20172  P-0002942-T02-IM3     -1                NaN   NaN                1\n",
       "20173  P-0002944-T02-IM3     -1                NaN   NaN                1\n",
       "20174  P-0002945-T02-IM3     -1                NaN   NaN                1\n",
       "20175  P-0002946-T02-IM3     -1                NaN   NaN                1\n",
       "20176  P-0002947-T02-IM3     -1                NaN   NaN                1\n",
       "20178  P-0002953-T02-IM3     -1                NaN   NaN                1\n",
       "20234  P-0003697-T01-IM5     -1                NaN   NaN                1\n",
       "20238  P-0003702-T01-IM5     -1                NaN   NaN                1\n",
       "20331  P-0004842-T02-IM5     -1                NaN   NaN                1\n",
       "20362  P-0005192-T01-IM5     -1  P-0005192-T01-IM5  True                1\n",
       "20404  P-0005666-T02-IM5     -1                NaN   NaN                1\n",
       "20658  P-0007961-T01-IM5     -1                NaN   NaN                1\n",
       "21085  P-0014831-T01-IM6     -1                NaN   NaN                1\n",
       "21118  P-0015364-T01-IM6     -1                NaN   NaN                1\n",
       "21147  P-0015881-T05-IM6     -1                NaN   NaN                1\n",
       "21247  P-0017878-T01-IM6     -1                NaN   NaN                1\n",
       "21352  P-0019732-T01-IM6     -1                NaN   NaN                1\n",
       "21356  P-0019817-T01-IM6     -1                NaN   NaN                1\n",
       "21402  P-0024195-T01-IM6     -1                NaN   NaN                1\n",
       "21406  P-0031102-T01-IM6     -1                NaN   NaN                1\n",
       "21706  P-0038201-T01-IM6     -1                NaN   NaN                1\n",
       "22078  P-0039073-T01-IM6     -1                NaN   NaN                1\n",
       "22185  P-0039361-T01-IM6     -1                NaN   NaN                1\n",
       "23053  P-0040935-T02-IM6     -1                NaN   NaN                1\n",
       "23118  P-0028971-T03-IM6     -1                NaN   NaN                1\n",
       "23205  P-0041308-T01-IM6     -1                NaN   NaN                1\n",
       "23218  P-0041349-T01-IM6     -1                NaN   NaN                1\n",
       "23392  P-0040989-T01-IM6     -1                NaN   NaN                1\n",
       "23393  P-0040989-T03-IM6     -1                NaN   NaN                1\n",
       "23648  P-0042229-T01-IM6     -1                NaN   NaN                1\n",
       "23779  P-0042670-T02-IM6     -1                NaN   NaN                1\n",
       "24406  P-0043869-T01-IM6     -1                NaN   NaN                1\n",
       "25605  P-0040699-T03-IM6     -1                NaN   NaN                1\n",
       "25731  P-0020747-T02-IM6     -1                NaN   NaN                1\n",
       "26045  P-0045498-T01-IM6     -1                NaN   NaN                1\n",
       "26062  P-0045536-T01-IM6     -1                NaN   NaN                1\n",
       "26234  P-0045745-T01-IM6     -1                NaN   NaN                1\n",
       "26442  P-0046202-T02-IM6     -1                NaN   NaN                1\n",
       "26789  P-0039879-T02-IM6     -1                NaN   NaN                1\n",
       "26792  P-0046981-T01-IM6     -1                NaN   NaN                1\n",
       "26879  P-0035894-T01-IM6     -1                NaN   NaN                1\n",
       "27032  P-0047440-T01-IM6     -1                NaN   NaN                1\n",
       "27419  P-0047819-T01-IM6     -1                NaN   NaN                1\n",
       "27497  P-0047363-T02-IM6     -1                NaN   NaN                1\n",
       "27620  P-0000088-T01-IM3     -1                NaN   NaN                1\n",
       "27623  P-0040058-T01-IM6     -1                NaN   NaN                1\n",
       "27650  P-0025278-T01-IM6     -1                NaN   NaN                1\n",
       "27651  P-0026935-T03-IM6     -1                NaN   NaN                1\n",
       "27665  P-0008769-T01-IM5     -1                NaN   NaN                1\n",
       "27666  P-0008759-T01-IM5     -1                NaN   NaN                1\n",
       "27667  P-0008493-T01-IM5     -1                NaN   NaN                1\n",
       "27668  P-0008557-T01-IM5     -1                NaN   NaN                1\n",
       "27669  P-0008531-T01-IM5     -1                NaN   NaN                1\n",
       "27670  P-0008536-T01-IM5     -1                NaN   NaN                1\n",
       "27671  P-0008670-T01-IM5     -1                NaN   NaN                1\n",
       "27672  P-0008761-T01-IM5     -1                NaN   NaN                1\n",
       "27674  P-0008488-T01-IM5     -1                NaN   NaN                1\n",
       "27675  P-0008492-T01-IM5     -1                NaN   NaN                1\n",
       "27676  P-0008679-T02-IM5     -1                NaN   NaN                1\n",
       "27677  P-0008484-T01-IM5     -1                NaN   NaN                1\n",
       "27678  P-0009175-T01-IM5     -1                NaN   NaN                1\n",
       "27679  P-0008459-T01-IM5     -1                NaN   NaN                1\n",
       "27680  P-0008758-T01-IM5     -1                NaN   NaN                1\n",
       "27682  P-0008458-T01-IM5     -1                NaN   NaN                1\n",
       "27683  P-0009324-T01-IM5     -1                NaN   NaN                1\n",
       "27684  P-0008457-T01-IM5     -1                NaN   NaN                1\n",
       "27685  P-0008558-T01-IM5     -1                NaN   NaN                1"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotated_cohort_wgd = annotated_cohort_filt[['tumor_sample', 'wgd']]\n",
    "merged = pd.merge(left=master, right=annotated_cohort_wgd, how='left', left_on='Tumor_Id', right_on='tumor_sample')\n",
    "merged['inconsistencies'] = merged.apply(check_inconsistencies, axis=1)\n",
    "merged[merged['inconsistencies'] == 1]#[~merged.wgd_y.isna()]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Conclusion:** We will keep the *wgd* column as the reference for wgd."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Filtering the annotated_data\n",
    "1. We do not want duplicates in Tumor_Id\n",
    "2. As maf_cohort is necessary to know if we have tp53 mutants or not, we will reduce our analysis to the samples of maf_cohort (loss of ~3 600 samples over 42 153)\n",
    "3. Moreover, we need to access tp53 cn state, so we have to be sure all samples are in gene_level file\n",
    "4. As we did for the TP53 cohort, we will filter samples with purity < 0.2 BUT we will rescue TP53 mutants with no purity a tp53_vaf > 0.15\n",
    "5. We remove samples with INDETERMINATE copy number state"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotated_cohort = pd.read_csv(data_path + 'impact-facets-tp53/new_data_facets/msk_impact_facets_annotated.cohort.txt', sep='\\t')\n",
    "maf_cohort = pd.read_csv(data_path + 'impact-facets-tp53/new_data_facets/msk_impact_facets_annotated.ccf.maf', sep='\\t')\n",
    "gene_level = pd.read_csv(data_path + 'impact-facets-tp53/new_data_facets/msk_impact_facets_annotated.gene_level_grepTP53.txt', sep='\\t')\n",
    "\n",
    "maf_samples = list(set(maf_cohort.Tumor_Sample_Barcode))\n",
    "\n",
    "annotated_cohort_filt = normal_samp_duplicates_filter(annotated_cohort, 'sample_id', 'purity')\n",
    "annotated_cohort_filt = normal_samp_duplicates_filter(annotated_cohort_filt, 'sample_id', 'purity')\n",
    "# the last duplicated samples are samples with purity = NaN so we simply use deplicated method\n",
    "cohort = annotated_cohort_filt.drop_duplicates(subset='tumor_sample')\n",
    "\n",
    "# We take only samples that are in maf_samples\n",
    "cohort = cohort[cohort.tumor_sample.isin(maf_samples)]\n",
    "\n",
    "# Finally we want to be able to accesss the TP53 locus copy number state, so we need to make sure all samples are\n",
    "# in gene_level\n",
    "\n",
    "# We remove the following samples that are not in gene_level file\n",
    "cohort = cohort[~cohort.tumor_sample.isin(['P-0000157-T01-IM3','P-0000158-T01-IM3','P-0000204-T01-IM3','P-0000206-T01-IM3','P-0000273-T01-IM3','P-0000308-T01-IM3','P-0000320-T01-IM3','P-0000396-T01-IM3','P-0007206-T01-IM5','P-0011437-T01-IM5','P-0037557-T01-IM6','P-0043334-T01-IM6','P-0043373-T01-IM6','P-0043377-T01-IM6','P-0045294-T01-IM6','P-0045404-T01-IM6'])]\n",
    "\n",
    "\n",
    "# Now we filter on purity\n",
    "\n",
    "# We create a list for the interesting samples that we will integrate to our data\n",
    "cohort_na_tp53 = maf_cohort[(maf_cohort.purity == 0.3) | (maf_cohort.purity.isna())][maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "samples_maxvaf=[]\n",
    "for sample in list(set(cohort_na_tp53.Tumor_Sample_Barcode)):\n",
    "    sample_1 = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == sample]\n",
    "    if max(sample_1.t_var_freq)>0.15:\n",
    "        samples_maxvaf.append(sample)    \n",
    "# we remove THE sample that have a tp53_vaf < 2%\n",
    "samples_maxvaf.remove('P-0035205-T03-IM6')\n",
    "\n",
    "# Samples with purity > 0.2\n",
    "samples_purity = list(set(maf_cohort[maf_cohort['purity'] > 0.2].Tumor_Sample_Barcode))\n",
    "\n",
    "samples = samples_maxvaf + samples_purity\n",
    "samples.remove('P-0016457-T04-IM6')\n",
    "samples.remove('P-0014703-T01-IM6')\n",
    "\n",
    "cohort = cohort[cohort.tumor_sample.isin(samples)]\n",
    "\n",
    "# OPTIONAL\n",
    "# Removing INDETERMINATE tp53 CN state samples\n",
    "gene_level_ = gene_level[gene_level['gene'] == 'TP53']\n",
    "samples_ind = list(set(gene_level_[gene_level_['cn_state'] == 'INDETERMINATE']['sample']))\n",
    "\n",
    "cohort = cohort[~cohort.sample_id.isin(samples_ind)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Final size of our cohort: 30289\n",
      "\n",
      "We verify that purity>0.2 in our maf_cohort file:\n",
      "0.200051999476184\n"
     ]
    }
   ],
   "source": [
    "print('Final size of our cohort: ' + str(len(list(cohort.tumor_sample))))\n",
    "\n",
    "maf_cohort_study = maf_cohort[maf_cohort.Tumor_Sample_Barcode.isin(list(cohort.tumor_sample))]\n",
    "print('\\nWe verify that purity>0.2 in our maf_cohort file:')\n",
    "print(maf_cohort_study[maf_cohort_study.Tumor_Sample_Barcode.isin(samples_purity)].purity.min())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('samples_cohort.pkl', 'wb') as f:\n",
    "    pickle.dump(list(cohort.tumor_sample), f)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now we have unique tumor_sample, we can begin the analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# WGD "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WGD Proportion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 195,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of samples with WGD: 11350 - 37.47 %\n",
      "Number of samples without WGD: 18939 - 62.53 %\n"
     ]
    }
   ],
   "source": [
    "print('Number of samples with WGD: ' + str(cohort[cohort['wgd'] == True].shape[0]) + ' - ' + str(round(100* cohort[cohort['wgd'] == True].shape[0]/cohort.shape[0],2))+ ' %')\n",
    "print('Number of samples without WGD: ' + str(cohort[cohort['wgd'] == False].shape[0])  + ' - ' + str(round(100* cohort[cohort['wgd'] == False].shape[0]/cohort.shape[0],2))+ ' %')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 196,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We filter only WGD Samples\n",
    "cohort_wgd = cohort[cohort['wgd'] == 1]\n",
    "\n",
    "# We store tumor sample ids and matched normal sample ids to filter gene_level anf maf_cohort\n",
    "samples_wgd = list(set(cohort_wgd.tumor_sample))\n",
    "samples_matched_wgd = list(set(cohort_wgd.sample_id))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 197,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We filter the maf_cohort on the wgd samples\n",
    "maf_cohort_wgd = maf_cohort[maf_cohort.Tumor_Sample_Barcode.isin(samples_wgd)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Filter out uncertain tp53 cn_state\n",
    "Further in the analysis, we will need to time TP53 mutations. This timing highly rely on TP53 copy number states. The problem is that we have some inconistencies between gene_level cn_state and the WGD state ie we have *CNLOH* at tp53 locus while we should have *CNLOH BEFORE* or *CNLOH AFTER*... \n",
    "\n",
    "Consequently we want to filter out all samples with uncertain/inconsistent copy number state at tp53 locus"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 213,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We show the distribution of CN state withtin the WGD cohort\n",
    "gene_level_tp53_wgd = gene_level[gene_level['gene'] == 'TP53'][gene_level['sample'].isin(samples_matched_wgd)]\n",
    "gene_level_tp53_wgd['tumor_id'] = gene_level.apply(lambda x: x['sample'][:17], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cn_state</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>LOSS BEFORE</th>\n",
       "      <td>4865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; LOSS</th>\n",
       "      <td>1887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS AFTER</th>\n",
       "      <td>1278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TETRAPLOID</th>\n",
       "      <td>652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOUBLE LOSS AFTER</th>\n",
       "      <td>533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE</th>\n",
       "      <td>464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS BEFORE &amp; AFTER</th>\n",
       "      <td>422</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GAIN</th>\n",
       "      <td>325</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH AFTER</th>\n",
       "      <td>146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH</th>\n",
       "      <td>144</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOMDEL</th>\n",
       "      <td>126</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DIPLOID</th>\n",
       "      <td>91</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP (BALANCED)</th>\n",
       "      <td>90</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; GAIN</th>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP</th>\n",
       "      <td>74</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HETLOSS</th>\n",
       "      <td>48</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP (LOH)</th>\n",
       "      <td>43</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS &amp; GAIN</th>\n",
       "      <td>40</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH &amp; GAIN</th>\n",
       "      <td>38</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     count\n",
       "cn_state                  \n",
       "LOSS BEFORE           4865\n",
       "CNLOH BEFORE & LOSS   1887\n",
       "LOSS AFTER            1278\n",
       "TETRAPLOID             652\n",
       "DOUBLE LOSS AFTER      533\n",
       "CNLOH BEFORE           464\n",
       "LOSS BEFORE & AFTER    422\n",
       "GAIN                   325\n",
       "CNLOH AFTER            146\n",
       "CNLOH                  144\n",
       "HOMDEL                 126\n",
       "DIPLOID                 91\n",
       "AMP (BALANCED)          90\n",
       "CNLOH BEFORE & GAIN     84\n",
       "AMP                     74\n",
       "HETLOSS                 48\n",
       "AMP (LOH)               43\n",
       "LOSS & GAIN             40\n",
       "CNLOH & GAIN            38"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Let's take a look to all cn_state\n",
    "get_groupby(gene_level_tp53_wgd, 'cn_state', 'count').sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We want to filter out the following cn_state that are inconsistent with a WGD state:\n",
    "* CNLOH\n",
    "* GAIN\n",
    "* DIPLOID\n",
    "* AMP (LOH)\n",
    "* HETLOSS\n",
    "* CNLOH & GAIN\n",
    "* AMP\n",
    "* LOSS & GAIN\n",
    "* AMP (BALANCED)\n",
    "* HOMDEL\n",
    "\n",
    "Additionnaly we will filter *LOSS BEFORE & AFTER* because it is impossible to know the timing of the mutation with this cn_state."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cn_state</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>LOSS BEFORE</th>\n",
       "      <td>4865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; LOSS</th>\n",
       "      <td>1887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS AFTER</th>\n",
       "      <td>1278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TETRAPLOID</th>\n",
       "      <td>652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOUBLE LOSS AFTER</th>\n",
       "      <td>533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE</th>\n",
       "      <td>464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH AFTER</th>\n",
       "      <td>146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; GAIN</th>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     count\n",
       "cn_state                  \n",
       "LOSS BEFORE           4865\n",
       "CNLOH BEFORE & LOSS   1887\n",
       "LOSS AFTER            1278\n",
       "TETRAPLOID             652\n",
       "DOUBLE LOSS AFTER      533\n",
       "CNLOH BEFORE           464\n",
       "CNLOH AFTER            146\n",
       "CNLOH BEFORE & GAIN     84"
      ]
     },
     "execution_count": 215,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Let's see how many samples are gathered in these cn_state\n",
    "to_filter = ['CNLOH', 'GAIN', 'DIPLOID', 'AMP (LOH)', 'HETLOSS',\n",
    "             'CNLOH & GAIN', 'AMP', 'LOSS & GAIN', 'AMP (BALANCED)',\n",
    "            'HOMDEL', 'LOSS BEFORE & AFTER']\n",
    "\n",
    "gene_level_tp53_wgd = gene_level_tp53_wgd[~gene_level_tp53_wgd.cn_state.isin(to_filter)]\n",
    "\n",
    "\n",
    "# Let's take a look to all cn_state\n",
    "get_groupby(gene_level_tp53_wgd, 'cn_state', 'count').sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Now we filter maf_cohort_wgd and cohort_wgd\n",
    "samples = list(set(gene_level_tp53_wgd.tumor_id))\n",
    "maf_cohort_wgd = maf_cohort_wgd[maf_cohort_wgd.Tumor_Sample_Barcode.isin(samples)]\n",
    "cohort_wgd = cohort_wgd[cohort_wgd.tumor_sample.isin(samples)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 217,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "9909"
      ]
     },
     "execution_count": 217,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# We check the length of cohort_wgd \n",
    "cohort_wgd.shape[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## LOH status creation\n",
    "\n",
    "1. We define LOH (Loss of Heterozygosity) as having lost the second allele, which in practice is traduced numerically by having a lcn (low copy number) of 0."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "First, le's take a look at the different TP53 Copy Number states involved in our WGD cohort:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cn_state</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>LOSS BEFORE</th>\n",
       "      <td>4865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; LOSS</th>\n",
       "      <td>1887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS AFTER</th>\n",
       "      <td>1278</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TETRAPLOID</th>\n",
       "      <td>652</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOUBLE LOSS AFTER</th>\n",
       "      <td>533</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE</th>\n",
       "      <td>464</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH AFTER</th>\n",
       "      <td>146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; GAIN</th>\n",
       "      <td>84</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     count\n",
       "cn_state                  \n",
       "LOSS BEFORE           4865\n",
       "CNLOH BEFORE & LOSS   1887\n",
       "LOSS AFTER            1278\n",
       "TETRAPLOID             652\n",
       "DOUBLE LOSS AFTER      533\n",
       "CNLOH BEFORE           464\n",
       "CNLOH AFTER            146\n",
       "CNLOH BEFORE & GAIN     84"
      ]
     },
     "execution_count": 218,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# We show the distribution of CN state withtin the WGD cohort\n",
    "get_groupby(gene_level_tp53_wgd, 'cn_state', 'count').sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We identify here that we have 3 major CN states: *LOSS BEFORE*, *CNLOH BEFORE & LOSS* and *LOSS AFTER*.\n",
    "Moreover we identify the CN states leading to a LOH:\n",
    "* LOSS BEFORE\n",
    "* CNLOH BEFORE & LOSS\n",
    "* CNLOH BEFORE\n",
    "* CNLOH BEFORE & GAIN\n",
    "\n",
    "**This is strictly equivalent to say lcn = 0 (lower copy number)**\n",
    "\n",
    "So we're able to identify LOH samples and compute the proportion of LOH:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 219,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 12.2 s, sys: 52.3 ms, total: 12.2 s\n",
      "Wall time: 12.2 s\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOH</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>False</th>\n",
       "      <td>2609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>True</th>\n",
       "      <td>7300</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       count\n",
       "LOH         \n",
       "False   2609\n",
       "True    7300"
      ]
     },
     "execution_count": 219,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "def get_loh(x):\n",
    "    if gene_level_tp53_wgd[gene_level_tp53_wgd['sample'] == x.sample_id]['cn_state'].values[0] in ['LOSS BEFORE', 'CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS BEFORE & AFTER', 'CNLOH BEFORE & GAIN']:\n",
    "        return True\n",
    "    else: return False\n",
    "                                                                                                                \n",
    "# we create a new column named LOH in our cohort_wgd\n",
    "cohort_wgd['LOH'] = cohort_wgd.apply(get_loh, axis=1)\n",
    "get_groupby(cohort_wgd, 'LOH', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of LOH samples in WGD cohort: 7300 - 73.67 %\n",
      "Number of non LOH samples in WGD cohort: 2609 - 26.33 %\n"
     ]
    }
   ],
   "source": [
    "n_loh = len(set(cohort_wgd[cohort_wgd['LOH'] == True].tumor_sample))\n",
    "n_noloh = len(set(cohort_wgd[cohort_wgd['LOH'] == False].tumor_sample))\n",
    "total = len(set(cohort_wgd.tumor_sample)) \n",
    "\n",
    "print('Number of LOH samples in WGD cohort: ' + str(n_loh) + ' - ' + str(round(100*n_loh/total, 2)) + ' %')\n",
    "print('Number of non LOH samples in WGD cohort: ' + str(n_noloh) + ' - ' + str(round(100*n_noloh/total, 2)) + ' %')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 221,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of TP53 mutated samples in WGD setting: 6714 - 67.76 %\n",
      "Number of TP53 wild-type samples in WGD setting: 3195 - 32.239999999999995 %\n"
     ]
    }
   ],
   "source": [
    "# We count the samples that has at least one tp53 mutation\n",
    "total = len(set(maf_cohort_wgd.Tumor_Sample_Barcode))\n",
    "n = len(set(maf_cohort_wgd[maf_cohort['Hugo_Symbol'] == 'TP53'].Tumor_Sample_Barcode))\n",
    "print('Number of TP53 mutated samples in WGD setting: ' + str(n) + ' - ' + str(round(100*n/len(cohort_wgd),2)) + ' %')\n",
    "print('Number of TP53 wild-type samples in WGD setting: ' + str(total - n) + ' - ' + str(100 -round(100*n/len(cohort_wgd),2)) + ' %')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Creation of tp53_count column"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 222,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 7.53 s, sys: 423 ms, total: 7.95 s\n",
      "Wall time: 7.95 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "# First we need to create a sample_mut_key to identify duplicated mutations\n",
    "maf_cohort_wgd['mut_key'] = maf_cohort_wgd.apply(lambda h: str(h.Chromosome)+'_'+str(h.Start_Position)+'_'+str(h.Reference_Allele)+'_'+str(h.Tumor_Seq_Allele2), axis=1) \n",
    "\n",
    "# Create a sample key to differentiate duplicates\n",
    "maf_cohort_wgd['sample_mut_key'] = maf_cohort_wgd.apply(lambda h: h.Tumor_Sample_Barcode + h.mut_key, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%%time\n",
    "# Here we count the number of tp53 mutations per sample\n",
    "def get_tp53_count(x):\n",
    "    h = maf_cohort_wgd[maf_cohort_wgd['Tumor_Sample_Barcode'] == x.tumor_sample][maf_cohort_wgd['Hugo_Symbol'] == 'TP53']\n",
    "    h.drop_duplicates(subset='sample_mut_key')\n",
    "    return h.shape[0]\n",
    "\n",
    "cohort_wgd['tp53_count'] = cohort_wgd.apply(get_tp53_count, axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Plot of the number of mutations + LOH Status\n",
    "Here we create a column *tp53_count* and compute the plot of the number of tp53 mutation per samples colorcoded by LOH status"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "# First we need to groupby our cohort_wgd file to have the size of the different groups\n",
    "loh_tp53count = pd.DataFrame(cohort_wgd[['tp53_count', 'LOH']].groupby(['tp53_count', 'LOH']).size()).unstack()\n",
    "loh_tp53count = loh_tp53count.fillna(0)\n",
    "loh_tp53count.columns=['NO LOH', 'LOH']\n",
    "loh_tp53count.loc['>=2'] = [loh_tp53count['NO LOH'][2:].sum(),loh_tp53count['LOH'][2:].sum()]\n",
    "loh_tp53count = loh_tp53count.drop([2,3,4,5],axis=0)\n",
    "display(loh_tp53count)\n",
    "\n",
    "sns.set_style(\"whitegrid\", {'grid.color': '1.'})\n",
    "fig = plt.figure(figsize=(11,5))\n",
    "ax = plt.subplot2grid(shape=(1,1), loc=(0,0), colspan=1)\n",
    "h = loh_tp53count.plot(kind = 'bar', stacked=True, ax=ax)\n",
    "ax.set_xticklabels(['0', '1','>=2'], rotation=0)\n",
    "ax.legend(['NO LOH', 'LOH'], fontsize=14)\n",
    "ax.set_title('# of TP53 muts in WGD cohort - colorcoded by LOH state',weight='bold', fontsize=20)\n",
    "ax.spines['right'].set_visible(False)\n",
    "ax.spines['top'].set_visible(False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "So we see an enrichment of LOH in samples with 1 tp53 mutation. The questions are:\n",
    "* What are the TP53 states involved in this LOH state? LOSS ? CNLOH ? \n",
    "    * For the different tp53 mutation count categories (0,1,>=2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WGD - TP53"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11350"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cohort_wgd.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We define gene_level_tp53 which represents gene level filtered on tp53 locus\n",
    "gene_level_tp53 = gene_level[gene_level['gene'] == 'TP53']"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### WGD - TP53 - LOH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "wgd_tp53_loh_samples = list(set(cohort_wgd[cohort_wgd['LOH'] == True][cohort_wgd['tp53_count'] >= 1].sample_id))\n",
    "gene_level_tp53_wgd_loh = gene_level_tp53[gene_level_tp53['sample'].isin(wgd_tp53_loh_samples)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6250"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(wgd_tp53_loh_samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of LOSS: 4156 - 66.5 %\n",
      "Number of CNLOH: 2094 - 33.5 %\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cn_state</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE</th>\n",
       "      <td>420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; GAIN</th>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; LOSS</th>\n",
       "      <td>1594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS BEFORE</th>\n",
       "      <td>3881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS BEFORE &amp; AFTER</th>\n",
       "      <td>275</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     count\n",
       "cn_state                  \n",
       "CNLOH BEFORE           420\n",
       "CNLOH BEFORE & GAIN     80\n",
       "CNLOH BEFORE & LOSS   1594\n",
       "LOSS BEFORE           3881\n",
       "LOSS BEFORE & AFTER    275"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h = get_groupby(gene_level_tp53_wgd_loh, 'cn_state', 'count')\n",
    "LOSS = h['count']['LOSS BEFORE'] + h['count']['LOSS BEFORE & AFTER']\n",
    "CNLOH = h.sum().values[0] - h['count']['LOSS BEFORE'] - h['count']['LOSS BEFORE & AFTER']\n",
    "print('Number of LOSS: ' + str(LOSS) + ' - ' + str(round(100*LOSS/h.sum().values[0],2)) + ' %')\n",
    "print('Number of CNLOH: ' + str(CNLOH) + ' - ' + str(round(100*CNLOH/h.sum().values[0],2)) + ' %')\n",
    "h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### WGD - TP53 - No LOH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1168"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "wgd_tp53_noloh_samples = list(set(cohort_wgd[cohort_wgd['LOH'] == False][cohort_wgd['tp53_count'] >= 1].sample_id))\n",
    "gene_level_tp53_wgd_noloh = gene_level_tp53[gene_level_tp53['sample'].isin(wgd_tp53_noloh_samples)]\n",
    "len(wgd_tp53_noloh_samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of WGD - 1tp53 - No LOH: 832 - 71.23 %\n"
     ]
    }
   ],
   "source": [
    "# Samples with One mutation\n",
    "wgd_1tp53_noloh_samples = list(set(cohort_wgd[cohort_wgd['LOH'] == False][cohort_wgd['tp53_count'] == 1].sample_id))\n",
    "print( 'Number of WGD - 1tp53 - No LOH: ' + str(len(wgd_1tp53_noloh_samples)) + ' - '+ str(round(100*len(wgd_1tp53_noloh_samples)/len(wgd_tp53_noloh_samples), 2)) + ' %')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of WGD - >=2 tp53 - No LOH: 336 - 28.77 %\n"
     ]
    }
   ],
   "source": [
    "# Samples with >=2 mutations\n",
    "wgd_2tp53_noloh_samples = list(set(cohort_wgd[cohort_wgd['LOH'] == False][cohort_wgd['tp53_count'] >= 2].sample_id))\n",
    "print( 'Number of WGD - >=2 tp53 - No LOH: ' + str(len(wgd_2tp53_noloh_samples)) + ' - '+ str(round(100*len(wgd_2tp53_noloh_samples)/len(wgd_tp53_noloh_samples), 2)) + ' %')\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WGD - No TP53"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "We verify that we have the good amount of samples: 3932\n"
     ]
    }
   ],
   "source": [
    "wgd_notp53_samples = list(set(cohort_wgd[cohort_wgd['tp53_count'] == 0].sample_id))\n",
    "gene_level_tp53_wgd_notp53 = gene_level_tp53[gene_level_tp53['sample'].isin(wgd_notp53_samples)]\n",
    "print('We verify that we have the good amount of samples: ' + str(len(wgd_notp53_samples)))"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### WGD - No TP53 - No LOH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of WGD - No TP53 - No LOH Samples: 2460 - 62.56 %\n"
     ]
    }
   ],
   "source": [
    "# Now we count the number of samples with WGD - No TP53 mut - No LOH\n",
    "wgd_notp53_noloh_samples = list(set(cohort_wgd[cohort_wgd['tp53_count'] == 0][cohort_wgd['LOH'] == False].sample_id))\n",
    "gene_level_tp53_wgd_notp53_noloh = gene_level_tp53[gene_level_tp53['sample'].isin(wgd_notp53_noloh_samples)]\n",
    "print('Number of WGD - No TP53 - No LOH Samples: ' + str(len(wgd_notp53_noloh_samples)) + ' - ' + str(round(100*len(wgd_notp53_noloh_samples)/len(wgd_notp53_samples), 2)) + ' %')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cn_state</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>LOSS AFTER</th>\n",
       "      <td>890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TETRAPLOID</th>\n",
       "      <td>527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOUBLE LOSS AFTER</th>\n",
       "      <td>395</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GAIN</th>\n",
       "      <td>238</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOMDEL</th>\n",
       "      <td>116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP (BALANCED)</th>\n",
       "      <td>70</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH AFTER</th>\n",
       "      <td>58</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DIPLOID</th>\n",
       "      <td>47</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP</th>\n",
       "      <td>45</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH</th>\n",
       "      <td>30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS &amp; GAIN</th>\n",
       "      <td>16</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HETLOSS</th>\n",
       "      <td>14</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP (LOH)</th>\n",
       "      <td>8</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH &amp; GAIN</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   count\n",
       "cn_state                \n",
       "LOSS AFTER           890\n",
       "TETRAPLOID           527\n",
       "DOUBLE LOSS AFTER    395\n",
       "GAIN                 238\n",
       "HOMDEL               116\n",
       "AMP (BALANCED)        70\n",
       "CNLOH AFTER           58\n",
       "DIPLOID               47\n",
       "AMP                   45\n",
       "CNLOH                 30\n",
       "LOSS & GAIN           16\n",
       "HETLOSS               14\n",
       "AMP (LOH)              8\n",
       "CNLOH & GAIN           6"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# What are the different CN_states\n",
    "get_groupby(gene_level_tp53_wgd_notp53_noloh, 'cn_state', 'count').sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### WGD - No TP53 - LOH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of WGD - No TP53 - LOH Samples: 1472 - 37.44 %\n"
     ]
    }
   ],
   "source": [
    "# Now we count the number of samples with WGD - No TP53 mut - LOH\n",
    "wgd_notp53_loh_samples = list(set(cohort_wgd[cohort_wgd['tp53_count'] == 0][cohort_wgd['LOH'] == True].sample_id))\n",
    "gene_level_tp53_wgd_notp53_loh = gene_level_tp53[gene_level_tp53['sample'].isin(wgd_notp53_loh_samples)]\n",
    "print('Number of WGD - No TP53 - LOH Samples: ' + str(len(wgd_notp53_loh_samples)) + ' - ' + str(round(100*len(wgd_notp53_loh_samples)/len(wgd_notp53_samples), 2)) + ' %')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's dig in the different Copy Number States involved in the *WGD - No TP53 - LOH* subcohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of LOSS: 1131 - 76.83 %\n",
      "Number of CNLOH: 341 - 23.17 %\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cn_state</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE</th>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; GAIN</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; LOSS</th>\n",
       "      <td>293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS BEFORE</th>\n",
       "      <td>984</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS BEFORE &amp; AFTER</th>\n",
       "      <td>147</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     count\n",
       "cn_state                  \n",
       "CNLOH BEFORE            44\n",
       "CNLOH BEFORE & GAIN      4\n",
       "CNLOH BEFORE & LOSS    293\n",
       "LOSS BEFORE            984\n",
       "LOSS BEFORE & AFTER    147"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h = get_groupby(gene_level_tp53_wgd_notp53_loh, 'cn_state', 'count')\n",
    "LOSS = h['count']['LOSS BEFORE'] + h['count']['LOSS BEFORE & AFTER']\n",
    "CNLOH = h.sum().values[0] - h['count']['LOSS BEFORE'] - h['count']['LOSS BEFORE & AFTER']\n",
    "print('Number of LOSS: ' + str(LOSS) + ' - ' + str(round(100*LOSS/h.sum().values[0],2)) + ' %')\n",
    "print('Number of CNLOH: ' + str(CNLOH) + ' - ' + str(round(100*CNLOH/h.sum().values[0],2)) + ' %')\n",
    "h"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Timing TP53 mutations"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We begin by defining the WGD - TP53 Cohort\n",
    "\n",
    "gene_level_tp53 = gene_level[gene_level['gene'] == 'TP53']\n",
    "\n",
    "cohort_wgd_tp53 = cohort_wgd[cohort_wgd['tp53_count'] >= 1]\n",
    "wgd_tp53_samples = list(set(cohort_wgd_tp53.sample_id))\n",
    "wgd_tp53_tumors = list(set(cohort_wgd_tp53.tumor_sample))\n",
    "\n",
    "gene_level_tp53_wgd = gene_level_tp53[gene_level_tp53['sample'].isin(wgd_tp53_samples)]\n",
    "maf_cohort_wgd_tp53 = maf_cohort[maf_cohort['Tumor_Sample_Barcode'].isin(wgd_tp53_tumors)][maf_cohort['Hugo_Symbol'] == 'TP53']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7925"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_cohort_wgd_tp53.shape[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will add a  column to **maf_cohort_wgd_tp53** called *tp53_res*."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 20.6 s, sys: 81.9 ms, total: 20.7 s\n",
      "Wall time: 20.7 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "def get_tp53_res(x):\n",
    "    tumor = x.Tumor_Sample_Barcode\n",
    "    pur = x.purity\n",
    "    vaf = x.t_var_freq\n",
    "    tcn = gene_level_tp53_wgd[gene_level_tp53_wgd['sample'].str[:17] == tumor]['tcn'].values[0]\n",
    "    \n",
    "    tp53_exp_nb = (vaf/pur) * (tcn * pur + 2 * (1 - pur))\n",
    "    tp53_res = tcn - tp53_exp_nb\n",
    "    \n",
    "    return tp53_res\n",
    "\n",
    "maf_cohort_wgd_tp53['tp53_res'] = maf_cohort_wgd_tp53.apply(get_tp53_res, axis=1)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Let's have a look to the distribution of tp53_res:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "23"
      ]
     },
     "execution_count": 99,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_cohort_wgd_tp53.tp53_res.isna().sum()#describe()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We see that we have 25% of our tp53 mutations with a negative tp53_res. What does that mean?\n",
    "\n",
    "Basically, having a look to the computation to reach tp53_res, there are two main actors: the *vaf* and the *purity*. If the purity is low, then we will have a low - or negative - tp53_res. But we cut the cohort with purity > 0.2. On the other hand the vaf can be very high. It is indeed. the case in the negative. tp53_res mutations. \n",
    "\n",
    "Let's see the difference of vaf for tp53_res > 0 and tp53_res <0 in the next cell:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    tp53<0     |     tp53>=0\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "                <table style=\"display:inline\" border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>t_var_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>2431.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>0.574134</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>0.205412</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.025140</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>0.402591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>0.555556</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>0.741409</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>0.992523</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table style=\"display:inline\">                <table style=\"display:inline\" border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>t_var_freq</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>5471.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>0.407922</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>0.196796</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>0.020173</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>0.256659</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>0.379888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>0.541667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>0.970268</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table style=\"display:inline\">"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# In this cell we get the tp53_res distribution for the cn_state considered\n",
    "\n",
    "h0 = pd.DataFrame(maf_cohort_wgd_tp53[maf_cohort_wgd_tp53['tp53_res']<0].t_var_freq.describe())#.tp53_res.describe()\n",
    "h1 = pd.DataFrame(maf_cohort_wgd_tp53[maf_cohort_wgd_tp53['tp53_res']>=0].t_var_freq.describe())\n",
    "\n",
    "print( '    tp53<0     |     tp53>=0')\n",
    "display_side_by_side(h0, h1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>cn_state</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>LOSS BEFORE</th>\n",
       "      <td>3881</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; LOSS</th>\n",
       "      <td>1594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE</th>\n",
       "      <td>420</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS AFTER</th>\n",
       "      <td>388</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS BEFORE &amp; AFTER</th>\n",
       "      <td>275</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DOUBLE LOSS AFTER</th>\n",
       "      <td>138</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>TETRAPLOID</th>\n",
       "      <td>125</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH</th>\n",
       "      <td>114</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH AFTER</th>\n",
       "      <td>88</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GAIN</th>\n",
       "      <td>87</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH BEFORE &amp; GAIN</th>\n",
       "      <td>80</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>DIPLOID</th>\n",
       "      <td>44</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP (LOH)</th>\n",
       "      <td>35</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HETLOSS</th>\n",
       "      <td>34</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>CNLOH &amp; GAIN</th>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP</th>\n",
       "      <td>29</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>LOSS &amp; GAIN</th>\n",
       "      <td>24</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AMP (BALANCED)</th>\n",
       "      <td>20</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOMDEL</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                     count\n",
       "cn_state                  \n",
       "LOSS BEFORE           3881\n",
       "CNLOH BEFORE & LOSS   1594\n",
       "CNLOH BEFORE           420\n",
       "LOSS AFTER             388\n",
       "LOSS BEFORE & AFTER    275\n",
       "DOUBLE LOSS AFTER      138\n",
       "TETRAPLOID             125\n",
       "CNLOH                  114\n",
       "CNLOH AFTER             88\n",
       "GAIN                    87\n",
       "CNLOH BEFORE & GAIN     80\n",
       "DIPLOID                 44\n",
       "AMP (LOH)               35\n",
       "HETLOSS                 34\n",
       "CNLOH & GAIN            32\n",
       "AMP                     29\n",
       "LOSS & GAIN             24\n",
       "AMP (BALANCED)          20\n",
       "HOMDEL                  10"
      ]
     },
     "execution_count": 75,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(gene_level_tp53_wgd, 'cn_state', 'count').sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Now let's create a column with the timing of the tp53 mutation.\n",
    "For that purpose, we use the cn_state of each sample to see what are the possibilities in terms of tp53_res.\n",
    "\n",
    "Let's take an example: **LOSS BEFORE** (aka the most represented cn_state in our subcohort)\n",
    "\n",
    "To sum up the different steps of this cn_state: \n",
    "1. we have a DIPLOID state (1/1)\n",
    "2. We have a LOSS (1/0)\n",
    "3. We have a WGD 'duplicating' this allele (2/0)\n",
    "\n",
    "We have 2 possibilities for the mutation:\n",
    "1. If it arose before WGD, the theoretical tp53_res is 0 because we have one allele mutated (after the LOSS) that is duplicated.\n",
    "2. If it arose after WGD, the theoretical tp53_res is 1 because we have one allele mutated AND one WT allele.\n",
    "\n",
    "\n",
    "In this case it is pretty easy: if tp53_res == 0   ==>  BEFORE\n",
    "\n",
    "But do not forget that tp53_res is a statistical measurement, so we have to set up a threshold to capture as many confident samples as we can.\n",
    "We can take thr = 0.5 for example.\n",
    "\n",
    "The mecanism is the same with all cn_states. But some are very difficult."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_tp53_timing_wgd(x):\n",
    "    tumor = x.Tumor_Sample_Barcode\n",
    "    tp53_res = x.tp53_res\n",
    "    cn_state = gene_level_tp53_wgd[gene_level_tp53_wgd['sample'].str[:17] == tumor]['cn_state'].values[0]\n",
    "    \n",
    "    # Now we enumerate all cn_state and put a threshold\n",
    "    if cn_state == 'LOSS BEFORE':\n",
    "        thr = 0.5\n",
    "    elif cn_state ==  'CNLOH BEFORE & LOSS':\n",
    "        thr = 1\n",
    "    elif cn_state == 'CNLOH BEFORE':\n",
    "        thr = 2.5\n",
    "    # The following is tricky because tp53_res=2 could be in both cases\n",
    "    elif cn_state == 'LOSS AFTER':\n",
    "        thr = 1.25\n",
    "    # The following is impossible to classifiy as both cases have tp53_res = 0\n",
    "    # In this case we put thr = -1\n",
    "    elif cn_state == 'LOSS BEFORE & AFTER':\n",
    "        thr = -1\n",
    "    elif cn_state == 'DOUBLE LOSS AFTER':\n",
    "        thr = 0.25\n",
    "    elif cn_state == 'TETRAPLOID':\n",
    "        thr = 2.5\n",
    "    elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "        thr = 2\n",
    "    # For all other cases we put thr = -1 for INDETERMINATE\n",
    "    else:\n",
    "        thr = -1\n",
    "    \n",
    "    if thr != -1:\n",
    "        if tp53_res <= thr:\n",
    "            return 'BEFORE'\n",
    "        elif tp53_res > thr:\n",
    "            return 'AFTER'\n",
    "        else: return 'INDETERMINATE'\n",
    "    elif thr == -1:\n",
    "        return 'INDETERMINATE'\n",
    "    \n",
    "maf_cohort_wgd_tp53['tp53_timing_wgd'] = maf_cohort_wgd_tp53.apply(get_tp53_timing_wgd, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 176,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tp53_timing_wgd</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>BEFORE</th>\n",
       "      <td>5974</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>AFTER</th>\n",
       "      <td>1047</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INDETERMINATE</th>\n",
       "      <td>904</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 count\n",
       "tp53_timing_wgd       \n",
       "BEFORE            5974\n",
       "AFTER             1047\n",
       "INDETERMINATE      904"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of TP53 mutations that had arisen BEFORE WGD: 5974 - 75.38 %\n",
      "Number of TP53 mutations that had arisen AFTER WGD: 1047 - 13.21 %\n",
      "Number of INDETERMINATE: 904 - 11.41 %\n"
     ]
    }
   ],
   "source": [
    "h = pd.DataFrame(get_groupby(maf_cohort_wgd_tp53,'tp53_timing_wgd', 'count').sort_values(by='count', ascending=False))\n",
    "display(h)\n",
    "\n",
    "\n",
    "print('Number of TP53 mutations that had arisen BEFORE WGD: ' + str(h.iloc[0,0]) + ' - ' + str(round(100*h.iloc[0,0]/h.sum().values[0],2)) + ' %')\n",
    "\n",
    "print('Number of TP53 mutations that had arisen AFTER WGD: ' + str(h.iloc[1,0]) + ' - ' + str(round(100*h.iloc[1,0]/h.sum().values[0],2)) + ' %')\n",
    "\n",
    "print('Number of INDETERMINATE: ' + str(h.iloc[2,0]) + ' - ' + str(round(100*h.iloc[2,0]/h.sum().values[0],2)) + ' %')\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We have a proportion of 73.2%, which slightly less than Bielsky's paper published in Nature on WGD in prognosis. They have a number of 97.3% on 1 174 mutations. \n",
    "\n",
    "BUT they excluded from the cohort all ambiguous cases, and they used a simplier way to assess the timing on mutations. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tp53_res</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>199.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>2.125703</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>0.797683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>-0.739883</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>1.928831</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>2.148834</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>2.523609</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>3.883283</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         tp53_res\n",
       "count  199.000000\n",
       "mean     2.125703\n",
       "std      0.797683\n",
       "min     -0.739883\n",
       "25%      1.928831\n",
       "50%      2.148834\n",
       "75%      2.523609\n",
       "max      3.883283"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# In this cell we check the tp53_res distribution for a given tp53_cn_state\n",
    "cn_state = 'TETRAPLOID'\n",
    "samples = list(set(gene_level_tp53_wgd[gene_level_tp53_wgd['cn_state'] == cn_state]['sample'].str[:17]))\n",
    "pd.DataFrame(maf_cohort_wgd_tp53[maf_cohort_wgd_tp53['Tumor_Sample_Barcode'].isin(samples)].tp53_res.describe())"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Allelic State before WGD\n",
    "Thanks to the tp53 mutation timing, we can compute pre-WGD allelic state of our samples.\n",
    "\n",
    "For that, we need to separate 1mut samples and multi-mut samples: indeed the allelic state highly depends on the number of mutaions"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1mut Samples"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_allelic_state(x):\n",
    "    tumor = x.tumor_sample\n",
    "    cn_state = gene_level_tp53_wgd[gene_level_tp53_wgd['sample'].str[:17] == tumor]['cn_state'].values[0]\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort_wgd_tp53[maf_cohort_wgd_tp53['Tumor_Sample_Barcode'] == tumor]\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    \n",
    "    # Samples with only one tp53 mutation\n",
    "    if nb_tp53muts == 1:\n",
    "        tp53_res = maf_muts.tp53_res.values[0]\n",
    "        timing = maf_muts.tp53_timing_wgd.values[0]\n",
    "        \n",
    "        if timing == 'INDETERMINATE':\n",
    "            return 'INDETERMINATE'\n",
    "        \n",
    "        if timing == 'BEFORE':\n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'bi'\n",
    "\n",
    "            elif cn_state == 'CNLOH BEFORE & LOSS':\n",
    "                thr = 0.5\n",
    "                if tp53_res < thr:\n",
    "                    return 'bi'\n",
    "                elif tp53_res >= thr:\n",
    "                    return 'mono'\n",
    "\n",
    "            elif cn_state == 'CNLOH BEFORE':\n",
    "                thr = 1\n",
    "                if tp53_res < thr:\n",
    "                    return 'bi'\n",
    "                elif tp53_res >= thr:\n",
    "                    return 'mono'\n",
    "\n",
    "            elif cn_state == 'LOSS AFTER': \n",
    "                return 'mono'\n",
    "            \n",
    "            elif cn_state == 'DOUBLE LOSS AFTER':\n",
    "                return 'mono'\n",
    "            \n",
    "            elif cn_state == 'TETRAPLOID':\n",
    "                return 'mono'\n",
    "            \n",
    "        if timing == 'AFTER':\n",
    "            if cn_state in ['CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS AFTER', 'DOUBLE LOSS AFTER', 'TETRAPLOID']:\n",
    "                return '2WT'\n",
    "            elif cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            \n",
    "            \n",
    "            \n",
    "    # Samples with 2 tp53 mutations\n",
    "    elif nb_tp53muts == 2:\n",
    "        tp53_res_1 = maf_muts.tp53_res.values[0]\n",
    "        tp53_res_2 = maf_muts.tp53_res.values[1]\n",
    "        timing_1 = maf_muts.tp53_timing_wgd.values[0]\n",
    "        timing_2 = maf_muts.tp53_timing_wgd.values[1]\n",
    "        \n",
    "        if timing_1 == 'INDETERMINATE' or timing_2 == 'INDETERMINATE':\n",
    "            return 'INDETERMINATE'\n",
    "        \n",
    "        elif timing_1 == 'AFTER' and timing_2 == 'AFTER':\n",
    "            # Same case as 1 mut AFTER\n",
    "            if cn_state in ['CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS AFTER', 'DOUBLE LOSS AFTER', 'TETRAPLOID']:\n",
    "                return '2WT'\n",
    "            elif cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "        \n",
    "        elif (timing_1 == 'AFTER' and timing_2 == 'BEFORE') :\n",
    "            # Similar to 1 mut BEFORE\n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'bi'\n",
    "\n",
    "            elif cn_state == 'CNLOH BEFORE & LOSS':\n",
    "                thr = 0.5\n",
    "                if tp53_res_2 < thr:\n",
    "                    return 'bi'\n",
    "                elif tp53_res_2 >= thr:\n",
    "                    return 'mono'\n",
    "\n",
    "            elif cn_state == 'CNLOH BEFORE':\n",
    "                thr = 1\n",
    "                if tp53_res_2 < thr:\n",
    "                    return 'bi'\n",
    "                elif tp53_res_2 >= thr:\n",
    "                    return 'mono'\n",
    "\n",
    "            elif cn_state == 'LOSS AFTER': \n",
    "                return 'mono'\n",
    "            \n",
    "            elif cn_state == 'DOUBLE LOSS AFTER':\n",
    "                return 'mono'\n",
    "            \n",
    "            elif cn_state == 'TETRAPLOID':\n",
    "                return 'mono'\n",
    "        \n",
    "        elif (timing_2 == 'AFTER' and timing_1 == 'BEFORE'):\n",
    "            # Similar to 1 mut BEFORE\n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'bi'\n",
    "\n",
    "            elif cn_state == 'CNLOH BEFORE & LOSS':\n",
    "                thr = 0.5\n",
    "                if tp53_res_1 < thr:\n",
    "                    return 'bi'\n",
    "                elif tp53_res_1 >= thr:\n",
    "                    return 'mono'\n",
    "\n",
    "            elif cn_state == 'CNLOH BEFORE':\n",
    "                thr = 1\n",
    "                if tp53_res_1 < thr:\n",
    "                    return 'bi'\n",
    "                elif tp53_res_1 >= thr:\n",
    "                    return 'mono'\n",
    "\n",
    "            elif cn_state == 'LOSS AFTER': \n",
    "                return 'mono'\n",
    "            \n",
    "            elif cn_state == 'DOUBLE LOSS AFTER':\n",
    "                return 'mono'\n",
    "            \n",
    "            elif cn_state == 'TETRAPLOID':\n",
    "                return 'mono'\n",
    "        \n",
    "        \n",
    "        elif timing_1 == 'BEFORE' and timing_2 == 'BEFORE':\n",
    "            tp53_ccf_1 = maf_muts.ccf_expected_copies.values[0]\n",
    "            tp53_ccf_2 = maf_muts.ccf_expected_copies.values[1]\n",
    "            \n",
    "            # We assess if the tp53 mutations are in the same cells or not\n",
    "            if (tp53_ccf_1 + tp53_ccf_2 > 1):\n",
    "                if cn_state in ['LOSS BEFORE','CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS AFTER', 'DOUBLE LOSS AFTER', 'TETRAPLOID']:\n",
    "                    return 'bi'\n",
    "                if cn_state == 'LOSS BEFORE & AFTER':\n",
    "                    return 'INDETERMINATE'\n",
    "            \n",
    "    # Samples with 2 tp53 mutations\n",
    "    elif nb_tp53muts == 3:\n",
    "        tp53_res_1 = maf_muts.tp53_res.values[0]\n",
    "        tp53_res_2 = maf_muts.tp53_res.values[1]\n",
    "        tp53_res_3 = maf_muts.tp53_res.values[2]\n",
    "        tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "        \n",
    "        timing_1 = maf_muts.tp53_timing_wgd.values[0]\n",
    "        timing_2 = maf_muts.tp53_timing_wgd.values[1]\n",
    "        timing_3 = maf_muts.tp53_timing_wgd.values[2]\n",
    "        timings = [timing_1, timing_2, timing_3]\n",
    "        BEF = [1 for x in timings if x=='BEFORE']\n",
    "        # Number of mutations happening before wgd\n",
    "        nb_bef = len(BEF)\n",
    "        \n",
    "        if nb_bef == 0 : # Only mutations after\n",
    "            # Same case as 1 mut AFTER\n",
    "            if cn_state in ['CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS AFTER', 'DOUBLE LOSS AFTER', 'TETRAPLOID']:\n",
    "                return '2WT'\n",
    "            elif cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "        \n",
    "        elif nb_bef == 1:\n",
    "            # Similar to 1 mut BEFORE\n",
    "            bef_index = timings.index('BEFORE')\n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'bi'\n",
    "\n",
    "            elif cn_state == 'CNLOH BEFORE & LOSS':\n",
    "                thr = 0.5\n",
    "                if tp53_res[bef_index] < thr:\n",
    "                    return 'bi'\n",
    "                elif tp53_res[bef_index] >= thr:\n",
    "                    return 'mono'\n",
    "\n",
    "            elif cn_state == 'CNLOH BEFORE':\n",
    "                thr = 1\n",
    "                if tp53_res[bef_index] < thr:\n",
    "                    return 'bi'\n",
    "                elif tp53_res[bef_index] >= thr:\n",
    "                    return 'mono'\n",
    "\n",
    "            elif cn_state == 'LOSS AFTER': \n",
    "                return 'mono'\n",
    "            \n",
    "            elif cn_state == 'DOUBLE LOSS AFTER':\n",
    "                return 'mono'\n",
    "            \n",
    "            elif cn_state == 'TETRAPLOID':\n",
    "                return 'mono'\n",
    "            \n",
    "        elif nb_bef == 2:\n",
    "            # 2 mutations\n",
    "            tp53_ccf_1 = maf_muts[maf_muts['tp53_timing_wgd'] == 'BEFORE'].ccf_expected_copies.values[0]\n",
    "            tp53_ccf_2 = maf_muts[maf_muts['tp53_timing_wgd'] == 'BEFORE'].ccf_expected_copies.values[1]\n",
    "            if (tp53_ccf_1 + tp53_ccf_2 > 1):\n",
    "                if cn_state in ['LOSS BEFORE','CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS AFTER', 'DOUBLE LOSS AFTER', 'TETRAPLOID']:\n",
    "                    return 'bi'\n",
    "                if cn_state == 'LOSS BEFORE & AFTER':\n",
    "                    return 'INDETERMINATE'\n",
    "        \n",
    "        elif nb_bef == 3:\n",
    "            tp53_ccf_1 = maf_muts[maf_muts['tp53_timing_wgd'] == 'BEFORE'].ccf_expected_copies.values[0]\n",
    "            tp53_ccf_2 = maf_muts[maf_muts['tp53_timing_wgd'] == 'BEFORE'].ccf_expected_copies.values[1]\n",
    "            tp53_ccf_3 = maf_muts[maf_muts['tp53_timing_wgd'] == 'BEFORE'].ccf_expected_copies.values[2]\n",
    "            ccfs = [tp53_ccf_2, tp53_ccf_1,tp53_ccf_3]\n",
    "            ccfs.sort()\n",
    "            if (ccfs[1] + ccfs[2] > 1):\n",
    "                if cn_state in ['LOSS BEFORE','CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS AFTER', 'DOUBLE LOSS AFTER', 'TETRAPLOID']:\n",
    "                    return 'bi'\n",
    "                if cn_state == 'LOSS BEFORE & AFTER':\n",
    "                    return 'INDETERMINATE'\n",
    "        \n",
    "        else: return 'INDETERMINATE'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "f916968002214a169ae8f9408c05d03b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=1.0, bar_style='info', max=1.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "5c9ac30d44fc42dcba747f8f071ae315",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=0.0, max=7418.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "from tqdm import tqdm,tqdm_notebook\n",
    "tqdm_notebook().pandas()\n",
    "cohort_wgd_tp53['allelic_state_wgd'] = cohort_wgd_tp53.progress_apply(get_allelic_state, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 174,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>allelic_state_wgd</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>bi</th>\n",
       "      <td>5097</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mono</th>\n",
       "      <td>992</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>INDETERMINATE</th>\n",
       "      <td>798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2WT</th>\n",
       "      <td>444</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                   count\n",
       "allelic_state_wgd       \n",
       "bi                  5097\n",
       "mono                 992\n",
       "INDETERMINATE        798\n",
       "2WT                  444"
      ]
     },
     "execution_count": 174,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(cohort_wgd_tp53, 'allelic_state_wgd', 'count').sort_values(by='count', ascending=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[0.4, 0.5, 0.9]"
      ]
     },
     "execution_count": 155,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tp53_ccf_1 = 0.4\n",
    "tp53_ccf_2 = 0.5\n",
    "tp53_ccf_3 = 0.9\n",
    "ccfs = [tp53_ccf_2, tp53_ccf_1,tp53_ccf_3]\n",
    "ccfs.sort()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>dbSNP_RS</th>\n",
       "      <th>dbSNP_Val_Status</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Match_Norm_Seq_Allele1</th>\n",
       "      <th>Match_Norm_Seq_Allele2</th>\n",
       "      <th>Tumor_Validation_Allele1</th>\n",
       "      <th>Tumor_Validation_Allele2</th>\n",
       "      <th>Match_Norm_Validation_Allele1</th>\n",
       "      <th>Match_Norm_Validation_Allele2</th>\n",
       "      <th>Verification_Status</th>\n",
       "      <th>Validation_Status</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>Sequencing_Phase</th>\n",
       "      <th>Sequence_Source</th>\n",
       "      <th>Validation_Method</th>\n",
       "      <th>Score</th>\n",
       "      <th>BAM_File</th>\n",
       "      <th>Sequencer</th>\n",
       "      <th>Tumor_Sample_UUID</th>\n",
       "      <th>Matched_Norm_Sample_UUID</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Exon_Number</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>n_depth</th>\n",
       "      <th>n_ref_count</th>\n",
       "      <th>n_alt_count</th>\n",
       "      <th>all_effects</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>ALLELE_NUM</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND_VEP</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>RefSeq</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>GMAF</th>\n",
       "      <th>AFR_MAF</th>\n",
       "      <th>AMR_MAF</th>\n",
       "      <th>ASN_MAF</th>\n",
       "      <th>EAS_MAF</th>\n",
       "      <th>EUR_MAF</th>\n",
       "      <th>SAS_MAF</th>\n",
       "      <th>AA_MAF</th>\n",
       "      <th>EA_MAF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>PICK</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>TSL</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>MINIMISED</th>\n",
       "      <th>ExAC_AF</th>\n",
       "      <th>ExAC_AF_AFR</th>\n",
       "      <th>ExAC_AF_AMR</th>\n",
       "      <th>ExAC_AF_EAS</th>\n",
       "      <th>ExAC_AF_FIN</th>\n",
       "      <th>ExAC_AF_NFE</th>\n",
       "      <th>ExAC_AF_OTH</th>\n",
       "      <th>ExAC_AF_SAS</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>flanking_bps</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_qual</th>\n",
       "      <th>ExAC_AF_Adj</th>\n",
       "      <th>ExAC_AC_AN_Adj</th>\n",
       "      <th>ExAC_AC_AN</th>\n",
       "      <th>ExAC_AC_AN_AFR</th>\n",
       "      <th>ExAC_AC_AN_AMR</th>\n",
       "      <th>ExAC_AC_AN_EAS</th>\n",
       "      <th>ExAC_AC_AN_FIN</th>\n",
       "      <th>ExAC_AC_AN_NFE</th>\n",
       "      <th>ExAC_AC_AN_OTH</th>\n",
       "      <th>ExAC_AC_AN_SAS</th>\n",
       "      <th>ExAC_FILTER</th>\n",
       "      <th>Caller</th>\n",
       "      <th>is-a-hotspot</th>\n",
       "      <th>is-a-3d-hotspot</th>\n",
       "      <th>mutation_effect</th>\n",
       "      <th>oncogenic</th>\n",
       "      <th>LEVEL_1</th>\n",
       "      <th>LEVEL_2A</th>\n",
       "      <th>LEVEL_2B</th>\n",
       "      <th>LEVEL_3A</th>\n",
       "      <th>LEVEL_3B</th>\n",
       "      <th>LEVEL_4</th>\n",
       "      <th>LEVEL_R1</th>\n",
       "      <th>LEVEL_R2</th>\n",
       "      <th>LEVEL_R3</th>\n",
       "      <th>Highest_level</th>\n",
       "      <th>citations</th>\n",
       "      <th>driver</th>\n",
       "      <th>tcn</th>\n",
       "      <th>lcn</th>\n",
       "      <th>cf</th>\n",
       "      <th>purity</th>\n",
       "      <th>t_var_freq</th>\n",
       "      <th>expected_alt_copies</th>\n",
       "      <th>ccf_Mcopies</th>\n",
       "      <th>ccf_Mcopies_lower</th>\n",
       "      <th>ccf_Mcopies_upper</th>\n",
       "      <th>ccf_Mcopies_prob95</th>\n",
       "      <th>ccf_Mcopies_prob90</th>\n",
       "      <th>ccf_1copy</th>\n",
       "      <th>ccf_1copy_lower</th>\n",
       "      <th>ccf_1copy_upper</th>\n",
       "      <th>ccf_1copy_prob95</th>\n",
       "      <th>ccf_1copy_prob90</th>\n",
       "      <th>ccf_expected_copies</th>\n",
       "      <th>ccf_expected_copies_lower</th>\n",
       "      <th>ccf_expected_copies_upper</th>\n",
       "      <th>ccf_expected_copies_prob95</th>\n",
       "      <th>ccf_expected_copies_prob90</th>\n",
       "      <th>facets_fit</th>\n",
       "      <th>reviewer_set_purity</th>\n",
       "      <th>use_only_purity_run</th>\n",
       "      <th>use_edited_cncf</th>\n",
       "      <th>cncf_file_used</th>\n",
       "      <th>tp53_res</th>\n",
       "      <th>tp53_timing_wgd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577121</td>\n",
       "      <td>7577121</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.817C&gt;T</td>\n",
       "      <td>p.Arg273Cys</td>\n",
       "      <td>p.R273C</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>8/11</td>\n",
       "      <td>756.0</td>\n",
       "      <td>520.0</td>\n",
       "      <td>236.0</td>\n",
       "      <td>584</td>\n",
       "      <td>584</td>\n",
       "      <td>0</td>\n",
       "      <td>TP53,missense_variant,p.Arg273Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg273Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg273Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg273Cys,ENST00000445888,;TP53,missense_variant,p.Arg273Cys,ENST00000359597,;TP53,missense_variant,p.Arg141Cys,ENST00000509690,;TP53,intron_variant,,ENST00000413465,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,downstream_gene_variant,,ENST00000514944,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,downstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,downstream_gene_variant,,ENST00000505014,;</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1007/2579</td>\n",
       "      <td>817/1182</td>\n",
       "      <td>273/393</td>\n",
       "      <td>R/C</td>\n",
       "      <td>Cgt/Tgt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.998)</td>\n",
       "      <td>8/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23863845;11900253;23612969;15037740</td>\n",
       "      <td>True</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.306121</td>\n",
       "      <td>0.391316</td>\n",
       "      <td>0.312169</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.798</td>\n",
       "      <td>0.747</td>\n",
       "      <td>0.850</td>\n",
       "      <td>3.269715e-04</td>\n",
       "      <td>0.010807</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.993</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.997653</td>\n",
       "      <td>0.999997</td>\n",
       "      <td>0.798</td>\n",
       "      <td>0.747</td>\n",
       "      <td>0.850</td>\n",
       "      <td>0.000327</td>\n",
       "      <td>0.010807</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00369/P-0036909-T01-IM6_P-0036909-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00369/P-0036909-T01-IM6_P-0036909-N01-IM6//default/P-0036909-T01-IM6_P-0036909-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>0.404515</td>\n",
       "      <td>BEFORE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>106</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578406</td>\n",
       "      <td>7578406</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0016553-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.524G&gt;A</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>p.R175H</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>5/11</td>\n",
       "      <td>863.0</td>\n",
       "      <td>159.0</td>\n",
       "      <td>704.0</td>\n",
       "      <td>714</td>\n",
       "      <td>712</td>\n",
       "      <td>2</td>\n",
       "      <td>TP53,missense_variant,p.Arg175His,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg175His,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg175His,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg175His,ENST00000445888,;TP53,missense_variant,p.Arg175His,ENST00000359597,;TP53,missense_variant,p.Arg175His,ENST00000413465,;TP53,missense_variant,p.Arg43His,ENST00000509690,;TP53,missense_variant,p.Arg82His,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,EN...</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>714/2579</td>\n",
       "      <td>524/1182</td>\n",
       "      <td>175/393</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGc/cAc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>tolerated(0.11)</td>\n",
       "      <td>benign(0.308)</td>\n",
       "      <td>5/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25584008;15781620;22114072;10713666;21445056</td>\n",
       "      <td>True</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.630313</td>\n",
       "      <td>0.630313</td>\n",
       "      <td>0.815759</td>\n",
       "      <td>6.0</td>\n",
       "      <td>0.975</td>\n",
       "      <td>0.956</td>\n",
       "      <td>0.994</td>\n",
       "      <td>9.367284e-01</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.975</td>\n",
       "      <td>0.956</td>\n",
       "      <td>0.994</td>\n",
       "      <td>0.936728</td>\n",
       "      <td>0.999995</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00165/P-0016553-T01-IM6_P-0016553-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00165/P-0016553-T01-IM6_P-0016553-N01-IM6//default/P-0016553-T01-IM6_P-0016553-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>0.148539</td>\n",
       "      <td>INDETERMINATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578442</td>\n",
       "      <td>7578442</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.488A&gt;G</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>p.Y163C</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>5/11</td>\n",
       "      <td>426.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>360.0</td>\n",
       "      <td>744</td>\n",
       "      <td>744</td>\n",
       "      <td>0</td>\n",
       "      <td>TP53,missense_variant,p.Tyr163Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Tyr163Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Tyr163Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Tyr163Cys,ENST00000445888,;TP53,missense_variant,p.Tyr163Cys,ENST00000359597,;TP53,missense_variant,p.Tyr163Cys,ENST00000413465,;TP53,missense_variant,p.Tyr31Cys,ENST00000509690,;TP53,missense_variant,p.Tyr163Cys,ENST00000508793,;TP53,missense_variant,p.Tyr70Cys,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,upstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,ENST0000050...</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>678/2579</td>\n",
       "      <td>488/1182</td>\n",
       "      <td>163/393</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>tAc/tGc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.999)</td>\n",
       "      <td>5/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GTA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25584008;8023157;11900253;15077194;23246812;15037740</td>\n",
       "      <td>True</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.835504</td>\n",
       "      <td>0.865628</td>\n",
       "      <td>0.845070</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.908</td>\n",
       "      <td>0.955</td>\n",
       "      <td>1.667143e-01</td>\n",
       "      <td>0.941853</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.997</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.908</td>\n",
       "      <td>0.955</td>\n",
       "      <td>0.166714</td>\n",
       "      <td>0.941853</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T01-IM6_P-0023546-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T01-IM6_P-0023546-N01-IM6//default/P-0023546-T01-IM6_P-0023546-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>0.202426</td>\n",
       "      <td>BEFORE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578442</td>\n",
       "      <td>7578442</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.488A&gt;G</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>p.Y163C</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>5/11</td>\n",
       "      <td>490.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>312.0</td>\n",
       "      <td>744</td>\n",
       "      <td>744</td>\n",
       "      <td>0</td>\n",
       "      <td>TP53,missense_variant,p.Tyr163Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Tyr163Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Tyr163Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Tyr163Cys,ENST00000445888,;TP53,missense_variant,p.Tyr163Cys,ENST00000359597,;TP53,missense_variant,p.Tyr163Cys,ENST00000413465,;TP53,missense_variant,p.Tyr31Cys,ENST00000509690,;TP53,missense_variant,p.Tyr163Cys,ENST00000508793,;TP53,missense_variant,p.Tyr70Cys,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,upstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,ENST0000050...</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>678/2579</td>\n",
       "      <td>488/1182</td>\n",
       "      <td>163/393</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>tAc/tGc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.999)</td>\n",
       "      <td>5/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GTA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25584008;8023157;11900253;15077194;23246812;15037740</td>\n",
       "      <td>True</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>0.636735</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.997</td>\n",
       "      <td>1.000</td>\n",
       "      <td>9.999955e-01</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.997</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.972</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.913083</td>\n",
       "      <td>0.998538</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default/P-0023546-T02-IM6_P-0023546-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>-2.069805</td>\n",
       "      <td>BEFORE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>162</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578406</td>\n",
       "      <td>7578406</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.524G&gt;A</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>p.R175H</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>5/11</td>\n",
       "      <td>801.0</td>\n",
       "      <td>540.0</td>\n",
       "      <td>261.0</td>\n",
       "      <td>714</td>\n",
       "      <td>712</td>\n",
       "      <td>2</td>\n",
       "      <td>TP53,missense_variant,p.Arg175His,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg175His,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg175His,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg175His,ENST00000445888,;TP53,missense_variant,p.Arg175His,ENST00000359597,;TP53,missense_variant,p.Arg175His,ENST00000413465,;TP53,missense_variant,p.Arg43His,ENST00000509690,;TP53,missense_variant,p.Arg82His,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,EN...</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>714/2579</td>\n",
       "      <td>524/1182</td>\n",
       "      <td>175/393</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGc/cAc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>tolerated(0.11)</td>\n",
       "      <td>benign(0.308)</td>\n",
       "      <td>5/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25584008;15781620;22114072;10713666;21445056</td>\n",
       "      <td>True</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.257400</td>\n",
       "      <td>0.351778</td>\n",
       "      <td>0.325843</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.726</td>\n",
       "      <td>0.683</td>\n",
       "      <td>0.771</td>\n",
       "      <td>3.198825e-09</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.995</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.999866</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.908549</td>\n",
       "      <td>0.996435</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00188/P-0018837-T01-IM6_P-0018837-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00188/P-0018837-T01-IM6_P-0018837-N01-IM6//default/P-0018837-T01-IM6_P-0018837-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>0.821608</td>\n",
       "      <td>BEFORE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339227</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577121</td>\n",
       "      <td>7577121</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0022777-T04-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.817C&gt;T</td>\n",
       "      <td>p.Arg273Cys</td>\n",
       "      <td>p.R273C</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>8/11</td>\n",
       "      <td>609.0</td>\n",
       "      <td>427.0</td>\n",
       "      <td>182.0</td>\n",
       "      <td>617</td>\n",
       "      <td>615</td>\n",
       "      <td>2</td>\n",
       "      <td>TP53,missense_variant,p.Arg273Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg273Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg273Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg273Cys,ENST00000445888,;TP53,missense_variant,p.Arg273Cys,ENST00000359597,;TP53,missense_variant,p.Arg141Cys,ENST00000509690,;TP53,intron_variant,,ENST00000413465,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,downstream_gene_variant,,ENST00000514944,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,downstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,downstream_gene_variant,,ENST00000505014,;</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1007/2579</td>\n",
       "      <td>817/1182</td>\n",
       "      <td>273/393</td>\n",
       "      <td>R/C</td>\n",
       "      <td>Cgt/Tgt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.998)</td>\n",
       "      <td>8/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>23863845;11900253;23612969;15037740</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.412643</td>\n",
       "      <td>0.412643</td>\n",
       "      <td>0.298851</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.000</td>\n",
       "      <td>8.899879e-01</td>\n",
       "      <td>0.993088</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.889988</td>\n",
       "      <td>0.993088</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.889988</td>\n",
       "      <td>0.993088</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00227/P-0022777-T04-IM6_P-0022777-N02-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00227/P-0022777-T04-IM6_P-0022777-N02-IM6//default/P-0022777-T04-IM6_P-0022777-N02-IM6_hisens.cncf.txt</td>\n",
       "      <td>-0.149619</td>\n",
       "      <td>INDETERMINATE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339266</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578406</td>\n",
       "      <td>7578406</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050750-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.524G&gt;A</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>p.R175H</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>5/11</td>\n",
       "      <td>589.0</td>\n",
       "      <td>421.0</td>\n",
       "      <td>168.0</td>\n",
       "      <td>566</td>\n",
       "      <td>566</td>\n",
       "      <td>0</td>\n",
       "      <td>TP53,missense_variant,p.Arg175His,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg175His,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg175His,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg175His,ENST00000445888,;TP53,missense_variant,p.Arg175His,ENST00000359597,;TP53,missense_variant,p.Arg175His,ENST00000413465,;TP53,missense_variant,p.Arg43His,ENST00000509690,;TP53,missense_variant,p.Arg82His,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,EN...</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>714/2579</td>\n",
       "      <td>524/1182</td>\n",
       "      <td>175/393</td>\n",
       "      <td>R/H</td>\n",
       "      <td>cGc/cAc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>tolerated(0.11)</td>\n",
       "      <td>benign(0.308)</td>\n",
       "      <td>5/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25584008;15781620;22114072;10713666;21445056</td>\n",
       "      <td>True</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.286332</td>\n",
       "      <td>0.301523</td>\n",
       "      <td>0.285229</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0.874</td>\n",
       "      <td>1.000</td>\n",
       "      <td>3.611258e-01</td>\n",
       "      <td>0.731527</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.992</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.993854</td>\n",
       "      <td>0.999975</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0.874</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.361126</td>\n",
       "      <td>0.731527</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050750-T01-IM6_P-0050750-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050750-T01-IM6_P-0050750-N01-IM6//default/P-0050750-T01-IM6_P-0050750-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>0.108078</td>\n",
       "      <td>BEFORE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339363</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578413</td>\n",
       "      <td>7578413</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0048760-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.517G&gt;A</td>\n",
       "      <td>p.Val173Met</td>\n",
       "      <td>p.V173M</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>5/11</td>\n",
       "      <td>706.0</td>\n",
       "      <td>229.0</td>\n",
       "      <td>477.0</td>\n",
       "      <td>726</td>\n",
       "      <td>725</td>\n",
       "      <td>1</td>\n",
       "      <td>TP53,missense_variant,p.Val173Met,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Val173Met,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Val173Met,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Val173Met,ENST00000445888,;TP53,missense_variant,p.Val173Met,ENST00000359597,;TP53,missense_variant,p.Val173Met,ENST00000413465,;TP53,missense_variant,p.Val41Met,ENST00000509690,;TP53,missense_variant,p.Val80Met,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,EN...</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>707/2579</td>\n",
       "      <td>517/1182</td>\n",
       "      <td>173/393</td>\n",
       "      <td>V/M</td>\n",
       "      <td>Gtg/Atg</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>deleterious(0.02)</td>\n",
       "      <td>probably_damaging(1)</td>\n",
       "      <td>5/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ACA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15037740</td>\n",
       "      <td>True</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.598599</td>\n",
       "      <td>0.614603</td>\n",
       "      <td>0.675637</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.958</td>\n",
       "      <td>0.928</td>\n",
       "      <td>0.988</td>\n",
       "      <td>6.114443e-01</td>\n",
       "      <td>0.988322</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.998</td>\n",
       "      <td>1.0</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>0.958</td>\n",
       "      <td>0.928</td>\n",
       "      <td>0.988</td>\n",
       "      <td>0.611444</td>\n",
       "      <td>0.988322</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00487/P-0048760-T01-IM6_P-0048760-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00487/P-0048760-T01-IM6_P-0048760-N01-IM6//default/P-0048760-T01-IM6_P-0048760-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>0.125750</td>\n",
       "      <td>BEFORE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339387</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577580</td>\n",
       "      <td>7577580</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0050744-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.701A&gt;G</td>\n",
       "      <td>p.Tyr234Cys</td>\n",
       "      <td>p.Y234C</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>7/11</td>\n",
       "      <td>276.0</td>\n",
       "      <td>210.0</td>\n",
       "      <td>66.0</td>\n",
       "      <td>511</td>\n",
       "      <td>511</td>\n",
       "      <td>0</td>\n",
       "      <td>TP53,missense_variant,p.Tyr234Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Tyr234Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Tyr234Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Tyr234Cys,ENST00000445888,;TP53,missense_variant,p.Tyr234Cys,ENST00000359597,;TP53,missense_variant,p.Tyr234Cys,ENST00000413465,;TP53,missense_variant,p.Tyr102Cys,ENST00000509690,;TP53,missense_variant,p.Tyr141Cys,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,downstream_gene_variant,,ENST0000050...</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>891/2579</td>\n",
       "      <td>701/1182</td>\n",
       "      <td>234/393</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>tAc/tGc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.992)</td>\n",
       "      <td>7/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GTA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>26748848;16861262;10229196</td>\n",
       "      <td>True</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.205450</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>0.239130</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.797</td>\n",
       "      <td>0.699</td>\n",
       "      <td>0.901</td>\n",
       "      <td>3.472834e-02</td>\n",
       "      <td>0.119830</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.976</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.814993</td>\n",
       "      <td>0.971027</td>\n",
       "      <td>0.797</td>\n",
       "      <td>0.699</td>\n",
       "      <td>0.901</td>\n",
       "      <td>0.034728</td>\n",
       "      <td>0.119830</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050744-T01-IM6_P-0050744-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050744-T01-IM6_P-0050744-N01-IM6//default/P-0050744-T01-IM6_P-0050744-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>0.405797</td>\n",
       "      <td>BEFORE</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339408</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7577094</td>\n",
       "      <td>7577094</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0049537-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.844C&gt;T</td>\n",
       "      <td>p.Arg282Trp</td>\n",
       "      <td>p.R282W</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>8/11</td>\n",
       "      <td>553.0</td>\n",
       "      <td>285.0</td>\n",
       "      <td>268.0</td>\n",
       "      <td>552</td>\n",
       "      <td>551</td>\n",
       "      <td>1</td>\n",
       "      <td>TP53,missense_variant,p.Arg282Trp,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg282Trp,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg282Trp,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg282Trp,ENST00000445888,;TP53,missense_variant,p.Arg282Trp,ENST00000359597,;TP53,missense_variant,p.Arg150Trp,ENST00000509690,;TP53,intron_variant,,ENST00000413465,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,downstream_gene_variant,,ENST00000514944,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,downstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,downstream_gene_variant,,ENST00000505014,;</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1034/2579</td>\n",
       "      <td>844/1182</td>\n",
       "      <td>282/393</td>\n",
       "      <td>R/W</td>\n",
       "      <td>Cgg/Tgg</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.997)</td>\n",
       "      <td>8/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>21445056</td>\n",
       "      <td>True</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.632827</td>\n",
       "      <td>0.663312</td>\n",
       "      <td>0.484629</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.977</td>\n",
       "      <td>0.926</td>\n",
       "      <td>1.000</td>\n",
       "      <td>6.310788e-01</td>\n",
       "      <td>0.950605</td>\n",
       "      <td>0.977</td>\n",
       "      <td>0.926</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.631079</td>\n",
       "      <td>0.950605</td>\n",
       "      <td>0.977</td>\n",
       "      <td>0.926</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.631079</td>\n",
       "      <td>0.950605</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00495/P-0049537-T01-IM6_P-0049537-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00495/P-0049537-T01-IM6_P-0049537-N01-IM6//default/P-0049537-T01-IM6_P-0049537-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>0.023388</td>\n",
       "      <td>INDETERMINATE</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>7925 rows Ã— 168 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "       Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  \\\n",
       "91            TP53            7157  MSKCC     GRCh37         17   \n",
       "106           TP53            7157  MSKCC     GRCh37         17   \n",
       "110           TP53            7157  MSKCC     GRCh37         17   \n",
       "113           TP53            7157  MSKCC     GRCh37         17   \n",
       "162           TP53            7157  MSKCC     GRCh37         17   \n",
       "...            ...             ...    ...        ...        ...   \n",
       "339227        TP53            7157  MSKCC     GRCh37         17   \n",
       "339266        TP53            7157  MSKCC     GRCh37         17   \n",
       "339363        TP53            7157  MSKCC     GRCh37         17   \n",
       "339387        TP53            7157  MSKCC     GRCh37         17   \n",
       "339408        TP53            7157  MSKCC     GRCh37         17   \n",
       "\n",
       "        Start_Position  End_Position Strand Variant_Classification  \\\n",
       "91             7577121       7577121      +      Missense_Mutation   \n",
       "106            7578406       7578406      +      Missense_Mutation   \n",
       "110            7578442       7578442      +      Missense_Mutation   \n",
       "113            7578442       7578442      +      Missense_Mutation   \n",
       "162            7578406       7578406      +      Missense_Mutation   \n",
       "...                ...           ...    ...                    ...   \n",
       "339227         7577121       7577121      +      Missense_Mutation   \n",
       "339266         7578406       7578406      +      Missense_Mutation   \n",
       "339363         7578413       7578413      +      Missense_Mutation   \n",
       "339387         7577580       7577580      +      Missense_Mutation   \n",
       "339408         7577094       7577094      +      Missense_Mutation   \n",
       "\n",
       "       Variant_Type Reference_Allele Tumor_Seq_Allele1 Tumor_Seq_Allele2  \\\n",
       "91              SNP                G                 G                 A   \n",
       "106             SNP                C                 C                 T   \n",
       "110             SNP                T                 T                 C   \n",
       "113             SNP                T                 T                 C   \n",
       "162             SNP                C                 C                 T   \n",
       "...             ...              ...               ...               ...   \n",
       "339227          SNP                G                 G                 A   \n",
       "339266          SNP                C                 C                 T   \n",
       "339363          SNP                C                 C                 T   \n",
       "339387          SNP                T                 T                 C   \n",
       "339408          SNP                G                 G                 A   \n",
       "\n",
       "       dbSNP_RS  dbSNP_Val_Status Tumor_Sample_Barcode  \\\n",
       "91        novel               NaN    P-0036909-T01-IM6   \n",
       "106       novel               NaN    P-0016553-T01-IM6   \n",
       "110       novel               NaN    P-0023546-T01-IM6   \n",
       "113       novel               NaN    P-0023546-T02-IM6   \n",
       "162       novel               NaN    P-0018837-T01-IM6   \n",
       "...         ...               ...                  ...   \n",
       "339227    novel               NaN    P-0022777-T04-IM6   \n",
       "339266    novel               NaN    P-0050750-T01-IM6   \n",
       "339363    novel               NaN    P-0048760-T01-IM6   \n",
       "339387    novel               NaN    P-0050744-T01-IM6   \n",
       "339408    novel               NaN    P-0049537-T01-IM6   \n",
       "\n",
       "        Matched_Norm_Sample_Barcode Match_Norm_Seq_Allele1  \\\n",
       "91                              NaN                    NaN   \n",
       "106                             NaN                    NaN   \n",
       "110                             NaN                    NaN   \n",
       "113                             NaN                    NaN   \n",
       "162                             NaN                    NaN   \n",
       "...                             ...                    ...   \n",
       "339227                          NaN                    NaN   \n",
       "339266                          NaN                    NaN   \n",
       "339363                          NaN                    NaN   \n",
       "339387                          NaN                    NaN   \n",
       "339408                          NaN                    NaN   \n",
       "\n",
       "       Match_Norm_Seq_Allele2  Tumor_Validation_Allele1  \\\n",
       "91                        NaN                       NaN   \n",
       "106                       NaN                       NaN   \n",
       "110                       NaN                       NaN   \n",
       "113                       NaN                       NaN   \n",
       "162                       NaN                       NaN   \n",
       "...                       ...                       ...   \n",
       "339227                    NaN                       NaN   \n",
       "339266                    NaN                       NaN   \n",
       "339363                    NaN                       NaN   \n",
       "339387                    NaN                       NaN   \n",
       "339408                    NaN                       NaN   \n",
       "\n",
       "        Tumor_Validation_Allele2  Match_Norm_Validation_Allele1  \\\n",
       "91                           NaN                            NaN   \n",
       "106                          NaN                            NaN   \n",
       "110                          NaN                            NaN   \n",
       "113                          NaN                            NaN   \n",
       "162                          NaN                            NaN   \n",
       "...                          ...                            ...   \n",
       "339227                       NaN                            NaN   \n",
       "339266                       NaN                            NaN   \n",
       "339363                       NaN                            NaN   \n",
       "339387                       NaN                            NaN   \n",
       "339408                       NaN                            NaN   \n",
       "\n",
       "        Match_Norm_Validation_Allele2  Verification_Status Validation_Status  \\\n",
       "91                                NaN                  NaN           Unknown   \n",
       "106                               NaN                  NaN           Unknown   \n",
       "110                               NaN                  NaN           Unknown   \n",
       "113                               NaN                  NaN           Unknown   \n",
       "162                               NaN                  NaN           Unknown   \n",
       "...                               ...                  ...               ...   \n",
       "339227                            NaN                  NaN           Unknown   \n",
       "339266                            NaN                  NaN           Unknown   \n",
       "339363                            NaN                  NaN           Unknown   \n",
       "339387                            NaN                  NaN           Unknown   \n",
       "339408                            NaN                  NaN           Unknown   \n",
       "\n",
       "       Mutation_Status  Sequencing_Phase  Sequence_Source  Validation_Method  \\\n",
       "91             SOMATIC               NaN              NaN                NaN   \n",
       "106            SOMATIC               NaN              NaN                NaN   \n",
       "110            SOMATIC               NaN              NaN                NaN   \n",
       "113            SOMATIC               NaN              NaN                NaN   \n",
       "162            SOMATIC               NaN              NaN                NaN   \n",
       "...                ...               ...              ...                ...   \n",
       "339227         SOMATIC               NaN              NaN                NaN   \n",
       "339266         SOMATIC               NaN              NaN                NaN   \n",
       "339363         SOMATIC               NaN              NaN                NaN   \n",
       "339387         SOMATIC               NaN              NaN                NaN   \n",
       "339408         SOMATIC               NaN              NaN                NaN   \n",
       "\n",
       "             Score  BAM_File  Sequencer  Tumor_Sample_UUID  \\\n",
       "91      MSK-IMPACT       NaN        NaN                NaN   \n",
       "106     MSK-IMPACT       NaN        NaN                NaN   \n",
       "110     MSK-IMPACT       NaN        NaN                NaN   \n",
       "113     MSK-IMPACT       NaN        NaN                NaN   \n",
       "162     MSK-IMPACT       NaN        NaN                NaN   \n",
       "...            ...       ...        ...                ...   \n",
       "339227  MSK-IMPACT       NaN        NaN                NaN   \n",
       "339266  MSK-IMPACT       NaN        NaN                NaN   \n",
       "339363  MSK-IMPACT       NaN        NaN                NaN   \n",
       "339387  MSK-IMPACT       NaN        NaN                NaN   \n",
       "339408  MSK-IMPACT       NaN        NaN                NaN   \n",
       "\n",
       "        Matched_Norm_Sample_UUID     HGVSc        HGVSp HGVSp_Short  \\\n",
       "91                           NaN  c.817C>T  p.Arg273Cys     p.R273C   \n",
       "106                          NaN  c.524G>A  p.Arg175His     p.R175H   \n",
       "110                          NaN  c.488A>G  p.Tyr163Cys     p.Y163C   \n",
       "113                          NaN  c.488A>G  p.Tyr163Cys     p.Y163C   \n",
       "162                          NaN  c.524G>A  p.Arg175His     p.R175H   \n",
       "...                          ...       ...          ...         ...   \n",
       "339227                       NaN  c.817C>T  p.Arg273Cys     p.R273C   \n",
       "339266                       NaN  c.524G>A  p.Arg175His     p.R175H   \n",
       "339363                       NaN  c.517G>A  p.Val173Met     p.V173M   \n",
       "339387                       NaN  c.701A>G  p.Tyr234Cys     p.Y234C   \n",
       "339408                       NaN  c.844C>T  p.Arg282Trp     p.R282W   \n",
       "\n",
       "          Transcript_ID Exon_Number  t_depth  t_ref_count  t_alt_count  \\\n",
       "91      ENST00000269305        8/11    756.0        520.0        236.0   \n",
       "106     ENST00000269305        5/11    863.0        159.0        704.0   \n",
       "110     ENST00000269305        5/11    426.0         66.0        360.0   \n",
       "113     ENST00000269305        5/11    490.0        178.0        312.0   \n",
       "162     ENST00000269305        5/11    801.0        540.0        261.0   \n",
       "...                 ...         ...      ...          ...          ...   \n",
       "339227  ENST00000269305        8/11    609.0        427.0        182.0   \n",
       "339266  ENST00000269305        5/11    589.0        421.0        168.0   \n",
       "339363  ENST00000269305        5/11    706.0        229.0        477.0   \n",
       "339387  ENST00000269305        7/11    276.0        210.0         66.0   \n",
       "339408  ENST00000269305        8/11    553.0        285.0        268.0   \n",
       "\n",
       "        n_depth n_ref_count n_alt_count  \\\n",
       "91          584         584           0   \n",
       "106         714         712           2   \n",
       "110         744         744           0   \n",
       "113         744         744           0   \n",
       "162         714         712           2   \n",
       "...         ...         ...         ...   \n",
       "339227      617         615           2   \n",
       "339266      566         566           0   \n",
       "339363      726         725           1   \n",
       "339387      511         511           0   \n",
       "339408      552         551           1   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    all_effects  \\\n",
       "91                               TP53,missense_variant,p.Arg273Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg273Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg273Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg273Cys,ENST00000445888,;TP53,missense_variant,p.Arg273Cys,ENST00000359597,;TP53,missense_variant,p.Arg141Cys,ENST00000509690,;TP53,intron_variant,,ENST00000413465,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,downstream_gene_variant,,ENST00000514944,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,downstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,downstream_gene_variant,,ENST00000505014,;   \n",
       "106     TP53,missense_variant,p.Arg175His,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg175His,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg175His,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg175His,ENST00000445888,;TP53,missense_variant,p.Arg175His,ENST00000359597,;TP53,missense_variant,p.Arg175His,ENST00000413465,;TP53,missense_variant,p.Arg43His,ENST00000509690,;TP53,missense_variant,p.Arg82His,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,EN...   \n",
       "110     TP53,missense_variant,p.Tyr163Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Tyr163Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Tyr163Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Tyr163Cys,ENST00000445888,;TP53,missense_variant,p.Tyr163Cys,ENST00000359597,;TP53,missense_variant,p.Tyr163Cys,ENST00000413465,;TP53,missense_variant,p.Tyr31Cys,ENST00000509690,;TP53,missense_variant,p.Tyr163Cys,ENST00000508793,;TP53,missense_variant,p.Tyr70Cys,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,upstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,ENST0000050...   \n",
       "113     TP53,missense_variant,p.Tyr163Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Tyr163Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Tyr163Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Tyr163Cys,ENST00000445888,;TP53,missense_variant,p.Tyr163Cys,ENST00000359597,;TP53,missense_variant,p.Tyr163Cys,ENST00000413465,;TP53,missense_variant,p.Tyr31Cys,ENST00000509690,;TP53,missense_variant,p.Tyr163Cys,ENST00000508793,;TP53,missense_variant,p.Tyr70Cys,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,upstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,ENST0000050...   \n",
       "162     TP53,missense_variant,p.Arg175His,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg175His,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg175His,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg175His,ENST00000445888,;TP53,missense_variant,p.Arg175His,ENST00000359597,;TP53,missense_variant,p.Arg175His,ENST00000413465,;TP53,missense_variant,p.Arg43His,ENST00000509690,;TP53,missense_variant,p.Arg82His,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,EN...   \n",
       "...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         ...   \n",
       "339227                           TP53,missense_variant,p.Arg273Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg273Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg273Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg273Cys,ENST00000445888,;TP53,missense_variant,p.Arg273Cys,ENST00000359597,;TP53,missense_variant,p.Arg141Cys,ENST00000509690,;TP53,intron_variant,,ENST00000413465,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,downstream_gene_variant,,ENST00000514944,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,downstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,downstream_gene_variant,,ENST00000505014,;   \n",
       "339266  TP53,missense_variant,p.Arg175His,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg175His,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg175His,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg175His,ENST00000445888,;TP53,missense_variant,p.Arg175His,ENST00000359597,;TP53,missense_variant,p.Arg175His,ENST00000413465,;TP53,missense_variant,p.Arg43His,ENST00000509690,;TP53,missense_variant,p.Arg82His,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,EN...   \n",
       "339363  TP53,missense_variant,p.Val173Met,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Val173Met,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Val173Met,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Val173Met,ENST00000445888,;TP53,missense_variant,p.Val173Met,ENST00000359597,;TP53,missense_variant,p.Val173Met,ENST00000413465,;TP53,missense_variant,p.Val41Met,ENST00000509690,;TP53,missense_variant,p.Val80Met,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,EN...   \n",
       "339387  TP53,missense_variant,p.Tyr234Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Tyr234Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Tyr234Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Tyr234Cys,ENST00000445888,;TP53,missense_variant,p.Tyr234Cys,ENST00000359597,;TP53,missense_variant,p.Tyr234Cys,ENST00000413465,;TP53,missense_variant,p.Tyr102Cys,ENST00000509690,;TP53,missense_variant,p.Tyr141Cys,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,non_coding_transcript_exon_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,downstream_gene_variant,,ENST0000050...   \n",
       "339408                           TP53,missense_variant,p.Arg282Trp,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Arg282Trp,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Arg282Trp,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Arg282Trp,ENST00000445888,;TP53,missense_variant,p.Arg282Trp,ENST00000359597,;TP53,missense_variant,p.Arg150Trp,ENST00000509690,;TP53,intron_variant,,ENST00000413465,;TP53,downstream_gene_variant,,ENST00000508793,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,downstream_gene_variant,,ENST00000514944,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,downstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,downstream_gene_variant,,ENST00000505014,;   \n",
       "\n",
       "       Allele             Gene          Feature Feature_type  \\\n",
       "91          A  ENSG00000141510  ENST00000269305   Transcript   \n",
       "106         T  ENSG00000141510  ENST00000269305   Transcript   \n",
       "110         C  ENSG00000141510  ENST00000269305   Transcript   \n",
       "113         C  ENSG00000141510  ENST00000269305   Transcript   \n",
       "162         T  ENSG00000141510  ENST00000269305   Transcript   \n",
       "...       ...              ...              ...          ...   \n",
       "339227      A  ENSG00000141510  ENST00000269305   Transcript   \n",
       "339266      T  ENSG00000141510  ENST00000269305   Transcript   \n",
       "339363      T  ENSG00000141510  ENST00000269305   Transcript   \n",
       "339387      C  ENSG00000141510  ENST00000269305   Transcript   \n",
       "339408      A  ENSG00000141510  ENST00000269305   Transcript   \n",
       "\n",
       "             Consequence cDNA_position CDS_position Protein_position  \\\n",
       "91      missense_variant     1007/2579     817/1182          273/393   \n",
       "106     missense_variant      714/2579     524/1182          175/393   \n",
       "110     missense_variant      678/2579     488/1182          163/393   \n",
       "113     missense_variant      678/2579     488/1182          163/393   \n",
       "162     missense_variant      714/2579     524/1182          175/393   \n",
       "...                  ...           ...          ...              ...   \n",
       "339227  missense_variant     1007/2579     817/1182          273/393   \n",
       "339266  missense_variant      714/2579     524/1182          175/393   \n",
       "339363  missense_variant      707/2579     517/1182          173/393   \n",
       "339387  missense_variant      891/2579     701/1182          234/393   \n",
       "339408  missense_variant     1034/2579     844/1182          282/393   \n",
       "\n",
       "       Amino_acids   Codons  Existing_variation  ALLELE_NUM  DISTANCE  \\\n",
       "91             R/C  Cgt/Tgt                 NaN           1       NaN   \n",
       "106            R/H  cGc/cAc                 NaN           1       NaN   \n",
       "110            Y/C  tAc/tGc                 NaN           1       NaN   \n",
       "113            Y/C  tAc/tGc                 NaN           1       NaN   \n",
       "162            R/H  cGc/cAc                 NaN           1       NaN   \n",
       "...            ...      ...                 ...         ...       ...   \n",
       "339227         R/C  Cgt/Tgt                 NaN           1       NaN   \n",
       "339266         R/H  cGc/cAc                 NaN           1       NaN   \n",
       "339363         V/M  Gtg/Atg                 NaN           1       NaN   \n",
       "339387         Y/C  tAc/tGc                 NaN           1       NaN   \n",
       "339408         R/W  Cgg/Tgg                 NaN           1       NaN   \n",
       "\n",
       "        STRAND_VEP SYMBOL SYMBOL_SOURCE  HGNC_ID         BIOTYPE CANONICAL  \\\n",
       "91              -1   TP53          HGNC  11998.0  protein_coding       YES   \n",
       "106             -1   TP53          HGNC  11998.0  protein_coding       YES   \n",
       "110             -1   TP53          HGNC  11998.0  protein_coding       YES   \n",
       "113             -1   TP53          HGNC  11998.0  protein_coding       YES   \n",
       "162             -1   TP53          HGNC  11998.0  protein_coding       YES   \n",
       "...            ...    ...           ...      ...             ...       ...   \n",
       "339227          -1   TP53          HGNC  11998.0  protein_coding       YES   \n",
       "339266          -1   TP53          HGNC  11998.0  protein_coding       YES   \n",
       "339363          -1   TP53          HGNC  11998.0  protein_coding       YES   \n",
       "339387          -1   TP53          HGNC  11998.0  protein_coding       YES   \n",
       "339408          -1   TP53          HGNC  11998.0  protein_coding       YES   \n",
       "\n",
       "               CCDS             ENSP SWISSPROT  \\\n",
       "91      CCDS11118.1  ENSP00000269305    P04637   \n",
       "106     CCDS11118.1  ENSP00000269305    P04637   \n",
       "110     CCDS11118.1  ENSP00000269305    P04637   \n",
       "113     CCDS11118.1  ENSP00000269305    P04637   \n",
       "162     CCDS11118.1  ENSP00000269305    P04637   \n",
       "...             ...              ...       ...   \n",
       "339227  CCDS11118.1  ENSP00000269305    P04637   \n",
       "339266  CCDS11118.1  ENSP00000269305    P04637   \n",
       "339363  CCDS11118.1  ENSP00000269305    P04637   \n",
       "339387  CCDS11118.1  ENSP00000269305    P04637   \n",
       "339408  CCDS11118.1  ENSP00000269305    P04637   \n",
       "\n",
       "                                                                                                                                                                                TREMBL  \\\n",
       "91      S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "106     S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "110     S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "113     S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "162     S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "...                                                                                                                                                                                ...   \n",
       "339227  S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "339266  S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "339363  S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "339387  S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "339408  S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "\n",
       "              UNIPARC  \\\n",
       "91      UPI000002ED67   \n",
       "106     UPI000002ED67   \n",
       "110     UPI000002ED67   \n",
       "113     UPI000002ED67   \n",
       "162     UPI000002ED67   \n",
       "...               ...   \n",
       "339227  UPI000002ED67   \n",
       "339266  UPI000002ED67   \n",
       "339363  UPI000002ED67   \n",
       "339387  UPI000002ED67   \n",
       "339408  UPI000002ED67   \n",
       "\n",
       "                                                                         RefSeq  \\\n",
       "91      NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "106     NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "110     NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "113     NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "162     NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "...                                                                         ...   \n",
       "339227  NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "339266  NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "339363  NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "339387  NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "339408  NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "\n",
       "                     SIFT                  PolyPhen  EXON INTRON  \\\n",
       "91         deleterious(0)  probably_damaging(0.998)  8/11    NaN   \n",
       "106       tolerated(0.11)             benign(0.308)  5/11    NaN   \n",
       "110        deleterious(0)  probably_damaging(0.999)  5/11    NaN   \n",
       "113        deleterious(0)  probably_damaging(0.999)  5/11    NaN   \n",
       "162       tolerated(0.11)             benign(0.308)  5/11    NaN   \n",
       "...                   ...                       ...   ...    ...   \n",
       "339227     deleterious(0)  probably_damaging(0.998)  8/11    NaN   \n",
       "339266    tolerated(0.11)             benign(0.308)  5/11    NaN   \n",
       "339363  deleterious(0.02)      probably_damaging(1)  5/11    NaN   \n",
       "339387     deleterious(0)  probably_damaging(0.992)  7/11    NaN   \n",
       "339408     deleterious(0)  probably_damaging(0.997)  8/11    NaN   \n",
       "\n",
       "                                                                                                                                        DOMAINS  \\\n",
       "91      Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "106     Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "110     Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "113     Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "162     Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "...                                                                                                                                         ...   \n",
       "339227  Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "339266  Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "339363  Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "339387  Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "339408  Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "\n",
       "        GMAF  AFR_MAF  AMR_MAF  ASN_MAF  EAS_MAF  EUR_MAF  SAS_MAF  AA_MAF  \\\n",
       "91       NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "106      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "110      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "113      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "162      NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "...      ...      ...      ...      ...      ...      ...      ...     ...   \n",
       "339227   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "339266   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "339363   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "339387   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "339408   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "\n",
       "        EA_MAF  CLIN_SIG  SOMATIC  PUBMED  MOTIF_NAME  MOTIF_POS  \\\n",
       "91         NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "106        NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "110        NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "113        NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "162        NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "...        ...       ...      ...     ...         ...        ...   \n",
       "339227     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "339266     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "339363     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "339387     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "339408     NaN       NaN      NaN     NaN         NaN        NaN   \n",
       "\n",
       "        HIGH_INF_POS  MOTIF_SCORE_CHANGE    IMPACT  PICK VARIANT_CLASS  TSL  \\\n",
       "91               NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "106              NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "110              NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "113              NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "162              NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "...              ...                 ...       ...   ...           ...  ...   \n",
       "339227           NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "339266           NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "339363           NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "339387           NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "339408           NaN                 NaN  MODERATE   1.0         indel  NaN   \n",
       "\n",
       "        HGVS_OFFSET  PHENO  MINIMISED  ExAC_AF  ExAC_AF_AFR  ExAC_AF_AMR  \\\n",
       "91              NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "106             NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "110             NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "113             NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "162             NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "...             ...    ...        ...      ...          ...          ...   \n",
       "339227          NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "339266          NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "339363          NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "339387          NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "339408          NaN    NaN        NaN      NaN          NaN          NaN   \n",
       "\n",
       "        ExAC_AF_EAS  ExAC_AF_FIN  ExAC_AF_NFE  ExAC_AF_OTH  ExAC_AF_SAS  \\\n",
       "91              NaN          NaN          NaN          NaN          NaN   \n",
       "106             NaN          NaN          NaN          NaN          NaN   \n",
       "110             NaN          NaN          NaN          NaN          NaN   \n",
       "113             NaN          NaN          NaN          NaN          NaN   \n",
       "162             NaN          NaN          NaN          NaN          NaN   \n",
       "...             ...          ...          ...          ...          ...   \n",
       "339227          NaN          NaN          NaN          NaN          NaN   \n",
       "339266          NaN          NaN          NaN          NaN          NaN   \n",
       "339363          NaN          NaN          NaN          NaN          NaN   \n",
       "339387          NaN          NaN          NaN          NaN          NaN   \n",
       "339408          NaN          NaN          NaN          NaN          NaN   \n",
       "\n",
       "        GENE_PHENO FILTER flanking_bps variant_id variant_qual  ExAC_AF_Adj  \\\n",
       "91             1.0    NaN          CGC          .            .          NaN   \n",
       "106            1.0    NaN          GCG          .            .          NaN   \n",
       "110            1.0    NaN          GTA          .            .          NaN   \n",
       "113            1.0    NaN          GTA          .            .          NaN   \n",
       "162            1.0    NaN          GCG          .            .          NaN   \n",
       "...            ...    ...          ...        ...          ...          ...   \n",
       "339227         1.0    NaN          CGC          .            .          NaN   \n",
       "339266         1.0    NaN          GCG          .            .          NaN   \n",
       "339363         1.0    NaN          ACA          .            .          NaN   \n",
       "339387         1.0    NaN          GTA          .            .          NaN   \n",
       "339408         1.0    NaN          CGG          .            .          NaN   \n",
       "\n",
       "       ExAC_AC_AN_Adj ExAC_AC_AN ExAC_AC_AN_AFR ExAC_AC_AN_AMR ExAC_AC_AN_EAS  \\\n",
       "91                NaN        NaN            NaN            NaN            NaN   \n",
       "106               NaN        NaN            NaN            NaN            NaN   \n",
       "110               NaN        NaN            NaN            NaN            NaN   \n",
       "113               NaN        NaN            NaN            NaN            NaN   \n",
       "162               NaN        NaN            NaN            NaN            NaN   \n",
       "...               ...        ...            ...            ...            ...   \n",
       "339227            NaN        NaN            NaN            NaN            NaN   \n",
       "339266            NaN        NaN            NaN            NaN            NaN   \n",
       "339363            NaN        NaN            NaN            NaN            NaN   \n",
       "339387            NaN        NaN            NaN            NaN            NaN   \n",
       "339408            NaN        NaN            NaN            NaN            NaN   \n",
       "\n",
       "       ExAC_AC_AN_FIN ExAC_AC_AN_NFE ExAC_AC_AN_OTH ExAC_AC_AN_SAS  \\\n",
       "91                NaN            NaN            NaN            NaN   \n",
       "106               NaN            NaN            NaN            NaN   \n",
       "110               NaN            NaN            NaN            NaN   \n",
       "113               NaN            NaN            NaN            NaN   \n",
       "162               NaN            NaN            NaN            NaN   \n",
       "...               ...            ...            ...            ...   \n",
       "339227            NaN            NaN            NaN            NaN   \n",
       "339266            NaN            NaN            NaN            NaN   \n",
       "339363            NaN            NaN            NaN            NaN   \n",
       "339387            NaN            NaN            NaN            NaN   \n",
       "339408            NaN            NaN            NaN            NaN   \n",
       "\n",
       "       ExAC_FILTER  Caller is-a-hotspot is-a-3d-hotspot  \\\n",
       "91             NaN     NaN            Y               Y   \n",
       "106            NaN     NaN            Y               Y   \n",
       "110            NaN     NaN            Y               Y   \n",
       "113            NaN     NaN            Y               Y   \n",
       "162            NaN     NaN            Y               Y   \n",
       "...            ...     ...          ...             ...   \n",
       "339227         NaN     NaN            Y               Y   \n",
       "339266         NaN     NaN            Y               Y   \n",
       "339363         NaN     NaN            Y               Y   \n",
       "339387         NaN     NaN            Y               Y   \n",
       "339408         NaN     NaN            Y               Y   \n",
       "\n",
       "                mutation_effect         oncogenic LEVEL_1 LEVEL_2A LEVEL_2B  \\\n",
       "91      Likely Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "106            Loss-of-function         Oncogenic     NaN      NaN      NaN   \n",
       "110            Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "113            Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "162            Loss-of-function         Oncogenic     NaN      NaN      NaN   \n",
       "...                         ...               ...     ...      ...      ...   \n",
       "339227  Likely Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "339266         Loss-of-function         Oncogenic     NaN      NaN      NaN   \n",
       "339363         Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "339387         Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "339408  Likely Loss-of-function  Likely Oncogenic     NaN      NaN      NaN   \n",
       "\n",
       "       LEVEL_3A LEVEL_3B LEVEL_4 LEVEL_R1 LEVEL_R2  LEVEL_R3 Highest_level  \\\n",
       "91          NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "106         NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "110         NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "113         NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "162         NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "...         ...      ...     ...      ...      ...       ...           ...   \n",
       "339227      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "339266      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "339363      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "339387      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "339408      NaN      NaN     NaN      NaN      NaN       NaN           NaN   \n",
       "\n",
       "                                                   citations  driver  tcn  \\\n",
       "91                       23863845;11900253;23612969;15037740    True  2.0   \n",
       "106             25584008;15781620;22114072;10713666;21445056    True  6.0   \n",
       "110     25584008;8023157;11900253;15077194;23246812;15037740    True  3.0   \n",
       "113     25584008;8023157;11900253;15077194;23246812;15037740    True  2.0   \n",
       "162             25584008;15781620;22114072;10713666;21445056    True  3.0   \n",
       "...                                                      ...     ...  ...   \n",
       "339227                   23863845;11900253;23612969;15037740    True  1.0   \n",
       "339266          25584008;15781620;22114072;10713666;21445056    True  2.0   \n",
       "339363                                              15037740    True  3.0   \n",
       "339387                            26748848;16861262;10229196    True  2.0   \n",
       "339408                                              21445056    True  1.0   \n",
       "\n",
       "        lcn        cf    purity  t_var_freq  expected_alt_copies  ccf_Mcopies  \\\n",
       "91      0.0  0.306121  0.391316    0.312169                  2.0        0.798   \n",
       "106     0.0  0.630313  0.630313    0.815759                  6.0        0.975   \n",
       "110     0.0  0.835504  0.865628    0.845070                  3.0        0.933   \n",
       "113     0.0  0.312907  0.312907    0.636735                  4.0        1.000   \n",
       "162     0.0  0.257400  0.351778    0.325843                  2.0        0.726   \n",
       "...     ...       ...       ...         ...                  ...          ...   \n",
       "339227  0.0  0.412643  0.412643    0.298851                  1.0        1.000   \n",
       "339266  0.0  0.286332  0.301523    0.285229                  2.0        0.946   \n",
       "339363  0.0  0.598599  0.614603    0.675637                  3.0        0.958   \n",
       "339387  0.0  0.205450  0.300000    0.239130                  2.0        0.797   \n",
       "339408  0.0  0.632827  0.663312    0.484629                  1.0        0.977   \n",
       "\n",
       "        ccf_Mcopies_lower  ccf_Mcopies_upper  ccf_Mcopies_prob95  \\\n",
       "91                  0.747              0.850        3.269715e-04   \n",
       "106                 0.956              0.994        9.367284e-01   \n",
       "110                 0.908              0.955        1.667143e-01   \n",
       "113                 0.997              1.000        9.999955e-01   \n",
       "162                 0.683              0.771        3.198825e-09   \n",
       "...                   ...                ...                 ...   \n",
       "339227              0.979              1.000        8.899879e-01   \n",
       "339266              0.874              1.000        3.611258e-01   \n",
       "339363              0.928              0.988        6.114443e-01   \n",
       "339387              0.699              0.901        3.472834e-02   \n",
       "339408              0.926              1.000        6.310788e-01   \n",
       "\n",
       "        ccf_Mcopies_prob90  ccf_1copy  ccf_1copy_lower  ccf_1copy_upper  \\\n",
       "91                0.010807      1.000            0.993              1.0   \n",
       "106               0.999995        NaN              NaN              1.0   \n",
       "110               0.941853      1.000            0.997              1.0   \n",
       "113               1.000000      1.000            0.997              1.0   \n",
       "162               0.000003      1.000            0.995              1.0   \n",
       "...                    ...        ...              ...              ...   \n",
       "339227            0.993088      1.000            0.979              1.0   \n",
       "339266            0.731527      1.000            0.992              1.0   \n",
       "339363            0.988322      1.000            0.998              1.0   \n",
       "339387            0.119830      1.000            0.976              1.0   \n",
       "339408            0.950605      0.977            0.926              1.0   \n",
       "\n",
       "        ccf_1copy_prob95  ccf_1copy_prob90  ccf_expected_copies  \\\n",
       "91              0.997653          0.999997                0.798   \n",
       "106                  NaN               NaN                0.975   \n",
       "110             1.000000          1.000000                0.933   \n",
       "113             1.000000          1.000000                1.000   \n",
       "162             0.999866          1.000000                1.000   \n",
       "...                  ...               ...                  ...   \n",
       "339227          0.889988          0.993088                1.000   \n",
       "339266          0.993854          0.999975                0.946   \n",
       "339363          1.000000          1.000000                0.958   \n",
       "339387          0.814993          0.971027                0.797   \n",
       "339408          0.631079          0.950605                0.977   \n",
       "\n",
       "        ccf_expected_copies_lower  ccf_expected_copies_upper  \\\n",
       "91                          0.747                      0.850   \n",
       "106                         0.956                      0.994   \n",
       "110                         0.908                      0.955   \n",
       "113                         0.972                      1.000   \n",
       "162                         0.979                      1.000   \n",
       "...                           ...                        ...   \n",
       "339227                      0.979                      1.000   \n",
       "339266                      0.874                      1.000   \n",
       "339363                      0.928                      0.988   \n",
       "339387                      0.699                      0.901   \n",
       "339408                      0.926                      1.000   \n",
       "\n",
       "        ccf_expected_copies_prob95  ccf_expected_copies_prob90  \\\n",
       "91                        0.000327                    0.010807   \n",
       "106                       0.936728                    0.999995   \n",
       "110                       0.166714                    0.941853   \n",
       "113                       0.913083                    0.998538   \n",
       "162                       0.908549                    0.996435   \n",
       "...                            ...                         ...   \n",
       "339227                    0.889988                    0.993088   \n",
       "339266                    0.361126                    0.731527   \n",
       "339363                    0.611444                    0.988322   \n",
       "339387                    0.034728                    0.119830   \n",
       "339408                    0.631079                    0.950605   \n",
       "\n",
       "                                                                                                    facets_fit  \\\n",
       "91      /juno/work/ccs/shared/resources/impact/facets/all/P-00369/P-0036909-T01-IM6_P-0036909-N01-IM6//default   \n",
       "106     /juno/work/ccs/shared/resources/impact/facets/all/P-00165/P-0016553-T01-IM6_P-0016553-N01-IM6//default   \n",
       "110     /juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T01-IM6_P-0023546-N01-IM6//default   \n",
       "113     /juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default   \n",
       "162     /juno/work/ccs/shared/resources/impact/facets/all/P-00188/P-0018837-T01-IM6_P-0018837-N01-IM6//default   \n",
       "...                                                                                                        ...   \n",
       "339227  /juno/work/ccs/shared/resources/impact/facets/all/P-00227/P-0022777-T04-IM6_P-0022777-N02-IM6//default   \n",
       "339266  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050750-T01-IM6_P-0050750-N01-IM6//default   \n",
       "339363  /juno/work/ccs/shared/resources/impact/facets/all/P-00487/P-0048760-T01-IM6_P-0048760-N01-IM6//default   \n",
       "339387  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050744-T01-IM6_P-0050744-N01-IM6//default   \n",
       "339408  /juno/work/ccs/shared/resources/impact/facets/all/P-00495/P-0049537-T01-IM6_P-0049537-N01-IM6//default   \n",
       "\n",
       "        reviewer_set_purity  use_only_purity_run  use_edited_cncf  \\\n",
       "91                      NaN                False            False   \n",
       "106                     NaN                False            False   \n",
       "110                     NaN                False            False   \n",
       "113                     NaN                False            False   \n",
       "162                     NaN                False            False   \n",
       "...                     ...                  ...              ...   \n",
       "339227                  NaN                False            False   \n",
       "339266                  NaN                False            False   \n",
       "339363                  NaN                False            False   \n",
       "339387                  NaN                False            False   \n",
       "339408                  NaN                False            False   \n",
       "\n",
       "                                                                                                                                                    cncf_file_used  \\\n",
       "91      /juno/work/ccs/shared/resources/impact/facets/all/P-00369/P-0036909-T01-IM6_P-0036909-N01-IM6//default/P-0036909-T01-IM6_P-0036909-N01-IM6_hisens.cncf.txt   \n",
       "106     /juno/work/ccs/shared/resources/impact/facets/all/P-00165/P-0016553-T01-IM6_P-0016553-N01-IM6//default/P-0016553-T01-IM6_P-0016553-N01-IM6_hisens.cncf.txt   \n",
       "110     /juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T01-IM6_P-0023546-N01-IM6//default/P-0023546-T01-IM6_P-0023546-N01-IM6_hisens.cncf.txt   \n",
       "113     /juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default/P-0023546-T02-IM6_P-0023546-N01-IM6_hisens.cncf.txt   \n",
       "162     /juno/work/ccs/shared/resources/impact/facets/all/P-00188/P-0018837-T01-IM6_P-0018837-N01-IM6//default/P-0018837-T01-IM6_P-0018837-N01-IM6_hisens.cncf.txt   \n",
       "...                                                                                                                                                            ...   \n",
       "339227  /juno/work/ccs/shared/resources/impact/facets/all/P-00227/P-0022777-T04-IM6_P-0022777-N02-IM6//default/P-0022777-T04-IM6_P-0022777-N02-IM6_hisens.cncf.txt   \n",
       "339266  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050750-T01-IM6_P-0050750-N01-IM6//default/P-0050750-T01-IM6_P-0050750-N01-IM6_hisens.cncf.txt   \n",
       "339363  /juno/work/ccs/shared/resources/impact/facets/all/P-00487/P-0048760-T01-IM6_P-0048760-N01-IM6//default/P-0048760-T01-IM6_P-0048760-N01-IM6_hisens.cncf.txt   \n",
       "339387  /juno/work/ccs/shared/resources/impact/facets/all/P-00507/P-0050744-T01-IM6_P-0050744-N01-IM6//default/P-0050744-T01-IM6_P-0050744-N01-IM6_hisens.cncf.txt   \n",
       "339408  /juno/work/ccs/shared/resources/impact/facets/all/P-00495/P-0049537-T01-IM6_P-0049537-N01-IM6//default/P-0049537-T01-IM6_P-0049537-N01-IM6_hisens.cncf.txt   \n",
       "\n",
       "        tp53_res tp53_timing_wgd  \n",
       "91      0.404515          BEFORE  \n",
       "106     0.148539   INDETERMINATE  \n",
       "110     0.202426          BEFORE  \n",
       "113    -2.069805          BEFORE  \n",
       "162     0.821608          BEFORE  \n",
       "...          ...             ...  \n",
       "339227 -0.149619   INDETERMINATE  \n",
       "339266  0.108078          BEFORE  \n",
       "339363  0.125750          BEFORE  \n",
       "339387  0.405797          BEFORE  \n",
       "339408  0.023388   INDETERMINATE  \n",
       "\n",
       "[7925 rows x 168 columns]"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_cohort_wgd_tp53"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Tests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# test to count the number of mutations per sample\n",
    "sample_muts = maf_cohort_wgd[maf_cohort_wgd.Tumor_Sample_Barcode == 'P-0025956-T01-IM6'][maf_cohort_wgd.Tumor_Sample_Barcode == 'P-0023546-T02-IM6'][maf_cohort['Hugo_Symbol'] == 'TP53'].drop_duplicates(subset='sample_mut_key')\n",
    "sample_muts.shape[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>dbSNP_RS</th>\n",
       "      <th>dbSNP_Val_Status</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Match_Norm_Seq_Allele1</th>\n",
       "      <th>Match_Norm_Seq_Allele2</th>\n",
       "      <th>Tumor_Validation_Allele1</th>\n",
       "      <th>Tumor_Validation_Allele2</th>\n",
       "      <th>Match_Norm_Validation_Allele1</th>\n",
       "      <th>Match_Norm_Validation_Allele2</th>\n",
       "      <th>Verification_Status</th>\n",
       "      <th>Validation_Status</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>Sequencing_Phase</th>\n",
       "      <th>Sequence_Source</th>\n",
       "      <th>Validation_Method</th>\n",
       "      <th>Score</th>\n",
       "      <th>BAM_File</th>\n",
       "      <th>Sequencer</th>\n",
       "      <th>Tumor_Sample_UUID</th>\n",
       "      <th>Matched_Norm_Sample_UUID</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Exon_Number</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>n_depth</th>\n",
       "      <th>n_ref_count</th>\n",
       "      <th>n_alt_count</th>\n",
       "      <th>all_effects</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>ALLELE_NUM</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND_VEP</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>RefSeq</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>GMAF</th>\n",
       "      <th>AFR_MAF</th>\n",
       "      <th>AMR_MAF</th>\n",
       "      <th>ASN_MAF</th>\n",
       "      <th>EAS_MAF</th>\n",
       "      <th>EUR_MAF</th>\n",
       "      <th>SAS_MAF</th>\n",
       "      <th>AA_MAF</th>\n",
       "      <th>EA_MAF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>PICK</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>TSL</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>MINIMISED</th>\n",
       "      <th>ExAC_AF</th>\n",
       "      <th>ExAC_AF_AFR</th>\n",
       "      <th>ExAC_AF_AMR</th>\n",
       "      <th>ExAC_AF_EAS</th>\n",
       "      <th>ExAC_AF_FIN</th>\n",
       "      <th>ExAC_AF_NFE</th>\n",
       "      <th>ExAC_AF_OTH</th>\n",
       "      <th>ExAC_AF_SAS</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>flanking_bps</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_qual</th>\n",
       "      <th>ExAC_AF_Adj</th>\n",
       "      <th>ExAC_AC_AN_Adj</th>\n",
       "      <th>ExAC_AC_AN</th>\n",
       "      <th>ExAC_AC_AN_AFR</th>\n",
       "      <th>ExAC_AC_AN_AMR</th>\n",
       "      <th>ExAC_AC_AN_EAS</th>\n",
       "      <th>ExAC_AC_AN_FIN</th>\n",
       "      <th>ExAC_AC_AN_NFE</th>\n",
       "      <th>ExAC_AC_AN_OTH</th>\n",
       "      <th>ExAC_AC_AN_SAS</th>\n",
       "      <th>ExAC_FILTER</th>\n",
       "      <th>Caller</th>\n",
       "      <th>is-a-hotspot</th>\n",
       "      <th>is-a-3d-hotspot</th>\n",
       "      <th>mutation_effect</th>\n",
       "      <th>oncogenic</th>\n",
       "      <th>LEVEL_1</th>\n",
       "      <th>LEVEL_2A</th>\n",
       "      <th>LEVEL_2B</th>\n",
       "      <th>LEVEL_3A</th>\n",
       "      <th>LEVEL_3B</th>\n",
       "      <th>LEVEL_4</th>\n",
       "      <th>LEVEL_R1</th>\n",
       "      <th>LEVEL_R2</th>\n",
       "      <th>LEVEL_R3</th>\n",
       "      <th>Highest_level</th>\n",
       "      <th>citations</th>\n",
       "      <th>driver</th>\n",
       "      <th>tcn</th>\n",
       "      <th>lcn</th>\n",
       "      <th>cf</th>\n",
       "      <th>purity</th>\n",
       "      <th>t_var_freq</th>\n",
       "      <th>expected_alt_copies</th>\n",
       "      <th>ccf_Mcopies</th>\n",
       "      <th>ccf_Mcopies_lower</th>\n",
       "      <th>ccf_Mcopies_upper</th>\n",
       "      <th>ccf_Mcopies_prob95</th>\n",
       "      <th>ccf_Mcopies_prob90</th>\n",
       "      <th>ccf_1copy</th>\n",
       "      <th>ccf_1copy_lower</th>\n",
       "      <th>ccf_1copy_upper</th>\n",
       "      <th>ccf_1copy_prob95</th>\n",
       "      <th>ccf_1copy_prob90</th>\n",
       "      <th>ccf_expected_copies</th>\n",
       "      <th>ccf_expected_copies_lower</th>\n",
       "      <th>ccf_expected_copies_upper</th>\n",
       "      <th>ccf_expected_copies_prob95</th>\n",
       "      <th>ccf_expected_copies_prob90</th>\n",
       "      <th>facets_fit</th>\n",
       "      <th>reviewer_set_purity</th>\n",
       "      <th>use_only_purity_run</th>\n",
       "      <th>use_edited_cncf</th>\n",
       "      <th>cncf_file_used</th>\n",
       "      <th>LOH</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>111</th>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>1499</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>3</td>\n",
       "      <td>41266098</td>\n",
       "      <td>41266098</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.95A&gt;C</td>\n",
       "      <td>p.Asp32Ala</td>\n",
       "      <td>p.D32A</td>\n",
       "      <td>ENST00000349496</td>\n",
       "      <td>3/15</td>\n",
       "      <td>386.0</td>\n",
       "      <td>339.0</td>\n",
       "      <td>47.0</td>\n",
       "      <td>390</td>\n",
       "      <td>390</td>\n",
       "      <td>0</td>\n",
       "      <td>CTNNB1,missense_variant,p.Asp32Ala,ENST00000349496,NM_001904.3;CTNNB1,missense_variant,p.Asp32Ala,ENST00000396185,NM_001098209.1;CTNNB1,missense_variant,p.Asp32Ala,ENST00000396183,NM_001098210.1;CTNNB1,missense_variant,p.Asp25Ala,ENST00000453024,;CTNNB1,missense_variant,p.Asp32Ala,ENST00000405570,;CTNNB1,missense_variant,p.Asp32Ala,ENST00000450969,;CTNNB1,missense_variant,p.Asp32Ala,ENST00000431914,;CTNNB1,missense_variant,p.Asp32Ala,ENST00000441708,;CTNNB1,missense_variant,p.Asp25Ala,ENST00000426215,;CTNNB1,downstream_gene_variant,,ENST00000433400,;CTNNB1,non_coding_transcript_exon_variant,,ENST00000488914,;,regulatory_region_variant,,ENSR00000203589,;</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000168036</td>\n",
       "      <td>ENST00000349496</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>375/3729</td>\n",
       "      <td>95/2346</td>\n",
       "      <td>32/781</td>\n",
       "      <td>D/A</td>\n",
       "      <td>gAc/gCc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>CTNNB1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>2514.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS2694.1</td>\n",
       "      <td>ENSP00000344456</td>\n",
       "      <td>P35222</td>\n",
       "      <td>G9GAG7,E9PDF9,E7EV28,E7EMJ5,C9IZ65,B4DGU4</td>\n",
       "      <td>UPI000012862F</td>\n",
       "      <td>NM_001904.3</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.959)</td>\n",
       "      <td>3/15</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hmmpanther:PTHR23315,hmmpanther:PTHR23315:SF53</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GAC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gain-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>15829978</td>\n",
       "      <td>True</td>\n",
       "      <td>6.0</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>0.121762</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.422</td>\n",
       "      <td>0.357</td>\n",
       "      <td>0.493</td>\n",
       "      <td>2.583183e-13</td>\n",
       "      <td>1.732627e-11</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.944</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.545539</td>\n",
       "      <td>0.816613</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.944</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.545539</td>\n",
       "      <td>0.816613</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default/P-0023546-T02-IM6_P-0023546-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>112</th>\n",
       "      <td>PIK3CG</td>\n",
       "      <td>5294</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>7</td>\n",
       "      <td>106519962</td>\n",
       "      <td>106519962</td>\n",
       "      <td>+</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.2392-2A&gt;T</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.X798_splice</td>\n",
       "      <td>ENST00000359195</td>\n",
       "      <td>NaN</td>\n",
       "      <td>208.0</td>\n",
       "      <td>188.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>244</td>\n",
       "      <td>243</td>\n",
       "      <td>1</td>\n",
       "      <td>PIK3CG,splice_acceptor_variant,,ENST00000359195,NM_002649.2,NM_001282427.1;PIK3CG,splice_acceptor_variant,,ENST00000496166,NM_001282426.1;PIK3CG,splice_acceptor_variant,,ENST00000440650,;PIK3CG,splice_acceptor_variant,,ENST00000473541,;PIK3CG,upstream_gene_variant,,ENST00000466738,;</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000105851</td>\n",
       "      <td>ENST00000359195</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>splice_acceptor_variant</td>\n",
       "      <td>-/5377</td>\n",
       "      <td>2392/3309</td>\n",
       "      <td>798/1102</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>PIK3CG</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>8978.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS5739.1</td>\n",
       "      <td>ENSP00000352121</td>\n",
       "      <td>P48736</td>\n",
       "      <td>Q24M88,E9PDN7</td>\n",
       "      <td>UPI00000746B8</td>\n",
       "      <td>NM_002649.2,NM_001282427.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>1.0</td>\n",
       "      <td>SNV</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CAG</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>0.096154</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.355</td>\n",
       "      <td>0.273</td>\n",
       "      <td>0.451</td>\n",
       "      <td>3.536964e-09</td>\n",
       "      <td>4.038132e-08</td>\n",
       "      <td>0.711</td>\n",
       "      <td>0.546</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.040386</td>\n",
       "      <td>0.099892</td>\n",
       "      <td>0.711</td>\n",
       "      <td>0.546</td>\n",
       "      <td>0.902</td>\n",
       "      <td>0.040386</td>\n",
       "      <td>0.099892</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default/P-0023546-T02-IM6_P-0023546-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>113</th>\n",
       "      <td>TP53</td>\n",
       "      <td>7157</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>17</td>\n",
       "      <td>7578442</td>\n",
       "      <td>7578442</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.488A&gt;G</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>p.Y163C</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>5/11</td>\n",
       "      <td>490.0</td>\n",
       "      <td>178.0</td>\n",
       "      <td>312.0</td>\n",
       "      <td>744</td>\n",
       "      <td>744</td>\n",
       "      <td>0</td>\n",
       "      <td>TP53,missense_variant,p.Tyr163Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Tyr163Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Tyr163Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Tyr163Cys,ENST00000445888,;TP53,missense_variant,p.Tyr163Cys,ENST00000359597,;TP53,missense_variant,p.Tyr163Cys,ENST00000413465,;TP53,missense_variant,p.Tyr31Cys,ENST00000509690,;TP53,missense_variant,p.Tyr163Cys,ENST00000508793,;TP53,missense_variant,p.Tyr70Cys,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,upstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,ENST0000050...</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000141510</td>\n",
       "      <td>ENST00000269305</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>678/2579</td>\n",
       "      <td>488/1182</td>\n",
       "      <td>163/393</td>\n",
       "      <td>Y/C</td>\n",
       "      <td>tAc/tGc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11998.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS11118.1</td>\n",
       "      <td>ENSP00000269305</td>\n",
       "      <td>P04637</td>\n",
       "      <td>S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0</td>\n",
       "      <td>UPI000002ED67</td>\n",
       "      <td>NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.999)</td>\n",
       "      <td>5/11</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GTA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>25584008;8023157;11900253;15077194;23246812;15037740</td>\n",
       "      <td>True</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>0.636735</td>\n",
       "      <td>4.0</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.997</td>\n",
       "      <td>1.000</td>\n",
       "      <td>9.999955e-01</td>\n",
       "      <td>1.000000e+00</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.997</td>\n",
       "      <td>1.000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.972</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.913083</td>\n",
       "      <td>0.998538</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default/P-0023546-T02-IM6_P-0023546-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \\\n",
       "111      CTNNB1            1499  MSKCC     GRCh37          3        41266098   \n",
       "112      PIK3CG            5294  MSKCC     GRCh37          7       106519962   \n",
       "113        TP53            7157  MSKCC     GRCh37         17         7578442   \n",
       "\n",
       "     End_Position Strand Variant_Classification Variant_Type Reference_Allele  \\\n",
       "111      41266098      +      Missense_Mutation          SNP                A   \n",
       "112     106519962      +            Splice_Site          SNP                A   \n",
       "113       7578442      +      Missense_Mutation          SNP                T   \n",
       "\n",
       "    Tumor_Seq_Allele1 Tumor_Seq_Allele2 dbSNP_RS  dbSNP_Val_Status  \\\n",
       "111                 A                 C    novel               NaN   \n",
       "112                 A                 T    novel               NaN   \n",
       "113                 T                 C    novel               NaN   \n",
       "\n",
       "    Tumor_Sample_Barcode  Matched_Norm_Sample_Barcode Match_Norm_Seq_Allele1  \\\n",
       "111    P-0023546-T02-IM6                          NaN                    NaN   \n",
       "112    P-0023546-T02-IM6                          NaN                    NaN   \n",
       "113    P-0023546-T02-IM6                          NaN                    NaN   \n",
       "\n",
       "    Match_Norm_Seq_Allele2  Tumor_Validation_Allele1  \\\n",
       "111                    NaN                       NaN   \n",
       "112                    NaN                       NaN   \n",
       "113                    NaN                       NaN   \n",
       "\n",
       "     Tumor_Validation_Allele2  Match_Norm_Validation_Allele1  \\\n",
       "111                       NaN                            NaN   \n",
       "112                       NaN                            NaN   \n",
       "113                       NaN                            NaN   \n",
       "\n",
       "     Match_Norm_Validation_Allele2  Verification_Status Validation_Status  \\\n",
       "111                            NaN                  NaN           Unknown   \n",
       "112                            NaN                  NaN           Unknown   \n",
       "113                            NaN                  NaN           Unknown   \n",
       "\n",
       "    Mutation_Status  Sequencing_Phase  Sequence_Source  Validation_Method  \\\n",
       "111         SOMATIC               NaN              NaN                NaN   \n",
       "112         SOMATIC               NaN              NaN                NaN   \n",
       "113         SOMATIC               NaN              NaN                NaN   \n",
       "\n",
       "          Score  BAM_File  Sequencer  Tumor_Sample_UUID  \\\n",
       "111  MSK-IMPACT       NaN        NaN                NaN   \n",
       "112  MSK-IMPACT       NaN        NaN                NaN   \n",
       "113  MSK-IMPACT       NaN        NaN                NaN   \n",
       "\n",
       "     Matched_Norm_Sample_UUID        HGVSc        HGVSp    HGVSp_Short  \\\n",
       "111                       NaN      c.95A>C   p.Asp32Ala         p.D32A   \n",
       "112                       NaN  c.2392-2A>T          NaN  p.X798_splice   \n",
       "113                       NaN     c.488A>G  p.Tyr163Cys        p.Y163C   \n",
       "\n",
       "       Transcript_ID Exon_Number  t_depth  t_ref_count  t_alt_count  n_depth  \\\n",
       "111  ENST00000349496        3/15    386.0        339.0         47.0      390   \n",
       "112  ENST00000359195         NaN    208.0        188.0         20.0      244   \n",
       "113  ENST00000269305        5/11    490.0        178.0        312.0      744   \n",
       "\n",
       "    n_ref_count n_alt_count  \\\n",
       "111         390           0   \n",
       "112         243           1   \n",
       "113         744           0   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                 all_effects  \\\n",
       "111                                                                                                                                                                                                                                                                                                                                                    CTNNB1,missense_variant,p.Asp32Ala,ENST00000349496,NM_001904.3;CTNNB1,missense_variant,p.Asp32Ala,ENST00000396185,NM_001098209.1;CTNNB1,missense_variant,p.Asp32Ala,ENST00000396183,NM_001098210.1;CTNNB1,missense_variant,p.Asp25Ala,ENST00000453024,;CTNNB1,missense_variant,p.Asp32Ala,ENST00000405570,;CTNNB1,missense_variant,p.Asp32Ala,ENST00000450969,;CTNNB1,missense_variant,p.Asp32Ala,ENST00000431914,;CTNNB1,missense_variant,p.Asp32Ala,ENST00000441708,;CTNNB1,missense_variant,p.Asp25Ala,ENST00000426215,;CTNNB1,downstream_gene_variant,,ENST00000433400,;CTNNB1,non_coding_transcript_exon_variant,,ENST00000488914,;,regulatory_region_variant,,ENSR00000203589,;   \n",
       "112                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              PIK3CG,splice_acceptor_variant,,ENST00000359195,NM_002649.2,NM_001282427.1;PIK3CG,splice_acceptor_variant,,ENST00000496166,NM_001282426.1;PIK3CG,splice_acceptor_variant,,ENST00000440650,;PIK3CG,splice_acceptor_variant,,ENST00000473541,;PIK3CG,upstream_gene_variant,,ENST00000466738,;   \n",
       "113  TP53,missense_variant,p.Tyr163Cys,ENST00000420246,NM_001126114.2,NM_001276696.1;TP53,missense_variant,p.Tyr163Cys,ENST00000455263,NM_001276695.1,NM_001126113.2;TP53,missense_variant,p.Tyr163Cys,ENST00000269305,NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1;TP53,missense_variant,p.Tyr163Cys,ENST00000445888,;TP53,missense_variant,p.Tyr163Cys,ENST00000359597,;TP53,missense_variant,p.Tyr163Cys,ENST00000413465,;TP53,missense_variant,p.Tyr31Cys,ENST00000509690,;TP53,missense_variant,p.Tyr163Cys,ENST00000508793,;TP53,missense_variant,p.Tyr70Cys,ENST00000514944,;TP53,downstream_gene_variant,,ENST00000604348,;TP53,downstream_gene_variant,,ENST00000503591,;TP53,upstream_gene_variant,,ENST00000576024,;TP53,upstream_gene_variant,,ENST00000574684,;TP53,non_coding_transcript_exon_variant,,ENST00000510385,;TP53,non_coding_transcript_exon_variant,,ENST00000504290,;TP53,non_coding_transcript_exon_variant,,ENST00000504937,;TP53,non_coding_transcript_exon_variant,,ENST0000050...   \n",
       "\n",
       "    Allele             Gene          Feature Feature_type  \\\n",
       "111      C  ENSG00000168036  ENST00000349496   Transcript   \n",
       "112      T  ENSG00000105851  ENST00000359195   Transcript   \n",
       "113      C  ENSG00000141510  ENST00000269305   Transcript   \n",
       "\n",
       "                 Consequence cDNA_position CDS_position Protein_position  \\\n",
       "111         missense_variant      375/3729      95/2346           32/781   \n",
       "112  splice_acceptor_variant        -/5377    2392/3309         798/1102   \n",
       "113         missense_variant      678/2579     488/1182          163/393   \n",
       "\n",
       "    Amino_acids   Codons  Existing_variation  ALLELE_NUM  DISTANCE  \\\n",
       "111         D/A  gAc/gCc                 NaN           1       NaN   \n",
       "112         NaN      NaN                 NaN           1       NaN   \n",
       "113         Y/C  tAc/tGc                 NaN           1       NaN   \n",
       "\n",
       "     STRAND_VEP  SYMBOL SYMBOL_SOURCE  HGNC_ID         BIOTYPE CANONICAL  \\\n",
       "111           1  CTNNB1          HGNC   2514.0  protein_coding       YES   \n",
       "112           1  PIK3CG          HGNC   8978.0  protein_coding       YES   \n",
       "113          -1    TP53          HGNC  11998.0  protein_coding       YES   \n",
       "\n",
       "            CCDS             ENSP SWISSPROT  \\\n",
       "111   CCDS2694.1  ENSP00000344456    P35222   \n",
       "112   CCDS5739.1  ENSP00000352121    P48736   \n",
       "113  CCDS11118.1  ENSP00000269305    P04637   \n",
       "\n",
       "                                                                                                                                                                             TREMBL  \\\n",
       "111                                                                                                                                       G9GAG7,E9PDF9,E7EV28,E7EMJ5,C9IZ65,B4DGU4   \n",
       "112                                                                                                                                                                   Q24M88,E9PDN7   \n",
       "113  S5LQU8,Q761V2,Q6IT77,Q1HGV1,Q0PKT5,L0ES54,L0EQ05,K7PPA8,H2EHT1,G4Y083,E9PCY9,E7ESS1,E7EMR6,B5AKF6,B4DNI2,A4GWD0,A4GWB8,A4GWB5,A4GW97,A4GW76,A4GW75,A4GW74,A4GW67,A2I9Z1,A2I9Z0   \n",
       "\n",
       "           UNIPARC  \\\n",
       "111  UPI000012862F   \n",
       "112  UPI00000746B8   \n",
       "113  UPI000002ED67   \n",
       "\n",
       "                                                                      RefSeq  \\\n",
       "111                                                              NM_001904.3   \n",
       "112                                               NM_002649.2,NM_001282427.1   \n",
       "113  NM_001126112.2,NM_001276761.1,NM_001276760.1,NM_000546.5,NM_001126118.1   \n",
       "\n",
       "               SIFT                  PolyPhen  EXON INTRON  \\\n",
       "111  deleterious(0)  probably_damaging(0.959)  3/15    NaN   \n",
       "112             NaN                       NaN   NaN   5/10   \n",
       "113  deleterious(0)  probably_damaging(0.999)  5/11    NaN   \n",
       "\n",
       "                                                                                                                                     DOMAINS  \\\n",
       "111                                                                                           hmmpanther:PTHR23315,hmmpanther:PTHR23315:SF53   \n",
       "112                                                                                                                                      NaN   \n",
       "113  Gene3D:2.60.40.720,Pfam_domain:PF00870,Prints_domain:PR00386,hmmpanther:PTHR11447,hmmpanther:PTHR11447:SF6,Superfamily_domains:SSF49417   \n",
       "\n",
       "     GMAF  AFR_MAF  AMR_MAF  ASN_MAF  EAS_MAF  EUR_MAF  SAS_MAF  AA_MAF  \\\n",
       "111   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "112   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "113   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN   \n",
       "\n",
       "     EA_MAF  CLIN_SIG  SOMATIC  PUBMED  MOTIF_NAME  MOTIF_POS  HIGH_INF_POS  \\\n",
       "111     NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "112     NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "113     NaN       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "\n",
       "     MOTIF_SCORE_CHANGE    IMPACT  PICK VARIANT_CLASS  TSL  HGVS_OFFSET  \\\n",
       "111                 NaN  MODERATE   1.0           SNV  NaN          NaN   \n",
       "112                 NaN      HIGH   1.0           SNV  NaN          NaN   \n",
       "113                 NaN  MODERATE   1.0         indel  NaN          NaN   \n",
       "\n",
       "     PHENO  MINIMISED  ExAC_AF  ExAC_AF_AFR  ExAC_AF_AMR  ExAC_AF_EAS  \\\n",
       "111    NaN        1.0      NaN          NaN          NaN          NaN   \n",
       "112    NaN        1.0      NaN          NaN          NaN          NaN   \n",
       "113    NaN        NaN      NaN          NaN          NaN          NaN   \n",
       "\n",
       "     ExAC_AF_FIN  ExAC_AF_NFE  ExAC_AF_OTH  ExAC_AF_SAS  GENE_PHENO FILTER  \\\n",
       "111          NaN          NaN          NaN          NaN         1.0    NaN   \n",
       "112          NaN          NaN          NaN          NaN         NaN    NaN   \n",
       "113          NaN          NaN          NaN          NaN         1.0    NaN   \n",
       "\n",
       "    flanking_bps variant_id variant_qual  ExAC_AF_Adj ExAC_AC_AN_Adj  \\\n",
       "111          GAC          .            .          NaN            NaN   \n",
       "112          CAG          .            .          NaN            NaN   \n",
       "113          GTA          .            .          NaN            NaN   \n",
       "\n",
       "    ExAC_AC_AN ExAC_AC_AN_AFR ExAC_AC_AN_AMR ExAC_AC_AN_EAS ExAC_AC_AN_FIN  \\\n",
       "111        NaN            NaN            NaN            NaN            NaN   \n",
       "112        NaN            NaN            NaN            NaN            NaN   \n",
       "113        NaN            NaN            NaN            NaN            NaN   \n",
       "\n",
       "    ExAC_AC_AN_NFE ExAC_AC_AN_OTH ExAC_AC_AN_SAS ExAC_FILTER  Caller  \\\n",
       "111            NaN            NaN            NaN         NaN     NaN   \n",
       "112            NaN            NaN            NaN         NaN     NaN   \n",
       "113            NaN            NaN            NaN         NaN     NaN   \n",
       "\n",
       "    is-a-hotspot is-a-3d-hotspot   mutation_effect         oncogenic LEVEL_1  \\\n",
       "111            Y             NaN  Gain-of-function  Likely Oncogenic     NaN   \n",
       "112          NaN             NaN               NaN               NaN     NaN   \n",
       "113            Y               Y  Loss-of-function  Likely Oncogenic     NaN   \n",
       "\n",
       "    LEVEL_2A LEVEL_2B LEVEL_3A LEVEL_3B LEVEL_4 LEVEL_R1 LEVEL_R2  LEVEL_R3  \\\n",
       "111      NaN      NaN      NaN      NaN     NaN      NaN      NaN       NaN   \n",
       "112      NaN      NaN      NaN      NaN     NaN      NaN      NaN       NaN   \n",
       "113      NaN      NaN      NaN      NaN     NaN      NaN      NaN       NaN   \n",
       "\n",
       "    Highest_level                                             citations  \\\n",
       "111           NaN                                              15829978   \n",
       "112           NaN                                                   NaN   \n",
       "113           NaN  25584008;8023157;11900253;15077194;23246812;15037740   \n",
       "\n",
       "     driver  tcn  lcn        cf    purity  t_var_freq  expected_alt_copies  \\\n",
       "111    True  6.0  3.0  0.312907  0.312907    0.121762                  1.0   \n",
       "112   False  3.0  1.0  0.312907  0.312907    0.096154                  1.0   \n",
       "113    True  2.0  0.0  0.312907  0.312907    0.636735                  4.0   \n",
       "\n",
       "     ccf_Mcopies  ccf_Mcopies_lower  ccf_Mcopies_upper  ccf_Mcopies_prob95  \\\n",
       "111        0.422              0.357              0.493        2.583183e-13   \n",
       "112        0.355              0.273              0.451        3.536964e-09   \n",
       "113        1.000              0.997              1.000        9.999955e-01   \n",
       "\n",
       "     ccf_Mcopies_prob90  ccf_1copy  ccf_1copy_lower  ccf_1copy_upper  \\\n",
       "111        1.732627e-11      1.000            0.944            1.000   \n",
       "112        4.038132e-08      0.711            0.546            0.902   \n",
       "113        1.000000e+00      1.000            0.997            1.000   \n",
       "\n",
       "     ccf_1copy_prob95  ccf_1copy_prob90  ccf_expected_copies  \\\n",
       "111          0.545539          0.816613                1.000   \n",
       "112          0.040386          0.099892                0.711   \n",
       "113          1.000000          1.000000                1.000   \n",
       "\n",
       "     ccf_expected_copies_lower  ccf_expected_copies_upper  \\\n",
       "111                      0.944                      1.000   \n",
       "112                      0.546                      0.902   \n",
       "113                      0.972                      1.000   \n",
       "\n",
       "     ccf_expected_copies_prob95  ccf_expected_copies_prob90  \\\n",
       "111                    0.545539                    0.816613   \n",
       "112                    0.040386                    0.099892   \n",
       "113                    0.913083                    0.998538   \n",
       "\n",
       "                                                                                                 facets_fit  \\\n",
       "111  /juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default   \n",
       "112  /juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default   \n",
       "113  /juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default   \n",
       "\n",
       "     reviewer_set_purity  use_only_purity_run  use_edited_cncf  \\\n",
       "111                  NaN                False            False   \n",
       "112                  NaN                False            False   \n",
       "113                  NaN                False            False   \n",
       "\n",
       "                                                                                                                                                 cncf_file_used  \\\n",
       "111  /juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default/P-0023546-T02-IM6_P-0023546-N01-IM6_hisens.cncf.txt   \n",
       "112  /juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default/P-0023546-T02-IM6_P-0023546-N01-IM6_hisens.cncf.txt   \n",
       "113  /juno/work/ccs/shared/resources/impact/facets/all/P-00235/P-0023546-T02-IM6_P-0023546-N01-IM6//default/P-0023546-T02-IM6_P-0023546-N01-IM6_hisens.cncf.txt   \n",
       "\n",
       "       LOH  \n",
       "111    NaN  \n",
       "112  False  \n",
       "113   True  "
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_cohort_wgd[maf_cohort_wgd.Tumor_Sample_Barcode == 'P-0023546-T02-IM6'][maf_cohort_wgd.Tumor_Sample_Barcode == 'P-0023546-T02-IM6']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We will merge maf_cohort and gene_level in order to count the samples that have at least one tp53 event (either\n",
    "# tp53 mutation or tp53 CN event)\n",
    "# We add the WGD information to know if DIPLOID or TETRAPLOID is our baseline in terms of tp53 CN\n",
    "\n",
    "# First we select the columns\n",
    "maf_cohort_merge = maf_cohort_wgd[['Tumor_Sample_Barcode', 'Hugo_Symbol']]\n",
    "gene_level_merge = gene_level_tp53_wgd[['tumor_id', 'cn_state']]\n",
    "cohort=cohort[['tumor_sample', 'wgd']]\n",
    "\n",
    "# We merge\n",
    "tp53_hit = pd.merge(maf_cohort_merge, gene_level_merge, how='left', left_on='Tumor_Sample_Barcode', right_on='tumor_id')\n",
    "tp53_hit = pd.merge(tp53_hit, cohort, how='left', left_on='Tumor_Sample_Barcode', right_on='tumor_sample')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>tumor_id</th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>wgd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>STK11</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>KEAP1</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>SMARCA4</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>PTPRT</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90473</th>\n",
       "      <td>P-0049537-T01-IM6</td>\n",
       "      <td>ELF3</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90474</th>\n",
       "      <td>P-0049537-T01-IM6</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90475</th>\n",
       "      <td>P-0049537-T01-IM6</td>\n",
       "      <td>TP53</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90476</th>\n",
       "      <td>P-0049537-T01-IM6</td>\n",
       "      <td>RNF43</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90477</th>\n",
       "      <td>P-0049537-T01-IM6</td>\n",
       "      <td>DNMT3B</td>\n",
       "      <td>HETLOSS</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>90478 rows Ã— 4 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                tumor_id Hugo_Symbol tp53_cn_state   wgd\n",
       "0      P-0025956-T01-IM6        KRAS    LOSS AFTER  True\n",
       "1      P-0025956-T01-IM6       STK11    LOSS AFTER  True\n",
       "2      P-0025956-T01-IM6       KEAP1    LOSS AFTER  True\n",
       "3      P-0025956-T01-IM6     SMARCA4    LOSS AFTER  True\n",
       "4      P-0025956-T01-IM6       PTPRT    LOSS AFTER  True\n",
       "...                  ...         ...           ...   ...\n",
       "90473  P-0049537-T01-IM6        ELF3       HETLOSS  True\n",
       "90474  P-0049537-T01-IM6        KRAS       HETLOSS  True\n",
       "90475  P-0049537-T01-IM6        TP53       HETLOSS  True\n",
       "90476  P-0049537-T01-IM6       RNF43       HETLOSS  True\n",
       "90477  P-0049537-T01-IM6      DNMT3B       HETLOSS  True\n",
       "\n",
       "[90478 rows x 4 columns]"
      ]
     },
     "execution_count": 126,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tp53_hit.columns = ['tumor_id', 'Hugo_Symbol', 'tumor_id_bis', 'tp53_cn_state', 'tumor_id_ter', 'wgd']\n",
    "tp53_hit = tp53_hit.drop(['tumor_id_bis', 'tumor_id_ter'], axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "master = pd.read_pickle(data_path + 'merged_data/master_file.pkl')\n",
    "# We eliminate unmatched samples\n",
    "master = master[master['Somatic_Status']!='Unmatched']\n",
    "#We eliminate samples with too low purity\n",
    "samples_cutoff = list(master[master['purity'] > 0.20]['Tumor_Id'])\n",
    "samples_na = list(master[master['purity'].isna()]['Tumor_Id'])\n",
    "samples_cutoff = samples_cutoff + samples_na\n",
    "master_cutoff = master[master.Tumor_Id.isin(samples_cutoff)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "335.797px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
