{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <span style='font-family:\"Times New Roman\"'> <span styel=''> **MASTER FILE CREATION**\n",
    "\n",
    "## <span style='font-family:\"Times New Roman\"'> <span styel=''> *Emile Cohen* \n",
    "*March 2020*"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "**Goal:** In this Notebook, we create a master file that summarizes all useful information.\n",
    "\n",
    "The Notebook is divided in 4 parts, representing the four parts of our Master file:\n",
    "   \n",
    "* **1. Patient/Sample Information**\n",
    "* **2. TP53 Mutations**\n",
    "* **3. TP53 Copy Numbers**\n",
    "* **4. TP53 Computed Metrics**\n",
    "* **5. Subgroup columns creation**\n",
    "* **6. Merge tables**\n",
    "\n",
    "**NB1:** In each part, you must run the cells from the begining in order to initialize the variables\n",
    "\n",
    "**NB2:** In order to launch the last script (Merge Tables), you have to define the functions in each part.\n",
    "\n",
    "**NB3:** All functions used for the plots are located in utils/custom_tools.py\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Setup environment... done!\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "<span style=\"color:green\">✅ Working on **mskimpact_env** conda environment.</span>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%run -i '../../utils/setup_environment.ipy'\n",
    "\n",
    "from pathlib import Path\n",
    "from utils.filters import *\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "\n",
    "data_path = '../../data/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "## 1. Patient/Sample Information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we focus on clinical information exported from CbioPortal. We use the maf file created in the script *./maf_cohort_creation.ipynb* and stored in *../../data/merged_data/maf_cohort.pkl*.\n",
    "\n",
    "The following columns are selected:\n",
    "* Sample_Id\n",
    "* Tumor_Id\n",
    "* Patient_Id\n",
    "* Cancer_Type\n",
    "* Cancer_Type_Detailed\n",
    "* Sample_Type\n",
    "* purity\n",
    "* ploidy\n",
    "* samples_per_patient\n",
    "* Overall Survival Status\n",
    "* Overall Survival (Months)\n",
    "* MSI Score\n",
    "* Tumor Mutational Burden"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_sample_info(path):\n",
    "    '''\n",
    "    This function aims to create a dataframe gathering all samples from the cohort with important clinical\n",
    "    information.\n",
    "    We use the function normal_samp_duplicates_filter to filter out the samples that have the same tumor\n",
    "    but different normal samples, we keep only the one with the highest purity.\n",
    "    '''\n",
    "    maf_cohort = pd.read_pickle(path)\n",
    "    \n",
    "    #We select only interesting columns\n",
    "    selected_cohort = maf_cohort[['Sample_Id','Tumor_Id', 'Patient_Id','Cancer_Type', 'Cancer_Type_Detailed', 'Sample_Type', 'purity', 'ploidy',\n",
    "                                  'samples_per_patient','Overall Survival Status', 'Overall Survival (Months)', \n",
    "                                  'MSI Score', 'TMB_Score', 'Patient_Current_Age', 'Sex', 'Ethnicity_Category', 'Race_Category', 'mutationStatus', 'Somatic_Status']]\n",
    "\n",
    "    # But we have many duplicates\n",
    "    # First, we remove the duplicates based on Sample_Id\n",
    "    selected_cohort = selected_cohort.drop_duplicates('Sample_Id')\n",
    "    \n",
    "    return selected_cohort #selected_cohort"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 85,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>samples_per_patient</th>\n",
       "      <th>Overall Survival Status</th>\n",
       "      <th>Overall Survival (Months)</th>\n",
       "      <th>MSI Score</th>\n",
       "      <th>TMB_Score</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Ethnicity_Category</th>\n",
       "      <th>Race_Category</th>\n",
       "      <th>mutationStatus</th>\n",
       "      <th>Somatic_Status</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>P-0034223-T01-IM6</td>\n",
       "      <td>P-0034223</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.941111</td>\n",
       "      <td>2.241830</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.55</td>\n",
       "      <td>5.3</td>\n",
       "      <td>63.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NO VALUE ENTERED</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>Matched</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>P-0009819-T01-IM5_P-0009819-N01-IM5</td>\n",
       "      <td>P-0009819-T01-IM5</td>\n",
       "      <td>P-0009819</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.275237</td>\n",
       "      <td>2.681075</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>23.441</td>\n",
       "      <td>0.00</td>\n",
       "      <td>1.0</td>\n",
       "      <td>72.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Non-Spanish; Non-Hispanic</td>\n",
       "      <td>NO VALUE ENTERED</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>Matched</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>P-0025956</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.185874</td>\n",
       "      <td>3.496971</td>\n",
       "      <td>1.0</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>3.584</td>\n",
       "      <td>0.00</td>\n",
       "      <td>5.3</td>\n",
       "      <td>71.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Non-Spanish; Non-Hispanic</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>Matched</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>P-0027408-T01-IM6_P-0027408-N01-IM6</td>\n",
       "      <td>P-0027408-T01-IM6</td>\n",
       "      <td>P-0027408</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.308886</td>\n",
       "      <td>1.811066</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>22.586</td>\n",
       "      <td>0.27</td>\n",
       "      <td>17.6</td>\n",
       "      <td>67.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Non-Spanish; Non-Hispanic</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>Matched</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>P-0006554-T01-IM5_P-0006554-N01-IM5</td>\n",
       "      <td>P-0006554-T01-IM5</td>\n",
       "      <td>P-0006554</td>\n",
       "      <td>Glioma</td>\n",
       "      <td>Anaplastic Oligodendroglioma</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.715208</td>\n",
       "      <td>1.910719</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>26.170</td>\n",
       "      <td>1.30</td>\n",
       "      <td>46.2</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Non-Spanish; Non-Hispanic</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>Matched</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263095</th>\n",
       "      <td>P-0050644-T01-IM6_P-0050644-N01-IM6</td>\n",
       "      <td>P-0050644-T01-IM6</td>\n",
       "      <td>P-0050644</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.576103</td>\n",
       "      <td>2.175837</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.249</td>\n",
       "      <td>0.00</td>\n",
       "      <td>0.9</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Spanish  NOS; Hispanic NOS, Latino NOS</td>\n",
       "      <td>OTHER</td>\n",
       "      <td>NA</td>\n",
       "      <td>Matched</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263098</th>\n",
       "      <td>P-0050741-T01-IM6_P-0050741-N01-IM6</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>P-0050741</td>\n",
       "      <td>Small Cell Lung Cancer</td>\n",
       "      <td>Small Cell Lung Cancer</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.833591</td>\n",
       "      <td>2.006039</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.940</td>\n",
       "      <td>0.22</td>\n",
       "      <td>7.0</td>\n",
       "      <td>46.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Non-Spanish; Non-Hispanic</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>Matched</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263106</th>\n",
       "      <td>P-0050747-T01-IM6_P-0050747-N01-IM6</td>\n",
       "      <td>P-0050747-T01-IM6</td>\n",
       "      <td>P-0050747</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.360576</td>\n",
       "      <td>2.187990</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.05</td>\n",
       "      <td>6.1</td>\n",
       "      <td>72.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>NaN</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>Matched</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263112</th>\n",
       "      <td>P-0050652-T01-IM6_P-0050652-N01-IM6</td>\n",
       "      <td>P-0050652-T01-IM6</td>\n",
       "      <td>P-0050652</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.171442</td>\n",
       "      <td>2.011650</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.085</td>\n",
       "      <td>0.26</td>\n",
       "      <td>2.6</td>\n",
       "      <td>56.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Non-Spanish; Non-Hispanic</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>Matched</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>263115</th>\n",
       "      <td>P-0050745-T01-IM6_P-0050745-N01-IM6</td>\n",
       "      <td>P-0050745-T01-IM6</td>\n",
       "      <td>P-0050745</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.597798</td>\n",
       "      <td>1.808634</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.841</td>\n",
       "      <td>1.85</td>\n",
       "      <td>5.3</td>\n",
       "      <td>68.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Non-Spanish; Non-Hispanic</td>\n",
       "      <td>WHITE</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>Matched</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29247 rows × 19 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0       P-0034223-T01-IM6_P-0034223-N01-IM6  P-0034223-T01-IM6  P-0034223   \n",
       "6       P-0009819-T01-IM5_P-0009819-N01-IM5  P-0009819-T01-IM5  P-0009819   \n",
       "9       P-0025956-T01-IM6_P-0025956-N01-IM6  P-0025956-T01-IM6  P-0025956   \n",
       "15      P-0027408-T01-IM6_P-0027408-N01-IM6  P-0027408-T01-IM6  P-0027408   \n",
       "38      P-0006554-T01-IM5_P-0006554-N01-IM5  P-0006554-T01-IM5  P-0006554   \n",
       "...                                     ...                ...        ...   \n",
       "263095  P-0050644-T01-IM6_P-0050644-N01-IM6  P-0050644-T01-IM6  P-0050644   \n",
       "263098  P-0050741-T01-IM6_P-0050741-N01-IM6  P-0050741-T01-IM6  P-0050741   \n",
       "263106  P-0050747-T01-IM6_P-0050747-N01-IM6  P-0050747-T01-IM6  P-0050747   \n",
       "263112  P-0050652-T01-IM6_P-0050652-N01-IM6  P-0050652-T01-IM6  P-0050652   \n",
       "263115  P-0050745-T01-IM6_P-0050745-N01-IM6  P-0050745-T01-IM6  P-0050745   \n",
       "\n",
       "                       Cancer_Type              Cancer_Type_Detailed  \\\n",
       "0                    Breast Cancer         Invasive Breast Carcinoma   \n",
       "6                  Prostate Cancer           Prostate Adenocarcinoma   \n",
       "9       Non-Small Cell Lung Cancer               Lung Adenocarcinoma   \n",
       "15      Non-Small Cell Lung Cancer        Non-Small Cell Lung Cancer   \n",
       "38                          Glioma      Anaplastic Oligodendroglioma   \n",
       "...                            ...                               ...   \n",
       "263095             Prostate Cancer           Prostate Adenocarcinoma   \n",
       "263098      Small Cell Lung Cancer            Small Cell Lung Cancer   \n",
       "263106           Pancreatic Cancer         Pancreatic Adenocarcinoma   \n",
       "263112           Pancreatic Cancer         Pancreatic Adenocarcinoma   \n",
       "263115               Breast Cancer  Breast Invasive Ductal Carcinoma   \n",
       "\n",
       "       Sample_Type    purity    ploidy  samples_per_patient  \\\n",
       "0       Metastasis  0.941111  2.241830                  1.0   \n",
       "6          Primary  0.275237  2.681075                  1.0   \n",
       "9          Primary  0.185874  3.496971                  1.0   \n",
       "15      Metastasis  0.308886  1.811066                  1.0   \n",
       "38         Primary  0.715208  1.910719                  1.0   \n",
       "...            ...       ...       ...                  ...   \n",
       "263095     Primary  0.576103  2.175837                  1.0   \n",
       "263098  Metastasis  0.833591  2.006039                  1.0   \n",
       "263106     Primary  0.360576  2.187990                  1.0   \n",
       "263112     Primary  0.171442  2.011650                  1.0   \n",
       "263115     Primary  0.597798  1.808634                  1.0   \n",
       "\n",
       "       Overall Survival Status  Overall Survival (Months)  MSI Score  \\\n",
       "0                       LIVING                        NaN       0.55   \n",
       "6                       LIVING                     23.441       0.00   \n",
       "9                     DECEASED                      3.584       0.00   \n",
       "15                      LIVING                     22.586       0.27   \n",
       "38                      LIVING                     26.170       1.30   \n",
       "...                        ...                        ...        ...   \n",
       "263095                  LIVING                      1.249       0.00   \n",
       "263098                  LIVING                      1.940       0.22   \n",
       "263106                  LIVING                        NaN       0.05   \n",
       "263112                  LIVING                      1.085       0.26   \n",
       "263115                  LIVING                      1.841       1.85   \n",
       "\n",
       "        TMB_Score  Patient_Current_Age     Sex  \\\n",
       "0             5.3                 63.0  Female   \n",
       "6             1.0                 72.0    Male   \n",
       "9             5.3                 71.0  Female   \n",
       "15           17.6                 67.0  Female   \n",
       "38           46.2                 55.0  Female   \n",
       "...           ...                  ...     ...   \n",
       "263095        0.9                 53.0    Male   \n",
       "263098        7.0                 46.0    Male   \n",
       "263106        6.1                 72.0  Female   \n",
       "263112        2.6                 56.0    Male   \n",
       "263115        5.3                 68.0  Female   \n",
       "\n",
       "                            Ethnicity_Category     Race_Category  \\\n",
       "0                                          NaN  NO VALUE ENTERED   \n",
       "6                    Non-Spanish; Non-Hispanic  NO VALUE ENTERED   \n",
       "9                    Non-Spanish; Non-Hispanic             WHITE   \n",
       "15                   Non-Spanish; Non-Hispanic             WHITE   \n",
       "38                   Non-Spanish; Non-Hispanic             WHITE   \n",
       "...                                        ...               ...   \n",
       "263095  Spanish  NOS; Hispanic NOS, Latino NOS             OTHER   \n",
       "263098               Non-Spanish; Non-Hispanic             WHITE   \n",
       "263106                                     NaN             WHITE   \n",
       "263112               Non-Spanish; Non-Hispanic             WHITE   \n",
       "263115               Non-Spanish; Non-Hispanic             WHITE   \n",
       "\n",
       "       mutationStatus Somatic_Status  \n",
       "0             SOMATIC        Matched  \n",
       "6             SOMATIC        Matched  \n",
       "9             SOMATIC        Matched  \n",
       "15            SOMATIC        Matched  \n",
       "38            SOMATIC        Matched  \n",
       "...               ...            ...  \n",
       "263095             NA        Matched  \n",
       "263098        SOMATIC        Matched  \n",
       "263106        SOMATIC        Matched  \n",
       "263112        SOMATIC        Matched  \n",
       "263115        SOMATIC        Matched  \n",
       "\n",
       "[29247 rows x 19 columns]"
      ]
     },
     "execution_count": 85,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info = create_sample_info(data_path + 'merged_data/maf_cohort.pkl')\n",
    "sample_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "## 2. TP53 Mutations"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we focus on mutational information exported from *default_qc_pass.ccf_TP53.maf* file. We use the maf file created in the script *./maf_tp53_creation.ipynb* and stored in *../../data/merged_data/maf_tp53.pkl*.\n",
    "\n",
    "We gather all mutations per sample, and split it into different columns. We have the following columns:\n",
    "* Tumor_Id\t\n",
    "* key_1 (2,3,4,5) --> Mutation key allowing to filter duplicates\n",
    "* vc_1 (2,3,4,5) --> Variant Classification\n",
    "* ccf_1 (2,3,4,5) --> Cancer Cell Fraction of the mutation\n",
    "* vaf_1 (2,3,4,5) --> Variant Allele Frequency of the mutation\n",
    "* HGVSp_1 (2,3,4,5) --> protein change\n",
    "* spot_1 (2,3,4,5) --> Integer that defines the spot of the tp53 mutation\n",
    "* tp53_count --> Number of tp53 mutations of the sample\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "def f(x):\n",
    "    # This function helps us to group mutations together in a single cell per patient\n",
    "    return pd.DataFrame(dict(Sample_Id = x['Sample_Id'],  \n",
    "                        muts = \"%s\" % ','.join(x['sample_mut_key_vc_ccf_vaf_hgv_spot'])))\n",
    "\n",
    "def count_tp53_muts(x):\n",
    "    count = 0\n",
    "    for i in range(1,6):\n",
    "        if x['tp53_key_' + str(i)]:\n",
    "            count+= 1\n",
    "    return count\n",
    "\n",
    "def create_tp53_muts(sample_info, path):\n",
    "    '''\n",
    "    This function aims to gather all tp53 mutation characteristics.\n",
    "    For each sample we gather the tp53 mutations and their characteristics for all patients.\n",
    "    '''\n",
    "    # We load the  table created in maf_tp53_creation.ipynb\n",
    "    maf_tp53 = pd.read_pickle(path)\n",
    "    \n",
    "    # We select only intresting columns\n",
    "    maf_tp53_filtered = maf_tp53[['Sample_Id','sample_mut_key', 'Variant_Classification',\\\n",
    "                                        'ccf_expected_copies', 't_var_freq', 'HGVSp','mut_spot' ]]\n",
    "\n",
    "    # Let's Merge mut_key,Variant_classification, CF, CCF, and VAF to gather them\n",
    "    maf_tp53_filtered['sample_mut_key_vc_ccf_vaf_hgv_spot'] = pd.Series([str(i)+'%'+str(j)+'%'+str(k)+'%'+str(l)+'%'+str(m)+'%'+str(n) for i,j,k,l,m,n\\\n",
    "                                                            in zip(maf_tp53_filtered.sample_mut_key, \\\n",
    "                                                                   maf_tp53_filtered.Variant_Classification,\\\n",
    "                                                                   maf_tp53_filtered.ccf_expected_copies,\\\n",
    "                                                                   maf_tp53_filtered.t_var_freq,\\\n",
    "                                                                   maf_tp53_filtered.HGVSp,\\\n",
    "                                                                   maf_tp53_filtered.mut_spot\\\n",
    "                                                                  )]) \n",
    "\n",
    " \n",
    "\n",
    "    # We Select important columns\n",
    "    final = maf_tp53_filtered[['Sample_Id', 'sample_mut_key_vc_ccf_vaf_hgv_spot']]\n",
    "    # We groupby Patient_Id and apply the function above to group mutations\n",
    "    final = final.groupby(['Sample_Id'], sort=False).apply(f)\n",
    "\n",
    "    # We separate the different mutations into 5 different columns (5 is the max number of tp53 mutations in our cohort)\n",
    "    final[['mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5']] = final.muts.str.split(',', expand=True)\n",
    "\n",
    "    # Split the columns into mut_key_ and vc_\n",
    "    final[['tp53_key_1','tp53_vc_1','tp53_ccf_1','tp53_vaf_1','tp53_HGVSp_1', 'tp53_spot_1']] = final.mut_key_1.str.split('%', expand=True)\n",
    "    final[['tp53_key_2','tp53_vc_2','tp53_ccf_2','tp53_vaf_2','tp53_HGVSp_2', 'tp53_spot_2']] = final.mut_key_2.str.split('%', expand=True)\n",
    "    final[['tp53_key_3','tp53_vc_3','tp53_ccf_3','tp53_vaf_3','tp53_HGVSp_3', 'tp53_spot_3']] = final.mut_key_3.str.split('%', expand=True)\n",
    "    final[['tp53_key_4','tp53_vc_4','tp53_ccf_4','tp53_vaf_4','tp53_HGVSp_4', 'tp53_spot_4']] = final.mut_key_4.str.split('%', expand=True)\n",
    "    final[['tp53_key_5','tp53_vc_5','tp53_ccf_5','tp53_vaf_5','tp53_HGVSp_5', 'tp53_spot_5']] = final.mut_key_5.str.split('%', expand=True)\n",
    "\n",
    "    # We remove the muts column\n",
    "    final = final.drop(['muts','mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5'], axis=1)\n",
    "\n",
    "    # We remove duplicates\n",
    "    final = final.drop_duplicates('Sample_Id')\n",
    "\n",
    "    # We add the cohort patients that are not tp53 positive\n",
    "    #First we create a dataframe with all missing samples\n",
    "    cohort_samples = set(sample_info.Tumor_Id)\n",
    "    final_samples = set(final.Sample_Id)\n",
    "    missing_samp = pd.DataFrame(cohort_samples - final_samples, columns = ['Sample_Id'])\n",
    "    #Then we append the two datframe\n",
    "    final = final.append(missing_samp)\n",
    "    \n",
    "    # We rename the Sample_Id column to have the same key as in other datframes\n",
    "    final = final.rename(columns={'Sample_Id': 'Tumor_Id'})\n",
    "    \n",
    "    # We add a last column tp53_count that represents the number of tp53 mutations per sample\n",
    "    final = final.where(final.notnull(), None)\n",
    "    final['tp53_count'] = final.apply(count_tp53_muts, axis = 1)\n",
    "    \n",
    "    # We change the type of vafs column to float64 instead of strings\n",
    "    final = final.astype({'tp53_vaf_1': 'float64', 'tp53_vaf_2': 'float64', 'tp53_vaf_3': 'float64', 'tp53_vaf_4': 'float64', 'tp53_vaf_5': 'float64',\n",
    "                       'tp53_ccf_1': 'float64', 'tp53_ccf_2': 'float64', 'tp53_ccf_3': 'float64', 'tp53_ccf_4': 'float64', 'tp53_ccf_5': 'float64'})\n",
    "\n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0027408-T01-IM6</td>\n",
       "      <td>P-0027408-T01-IM6_17_7578409_CT_TC</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.925</td>\n",
       "      <td>0.168901</td>\n",
       "      <td>p.Arg174Glu</td>\n",
       "      <td>174</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909-T01-IM6_17_7577121_G_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.812</td>\n",
       "      <td>0.312169</td>\n",
       "      <td>p.Arg273Cys</td>\n",
       "      <td>273</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546-T01-IM6_17_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.935</td>\n",
       "      <td>0.845070</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546-T02-IM6_17_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.636735</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0025997-T01-IM6</td>\n",
       "      <td>P-0025997-T01-IM6_17_7578471_G_-</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.912621</td>\n",
       "      <td>p.Gly154AlafsTer16</td>\n",
       "      <td>154</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "            Tumor_Id                          tp53_key_1          tp53_vc_1  \\\n",
       "0  P-0027408-T01-IM6  P-0027408-T01-IM6_17_7578409_CT_TC  Missense_Mutation   \n",
       "1  P-0036909-T01-IM6    P-0036909-T01-IM6_17_7577121_G_A  Missense_Mutation   \n",
       "2  P-0023546-T01-IM6    P-0023546-T01-IM6_17_7578442_T_C  Missense_Mutation   \n",
       "3  P-0023546-T02-IM6    P-0023546-T02-IM6_17_7578442_T_C  Missense_Mutation   \n",
       "4  P-0025997-T01-IM6    P-0025997-T01-IM6_17_7578471_G_-    Frame_Shift_Del   \n",
       "\n",
       "   tp53_ccf_1  tp53_vaf_1        tp53_HGVSp_1 tp53_spot_1 tp53_key_2  \\\n",
       "0       0.925    0.168901         p.Arg174Glu         174       None   \n",
       "1       0.812    0.312169         p.Arg273Cys         273       None   \n",
       "2       0.935    0.845070         p.Tyr163Cys         163       None   \n",
       "3       1.000    0.636735         p.Tyr163Cys         163       None   \n",
       "4       1.000    0.912621  p.Gly154AlafsTer16         154       None   \n",
       "\n",
       "  tp53_vc_2  tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2 tp53_spot_2 tp53_key_3  \\\n",
       "0      None         NaN         NaN         None        None       None   \n",
       "1      None         NaN         NaN         None        None       None   \n",
       "2      None         NaN         NaN         None        None       None   \n",
       "3      None         NaN         NaN         None        None       None   \n",
       "4      None         NaN         NaN         None        None       None   \n",
       "\n",
       "  tp53_vc_3  tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3 tp53_spot_3 tp53_key_4  \\\n",
       "0      None         NaN         NaN         None        None       None   \n",
       "1      None         NaN         NaN         None        None       None   \n",
       "2      None         NaN         NaN         None        None       None   \n",
       "3      None         NaN         NaN         None        None       None   \n",
       "4      None         NaN         NaN         None        None       None   \n",
       "\n",
       "  tp53_vc_4  tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4 tp53_spot_4 tp53_key_5  \\\n",
       "0      None         NaN         NaN         None        None       None   \n",
       "1      None         NaN         NaN         None        None       None   \n",
       "2      None         NaN         NaN         None        None       None   \n",
       "3      None         NaN         NaN         None        None       None   \n",
       "4      None         NaN         NaN         None        None       None   \n",
       "\n",
       "  tp53_vc_5  tp53_ccf_5  tp53_vaf_5 tp53_HGVSp_5 tp53_spot_5  tp53_count  \n",
       "0      None         NaN         NaN         None        None           1  \n",
       "1      None         NaN         NaN         None        None           1  \n",
       "2      None         NaN         NaN         None        None           1  \n",
       "3      None         NaN         NaN         None        None           1  \n",
       "4      None         NaN         NaN         None        None           1  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tp53_muts = create_tp53_muts(sample_info, data_path + 'merged_data/maf_tp53.pkl')\n",
    "tp53_muts.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "## 3. TP53 Copy Numbers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we gather the information from gene_level table.\n",
    "We creaste the following columns:\n",
    "* Sample_Id \n",
    "* tcn --> total copy number\n",
    "* mcn --> major copy number\n",
    "* lcn --> lower copy number\n",
    "* seg_length --> length of the segment\n",
    "* cn_state --> copy number state\n",
    "* cf --> Cell fraction of the cn_state\n",
    "* wgd --> Wholde Genome Doubling (1 or -1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "def wgd_condition(x):\n",
    "    arm_level = pd.read_csv(data_path + 'impact-facets-tp53/raw/default_qc_pass.arm_level.txt', sep='\\t')\n",
    "    cond_wgd = ['LOSS BEFORE & AFTER', 'LOSS BEFORE', 'CNLOH BEFORE & AFTER',\n",
    "           'CNLOH BEFORE', 'CNLOH BEFORE & GAIN', 'DOUBLE LOSS AFTER',\n",
    "           'LOSS AFTER', 'CNLOH AFTER', 'LOSS & GAIN']\n",
    "    cond_no_wgd = ['CNLOH', 'HETLOSS', 'CNLOH & GAIN', 'DIPLOID']\n",
    "    \n",
    "    for tp53_cn_state in list(arm_level[arm_level['sample'] == x.Sample_Id]['tp53_cn_state']):\n",
    "        if tp53_cn_state in cond_wgd:\n",
    "            return 1\n",
    "        \n",
    "    if x.tp53_cn_state in cond_no_wgd :\n",
    "        return -1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_copy_number_state(sample_info, path):\n",
    "    arm_level = pd.read_csv(data_path + 'impact-facets-tp53/raw/default_qc_pass.arm_level.txt', sep='\\t')\n",
    "    \n",
    "    gene_level = pd.read_csv(path, sep='\\t')\n",
    "    gene_level['Tumor_Id'] = gene_level['sample'].str[:17]\n",
    "    gene_level_subset = gene_level[['sample','tcn','mcn','lcn','seg_length','cn_state', 'cf.em']]\n",
    "    \n",
    "    # We add the cohort patients that are not in the dataframe\n",
    "    #First we create a dataframe with all missing samples\n",
    "    cohort_samples = set(sample_info.Sample_Id)\n",
    "    gene_level_samples = set(gene_level_subset['sample'])\n",
    "    missing_samp = pd.DataFrame(cohort_samples - gene_level_samples, columns = ['sample'])\n",
    "    \n",
    "    \n",
    "    #Then we append the two dataframe\n",
    "    gene_level_subset = gene_level_subset.append(missing_samp)\n",
    "    \n",
    "    # We rename the cf.em column \n",
    "    gene_level_subset = gene_level_subset.rename(columns={'cf.em': 'tp53_cf', \n",
    "                                                          'sample':'Sample_Id',\n",
    "                                                          'tcn': 'tp53_tcn',\n",
    "                                                          'mcn': 'tp53_mcn',\n",
    "                                                          'lcn': 'tp53_lcn',\n",
    "                                                          'seg_length': 'tp53_seg_length',\n",
    "                                                          'cn_state':'tp53_cn_state'})\n",
    "    \n",
    "    # We add WGD information\n",
    "    gene_level_subset['wgd'] = gene_level_subset.apply(wgd_condition, axis = 1)\n",
    "    \n",
    "    return gene_level_subset"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "metadata": {},
   "outputs": [],
   "source": [
    "def condition_CNLOH(x):\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    if x.cn_state in CNLOH:\n",
    "        return 'CNLOH' + x.chr\n",
    "    else:\n",
    "        return 'NO_CNLOH'  + x.chr \n",
    "    \n",
    "def condition_GAIN(x):\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    if x.cn_state in GAIN:\n",
    "        return 'GAIN'+ x.chr\n",
    "    else:\n",
    "        return 'NO_GAIN'+ x.chr\n",
    "\n",
    "def condition_LOSS(x):\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    if x.cn_state in LOSS:\n",
    "        return 'LOSS'+ x.chr\n",
    "    else:\n",
    "        return 'NO_LOSS'+ x.chr\n",
    "\n",
    "\n",
    "def compute_frac_genome(x, arm_level: pd.DataFrame):\n",
    "    copy_number_state =  pd.read_pickle(data_path + 'merged_data/copy_number.pkl')\n",
    "    \n",
    "    lookup_table = arm_level[arm_level['sample'] == x.Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    if float(copy_number_state[copy_number_state['Sample_Id'] == x.Sample_Id]['wgd']) == 1:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "    altered_length = lookup_table_altered.cn_length.sum()\n",
    "    total_length = lookup_table.arm_length.sum()\n",
    "    \n",
    "    frac_gen_altered = round(altered_length/total_length,3)\n",
    "    \n",
    "    return frac_gen_altered\n",
    "\n",
    "# Here is the function that allws to compute genome instability columns\n",
    "def chr_computations(x):\n",
    "    arm_level = pd.read_csv(data_path + 'impact-facets-tp53/raw/default_qc_pass.arm_level.txt', sep='\\t')\n",
    "    copy_number_state =  pd.read_pickle(data_path + 'merged_data/copy_number.pkl')\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    arm_level_samples = list(set(arm_level['sample']))\n",
    "    \n",
    "    if x.Sample_Id not in arm_level_samples:\n",
    "        return ['NaN','NaN','NaN','NaN', 'NaN']\n",
    "    \n",
    "    lookup_table = arm_level[arm_level['sample'] == x.Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    if float(copy_number_state[copy_number_state['Sample_Id'] == x.Sample_Id]['wgd']) == 1:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "    lookup_table['state_chr'] = lookup_table['cn_state']+lookup_table['chr']\n",
    "    \n",
    "    # If only DIPLOID or TETRAPLOID\n",
    "    if lookup_table.empty:\n",
    "        return [float(0)]*5\n",
    "    \n",
    "    lookup_table['cnloh_chr'] = lookup_table.apply(condition_CNLOH, axis=1)\n",
    "    lookup_table['loss_chr'] = lookup_table.apply(condition_LOSS, axis=1)\n",
    "    lookup_table['gain_chr'] = lookup_table.apply(condition_GAIN, axis=1)\n",
    "\n",
    "    #chr_affected colum\n",
    "    lookup_table_chr = lookup_table.drop_duplicates(subset=['chr'])\n",
    "    chr_affected = len(lookup_table_chr)\n",
    "    \n",
    "    #chr_loss, chr_gain, chr_cnloh columns\n",
    "    lookup_table_cnloh = lookup_table.drop_duplicates(subset=['cnloh_chr'])['cnloh_chr']\n",
    "    lookup_table_loss = lookup_table.drop_duplicates(subset=['loss_chr'])['loss_chr']\n",
    "    lookup_table_gain = lookup_table.drop_duplicates(subset=['gain_chr'])['gain_chr']\n",
    "\n",
    "    chr_loss = len(lookup_table_loss[lookup_table_loss.str.startswith('LOSS')])\n",
    "    chr_gain = len(lookup_table_gain[lookup_table_gain.str.startswith('GAIN')])\n",
    "    chr_cnloh = len(lookup_table_cnloh[lookup_table_cnloh.str.startswith('CNLOH')])\n",
    "    \n",
    "    #frac_gen_altered column\n",
    "    frac_gen_altered = compute_frac_genome(x, arm_level)\n",
    "    \n",
    "    return [chr_affected, chr_loss, chr_gain, chr_cnloh, frac_gen_altered]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>samples_per_patient</th>\n",
       "      <th>Overall Survival Status</th>\n",
       "      <th>Overall Survival (Months)</th>\n",
       "      <th>MSI Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>P-0034223-T01-IM6</td>\n",
       "      <td>P-0034223</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.941111</td>\n",
       "      <td>2.241830</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>P-0009819-T01-IM5_P-0009819-N01-IM5</td>\n",
       "      <td>P-0009819-T01-IM5</td>\n",
       "      <td>P-0009819</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.275237</td>\n",
       "      <td>2.681075</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>23.441</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>P-0025956</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.185874</td>\n",
       "      <td>3.496971</td>\n",
       "      <td>1.0</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>3.584</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>P-0027408-T01-IM6_P-0027408-N01-IM6</td>\n",
       "      <td>P-0027408-T01-IM6</td>\n",
       "      <td>P-0027408</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.308886</td>\n",
       "      <td>1.811066</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>22.586</td>\n",
       "      <td>0.27</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>P-0006554-T01-IM5_P-0006554-N01-IM5</td>\n",
       "      <td>P-0006554-T01-IM5</td>\n",
       "      <td>P-0006554</td>\n",
       "      <td>Glioma</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.715208</td>\n",
       "      <td>1.910719</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>26.170</td>\n",
       "      <td>1.30</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260784</th>\n",
       "      <td>P-0050644-T01-IM6_P-0050644-N01-IM6</td>\n",
       "      <td>P-0050644-T01-IM6</td>\n",
       "      <td>P-0050644</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.576103</td>\n",
       "      <td>2.175837</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.249</td>\n",
       "      <td>0.00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260787</th>\n",
       "      <td>P-0050741-T01-IM6_P-0050741-N01-IM6</td>\n",
       "      <td>P-0050741-T01-IM6</td>\n",
       "      <td>P-0050741</td>\n",
       "      <td>Small Cell Lung Cancer</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.833591</td>\n",
       "      <td>2.006039</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.940</td>\n",
       "      <td>0.22</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260795</th>\n",
       "      <td>P-0050747-T01-IM6_P-0050747-N01-IM6</td>\n",
       "      <td>P-0050747-T01-IM6</td>\n",
       "      <td>P-0050747</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.360576</td>\n",
       "      <td>2.187990</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260801</th>\n",
       "      <td>P-0050652-T01-IM6_P-0050652-N01-IM6</td>\n",
       "      <td>P-0050652-T01-IM6</td>\n",
       "      <td>P-0050652</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.171442</td>\n",
       "      <td>2.011650</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.085</td>\n",
       "      <td>0.26</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>260804</th>\n",
       "      <td>P-0050745-T01-IM6_P-0050745-N01-IM6</td>\n",
       "      <td>P-0050745-T01-IM6</td>\n",
       "      <td>P-0050745</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.597798</td>\n",
       "      <td>1.808634</td>\n",
       "      <td>1.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.841</td>\n",
       "      <td>1.85</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29304 rows × 11 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                  Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0       P-0034223-T01-IM6_P-0034223-N01-IM6  P-0034223-T01-IM6  P-0034223   \n",
       "6       P-0009819-T01-IM5_P-0009819-N01-IM5  P-0009819-T01-IM5  P-0009819   \n",
       "9       P-0025956-T01-IM6_P-0025956-N01-IM6  P-0025956-T01-IM6  P-0025956   \n",
       "15      P-0027408-T01-IM6_P-0027408-N01-IM6  P-0027408-T01-IM6  P-0027408   \n",
       "36      P-0006554-T01-IM5_P-0006554-N01-IM5  P-0006554-T01-IM5  P-0006554   \n",
       "...                                     ...                ...        ...   \n",
       "260784  P-0050644-T01-IM6_P-0050644-N01-IM6  P-0050644-T01-IM6  P-0050644   \n",
       "260787  P-0050741-T01-IM6_P-0050741-N01-IM6  P-0050741-T01-IM6  P-0050741   \n",
       "260795  P-0050747-T01-IM6_P-0050747-N01-IM6  P-0050747-T01-IM6  P-0050747   \n",
       "260801  P-0050652-T01-IM6_P-0050652-N01-IM6  P-0050652-T01-IM6  P-0050652   \n",
       "260804  P-0050745-T01-IM6_P-0050745-N01-IM6  P-0050745-T01-IM6  P-0050745   \n",
       "\n",
       "                       Cancer_Type Sample_Type    purity    ploidy  \\\n",
       "0                    Breast Cancer  Metastasis  0.941111  2.241830   \n",
       "6                  Prostate Cancer     Primary  0.275237  2.681075   \n",
       "9       Non-Small Cell Lung Cancer     Primary  0.185874  3.496971   \n",
       "15      Non-Small Cell Lung Cancer  Metastasis  0.308886  1.811066   \n",
       "36                          Glioma     Primary  0.715208  1.910719   \n",
       "...                            ...         ...       ...       ...   \n",
       "260784             Prostate Cancer     Primary  0.576103  2.175837   \n",
       "260787      Small Cell Lung Cancer  Metastasis  0.833591  2.006039   \n",
       "260795           Pancreatic Cancer     Primary  0.360576  2.187990   \n",
       "260801           Pancreatic Cancer     Primary  0.171442  2.011650   \n",
       "260804               Breast Cancer     Primary  0.597798  1.808634   \n",
       "\n",
       "        samples_per_patient Overall Survival Status  \\\n",
       "0                       1.0                  LIVING   \n",
       "6                       1.0                  LIVING   \n",
       "9                       1.0                DECEASED   \n",
       "15                      1.0                  LIVING   \n",
       "36                      1.0                  LIVING   \n",
       "...                     ...                     ...   \n",
       "260784                  1.0                  LIVING   \n",
       "260787                  1.0                  LIVING   \n",
       "260795                  1.0                  LIVING   \n",
       "260801                  1.0                  LIVING   \n",
       "260804                  1.0                  LIVING   \n",
       "\n",
       "        Overall Survival (Months)  MSI Score  \n",
       "0                             NaN       0.55  \n",
       "6                          23.441       0.00  \n",
       "9                           3.584       0.00  \n",
       "15                         22.586       0.27  \n",
       "36                         26.170       1.30  \n",
       "...                           ...        ...  \n",
       "260784                      1.249       0.00  \n",
       "260787                      1.940       0.22  \n",
       "260795                        NaN       0.05  \n",
       "260801                      1.085       0.26  \n",
       "260804                      1.841       1.85  \n",
       "\n",
       "[29304 rows x 11 columns]"
      ]
     },
     "execution_count": 110,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>arm</th>\n",
       "      <th>tcn</th>\n",
       "      <th>lcn</th>\n",
       "      <th>cn_length</th>\n",
       "      <th>arm_length</th>\n",
       "      <th>frac_of_arm</th>\n",
       "      <th>cn_state</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>1p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>120534257</td>\n",
       "      <td>120534257</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>1q</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "      <td>101461100</td>\n",
       "      <td>125135032</td>\n",
       "      <td>0.81</td>\n",
       "      <td>GAIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>2p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>87764506</td>\n",
       "      <td>87764506</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>2q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>150656179</td>\n",
       "      <td>150656179</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>3p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>89055004</td>\n",
       "      <td>89055004</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>3q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>106857923</td>\n",
       "      <td>106857923</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>4p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>48257537</td>\n",
       "      <td>48257537</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>4q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>140602481</td>\n",
       "      <td>140602481</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>5p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>46187241</td>\n",
       "      <td>46187241</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>5q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>134121897</td>\n",
       "      <td>134121897</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>6p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>58437159</td>\n",
       "      <td>58437159</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>6q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>111968888</td>\n",
       "      <td>111968888</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>7p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>56381732</td>\n",
       "      <td>56381732</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>7q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>100573669</td>\n",
       "      <td>100573669</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>8q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>102347589</td>\n",
       "      <td>102347589</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>9p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>46945819</td>\n",
       "      <td>46945819</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>9q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>92452921</td>\n",
       "      <td>92452921</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>10p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>37744888</td>\n",
       "      <td>37744888</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>10q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>95926530</td>\n",
       "      <td>95926530</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>11p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>51319187</td>\n",
       "      <td>51319187</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>12p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>22819044</td>\n",
       "      <td>22848994</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>12q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>98407356</td>\n",
       "      <td>98407356</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>13q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>93122091</td>\n",
       "      <td>93122091</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>14q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>86418558</td>\n",
       "      <td>86418558</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>15q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>80605814</td>\n",
       "      <td>80605814</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>16p</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "      <td>33258581</td>\n",
       "      <td>35220670</td>\n",
       "      <td>0.94</td>\n",
       "      <td>GAIN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>16q</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>54904790</td>\n",
       "      <td>54904790</td>\n",
       "      <td>1.00</td>\n",
       "      <td>HETLOSS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>17p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>22148337</td>\n",
       "      <td>22148337</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>17q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>58520255</td>\n",
       "      <td>58520255</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>18p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>14736648</td>\n",
       "      <td>14736648</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>18q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>59002560</td>\n",
       "      <td>59002560</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>19p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>23475032</td>\n",
       "      <td>23475032</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>19q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>34414558</td>\n",
       "      <td>34414558</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>20p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>26023705</td>\n",
       "      <td>26023705</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>20q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>36311101</td>\n",
       "      <td>36311101</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>21q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>34964871</td>\n",
       "      <td>34964871</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>22q</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>31302129</td>\n",
       "      <td>31302129</td>\n",
       "      <td>1.00</td>\n",
       "      <td>HETLOSS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>23p</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>58180962</td>\n",
       "      <td>58180962</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>P-0034223-T01-IM6_P-0034223-N01-IM6</td>\n",
       "      <td>23q</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "      <td>94232388</td>\n",
       "      <td>94232388</td>\n",
       "      <td>1.00</td>\n",
       "      <td>DIPLOID</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 sample  arm  tcn  lcn  cn_length  arm_length  \\\n",
       "0   P-0034223-T01-IM6_P-0034223-N01-IM6   1p    2    1  120534257   120534257   \n",
       "1   P-0034223-T01-IM6_P-0034223-N01-IM6   1q    4    1  101461100   125135032   \n",
       "2   P-0034223-T01-IM6_P-0034223-N01-IM6   2p    2    1   87764506    87764506   \n",
       "3   P-0034223-T01-IM6_P-0034223-N01-IM6   2q    2    1  150656179   150656179   \n",
       "4   P-0034223-T01-IM6_P-0034223-N01-IM6   3p    2    1   89055004    89055004   \n",
       "5   P-0034223-T01-IM6_P-0034223-N01-IM6   3q    2    1  106857923   106857923   \n",
       "6   P-0034223-T01-IM6_P-0034223-N01-IM6   4p    2    1   48257537    48257537   \n",
       "7   P-0034223-T01-IM6_P-0034223-N01-IM6   4q    2    1  140602481   140602481   \n",
       "8   P-0034223-T01-IM6_P-0034223-N01-IM6   5p    2    1   46187241    46187241   \n",
       "9   P-0034223-T01-IM6_P-0034223-N01-IM6   5q    2    1  134121897   134121897   \n",
       "10  P-0034223-T01-IM6_P-0034223-N01-IM6   6p    2    1   58437159    58437159   \n",
       "11  P-0034223-T01-IM6_P-0034223-N01-IM6   6q    2    1  111968888   111968888   \n",
       "12  P-0034223-T01-IM6_P-0034223-N01-IM6   7p    2    1   56381732    56381732   \n",
       "13  P-0034223-T01-IM6_P-0034223-N01-IM6   7q    2    1  100573669   100573669   \n",
       "14  P-0034223-T01-IM6_P-0034223-N01-IM6   8q    2    1  102347589   102347589   \n",
       "15  P-0034223-T01-IM6_P-0034223-N01-IM6   9p    2    1   46945819    46945819   \n",
       "16  P-0034223-T01-IM6_P-0034223-N01-IM6   9q    2    1   92452921    92452921   \n",
       "17  P-0034223-T01-IM6_P-0034223-N01-IM6  10p    2    1   37744888    37744888   \n",
       "18  P-0034223-T01-IM6_P-0034223-N01-IM6  10q    2    1   95926530    95926530   \n",
       "19  P-0034223-T01-IM6_P-0034223-N01-IM6  11p    2    1   51319187    51319187   \n",
       "20  P-0034223-T01-IM6_P-0034223-N01-IM6  12p    2    1   22819044    22848994   \n",
       "21  P-0034223-T01-IM6_P-0034223-N01-IM6  12q    2    1   98407356    98407356   \n",
       "22  P-0034223-T01-IM6_P-0034223-N01-IM6  13q    2    1   93122091    93122091   \n",
       "23  P-0034223-T01-IM6_P-0034223-N01-IM6  14q    2    1   86418558    86418558   \n",
       "24  P-0034223-T01-IM6_P-0034223-N01-IM6  15q    2    1   80605814    80605814   \n",
       "25  P-0034223-T01-IM6_P-0034223-N01-IM6  16p    3    1   33258581    35220670   \n",
       "26  P-0034223-T01-IM6_P-0034223-N01-IM6  16q    1    0   54904790    54904790   \n",
       "27  P-0034223-T01-IM6_P-0034223-N01-IM6  17p    2    1   22148337    22148337   \n",
       "28  P-0034223-T01-IM6_P-0034223-N01-IM6  17q    2    1   58520255    58520255   \n",
       "29  P-0034223-T01-IM6_P-0034223-N01-IM6  18p    2    1   14736648    14736648   \n",
       "30  P-0034223-T01-IM6_P-0034223-N01-IM6  18q    2    1   59002560    59002560   \n",
       "31  P-0034223-T01-IM6_P-0034223-N01-IM6  19p    2    1   23475032    23475032   \n",
       "32  P-0034223-T01-IM6_P-0034223-N01-IM6  19q    2    1   34414558    34414558   \n",
       "33  P-0034223-T01-IM6_P-0034223-N01-IM6  20p    2    1   26023705    26023705   \n",
       "34  P-0034223-T01-IM6_P-0034223-N01-IM6  20q    2    1   36311101    36311101   \n",
       "35  P-0034223-T01-IM6_P-0034223-N01-IM6  21q    2    1   34964871    34964871   \n",
       "36  P-0034223-T01-IM6_P-0034223-N01-IM6  22q    1    0   31302129    31302129   \n",
       "37  P-0034223-T01-IM6_P-0034223-N01-IM6  23p    2    1   58180962    58180962   \n",
       "38  P-0034223-T01-IM6_P-0034223-N01-IM6  23q    2    1   94232388    94232388   \n",
       "\n",
       "    frac_of_arm cn_state  \n",
       "0          1.00  DIPLOID  \n",
       "1          0.81     GAIN  \n",
       "2          1.00  DIPLOID  \n",
       "3          1.00  DIPLOID  \n",
       "4          1.00  DIPLOID  \n",
       "5          1.00  DIPLOID  \n",
       "6          1.00  DIPLOID  \n",
       "7          1.00  DIPLOID  \n",
       "8          1.00  DIPLOID  \n",
       "9          1.00  DIPLOID  \n",
       "10         1.00  DIPLOID  \n",
       "11         1.00  DIPLOID  \n",
       "12         1.00  DIPLOID  \n",
       "13         1.00  DIPLOID  \n",
       "14         1.00  DIPLOID  \n",
       "15         1.00  DIPLOID  \n",
       "16         1.00  DIPLOID  \n",
       "17         1.00  DIPLOID  \n",
       "18         1.00  DIPLOID  \n",
       "19         1.00  DIPLOID  \n",
       "20         1.00  DIPLOID  \n",
       "21         1.00  DIPLOID  \n",
       "22         1.00  DIPLOID  \n",
       "23         1.00  DIPLOID  \n",
       "24         1.00  DIPLOID  \n",
       "25         0.94     GAIN  \n",
       "26         1.00  HETLOSS  \n",
       "27         1.00  DIPLOID  \n",
       "28         1.00  DIPLOID  \n",
       "29         1.00  DIPLOID  \n",
       "30         1.00  DIPLOID  \n",
       "31         1.00  DIPLOID  \n",
       "32         1.00  DIPLOID  \n",
       "33         1.00  DIPLOID  \n",
       "34         1.00  DIPLOID  \n",
       "35         1.00  DIPLOID  \n",
       "36         1.00  HETLOSS  \n",
       "37         1.00  DIPLOID  \n",
       "38         1.00  DIPLOID  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "0.08"
      ]
     },
     "execution_count": 48,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "arm_level = pd.read_csv(data_path + 'impact-facets-tp53/raw/default_qc_pass.arm_level.txt', sep='\\t')\n",
    "lookup_table = arm_level[arm_level['sample'] == 'P-0034223-T01-IM6_P-0034223-N01-IM6']\n",
    "lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'DIPLOID']\n",
    "display(lookup_table)\n",
    "altered_length = lookup_table_altered.cn_length.sum()\n",
    "total_length = lookup_table.arm_length.sum()\n",
    "frac_gen_altered = round(altered_length/total_length,3)\n",
    "frac_gen_altered"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>wgd</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>-1.0</th>\n",
       "      <td>17920</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1.0</th>\n",
       "      <td>9286</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      count\n",
       "wgd        \n",
       "-1.0  17920\n",
       " 1.0   9286"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(table, 'wgd', 'count')  "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "## 4. TP53 Computed Metrics"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we compute mainly 4 metrics:\n",
    "\n",
    "* mutation_count (*create_mut_count*) --> It is the total mutation count per sample\n",
    "* gene_count (*create_gene_count*)--> It is the number of mutated genes per sample\n",
    "* max_vaf --> It is the maximum Variant Allele Frequency within all the mutations of a sample\n",
    "* exp_nb_1 (2,3,4,5) --> It is the expected number of copies of tp53 mutations in a cell \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_gene_count(maf_cohort):\n",
    "    '''\n",
    "    This function create the count of genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    \n",
    "    # First we create the gene_count table by groupbying and sizing, we then change the index\n",
    "    selected_cohort = maf_cohort[['Sample_Id','Tumor_Id', 'Gene_Id']]\n",
    "    gene_count = pd.DataFrame(pd.DataFrame(selected_cohort[['Sample_Id', 'Gene_Id']].groupby(['Sample_Id', 'Gene_Id']).size(), columns = ['count']).groupby(['Sample_Id']).size(), columns = ['gene_count'])\n",
    "    gene_count = gene_count.reset_index()\n",
    "\n",
    "    # We add missing patients to the gene_count to have all the cohort\n",
    "    no_gene_id = selected_cohort.Gene_Id.isna()\n",
    "    no_gene_samples = set(selected_cohort[selected_cohort.index.isin(list(no_gene_id[no_gene_id == True].index))]['Sample_Id'])\n",
    "    missing_samp = pd.DataFrame(no_gene_samples, columns = ['Sample_Id'])\n",
    "\n",
    "    # We append the two dataframes\n",
    "    gene_count = gene_count.append(missing_samp)\n",
    "    \n",
    "    #Fillna with 0\n",
    "    gene_count = gene_count.fillna(0)\n",
    "\n",
    "    return gene_count\n",
    "\n",
    "def create_mut_count(maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all mutation count per sample.\n",
    "    '''\n",
    "    selected_cohort = maf_cohort[['Sample_Id','Tumor_Id', 'Gene_Id']]\n",
    "    mut_count = get_groupby(selected_cohort, 'Sample_Id', 'mutation_count')\n",
    "    \n",
    "    return mut_count\n",
    "\n",
    "\n",
    "def get_driver_count(x, maf_cohort):\n",
    "    lookup_table = maf_cohort[maf_cohort['Sample_Id'] == x.Sample_Id]\n",
    "    h = get_groupby(lookup_table, 'oncogenic', 'count')\n",
    "    count = (int(h.loc['Oncogenic']) if 'Oncogenic' in h.index else 0) + (int(h.loc['Likely Oncogenic']) if 'Likely Oncogenic' in h.index else 0) +(int(h.loc['Predicted Oncogenic']) if 'Predicted Oncogenic' in h.index else 0)\n",
    "    return count\n",
    "\n",
    "def create_driver_count(maf_cohort):\n",
    "    '''In this function we count the number of mutation driver per sample.'''\n",
    "    samples=list(set(maf_cohort.Sample_Id))\n",
    "    driver_count = pd.DataFrame(columns=['Sample_Id', 'driver_count'])\n",
    "    driver_count.Sample_Id = samples\n",
    "    driver_count['driver_count'] = driver_count.apply(get_driver_count, maf_cohort=maf_cohort, axis=1)\n",
    "\n",
    "    return driver_count\n",
    "    \n",
    "\n",
    "# The following function needs to be called on the complete master file because it needs info from different parts\n",
    "def create_copies_tp53_muts(master):\n",
    "    master['tp53_exp_nb_1'] = master.apply(lambda x:(x.tp53_vaf_1 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_2'] = master.apply(lambda x:(x.tp53_vaf_2 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_3'] = master.apply(lambda x:(x.tp53_vaf_3 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_4'] = master.apply(lambda x:(x.tp53_vaf_4 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_5'] = master.apply(lambda x:(x.tp53_vaf_5 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    \n",
    "    return master\n",
    "\n",
    "\n",
    "def vc_group_cond_1(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_1 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_1 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_1 in missense: \n",
    "        if x.tp53_spot_1 in ['273','248','175']: return x.tp53_spot_1\n",
    "        elif x.tp53_spot_1 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_2(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_2 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_2 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_2 in missense: \n",
    "        if x.tp53_spot_2 in ['273','248','175']: return x.tp53_spot_2\n",
    "        elif x.tp53_spot_2 in['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'   \n",
    "def vc_group_cond_3(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_3 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_3 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_3 in missense: \n",
    "        if x.tp53_spot_3 in ['273','248','175']: return x.tp53_spot_3\n",
    "        elif x.tp53_spot_3 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense' \n",
    "def vc_group_cond_4(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_4 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_4 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_4 in missense: \n",
    "        if x.tp53_spot_4 in ['273','248','175']: return x.tp53_spot_4\n",
    "        elif x.tp53_spot_4 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_5(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_5 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_5 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_5 in missense: \n",
    "        if x.tp53_spot_5 in ['273','248','175']: return x.tp53_spot_5\n",
    "        elif x.tp53_spot_5 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-81-389561220930>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     10\u001b[0m \u001b[0mdriver_count\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Sample_Id'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'driver_count'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     11\u001b[0m \u001b[0mdriver_count\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSample_Id\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msamples\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 12\u001b[0;31m \u001b[0mdriver_count\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'driver_count'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdriver_count\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mget_driver_count\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmaf_cohort\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mmaf_cohort\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/opt/anaconda3/envs/mskimpact_env/lib/python3.7/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mapply\u001b[0;34m(self, func, axis, raw, result_type, args, **kwds)\u001b[0m\n\u001b[1;32m   6876\u001b[0m             \u001b[0mkwds\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6877\u001b[0m         )\n\u001b[0;32m-> 6878\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   6879\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   6880\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mapplymap\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m->\u001b[0m \u001b[0;34m\"DataFrame\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/envs/mskimpact_env/lib/python3.7/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mget_result\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    184\u001b[0m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_raw\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    185\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 186\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply_standard\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    187\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    188\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mapply_empty_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/envs/mskimpact_env/lib/python3.7/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mapply_standard\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    294\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    295\u001b[0m                 result = libreduction.compute_reduction(\n\u001b[0;32m--> 296\u001b[0;31m                     \u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdummy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdummy\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlabels\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    297\u001b[0m                 )\n\u001b[1;32m    298\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mValueError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/reduction.pyx\u001b[0m in \u001b[0;36mpandas._libs.reduction.compute_reduction\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/reduction.pyx\u001b[0m in \u001b[0;36mpandas._libs.reduction.Reducer.get_result\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/envs/mskimpact_env/lib/python3.7/site-packages/pandas/core/apply.py\u001b[0m in \u001b[0;36mf\u001b[0;34m(x)\u001b[0m\n\u001b[1;32m    111\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    112\u001b[0m             \u001b[0;32mdef\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 113\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwds\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    114\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    115\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m<ipython-input-81-389561220930>\u001b[0m in \u001b[0;36mget_driver_count\u001b[0;34m(x, maf_cohort)\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0mget_driver_count\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmaf_cohort\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m     \u001b[0mlookup_table\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmaf_cohort\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mmaf_cohort\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Sample_Id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mSample_Id\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m     \u001b[0mh\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_groupby\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlookup_table\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'oncogenic'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'count'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mcount\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mh\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Oncogenic'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0;34m'Oncogenic'\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mh\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0;34m(\u001b[0m\u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mh\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Likely Oncogenic'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0;34m'Likely Oncogenic'\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mh\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m+\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mh\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'Predicted Oncogenic'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mif\u001b[0m \u001b[0;34m'Predicted Oncogenic'\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mh\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m \u001b[0;32melse\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mcount\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/envs/mskimpact_env/lib/python3.7/site-packages/pandas/core/ops/common.py\u001b[0m in \u001b[0;36mnew_method\u001b[0;34m(self, other)\u001b[0m\n\u001b[1;32m     62\u001b[0m         \u001b[0mother\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mitem_from_zerodim\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     63\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 64\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mother\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     65\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     66\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mnew_method\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/envs/mskimpact_env/lib/python3.7/site-packages/pandas/core/ops/__init__.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(self, other)\u001b[0m\n\u001b[1;32m    524\u001b[0m         \u001b[0mrvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mextract_array\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mother\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mextract_numpy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    525\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 526\u001b[0;31m         \u001b[0mres_values\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcomparison_op\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    527\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    528\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0m_construct_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mres_values\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mname\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mres_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/envs/mskimpact_env/lib/python3.7/site-packages/pandas/core/ops/array_ops.py\u001b[0m in \u001b[0;36mcomparison_op\u001b[0;34m(left, right, op)\u001b[0m\n\u001b[1;32m    245\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    246\u001b[0m     \u001b[0;32melif\u001b[0m \u001b[0mis_object_dtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlvalues\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 247\u001b[0;31m         \u001b[0mres_values\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcomp_method_OBJECT_ARRAY\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mop\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mrvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    248\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    249\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/opt/anaconda3/envs/mskimpact_env/lib/python3.7/site-packages/pandas/core/ops/array_ops.py\u001b[0m in \u001b[0;36mcomp_method_OBJECT_ARRAY\u001b[0;34m(op, x, y)\u001b[0m\n\u001b[1;32m     55\u001b[0m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlibops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mvec_compare\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mravel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     56\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 57\u001b[0;31m         \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlibops\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mscalar_compare\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mravel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0my\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     58\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mreshape\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     59\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "def get_driver_count(x, maf_cohort):\n",
    "    lookup_table = maf_cohort[maf_cohort['Sample_Id'] == x.Sample_Id]\n",
    "    h = get_groupby(lookup_table, 'oncogenic', 'count')\n",
    "    count = (int(h.loc['Oncogenic']) if 'Oncogenic' in h.index else 0) + (int(h.loc['Likely Oncogenic']) if 'Likely Oncogenic' in h.index else 0) +(int(h.loc['Predicted Oncogenic']) if 'Predicted Oncogenic' in h.index else 0)\n",
    "    return count\n",
    "\n",
    "maf_cohort = pd.read_pickle(data_path + 'merged_data/maf_cohort.pkl')\n",
    "selected_cohort = maf_cohort[['Sample_Id','oncogenic']]\n",
    "samples=list(set(selected_cohort.Sample_Id))\n",
    "driver_count = pd.DataFrame(columns=['Sample_Id', 'driver_count'])\n",
    "driver_count.Sample_Id = samples\n",
    "driver_count['driver_count'] = driver_count.apply(get_driver_count, maf_cohort=maf_cohort, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>driver_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0009708-T01-IM5_P-0009708-N01-IM5</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0006506-T02-IM6_P-0006506-N01-IM6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0010938-T01-IM5_P-0010938-N01-IM5</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0001321-T01-IM3_P-0001321-N01-IM3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0004129-T01-IM5_P-0004129-N01-IM5</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29299</th>\n",
       "      <td>P-0016238-T01-IM6_P-0016238-N01-IM6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29300</th>\n",
       "      <td>P-0020748-T01-IM6_P-0020748-N01-IM6</td>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29301</th>\n",
       "      <td>P-0000698-T01-IM3_P-0000698-N01-IM3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29302</th>\n",
       "      <td>P-0043424-T02-IM6_P-0043424-N01-IM6</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29303</th>\n",
       "      <td>P-0033622-T01-IM6_P-0033622-N01-IM6</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29304 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                 Sample_Id  driver_count\n",
       "0      P-0009708-T01-IM5_P-0009708-N01-IM5             4\n",
       "1      P-0006506-T02-IM6_P-0006506-N01-IM6             0\n",
       "2      P-0010938-T01-IM5_P-0010938-N01-IM5             0\n",
       "3      P-0001321-T01-IM3_P-0001321-N01-IM3             1\n",
       "4      P-0004129-T01-IM5_P-0004129-N01-IM5             1\n",
       "...                                    ...           ...\n",
       "29299  P-0016238-T01-IM6_P-0016238-N01-IM6             0\n",
       "29300  P-0020748-T01-IM6_P-0020748-N01-IM6            10\n",
       "29301  P-0000698-T01-IM3_P-0000698-N01-IM3             1\n",
       "29302  P-0043424-T02-IM6_P-0043424-N01-IM6             0\n",
       "29303  P-0033622-T01-IM6_P-0033622-N01-IM6             1\n",
       "\n",
       "[29304 rows x 2 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "driver_count.driver_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>driver_count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Sample_Id</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>P-0000004-T01-IM3_P-0000004-N01-IM3</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-0000012-T02-IM3_P-0000012-N01-IM3</th>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-0000024-T01-IM3_P-0000024-N01-IM3</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-0000025-T02-IM5_P-0000025-N01-IM5</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-0000026-T01-IM3_P-0000026-N01-IM3</th>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-0050745-T01-IM6_P-0050745-N01-IM6</th>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-0050746-T01-IM6_P-0050746-N01-IM6</th>\n",
       "      <td>89</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-0050747-T01-IM6_P-0050747-N01-IM6</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-0050748-T01-IM6_P-0050748-N01-IM6</th>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>P-0050749-T01-IM6_P-0050749-N01-IM6</th>\n",
       "      <td>10</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>29304 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                     driver_count\n",
       "Sample_Id                                        \n",
       "P-0000004-T01-IM3_P-0000004-N01-IM3             4\n",
       "P-0000012-T02-IM3_P-0000012-N01-IM3             1\n",
       "P-0000024-T01-IM3_P-0000024-N01-IM3             6\n",
       "P-0000025-T02-IM5_P-0000025-N01-IM5             2\n",
       "P-0000026-T01-IM3_P-0000026-N01-IM3             4\n",
       "...                                           ...\n",
       "P-0050745-T01-IM6_P-0050745-N01-IM6             7\n",
       "P-0050746-T01-IM6_P-0050746-N01-IM6            89\n",
       "P-0050747-T01-IM6_P-0050747-N01-IM6             6\n",
       "P-0050748-T01-IM6_P-0050748-N01-IM6             2\n",
       "P-0050749-T01-IM6_P-0050749-N01-IM6            10\n",
       "\n",
       "[29304 rows x 1 columns]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "maf_cohort = pd.read_pickle(data_path + 'merged_data/maf_cohort.pkl')\n",
    "h = create_driver_count(maf_cohort)\n",
    "display(h)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_computed_metrics(path):\n",
    "    # We add mutation_count and max_vaf\n",
    "    maf_cohort = pd.read_pickle(path)\n",
    "\n",
    "    # MUTATION COUNT\n",
    "    #We create the table for mutation_count\n",
    "    mut_count = create_mut_count(maf_cohort)\n",
    "    \n",
    "    # We create the table for gene_count\n",
    "    gene_count = create_gene_count(maf_cohort)\n",
    "    \n",
    "    # We create the table for the driver count per sample\n",
    "    driver_count = create_driver_count(maf_cohort)\n",
    "\n",
    "    # MAX_VAF\n",
    "    # To do so, we groupby Tumor_Id and apply the max() function\n",
    "    # But first we need to transform None values in Nan to compute the max\n",
    "    maf_cohort['vaf'].replace('None', np.nan, inplace=True)\n",
    "    max_vaf = maf_cohort[['Sample_Id','vaf']].groupby(['Sample_Id']).max()\n",
    "    max_vaf = max_vaf.rename(columns={'vaf': 'max_vaf'})\n",
    "    \n",
    "    # Merge the tables\n",
    "    computed_metrics = pd.merge(mut_count, gene_count, on=['Sample_Id'])\n",
    "    computed_metrics = pd.merge(computed_metrics, driver_count, on=['Sample_Id'])\n",
    "    computed_metrics = pd.merge(computed_metrics, max_vaf, on=['Sample_Id'])\n",
    "    \n",
    "    \n",
    "    return computed_metrics"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>mutation_count</th>\n",
       "      <th>gene_count</th>\n",
       "      <th>driver_count</th>\n",
       "      <th>max_vaf</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0000004-T01-IM3_P-0000004-N01-IM3</td>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.547085</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0000012-T02-IM3_P-0000012-N01-IM3</td>\n",
       "      <td>1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.502203</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0000024-T01-IM3_P-0000024-N01-IM3</td>\n",
       "      <td>6</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.368683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0000025-T02-IM5_P-0000025-N01-IM5</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2</td>\n",
       "      <td>0.203236</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0000026-T01-IM3_P-0000026-N01-IM3</td>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0</td>\n",
       "      <td>0.590164</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                             Sample_Id  mutation_count  gene_count  \\\n",
       "0  P-0000004-T01-IM3_P-0000004-N01-IM3               4         4.0   \n",
       "1  P-0000012-T02-IM3_P-0000012-N01-IM3               1         1.0   \n",
       "2  P-0000024-T01-IM3_P-0000024-N01-IM3               6         5.0   \n",
       "3  P-0000025-T02-IM5_P-0000025-N01-IM5               2         2.0   \n",
       "4  P-0000026-T01-IM3_P-0000026-N01-IM3               4         4.0   \n",
       "\n",
       "   driver_count   max_vaf  \n",
       "0             2  0.547085  \n",
       "1             0  0.502203  \n",
       "2             2  0.368683  \n",
       "3             2  0.203236  \n",
       "4             0  0.590164  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "computed_metrics = create_computed_metrics(data_path + 'merged_data/maf_cohort.pkl')\n",
    "computed_metrics.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 5. Subgroup Columns Creation"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### tp53_group\n",
    "First, we group the different COpy Number States *cn_state* in subgroups, under the column *cn_group*:\n",
    "* Group 1: cnLOH gathering ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'CNLOH BEFORE & GAIN']\n",
    "* Group 2: LOSS gathering ['LOSS BEFORE', 'HETLOSS', 'LOSS BEFORE & AFTER']\n",
    "* Group 3: HOMDEL gathering ['HOMDEL']\n",
    "* Group 4: DOUBLE LOSS AFTER gathering ['DOUBLE LOSS AFTER']\n",
    "* Group 5: WILD_TYPE gathering ['LOSS AFTER', 'DIPLOID', 'TETRAPLOID']\n",
    "* Group 6: GAIN gathering ['GAIN']\n",
    "* Group 7: OTHER gathering ['CNLOH AFTER', 'AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']\n",
    "\n",
    "\n",
    "Based on this first column we define 7 final groups of patients adding the mutational information. These groups will be under the column *mut_cn_group*.\n",
    "* Group 1: Samples with 0 tp53 mutations and HETLOSS\n",
    "* Group 2: Samples with HOMDEL\n",
    "* Group 3: Samples with 1 tp53 mutation and WILD_TYPE (DIPLOID, LOSS AFTER, TETRAPLOID)\n",
    "* Group 4: Samples with 1 tp53 mutation or more and LOSS\n",
    "* Group 5: Samples with 1 tp53 mutation or more and cnLOH\n",
    "* Group 6: Samples with 2/3/4/5 tp53 mutations and WILD_TYPE or GAIN\n",
    "\n",
    "We define the columns thanks to 2 functions that we call in the **Merge Tables** part through *create_master* function."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "def cn_group_cond(x):\n",
    "    if x.tp53_cn_state in ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'CNLOH BEFORE & GAIN', 'LOSS BEFORE']:\n",
    "        return 'cnLOH'\n",
    "    if x.tp53_cn_state in ['HETLOSS', 'LOSS BEFORE & AFTER']:\n",
    "        return 'LOSS'\n",
    "    if x.tp53_cn_state == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_cn_state in ['LOSS AFTER', 'DIPLOID', 'TETRAPLOID']:\n",
    "        return 'WILD_TYPE'\n",
    "    if x.tp53_cn_state == 'DOUBLE LOSS AFTER':\n",
    "        return 'DOUBLE LOSS AFTER'\n",
    "    if x.tp53_cn_state == 'GAIN':\n",
    "        return 'GAIN'\n",
    "    if x.tp53_cn_state in ['CNLOH AFTER', 'AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']:\n",
    "        return 'OTHER'\n",
    "\n",
    "def mut_cn_group_cond(x):\n",
    "    if x.tp53_cn_state == 'HETLOSS' and x.tp53_count == 0:\n",
    "        return '0_HETLOSS'\n",
    "    if x.tp53_first_group == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_first_group == 'WILD_TYPE' and x.tp53_count == 1 :\n",
    "        return '1_WILD_TYPE'\n",
    "    if x.tp53_first_group == 'LOSS' and x.tp53_count >=1:\n",
    "        return '>=1_LOSS'\n",
    "    if x.tp53_first_group == 'cnLOH' and x.tp53_count >=1:\n",
    "        return '>=1_cnLOH'\n",
    "    if (x.tp53_first_group == 'WILD_TYPE' or x.tp53_first_group == 'DOUBLE LOSS AFTER' or x.tp53_first_group == 'GAIN') and x.tp53_count > 1:\n",
    "        return '>1muts'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### tp53_residual subgroups"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "def tp53_residual_group(x):\n",
    "    if x.tp53_group == '1_WILD_TYPE' or x.tp53_group == '0_HETLOSS':\n",
    "        return 'tp53_res'\n",
    "    if x.tp53_group == 'HOMDEL':\n",
    "        return 'no_tp53_res'\n",
    "    if x.tp53_group == '>=1_LOSS' or x.tp53_group == '>=1_cnLOH':\n",
    "        if (x.tp53_residual_1 < 0.5) or (x.tp53_residual_2 < 0.5):\n",
    "            return 'no_tp53_res'\n",
    "        else:\n",
    "            if (x.cf + max(x.tp53_ccf_1, x.tp53_ccf_2, x.tp53_ccf_3, x.tp53_ccf_4, x.tp53_ccf_5)) > 1:\n",
    "                return 'no_tp53_res'\n",
    "            else:\n",
    "                return 'uncertain'\n",
    "    if x.tp53_group == '>1muts':\n",
    "        if (x.tp53_residual_1 + x.tp53_residual_2 < 2.5):\n",
    "            if (x.tp53_ccf_1 + x.tp53_ccf_2 > 1):\n",
    "                return 'no_tp53_res'\n",
    "            else: \n",
    "                return 'uncertain'\n",
    "           \n",
    "        elif (x.tp53_residual_1 + x.tp53_residual_2 > 2.5):\n",
    "            return 'tp53_res'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "## Merge Tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>samples_per_patient</th>\n",
       "      <th>Overall Survival Status</th>\n",
       "      <th>Overall Survival (Months)</th>\n",
       "      <th>MSI Score</th>\n",
       "      <th>TMB_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>174</th>\n",
       "      <td>P-0032161-T02-IM6_P-0032161-N01-IM6</td>\n",
       "      <td>P-0032161-T02-IM6</td>\n",
       "      <td>P-0032161</td>\n",
       "      <td>Endometrial Cancer</td>\n",
       "      <td>Uterine Endometrioid Carcinoma</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.608041</td>\n",
       "      <td>2.230969</td>\n",
       "      <td>2.0</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.874</td>\n",
       "      <td>40.19</td>\n",
       "      <td>57.9</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                               Sample_Id           Tumor_Id Patient_Id  \\\n",
       "174  P-0032161-T02-IM6_P-0032161-N01-IM6  P-0032161-T02-IM6  P-0032161   \n",
       "\n",
       "            Cancer_Type            Cancer_Type_Detailed Sample_Type    purity  \\\n",
       "174  Endometrial Cancer  Uterine Endometrioid Carcinoma     Primary  0.608041   \n",
       "\n",
       "       ploidy  samples_per_patient Overall Survival Status  \\\n",
       "174  2.230969                  2.0                  LIVING   \n",
       "\n",
       "     Overall Survival (Months)  MSI Score  TMB_Score  \n",
       "174                      1.874      40.19       57.9  "
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info = create_sample_info(data_path + 'merged_data/maf_cohort.pkl')\n",
    "sample_info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "7de67d784cbd45a589ec73a914dc0d44",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=1.0, bar_style='info', max=1.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "1544a37c079c4f2e97c1c970db222ad9",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=0.0, max=29304.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "checkpoint 1\n",
      "checkpoint 2\n"
     ]
    }
   ],
   "source": [
    "from tqdm import tqdm,tqdm_notebook\n",
    "\n",
    "def compute_genome_instability():\n",
    "    tqdm_notebook().pandas()\n",
    "    sample_info = create_sample_info(data_path + 'merged_data/maf_cohort.pkl')\n",
    "    sample_info = sample_info\n",
    "    sample_info['chr_comput'] = sample_info.progress_apply(chr_computations, axis=1)\n",
    "    print('checkpoint 1')\n",
    "    sample_info[['chr_affected', 'chr_loss', 'chr_gain', 'chr_cnloh', 'frac_genome_altered']] = pd.DataFrame(sample_info.chr_comput.values.tolist(), index= sample_info.index)\n",
    "    print('checkpoint 2')\n",
    "    \n",
    "    return sample_info\n",
    "\n",
    "sample_info = compute_genome_instability()\n",
    "sample_info[['Sample_Id','chr_affected', 'chr_loss', 'chr_gain', 'chr_cnloh', 'frac_genome_altered']].to_pickle(data_path + 'merged_data/chr_metrics_new.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_master():\n",
    "    '''\n",
    "    This function creates the tables and merges them.\n",
    "    '''\n",
    "    sample_info = create_sample_info(data_path + 'merged_data/maf_cohort.pkl')\n",
    "    tp53_muts = create_tp53_muts(sample_info, data_path + 'merged_data/maf_tp53.pkl')\n",
    "    \n",
    "    # We load the copy number table if it is already stored \n",
    "    file = Path(data_path + 'merged_data/copy_number.pkl')\n",
    "    if file.is_file():\n",
    "        copy_number_state =  pd.read_pickle(data_path + 'merged_data/copy_number.pkl')\n",
    "        copy_number_state = copy_number_state.rename(columns={'tcn': 'tp53_tcn',\n",
    "                                                          'mcn': 'tp53_mcn',\n",
    "                                                          'lcn': 'tp53_lcn',\n",
    "                                                          'seg_length': 'tp53_seg_length',\n",
    "                                                          'cn_state':'tp53_cn_state'})\n",
    "    else:\n",
    "        copy_number_state = create_copy_number_state(sample_info, data_path + 'impact-facets-tp53/raw/default_qc_pass.gene_level_TP53.txt')\n",
    "    computed_metrics = create_computed_metrics(data_path + 'merged_data/maf_cohort.pkl')\n",
    "    \n",
    "    # We first merge sample_info and tp53_muts because they have the same list of keys\n",
    "    master_file = pd.merge(sample_info, tp53_muts, on=['Tumor_Id'])\n",
    "    #For copy_number_state we have to do a right join because it contains less Tumor_Ids\n",
    "    master_file = pd.merge(master_file, copy_number_state, on=['Sample_Id'])\n",
    "    # Finally we merge the computedmetrics table\n",
    "    master_file = pd.merge(master_file, computed_metrics, on=['Sample_Id'])\n",
    "    \n",
    "    # We filter out the sample duplicates\n",
    "    master_file = master_file.drop_duplicates('Sample_Id')\n",
    "    \n",
    "    # At this step we need to remove samples that comes from same tumor but different normal sample\n",
    "    # BUT this step makes us loose important clinical information for some samples\n",
    "    # So we wil spread the Patient_Id and Cancer_Type by front and backpropagating the non-NaN values\n",
    "    master_file['Patient_Id'] = master_file.Tumor_Id.str[:9]\n",
    "    master_file['Cancer_Type'] = master_file[['Patient_Id','Cancer_Type']].groupby(['Patient_Id']).bfill().ffill()\n",
    "    \n",
    "    # Then we filter out samples with the same Tumor_Id but different Sample_Id with a filter function\n",
    "    master_file = normal_samp_duplicates_filter(master_file, 'Sample_Id', 'purity')\n",
    "    master_file = normal_samp_duplicates_filter(master_file, 'Sample_Id', 'purity')\n",
    "    \n",
    "    #We compute the expected number of copies of tp53 mutations\n",
    "    master_file = create_copies_tp53_muts(master_file)\n",
    "    \n",
    "    master_file['tp53_residual_1'] = master_file['tp53_tcn'] - master_file['tp53_exp_nb_1']\n",
    "    master_file['tp53_residual_2'] = master_file['tp53_tcn'] - master_file['tp53_exp_nb_2']\n",
    "    master_file['tp53_residual_3'] = master_file['tp53_tcn'] - master_file['tp53_exp_nb_3']\n",
    "    master_file['tp53_residual_4'] = master_file['tp53_tcn'] - master_file['tp53_exp_nb_4']\n",
    "    master_file['tp53_residual_5'] = master_file['tp53_tcn'] - master_file['tp53_exp_nb_5']\n",
    "\n",
    "    #Finally we add the subgroup columns defined in Part 5\n",
    "    master_file['tp53_first_group'] = master_file.apply(cn_group_cond, axis = 1)\n",
    "    master_file['tp53_group'] = master_file.apply(mut_cn_group_cond, axis = 1)\n",
    "    master_file['tp53_res_group'] = master_file.apply(tp53_residual_group, axis = 1)\n",
    "    \n",
    "    # We add Genome Instability columns\n",
    "    # Genome Instability columns computed from arm_level file\n",
    "    arm_level = pd.read_csv(data_path + 'impact-facets-tp53/raw/default_qc_pass.arm_level.txt', sep='\\t')\n",
    "    arm_level['chr'] = arm_level.arm.str.extract('(\\d+)')\n",
    "    print('checkpoint_1')\n",
    "    #master_file['chr_comput'] = master_file.apply(chr_computations, axis=1)\n",
    "    print('checkpoint_2')\n",
    "    chr_metrics =  pd.read_pickle(data_path + 'merged_data/chr_metrics_new.pkl')\n",
    "    master_file = pd.merge(master_file, chr_metrics, on=['Sample_Id'])\n",
    "    \n",
    "    print('checkpoint_3')\n",
    "    \n",
    "    # Grouping the Variant Classificationb into 3 Classes\n",
    "    master_file['tp53_vc_group_1'] = master_file.apply(vc_group_cond_1, axis = 1)\n",
    "    master_file['tp53_vc_group_2'] = master_file.apply(vc_group_cond_2, axis = 1)\n",
    "    master_file['tp53_vc_group_3'] = master_file.apply(vc_group_cond_3, axis = 1)\n",
    "    master_file['tp53_vc_group_4'] = master_file.apply(vc_group_cond_4, axis = 1)\n",
    "    master_file['tp53_vc_group_5'] = master_file.apply(vc_group_cond_5, axis = 1)\n",
    "    return master_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "checkpoint_1\n",
      "checkpoint_2\n",
      "checkpoint_3\n",
      "CPU times: user 7min 14s, sys: 1.86 s, total: 7min 16s\n",
      "Wall time: 7min 17s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "master_file = create_master()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tp53_res_group</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>no_tp53_res</th>\n",
       "      <td>9146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tp53_res</th>\n",
       "      <td>4284</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>uncertain</th>\n",
       "      <td>828</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                count\n",
       "tp53_res_group       \n",
       "no_tp53_res      9146\n",
       "tp53_res         4284\n",
       "uncertain         828"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(master_file, 'tp53_res_group', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tp53_group</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0_HETLOSS</th>\n",
       "      <td>2827</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_WILD_TYPE</th>\n",
       "      <td>1457</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;1muts</th>\n",
       "      <td>697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;=1_LOSS</th>\n",
       "      <td>3956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>&gt;=1_cnLOH</th>\n",
       "      <td>5026</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>HOMDEL</th>\n",
       "      <td>295</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             count\n",
       "tp53_group        \n",
       "0_HETLOSS     2827\n",
       "1_WILD_TYPE   1457\n",
       ">1muts         697\n",
       ">=1_LOSS      3956\n",
       ">=1_cnLOH     5026\n",
       "HOMDEL         295"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(master_file, 'tp53_group', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Number of samples: 29259\n",
      "Number of patients: 27021\n",
      "Number of tp53 positive samples: 12731\n",
      "Number of tp53 positive patients: 11885\n",
      "Number of samples with missing wgd : 2092\n",
      "Number of samples with missing cf : 1746\n",
      "Number of samples with missing max_vaf : 1421\n",
      "Number of samples with missing Cn state: 517\n",
      "Number of samples with missing Sample_Type : 104\n"
     ]
    }
   ],
   "source": [
    "print('Number of samples: ' + str(len(set(master_file.Tumor_Id))))\n",
    "print('Number of patients: ' + str(len(set(master_file.Patient_Id))))\n",
    "print('Number of tp53 positive samples: ' + str( len(set(master_file.Tumor_Id)) - master_file.tp53_key_1.isna().sum()))\n",
    "print('Number of tp53 positive patients: ' + str( len(set(master_file.Patient_Id)) - master_file.drop_duplicates('Patient_Id').tp53_key_1.isna().sum()))\n",
    "print('Number of samples with missing wgd : ' + str(master_file.wgd.isna().sum()))\n",
    "print('Number of samples with missing cf : ' + str(master_file.cf.isna().sum()))\n",
    "print('Number of samples with missing max_vaf : ' + str(master_file.max_vaf.isna().sum()))\n",
    "print('Number of samples with missing Cn state: ' + str(master_file.tp53_cn_state.isna().sum()))\n",
    "print('Number of samples with missing Sample_Type : ' + str(master_file.Sample_Type.isna().sum()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Saving to pickle File\n",
    "master_file.to_pickle(data_path + 'merged_data/master_file.pkl')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11527"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master = load_clean_up_master(data_path + 'merged_data/master_file.pkl')\n",
    "len(list(master[master['tp53_count']>=1]['Sample_Id']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "3"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "max(1,2,3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "count    29202.000000\n",
       "mean         0.495941\n",
       "std          0.207579\n",
       "min          0.018483\n",
       "25%          0.319273\n",
       "50%          0.469923\n",
       "75%          0.658398\n",
       "max          0.989434\n",
       "Name: purity, dtype: float64"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_file.purity.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {},
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
