{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <span style='font-family:\"Times New Roman\"'> <span styel=''> **MASTER FILE CREATION**\n",
    "\n",
    "*Emile Cohen* \n",
    "    \n",
    "*May 2020*\n",
    "\n",
    "**Goal:** In this Notebook, we create a master file that summarizes all useful information.\n",
    "\n",
    "The Notebook is divided in 4 parts, representing the four parts of our Master file:\n",
    "   \n",
    "* **1. Patient/Sample Information**\n",
    "* **2. TP53 Mutations**\n",
    "* **3. TP53 Copy Numbers**\n",
    "* **4. TP53 Computed Metrics**\n",
    "* **5. Subgroup columns creation**\n",
    "* **6. Merge tables**\n",
    "\n",
    "**NB1:** In each part, you must run the cells from the begining in order to initialize the variables\n",
    "\n",
    "**NB2:** In order to launch the last script (Merge Tables), you have to define the functions in each part.\n",
    "\n",
    "**NB3:** All functions used for the plots are located in utils/custom_tools.py\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Setup environment... done!\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "<span style=\"color:green\">✅ Working on **mskimpact_env** conda environment.</span>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%run -i '../../utils/setup_environment.ipy'\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "from scipy.stats import fisher_exact, ranksums, chi2, norm\n",
    "from statsmodels.sandbox.stats.multicomp import multipletests\n",
    "import matplotlib.gridspec as gridspec\n",
    "import pickle\n",
    "\n",
    "data_path = '../../data/'\n",
    "data_wgd = data_path + 'impact-facets-tp53/processed/wgd/'\n",
    "data_no_wgd = data_path + 'impact-facets-tp53/processed/no_wgd/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# first we load all files for WGD\n",
    "maf_cohort_wgd = pd.read_csv(data_wgd + 'maf_cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_wgd = pd.read_csv(data_wgd + 'cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_wgd = pd.read_csv(data_wgd + 'arm_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_wgd = pd.read_csv(data_wgd + 'gene_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We load all files for non WGD\n",
    "\n",
    "maf_cohort_nowgd = pd.read_csv(data_no_wgd + 'maf_cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_nowgd = pd.read_csv(data_no_wgd + 'cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_nowgd = pd.read_csv(data_no_wgd + 'arm_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_nowgd = pd.read_csv(data_no_wgd + 'gene_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "def removing_germ_na_mut(maf_cohort: pd.DataFrame):\n",
    "    '''\n",
    "    This function is meant to remove non TP53 GERMLINE mutations from the mafs, to have only SOMATIC mut \n",
    "    except for TP53 muts where we allow GERMLINE status.\n",
    "    '''\n",
    "    maf_cohort_final = maf_cohort[maf_cohort['Mutation_Status'] != 'UNKNOWN']\n",
    "\n",
    "    key_list = list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] != 'TP53'][maf_cohort_final['Mutation_Status'] != 'GERMLINE'].sample_mut_key)\n",
    "    key_list = key_list + list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] == 'TP53'].sample_mut_key)\n",
    "    maf_cohort_final = maf_cohort_final[maf_cohort_final.sample_mut_key.isin(key_list)]\n",
    "    \n",
    "    return maf_cohort_final\n",
    "\n",
    "maf_cohort_wgd = removing_germ_na_mut(maf_cohort_wgd)\n",
    "maf_cohort_nowgd = removing_germ_na_mut(maf_cohort_nowgd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load clinical data\n",
    "clinical_data = pd.read_csv(data_path + 'cbioportal/raw/mskimpact_clinical_data-2.tsv', sep= '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filtering the clinical data\n",
    "samples_wgd = list(set(cohort_wgd.tumor_sample))\n",
    "samples_nowgd = list(set(cohort_nowgd.tumor_sample))\n",
    "\n",
    "clinical_wgd = clinical_data[clinical_data['Sample ID'].isin(samples_wgd)]\n",
    "clinical_nowgd = clinical_data[clinical_data['Sample ID'].isin(samples_nowgd)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IMPORTANT: Parameter Definition\n",
    "In this script we can create two master files: one for WGD samples and one for non WGD samples.\n",
    "So this parameter allows to select the type of master you want.\n",
    "On top of that you can specify where you want to store the file you created"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "# cohort_type must be 'wgd' or 'no_wgd'\n",
    "cohort_type = 'wgd'\n",
    "\n",
    "output_path = data_path + 'impact-facets-tp53/processed/'\n",
    "\n",
    "total_output = output_path + cohort_type + '/'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "# Patient/Sample Information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we focus on clinical information exported from CbioPortal.\n",
    "\n",
    "The following columns are selected:\n",
    "* Sample_Id\n",
    "* Tumor_Id\n",
    "* Patient_Id\n",
    "* Patient Current Age\n",
    "* Cancer_Type\n",
    "* Cancer_Type_Detailed\n",
    "* Sample_Type\n",
    "* purity\n",
    "* ploidy\n",
    "* Overall Survival Status\n",
    "* Overall Survival (Months)\n",
    "* MSI Score\n",
    "* MSI Type\n",
    "* Tumor Mutational Burden"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_sample_info(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    cohort_filt = cohort[['sample_id', 'tumor_sample', 'patient', 'ploidy']]\n",
    "    clinical_filt = clinical[['Sample ID', 'Patient Current Age', 'Sex', 'Cancer Type', 'Cancer Type Detailed', 'Sample Type',\n",
    "                  'Overall Survival (Months)', 'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "    purity = maf_cohort.drop_duplicates('Tumor_Sample_Barcode')[['Tumor_Sample_Barcode', 'purity']]\n",
    "\n",
    "    # Merging these files\n",
    "    sample_info = pd.merge(cohort_filt, purity, how='left', left_on='tumor_sample', right_on='Tumor_Sample_Barcode')\n",
    "    sample_info =  pd.merge(sample_info, clinical_filt, left_on='tumor_sample', right_on='Sample ID').drop(['Tumor_Sample_Barcode', 'Sample ID'], axis=1)\n",
    "\n",
    "    sample_info = sample_info[['sample_id', 'tumor_sample', 'patient', 'Cancer Type', 'Cancer Type Detailed',\n",
    "                               'Patient Current Age', 'Sex','Sample Type', 'purity','ploidy','Overall Survival (Months)', \n",
    "                               'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "\n",
    "    sample_info.columns = ['Sample_Id', 'Tumor_Id', 'Patient_Id','Cancer_Type', 'Cancer_Type_Detailed', 'Patient_Current_Age',\n",
    "                          'Sex', 'Sample_Type', 'purity', 'ploidy', 'Overall_Survival_Months', 'Overall_Survival_Status',\n",
    "                          'MSI_Score', 'MSI_Type', 'TMB_Score']\n",
    "    \n",
    "    return sample_info\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sex</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>Overall_Survival_Months</th>\n",
       "      <th>Overall_Survival_Status</th>\n",
       "      <th>MSI_Score</th>\n",
       "      <th>MSI_Type</th>\n",
       "      <th>TMB_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>P-0025956</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>71.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>3.496971</td>\n",
       "      <td>3.584</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.391316</td>\n",
       "      <td>2.871793</td>\n",
       "      <td>14.137</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Neuroendocrine Carcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.865628</td>\n",
       "      <td>3.115253</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>2.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>3.136841</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.82</td>\n",
       "      <td>Stable</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Colon Adenocarcinoma</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.351778</td>\n",
       "      <td>3.727190</td>\n",
       "      <td>34.060</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.45</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6362</th>\n",
       "      <td>P-0050671-T01-IM6_P-0050671-N01-IM6</td>\n",
       "      <td>P-0050671-T01-IM6</td>\n",
       "      <td>P-0050671</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Invasive Breast Carcinoma</td>\n",
       "      <td>34.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.300360</td>\n",
       "      <td>3.776265</td>\n",
       "      <td>2.005</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Stable</td>\n",
       "      <td>7.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6363</th>\n",
       "      <td>P-0009110-T02-IM6_P-0009110-N01-IM6</td>\n",
       "      <td>P-0009110-T02-IM6</td>\n",
       "      <td>P-0009110</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>57.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.239690</td>\n",
       "      <td>3.480349</td>\n",
       "      <td>49.414</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>30.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6364</th>\n",
       "      <td>P-0050668-T01-IM6_P-0050668-N01-IM6</td>\n",
       "      <td>P-0050668-T01-IM6</td>\n",
       "      <td>P-0050668</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Squamous Cell Carcinoma</td>\n",
       "      <td>73.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.211949</td>\n",
       "      <td>3.144416</td>\n",
       "      <td>1.710</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>6.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6365</th>\n",
       "      <td>P-0050675-T01-IM6_P-0050675-N01-IM6</td>\n",
       "      <td>P-0050675-T01-IM6</td>\n",
       "      <td>P-0050675</td>\n",
       "      <td>Cancer of Unknown Primary</td>\n",
       "      <td>Cancer of Unknown Primary</td>\n",
       "      <td>64.0</td>\n",
       "      <td>Female</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.386871</td>\n",
       "      <td>2.760248</td>\n",
       "      <td>1.973</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>1.32</td>\n",
       "      <td>Stable</td>\n",
       "      <td>23.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6366</th>\n",
       "      <td>P-0050660-T01-IM6_P-0050660-N01-IM6</td>\n",
       "      <td>P-0050660-T01-IM6</td>\n",
       "      <td>P-0050660</td>\n",
       "      <td>Bladder Cancer</td>\n",
       "      <td>Bladder Urothelial Carcinoma</td>\n",
       "      <td>70.0</td>\n",
       "      <td>Male</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.683828</td>\n",
       "      <td>3.459433</td>\n",
       "      <td>2.038</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.41</td>\n",
       "      <td>Stable</td>\n",
       "      <td>6.1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6367 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6  P-0025956-T01-IM6  P-0025956   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6  P-0036909-T01-IM6  P-0036909   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6  P-0023546-T01-IM6  P-0023546   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6  P-0023546-T02-IM6  P-0023546   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6  P-0018837-T01-IM6  P-0018837   \n",
       "...                                   ...                ...        ...   \n",
       "6362  P-0050671-T01-IM6_P-0050671-N01-IM6  P-0050671-T01-IM6  P-0050671   \n",
       "6363  P-0009110-T02-IM6_P-0009110-N01-IM6  P-0009110-T02-IM6  P-0009110   \n",
       "6364  P-0050668-T01-IM6_P-0050668-N01-IM6  P-0050668-T01-IM6  P-0050668   \n",
       "6365  P-0050675-T01-IM6_P-0050675-N01-IM6  P-0050675-T01-IM6  P-0050675   \n",
       "6366  P-0050660-T01-IM6_P-0050660-N01-IM6  P-0050660-T01-IM6  P-0050660   \n",
       "\n",
       "                     Cancer_Type               Cancer_Type_Detailed  \\\n",
       "0     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "1     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "2                Prostate Cancer  Prostate Neuroendocrine Carcinoma   \n",
       "3                Prostate Cancer            Prostate Adenocarcinoma   \n",
       "4              Colorectal Cancer               Colon Adenocarcinoma   \n",
       "...                          ...                                ...   \n",
       "6362               Breast Cancer          Invasive Breast Carcinoma   \n",
       "6363  Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "6364  Non-Small Cell Lung Cancer       Lung Squamous Cell Carcinoma   \n",
       "6365   Cancer of Unknown Primary          Cancer of Unknown Primary   \n",
       "6366              Bladder Cancer       Bladder Urothelial Carcinoma   \n",
       "\n",
       "      Patient_Current_Age     Sex Sample_Type    purity    ploidy  \\\n",
       "0                    71.0  Female     Primary  0.273767  3.496971   \n",
       "1                    47.0  Female  Metastasis  0.391316  2.871793   \n",
       "2                    50.0    Male     Primary  0.865628  3.115253   \n",
       "3                    50.0    Male     Primary  0.312907  3.136841   \n",
       "4                    60.0    Male     Primary  0.351778  3.727190   \n",
       "...                   ...     ...         ...       ...       ...   \n",
       "6362                 34.0  Female  Metastasis  0.300360  3.776265   \n",
       "6363                 57.0  Female  Metastasis  0.239690  3.480349   \n",
       "6364                 73.0    Male     Primary  0.211949  3.144416   \n",
       "6365                 64.0  Female     Primary  0.386871  2.760248   \n",
       "6366                 70.0    Male     Primary  0.683828  3.459433   \n",
       "\n",
       "      Overall_Survival_Months Overall_Survival_Status  MSI_Score MSI_Type  \\\n",
       "0                       3.584                DECEASED       0.00   Stable   \n",
       "1                      14.137                  LIVING       0.37   Stable   \n",
       "2                       4.800                DECEASED       2.37   Stable   \n",
       "3                       4.800                DECEASED       0.82   Stable   \n",
       "4                      34.060                  LIVING       0.45   Stable   \n",
       "...                       ...                     ...        ...      ...   \n",
       "6362                    2.005                  LIVING       0.05   Stable   \n",
       "6363                   49.414                  LIVING       0.00   Stable   \n",
       "6364                    1.710                  LIVING       0.00   Stable   \n",
       "6365                    1.973                  LIVING       1.32   Stable   \n",
       "6366                    2.038                  LIVING       0.41   Stable   \n",
       "\n",
       "      TMB_Score  \n",
       "0           5.3  \n",
       "1           3.5  \n",
       "2           3.5  \n",
       "3           2.6  \n",
       "4           5.3  \n",
       "...         ...  \n",
       "6362        7.0  \n",
       "6363       30.7  \n",
       "6364        6.1  \n",
       "6365       23.7  \n",
       "6366        6.1  \n",
       "\n",
       "[6367 rows x 15 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sample_info = create_sample_info(cohort=cohort_type)\n",
    "sample_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Mutations\n",
    "In this part, we focus on tp53 mutation information.\n",
    "\n",
    "We gather all mutations per sample, and split it into different columns. We have the following columns:\n",
    "* Tumor_Id\t\n",
    "* key_1 (2,3,4,5) --> Mutation key allowing to filter duplicates\n",
    "* vc_1 (2,3,4,5) --> Variant Classification\n",
    "* ccf_1 (2,3,4,5) --> Cancer Cell Fraction of the mutation\n",
    "* vaf_1 (2,3,4,5) --> Variant Allele Frequency of the mutation\n",
    "* HGVSp_1 (2,3,4,5) --> protein change\n",
    "* spot_1 (2,3,4,5) --> Integer that defines the spot of the tp53 mutation\n",
    "* tp53_count --> Number of tp53 mutations of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def f_(x):\n",
    "    # This function helps us to group mutations together in a single cell per patient\n",
    "    return pd.DataFrame(dict(Tumor_Sample_Barcode = x['Tumor_Sample_Barcode'],  \n",
    "                        muts = \"%s\" % ','.join(x['sample_mut_key_vc_ccf_vaf_hgv_spot_vt'])))\n",
    "\n",
    "def count_tp53_muts(x):\n",
    "    count = 0\n",
    "    for i in range(1,6):\n",
    "        if x['tp53_key_' + str(i)]:\n",
    "            count+= 1\n",
    "    return count\n",
    "\n",
    "# WARNING: THis function needs sample_info to work\n",
    "def create_tp53_muts(cohort:str, sample_info):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        print('yes')\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    '''\n",
    "    This function aims to gather all tp53 mutation characteristics.\n",
    "    For each sample we gather the tp53 mutations and their characteristics for all patients.\n",
    "    '''\n",
    "    # We load the  table created in maf_tp53_creation.ipynb\n",
    "    maf_tp53 = maf_cohort[maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    maf_tp53['mut_spot'] = maf_tp53.HGVSp.str.extract('(\\d+)')\n",
    "\n",
    "    # We select only intresting columns\n",
    "    maf_tp53_filtered = maf_tp53[['Tumor_Sample_Barcode','sample_mut_key', 'Variant_Classification',\\\n",
    "                                        'ccf_expected_copies', 't_var_freq', 'HGVSp','mut_spot', 'Variant_Type' ]]\n",
    "\n",
    "    # Let's Merge mut_key,Variant_classification, CF, CCF, and VAF to gather them\n",
    "    maf_tp53_filtered['sample_mut_key_vc_ccf_vaf_hgv_spot_vt'] = maf_tp53_filtered.apply(lambda x: str(x.sample_mut_key)+'%'+str(x.Variant_Classification)+'%'+str(x.ccf_expected_copies)+'%'+str(x.t_var_freq)+'%'+str(x.HGVSp)+'%'+str(x.mut_spot)+'%'+str(x.Variant_Type), axis=1)\n",
    "\n",
    "    # We Select important columns\n",
    "    final = maf_tp53_filtered[['Tumor_Sample_Barcode', 'sample_mut_key_vc_ccf_vaf_hgv_spot_vt']]\n",
    "    # We groupby Patient_Id and apply the function above to group mutations\n",
    "    final = final.groupby(['Tumor_Sample_Barcode'], sort=False).apply(f_)\n",
    "\n",
    "    # We separate the different mutations into 5 different columns (5 is the max number of tp53 mutations in our cohort)\n",
    "    final[['mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5']] = final.muts.str.split(',', expand=True)\n",
    "    #final = final.drop(['mut_key_6'],axis=1)\n",
    "    # Split the columns into mut_key_ and vc_\n",
    "    final[['tp53_key_1','tp53_vc_1','tp53_ccf_1','tp53_vaf_1','tp53_HGVSp_1', 'tp53_spot_1','tp53_vt_1']] = final.mut_key_1.str.split('%', expand=True)\n",
    "    final[['tp53_key_2','tp53_vc_2','tp53_ccf_2','tp53_vaf_2','tp53_HGVSp_2', 'tp53_spot_2','tp53_vt_2']] = final.mut_key_2.str.split('%', expand=True)\n",
    "    final[['tp53_key_3','tp53_vc_3','tp53_ccf_3','tp53_vaf_3','tp53_HGVSp_3', 'tp53_spot_3','tp53_vt_3']] = final.mut_key_3.str.split('%', expand=True)\n",
    "    final[['tp53_key_4','tp53_vc_4','tp53_ccf_4','tp53_vaf_4','tp53_HGVSp_4', 'tp53_spot_4','tp53_vt_4']] = final.mut_key_4.str.split('%', expand=True)\n",
    "    final[['tp53_key_5','tp53_vc_5','tp53_ccf_5','tp53_vaf_5','tp53_HGVSp_5', 'tp53_spot_5','tp53_vt_5']] = final.mut_key_5.str.split('%', expand=True)\n",
    "\n",
    "    # We remove the muts column\n",
    "    final = final.drop(['muts','mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5'], axis=1)\n",
    "\n",
    "    # We remove duplicates\n",
    "    final = final.drop_duplicates('Tumor_Sample_Barcode')\n",
    "\n",
    "    # We add the cohort patients that are not tp53 positive\n",
    "    #First we create a dataframe with all missing samples\n",
    "\n",
    "    cohort_samples = set(sample_info.Tumor_Id)\n",
    "    final_samples = set(final.Tumor_Sample_Barcode)\n",
    "    missing_samp = pd.DataFrame(cohort_samples - final_samples, columns = ['Tumor_Sample_Barcode'])\n",
    "    #Then we append the two datframe\n",
    "    final = final.append(missing_samp)\n",
    "\n",
    "    # We rename the Tumor_Sample_Barcode column to have the same key as in other datframes\n",
    "    final = final.rename(columns={'Tumor_Sample_Barcode': 'Tumor_Id'})\n",
    "\n",
    "    # We add a last column tp53_count that represents the number of tp53 mutations per sample\n",
    "    final = final.where(final.notnull(), None)\n",
    "    final['tp53_count'] = final.apply(count_tp53_muts, axis = 1)\n",
    "\n",
    "    # We change the type of vafs column to float64 instead of strings\n",
    "    final = final.astype({'tp53_vaf_1': 'float64', 'tp53_vaf_2': 'float64', 'tp53_vaf_3': 'float64', 'tp53_vaf_4': 'float64', 'tp53_vaf_5': 'float64',\n",
    "                       'tp53_ccf_1': 'float64', 'tp53_ccf_2': 'float64', 'tp53_ccf_3': 'float64', 'tp53_ccf_4': 'float64', 'tp53_ccf_5': 'float64'})\n",
    "\n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_vt_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_vt_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_vt_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_vt_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_vt_5</th>\n",
       "      <th>tp53_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909-T01-IM6-17_7577121_G_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.798</td>\n",
       "      <td>0.312169</td>\n",
       "      <td>p.Arg273Cys</td>\n",
       "      <td>273</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546-T01-IM6-17_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.845070</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546-T02-IM6-17_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.636735</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837-T01-IM6-17_7578406_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.325843</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>175</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>P-0019444-T01-IM6</td>\n",
       "      <td>P-0019444-T01-IM6-17_7578461_C_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.169002</td>\n",
       "      <td>p.Val157Phe</td>\n",
       "      <td>157</td>\n",
       "      <td>SNP</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2134</th>\n",
       "      <td>P-0049354-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2135</th>\n",
       "      <td>P-0027276-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2136</th>\n",
       "      <td>P-0001014-T01-IM3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2137</th>\n",
       "      <td>P-0003210-T01-IM5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2138</th>\n",
       "      <td>P-0000386-T02-IM5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6367 rows × 37 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Tumor_Id                        tp53_key_1          tp53_vc_1  \\\n",
       "8     P-0036909-T01-IM6  P-0036909-T01-IM6-17_7577121_G_A  Missense_Mutation   \n",
       "13    P-0023546-T01-IM6  P-0023546-T01-IM6-17_7578442_T_C  Missense_Mutation   \n",
       "16    P-0023546-T02-IM6  P-0023546-T02-IM6-17_7578442_T_C  Missense_Mutation   \n",
       "21    P-0018837-T01-IM6  P-0018837-T01-IM6-17_7578406_C_T  Missense_Mutation   \n",
       "48    P-0019444-T01-IM6  P-0019444-T01-IM6-17_7578461_C_A  Missense_Mutation   \n",
       "...                 ...                               ...                ...   \n",
       "2134  P-0049354-T01-IM6                              None               None   \n",
       "2135  P-0027276-T01-IM6                              None               None   \n",
       "2136  P-0001014-T01-IM3                              None               None   \n",
       "2137  P-0003210-T01-IM5                              None               None   \n",
       "2138  P-0000386-T02-IM5                              None               None   \n",
       "\n",
       "      tp53_ccf_1  tp53_vaf_1 tp53_HGVSp_1 tp53_spot_1 tp53_vt_1 tp53_key_2  \\\n",
       "8          0.798    0.312169  p.Arg273Cys         273       SNP       None   \n",
       "13         0.933    0.845070  p.Tyr163Cys         163       SNP       None   \n",
       "16         1.000    0.636735  p.Tyr163Cys         163       SNP       None   \n",
       "21         1.000    0.325843  p.Arg175His         175       SNP       None   \n",
       "48         1.000    0.169002  p.Val157Phe         157       SNP       None   \n",
       "...          ...         ...          ...         ...       ...        ...   \n",
       "2134         NaN         NaN         None        None      None       None   \n",
       "2135         NaN         NaN         None        None      None       None   \n",
       "2136         NaN         NaN         None        None      None       None   \n",
       "2137         NaN         NaN         None        None      None       None   \n",
       "2138         NaN         NaN         None        None      None       None   \n",
       "\n",
       "     tp53_vc_2  tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2 tp53_spot_2 tp53_vt_2  \\\n",
       "8         None         NaN         NaN         None        None      None   \n",
       "13        None         NaN         NaN         None        None      None   \n",
       "16        None         NaN         NaN         None        None      None   \n",
       "21        None         NaN         NaN         None        None      None   \n",
       "48        None         NaN         NaN         None        None      None   \n",
       "...        ...         ...         ...          ...         ...       ...   \n",
       "2134      None         NaN         NaN         None        None      None   \n",
       "2135      None         NaN         NaN         None        None      None   \n",
       "2136      None         NaN         NaN         None        None      None   \n",
       "2137      None         NaN         NaN         None        None      None   \n",
       "2138      None         NaN         NaN         None        None      None   \n",
       "\n",
       "     tp53_key_3 tp53_vc_3  tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3 tp53_spot_3  \\\n",
       "8          None      None         NaN         NaN         None        None   \n",
       "13         None      None         NaN         NaN         None        None   \n",
       "16         None      None         NaN         NaN         None        None   \n",
       "21         None      None         NaN         NaN         None        None   \n",
       "48         None      None         NaN         NaN         None        None   \n",
       "...         ...       ...         ...         ...          ...         ...   \n",
       "2134       None      None         NaN         NaN         None        None   \n",
       "2135       None      None         NaN         NaN         None        None   \n",
       "2136       None      None         NaN         NaN         None        None   \n",
       "2137       None      None         NaN         NaN         None        None   \n",
       "2138       None      None         NaN         NaN         None        None   \n",
       "\n",
       "     tp53_vt_3 tp53_key_4 tp53_vc_4  tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4  \\\n",
       "8         None       None      None         NaN         NaN         None   \n",
       "13        None       None      None         NaN         NaN         None   \n",
       "16        None       None      None         NaN         NaN         None   \n",
       "21        None       None      None         NaN         NaN         None   \n",
       "48        None       None      None         NaN         NaN         None   \n",
       "...        ...        ...       ...         ...         ...          ...   \n",
       "2134      None       None      None         NaN         NaN         None   \n",
       "2135      None       None      None         NaN         NaN         None   \n",
       "2136      None       None      None         NaN         NaN         None   \n",
       "2137      None       None      None         NaN         NaN         None   \n",
       "2138      None       None      None         NaN         NaN         None   \n",
       "\n",
       "     tp53_spot_4 tp53_vt_4 tp53_key_5 tp53_vc_5  tp53_ccf_5  tp53_vaf_5  \\\n",
       "8           None      None       None      None         NaN         NaN   \n",
       "13          None      None       None      None         NaN         NaN   \n",
       "16          None      None       None      None         NaN         NaN   \n",
       "21          None      None       None      None         NaN         NaN   \n",
       "48          None      None       None      None         NaN         NaN   \n",
       "...          ...       ...        ...       ...         ...         ...   \n",
       "2134        None      None       None      None         NaN         NaN   \n",
       "2135        None      None       None      None         NaN         NaN   \n",
       "2136        None      None       None      None         NaN         NaN   \n",
       "2137        None      None       None      None         NaN         NaN   \n",
       "2138        None      None       None      None         NaN         NaN   \n",
       "\n",
       "     tp53_HGVSp_5 tp53_spot_5 tp53_vt_5  tp53_count  \n",
       "8            None        None      None           1  \n",
       "13           None        None      None           1  \n",
       "16           None        None      None           1  \n",
       "21           None        None      None           1  \n",
       "48           None        None      None           1  \n",
       "...           ...         ...       ...         ...  \n",
       "2134         None        None      None           0  \n",
       "2135         None        None      None           0  \n",
       "2136         None        None      None           0  \n",
       "2137         None        None      None           0  \n",
       "2138         None        None      None           0  \n",
       "\n",
       "[6367 rows x 37 columns]"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_tp53_muts(cohort=cohort_type, sample_info=sample_info)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Copy Numbers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we gather the information from gene_level table.\n",
    "We creaste the following columns:\n",
    "* Sample_Id \n",
    "* tcn --> total copy number\n",
    "* mcn --> major copy number\n",
    "* lcn --> lower copy number\n",
    "* seg_length --> length of the segment\n",
    "* cn_state --> copy number state\n",
    "* cf --> Cell fraction of the cn_state\n",
    "* wgd --> Wholde Genome Doubling (True or False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "# WARNING: THis function needs sample_info to work\n",
    "def create_copy_number_state(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    # We want TP53 locus so we have to filter the gene\n",
    "    gene_level = gene_level[gene_level['gene'] == 'TP53']\n",
    "\n",
    "    gene_level['Tumor_Id'] = gene_level['sample'].str[:17]\n",
    "    gene_level_subset = gene_level[['sample','tcn','mcn','lcn','seg_length','cn_state', 'cf.em']]\n",
    "    \n",
    "    # We rename the cf.em column \n",
    "    gene_level_subset = gene_level_subset.rename(columns={'cf.em': 'tp53_cf', \n",
    "                                                          'sample':'Sample_Id',\n",
    "                                                          'tcn': 'tp53_tcn',\n",
    "                                                          'mcn': 'tp53_mcn',\n",
    "                                                          'lcn': 'tp53_lcn',\n",
    "                                                          'seg_length': 'tp53_seg_length',\n",
    "                                                          'cn_state':'tp53_cn_state'})\n",
    "    \n",
    "    # We add WGD information\n",
    "    wgd = cohort[['sample_id', 'wgd']]\n",
    "    \n",
    "    final = pd.merge(gene_level_subset, wgd, left_on='Sample_Id', right_on='sample_id').drop(['sample_id'], axis=1)\n",
    "    \n",
    "    \n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 17.9 ms, sys: 2.72 ms, total: 20.6 ms\n",
      "Wall time: 19.1 ms\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>tp53_tcn</th>\n",
       "      <th>tp53_mcn</th>\n",
       "      <th>tp53_lcn</th>\n",
       "      <th>tp53_seg_length</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>tp53_cf</th>\n",
       "      <th>wgd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668431</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>0.169170</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18101929</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.306121</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25250470</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.835504</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25237770</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25231975</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.257400</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6362</th>\n",
       "      <td>P-0050671-T01-IM6_P-0050671-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>46689400</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.300360</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6363</th>\n",
       "      <td>P-0009110-T02-IM6_P-0009110-N01-IM6</td>\n",
       "      <td>4</td>\n",
       "      <td>4.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7438495</td>\n",
       "      <td>CNLOH BEFORE</td>\n",
       "      <td>0.239690</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6364</th>\n",
       "      <td>P-0050668-T01-IM6_P-0050668-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>80668561</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.197170</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6365</th>\n",
       "      <td>P-0050675-T01-IM6_P-0050675-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>21375632</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.376084</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6366</th>\n",
       "      <td>P-0050660-T01-IM6_P-0050660-N01-IM6</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>29371203</td>\n",
       "      <td>TETRAPLOID</td>\n",
       "      <td>0.657819</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6367 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id  tp53_tcn  tp53_mcn  tp53_lcn  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6         3       2.0       1.0   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6         2       2.0       0.0   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6         3       3.0       0.0   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6         2       2.0       0.0   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6         3       3.0       0.0   \n",
       "...                                   ...       ...       ...       ...   \n",
       "6362  P-0050671-T01-IM6_P-0050671-N01-IM6         2       2.0       0.0   \n",
       "6363  P-0009110-T02-IM6_P-0009110-N01-IM6         4       4.0       0.0   \n",
       "6364  P-0050668-T01-IM6_P-0050668-N01-IM6         3       3.0       0.0   \n",
       "6365  P-0050675-T01-IM6_P-0050675-N01-IM6         2       2.0       0.0   \n",
       "6366  P-0050660-T01-IM6_P-0050660-N01-IM6         4       2.0       2.0   \n",
       "\n",
       "      tp53_seg_length        tp53_cn_state   tp53_cf   wgd  \n",
       "0            80668431           LOSS AFTER  0.169170  True  \n",
       "1            18101929          LOSS BEFORE  0.306121  True  \n",
       "2            25250470  CNLOH BEFORE & LOSS  0.835504  True  \n",
       "3            25237770          LOSS BEFORE  0.312907  True  \n",
       "4            25231975  CNLOH BEFORE & LOSS  0.257400  True  \n",
       "...               ...                  ...       ...   ...  \n",
       "6362         46689400          LOSS BEFORE  0.300360  True  \n",
       "6363          7438495         CNLOH BEFORE  0.239690  True  \n",
       "6364         80668561  CNLOH BEFORE & LOSS  0.197170  True  \n",
       "6365         21375632          LOSS BEFORE  0.376084  True  \n",
       "6366         29371203           TETRAPLOID  0.657819  True  \n",
       "\n",
       "[6367 rows x 8 columns]"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time \n",
    "copy_number_info = create_copy_number_state(cohort = cohort_type)\n",
    "copy_number_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computed metrics\n",
    "In this part we define functions to be applied on the master file to compute specific metrics.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genes, mutations and max_vaf\n",
    "* *create_gene_count*: count of mutated genes for the given sample\n",
    "* *create_mut_count*: count of mutations for the given sample\n",
    "* *get_max_vaf*: the maximum of Variant Allele Frequency within all mutation of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# These 2 first functions allow  to count the total number of genes/mutations per sample.\n",
    "def create_gene_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function create the count of genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    \n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "\n",
    "    nb_genes = selected_cohort.groupby('Entrez_Gene_Id').size().shape[0]\n",
    "\n",
    "    return nb_genes\n",
    "\n",
    "def create_mut_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "    \n",
    "    return selected_cohort.shape[0]\n",
    "\n",
    "# The following calculates the max_vaf for a given sample\n",
    "def get_max_vaf(x, maf_cohort):\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    \n",
    "    return selected_cohort['t_var_freq'].max()\n",
    "\n",
    "\n",
    "def create_driver_gene_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function create the count of driver mut genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    \n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['driver'] == True][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "\n",
    "    nb_genes = selected_cohort.groupby('Entrez_Gene_Id').size().shape[0]\n",
    "\n",
    "    return nb_genes\n",
    "\n",
    "def create_driver_mut_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all driver mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['driver'] == True][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "    \n",
    "    return selected_cohort.shape[0]\n",
    "\n",
    "def create_snv_driver_mut_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all driver mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['driver'] == True][maf_cohort['Variant_Type'].isin(['SNP', 'DNP', 'TNP', 'ONP'])][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "    \n",
    "    return selected_cohort.shape[0]\n",
    "\n",
    "def create_indel_driver_mut_count(x, maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all driver mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['driver'] == True][maf_cohort['Variant_Type'].isin(['DEL', 'INS'])][maf_cohort['Hugo_Symbol'] != 'TP53']\n",
    "    \n",
    "    return selected_cohort.shape[0]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genome Instability (frac_genome_aletered, chr_affected by loss, cnkloh, gain )\n",
    "This script is very long to run, so I decide to run it apart and save the results in a pickle fil that we will merge with our master.\n",
    "\n",
    "* For no_wgd: 40 minutes (MacBook Pro 2019)\n",
    "* For wgd: 11 minutes (MacBook Pro 2019)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def condition_CNLOH(x):\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    if x.cn_state in CNLOH:\n",
    "        return 'CNLOH' + x.chr\n",
    "    else:\n",
    "        return 'NO_CNLOH'  + x.chr \n",
    "    \n",
    "def condition_GAIN(x):\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    if x.cn_state in GAIN:\n",
    "        return 'GAIN'+ x.chr\n",
    "    else:\n",
    "        return 'NO_GAIN'+ x.chr\n",
    "\n",
    "def condition_LOSS(x):\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    if x.cn_state in LOSS:\n",
    "        return 'LOSS'+ x.chr\n",
    "    else:\n",
    "        return 'NO_LOSS'+ x.chr\n",
    "\n",
    "def compute_frac_genome(x, arm_level):\n",
    "    Sample_Id = x.Sample_Id\n",
    "    wgd = x.wgd\n",
    "    lookup_table = arm_level[arm_level['sample'] == Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table_altered = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "        \n",
    "    altered_length = lookup_table_altered.cn_length.sum()\n",
    "    total_length = lookup_table.arm_length.sum()\n",
    "    \n",
    "    frac_gen_altered = round(altered_length/total_length,3)\n",
    "    \n",
    "    return frac_gen_altered\n",
    "\n",
    "def compute_frac_genome_without_cnloh(x, arm_level):\n",
    "    Sample_Id = x.Sample_Id\n",
    "    wgd = x.wgd\n",
    "    lookup_table = arm_level[arm_level['sample'] == Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table_altered = lookup_table[~lookup_table['cn_state'].isin(['TETRAPLOID', 'CNLOH BEFORE', 'CNLOH AFTER'])][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table_altered = lookup_table[~lookup_table['cn_state'].isin(['DIPLOID', 'CNLOH', 'CNLOH & GAIN'])][lookup_table['chr'] != '17']\n",
    "        \n",
    "    altered_length = lookup_table_altered.cn_length.sum()\n",
    "    total_length = lookup_table.arm_length.sum()\n",
    "    \n",
    "    frac_gen_altered = round(altered_length/total_length,3)\n",
    "    \n",
    "    return frac_gen_altered\n",
    "\n",
    "\n",
    "# Here is the function that allws to compute genome instability columns\n",
    "def chr_computations(x, arm_level):\n",
    "    CNLOH = ['CNLOH', 'CNLOH BEFORE & LOSS', 'CNLOH AFTER', 'CNLOH BEFORE', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN', 'AMP (LOH)']\n",
    "    LOSS = ['HETLOSS', 'LOSS BEFORE', 'LOSS AFTER', 'HOMDEL', 'LOSS BEFORE & AFTER', 'DOUBLE LOSS AFTER', 'LOSS & GAIN', 'CNLOH BEFORE & LOSS']\n",
    "    GAIN = ['GAIN', 'AMP', 'AMP (BALANCED)', 'LOSS & GAIN', 'CNLOH & GAIN', 'CNLOH BEFORE & GAIN']\n",
    "    wgd=x.wgd\n",
    "    arm_level_samples = list(set(arm_level['sample']))\n",
    "    \n",
    "    if x.Sample_Id not in arm_level_samples:\n",
    "        return ['NaN','NaN','NaN','NaN', 'NaN']\n",
    "    \n",
    "    lookup_table = arm_level[arm_level['sample'] == x.Sample_Id]\n",
    "    lookup_table['chr'] = lookup_table.arm.str.extract('(\\d+)')\n",
    "    \n",
    "    if wgd:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'TETRAPLOID'][lookup_table['chr'] != '17']\n",
    "    else:\n",
    "        lookup_table = lookup_table[lookup_table['cn_state'] != 'DIPLOID'][lookup_table['chr'] != '17']\n",
    "    lookup_table['state_chr'] = lookup_table['cn_state']+lookup_table['chr']\n",
    "    \n",
    "    # If only DIPLOID or TETRAPLOID\n",
    "    if lookup_table.empty:\n",
    "        return [float(0)]*5\n",
    "    \n",
    "    lookup_table['cnloh_chr'] = lookup_table.apply(condition_CNLOH, axis=1)\n",
    "    lookup_table['loss_chr'] = lookup_table.apply(condition_LOSS, axis=1)\n",
    "    lookup_table['gain_chr'] = lookup_table.apply(condition_GAIN, axis=1)\n",
    "\n",
    "    #chr_affected colum\n",
    "    lookup_table_chr = lookup_table.drop_duplicates(subset=['chr'])\n",
    "    chr_affected = len(lookup_table_chr)\n",
    "    \n",
    "    #chr_loss, chr_gain, chr_cnloh columns\n",
    "    lookup_table_cnloh = lookup_table.drop_duplicates(subset=['cnloh_chr'])['cnloh_chr']\n",
    "    lookup_table_loss = lookup_table.drop_duplicates(subset=['loss_chr'])['loss_chr']\n",
    "    lookup_table_gain = lookup_table.drop_duplicates(subset=['gain_chr'])['gain_chr']\n",
    "\n",
    "    chr_loss = len(lookup_table_loss[lookup_table_loss.str.startswith('LOSS')])\n",
    "    chr_gain = len(lookup_table_gain[lookup_table_gain.str.startswith('GAIN')])\n",
    "    chr_cnloh = len(lookup_table_cnloh[lookup_table_cnloh.str.startswith('CNLOH')])\n",
    "    \n",
    "    #frac_gen_altered column\n",
    "    frac_gen_altered = compute_frac_genome(x, arm_level=arm_level)\n",
    "    \n",
    "    # frac_gen_altered_without_cnloh\n",
    "    frac_gen_altered_without_cnloh = compute_frac_genome_without_cnloh(x, arm_level=arm_level)\n",
    "    \n",
    "    return [chr_affected, chr_loss, chr_gain, chr_cnloh, frac_gen_altered, frac_gen_altered_without_cnloh]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You do not need to run the following, excpet if you want to recompute the genome instability metrics.\n",
    "\n",
    "The following cell allows to run the above functions and to save it in a file that we will call in the final merging cell."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "1a6614fc330f431f8df2a64c23fc611b",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=1.0, bar_style='info', max=1.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "ec41519cf8ce4016b5844d51fb605300",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "HBox(children=(FloatProgress(value=0.0, max=6367.0), HTML(value='')))"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "\n",
      "checkpoint 1\n",
      "checkpoint 2\n"
     ]
    }
   ],
   "source": [
    "'''from tqdm import tqdm,tqdm_notebook\n",
    "\n",
    "def compute_genome_instability(cohort:str):\n",
    "    tqdm_notebook().pandas()\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    copy_number_info = create_copy_number_state(cohort = cohort_type)\n",
    "    copy_number_info_ = copy_number_info\n",
    "    copy_number_info_['chr_comput'] = copy_number_info_.progress_apply(chr_computations,arm_level=arm_level, axis=1)\n",
    "    print('checkpoint 1')\n",
    "    copy_number_info[['chr_affected', 'chr_loss', 'chr_gain', 'chr_cnloh', 'frac_genome_altered', 'frac_gen_altered_without_cnloh']] = pd.DataFrame(copy_number_info_.chr_comput.values.tolist(), index= copy_number_info_.index)\n",
    "    print('checkpoint 2')\n",
    "    \n",
    "    return copy_number_info_\n",
    "\n",
    "copy_number_info = compute_genome_instability(cohort=cohort_type)\n",
    "copy_number_info[['Sample_Id','chr_affected', 'chr_loss', 'chr_gain', 'chr_cnloh', 'frac_genome_altered', 'frac_gen_altered_without_cnloh']].to_pickle(total_output + 'gi_metrics_{}.pkl'.format(cohort_type))'''"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## TP53 Residual"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following function needs to be called on the complete master file because it needs info from different parts\n",
    "# It computes the expected number of tp53 mutant copies in a cell\n",
    "def create_copies_tp53_muts(master):\n",
    "    master['tp53_exp_nb_1'] = master.apply(lambda x:(x.tp53_vaf_1 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_2'] = master.apply(lambda x:(x.tp53_vaf_2 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_3'] = master.apply(lambda x:(x.tp53_vaf_3 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_4'] = master.apply(lambda x:(x.tp53_vaf_4 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_5'] = master.apply(lambda x:(x.tp53_vaf_5 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    \n",
    "    return master\n",
    "\n",
    "\n",
    "# The following computes the expected number of copies of tp53 residuals \n",
    "def create_tp53_res(master):\n",
    "    master['tp53_res_1'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_1, axis = 1)\n",
    "    master['tp53_res_2'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_2, axis = 1)\n",
    "    master['tp53_res_3'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_3, axis = 1)\n",
    "    master['tp53_res_4'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_4, axis = 1)\n",
    "    master['tp53_res_5'] = master.apply(lambda x:x.tp53_tcn - x.tp53_exp_nb_5, axis = 1)\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Mutation Types + Variant Type Grouping"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# The following functions allow to group the Mutation Types\n",
    "def vc_group_cond_1(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_1 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_1 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_1 in missense: \n",
    "        if x.tp53_spot_1 in ['273','248','175']: return x.tp53_spot_1\n",
    "        elif x.tp53_spot_1 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_2(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_2 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_2 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_2 in missense: \n",
    "        if x.tp53_spot_2 in ['273','248','175']: return x.tp53_spot_2\n",
    "        elif x.tp53_spot_2 in['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'   \n",
    "def vc_group_cond_3(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_3 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_3 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_3 in missense: \n",
    "        if x.tp53_spot_3 in ['273','248','175']: return x.tp53_spot_3\n",
    "        elif x.tp53_spot_3 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense' \n",
    "def vc_group_cond_4(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_4 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_4 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_4 in missense: \n",
    "        if x.tp53_spot_4 in ['273','248','175']: return x.tp53_spot_4\n",
    "        elif x.tp53_spot_4 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_5(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_5 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_5 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_5 in missense: \n",
    "        if x.tp53_spot_5 in ['273','248','175']: return x.tp53_spot_5\n",
    "        elif x.tp53_spot_5 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "        \n",
    "\n",
    "#############################################################################################################\n",
    "\n",
    "def vt_group_cond_1(x):\n",
    "    indel = ['DEL', 'INS']\n",
    "    snv = ['SNP', 'DNP', 'TNP', 'ONP']\n",
    "    if x.tp53_vt_1 in indel:\n",
    "        return 'INDEL'\n",
    "    elif x.tp53_vt_1 in snv:\n",
    "        return 'SNV'\n",
    "    \n",
    "def vt_group_cond_2(x):\n",
    "    indel = ['DEL', 'INS']\n",
    "    snv = ['SNP', 'DNP', 'TNP', 'ONP']\n",
    "    if x.tp53_vt_2 in indel:\n",
    "        return 'INDEL'\n",
    "    elif x.tp53_vt_2 in snv:\n",
    "        return 'SNV'\n",
    "    \n",
    "def vt_group_cond_3(x):\n",
    "    indel = ['DEL', 'INS']\n",
    "    snv = ['SNP', 'DNP', 'TNP', 'ONP']\n",
    "    if x.tp53_vt_3 in indel:\n",
    "        return 'INDEL'\n",
    "    elif x.tp53_vt_3 in snv:\n",
    "        return 'SNV'\n",
    "    \n",
    "def vt_group_cond_4(x):\n",
    "    indel = ['DEL', 'INS']\n",
    "    snv = ['SNP', 'DNP', 'TNP', 'ONP']\n",
    "    if x.tp53_vt_4 in indel:\n",
    "        return 'INDEL'\n",
    "    elif x.tp53_vt_4 in snv:\n",
    "        return 'SNV'\n",
    "    \n",
    "def vt_group_cond_5(x):\n",
    "    indel = ['DEL', 'INS']\n",
    "    snv = ['SNP', 'DNP', 'TNP', 'ONP']\n",
    "    if x.tp53_vt_5 in indel:\n",
    "        return 'INDEL'\n",
    "    elif x.tp53_vt_5 in snv:\n",
    "        return 'SNV'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Non-WGD specific functions\n",
    "\n",
    "* *cn_group_cond*\n",
    "\n",
    "First, we group the different Copy Number States *cn_state* in subgroups, under the column *cn_group*:\n",
    "    * Group 1: cnLOH gathering ['CNLOH']\n",
    "    * Group 2: LOSS gathering ['HETLOSS']\n",
    "    * Group 3: HOMDEL gathering ['HOMDEL']\n",
    "    * Group 4: WILD_TYPE gathering ['DIPLOID', 'TETRAPLOID']\n",
    "    * Group 5: GAIN gathering ['GAIN']\n",
    "    * Group 6: OTHER gathering ['AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']\n",
    "\n",
    "* *mut_cn_group_cond*\n",
    "\n",
    "Based on this first column we define 7 final groups of patients adding the mutational information. These groups will be under the column *tp53_group*.\n",
    "    * Group 1: Samples with 0 tp53 mutations and HETLOSS\n",
    "    * Group 2: Samples with HOMDEL\n",
    "    * Group 3: Samples with 1 tp53 mutation and WILD_TYPE (DIPLOID, LOSS AFTER, TETRAPLOID)\n",
    "    * Group 4: Samples with 1 tp53 mutation or more and LOSS\n",
    "    * Group 5: Samples with 1 tp53 mutation or more and cnLOH\n",
    "    * Group 6: Samples with 2/3/4/5 tp53 mutations and WILD_TYPE or GAIN\n",
    "\n",
    "* *tp53_residual_group*: defines if we have residual tp53 or not based on the expected WT tp53 residual\n",
    "\n",
    "\n",
    "* *get_loh_nowgd*\n",
    "\n",
    "The last function is meant to compute the LOH state"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "def cn_group_cond(x):\n",
    "    if x.tp53_cn_state in ['CNLOH']:\n",
    "        return 'cnLOH'\n",
    "    if x.tp53_cn_state in ['HETLOSS']:\n",
    "        return 'LOSS'\n",
    "    if x.tp53_cn_state == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_cn_state in ['DIPLOID', 'TETRAPLOID']:\n",
    "        return 'WILD_TYPE'\n",
    "    if x.tp53_cn_state == 'GAIN':\n",
    "        return 'GAIN'\n",
    "    if x.tp53_cn_state in ['AMP (BALANCED)', 'AMP (LOH)', 'AMP','LOSS & GAIN', 'CNLOH & GAIN']:\n",
    "        return 'OTHER'\n",
    "\n",
    "def mut_cn_group_cond(x):\n",
    "    if x.tp53_cn_state == 'HETLOSS' and x.tp53_count == 0:\n",
    "        return '0_HETLOSS'\n",
    "    if x.tp53_first_group == 'HOMDEL':\n",
    "        return 'HOMDEL'\n",
    "    if x.tp53_first_group == 'WILD_TYPE' and x.tp53_count == 1 :\n",
    "        return '1_WILD_TYPE'\n",
    "    if x.tp53_first_group == 'LOSS' and x.tp53_count >=1:\n",
    "        return '>=1_LOSS'\n",
    "    if x.tp53_first_group == 'cnLOH' and x.tp53_count >=1:\n",
    "        return '>=1_cnLOH'\n",
    "    if (x.tp53_first_group == 'WILD_TYPE' or x.tp53_first_group == 'GAIN') and x.tp53_count > 1:\n",
    "        return '>1muts'\n",
    "    \n",
    "    \n",
    "def tp53_residual_group(x):\n",
    "    if x.tp53_group == '1_WILD_TYPE' or x.tp53_group == '0_HETLOSS':\n",
    "        return 'tp53_res'\n",
    "    if x.tp53_group == 'HOMDEL':\n",
    "        return 'no_tp53_res'\n",
    "    if x.tp53_group == '>=1_LOSS' or x.tp53_group == '>=1_cnLOH':\n",
    "        if (x.tp53_res_1 < 0.5) or (x.tp53_res_2 < 0.5):\n",
    "            return 'no_tp53_res'\n",
    "        else:\n",
    "            if (x.tp53_cf + max(x.tp53_ccf_1, x.tp53_ccf_2, x.tp53_ccf_3, x.tp53_ccf_4, x.tp53_ccf_5)) > 1:\n",
    "                return 'no_tp53_res'\n",
    "            else:\n",
    "                return 'uncertain'\n",
    "    if x.tp53_group == '>1muts':\n",
    "        if (x.tp53_res_1 + x.tp53_res_2 < 2.5):\n",
    "            if (x.tp53_ccf_1 + x.tp53_ccf_2 > 1):\n",
    "                return 'no_tp53_res'\n",
    "            else: \n",
    "                return 'uncertain'\n",
    "           \n",
    "        elif (x.tp53_res_1 + x.tp53_res_2 > 2.5):\n",
    "            return 'tp53_res'\n",
    "        \n",
    "        \n",
    "def get_loh_nowgd(x):\n",
    "    if x.tp53_lcn == 0:\n",
    "        return True\n",
    "    else: return False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## WGD specific functions\n",
    "Finally we have functions to compute TP53 allelic state BEFORE WGD (for the WGD cohort only):\n",
    "* *get_bi_nobi* allows to say if the pre WGD allelic state was Bi-Allelic or not\n",
    "* *get_mono* allows to say if, within Not Bi-allelic samples, the state was mono-allelic or with 2 WT alleles\n",
    "* *get_loh_wgd* that computes if the samples is in LOH state or not"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "'''\n",
    "            NUMBER OF TP53 residual ASSOCIATED WITH ALLELIC STATE BEFORE WGD\n",
    "            \n",
    "            \n",
    "                                     BI-ALLELIC | MONO - ALLELIC | 2WT\n",
    "\n",
    "                LOSS BEFORE :            0      |        1       | -\n",
    "                CNLOH BEFORE & LOSS:     0      |       1,2      | 2\n",
    "                CNLOH BEFORE:            0      |        2       | 3\n",
    "                LOSS AFTER:              -      |       1,2      | 2\n",
    "                DOUBLE LOSS AFTER:       -      |       0,1      | 1\n",
    "                TETRAPLOID:              -      |        2       | 3\n",
    "                CNLOH AFTER:             -      |       1,3      | 3\n",
    "                CNLOH BEFORE & GAIN:     0      |        2,3     | 3,4\n",
    "                \n",
    "\n",
    "                        THRESHOLDS BETWEEN ALLELIC STATES BEFORE WGD\n",
    "                    \n",
    "                                     BI-ALLELIC | MONO - ALLELIC | 2WT\n",
    "\n",
    "                LOSS BEFORE :            <0.4   |        >0.6    | -\n",
    "                CNLOH BEFORE & LOSS:     <0.4   |      0.6< <1.5 | uncertain\n",
    "                CNLOH BEFORE:            <1.5   |   1.5< <2.5    | >2.5\n",
    "                LOSS AFTER:              -      |      <1.5      | uncertain \n",
    "                DOUBLE LOSS AFTER:       -      |       <0.5     | uncertain\n",
    "                TETRAPLOID:              -      |       <2.5     | >2.5\n",
    "                CNLOH AFTER:             -      |       <1.5     | uncertain\n",
    "                CNLOH BEFORE & GAIN:     <1.4   |   1.6< <2.5    | uncertain\n",
    "                \n",
    "    \n",
    "\n",
    " '''\n",
    "\n",
    "\n",
    "# The following functions are for WGD cohort, computing the tp53 allelic state before WGD\n",
    "def get_bi_nobi(x, maf_cohort):\n",
    "    tumor = x.Tumor_Id\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    \n",
    "    #samples without TP53 mutation\n",
    "    if nb_tp53muts == 0:\n",
    "        return 'no_bi'\n",
    "    \n",
    "    # Samples with only one tp53 mutation\n",
    "    if nb_tp53muts == 1:\n",
    "        tp53_res = x.tp53_res_1\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE' or cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr = 0.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr:\n",
    "                return 'bi'\n",
    "            elif tp53_res >= thr:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state in ['LOSS AFTER','DOUBLE LOSS AFTER','TETRAPLOID','CNLOH AFTER']: \n",
    "            return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "            \n",
    "            \n",
    "            \n",
    "    # Samples with 2 tp53 mutations\n",
    "    elif nb_tp53muts == 2:\n",
    "        tp53_res_1 = x.tp53_res_1\n",
    "        tp53_res_2 = x.tp53_res_2\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr - 0.1) or (tp53_res_2 < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr + 0.1) and (tp53_res_2 > thr + 0.1):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (tp53_res_1 < thr_1 - 0.1) or (tp53_res_2 < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (tp53_res_1 < thr_1) or (tp53_res_2 < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if tp53_res_1 < thr_1 - 0.1 or tp53_res_2 < thr_1 - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr_2) and (tp53_res_2 > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "            \n",
    "    # Samples with 3 tp53 mutations\n",
    "    elif nb_tp53muts == 3:\n",
    "        tp53_res_1 = x.tp53_res_1\n",
    "        tp53_res_2 = x.tp53_res_2\n",
    "        tp53_res_3 = x.tp53_res_3\n",
    "        tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "        tp53_res.sort()\n",
    "    \n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (min(tp53_res) < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (max(tp53_res) > thr + 0.1) :\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (min(tp53_res) < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2): # we want that exactly 2 mutation arose before WGD\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (min(tp53_res) < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if min(tp53_res) < thr_1 - 0.1 :\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "    \n",
    "    else: return 'uncertain'\n",
    "\n",
    "\n",
    "# This functions can only be called after get_bi_nobi() has been called on master\n",
    "def get_mono(x, maf_cohort):\n",
    "    tumor = x.Tumor_Id\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor][maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    bi_state = x.tp53_bi_state\n",
    "    \n",
    "    if bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    \n",
    "    elif bi_state == 'uncertain':\n",
    "        return 'uncertain_bi'\n",
    "    \n",
    "    \n",
    "    elif bi_state == 'no_bi': # We are already in the  mono/2WT distinction \n",
    "        \n",
    "        if nb_tp53muts == 0:\n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            else: return '2WT'\n",
    "            \n",
    "            \n",
    "        #1 mut samples\n",
    "        if nb_tp53muts == 1:\n",
    "            tp53_res = x.tp53_res_1\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "\n",
    "        \n",
    "        # 2 mut samples   \n",
    "        elif nb_tp53muts == 2:\n",
    "            tp53_res_1 = x.tp53_res_1\n",
    "            tp53_res_2 = x.tp53_res_2\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res_1 < 2.5 or tp53_res_2 < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res_1 > 2.5 and tp53_res_2 > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "            \n",
    "        # 3 mut samples\n",
    "        elif nb_tp53muts == 3:\n",
    "            tp53_res_1 = x.tp53_res_1\n",
    "            tp53_res_2 = x.tp53_res_2\n",
    "            tp53_res_3 = x.tp53_res_3\n",
    "            tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "            tp53_res.sort()\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if min(tp53_res) < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res[0] > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "        \n",
    "        else: return 'uncertain_mono' \n",
    "        \n",
    "def get_loh_wgd(x):\n",
    "    if x.tp53_cn_state in ['LOSS BEFORE', 'CNLOH BEFORE & LOSS', 'CNLOH BEFORE', 'LOSS BEFORE & AFTER', 'CNLOH BEFORE & GAIN']:\n",
    "        return True\n",
    "    else: return False\n",
    "    \n",
    "    \n",
    "def get_prewgd_tp53_group_1(x):\n",
    "    if x.tp53_bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    elif x.tp53_bi_state == 'no_bi':\n",
    "        return 'tp53_res'\n",
    "    \n",
    "    elif x.tp53_count == 0:\n",
    "        return 'tp53_res'\n",
    "    \n",
    "    \n",
    "def get_prewgd_tp53_group_2(x):\n",
    "    if x.tp53_bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    \n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == True and x.tp53_cn_state == 'LOSS BEFORE':\n",
    "        return 'mono'\n",
    "    elif x.tp53_state_wgd == 'mono':\n",
    "        return 'mono'\n",
    "    \n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == True and x.tp53_cn_state in ['CNLOH BEFORE', 'CNLOH BEFORE & GAIN', 'CNLOH BEFORE & LOSS']:\n",
    "        return 'cnloh_2WT'\n",
    "    \n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == False:\n",
    "        return '2WT'\n",
    "    elif x.tp53_state_wgd == '2WT':\n",
    "        return '2WT'\n",
    "    \n",
    "    \n",
    "def get_loh_mut_status(x):\n",
    "    if x.tp53_count == 0 and x.tp53_loh_status == True:\n",
    "        return '0_loh'\n",
    "    elif x.tp53_count == 0 and x.tp53_loh_status == False:\n",
    "        return '0_noloh'\n",
    "    elif x.tp53_count >= 1 and x.tp53_loh_status == True:\n",
    "        return '1_loh'\n",
    "    elif x.tp53_count >= 1 and x.tp53_loh_status == False:\n",
    "        return '1_noloh'\n",
    "    "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge and save tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "def merge_tables(cohort_type: str):\n",
    "    \n",
    "    if cohort_type == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort_type == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    sample_info = create_sample_info(cohort=cohort_type)\n",
    "    tp53_muts = create_tp53_muts(cohort=cohort_type, sample_info=sample_info)\n",
    "    copy_number_info = create_copy_number_state(cohort=cohort_type)\n",
    "    \n",
    "    master = pd.merge(sample_info, tp53_muts, on='Tumor_Id')\n",
    "    master = pd.merge(master, copy_number_info, on='Sample_Id')\n",
    "    print('checkpoint_1')\n",
    "    # Now we compute the columns\n",
    "    master['gene_count'] = master.apply(create_gene_count, maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_1')\n",
    "    master['mutation_count'] = master.apply(create_mut_count,maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_1')\n",
    "    master['driver_gene_count'] = master.apply(create_driver_gene_count, maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_1')\n",
    "    master['driver_mutation_count'] = master.apply(create_driver_mut_count,maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_1')\n",
    "    master['snv_driver_mutation_count'] = master.apply(create_snv_driver_mut_count, maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_1')\n",
    "    master['indel_driver_mutation_count'] = master.apply(create_indel_driver_mut_count, maf_cohort=maf_cohort, axis=1)\n",
    "    master['max_vaf'] = master.apply(get_max_vaf,maf_cohort=maf_cohort, axis=1)\n",
    "    print('checkpoint_2')\n",
    "    # Tp53 residual\n",
    "    master = create_copies_tp53_muts(master)\n",
    "    master = create_tp53_res(master)\n",
    "    print('checkpoint_3')\n",
    "    # Mutation Type Grouping\n",
    "    master['tp53_vc_group_1'] = master.apply(vc_group_cond_1, axis = 1)\n",
    "    master['tp53_vc_group_2'] = master.apply(vc_group_cond_2, axis = 1)\n",
    "    master['tp53_vc_group_3'] = master.apply(vc_group_cond_3, axis = 1)\n",
    "    master['tp53_vc_group_4'] = master.apply(vc_group_cond_4, axis = 1)\n",
    "    master['tp53_vc_group_5'] = master.apply(vc_group_cond_5, axis = 1)\n",
    "    print('checkpoint_4')\n",
    "    # Variant Type Grouping\n",
    "    master['tp53_vt_group_1'] = master.apply(vt_group_cond_1, axis = 1)\n",
    "    master['tp53_vt_group_2'] = master.apply(vt_group_cond_2, axis = 1)\n",
    "    master['tp53_vt_group_3'] = master.apply(vt_group_cond_3, axis = 1)\n",
    "    master['tp53_vt_group_4'] = master.apply(vt_group_cond_4, axis = 1)\n",
    "    master['tp53_vt_group_5'] = master.apply(vt_group_cond_5, axis = 1)\n",
    "    print('checkpoint_5')\n",
    "    \n",
    "    if cohort_type == 'wgd':\n",
    "        # These groups are BEFORE WGD !!!\n",
    "        master['tp53_bi_state'] = master.apply(get_bi_nobi, maf_cohort=maf_cohort, axis=1)\n",
    "        master['tp53_state_wgd'] = master.apply(get_mono, maf_cohort=maf_cohort,axis=1)\n",
    "        # This is after WGD\n",
    "        master['tp53_loh_status'] = master.apply(get_loh_wgd, axis=1)\n",
    "        master['tp53_mut_loh_group'] = master.apply(get_loh_mut_status, axis=1)\n",
    "        \n",
    "        # Pre WGD Allelic groups\n",
    "        master['prewgd_tp53_group_1'] = master.apply(get_prewgd_tp53_group_1, axis=1)\n",
    "        master['prewgd_tp53_group_2'] = master.apply(get_prewgd_tp53_group_2, axis=1)\n",
    "        \n",
    "    if cohort_type == 'no_wgd':\n",
    "        master['tp53_first_group'] = master.apply(cn_group_cond, axis = 1)\n",
    "        master['tp53_group'] = master.apply(mut_cn_group_cond, axis = 1)\n",
    "        master['tp53_res_group'] = master.apply(tp53_residual_group, axis = 1)\n",
    "        master['tp53_loh_status'] = master.apply(get_loh_nowgd, axis=1)\n",
    "        print('checkpoint_6')\n",
    "        \n",
    "    # Finally, we merge Genome Instability metrics\n",
    "    gi_path = total_output + 'gi_metrics_{}.pkl'.format(cohort_type)\n",
    "    chr_metrics =  pd.read_pickle(gi_path)\n",
    "    print('checkpoint_7')\n",
    "    master = pd.merge(master, chr_metrics, on=['Sample_Id'])\n",
    "    print('checkpoint_8')\n",
    "    master.to_pickle(total_output + 'master_{}.pkl'.format(cohort_type))\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "checkpoint_1\n",
      "checkpoint_1\n",
      "checkpoint_1\n",
      "checkpoint_1\n",
      "checkpoint_1\n",
      "checkpoint_1\n",
      "checkpoint_2\n",
      "checkpoint_3\n",
      "checkpoint_4\n",
      "checkpoint_5\n",
      "checkpoint_7\n",
      "checkpoint_8\n",
      "CPU times: user 7min 31s, sys: 2.15 s, total: 7min 33s\n",
      "Wall time: 7min 35s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "master = merge_tables(cohort_type=cohort_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>tp53_mut_loh_group</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0_loh</th>\n",
       "      <td>2139</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_loh</th>\n",
       "      <td>3814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1_noloh</th>\n",
       "      <td>414</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    count\n",
       "tp53_mut_loh_group       \n",
       "0_loh                2139\n",
       "1_loh                3814\n",
       "1_noloh               414"
      ]
     },
     "execution_count": 311,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master = pd.read_pickle(total_output + 'master_{}.pkl'.format(cohort_type))\n",
    "#master_tp53_loh = master[master['tp53_count']==0][master['tp53_loh_status']==True][~master['tp53_cn_state'].isin(['LOSS BEFORE'])]\n",
    "#get_groupby(master_tp53_loh, 'prewgd_tp53_group_1', 'count')\n",
    "#get_groupby(master,'tp53_vt_group_1','count')\n",
    "get_groupby(master, 'tp53_mut_loh_group', 'count')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_nowgd = pd.read_csv(data_no_wgd + 'cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GERMLINE</th>\n",
       "      <td>799</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOMATIC</th>\n",
       "      <td>147625</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>UNKNOWN</th>\n",
       "      <td>1380</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  count\n",
       "Mutation_Status        \n",
       "GERMLINE            799\n",
       "SOMATIC          147625\n",
       "UNKNOWN            1380"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(maf_cohort_nowgd, 'Mutation_Status', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [],
   "source": [
    "#maf_cohort_wgd\n",
    "\n",
    "def removing_germ_na_mut(maf_cohort: pd.DataFrame):\n",
    "    '''\n",
    "    This function is meant to remove non TP53 GERMLINE mutations from the mafs, to have only SOMATIC mut \n",
    "    except for TP53 muts where we allow GERMLINE status.\n",
    "    '''\n",
    "    maf_cohort_final = maf_cohort[maf_cohort['Mutation_Status'] != 'UNKNOWN']\n",
    "\n",
    "    key_list = list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] != 'TP53'][maf_cohort_final['Mutation_Status'] != 'GERMLINE'].sample_mut_key)\n",
    "    key_list = key_list + list(maf_cohort_final[maf_cohort_final['Hugo_Symbol'] == 'TP53'].sample_mut_key)\n",
    "    maf_cohort_final = maf_cohort_final[maf_cohort_final.sample_mut_key.isin(key_list)]\n",
    "    \n",
    "    return maf_cohort_final\n",
    "\n",
    "new = removing_germ_na_mut(maf_cohort_nowgd)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>GERMLINE</th>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>SOMATIC</th>\n",
       "      <td>147625</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                  count\n",
       "Mutation_Status        \n",
       "GERMLINE              6\n",
       "SOMATIC          147625"
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_groupby(new, 'Mutation_Status', 'count')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "336px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
