{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# <span style='font-family:\"Times New Roman\"'> <span styel=''> **MASTER FILE CREATION**\n",
    "\n",
    "*Emile Cohen* \n",
    "    \n",
    "*March 2020*\n",
    "\n",
    "**Goal:** In this Notebook, we create a master file that summarizes all useful information.\n",
    "\n",
    "The Notebook is divided in 4 parts, representing the four parts of our Master file:\n",
    "   \n",
    "* **1. Patient/Sample Information**\n",
    "* **2. TP53 Mutations**\n",
    "* **3. TP53 Copy Numbers**\n",
    "* **4. TP53 Computed Metrics**\n",
    "* **5. Subgroup columns creation**\n",
    "* **6. Merge tables**\n",
    "\n",
    "**NB1:** In each part, you must run the cells from the begining in order to initialize the variables\n",
    "\n",
    "**NB2:** In order to launch the last script (Merge Tables), you have to define the functions in each part.\n",
    "\n",
    "**NB3:** All functions used for the plots are located in utils/custom_tools.py\n",
    "\n",
    "---"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Setup environment... done!\n"
     ]
    },
    {
     "data": {
      "text/markdown": [
       "<span style=\"color:green\">âœ… Working on **mskimpact_env** conda environment.</span>"
      ],
      "text/plain": [
       "<IPython.core.display.Markdown object>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%run -i '../../../utils/setup_environment.ipy'\n",
    "\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "from scipy.stats import fisher_exact, ranksums, chi2, norm\n",
    "from statsmodels.sandbox.stats.multicomp import multipletests\n",
    "import matplotlib.gridspec as gridspec\n",
    "import pickle\n",
    "\n",
    "data_path = '../../../data/'\n",
    "data_wgd = data_path + 'impact-facets-tp53/processed/wgd/'\n",
    "data_no_wgd = data_path + 'impact-facets-tp53/processed/no_wgd/'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {},
   "outputs": [],
   "source": [
    "# first we load all files for WGD\n",
    "maf_cohort_wgd = pd.read_csv(data_wgd + 'maf_cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_wgd = pd.read_csv(data_wgd + 'cohort_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_wgd = pd.read_csv(data_wgd + 'arm_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_wgd = pd.read_csv(data_wgd + 'gene_level_wgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [],
   "source": [
    "# We load all files for non WGD\n",
    "\n",
    "maf_cohort_nowgd = pd.read_csv(data_no_wgd + 'maf_cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "cohort_nowgd = pd.read_csv(data_no_wgd + 'cohort_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "arm_level_nowgd = pd.read_csv(data_no_wgd + 'arm_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)\n",
    "gene_level_nowgd = pd.read_csv(data_no_wgd + 'gene_level_nowgd.txt', sep='\\t').drop('Unnamed: 0', axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Creating keys for mutations in maf files\n",
    "\n",
    "# First we need to create a sample_mut_key to identify duplicated mutations\n",
    "maf_cohort_wgd['mut_key'] = maf_cohort_wgd.apply(lambda h: str(h.Chromosome)+'_'+str(h.Start_Position)+'_'+str(h.Reference_Allele)+'_'+str(h.Tumor_Seq_Allele2), axis=1) \n",
    "maf_cohort_nowgd['mut_key'] = maf_cohort_nowgd.apply(lambda h: str(h.Chromosome)+'_'+str(h.Start_Position)+'_'+str(h.Reference_Allele)+'_'+str(h.Tumor_Seq_Allele2), axis=1) \n",
    "\n",
    "# Create a sample key to differentiate duplicates\n",
    "maf_cohort_wgd['sample_mut_key'] = maf_cohort_wgd.apply(lambda h: h.Tumor_Sample_Barcode + h.mut_key, axis = 1)\n",
    "maf_cohort_nowgd['sample_mut_key'] = maf_cohort_nowgd.apply(lambda h: h.Tumor_Sample_Barcode + h.mut_key, axis = 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Load clinical data\n",
    "clinical_data = pd.read_csv(data_path + 'cbioportal/raw/mskimpact_clinical_data-2.tsv', sep= '\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "# Filtering the clinical data\n",
    "samples_wgd = list(set(cohort_wgd.tumor_sample))\n",
    "samples_nowgd = list(set(cohort_nowgd.tumor_sample))\n",
    "\n",
    "clinical_wgd = clinical_data[clinical_data['Sample ID'].isin(samples_wgd)]\n",
    "clinical_nowgd = clinical_data[clinical_data['Sample ID'].isin(samples_nowgd)]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# IMPORTANT: Defining the master type you want\n",
    "In this script we can create two master files: one for WGD samples and one for non WGD samples.\n",
    "So this parameter allows to select the type of master you want."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [],
   "source": [
    "cohort_type = 'wgd'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "---\n",
    "# Patient/Sample Information"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we focus on clinical information exported from CbioPortal.\n",
    "\n",
    "The following columns are selected:\n",
    "* Sample_Id\n",
    "* Tumor_Id\n",
    "* Patient_Id\n",
    "* Patient Current Age\n",
    "* Cancer_Type\n",
    "* Cancer_Type_Detailed\n",
    "* Sample_Type\n",
    "* purity\n",
    "* ploidy\n",
    "* Overall Survival Status\n",
    "* Overall Survival (Months)\n",
    "* MSI Score\n",
    "* MSI Type\n",
    "* Tumor Mutational Burden"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_sample_info(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    cohort_filt = cohort[['sample_id', 'tumor_sample', 'patient', 'ploidy']]\n",
    "    clinical_filt = clinical[['Sample ID', 'Patient Current Age', 'Cancer Type', 'Cancer Type Detailed', 'Sample Type',\n",
    "                  'Overall Survival (Months)', 'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "    purity = maf_cohort.drop_duplicates('Tumor_Sample_Barcode')[['Tumor_Sample_Barcode', 'purity']]\n",
    "\n",
    "    # Merging these files\n",
    "    sample_info = pd.merge(cohort_filt, purity, left_on='tumor_sample', right_on='Tumor_Sample_Barcode')\n",
    "    sample_info =  pd.merge(sample_info, clinical_filt, left_on='tumor_sample', right_on='Sample ID').drop(['Tumor_Sample_Barcode', 'Sample ID'], axis=1)\n",
    "\n",
    "\n",
    "    sample_info = sample_info[['sample_id', 'tumor_sample', 'patient', 'Cancer Type', 'Cancer Type Detailed',\n",
    "                               'Patient Current Age','Sample Type', 'purity','ploidy','Overall Survival (Months)', \n",
    "                               'Overall Survival Status','MSI Score', 'MSI Type','Impact TMB Score']]\n",
    "\n",
    "    sample_info.columns = ['Sample_Id', 'Tumor_Id', 'Patient_Id','Cancer_Type', 'Cancer_Type_Detailed', 'Patient_Current_Age',\n",
    "                          'Sample_Type', 'purity', 'ploidy', 'Overall_Survival_Months', 'Overall_Survival_Status',\n",
    "                          'MSI_Score', 'MSI_Type', 'TMB_Score']\n",
    "    \n",
    "    return sample_info\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>Overall_Survival_Months</th>\n",
       "      <th>Overall_Survival_Status</th>\n",
       "      <th>MSI_Score</th>\n",
       "      <th>MSI_Type</th>\n",
       "      <th>TMB_Score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>P-0025956</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>71.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>3.496971</td>\n",
       "      <td>3.584</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.391316</td>\n",
       "      <td>2.871793</td>\n",
       "      <td>14.137</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Neuroendocrine Carcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.865628</td>\n",
       "      <td>3.115253</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>2.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>3.136841</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.82</td>\n",
       "      <td>Stable</td>\n",
       "      <td>2.6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Colon Adenocarcinoma</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.351778</td>\n",
       "      <td>3.727190</td>\n",
       "      <td>34.060</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.45</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9893</th>\n",
       "      <td>P-0050750-T01-IM6_P-0050750-N01-IM6</td>\n",
       "      <td>P-0050750-T01-IM6</td>\n",
       "      <td>P-0050750</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.301523</td>\n",
       "      <td>3.845451</td>\n",
       "      <td>0.099</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9894</th>\n",
       "      <td>P-0050643-T01-IM6_P-0050643-N01-IM6</td>\n",
       "      <td>P-0050643-T01-IM6</td>\n",
       "      <td>P-0050643</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.667936</td>\n",
       "      <td>4.419218</td>\n",
       "      <td>1.315</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.30</td>\n",
       "      <td>Stable</td>\n",
       "      <td>4.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9895</th>\n",
       "      <td>P-0050223-T02-IM6_P-0050223-N02-IM6</td>\n",
       "      <td>P-0050223-T02-IM6</td>\n",
       "      <td>P-0050223</td>\n",
       "      <td>Hepatobiliary Cancer</td>\n",
       "      <td>Cholangiocarcinoma</td>\n",
       "      <td>67.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.215128</td>\n",
       "      <td>4.725790</td>\n",
       "      <td>1.940</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9896</th>\n",
       "      <td>P-0048760-T01-IM6_P-0048760-N01-IM6</td>\n",
       "      <td>P-0048760-T01-IM6</td>\n",
       "      <td>P-0048760</td>\n",
       "      <td>Bone Cancer</td>\n",
       "      <td>Osteosarcoma</td>\n",
       "      <td>18.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.614603</td>\n",
       "      <td>3.947721</td>\n",
       "      <td>3.682</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>5.79</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>6.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9897</th>\n",
       "      <td>P-0050744-T01-IM6_P-0050744-N01-IM6</td>\n",
       "      <td>P-0050744-T01-IM6</td>\n",
       "      <td>P-0050744</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>69.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>2.426452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.06</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9898 rows Ã— 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6  P-0025956-T01-IM6  P-0025956   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6  P-0036909-T01-IM6  P-0036909   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6  P-0023546-T01-IM6  P-0023546   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6  P-0023546-T02-IM6  P-0023546   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6  P-0018837-T01-IM6  P-0018837   \n",
       "...                                   ...                ...        ...   \n",
       "9893  P-0050750-T01-IM6_P-0050750-N01-IM6  P-0050750-T01-IM6  P-0050750   \n",
       "9894  P-0050643-T01-IM6_P-0050643-N01-IM6  P-0050643-T01-IM6  P-0050643   \n",
       "9895  P-0050223-T02-IM6_P-0050223-N02-IM6  P-0050223-T02-IM6  P-0050223   \n",
       "9896  P-0048760-T01-IM6_P-0048760-N01-IM6  P-0048760-T01-IM6  P-0048760   \n",
       "9897  P-0050744-T01-IM6_P-0050744-N01-IM6  P-0050744-T01-IM6  P-0050744   \n",
       "\n",
       "                     Cancer_Type               Cancer_Type_Detailed  \\\n",
       "0     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "1     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "2                Prostate Cancer  Prostate Neuroendocrine Carcinoma   \n",
       "3                Prostate Cancer            Prostate Adenocarcinoma   \n",
       "4              Colorectal Cancer               Colon Adenocarcinoma   \n",
       "...                          ...                                ...   \n",
       "9893           Pancreatic Cancer          Pancreatic Adenocarcinoma   \n",
       "9894               Breast Cancer   Breast Invasive Ductal Carcinoma   \n",
       "9895        Hepatobiliary Cancer                 Cholangiocarcinoma   \n",
       "9896                 Bone Cancer                       Osteosarcoma   \n",
       "9897           Pancreatic Cancer          Pancreatic Adenocarcinoma   \n",
       "\n",
       "      Patient_Current_Age Sample_Type    purity    ploidy  \\\n",
       "0                    71.0     Primary  0.273767  3.496971   \n",
       "1                    47.0  Metastasis  0.391316  2.871793   \n",
       "2                    50.0     Primary  0.865628  3.115253   \n",
       "3                    50.0     Primary  0.312907  3.136841   \n",
       "4                    60.0     Primary  0.351778  3.727190   \n",
       "...                   ...         ...       ...       ...   \n",
       "9893                 55.0  Metastasis  0.301523  3.845451   \n",
       "9894                 53.0     Primary  0.667936  4.419218   \n",
       "9895                 67.0     Primary  0.215128  4.725790   \n",
       "9896                 18.0     Primary  0.614603  3.947721   \n",
       "9897                 69.0     Primary  0.300000  2.426452   \n",
       "\n",
       "      Overall_Survival_Months Overall_Survival_Status  MSI_Score  \\\n",
       "0                       3.584                DECEASED       0.00   \n",
       "1                      14.137                  LIVING       0.37   \n",
       "2                       4.800                DECEASED       2.37   \n",
       "3                       4.800                DECEASED       0.82   \n",
       "4                      34.060                  LIVING       0.45   \n",
       "...                       ...                     ...        ...   \n",
       "9893                    0.099                  LIVING       0.05   \n",
       "9894                    1.315                  LIVING       0.30   \n",
       "9895                    1.940                  LIVING       0.00   \n",
       "9896                    3.682                  LIVING       5.79   \n",
       "9897                      NaN                  LIVING       0.06   \n",
       "\n",
       "           MSI_Type  TMB_Score  \n",
       "0            Stable        5.3  \n",
       "1            Stable        3.5  \n",
       "2            Stable        3.5  \n",
       "3            Stable        2.6  \n",
       "4            Stable        5.3  \n",
       "...             ...        ...  \n",
       "9893         Stable        5.3  \n",
       "9894         Stable        4.4  \n",
       "9895         Stable        3.5  \n",
       "9896  Indeterminate        6.1  \n",
       "9897         Stable        5.3  \n",
       "\n",
       "[9898 rows x 14 columns]"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_sample_info(cohort=cohort_type)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Mutations\n",
    "In this part, we focus on tp53 mutation information.\n",
    "\n",
    "We gather all mutations per sample, and split it into different columns. We have the following columns:\n",
    "* Tumor_Id\t\n",
    "* key_1 (2,3,4,5) --> Mutation key allowing to filter duplicates\n",
    "* vc_1 (2,3,4,5) --> Variant Classification\n",
    "* ccf_1 (2,3,4,5) --> Cancer Cell Fraction of the mutation\n",
    "* vaf_1 (2,3,4,5) --> Variant Allele Frequency of the mutation\n",
    "* HGVSp_1 (2,3,4,5) --> protein change\n",
    "* spot_1 (2,3,4,5) --> Integer that defines the spot of the tp53 mutation\n",
    "* tp53_count --> Number of tp53 mutations of the sample"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {
    "scrolled": true
   },
   "outputs": [],
   "source": [
    "def f_(x):\n",
    "    # This function helps us to group mutations together in a single cell per patient\n",
    "    return pd.DataFrame(dict(Tumor_Sample_Barcode = x['Tumor_Sample_Barcode'],  \n",
    "                        muts = \"%s\" % ','.join(x['sample_mut_key_vc_ccf_vaf_hgv_spot'])))\n",
    "\n",
    "def count_tp53_muts(x):\n",
    "    count = 0\n",
    "    for i in range(1,6):\n",
    "        if x['tp53_key_' + str(i)]:\n",
    "            count+= 1\n",
    "    return count\n",
    "\n",
    "cohort = cohort_type\n",
    "\n",
    "\n",
    "# WARNING: THis function needs sample_info to work\n",
    "def create_tp53_muts(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "\n",
    "    '''\n",
    "    This function aims to gather all tp53 mutation characteristics.\n",
    "    For each sample we gather the tp53 mutations and their characteristics for all patients.\n",
    "    '''\n",
    "    # We load the  table created in maf_tp53_creation.ipynb\n",
    "    maf_tp53 = maf_cohort[maf_cohort['Hugo_Symbol'] == 'TP53']\n",
    "    maf_tp53['mut_spot'] = maf_tp53.HGVSp.str.extract('(\\d+)')\n",
    "\n",
    "    # We select only intresting columns\n",
    "    maf_tp53_filtered = maf_tp53[['Tumor_Sample_Barcode','sample_mut_key', 'Variant_Classification',\\\n",
    "                                        'ccf_expected_copies', 't_var_freq', 'HGVSp','mut_spot' ]]\n",
    "\n",
    "    # Let's Merge mut_key,Variant_classification, CF, CCF, and VAF to gather them\n",
    "    maf_tp53_filtered['sample_mut_key_vc_ccf_vaf_hgv_spot'] = maf_tp53_filtered.apply(lambda x: str(x.sample_mut_key)+'%'+str(x.Variant_Classification)+'%'+str(x.ccf_expected_copies)+'%'+str(x.t_var_freq)+'%'+str(x.HGVSp)+'%'+str(x.mut_spot), axis=1)\n",
    "\n",
    "    # We Select important columns\n",
    "    final = maf_tp53_filtered[['Tumor_Sample_Barcode', 'sample_mut_key_vc_ccf_vaf_hgv_spot']]\n",
    "    # We groupby Patient_Id and apply the function above to group mutations\n",
    "    final = final.groupby(['Tumor_Sample_Barcode'], sort=False).apply(f_)\n",
    "\n",
    "    # We separate the different mutations into 5 different columns (5 is the max number of tp53 mutations in our cohort)\n",
    "    final[['mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5']] = final.muts.str.split(',', expand=True)\n",
    "    #final = final.drop(['mut_key_6'],axis=1)\n",
    "    # Split the columns into mut_key_ and vc_\n",
    "    final[['tp53_key_1','tp53_vc_1','tp53_ccf_1','tp53_vaf_1','tp53_HGVSp_1', 'tp53_spot_1']] = final.mut_key_1.str.split('%', expand=True)\n",
    "    final[['tp53_key_2','tp53_vc_2','tp53_ccf_2','tp53_vaf_2','tp53_HGVSp_2', 'tp53_spot_2']] = final.mut_key_2.str.split('%', expand=True)\n",
    "    final[['tp53_key_3','tp53_vc_3','tp53_ccf_3','tp53_vaf_3','tp53_HGVSp_3', 'tp53_spot_3']] = final.mut_key_3.str.split('%', expand=True)\n",
    "    final[['tp53_key_4','tp53_vc_4','tp53_ccf_4','tp53_vaf_4','tp53_HGVSp_4', 'tp53_spot_4']] = final.mut_key_4.str.split('%', expand=True)\n",
    "    final[['tp53_key_5','tp53_vc_5','tp53_ccf_5','tp53_vaf_5','tp53_HGVSp_5', 'tp53_spot_5']] = final.mut_key_5.str.split('%', expand=True)\n",
    "\n",
    "    # We remove the muts column\n",
    "    final = final.drop(['muts','mut_key_1','mut_key_2','mut_key_3','mut_key_4','mut_key_5'], axis=1)\n",
    "\n",
    "    # We remove duplicates\n",
    "    final = final.drop_duplicates('Tumor_Sample_Barcode')\n",
    "\n",
    "    # We add the cohort patients that are not tp53 positive\n",
    "    #First we create a dataframe with all missing samples\n",
    "\n",
    "    cohort_samples = set(sample_info.Tumor_Id)\n",
    "    final_samples = set(final.Tumor_Sample_Barcode)\n",
    "    missing_samp = pd.DataFrame(cohort_samples - final_samples, columns = ['Tumor_Sample_Barcode'])\n",
    "    #Then we append the two datframe\n",
    "    final = final.append(missing_samp)\n",
    "\n",
    "    # We rename the Tumor_Sample_Barcode column to have the same key as in other datframes\n",
    "    final = final.rename(columns={'Tumor_Sample_Barcode': 'Tumor_Id'})\n",
    "\n",
    "    # We add a last column tp53_count that represents the number of tp53 mutations per sample\n",
    "    final = final.where(final.notnull(), None)\n",
    "    final['tp53_count'] = final.apply(count_tp53_muts, axis = 1)\n",
    "\n",
    "    # We change the type of vafs column to float64 instead of strings\n",
    "    final = final.astype({'tp53_vaf_1': 'float64', 'tp53_vaf_2': 'float64', 'tp53_vaf_3': 'float64', 'tp53_vaf_4': 'float64', 'tp53_vaf_5': 'float64',\n",
    "                       'tp53_ccf_1': 'float64', 'tp53_ccf_2': 'float64', 'tp53_ccf_3': 'float64', 'tp53_ccf_4': 'float64', 'tp53_ccf_5': 'float64'})\n",
    "\n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {
    "collapsed": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909-T01-IM617_7577121_G_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.798</td>\n",
       "      <td>0.312169</td>\n",
       "      <td>p.Arg273Cys</td>\n",
       "      <td>273</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546-T01-IM617_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.845070</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546-T02-IM617_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.636735</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837-T01-IM617_7578406_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.325843</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>175</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>P-0019444-T01-IM6</td>\n",
       "      <td>P-0019444-T01-IM617_7578461_C_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.169002</td>\n",
       "      <td>p.Val157Phe</td>\n",
       "      <td>157</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3190</th>\n",
       "      <td>P-0029543-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3191</th>\n",
       "      <td>P-0023447-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3192</th>\n",
       "      <td>P-0002901-T01-IM3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3193</th>\n",
       "      <td>P-0027887-T01-IM6</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3194</th>\n",
       "      <td>P-0012282-T02-IM5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9898 rows Ã— 32 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               Tumor_Id                       tp53_key_1          tp53_vc_1  \\\n",
       "8     P-0036909-T01-IM6  P-0036909-T01-IM617_7577121_G_A  Missense_Mutation   \n",
       "13    P-0023546-T01-IM6  P-0023546-T01-IM617_7578442_T_C  Missense_Mutation   \n",
       "16    P-0023546-T02-IM6  P-0023546-T02-IM617_7578442_T_C  Missense_Mutation   \n",
       "21    P-0018837-T01-IM6  P-0018837-T01-IM617_7578406_C_T  Missense_Mutation   \n",
       "48    P-0019444-T01-IM6  P-0019444-T01-IM617_7578461_C_A  Missense_Mutation   \n",
       "...                 ...                              ...                ...   \n",
       "3190  P-0029543-T01-IM6                             None               None   \n",
       "3191  P-0023447-T01-IM6                             None               None   \n",
       "3192  P-0002901-T01-IM3                             None               None   \n",
       "3193  P-0027887-T01-IM6                             None               None   \n",
       "3194  P-0012282-T02-IM5                             None               None   \n",
       "\n",
       "      tp53_ccf_1  tp53_vaf_1 tp53_HGVSp_1 tp53_spot_1 tp53_key_2 tp53_vc_2  \\\n",
       "8          0.798    0.312169  p.Arg273Cys         273       None      None   \n",
       "13         0.933    0.845070  p.Tyr163Cys         163       None      None   \n",
       "16         1.000    0.636735  p.Tyr163Cys         163       None      None   \n",
       "21         1.000    0.325843  p.Arg175His         175       None      None   \n",
       "48         1.000    0.169002  p.Val157Phe         157       None      None   \n",
       "...          ...         ...          ...         ...        ...       ...   \n",
       "3190         NaN         NaN         None        None       None      None   \n",
       "3191         NaN         NaN         None        None       None      None   \n",
       "3192         NaN         NaN         None        None       None      None   \n",
       "3193         NaN         NaN         None        None       None      None   \n",
       "3194         NaN         NaN         None        None       None      None   \n",
       "\n",
       "      tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2 tp53_spot_2 tp53_key_3 tp53_vc_3  \\\n",
       "8            NaN         NaN         None        None       None      None   \n",
       "13           NaN         NaN         None        None       None      None   \n",
       "16           NaN         NaN         None        None       None      None   \n",
       "21           NaN         NaN         None        None       None      None   \n",
       "48           NaN         NaN         None        None       None      None   \n",
       "...          ...         ...          ...         ...        ...       ...   \n",
       "3190         NaN         NaN         None        None       None      None   \n",
       "3191         NaN         NaN         None        None       None      None   \n",
       "3192         NaN         NaN         None        None       None      None   \n",
       "3193         NaN         NaN         None        None       None      None   \n",
       "3194         NaN         NaN         None        None       None      None   \n",
       "\n",
       "      tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3 tp53_spot_3 tp53_key_4 tp53_vc_4  \\\n",
       "8            NaN         NaN         None        None       None      None   \n",
       "13           NaN         NaN         None        None       None      None   \n",
       "16           NaN         NaN         None        None       None      None   \n",
       "21           NaN         NaN         None        None       None      None   \n",
       "48           NaN         NaN         None        None       None      None   \n",
       "...          ...         ...          ...         ...        ...       ...   \n",
       "3190         NaN         NaN         None        None       None      None   \n",
       "3191         NaN         NaN         None        None       None      None   \n",
       "3192         NaN         NaN         None        None       None      None   \n",
       "3193         NaN         NaN         None        None       None      None   \n",
       "3194         NaN         NaN         None        None       None      None   \n",
       "\n",
       "      tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4 tp53_spot_4 tp53_key_5 tp53_vc_5  \\\n",
       "8            NaN         NaN         None        None       None      None   \n",
       "13           NaN         NaN         None        None       None      None   \n",
       "16           NaN         NaN         None        None       None      None   \n",
       "21           NaN         NaN         None        None       None      None   \n",
       "48           NaN         NaN         None        None       None      None   \n",
       "...          ...         ...          ...         ...        ...       ...   \n",
       "3190         NaN         NaN         None        None       None      None   \n",
       "3191         NaN         NaN         None        None       None      None   \n",
       "3192         NaN         NaN         None        None       None      None   \n",
       "3193         NaN         NaN         None        None       None      None   \n",
       "3194         NaN         NaN         None        None       None      None   \n",
       "\n",
       "      tp53_ccf_5  tp53_vaf_5 tp53_HGVSp_5 tp53_spot_5  tp53_count  \n",
       "8            NaN         NaN         None        None           1  \n",
       "13           NaN         NaN         None        None           1  \n",
       "16           NaN         NaN         None        None           1  \n",
       "21           NaN         NaN         None        None           1  \n",
       "48           NaN         NaN         None        None           1  \n",
       "...          ...         ...          ...         ...         ...  \n",
       "3190         NaN         NaN         None        None           0  \n",
       "3191         NaN         NaN         None        None           0  \n",
       "3192         NaN         NaN         None        None           0  \n",
       "3193         NaN         NaN         None        None           0  \n",
       "3194         NaN         NaN         None        None           0  \n",
       "\n",
       "[9898 rows x 32 columns]"
      ]
     },
     "execution_count": 105,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "create_tp53_muts(cohort=cohort_type)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# TP53 Copy Numbers"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "In this part, we gather the information from gene_level table.\n",
    "We creaste the following columns:\n",
    "* Sample_Id \n",
    "* tcn --> total copy number\n",
    "* mcn --> major copy number\n",
    "* lcn --> lower copy number\n",
    "* seg_length --> length of the segment\n",
    "* cn_state --> copy number state\n",
    "* cf --> Cell fraction of the cn_state\n",
    "* wgd --> Wholde Genome Doubling (True or False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "# WARNING: THis function needs sample_info to work\n",
    "def create_copy_number_state(cohort:str):\n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    # We want TP53 locus so we have to filter the gene\n",
    "    gene_level = gene_level[gene_level['gene'] == 'TP53']\n",
    "\n",
    "    gene_level['Tumor_Id'] = gene_level['sample'].str[:17]\n",
    "    gene_level_subset = gene_level[['sample','tcn','mcn','lcn','seg_length','cn_state', 'cf.em']]\n",
    "    \n",
    "    # We rename the cf.em column \n",
    "    gene_level_subset = gene_level_subset.rename(columns={'cf.em': 'tp53_cf', \n",
    "                                                          'sample':'Sample_Id',\n",
    "                                                          'tcn': 'tp53_tcn',\n",
    "                                                          'mcn': 'tp53_mcn',\n",
    "                                                          'lcn': 'tp53_lcn',\n",
    "                                                          'seg_length': 'tp53_seg_length',\n",
    "                                                          'cn_state':'tp53_cn_state'})\n",
    "    \n",
    "    # We add WGD information\n",
    "    wgd = cohort[['sample_id', 'wgd']]\n",
    "    \n",
    "    final = pd.merge(gene_level_subset, wgd, left_on='Sample_Id', right_on='sample_id').drop(['sample_id'], axis=1)\n",
    "    \n",
    "    \n",
    "    return final"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 19.6 ms, sys: 2.26 ms, total: 21.8 ms\n",
      "Wall time: 20.4 ms\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>tp53_tcn</th>\n",
       "      <th>tp53_mcn</th>\n",
       "      <th>tp53_lcn</th>\n",
       "      <th>tp53_seg_length</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>tp53_cf</th>\n",
       "      <th>wgd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668431</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>0.169170</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18101929</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.306121</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25250470</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.835504</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25237770</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25231975</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.257400</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9893</th>\n",
       "      <td>P-0050750-T01-IM6_P-0050750-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13849975</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.286332</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9894</th>\n",
       "      <td>P-0050643-T01-IM6_P-0050643-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11917531</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.545007</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9895</th>\n",
       "      <td>P-0050223-T02-IM6_P-0050223-N02-IM6</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>80668650</td>\n",
       "      <td>TETRAPLOID</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9896</th>\n",
       "      <td>P-0048760-T01-IM6_P-0048760-N01-IM6</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7438283</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.598599</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9897</th>\n",
       "      <td>P-0050744-T01-IM6_P-0050744-N01-IM6</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30207093</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.205450</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9898 rows Ã— 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id  tp53_tcn  tp53_mcn  tp53_lcn  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6         3       2.0       1.0   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6         2       2.0       0.0   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6         3       3.0       0.0   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6         2       2.0       0.0   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6         3       3.0       0.0   \n",
       "...                                   ...       ...       ...       ...   \n",
       "9893  P-0050750-T01-IM6_P-0050750-N01-IM6         2       2.0       0.0   \n",
       "9894  P-0050643-T01-IM6_P-0050643-N01-IM6         3       3.0       0.0   \n",
       "9895  P-0050223-T02-IM6_P-0050223-N02-IM6         4       2.0       2.0   \n",
       "9896  P-0048760-T01-IM6_P-0048760-N01-IM6         3       3.0       0.0   \n",
       "9897  P-0050744-T01-IM6_P-0050744-N01-IM6         2       2.0       0.0   \n",
       "\n",
       "      tp53_seg_length        tp53_cn_state   tp53_cf   wgd  \n",
       "0            80668431           LOSS AFTER  0.169170  True  \n",
       "1            18101929          LOSS BEFORE  0.306121  True  \n",
       "2            25250470  CNLOH BEFORE & LOSS  0.835504  True  \n",
       "3            25237770          LOSS BEFORE  0.312907  True  \n",
       "4            25231975  CNLOH BEFORE & LOSS  0.257400  True  \n",
       "...               ...                  ...       ...   ...  \n",
       "9893         13849975          LOSS BEFORE  0.286332  True  \n",
       "9894         11917531  CNLOH BEFORE & LOSS  0.545007  True  \n",
       "9895         80668650           TETRAPLOID       NaN  True  \n",
       "9896          7438283  CNLOH BEFORE & LOSS  0.598599  True  \n",
       "9897         30207093          LOSS BEFORE  0.205450  True  \n",
       "\n",
       "[9898 rows x 8 columns]"
      ]
     },
     "execution_count": 120,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time \n",
    "copy_number_info = create_copy_number_state(cohort = cohort_type)\n",
    "copy_number_info"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Computed metrics\n",
    "In this part we define functions to be applied on the master file to compute specific metrics.\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_gene_count(maf_cohort):\n",
    "    '''\n",
    "    This function create the count of genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    \n",
    "    # First we create the gene_count table by groupbying and sizing, we then change the index\n",
    "    selected_cohort = maf_cohort[['Sample_Id','Tumor_Id', 'Gene_Id']]\n",
    "    gene_count = pd.DataFrame(pd.DataFrame(selected_cohort[['Sample_Id', 'Gene_Id']].groupby(['Sample_Id', 'Gene_Id']).size(), columns = ['count']).groupby(['Sample_Id']).size(), columns = ['gene_count'])\n",
    "    gene_count = gene_count.reset_index()\n",
    "\n",
    "    # We add missing patients to the gene_count to have all the cohort\n",
    "    no_gene_id = selected_cohort.Gene_Id.isna()\n",
    "    no_gene_samples = set(selected_cohort[selected_cohort.index.isin(list(no_gene_id[no_gene_id == True].index))]['Sample_Id'])\n",
    "    missing_samp = pd.DataFrame(no_gene_samples, columns = ['Sample_Id'])\n",
    "\n",
    "    # We append the two dataframes\n",
    "    gene_count = gene_count.append(missing_samp)\n",
    "    \n",
    "    #Fillna with 0\n",
    "    gene_count = gene_count.fillna(0)\n",
    "\n",
    "    return gene_count\n",
    "\n",
    "def create_mut_count(maf_cohort):\n",
    "    '''\n",
    "    This function computes the dataframe of all mutation count per sample.\n",
    "    '''\n",
    "    selected_cohort = maf_cohort[['Sample_Id','Tumor_Id', 'Gene_Id']]\n",
    "    mut_count = get_groupby(selected_cohort, 'Sample_Id', 'mutation_count')\n",
    "    \n",
    "    return mut_count\n",
    "\n",
    "\n",
    "def get_driver_count(x, maf_cohort):\n",
    "    lookup_table = maf_cohort[maf_cohort['Sample_Id'] == x.Sample_Id]\n",
    "    h = get_groupby(lookup_table, 'oncogenic', 'count')\n",
    "    count = (int(h.loc['Oncogenic']) if 'Oncogenic' in h.index else 0) + (int(h.loc['Likely Oncogenic']) if 'Likely Oncogenic' in h.index else 0) +(int(h.loc['Predicted Oncogenic']) if 'Predicted Oncogenic' in h.index else 0)\n",
    "    return count\n",
    "\n",
    "def create_driver_count(maf_cohort):\n",
    "    '''In this function we count the number of mutation driver per sample.'''\n",
    "    samples=list(set(maf_cohort.Sample_Id))\n",
    "    driver_count = pd.DataFrame(columns=['Sample_Id', 'driver_count'])\n",
    "    driver_count.Sample_Id = samples\n",
    "    driver_count['driver_count'] = driver_count.apply(get_driver_count, maf_cohort=maf_cohort, axis=1)\n",
    "\n",
    "    return driver_count\n",
    "    \n",
    "\n",
    "# The following function needs to be called on the complete master file because it needs info from different parts\n",
    "# It computes the expected number of tp53 mutant copies in a cell\n",
    "def create_copies_tp53_muts(master):\n",
    "    master['tp53_exp_nb_1'] = master.apply(lambda x:(x.tp53_vaf_1 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_2'] = master.apply(lambda x:(x.tp53_vaf_2 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_3'] = master.apply(lambda x:(x.tp53_vaf_3 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_4'] = master.apply(lambda x:(x.tp53_vaf_4 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    master['tp53_exp_nb_5'] = master.apply(lambda x:(x.tp53_vaf_5 / x.purity) * (x.tp53_tcn * x.purity + 2*(1 - x.purity)), axis = 1)\n",
    "    \n",
    "    return master\n",
    "\n",
    "\n",
    "# The following computes the expected number of copies of tp53 residuals \n",
    "def create_tp53_res(master):\n",
    "    master['tp53_res_1'] = master.apply(lambda x:x.tcn - x.tp53_exp_nb_1, axis = 1)\n",
    "    master['tp53_res_2'] = master.apply(lambda x:x.tcn - x.tp53_exp_nb_2, axis = 1)\n",
    "    master['tp53_res_3'] = master.apply(lambda x:x.tcn - x.tp53_exp_nb_3, axis = 1)\n",
    "    master['tp53_res_4'] = master.apply(lambda x:x.tcn - x.tp53_exp_nb_4, axis = 1)\n",
    "    master['tp53_res_5'] = master.apply(lambda x:x.tcn - x.tp53_exp_nb_5, axis = 1)\n",
    "    \n",
    "    return master\n",
    "\n",
    "\n",
    "\n",
    "# The following functions allow to group the Mutation Types\n",
    "def vc_group_cond_1(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_1 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_1 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_1 in missense: \n",
    "        if x.tp53_spot_1 in ['273','248','175']: return x.tp53_spot_1\n",
    "        elif x.tp53_spot_1 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_2(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_2 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_2 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_2 in missense: \n",
    "        if x.tp53_spot_2 in ['273','248','175']: return x.tp53_spot_2\n",
    "        elif x.tp53_spot_2 in['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'   \n",
    "def vc_group_cond_3(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_3 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_3 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_3 in missense: \n",
    "        if x.tp53_spot_3 in ['273','248','175']: return x.tp53_spot_3\n",
    "        elif x.tp53_spot_3 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense' \n",
    "def vc_group_cond_4(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_4 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_4 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_4 in missense: \n",
    "        if x.tp53_spot_4 in ['273','248','175']: return x.tp53_spot_4\n",
    "        elif x.tp53_spot_4 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "def vc_group_cond_5(x):\n",
    "    truncated = ['Splice_Site','Intron','Nonsense_Mutation','Splice_Region','Frame_Shift_Del','Frame_Shift_Ins']\n",
    "    in_frame = ['In_Frame_Ins','In_Frame_Del']\n",
    "    missense = ['Missense_Mutation']\n",
    "    \n",
    "    if x.tp53_vc_5 in truncated: return 'truncated'\n",
    "    if x.tp53_vc_5 in in_frame: return 'in_frame'\n",
    "    if x.tp53_vc_5 in missense: \n",
    "        if x.tp53_spot_5 in ['273','248','175']: return x.tp53_spot_5\n",
    "        elif x.tp53_spot_5 in ['245', '282', '213', '352', '220', '196']: return 'hotspot'\n",
    "        else: return 'missense'\n",
    "\n",
    "        \n",
    "        \n",
    "# The following functions are for WGD cohort, computing the tp53 allelic state before WGD\n",
    "def get_bi_nobi(x):\n",
    "    tumor = x.tumor_sample\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    \n",
    "    # Samples with only one tp53 mutation\n",
    "    if nb_tp53muts == 1:\n",
    "        tp53_res = maf_muts.tp53_res.values[0]\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE' or cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr = 0.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr:\n",
    "                return 'bi'\n",
    "            elif tp53_res >= thr:\n",
    "                return 'no_bi'\n",
    "\n",
    "        elif cn_state in ['LOSS AFTER','DOUBLE LOSS AFTER','TETRAPLOID','CNLOH AFTER']: \n",
    "            return 'no_bi'\n",
    "\n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr = 1.5\n",
    "            if tp53_res < thr - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res < thr + 0.1) and (tp53_res > thr - 0.1):\n",
    "                return 'uncertain'\n",
    "            elif tp53_res > thr + 0.1:\n",
    "                return 'no_bi'\n",
    "            \n",
    "            \n",
    "            \n",
    "    # Samples with 2 tp53 mutations\n",
    "    elif nb_tp53muts == 2:\n",
    "        tp53_res_1 = maf_muts.tp53_res.values[0]\n",
    "        tp53_res_2 = maf_muts.tp53_res.values[1]\n",
    "        timing_1 = maf_muts.tp53_timing_wgd.values[0]\n",
    "        timing_2 = maf_muts.tp53_timing_wgd.values[1]\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr - 0.1) or (tp53_res_2 < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr + 0.1) and (tp53_res_2 > thr + 0.1):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (tp53_res_1 < thr_1 - 0.1) or (tp53_res_2 < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (tp53_res_1 < thr_1) or (tp53_res_2 < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 < thr_2 and tp53_res_2 < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res_1 < thr) and (tp53_res_2 < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if tp53_res_1 < thr_1 - 0.1 or tp53_res_2 < thr_1 - 0.1:\n",
    "                return 'bi'\n",
    "            elif (tp53_res_1 > thr_2) and (tp53_res_2 > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "            \n",
    "    # Samples with 3 tp53 mutations\n",
    "    elif nb_tp53muts == 3:\n",
    "        tp53_res_1 = maf_muts.tp53_res.values[0]\n",
    "        tp53_res_2 = maf_muts.tp53_res.values[1]\n",
    "        tp53_res_3 = maf_muts.tp53_res.values[2]\n",
    "        tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "        tp53_res.sort()\n",
    "        \n",
    "        timing_1 = maf_muts.tp53_timing_wgd.values[0]\n",
    "        timing_2 = maf_muts.tp53_timing_wgd.values[1]\n",
    "        timing_3 = maf_muts.tp53_timing_wgd.values[2]\n",
    "        timings = [timing_1, timing_2, timing_3]\n",
    "        BEF = [1 for x in timings if x=='BEFORE']\n",
    "        \n",
    "        # Number of mutations happening before wgd\n",
    "        nb_bef = len(BEF)\n",
    "        \n",
    "        if cn_state == 'LOSS BEFORE':\n",
    "            thr = 0.5\n",
    "            if (min(tp53_res) < thr - 0.1):\n",
    "                return 'bi'\n",
    "            elif (max(tp53_res) > thr + 0.1) :\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "            \n",
    "        if cn_state == 'CNLOH BEFORE & LOSS':\n",
    "            thr_1 = 0.5\n",
    "            thr_2 = 1.5\n",
    "            if (min(tp53_res) < thr_1 - 0.1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2): # we want that exactly 2 mutation arose before WGD\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state == 'CNLOH BEFORE':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if (min(tp53_res) < thr_1):\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] < thr_2):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "\n",
    "        elif cn_state =='LOSS AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='DOUBLE LOSS AFTER': \n",
    "            thr = 0.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state =='TETRAPLOID': \n",
    "            thr = 2.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "            \n",
    "        elif cn_state =='CNLOH AFTER': \n",
    "            thr = 1.5\n",
    "            if (tp53_res[1] < thr):\n",
    "                return 'bi'\n",
    "            else: return 'no_bi'\n",
    "        \n",
    "        elif cn_state == 'CNLOH BEFORE & GAIN':\n",
    "            thr_1 = 1.5\n",
    "            thr_2 = 2.5\n",
    "            if min(tp53_res) < thr_1 - 0.1 :\n",
    "                return 'bi'\n",
    "            elif (tp53_res[1] > thr_2):\n",
    "                return 'no_bi'\n",
    "            else: return 'uncertain'\n",
    "    \n",
    "    else: return 'uncertain'\n",
    "\n",
    "\n",
    "# This functions can only be called after get_bi_nobi() has been called on master\n",
    "def get_mono(x):\n",
    "    tumor = x.tumor_sample\n",
    "    cn_state = x.tp53_cn_state\n",
    "    tp53_count = x.tp53_count\n",
    "    maf_muts = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    nb_tp53muts = maf_muts.shape[0]\n",
    "    bi_state = x.bi_state_wgd\n",
    "    \n",
    "    if bi_state == 'bi':\n",
    "        return 'bi'\n",
    "    \n",
    "    elif bi_state == 'uncertain':\n",
    "        return 'uncertain_bi'\n",
    "    \n",
    "    \n",
    "    elif bi_state == 'no_bi': # We are already in the  mono/2WT distinction \n",
    "        \n",
    "        #1 mut samples\n",
    "        if nb_tp53muts == 1:\n",
    "            tp53_res = maf_muts.tp53_res.values[0]\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "\n",
    "        \n",
    "        # 2 mut samples   \n",
    "        elif nb_tp53muts == 2:\n",
    "            tp53_res_1 = maf_muts.tp53_res.values[0]\n",
    "            tp53_res_2 = maf_muts.tp53_res.values[1]\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if tp53_res_1 < 2.5 or tp53_res_2 < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res_1 > 2.5 and tp53_res_2 > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "            \n",
    "        # 3 mut samples\n",
    "        elif nb_tp53muts == 3:\n",
    "            tp53_res_1 = maf_muts.tp53_res.values[0]\n",
    "            tp53_res_2 = maf_muts.tp53_res.values[1]\n",
    "            tp53_res_3 = maf_muts.tp53_res.values[2]\n",
    "            tp53_res = [tp53_res_1,tp53_res_2,tp53_res_3]\n",
    "            tp53_res.sort()\n",
    "            \n",
    "            if cn_state == 'LOSS BEFORE':\n",
    "                return 'mono'\n",
    "            elif cn_state == 'CNLOH BEFORE' or cn_state == 'TETRAPLOID':\n",
    "                if min(tp53_res) < 2.5:\n",
    "                    return 'mono'\n",
    "                elif tp53_res[0] > 2.5:\n",
    "                    return '2WT'\n",
    "                \n",
    "            else: return 'uncertain_mono'\n",
    "        \n",
    "        else: return 'uncertain_mono' "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Merge Tables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "def merge_tables(cohort_type: str):\n",
    "    \n",
    "    if cohort == 'wgd':\n",
    "        cohort = cohort_wgd\n",
    "        maf_cohort = maf_cohort_wgd\n",
    "        clinical = clinical_wgd\n",
    "        arm_level = arm_level_wgd\n",
    "        gene_level = gene_level_wgd\n",
    "    elif cohort == 'no_wgd':\n",
    "        cohort = cohort_nowgd\n",
    "        maf_cohort = maf_cohort_nowgd\n",
    "        clinical = clinical_nowgd\n",
    "        arm_level = arm_level_nowgd\n",
    "        gene_level = gene_level_nowgd\n",
    "    \n",
    "    sample_info = create_sample_info(cohort=cohort_type)\n",
    "    tp53_muts = create_tp53_muts(cohort=cohort_type)\n",
    "    copy_number_info = create_copy_number_state(cohort=cohort_type)\n",
    "    \n",
    "    master = pd.merge(sample_info, tp53_muts, on='Tumor_Id')\n",
    "    master = pd.merge(master, copy_number_info, on='Sample_Id')\n",
    "    \n",
    "    return master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 5.65 s, sys: 71.6 ms, total: 5.73 s\n",
      "Wall time: 5.7 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "master = merge_tables(cohort_type=cohort_type)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [],
   "source": [
    "def create_gene_count(maf_cohort):\n",
    "    '''\n",
    "    This function create the count of genes mutated for each sample.\n",
    "    Arguments:\n",
    "        - maf_cohort: the maf_cohort file located in data/merged/data\n",
    "    '''\n",
    "    \n",
    "    # First we create the gene_count table by groupbying and sizing, we then change the index\n",
    "    selected_cohort = maf_cohort[['Sample_Id','Tumor_Id', 'Gene_Id']]\n",
    "    gene_count = pd.DataFrame(pd.DataFrame(selected_cohort[['Sample_Id', 'Gene_Id']].groupby(['Sample_Id', 'Gene_Id']).size(), columns = ['count']).groupby(['Sample_Id']).size(), columns = ['gene_count'])\n",
    "    gene_count = gene_count.reset_index()\n",
    "\n",
    "    # We add missing patients to the gene_count to have all the cohort\n",
    "    no_gene_id = selected_cohort.Gene_Id.isna()\n",
    "    no_gene_samples = set(selected_cohort[selected_cohort.index.isin(list(no_gene_id[no_gene_id == True].index))]['Sample_Id'])\n",
    "    missing_samp = pd.DataFrame(no_gene_samples, columns = ['Sample_Id'])\n",
    "\n",
    "    # We append the two dataframes\n",
    "    gene_count = gene_count.append(missing_samp)\n",
    "    \n",
    "    #Fillna with 0\n",
    "    gene_count = gene_count.fillna(0)\n",
    "\n",
    "    return gene_count\n",
    "\n",
    "def create_mut_count(x):\n",
    "    '''\n",
    "    This function computes the dataframe of all mutation count per sample.\n",
    "    '''\n",
    "    tumor = x.Tumor_Id\n",
    "    selected_cohort = maf_cohort[maf_cohort['Tumor_Sample_Barcode'] == tumor]\n",
    "    \n",
    "    return selected_cohort.shape[0]\n",
    "\n",
    "\n",
    "def get_driver_count(x, maf_cohort):\n",
    "    lookup_table = maf_cohort[maf_cohort['Sample_Id'] == x.Sample_Id]\n",
    "    h = get_groupby(lookup_table, 'oncogenic', 'count')\n",
    "    count = (int(h.loc['Oncogenic']) if 'Oncogenic' in h.index else 0) + (int(h.loc['Likely Oncogenic']) if 'Likely Oncogenic' in h.index else 0) +(int(h.loc['Predicted Oncogenic']) if 'Predicted Oncogenic' in h.index else 0)\n",
    "    return count\n",
    "\n",
    "def create_driver_count(maf_cohort):\n",
    "    '''In this function we count the number of mutation driver per sample.'''\n",
    "    samples=list(set(maf_cohort.Sample_Id))\n",
    "    driver_count = pd.DataFrame(columns=['Sample_Id', 'driver_count'])\n",
    "    driver_count.Sample_Id = samples\n",
    "    driver_count['driver_count'] = driver_count.apply(get_driver_count, maf_cohort=maf_cohort, axis=1)\n",
    "\n",
    "    return driver_count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 132,
   "metadata": {},
   "outputs": [],
   "source": [
    "master['mut_count'] = master.apply(create_mut_count, axis=1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Sample_Id</th>\n",
       "      <th>Tumor_Id</th>\n",
       "      <th>Patient_Id</th>\n",
       "      <th>Cancer_Type</th>\n",
       "      <th>Cancer_Type_Detailed</th>\n",
       "      <th>Patient_Current_Age</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>purity</th>\n",
       "      <th>ploidy</th>\n",
       "      <th>Overall_Survival_Months</th>\n",
       "      <th>Overall_Survival_Status</th>\n",
       "      <th>MSI_Score</th>\n",
       "      <th>MSI_Type</th>\n",
       "      <th>TMB_Score</th>\n",
       "      <th>tp53_key_1</th>\n",
       "      <th>tp53_vc_1</th>\n",
       "      <th>tp53_ccf_1</th>\n",
       "      <th>tp53_vaf_1</th>\n",
       "      <th>tp53_HGVSp_1</th>\n",
       "      <th>tp53_spot_1</th>\n",
       "      <th>tp53_key_2</th>\n",
       "      <th>tp53_vc_2</th>\n",
       "      <th>tp53_ccf_2</th>\n",
       "      <th>tp53_vaf_2</th>\n",
       "      <th>tp53_HGVSp_2</th>\n",
       "      <th>tp53_spot_2</th>\n",
       "      <th>tp53_key_3</th>\n",
       "      <th>tp53_vc_3</th>\n",
       "      <th>tp53_ccf_3</th>\n",
       "      <th>tp53_vaf_3</th>\n",
       "      <th>tp53_HGVSp_3</th>\n",
       "      <th>tp53_spot_3</th>\n",
       "      <th>tp53_key_4</th>\n",
       "      <th>tp53_vc_4</th>\n",
       "      <th>tp53_ccf_4</th>\n",
       "      <th>tp53_vaf_4</th>\n",
       "      <th>tp53_HGVSp_4</th>\n",
       "      <th>tp53_spot_4</th>\n",
       "      <th>tp53_key_5</th>\n",
       "      <th>tp53_vc_5</th>\n",
       "      <th>tp53_ccf_5</th>\n",
       "      <th>tp53_vaf_5</th>\n",
       "      <th>tp53_HGVSp_5</th>\n",
       "      <th>tp53_spot_5</th>\n",
       "      <th>tp53_count</th>\n",
       "      <th>tp53_tcn</th>\n",
       "      <th>tp53_mcn</th>\n",
       "      <th>tp53_lcn</th>\n",
       "      <th>tp53_seg_length</th>\n",
       "      <th>tp53_cn_state</th>\n",
       "      <th>tp53_cf</th>\n",
       "      <th>wgd</th>\n",
       "      <th>mut_count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>P-0025956-T01-IM6_P-0025956-N01-IM6</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>P-0025956</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>71.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>3.496971</td>\n",
       "      <td>3.584</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>80668431</td>\n",
       "      <td>LOSS AFTER</td>\n",
       "      <td>0.169170</td>\n",
       "      <td>True</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>P-0036909-T01-IM6_P-0036909-N01-IM6</td>\n",
       "      <td>P-0036909-T01-IM6</td>\n",
       "      <td>P-0036909</td>\n",
       "      <td>Non-Small Cell Lung Cancer</td>\n",
       "      <td>Lung Adenocarcinoma</td>\n",
       "      <td>47.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.391316</td>\n",
       "      <td>2.871793</td>\n",
       "      <td>14.137</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "      <td>P-0036909-T01-IM617_7577121_G_A</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.798</td>\n",
       "      <td>0.312169</td>\n",
       "      <td>p.Arg273Cys</td>\n",
       "      <td>273</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>18101929</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.306121</td>\n",
       "      <td>True</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>P-0023546-T01-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T01-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Neuroendocrine Carcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.865628</td>\n",
       "      <td>3.115253</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>2.37</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "      <td>P-0023546-T01-IM617_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.933</td>\n",
       "      <td>0.845070</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25250470</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.835504</td>\n",
       "      <td>True</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>P-0023546-T02-IM6_P-0023546-N01-IM6</td>\n",
       "      <td>P-0023546-T02-IM6</td>\n",
       "      <td>P-0023546</td>\n",
       "      <td>Prostate Cancer</td>\n",
       "      <td>Prostate Adenocarcinoma</td>\n",
       "      <td>50.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>3.136841</td>\n",
       "      <td>4.800</td>\n",
       "      <td>DECEASED</td>\n",
       "      <td>0.82</td>\n",
       "      <td>Stable</td>\n",
       "      <td>2.6</td>\n",
       "      <td>P-0023546-T02-IM617_7578442_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.636735</td>\n",
       "      <td>p.Tyr163Cys</td>\n",
       "      <td>163</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25237770</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.312907</td>\n",
       "      <td>True</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>P-0018837-T01-IM6_P-0018837-N01-IM6</td>\n",
       "      <td>P-0018837-T01-IM6</td>\n",
       "      <td>P-0018837</td>\n",
       "      <td>Colorectal Cancer</td>\n",
       "      <td>Colon Adenocarcinoma</td>\n",
       "      <td>60.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.351778</td>\n",
       "      <td>3.727190</td>\n",
       "      <td>34.060</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.45</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>P-0018837-T01-IM617_7578406_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>1.000</td>\n",
       "      <td>0.325843</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>175</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>25231975</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.257400</td>\n",
       "      <td>True</td>\n",
       "      <td>6</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9893</th>\n",
       "      <td>P-0050750-T01-IM6_P-0050750-N01-IM6</td>\n",
       "      <td>P-0050750-T01-IM6</td>\n",
       "      <td>P-0050750</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>55.0</td>\n",
       "      <td>Metastasis</td>\n",
       "      <td>0.301523</td>\n",
       "      <td>3.845451</td>\n",
       "      <td>0.099</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.05</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>P-0050750-T01-IM617_7578406_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.946</td>\n",
       "      <td>0.285229</td>\n",
       "      <td>p.Arg175His</td>\n",
       "      <td>175</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>13849975</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.286332</td>\n",
       "      <td>True</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9894</th>\n",
       "      <td>P-0050643-T01-IM6_P-0050643-N01-IM6</td>\n",
       "      <td>P-0050643-T01-IM6</td>\n",
       "      <td>P-0050643</td>\n",
       "      <td>Breast Cancer</td>\n",
       "      <td>Breast Invasive Ductal Carcinoma</td>\n",
       "      <td>53.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.667936</td>\n",
       "      <td>4.419218</td>\n",
       "      <td>1.315</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.30</td>\n",
       "      <td>Stable</td>\n",
       "      <td>4.4</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>11917531</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.545007</td>\n",
       "      <td>True</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9895</th>\n",
       "      <td>P-0050223-T02-IM6_P-0050223-N02-IM6</td>\n",
       "      <td>P-0050223-T02-IM6</td>\n",
       "      <td>P-0050223</td>\n",
       "      <td>Hepatobiliary Cancer</td>\n",
       "      <td>Cholangiocarcinoma</td>\n",
       "      <td>67.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.215128</td>\n",
       "      <td>4.725790</td>\n",
       "      <td>1.940</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.00</td>\n",
       "      <td>Stable</td>\n",
       "      <td>3.5</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>0</td>\n",
       "      <td>4</td>\n",
       "      <td>2.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>80668650</td>\n",
       "      <td>TETRAPLOID</td>\n",
       "      <td>NaN</td>\n",
       "      <td>True</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9896</th>\n",
       "      <td>P-0048760-T01-IM6_P-0048760-N01-IM6</td>\n",
       "      <td>P-0048760-T01-IM6</td>\n",
       "      <td>P-0048760</td>\n",
       "      <td>Bone Cancer</td>\n",
       "      <td>Osteosarcoma</td>\n",
       "      <td>18.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.614603</td>\n",
       "      <td>3.947721</td>\n",
       "      <td>3.682</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>5.79</td>\n",
       "      <td>Indeterminate</td>\n",
       "      <td>6.1</td>\n",
       "      <td>P-0048760-T01-IM617_7578413_C_T</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.958</td>\n",
       "      <td>0.675637</td>\n",
       "      <td>p.Val173Met</td>\n",
       "      <td>173</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>3</td>\n",
       "      <td>3.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>7438283</td>\n",
       "      <td>CNLOH BEFORE &amp; LOSS</td>\n",
       "      <td>0.598599</td>\n",
       "      <td>True</td>\n",
       "      <td>7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9897</th>\n",
       "      <td>P-0050744-T01-IM6_P-0050744-N01-IM6</td>\n",
       "      <td>P-0050744-T01-IM6</td>\n",
       "      <td>P-0050744</td>\n",
       "      <td>Pancreatic Cancer</td>\n",
       "      <td>Pancreatic Adenocarcinoma</td>\n",
       "      <td>69.0</td>\n",
       "      <td>Primary</td>\n",
       "      <td>0.300000</td>\n",
       "      <td>2.426452</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LIVING</td>\n",
       "      <td>0.06</td>\n",
       "      <td>Stable</td>\n",
       "      <td>5.3</td>\n",
       "      <td>P-0050744-T01-IM617_7577580_T_C</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>0.797</td>\n",
       "      <td>0.239130</td>\n",
       "      <td>p.Tyr234Cys</td>\n",
       "      <td>234</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>30207093</td>\n",
       "      <td>LOSS BEFORE</td>\n",
       "      <td>0.205450</td>\n",
       "      <td>True</td>\n",
       "      <td>5</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>9898 rows Ã— 53 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                Sample_Id           Tumor_Id Patient_Id  \\\n",
       "0     P-0025956-T01-IM6_P-0025956-N01-IM6  P-0025956-T01-IM6  P-0025956   \n",
       "1     P-0036909-T01-IM6_P-0036909-N01-IM6  P-0036909-T01-IM6  P-0036909   \n",
       "2     P-0023546-T01-IM6_P-0023546-N01-IM6  P-0023546-T01-IM6  P-0023546   \n",
       "3     P-0023546-T02-IM6_P-0023546-N01-IM6  P-0023546-T02-IM6  P-0023546   \n",
       "4     P-0018837-T01-IM6_P-0018837-N01-IM6  P-0018837-T01-IM6  P-0018837   \n",
       "...                                   ...                ...        ...   \n",
       "9893  P-0050750-T01-IM6_P-0050750-N01-IM6  P-0050750-T01-IM6  P-0050750   \n",
       "9894  P-0050643-T01-IM6_P-0050643-N01-IM6  P-0050643-T01-IM6  P-0050643   \n",
       "9895  P-0050223-T02-IM6_P-0050223-N02-IM6  P-0050223-T02-IM6  P-0050223   \n",
       "9896  P-0048760-T01-IM6_P-0048760-N01-IM6  P-0048760-T01-IM6  P-0048760   \n",
       "9897  P-0050744-T01-IM6_P-0050744-N01-IM6  P-0050744-T01-IM6  P-0050744   \n",
       "\n",
       "                     Cancer_Type               Cancer_Type_Detailed  \\\n",
       "0     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "1     Non-Small Cell Lung Cancer                Lung Adenocarcinoma   \n",
       "2                Prostate Cancer  Prostate Neuroendocrine Carcinoma   \n",
       "3                Prostate Cancer            Prostate Adenocarcinoma   \n",
       "4              Colorectal Cancer               Colon Adenocarcinoma   \n",
       "...                          ...                                ...   \n",
       "9893           Pancreatic Cancer          Pancreatic Adenocarcinoma   \n",
       "9894               Breast Cancer   Breast Invasive Ductal Carcinoma   \n",
       "9895        Hepatobiliary Cancer                 Cholangiocarcinoma   \n",
       "9896                 Bone Cancer                       Osteosarcoma   \n",
       "9897           Pancreatic Cancer          Pancreatic Adenocarcinoma   \n",
       "\n",
       "      Patient_Current_Age Sample_Type    purity    ploidy  \\\n",
       "0                    71.0     Primary  0.273767  3.496971   \n",
       "1                    47.0  Metastasis  0.391316  2.871793   \n",
       "2                    50.0     Primary  0.865628  3.115253   \n",
       "3                    50.0     Primary  0.312907  3.136841   \n",
       "4                    60.0     Primary  0.351778  3.727190   \n",
       "...                   ...         ...       ...       ...   \n",
       "9893                 55.0  Metastasis  0.301523  3.845451   \n",
       "9894                 53.0     Primary  0.667936  4.419218   \n",
       "9895                 67.0     Primary  0.215128  4.725790   \n",
       "9896                 18.0     Primary  0.614603  3.947721   \n",
       "9897                 69.0     Primary  0.300000  2.426452   \n",
       "\n",
       "      Overall_Survival_Months Overall_Survival_Status  MSI_Score  \\\n",
       "0                       3.584                DECEASED       0.00   \n",
       "1                      14.137                  LIVING       0.37   \n",
       "2                       4.800                DECEASED       2.37   \n",
       "3                       4.800                DECEASED       0.82   \n",
       "4                      34.060                  LIVING       0.45   \n",
       "...                       ...                     ...        ...   \n",
       "9893                    0.099                  LIVING       0.05   \n",
       "9894                    1.315                  LIVING       0.30   \n",
       "9895                    1.940                  LIVING       0.00   \n",
       "9896                    3.682                  LIVING       5.79   \n",
       "9897                      NaN                  LIVING       0.06   \n",
       "\n",
       "           MSI_Type  TMB_Score                       tp53_key_1  \\\n",
       "0            Stable        5.3                             None   \n",
       "1            Stable        3.5  P-0036909-T01-IM617_7577121_G_A   \n",
       "2            Stable        3.5  P-0023546-T01-IM617_7578442_T_C   \n",
       "3            Stable        2.6  P-0023546-T02-IM617_7578442_T_C   \n",
       "4            Stable        5.3  P-0018837-T01-IM617_7578406_C_T   \n",
       "...             ...        ...                              ...   \n",
       "9893         Stable        5.3  P-0050750-T01-IM617_7578406_C_T   \n",
       "9894         Stable        4.4                             None   \n",
       "9895         Stable        3.5                             None   \n",
       "9896  Indeterminate        6.1  P-0048760-T01-IM617_7578413_C_T   \n",
       "9897         Stable        5.3  P-0050744-T01-IM617_7577580_T_C   \n",
       "\n",
       "              tp53_vc_1  tp53_ccf_1  tp53_vaf_1 tp53_HGVSp_1 tp53_spot_1  \\\n",
       "0                  None         NaN         NaN         None        None   \n",
       "1     Missense_Mutation       0.798    0.312169  p.Arg273Cys         273   \n",
       "2     Missense_Mutation       0.933    0.845070  p.Tyr163Cys         163   \n",
       "3     Missense_Mutation       1.000    0.636735  p.Tyr163Cys         163   \n",
       "4     Missense_Mutation       1.000    0.325843  p.Arg175His         175   \n",
       "...                 ...         ...         ...          ...         ...   \n",
       "9893  Missense_Mutation       0.946    0.285229  p.Arg175His         175   \n",
       "9894               None         NaN         NaN         None        None   \n",
       "9895               None         NaN         NaN         None        None   \n",
       "9896  Missense_Mutation       0.958    0.675637  p.Val173Met         173   \n",
       "9897  Missense_Mutation       0.797    0.239130  p.Tyr234Cys         234   \n",
       "\n",
       "     tp53_key_2 tp53_vc_2  tp53_ccf_2  tp53_vaf_2 tp53_HGVSp_2 tp53_spot_2  \\\n",
       "0          None      None         NaN         NaN         None        None   \n",
       "1          None      None         NaN         NaN         None        None   \n",
       "2          None      None         NaN         NaN         None        None   \n",
       "3          None      None         NaN         NaN         None        None   \n",
       "4          None      None         NaN         NaN         None        None   \n",
       "...         ...       ...         ...         ...          ...         ...   \n",
       "9893       None      None         NaN         NaN         None        None   \n",
       "9894       None      None         NaN         NaN         None        None   \n",
       "9895       None      None         NaN         NaN         None        None   \n",
       "9896       None      None         NaN         NaN         None        None   \n",
       "9897       None      None         NaN         NaN         None        None   \n",
       "\n",
       "     tp53_key_3 tp53_vc_3  tp53_ccf_3  tp53_vaf_3 tp53_HGVSp_3 tp53_spot_3  \\\n",
       "0          None      None         NaN         NaN         None        None   \n",
       "1          None      None         NaN         NaN         None        None   \n",
       "2          None      None         NaN         NaN         None        None   \n",
       "3          None      None         NaN         NaN         None        None   \n",
       "4          None      None         NaN         NaN         None        None   \n",
       "...         ...       ...         ...         ...          ...         ...   \n",
       "9893       None      None         NaN         NaN         None        None   \n",
       "9894       None      None         NaN         NaN         None        None   \n",
       "9895       None      None         NaN         NaN         None        None   \n",
       "9896       None      None         NaN         NaN         None        None   \n",
       "9897       None      None         NaN         NaN         None        None   \n",
       "\n",
       "     tp53_key_4 tp53_vc_4  tp53_ccf_4  tp53_vaf_4 tp53_HGVSp_4 tp53_spot_4  \\\n",
       "0          None      None         NaN         NaN         None        None   \n",
       "1          None      None         NaN         NaN         None        None   \n",
       "2          None      None         NaN         NaN         None        None   \n",
       "3          None      None         NaN         NaN         None        None   \n",
       "4          None      None         NaN         NaN         None        None   \n",
       "...         ...       ...         ...         ...          ...         ...   \n",
       "9893       None      None         NaN         NaN         None        None   \n",
       "9894       None      None         NaN         NaN         None        None   \n",
       "9895       None      None         NaN         NaN         None        None   \n",
       "9896       None      None         NaN         NaN         None        None   \n",
       "9897       None      None         NaN         NaN         None        None   \n",
       "\n",
       "     tp53_key_5 tp53_vc_5  tp53_ccf_5  tp53_vaf_5 tp53_HGVSp_5 tp53_spot_5  \\\n",
       "0          None      None         NaN         NaN         None        None   \n",
       "1          None      None         NaN         NaN         None        None   \n",
       "2          None      None         NaN         NaN         None        None   \n",
       "3          None      None         NaN         NaN         None        None   \n",
       "4          None      None         NaN         NaN         None        None   \n",
       "...         ...       ...         ...         ...          ...         ...   \n",
       "9893       None      None         NaN         NaN         None        None   \n",
       "9894       None      None         NaN         NaN         None        None   \n",
       "9895       None      None         NaN         NaN         None        None   \n",
       "9896       None      None         NaN         NaN         None        None   \n",
       "9897       None      None         NaN         NaN         None        None   \n",
       "\n",
       "      tp53_count  tp53_tcn  tp53_mcn  tp53_lcn  tp53_seg_length  \\\n",
       "0              0         3       2.0       1.0         80668431   \n",
       "1              1         2       2.0       0.0         18101929   \n",
       "2              1         3       3.0       0.0         25250470   \n",
       "3              1         2       2.0       0.0         25237770   \n",
       "4              1         3       3.0       0.0         25231975   \n",
       "...          ...       ...       ...       ...              ...   \n",
       "9893           1         2       2.0       0.0         13849975   \n",
       "9894           0         3       3.0       0.0         11917531   \n",
       "9895           0         4       2.0       2.0         80668650   \n",
       "9896           1         3       3.0       0.0          7438283   \n",
       "9897           1         2       2.0       0.0         30207093   \n",
       "\n",
       "            tp53_cn_state   tp53_cf   wgd  mut_count  \n",
       "0              LOSS AFTER  0.169170  True          6  \n",
       "1             LOSS BEFORE  0.306121  True          4  \n",
       "2     CNLOH BEFORE & LOSS  0.835504  True          4  \n",
       "3             LOSS BEFORE  0.312907  True          3  \n",
       "4     CNLOH BEFORE & LOSS  0.257400  True          6  \n",
       "...                   ...       ...   ...        ...  \n",
       "9893          LOSS BEFORE  0.286332  True          5  \n",
       "9894  CNLOH BEFORE & LOSS  0.545007  True          5  \n",
       "9895           TETRAPLOID       NaN  True          4  \n",
       "9896  CNLOH BEFORE & LOSS  0.598599  True          7  \n",
       "9897          LOSS BEFORE  0.205450  True          5  \n",
       "\n",
       "[9898 rows x 53 columns]"
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 134,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Hugo_Symbol</th>\n",
       "      <th>Entrez_Gene_Id</th>\n",
       "      <th>Center</th>\n",
       "      <th>NCBI_Build</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Start_Position</th>\n",
       "      <th>End_Position</th>\n",
       "      <th>Strand</th>\n",
       "      <th>Variant_Classification</th>\n",
       "      <th>Variant_Type</th>\n",
       "      <th>Reference_Allele</th>\n",
       "      <th>Tumor_Seq_Allele1</th>\n",
       "      <th>Tumor_Seq_Allele2</th>\n",
       "      <th>dbSNP_RS</th>\n",
       "      <th>dbSNP_Val_Status</th>\n",
       "      <th>Tumor_Sample_Barcode</th>\n",
       "      <th>Matched_Norm_Sample_Barcode</th>\n",
       "      <th>Match_Norm_Seq_Allele1</th>\n",
       "      <th>Match_Norm_Seq_Allele2</th>\n",
       "      <th>Tumor_Validation_Allele1</th>\n",
       "      <th>Tumor_Validation_Allele2</th>\n",
       "      <th>Match_Norm_Validation_Allele1</th>\n",
       "      <th>Match_Norm_Validation_Allele2</th>\n",
       "      <th>Verification_Status</th>\n",
       "      <th>Validation_Status</th>\n",
       "      <th>Mutation_Status</th>\n",
       "      <th>Sequencing_Phase</th>\n",
       "      <th>Sequence_Source</th>\n",
       "      <th>Validation_Method</th>\n",
       "      <th>Score</th>\n",
       "      <th>BAM_File</th>\n",
       "      <th>Sequencer</th>\n",
       "      <th>Tumor_Sample_UUID</th>\n",
       "      <th>Matched_Norm_Sample_UUID</th>\n",
       "      <th>HGVSc</th>\n",
       "      <th>HGVSp</th>\n",
       "      <th>HGVSp_Short</th>\n",
       "      <th>Transcript_ID</th>\n",
       "      <th>Exon_Number</th>\n",
       "      <th>t_depth</th>\n",
       "      <th>t_ref_count</th>\n",
       "      <th>t_alt_count</th>\n",
       "      <th>n_depth</th>\n",
       "      <th>n_ref_count</th>\n",
       "      <th>n_alt_count</th>\n",
       "      <th>all_effects</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>Amino_acids</th>\n",
       "      <th>Codons</th>\n",
       "      <th>Existing_variation</th>\n",
       "      <th>ALLELE_NUM</th>\n",
       "      <th>DISTANCE</th>\n",
       "      <th>STRAND_VEP</th>\n",
       "      <th>SYMBOL</th>\n",
       "      <th>SYMBOL_SOURCE</th>\n",
       "      <th>HGNC_ID</th>\n",
       "      <th>BIOTYPE</th>\n",
       "      <th>CANONICAL</th>\n",
       "      <th>CCDS</th>\n",
       "      <th>ENSP</th>\n",
       "      <th>SWISSPROT</th>\n",
       "      <th>TREMBL</th>\n",
       "      <th>UNIPARC</th>\n",
       "      <th>RefSeq</th>\n",
       "      <th>SIFT</th>\n",
       "      <th>PolyPhen</th>\n",
       "      <th>EXON</th>\n",
       "      <th>INTRON</th>\n",
       "      <th>DOMAINS</th>\n",
       "      <th>GMAF</th>\n",
       "      <th>AFR_MAF</th>\n",
       "      <th>AMR_MAF</th>\n",
       "      <th>ASN_MAF</th>\n",
       "      <th>EAS_MAF</th>\n",
       "      <th>EUR_MAF</th>\n",
       "      <th>SAS_MAF</th>\n",
       "      <th>AA_MAF</th>\n",
       "      <th>EA_MAF</th>\n",
       "      <th>CLIN_SIG</th>\n",
       "      <th>SOMATIC</th>\n",
       "      <th>PUBMED</th>\n",
       "      <th>MOTIF_NAME</th>\n",
       "      <th>MOTIF_POS</th>\n",
       "      <th>HIGH_INF_POS</th>\n",
       "      <th>MOTIF_SCORE_CHANGE</th>\n",
       "      <th>IMPACT</th>\n",
       "      <th>PICK</th>\n",
       "      <th>VARIANT_CLASS</th>\n",
       "      <th>TSL</th>\n",
       "      <th>HGVS_OFFSET</th>\n",
       "      <th>PHENO</th>\n",
       "      <th>MINIMISED</th>\n",
       "      <th>ExAC_AF</th>\n",
       "      <th>ExAC_AF_AFR</th>\n",
       "      <th>ExAC_AF_AMR</th>\n",
       "      <th>ExAC_AF_EAS</th>\n",
       "      <th>ExAC_AF_FIN</th>\n",
       "      <th>ExAC_AF_NFE</th>\n",
       "      <th>ExAC_AF_OTH</th>\n",
       "      <th>ExAC_AF_SAS</th>\n",
       "      <th>GENE_PHENO</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>flanking_bps</th>\n",
       "      <th>variant_id</th>\n",
       "      <th>variant_qual</th>\n",
       "      <th>ExAC_AF_Adj</th>\n",
       "      <th>ExAC_AC_AN_Adj</th>\n",
       "      <th>ExAC_AC_AN</th>\n",
       "      <th>ExAC_AC_AN_AFR</th>\n",
       "      <th>ExAC_AC_AN_AMR</th>\n",
       "      <th>ExAC_AC_AN_EAS</th>\n",
       "      <th>ExAC_AC_AN_FIN</th>\n",
       "      <th>ExAC_AC_AN_NFE</th>\n",
       "      <th>ExAC_AC_AN_OTH</th>\n",
       "      <th>ExAC_AC_AN_SAS</th>\n",
       "      <th>ExAC_FILTER</th>\n",
       "      <th>Caller</th>\n",
       "      <th>is-a-hotspot</th>\n",
       "      <th>is-a-3d-hotspot</th>\n",
       "      <th>mutation_effect</th>\n",
       "      <th>oncogenic</th>\n",
       "      <th>LEVEL_1</th>\n",
       "      <th>LEVEL_2A</th>\n",
       "      <th>LEVEL_2B</th>\n",
       "      <th>LEVEL_3A</th>\n",
       "      <th>LEVEL_3B</th>\n",
       "      <th>LEVEL_4</th>\n",
       "      <th>LEVEL_R1</th>\n",
       "      <th>LEVEL_R2</th>\n",
       "      <th>LEVEL_R3</th>\n",
       "      <th>Highest_level</th>\n",
       "      <th>citations</th>\n",
       "      <th>driver</th>\n",
       "      <th>tcn</th>\n",
       "      <th>lcn</th>\n",
       "      <th>cf</th>\n",
       "      <th>purity</th>\n",
       "      <th>t_var_freq</th>\n",
       "      <th>expected_alt_copies</th>\n",
       "      <th>ccf_Mcopies</th>\n",
       "      <th>ccf_Mcopies_lower</th>\n",
       "      <th>ccf_Mcopies_upper</th>\n",
       "      <th>ccf_Mcopies_prob95</th>\n",
       "      <th>ccf_Mcopies_prob90</th>\n",
       "      <th>ccf_1copy</th>\n",
       "      <th>ccf_1copy_lower</th>\n",
       "      <th>ccf_1copy_upper</th>\n",
       "      <th>ccf_1copy_prob95</th>\n",
       "      <th>ccf_1copy_prob90</th>\n",
       "      <th>ccf_expected_copies</th>\n",
       "      <th>ccf_expected_copies_lower</th>\n",
       "      <th>ccf_expected_copies_upper</th>\n",
       "      <th>ccf_expected_copies_prob95</th>\n",
       "      <th>ccf_expected_copies_prob90</th>\n",
       "      <th>facets_fit</th>\n",
       "      <th>reviewer_set_purity</th>\n",
       "      <th>use_only_purity_run</th>\n",
       "      <th>use_edited_cncf</th>\n",
       "      <th>cncf_file_used</th>\n",
       "      <th>mut_key</th>\n",
       "      <th>sample_mut_key</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>KRAS</td>\n",
       "      <td>3845</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>12</td>\n",
       "      <td>25398285</td>\n",
       "      <td>25398285</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.34G&gt;T</td>\n",
       "      <td>p.Gly12Cys</td>\n",
       "      <td>p.G12C</td>\n",
       "      <td>ENST00000256078</td>\n",
       "      <td>2/6</td>\n",
       "      <td>588.0</td>\n",
       "      <td>495.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>908</td>\n",
       "      <td>908</td>\n",
       "      <td>0</td>\n",
       "      <td>KRAS,missense_variant,p.Gly12Cys,ENST00000311936,NM_004985.3;KRAS,missense_variant,p.Gly12Cys,ENST00000556131,;KRAS,missense_variant,p.Gly12Cys,ENST00000256078,NM_033360.2;KRAS,missense_variant,p.Gly12Cys,ENST00000557334,;</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000133703</td>\n",
       "      <td>ENST00000256078</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>98/1119</td>\n",
       "      <td>34/570</td>\n",
       "      <td>12/189</td>\n",
       "      <td>G/C</td>\n",
       "      <td>Ggt/Tgt</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>KRAS</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>6407.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS8703.1</td>\n",
       "      <td>ENSP00000256078</td>\n",
       "      <td>P01116</td>\n",
       "      <td>Q9UM97,Q71SP6,P78460,L7RSL8,I1SRC5</td>\n",
       "      <td>UPI0000133132</td>\n",
       "      <td>NM_033360.2</td>\n",
       "      <td>deleterious(0.04)</td>\n",
       "      <td>probably_damaging(0.993)</td>\n",
       "      <td>2/6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:3.40.50.300,Pfam_domain:PF00071,Prints_domain:PR00449,PROSITE_profiles:PS51421,hmmpanther:PTHR24070,hmmpanther:PTHR24070:SF186,Low_complexity_(Seg):seg,SMART_domains:SM00173,SMART_domains:SM00174,SMART_domains:SM00175,SMART_domains:SM00176,Superfamily_domains:SSF52540,TIGRFAM_domain:TIGR00231</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Y</td>\n",
       "      <td>Y</td>\n",
       "      <td>Gain-of-function</td>\n",
       "      <td>Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Binimetinib,Cobimetinib,Trametinib</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>LEVEL_4</td>\n",
       "      <td>16051643;25705018;26841430;24256730;28783719;29247021;Bhagwat et al. Abstract# 4973, AACR 2017(http://cancerres.aacrjournals.org/content/77/13_Supplement/4973);Robarge et al. Abstract# DDT02-03, AACR 2014(http://cancerres.aacrjournals.org/content/74/19_Supplement/DDT02-03);Burrows et al. Abstract# 5168, AACR 2017(http://cancerres.aacrjournals.org/content/77/13_Supplement/5168)</td>\n",
       "      <td>True</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.138437</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.158163</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.736</td>\n",
       "      <td>0.655</td>\n",
       "      <td>0.821</td>\n",
       "      <td>0.002236</td>\n",
       "      <td>0.014834</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.860769</td>\n",
       "      <td>0.984692</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.860769</td>\n",
       "      <td>0.984692</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>12_25398285_C_A</td>\n",
       "      <td>P-0025956-T01-IM612_25398285_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>STK11</td>\n",
       "      <td>6794</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>19</td>\n",
       "      <td>1221314</td>\n",
       "      <td>1221314</td>\n",
       "      <td>+</td>\n",
       "      <td>Frame_Shift_Del</td>\n",
       "      <td>DEL</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>-</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.837delC</td>\n",
       "      <td>p.Pro281ArgfsTer6</td>\n",
       "      <td>p.P281Rfs*6</td>\n",
       "      <td>ENST00000326873</td>\n",
       "      <td>6/10</td>\n",
       "      <td>615.0</td>\n",
       "      <td>492.0</td>\n",
       "      <td>123.0</td>\n",
       "      <td>885</td>\n",
       "      <td>884</td>\n",
       "      <td>1</td>\n",
       "      <td>STK11,frameshift_variant,p.Pro281ArgfsTer6,ENST00000326873,NM_000455.4;STK11,frameshift_variant,p.Pro37ArgfsTer6,ENST00000586243,;STK11,downstream_gene_variant,,ENST00000585851,;STK11,downstream_gene_variant,,ENST00000585748,;STK11,3_prime_UTR_variant,,ENST00000593219,;STK11,non_coding_transcript_exon_variant,,ENST00000589152,;STK11,non_coding_transcript_exon_variant,,ENST00000591133,;STK11,non_coding_transcript_exon_variant,,ENST00000586358,;STK11,upstream_gene_variant,,ENST00000585465,;,regulatory_region_variant,,ENSR00000377818,;</td>\n",
       "      <td>-</td>\n",
       "      <td>ENSG00000118046</td>\n",
       "      <td>ENST00000326873</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>2010/3328</td>\n",
       "      <td>837/1302</td>\n",
       "      <td>279/433</td>\n",
       "      <td>G/X</td>\n",
       "      <td>ggC/gg</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>STK11</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11389.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS45896.1</td>\n",
       "      <td>ENSP00000324856</td>\n",
       "      <td>Q15831</td>\n",
       "      <td>Q9NS52</td>\n",
       "      <td>UPI0000136105</td>\n",
       "      <td>NM_000455.4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:1.10.510.10,Pfam_domain:PF00069,PROSITE_profiles:PS50011,hmmpanther:PTHR24347,hmmpanther:PTHR24347:SF1,SMART_domains:SM00220,Superfamily_domains:SSF56112</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>1.0</td>\n",
       "      <td>sequence_alteration</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGCC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24652667;19892943;25079552;21516316;19340305</td>\n",
       "      <td>True</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.183720</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.200000</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.731</td>\n",
       "      <td>0.662</td>\n",
       "      <td>0.802</td>\n",
       "      <td>0.000266</td>\n",
       "      <td>0.003571</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.984</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.926527</td>\n",
       "      <td>0.996079</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.984</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.926527</td>\n",
       "      <td>0.996079</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>19_1221314_C_-</td>\n",
       "      <td>P-0025956-T01-IM619_1221314_C_-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>KEAP1</td>\n",
       "      <td>9817</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>19</td>\n",
       "      <td>10602584</td>\n",
       "      <td>10602584</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.994G&gt;A</td>\n",
       "      <td>p.Gly332Ser</td>\n",
       "      <td>p.G332S</td>\n",
       "      <td>ENST00000171111</td>\n",
       "      <td>3/6</td>\n",
       "      <td>641.0</td>\n",
       "      <td>523.0</td>\n",
       "      <td>118.0</td>\n",
       "      <td>471</td>\n",
       "      <td>471</td>\n",
       "      <td>0</td>\n",
       "      <td>KEAP1,missense_variant,p.Gly332Ser,ENST00000171111,NM_203500.1;KEAP1,missense_variant,p.Gly332Ser,ENST00000393623,NM_012289.3;KEAP1,upstream_gene_variant,,ENST00000592478,;KEAP1,downstream_gene_variant,,ENST00000591419,;KEAP1,downstream_gene_variant,,ENST00000592055,;CTC-429L19.3,upstream_gene_variant,,ENST00000592671,;KEAP1,downstream_gene_variant,,ENST00000588024,;KEAP1,downstream_gene_variant,,ENST00000585845,;KEAP1,upstream_gene_variant,,ENST00000590593,;KEAP1,upstream_gene_variant,,ENST00000590237,;</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000079999</td>\n",
       "      <td>ENST00000171111</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>1542/2955</td>\n",
       "      <td>994/1875</td>\n",
       "      <td>332/624</td>\n",
       "      <td>G/S</td>\n",
       "      <td>Ggc/Agc</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>KEAP1</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>23177.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>YES</td>\n",
       "      <td>CCDS12239.1</td>\n",
       "      <td>ENSP00000171111</td>\n",
       "      <td>Q14145</td>\n",
       "      <td>K7ESE0,K7EJD8,K7EJ49</td>\n",
       "      <td>UPI000007139C</td>\n",
       "      <td>NM_203500.1</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.998)</td>\n",
       "      <td>3/6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>hmmpanther:PTHR24412:SF162,hmmpanther:PTHR24412,Pfam_domain:PF01344,Gene3D:1k3iA02,PIRSF_domain:PIRSF037037,SMART_domains:SM00612,Superfamily_domains:0052715</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>1.0</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCC</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.183720</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.184087</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.672</td>\n",
       "      <td>0.608</td>\n",
       "      <td>0.740</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>0.000093</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.880435</td>\n",
       "      <td>0.989551</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.880435</td>\n",
       "      <td>0.989551</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>19_10602584_C_T</td>\n",
       "      <td>P-0025956-T01-IM619_10602584_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SMARCA4</td>\n",
       "      <td>6597</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>19</td>\n",
       "      <td>11113836</td>\n",
       "      <td>11113836</td>\n",
       "      <td>+</td>\n",
       "      <td>Splice_Site</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1943+1G&gt;A</td>\n",
       "      <td>NaN</td>\n",
       "      <td>p.X648_splice</td>\n",
       "      <td>ENST00000344626</td>\n",
       "      <td>NaN</td>\n",
       "      <td>445.0</td>\n",
       "      <td>371.0</td>\n",
       "      <td>74.0</td>\n",
       "      <td>362</td>\n",
       "      <td>361</td>\n",
       "      <td>1</td>\n",
       "      <td>SMARCA4,splice_donor_variant,,ENST00000358026,NM_001128849.1;SMARCA4,splice_donor_variant,,ENST00000429416,NM_001128844.1;SMARCA4,splice_donor_variant,,ENST00000344626,NM_003072.3;SMARCA4,splice_donor_variant,,ENST00000413806,NM_001128845.1,NM_001128847.1;SMARCA4,splice_donor_variant,,ENST00000450717,NM_001128846.1,NM_001128848.1;SMARCA4,splice_donor_variant,,ENST00000590574,;SMARCA4,splice_donor_variant,,ENST00000589677,;SMARCA4,splice_donor_variant,,ENST00000541122,;SMARCA4,splice_donor_variant,,ENST00000444061,;SMARCA4,splice_donor_variant,,ENST00000591545,;</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000127616</td>\n",
       "      <td>ENST00000344626</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>splice_donor_variant</td>\n",
       "      <td>-/5392</td>\n",
       "      <td>1943/4944</td>\n",
       "      <td>648/1647</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>SMARCA4</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>11100.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS12253.1</td>\n",
       "      <td>ENSP00000343896</td>\n",
       "      <td>P51532</td>\n",
       "      <td>B4DSI8,A7E2E1</td>\n",
       "      <td>UPI000006F973</td>\n",
       "      <td>NM_003072.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>12/34</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GGT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Likely Loss-of-function</td>\n",
       "      <td>Likely Oncogenic</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>24658002;24658001;24658004;25060813;18301784</td>\n",
       "      <td>True</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.183720</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.166292</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.607</td>\n",
       "      <td>0.534</td>\n",
       "      <td>0.686</td>\n",
       "      <td>0.000001</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.959</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.659996</td>\n",
       "      <td>0.904988</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.959</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.659996</td>\n",
       "      <td>0.904988</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>19_11113836_G_A</td>\n",
       "      <td>P-0025956-T01-IM619_11113836_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PTPRT</td>\n",
       "      <td>11122</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>20</td>\n",
       "      <td>40735560</td>\n",
       "      <td>40735560</td>\n",
       "      <td>+</td>\n",
       "      <td>Missense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.3313G&gt;T</td>\n",
       "      <td>p.Ala1105Ser</td>\n",
       "      <td>p.A1105S</td>\n",
       "      <td>ENST00000373198</td>\n",
       "      <td>25/32</td>\n",
       "      <td>471.0</td>\n",
       "      <td>378.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>349</td>\n",
       "      <td>349</td>\n",
       "      <td>0</td>\n",
       "      <td>PTPRT,missense_variant,p.Ala1105Ser,ENST00000373198,NM_133170.3;PTPRT,missense_variant,p.Ala1076Ser,ENST00000373201,;PTPRT,missense_variant,p.Ala1089Ser,ENST00000373193,NM_007050.5;PTPRT,missense_variant,p.Ala1085Ser,ENST00000373190,;PTPRT,missense_variant,p.Ala1096Ser,ENST00000373184,;PTPRT,missense_variant,p.Ala1095Ser,ENST00000356100,;PTPRT,missense_variant,p.Ala1086Ser,ENST00000373187,;</td>\n",
       "      <td>A</td>\n",
       "      <td>ENSG00000196090</td>\n",
       "      <td>ENST00000373198</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>missense_variant,splice_region_variant</td>\n",
       "      <td>3549/12746</td>\n",
       "      <td>3313/4383</td>\n",
       "      <td>1105/1460</td>\n",
       "      <td>A/S</td>\n",
       "      <td>Gct/Tct</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>PTPRT</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>9682.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS68127.1</td>\n",
       "      <td>ENSP00000362294</td>\n",
       "      <td>O14522</td>\n",
       "      <td>NaN</td>\n",
       "      <td>UPI00001AF6FA</td>\n",
       "      <td>NM_133170.3</td>\n",
       "      <td>deleterious(0)</td>\n",
       "      <td>probably_damaging(0.973)</td>\n",
       "      <td>25/32</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gene3D:3.90.190.10,Pfam_domain:PF00102,Prints_domain:PR00700,PROSITE_patterns:PS00383,PROSITE_profiles:PS50055,PROSITE_profiles:PS50056,hmmpanther:PTHR19134,hmmpanther:PTHR19134:SF208,SMART_domains:SM00194,SMART_domains:SM00404,Superfamily_domains:SSF52799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>NaN</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>GCA</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.183720</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.197452</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.721</td>\n",
       "      <td>0.644</td>\n",
       "      <td>0.803</td>\n",
       "      <td>0.000762</td>\n",
       "      <td>0.006532</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.859171</td>\n",
       "      <td>0.984420</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.979</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.859171</td>\n",
       "      <td>0.984420</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>20_40735560_C_A</td>\n",
       "      <td>P-0025956-T01-IM620_40735560_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ERG</td>\n",
       "      <td>2078</td>\n",
       "      <td>MSKCC</td>\n",
       "      <td>GRCh37</td>\n",
       "      <td>21</td>\n",
       "      <td>39755607</td>\n",
       "      <td>39755607</td>\n",
       "      <td>+</td>\n",
       "      <td>Nonsense_Mutation</td>\n",
       "      <td>SNP</td>\n",
       "      <td>G</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>novel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>P-0025956-T01-IM6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>SOMATIC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>MSK-IMPACT</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>c.1158C&gt;G</td>\n",
       "      <td>p.Tyr386Ter</td>\n",
       "      <td>p.Y386*</td>\n",
       "      <td>ENST00000288319</td>\n",
       "      <td>10/10</td>\n",
       "      <td>584.0</td>\n",
       "      <td>490.0</td>\n",
       "      <td>94.0</td>\n",
       "      <td>371</td>\n",
       "      <td>371</td>\n",
       "      <td>0</td>\n",
       "      <td>ERG,stop_gained,p.Tyr369Ter,ENST00000442448,NM_004449.4;ERG,stop_gained,p.Tyr393Ter,ENST00000417133,NM_001136154.1,NM_001243432.1;ERG,stop_gained,p.Tyr370Ter,ENST00000398910,;ERG,stop_gained,p.Tyr386Ter,ENST00000288319,NM_182918.3;ERG,stop_gained,p.Tyr369Ter,ENST00000398911,;ERG,stop_gained,p.Tyr363Ter,ENST00000398907,;ERG,stop_gained,p.Tyr362Ter,ENST00000398905,;ERG,stop_gained,p.Tyr270Ter,ENST00000398897,NM_001243429.1;ERG,stop_gained,p.Tyr393Ter,ENST00000398919,NM_001243428.1;ERG,stop_gained,p.Tyr294Ter,ENST00000453032,NM_001136155.1;</td>\n",
       "      <td>C</td>\n",
       "      <td>ENSG00000157554</td>\n",
       "      <td>ENST00000288319</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>stop_gained</td>\n",
       "      <td>1261/4919</td>\n",
       "      <td>1158/1440</td>\n",
       "      <td>386/479</td>\n",
       "      <td>Y/*</td>\n",
       "      <td>taC/taG</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>-1</td>\n",
       "      <td>ERG</td>\n",
       "      <td>HGNC</td>\n",
       "      <td>3446.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CCDS13658.1</td>\n",
       "      <td>ENSP00000288319</td>\n",
       "      <td>P11308</td>\n",
       "      <td>Q16031,B4DVX5</td>\n",
       "      <td>UPI0000074389</td>\n",
       "      <td>NM_182918.3</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>10/10</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PROSITE_profiles:PS50061,hmmpanther:PTHR11849,hmmpanther:PTHR11849:SF161,Pfam_domain:PF00178,Gene3D:1.10.10.10,SMART_domains:SM00413,Superfamily_domains:SSF46785,Prints_domain:PR00454</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>NaN</td>\n",
       "      <td>indel</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGT</td>\n",
       "      <td>.</td>\n",
       "      <td>.</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.0</td>\n",
       "      <td>0.138437</td>\n",
       "      <td>0.273767</td>\n",
       "      <td>0.160959</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.749</td>\n",
       "      <td>0.668</td>\n",
       "      <td>0.835</td>\n",
       "      <td>0.003877</td>\n",
       "      <td>0.023297</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.871368</td>\n",
       "      <td>0.986954</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.980</td>\n",
       "      <td>1.0</td>\n",
       "      <td>0.871368</td>\n",
       "      <td>0.986954</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default</td>\n",
       "      <td>NaN</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>/juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt</td>\n",
       "      <td>21_39755607_G_C</td>\n",
       "      <td>P-0025956-T01-IM621_39755607_G_C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Hugo_Symbol  Entrez_Gene_Id Center NCBI_Build Chromosome  Start_Position  \\\n",
       "0        KRAS            3845  MSKCC     GRCh37         12        25398285   \n",
       "1       STK11            6794  MSKCC     GRCh37         19         1221314   \n",
       "2       KEAP1            9817  MSKCC     GRCh37         19        10602584   \n",
       "3     SMARCA4            6597  MSKCC     GRCh37         19        11113836   \n",
       "4       PTPRT           11122  MSKCC     GRCh37         20        40735560   \n",
       "5         ERG            2078  MSKCC     GRCh37         21        39755607   \n",
       "\n",
       "   End_Position Strand Variant_Classification Variant_Type Reference_Allele  \\\n",
       "0      25398285      +      Missense_Mutation          SNP                C   \n",
       "1       1221314      +        Frame_Shift_Del          DEL                C   \n",
       "2      10602584      +      Missense_Mutation          SNP                C   \n",
       "3      11113836      +            Splice_Site          SNP                G   \n",
       "4      40735560      +      Missense_Mutation          SNP                C   \n",
       "5      39755607      +      Nonsense_Mutation          SNP                G   \n",
       "\n",
       "  Tumor_Seq_Allele1 Tumor_Seq_Allele2 dbSNP_RS  dbSNP_Val_Status  \\\n",
       "0                 C                 A    novel               NaN   \n",
       "1                 C                 -    novel               NaN   \n",
       "2                 C                 T    novel               NaN   \n",
       "3                 G                 A    novel               NaN   \n",
       "4                 C                 A    novel               NaN   \n",
       "5                 G                 C    novel               NaN   \n",
       "\n",
       "  Tumor_Sample_Barcode  Matched_Norm_Sample_Barcode Match_Norm_Seq_Allele1  \\\n",
       "0    P-0025956-T01-IM6                          NaN                    NaN   \n",
       "1    P-0025956-T01-IM6                          NaN                    NaN   \n",
       "2    P-0025956-T01-IM6                          NaN                    NaN   \n",
       "3    P-0025956-T01-IM6                          NaN                    NaN   \n",
       "4    P-0025956-T01-IM6                          NaN                    NaN   \n",
       "5    P-0025956-T01-IM6                          NaN                    NaN   \n",
       "\n",
       "  Match_Norm_Seq_Allele2  Tumor_Validation_Allele1  Tumor_Validation_Allele2  \\\n",
       "0                    NaN                       NaN                       NaN   \n",
       "1                    NaN                       NaN                       NaN   \n",
       "2                    NaN                       NaN                       NaN   \n",
       "3                    NaN                       NaN                       NaN   \n",
       "4                    NaN                       NaN                       NaN   \n",
       "5                    NaN                       NaN                       NaN   \n",
       "\n",
       "   Match_Norm_Validation_Allele1  Match_Norm_Validation_Allele2  \\\n",
       "0                            NaN                            NaN   \n",
       "1                            NaN                            NaN   \n",
       "2                            NaN                            NaN   \n",
       "3                            NaN                            NaN   \n",
       "4                            NaN                            NaN   \n",
       "5                            NaN                            NaN   \n",
       "\n",
       "   Verification_Status Validation_Status Mutation_Status  Sequencing_Phase  \\\n",
       "0                  NaN           Unknown         SOMATIC               NaN   \n",
       "1                  NaN           Unknown         SOMATIC               NaN   \n",
       "2                  NaN           Unknown         SOMATIC               NaN   \n",
       "3                  NaN           Unknown         SOMATIC               NaN   \n",
       "4                  NaN           Unknown         SOMATIC               NaN   \n",
       "5                  NaN           Unknown         SOMATIC               NaN   \n",
       "\n",
       "   Sequence_Source  Validation_Method       Score  BAM_File  Sequencer  \\\n",
       "0              NaN                NaN  MSK-IMPACT       NaN        NaN   \n",
       "1              NaN                NaN  MSK-IMPACT       NaN        NaN   \n",
       "2              NaN                NaN  MSK-IMPACT       NaN        NaN   \n",
       "3              NaN                NaN  MSK-IMPACT       NaN        NaN   \n",
       "4              NaN                NaN  MSK-IMPACT       NaN        NaN   \n",
       "5              NaN                NaN  MSK-IMPACT       NaN        NaN   \n",
       "\n",
       "   Tumor_Sample_UUID  Matched_Norm_Sample_UUID        HGVSc  \\\n",
       "0                NaN                       NaN      c.34G>T   \n",
       "1                NaN                       NaN    c.837delC   \n",
       "2                NaN                       NaN     c.994G>A   \n",
       "3                NaN                       NaN  c.1943+1G>A   \n",
       "4                NaN                       NaN    c.3313G>T   \n",
       "5                NaN                       NaN    c.1158C>G   \n",
       "\n",
       "               HGVSp    HGVSp_Short    Transcript_ID Exon_Number  t_depth  \\\n",
       "0         p.Gly12Cys         p.G12C  ENST00000256078         2/6    588.0   \n",
       "1  p.Pro281ArgfsTer6    p.P281Rfs*6  ENST00000326873        6/10    615.0   \n",
       "2        p.Gly332Ser        p.G332S  ENST00000171111         3/6    641.0   \n",
       "3                NaN  p.X648_splice  ENST00000344626         NaN    445.0   \n",
       "4       p.Ala1105Ser       p.A1105S  ENST00000373198       25/32    471.0   \n",
       "5        p.Tyr386Ter        p.Y386*  ENST00000288319       10/10    584.0   \n",
       "\n",
       "   t_ref_count  t_alt_count  n_depth n_ref_count n_alt_count  \\\n",
       "0        495.0         93.0      908         908           0   \n",
       "1        492.0        123.0      885         884           1   \n",
       "2        523.0        118.0      471         471           0   \n",
       "3        371.0         74.0      362         361           1   \n",
       "4        378.0         93.0      349         349           0   \n",
       "5        490.0         94.0      371         371           0   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                               all_effects  \\\n",
       "0                                                                                                                                                                                                                                                                                                                                                           KRAS,missense_variant,p.Gly12Cys,ENST00000311936,NM_004985.3;KRAS,missense_variant,p.Gly12Cys,ENST00000556131,;KRAS,missense_variant,p.Gly12Cys,ENST00000256078,NM_033360.2;KRAS,missense_variant,p.Gly12Cys,ENST00000557334,;   \n",
       "1                               STK11,frameshift_variant,p.Pro281ArgfsTer6,ENST00000326873,NM_000455.4;STK11,frameshift_variant,p.Pro37ArgfsTer6,ENST00000586243,;STK11,downstream_gene_variant,,ENST00000585851,;STK11,downstream_gene_variant,,ENST00000585748,;STK11,3_prime_UTR_variant,,ENST00000593219,;STK11,non_coding_transcript_exon_variant,,ENST00000589152,;STK11,non_coding_transcript_exon_variant,,ENST00000591133,;STK11,non_coding_transcript_exon_variant,,ENST00000586358,;STK11,upstream_gene_variant,,ENST00000585465,;,regulatory_region_variant,,ENSR00000377818,;   \n",
       "2                                                            KEAP1,missense_variant,p.Gly332Ser,ENST00000171111,NM_203500.1;KEAP1,missense_variant,p.Gly332Ser,ENST00000393623,NM_012289.3;KEAP1,upstream_gene_variant,,ENST00000592478,;KEAP1,downstream_gene_variant,,ENST00000591419,;KEAP1,downstream_gene_variant,,ENST00000592055,;CTC-429L19.3,upstream_gene_variant,,ENST00000592671,;KEAP1,downstream_gene_variant,,ENST00000588024,;KEAP1,downstream_gene_variant,,ENST00000585845,;KEAP1,upstream_gene_variant,,ENST00000590593,;KEAP1,upstream_gene_variant,,ENST00000590237,;   \n",
       "3  SMARCA4,splice_donor_variant,,ENST00000358026,NM_001128849.1;SMARCA4,splice_donor_variant,,ENST00000429416,NM_001128844.1;SMARCA4,splice_donor_variant,,ENST00000344626,NM_003072.3;SMARCA4,splice_donor_variant,,ENST00000413806,NM_001128845.1,NM_001128847.1;SMARCA4,splice_donor_variant,,ENST00000450717,NM_001128846.1,NM_001128848.1;SMARCA4,splice_donor_variant,,ENST00000590574,;SMARCA4,splice_donor_variant,,ENST00000589677,;SMARCA4,splice_donor_variant,,ENST00000541122,;SMARCA4,splice_donor_variant,,ENST00000444061,;SMARCA4,splice_donor_variant,,ENST00000591545,;   \n",
       "4                                                                                                                                                                                PTPRT,missense_variant,p.Ala1105Ser,ENST00000373198,NM_133170.3;PTPRT,missense_variant,p.Ala1076Ser,ENST00000373201,;PTPRT,missense_variant,p.Ala1089Ser,ENST00000373193,NM_007050.5;PTPRT,missense_variant,p.Ala1085Ser,ENST00000373190,;PTPRT,missense_variant,p.Ala1096Ser,ENST00000373184,;PTPRT,missense_variant,p.Ala1095Ser,ENST00000356100,;PTPRT,missense_variant,p.Ala1086Ser,ENST00000373187,;   \n",
       "5                          ERG,stop_gained,p.Tyr369Ter,ENST00000442448,NM_004449.4;ERG,stop_gained,p.Tyr393Ter,ENST00000417133,NM_001136154.1,NM_001243432.1;ERG,stop_gained,p.Tyr370Ter,ENST00000398910,;ERG,stop_gained,p.Tyr386Ter,ENST00000288319,NM_182918.3;ERG,stop_gained,p.Tyr369Ter,ENST00000398911,;ERG,stop_gained,p.Tyr363Ter,ENST00000398907,;ERG,stop_gained,p.Tyr362Ter,ENST00000398905,;ERG,stop_gained,p.Tyr270Ter,ENST00000398897,NM_001243429.1;ERG,stop_gained,p.Tyr393Ter,ENST00000398919,NM_001243428.1;ERG,stop_gained,p.Tyr294Ter,ENST00000453032,NM_001136155.1;   \n",
       "\n",
       "  Allele             Gene          Feature Feature_type  \\\n",
       "0      A  ENSG00000133703  ENST00000256078   Transcript   \n",
       "1      -  ENSG00000118046  ENST00000326873   Transcript   \n",
       "2      T  ENSG00000079999  ENST00000171111   Transcript   \n",
       "3      A  ENSG00000127616  ENST00000344626   Transcript   \n",
       "4      A  ENSG00000196090  ENST00000373198   Transcript   \n",
       "5      C  ENSG00000157554  ENST00000288319   Transcript   \n",
       "\n",
       "                              Consequence cDNA_position CDS_position  \\\n",
       "0                        missense_variant       98/1119       34/570   \n",
       "1                      frameshift_variant     2010/3328     837/1302   \n",
       "2                        missense_variant     1542/2955     994/1875   \n",
       "3                    splice_donor_variant        -/5392    1943/4944   \n",
       "4  missense_variant,splice_region_variant    3549/12746    3313/4383   \n",
       "5                             stop_gained     1261/4919    1158/1440   \n",
       "\n",
       "  Protein_position Amino_acids   Codons  Existing_variation  ALLELE_NUM  \\\n",
       "0           12/189         G/C  Ggt/Tgt                 NaN           1   \n",
       "1          279/433         G/X   ggC/gg                 NaN           1   \n",
       "2          332/624         G/S  Ggc/Agc                 NaN           1   \n",
       "3         648/1647         NaN      NaN                 NaN           1   \n",
       "4        1105/1460         A/S  Gct/Tct                 NaN           1   \n",
       "5          386/479         Y/*  taC/taG                 NaN           1   \n",
       "\n",
       "   DISTANCE  STRAND_VEP   SYMBOL SYMBOL_SOURCE  HGNC_ID         BIOTYPE  \\\n",
       "0       NaN          -1     KRAS          HGNC   6407.0  protein_coding   \n",
       "1       NaN           1    STK11          HGNC  11389.0  protein_coding   \n",
       "2       NaN          -1    KEAP1          HGNC  23177.0  protein_coding   \n",
       "3       NaN           1  SMARCA4          HGNC  11100.0  protein_coding   \n",
       "4       NaN          -1    PTPRT          HGNC   9682.0  protein_coding   \n",
       "5       NaN          -1      ERG          HGNC   3446.0  protein_coding   \n",
       "\n",
       "  CANONICAL         CCDS             ENSP SWISSPROT  \\\n",
       "0       YES   CCDS8703.1  ENSP00000256078    P01116   \n",
       "1       YES  CCDS45896.1  ENSP00000324856    Q15831   \n",
       "2       YES  CCDS12239.1  ENSP00000171111    Q14145   \n",
       "3       NaN  CCDS12253.1  ENSP00000343896    P51532   \n",
       "4       NaN  CCDS68127.1  ENSP00000362294    O14522   \n",
       "5       NaN  CCDS13658.1  ENSP00000288319    P11308   \n",
       "\n",
       "                               TREMBL        UNIPARC       RefSeq  \\\n",
       "0  Q9UM97,Q71SP6,P78460,L7RSL8,I1SRC5  UPI0000133132  NM_033360.2   \n",
       "1                              Q9NS52  UPI0000136105  NM_000455.4   \n",
       "2                K7ESE0,K7EJD8,K7EJ49  UPI000007139C  NM_203500.1   \n",
       "3                       B4DSI8,A7E2E1  UPI000006F973  NM_003072.3   \n",
       "4                                 NaN  UPI00001AF6FA  NM_133170.3   \n",
       "5                       Q16031,B4DVX5  UPI0000074389  NM_182918.3   \n",
       "\n",
       "                SIFT                  PolyPhen   EXON INTRON  \\\n",
       "0  deleterious(0.04)  probably_damaging(0.993)    2/6    NaN   \n",
       "1                NaN                       NaN   6/10    NaN   \n",
       "2     deleterious(0)  probably_damaging(0.998)    3/6    NaN   \n",
       "3                NaN                       NaN    NaN  12/34   \n",
       "4     deleterious(0)  probably_damaging(0.973)  25/32    NaN   \n",
       "5                NaN                       NaN  10/10    NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                        DOMAINS  \\\n",
       "0  Gene3D:3.40.50.300,Pfam_domain:PF00071,Prints_domain:PR00449,PROSITE_profiles:PS51421,hmmpanther:PTHR24070,hmmpanther:PTHR24070:SF186,Low_complexity_(Seg):seg,SMART_domains:SM00173,SMART_domains:SM00174,SMART_domains:SM00175,SMART_domains:SM00176,Superfamily_domains:SSF52540,TIGRFAM_domain:TIGR00231   \n",
       "1                                                                                                                                              Gene3D:1.10.510.10,Pfam_domain:PF00069,PROSITE_profiles:PS50011,hmmpanther:PTHR24347,hmmpanther:PTHR24347:SF1,SMART_domains:SM00220,Superfamily_domains:SSF56112   \n",
       "2                                                                                                                                                 hmmpanther:PTHR24412:SF162,hmmpanther:PTHR24412,Pfam_domain:PF01344,Gene3D:1k3iA02,PIRSF_domain:PIRSF037037,SMART_domains:SM00612,Superfamily_domains:0052715   \n",
       "3                                                                                                                                                                                                                                                                                                           NaN   \n",
       "4                                              Gene3D:3.90.190.10,Pfam_domain:PF00102,Prints_domain:PR00700,PROSITE_patterns:PS00383,PROSITE_profiles:PS50055,PROSITE_profiles:PS50056,hmmpanther:PTHR19134,hmmpanther:PTHR19134:SF208,SMART_domains:SM00194,SMART_domains:SM00404,Superfamily_domains:SSF52799   \n",
       "5                                                                                                                       PROSITE_profiles:PS50061,hmmpanther:PTHR11849,hmmpanther:PTHR11849:SF161,Pfam_domain:PF00178,Gene3D:1.10.10.10,SMART_domains:SM00413,Superfamily_domains:SSF46785,Prints_domain:PR00454   \n",
       "\n",
       "   GMAF  AFR_MAF  AMR_MAF  ASN_MAF  EAS_MAF  EUR_MAF  SAS_MAF  AA_MAF  EA_MAF  \\\n",
       "0   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN     NaN   \n",
       "1   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN     NaN   \n",
       "2   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN     NaN   \n",
       "3   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN     NaN   \n",
       "4   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN     NaN   \n",
       "5   NaN      NaN      NaN      NaN      NaN      NaN      NaN     NaN     NaN   \n",
       "\n",
       "   CLIN_SIG  SOMATIC  PUBMED  MOTIF_NAME  MOTIF_POS  HIGH_INF_POS  \\\n",
       "0       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "1       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "2       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "3       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "4       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "5       NaN      NaN     NaN         NaN        NaN           NaN   \n",
       "\n",
       "   MOTIF_SCORE_CHANGE    IMPACT  PICK        VARIANT_CLASS  TSL  HGVS_OFFSET  \\\n",
       "0                 NaN  MODERATE   1.0                indel  NaN          NaN   \n",
       "1                 NaN      HIGH   1.0  sequence_alteration  NaN          NaN   \n",
       "2                 NaN  MODERATE   1.0                indel  NaN          NaN   \n",
       "3                 NaN      HIGH   NaN                indel  NaN          NaN   \n",
       "4                 NaN  MODERATE   NaN                indel  NaN          NaN   \n",
       "5                 NaN      HIGH   NaN                indel  NaN          NaN   \n",
       "\n",
       "   PHENO  MINIMISED  ExAC_AF  ExAC_AF_AFR  ExAC_AF_AMR  ExAC_AF_EAS  \\\n",
       "0    NaN        NaN      NaN          NaN          NaN          NaN   \n",
       "1    NaN        NaN      NaN          NaN          NaN          NaN   \n",
       "2    NaN        NaN      NaN          NaN          NaN          NaN   \n",
       "3    NaN        NaN      NaN          NaN          NaN          NaN   \n",
       "4    NaN        NaN      NaN          NaN          NaN          NaN   \n",
       "5    NaN        NaN      NaN          NaN          NaN          NaN   \n",
       "\n",
       "   ExAC_AF_FIN  ExAC_AF_NFE  ExAC_AF_OTH  ExAC_AF_SAS  GENE_PHENO FILTER  \\\n",
       "0          NaN          NaN          NaN          NaN         1.0    NaN   \n",
       "1          NaN          NaN          NaN          NaN         1.0    NaN   \n",
       "2          NaN          NaN          NaN          NaN         NaN    NaN   \n",
       "3          NaN          NaN          NaN          NaN         1.0    NaN   \n",
       "4          NaN          NaN          NaN          NaN         1.0    NaN   \n",
       "5          NaN          NaN          NaN          NaN         1.0    NaN   \n",
       "\n",
       "  flanking_bps variant_id variant_qual  ExAC_AF_Adj ExAC_AC_AN_Adj ExAC_AC_AN  \\\n",
       "0          CCA          .            .          NaN            NaN        NaN   \n",
       "1         GGCC          .            .          NaN            NaN        NaN   \n",
       "2          CCC          .            .          NaN            NaN        NaN   \n",
       "3          GGT          .            .          NaN            NaN        NaN   \n",
       "4          GCA          .            .          NaN            NaN        NaN   \n",
       "5          CGT          .            .          NaN            NaN        NaN   \n",
       "\n",
       "  ExAC_AC_AN_AFR ExAC_AC_AN_AMR ExAC_AC_AN_EAS ExAC_AC_AN_FIN ExAC_AC_AN_NFE  \\\n",
       "0            NaN            NaN            NaN            NaN            NaN   \n",
       "1            NaN            NaN            NaN            NaN            NaN   \n",
       "2            NaN            NaN            NaN            NaN            NaN   \n",
       "3            NaN            NaN            NaN            NaN            NaN   \n",
       "4            NaN            NaN            NaN            NaN            NaN   \n",
       "5            NaN            NaN            NaN            NaN            NaN   \n",
       "\n",
       "  ExAC_AC_AN_OTH ExAC_AC_AN_SAS ExAC_FILTER  Caller is-a-hotspot  \\\n",
       "0            NaN            NaN         NaN     NaN            Y   \n",
       "1            NaN            NaN         NaN     NaN          NaN   \n",
       "2            NaN            NaN         NaN     NaN          NaN   \n",
       "3            NaN            NaN         NaN     NaN          NaN   \n",
       "4            NaN            NaN         NaN     NaN          NaN   \n",
       "5            NaN            NaN        PASS     NaN          NaN   \n",
       "\n",
       "  is-a-3d-hotspot          mutation_effect         oncogenic LEVEL_1 LEVEL_2A  \\\n",
       "0               Y         Gain-of-function         Oncogenic     NaN      NaN   \n",
       "1             NaN  Likely Loss-of-function  Likely Oncogenic     NaN      NaN   \n",
       "2             NaN                      NaN               NaN     NaN      NaN   \n",
       "3             NaN  Likely Loss-of-function  Likely Oncogenic     NaN      NaN   \n",
       "4             NaN                      NaN               NaN     NaN      NaN   \n",
       "5             NaN                      NaN               NaN     NaN      NaN   \n",
       "\n",
       "  LEVEL_2B LEVEL_3A LEVEL_3B                             LEVEL_4 LEVEL_R1  \\\n",
       "0      NaN      NaN      NaN  Binimetinib,Cobimetinib,Trametinib      NaN   \n",
       "1      NaN      NaN      NaN                                 NaN      NaN   \n",
       "2      NaN      NaN      NaN                                 NaN      NaN   \n",
       "3      NaN      NaN      NaN                                 NaN      NaN   \n",
       "4      NaN      NaN      NaN                                 NaN      NaN   \n",
       "5      NaN      NaN      NaN                                 NaN      NaN   \n",
       "\n",
       "  LEVEL_R2  LEVEL_R3 Highest_level  \\\n",
       "0      NaN       NaN       LEVEL_4   \n",
       "1      NaN       NaN           NaN   \n",
       "2      NaN       NaN           NaN   \n",
       "3      NaN       NaN           NaN   \n",
       "4      NaN       NaN           NaN   \n",
       "5      NaN       NaN           NaN   \n",
       "\n",
       "                                                                                                                                                                                                                                                                                                                                                                                     citations  \\\n",
       "0  16051643;25705018;26841430;24256730;28783719;29247021;Bhagwat et al. Abstract# 4973, AACR 2017(http://cancerres.aacrjournals.org/content/77/13_Supplement/4973);Robarge et al. Abstract# DDT02-03, AACR 2014(http://cancerres.aacrjournals.org/content/74/19_Supplement/DDT02-03);Burrows et al. Abstract# 5168, AACR 2017(http://cancerres.aacrjournals.org/content/77/13_Supplement/5168)   \n",
       "1                                                                                                                                                                                                                                                                                                                                                 24652667;19892943;25079552;21516316;19340305   \n",
       "2                                                                                                                                                                                                                                                                                                                                                                                          NaN   \n",
       "3                                                                                                                                                                                                                                                                                                                                                 24658002;24658001;24658004;25060813;18301784   \n",
       "4                                                                                                                                                                                                                                                                                                                                                                                          NaN   \n",
       "5                                                                                                                                                                                                                                                                                                                                                                                          NaN   \n",
       "\n",
       "   driver  tcn  lcn        cf    purity  t_var_freq  expected_alt_copies  \\\n",
       "0    True  4.0  2.0  0.138437  0.273767    0.158163                  1.0   \n",
       "1    True  2.0  0.0  0.183720  0.273767    0.200000                  1.0   \n",
       "2   False  2.0  0.0  0.183720  0.273767    0.184087                  1.0   \n",
       "3    True  2.0  0.0  0.183720  0.273767    0.166292                  1.0   \n",
       "4   False  2.0  0.0  0.183720  0.273767    0.197452                  1.0   \n",
       "5   False  4.0  2.0  0.138437  0.273767    0.160959                  1.0   \n",
       "\n",
       "   ccf_Mcopies  ccf_Mcopies_lower  ccf_Mcopies_upper  ccf_Mcopies_prob95  \\\n",
       "0        0.736              0.655              0.821            0.002236   \n",
       "1        0.731              0.662              0.802            0.000266   \n",
       "2        0.672              0.608              0.740            0.000003   \n",
       "3        0.607              0.534              0.686            0.000001   \n",
       "4        0.721              0.644              0.803            0.000762   \n",
       "5        0.749              0.668              0.835            0.003877   \n",
       "\n",
       "   ccf_Mcopies_prob90  ccf_1copy  ccf_1copy_lower  ccf_1copy_upper  \\\n",
       "0            0.014834        1.0            0.979              1.0   \n",
       "1            0.003571        1.0            0.984              1.0   \n",
       "2            0.000093        1.0            0.980              1.0   \n",
       "3            0.000027        1.0            0.959              1.0   \n",
       "4            0.006532        1.0            0.979              1.0   \n",
       "5            0.023297        1.0            0.980              1.0   \n",
       "\n",
       "   ccf_1copy_prob95  ccf_1copy_prob90  ccf_expected_copies  \\\n",
       "0          0.860769          0.984692                  1.0   \n",
       "1          0.926527          0.996079                  1.0   \n",
       "2          0.880435          0.989551                  1.0   \n",
       "3          0.659996          0.904988                  1.0   \n",
       "4          0.859171          0.984420                  1.0   \n",
       "5          0.871368          0.986954                  1.0   \n",
       "\n",
       "   ccf_expected_copies_lower  ccf_expected_copies_upper  \\\n",
       "0                      0.979                        1.0   \n",
       "1                      0.984                        1.0   \n",
       "2                      0.980                        1.0   \n",
       "3                      0.959                        1.0   \n",
       "4                      0.979                        1.0   \n",
       "5                      0.980                        1.0   \n",
       "\n",
       "   ccf_expected_copies_prob95  ccf_expected_copies_prob90  \\\n",
       "0                    0.860769                    0.984692   \n",
       "1                    0.926527                    0.996079   \n",
       "2                    0.880435                    0.989551   \n",
       "3                    0.659996                    0.904988   \n",
       "4                    0.859171                    0.984420   \n",
       "5                    0.871368                    0.986954   \n",
       "\n",
       "                                                                                               facets_fit  \\\n",
       "0  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "1  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "2  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "3  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "4  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "5  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default   \n",
       "\n",
       "   reviewer_set_purity  use_only_purity_run  use_edited_cncf  \\\n",
       "0                  NaN                False            False   \n",
       "1                  NaN                False            False   \n",
       "2                  NaN                False            False   \n",
       "3                  NaN                False            False   \n",
       "4                  NaN                False            False   \n",
       "5                  NaN                False            False   \n",
       "\n",
       "                                                                                                                                               cncf_file_used  \\\n",
       "0  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "1  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "2  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "3  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "4  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "5  /juno/work/ccs/shared/resources/impact/facets/all/P-00259/P-0025956-T01-IM6_P-0025956-N01-IM6//default/P-0025956-T01-IM6_P-0025956-N01-IM6_hisens.cncf.txt   \n",
       "\n",
       "           mut_key                    sample_mut_key  \n",
       "0  12_25398285_C_A  P-0025956-T01-IM612_25398285_C_A  \n",
       "1   19_1221314_C_-   P-0025956-T01-IM619_1221314_C_-  \n",
       "2  19_10602584_C_T  P-0025956-T01-IM619_10602584_C_T  \n",
       "3  19_11113836_G_A  P-0025956-T01-IM619_11113836_G_A  \n",
       "4  20_40735560_C_A  P-0025956-T01-IM620_40735560_C_A  \n",
       "5  21_39755607_G_C  P-0025956-T01-IM621_39755607_G_C  "
      ]
     },
     "execution_count": 134,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "maf_cohort_wgd[maf_cohort_wgd['Tumor_Sample_Barcode'] == 'P-0025956-T01-IM6']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  },
  "toc": {
   "base_numbering": 1,
   "nav_menu": {},
   "number_sections": true,
   "sideBar": true,
   "skip_h1_title": false,
   "title_cell": "Table of Contents",
   "title_sidebar": "Contents",
   "toc_cell": false,
   "toc_position": {
    "height": "calc(100% - 180px)",
    "left": "10px",
    "top": "150px",
    "width": "384px"
   },
   "toc_section_display": true,
   "toc_window_display": true
  },
  "varInspector": {
   "cols": {
    "lenName": 16,
    "lenType": 16,
    "lenVar": 40
   },
   "kernels_config": {
    "python": {
     "delete_cmd_postfix": "",
     "delete_cmd_prefix": "del ",
     "library": "var_list.py",
     "varRefreshCmd": "print(var_dic_list())"
    },
    "r": {
     "delete_cmd_postfix": ") ",
     "delete_cmd_prefix": "rm(",
     "library": "var_list.r",
     "varRefreshCmd": "cat(var_dic_list()) "
    }
   },
   "types_to_exclude": [
    "module",
    "function",
    "builtin_function_or_method",
    "instance",
    "_Feature"
   ],
   "window_display": false
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
